{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# This is used for running data vs human_msigdb_c1 annotations located in: \n",
    "### /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import pickle\n",
    "import seaborn as sns\n",
    "import scipy.stats\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### get index of file names to loop through "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "cd fc_values\n",
    "ls FC* > filelist.txt\n",
    "cd .."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### get ensembl genes represented in iPAGE database"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash \n",
    "\n",
    "cp /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c1/human_ensembl_msigdb_c1_index.txt human_ensembl_index.txt\n",
    "awk '{print $1}' human_ensembl_index.txt > ensembl_index.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### make sure that genes in file match the index "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "###################!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n",
    "###################REMEMBER TO SPECIFY INDEX USED IN THE IPAGE COMMAND!!!!!!!!\n",
    "###################!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n",
    "\n",
    "\n",
    "#####################\n",
    "#######read in data from logfc calculations\n",
    "#####################\n",
    "####define index file to loop through\n",
    "index_file = \"fc_values/filelist.txt\"\n",
    "####define matching ensg index\n",
    "GO_index_filename = \"ensembl_index.txt\"\n",
    "genes_with_GOs = []\n",
    "with open(GO_index_filename,'r') as rf:\n",
    "    for line in rf:\n",
    "        gene_name = line.split('\\n')[0]\n",
    "        genes_with_GOs.append(gene_name)\n",
    "genes_with_GOs_set = set(genes_with_GOs)\n",
    "####open file and loop through each line\n",
    "###\n",
    "iPAGE_sh = \"iPAGE_JY.sh\"\n",
    "cwd = os.getcwd()+\"/\"\n",
    "os.system(\"rm \"+iPAGE_sh)\n",
    "###\n",
    "with open(index_file,'r') as rf:\n",
    "    for line in rf:\n",
    "        #####do this to each file\n",
    "        ###read in file into pandas df\n",
    "        FC_file = \"fc_values/\"+line.split('\\n')[0]\n",
    "        FC_df = pd.read_csv(FC_file,sep=',',index_col=0)\n",
    "        FC_df.columns = ['log2FoldChange']\n",
    "        FC_df.index.names = ['Gene']\n",
    "        ###filter\n",
    "        FC_df = FC_df.loc[[x for x in FC_df.index if x in genes_with_GOs_set]]\n",
    "        ###save\n",
    "        save_file = \"fc_values/filt_\"+line.split('\\n')[0]\n",
    "        FC_df.to_csv(save_file,sep='\\t',header=True)\n",
    "        ###write final iPAGE command to an ipage .sh file\n",
    "        iPAGE_cmd = \"perl /nvme/bins/iPAGEv1.0/page.pl --expfile \"+cwd+save_file+\" --species=human_ensembl_msigdb_c1 --exptype=continuous --ebins=9 \\n\"\n",
    "        with open(iPAGE_sh, \"a\") as myfile:\n",
    "            myfile.write(iPAGE_cmd)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## then run the command from this folder:\n",
    "\n",
    "nohup sh run.sh iPAGE_JY.sh &"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM1_1.csv --species=human_ensembl_msigdb_c1 --exptype=continuous --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM1_2.csv --species=human_ensembl_msigdb_c1 --exptype=continuous --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM1_3.csv --species=human_ensembl_msigdb_c1 --exptype=continuous --ebins=9\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM1_1.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c1.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c1/human_ensembl_msigdb_c1.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM1_1.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 326\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t326\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 326\n",
      "cat 46, mi = 0.00402238, chr13q14\n",
      "cat 44, mi = 0.00341812, chr13q12\n",
      "cat 95, mi = 0.00297485, chr17q25\n",
      "cat 50, mi = 0.00260504, chr13q32\n",
      "cat 52, mi = 0.00260327, chr13q34\n",
      "cat 309, mi = 0.00245843, chrxp11\n",
      "cat 274, mi = 0.00194427, chr7q31\n",
      "cat 48, mi = 0.00169527, chr13q22\n",
      "cat 315, mi = 0.00160654, chrxq13\n",
      "cat 271, mi = 0.00158622, chr7q11\n",
      "cat 143, mi = 0.00152913, chr21q22\n",
      "cat 45, mi = 0.00150993, chr13q13\n",
      "cat 149, mi = 0.00149085, chr2p11\n",
      "cat 211, mi = 0.00137748, chr4q22\n",
      "cat 79, mi = 0.00137627, chr16q12\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 46, mi = 0.004, pvalue = 0.000, chr13q14\n",
      "Passed: cat 44, mi = 0.003, pvalue = 0.000, chr13q12\n",
      "Passed: cat 95, mi = 0.003, pvalue = 0.000, chr17q25\n",
      "Passed: cat 50, mi = 0.003, pvalue = 0.000, chr13q32\n",
      "Passed: cat 52, mi = 0.003, pvalue = 0.000, chr13q34\n",
      "Passed: cat 309, mi = 0.002, pvalue = 0.000, chrxp11\n",
      "Passed: cat 274, mi = 0.002, pvalue = 0.000, chr7q31\n",
      "Passed: cat 48, mi = 0.002, pvalue = 0.000, chr13q22\n",
      "Passed: cat 315, mi = 0.002, pvalue = 0.002, chrxq13\n",
      "Passed: cat 271, mi = 0.002, pvalue = 0.001, chr7q11\n",
      "Passed: cat 143, mi = 0.002, pvalue = 0.002, chr21q22\n",
      "Passed: cat 45, mi = 0.002, pvalue = 0.001, chr13q13\n",
      "Passed: cat 149, mi = 0.001, pvalue = 0.002, chr2p11\n",
      "Passed: cat 211, mi = 0.001, pvalue = 0.002, chr4q22\n",
      "Didn't pass (cutoff=0.0050): cat 79, mi = 0.001, pvalue = 0.006, chr16q12\n",
      "Didn't pass (cutoff=0.0050): cat 204, mi = 0.001, pvalue = 0.010, chr4p15\n",
      "Didn't pass (cutoff=0.0050): cat 311, mi = 0.001, pvalue = 0.008, chrxp22\n",
      "Passed: cat 51, mi = 0.001, pvalue = 0.002, chr13q33\n",
      "Didn't pass (cutoff=0.0050): cat 230, mi = 0.001, pvalue = 0.014, chr5q13\n",
      "Didn't pass (cutoff=0.0050): cat 243, mi = 0.001, pvalue = 0.014, chr6p12\n",
      "Didn't pass (cutoff=0.0050): cat 284, mi = 0.001, pvalue = 0.014, chr8p23\n",
      "Didn't pass (cutoff=0.0050): cat 129, mi = 0.001, pvalue = 0.012, chr1q32\n",
      "Didn't pass (cutoff=0.0050): cat 152, mi = 0.001, pvalue = 0.020, chr2p14\n",
      "Didn't pass (cutoff=0.0050): cat 2, mi = 0.001, pvalue = 0.018, chr10p13\n",
      "Didn't pass (cutoff=0.0050): cat 259, mi = 0.001, pvalue = 0.022, chr6q25\n",
      "Didn't pass (cutoff=0.0050): cat 267, mi = 0.001, pvalue = 0.019, chr7p15\n",
      "Didn't pass (cutoff=0.0050): cat 161, mi = 0.001, pvalue = 0.027, chr2q11\n",
      "Didn't pass (cutoff=0.0050): cat 134, mi = 0.001, pvalue = 0.028, chr20p11\n",
      "Didn't pass (cutoff=0.0050): cat 3, mi = 0.001, pvalue = 0.017, chr10p14\n",
      "Didn't pass (cutoff=0.0050): cat 90, mi = 0.001, pvalue = 0.029, chr17q12\n",
      "Didn't pass (cutoff=0.0050): cat 291, mi = 0.001, pvalue = 0.020, chr8q23\n",
      "Didn't pass (cutoff=0.0050): cat 273, mi = 0.001, pvalue = 0.023, chr7q22\n",
      "Didn't pass (cutoff=0.0050): cat 289, mi = 0.001, pvalue = 0.034, chr8q21\n",
      "Didn't pass (cutoff=0.0050): cat 210, mi = 0.001, pvalue = 0.031, chr4q21\n",
      "Didn't pass (cutoff=0.0050): cat 222, mi = 0.001, pvalue = 0.035, chr4q35\n",
      "Didn't pass (cutoff=0.0050): cat 290, mi = 0.001, pvalue = 0.032, chr8q22\n",
      "Didn't pass (cutoff=0.0050): cat 258, mi = 0.001, pvalue = 0.044, chr6q24\n",
      "Didn't pass (cutoff=0.0050): cat 205, mi = 0.001, pvalue = 0.030, chr4p16\n",
      "Total number of categories evaluated = 29.\n",
      "Number of categories that passed the tests = 15\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM1_1.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM1_1.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 3, 7, 0.00883931956597828\n",
      "best merge is 0, 1, 0.012779702669598\n",
      "best merge is 3, 4, 0.0342687999877158\n",
      "best merge is 10, 11, 0.0390861630969336\n",
      "best merge is 3, 14, 0.0664207067397538\n",
      "best merge is 0, 10, 0.0842547852185983\n",
      "best merge is 0, 3, 0.159988419002771\n",
      "best merge is 2, 5, 0.189414648348862\n",
      "best merge is 0, 6, 0.226222392621106\n",
      "best merge is 2, 8, 0.365446820165274\n",
      "best merge is 0, 2, 0.504578635198886\n",
      "order classes\n",
      "9 - 0.207206497589085\n",
      "3 - 1.02256476519934\n",
      "0 1 10 11 3 7 4 14 6\n",
      "2 5 8\n",
      "9\n",
      "12\n",
      "13\n",
      "Done.Start drawing\n",
      "23.77\t0.00\n",
      "chr2p11\n",
      "chr4q22\n",
      "chr7q11\n",
      "chr17q25\n",
      "chrxp11\n",
      "chrxq13\n",
      "chr13q14\n",
      "chr13q12\n",
      "chr21q22\n",
      "chr13q13\n",
      "chr13q32\n",
      "chr13q22\n",
      "chr13q34\n",
      "chr13q33\n",
      "chr7q31\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM1_1.csv_PAGE/filt_FC_LVM1_1.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM1_1.csv_PAGE/filt_FC_LVM1_1.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM1_1.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM1_1.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 3, 7, 0.00883931956597828\n",
      "best merge is 0, 1, 0.012779702669598\n",
      "best merge is 3, 4, 0.0342687999877158\n",
      "best merge is 10, 11, 0.0390861630969336\n",
      "best merge is 3, 14, 0.0664207067397538\n",
      "best merge is 0, 10, 0.0842547852185983\n",
      "best merge is 0, 3, 0.159988419002771\n",
      "best merge is 2, 5, 0.189414648348862\n",
      "best merge is 0, 6, 0.226222392621106\n",
      "best merge is 2, 8, 0.365446820165274\n",
      "best merge is 0, 2, 0.504578635198886\n",
      "order classes\n",
      "9 - 0.207206497589085\n",
      "3 - 1.02256476519934\n",
      "0 1 10 11 3 7 4 14 6\n",
      "2 5 8\n",
      "9\n",
      "12\n",
      "13\n",
      "Done.-19.2932267564156\t42\t0\n",
      "-18.4450988641144\t42\t1.4135464871687\n",
      "-18.2153975599495\t42\t1.69625578460244\n",
      "-18.056373580143\t42\t1.87294909549853\n",
      "-17.8973496003366\t42\t2.0143037442154\n",
      "-17.7206562894405\t42\t2.19099705511149\n",
      "-17.51745898191\t42\t2.36769036600757\n",
      "-17.2082456878418\t42\t2.59739167017249\n",
      "2.49305847707194\t42\t2.98611695414388\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM1_1.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM1_4.csv --species=human_ensembl_msigdb_c1 --exptype=continuous --ebins=9\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM1_2.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c1.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c1/human_ensembl_msigdb_c1.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM1_2.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 326\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t326\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 326\n",
      "cat 143, mi = 0.00307619, chr21q22\n",
      "cat 44, mi = 0.00231676, chr13q12\n",
      "cat 50, mi = 0.00228918, chr13q32\n",
      "cat 46, mi = 0.00202174, chr13q14\n",
      "cat 52, mi = 0.00180099, chr13q34\n",
      "cat 309, mi = 0.00170071, chrxp11\n",
      "cat 311, mi = 0.00159708, chrxp22\n",
      "cat 231, mi = 0.00157205, chr5q14\n",
      "cat 152, mi = 0.00151925, chr2p14\n",
      "cat 287, mi = 0.00145037, chr8q12\n",
      "cat 274, mi = 0.00141010, chr7q31\n",
      "cat 290, mi = 0.00140137, chr8q22\n",
      "cat 245, mi = 0.00138521, chr6p22\n",
      "cat 167, mi = 0.00133322, chr2q23\n",
      "cat 256, mi = 0.00129875, chr6q22\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 143, mi = 0.003, pvalue = 0.000, chr21q22\n",
      "Passed: cat 44, mi = 0.002, pvalue = 0.000, chr13q12\n",
      "Passed: cat 50, mi = 0.002, pvalue = 0.000, chr13q32\n",
      "Passed: cat 46, mi = 0.002, pvalue = 0.000, chr13q14\n",
      "Passed: cat 52, mi = 0.002, pvalue = 0.000, chr13q34\n",
      "Passed: cat 309, mi = 0.002, pvalue = 0.001, chrxp11\n",
      "Passed: cat 311, mi = 0.002, pvalue = 0.001, chrxp22\n",
      "Passed: cat 231, mi = 0.002, pvalue = 0.002, chr5q14\n",
      "Passed: cat 152, mi = 0.002, pvalue = 0.001, chr2p14\n",
      "Passed: cat 287, mi = 0.001, pvalue = 0.003, chr8q12\n",
      "Passed: cat 274, mi = 0.001, pvalue = 0.004, chr7q31\n",
      "Didn't pass (cutoff=0.0050): cat 290, mi = 0.001, pvalue = 0.006, chr8q22\n",
      "Passed: cat 245, mi = 0.001, pvalue = 0.004, chr6p22\n",
      "Passed: cat 167, mi = 0.001, pvalue = 0.004, chr2q23\n",
      "Didn't pass (cutoff=0.0050): cat 256, mi = 0.001, pvalue = 0.012, chr6q22\n",
      "Didn't pass (cutoff=0.0050): cat 303, mi = 0.001, pvalue = 0.013, chr9q21\n",
      "Passed: cat 48, mi = 0.001, pvalue = 0.003, chr13q22\n",
      "Didn't pass (cutoff=0.0050): cat 205, mi = 0.001, pvalue = 0.009, chr4p16\n",
      "Didn't pass (cutoff=0.0050): cat 228, mi = 0.001, pvalue = 0.014, chr5q11\n",
      "Passed: cat 142, mi = 0.001, pvalue = 0.004, chr21q21\n",
      "Didn't pass (cutoff=0.0050): cat 208, mi = 0.001, pvalue = 0.014, chr4q12\n",
      "Didn't pass (cutoff=0.0050): cat 51, mi = 0.001, pvalue = 0.005, chr13q33\n",
      "Didn't pass (cutoff=0.0050): cat 156, mi = 0.001, pvalue = 0.023, chr2p22\n",
      "Didn't pass (cutoff=0.0050): cat 269, mi = 0.001, pvalue = 0.022, chr7p22\n",
      "Didn't pass (cutoff=0.0050): cat 97, mi = 0.001, pvalue = 0.027, chr18p11\n",
      "Didn't pass (cutoff=0.0050): cat 129, mi = 0.001, pvalue = 0.019, chr1q32\n",
      "Didn't pass (cutoff=0.0050): cat 163, mi = 0.001, pvalue = 0.026, chr2q13\n",
      "Didn't pass (cutoff=0.0050): cat 219, mi = 0.001, pvalue = 0.022, chr4q32\n",
      "Didn't pass (cutoff=0.0050): cat 305, mi = 0.001, pvalue = 0.031, chr9q31\n",
      "Didn't pass (cutoff=0.0050): cat 226, mi = 0.001, pvalue = 0.037, chr5p15\n",
      "Didn't pass (cutoff=0.0050): cat 119, mi = 0.001, pvalue = 0.029, chr1p35\n",
      "Didn't pass (cutoff=0.0050): cat 283, mi = 0.001, pvalue = 0.047, chr8p22\n",
      "Didn't pass (cutoff=0.0050): cat 66, mi = 0.001, pvalue = 0.037, chr15q14\n",
      "Didn't pass (cutoff=0.0050): cat 292, mi = 0.001, pvalue = 0.027, chr8q24\n",
      "Didn't pass (cutoff=0.0050): cat 23, mi = 0.001, pvalue = 0.049, chr11q14\n",
      "Didn't pass (cutoff=0.0050): cat 14, mi = 0.001, pvalue = 0.046, chr11p11\n",
      "Didn't pass (cutoff=0.0050): cat 199, mi = 0.001, pvalue = 0.050, chr3q29\n",
      "Didn't pass (cutoff=0.0050): cat 306, mi = 0.001, pvalue = 0.063, chr9q32\n",
      "Didn't pass (cutoff=0.0050): cat 277, mi = 0.001, pvalue = 0.058, chr7q34\n",
      "Didn't pass (cutoff=0.0050): cat 92, mi = 0.001, pvalue = 0.068, chr17q22\n",
      "Total number of categories evaluated = 30.\n",
      "Number of categories that passed the tests = 15\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM1_2.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM1_2.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 6, 14, 0.0629899810942742\n",
      "best merge is 0, 3, 0.0799558120640301\n",
      "best merge is 2, 13, 0.110596085660057\n",
      "best merge is 0, 1, 0.139228260797042\n",
      "best merge is 0, 4, 0.156726734565951\n",
      "best merge is 0, 6, 0.227069410273795\n",
      "best merge is 5, 10, 0.258363171315387\n",
      "best merge is 0, 5, 0.331839123770227\n",
      "best merge is 0, 7, 0.493183230210246\n",
      "best merge is 0, 9, 0.524900553332698\n",
      "best merge is 8, 11, 0.527540834141488\n",
      "order classes\n",
      "10 - 0.468544238366249\n",
      "2 - 0.414735321225214\n",
      "2 13\n",
      "0 3 1 4 6 14 5 10 7 9\n",
      "8\n",
      "11\n",
      "12\n",
      "Done.Start drawing\n",
      "62.96\t0.00\n",
      "chr6p22\n",
      "chr2p14\n",
      "chr5q14\n",
      "chr2q23\n",
      "chr8q12\n",
      "chr21q22\n",
      "chr13q14\n",
      "chr13q12\n",
      "chr13q34\n",
      "chrxp22\n",
      "chr21q21\n",
      "chr13q32\n",
      "chr13q22\n",
      "chrxp11\n",
      "chr7q31\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM1_2.csv_PAGE/filt_FC_LVM1_2.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM1_2.csv_PAGE/filt_FC_LVM1_2.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM1_2.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM1_2.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 6, 14, 0.0629899810942742\n",
      "best merge is 0, 3, 0.0799558120640301\n",
      "best merge is 2, 13, 0.110596085660057\n",
      "best merge is 0, 1, 0.139228260797042\n",
      "best merge is 0, 4, 0.156726734565951\n",
      "best merge is 0, 6, 0.227069410273795\n",
      "best merge is 5, 10, 0.258363171315387\n",
      "best merge is 0, 5, 0.331839123770227\n",
      "best merge is 0, 7, 0.493183230210246\n",
      "best merge is 0, 9, 0.524900553332698\n",
      "best merge is 8, 11, 0.527540834141488\n",
      "order classes\n",
      "10 - 0.468544238366249\n",
      "2 - 0.414735321225214\n",
      "2 13\n",
      "0 3 1 4 6 14 5 10 7 9\n",
      "8\n",
      "11\n",
      "12\n",
      "Done.-19.7198221092757\t42\t0\n",
      "-19.3562579415502\t42\t0.560355781448539\n",
      "-19.2195044472681\t42\t0.72712833545108\n",
      "-19.1227763659466\t42\t0.833862770012706\n",
      "-19.0360546378653\t42\t0.920584498094028\n",
      "-18.9459974587039\t42\t1.00730622617535\n",
      "-18.8359275730623\t42\t1.10069885641677\n",
      "-18.6558132147395\t42\t1.2274459974587\n",
      "1.73046378653113\t42\t1.46092757306226\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM1_2.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM2_1.csv --species=human_ensembl_msigdb_c1 --exptype=continuous --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM2_2.csv --species=human_ensembl_msigdb_c1 --exptype=continuous --ebins=9\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM1_3.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c1.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c1/human_ensembl_msigdb_c1.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM1_3.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 326\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t326\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 326\n",
      "cat 46, mi = 0.00234733, chr13q14\n",
      "cat 50, mi = 0.00198145, chr13q32\n",
      "cat 44, mi = 0.00178679, chr13q12\n",
      "cat 157, mi = 0.00174527, chr2p23\n",
      "cat 52, mi = 0.00170895, chr13q34\n",
      "cat 143, mi = 0.00169996, chr21q22\n",
      "cat 151, mi = 0.00168198, chr2p13\n",
      "cat 95, mi = 0.00154285, chr17q25\n",
      "cat 16, mi = 0.00152463, chr11p13\n",
      "cat 129, mi = 0.00147192, chr1q32\n",
      "cat 245, mi = 0.00141786, chr6p22\n",
      "cat 48, mi = 0.00141460, chr13q22\n",
      "cat 162, mi = 0.00140020, chr2q12\n",
      "cat 61, mi = 0.00135644, chr14q31\n",
      "cat 156, mi = 0.00133046, chr2p22\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 46, mi = 0.002, pvalue = 0.000, chr13q14\n",
      "Passed: cat 50, mi = 0.002, pvalue = 0.000, chr13q32\n",
      "Passed: cat 44, mi = 0.002, pvalue = 0.000, chr13q12\n",
      "Passed: cat 157, mi = 0.002, pvalue = 0.000, chr2p23\n",
      "Passed: cat 52, mi = 0.002, pvalue = 0.000, chr13q34\n",
      "Passed: cat 143, mi = 0.002, pvalue = 0.000, chr21q22\n",
      "Passed: cat 151, mi = 0.002, pvalue = 0.001, chr2p13\n",
      "Passed: cat 95, mi = 0.002, pvalue = 0.001, chr17q25\n",
      "Passed: cat 16, mi = 0.002, pvalue = 0.002, chr11p13\n",
      "Passed: cat 129, mi = 0.001, pvalue = 0.002, chr1q32\n",
      "Passed: cat 245, mi = 0.001, pvalue = 0.004, chr6p22\n",
      "Passed: cat 48, mi = 0.001, pvalue = 0.001, chr13q22\n",
      "Passed: cat 162, mi = 0.001, pvalue = 0.005, chr2q12\n",
      "Passed: cat 61, mi = 0.001, pvalue = 0.004, chr14q31\n",
      "Didn't pass (cutoff=0.0050): cat 156, mi = 0.001, pvalue = 0.006, chr2p22\n",
      "Didn't pass (cutoff=0.0050): cat 192, mi = 0.001, pvalue = 0.010, chr3q22\n",
      "Didn't pass (cutoff=0.0050): cat 182, mi = 0.001, pvalue = 0.013, chr3p22\n",
      "Didn't pass (cutoff=0.0050): cat 139, mi = 0.001, pvalue = 0.007, chr20q13\n",
      "Passed: cat 51, mi = 0.001, pvalue = 0.002, chr13q33\n",
      "Didn't pass (cutoff=0.0050): cat 226, mi = 0.001, pvalue = 0.015, chr5p15\n",
      "Didn't pass (cutoff=0.0050): cat 295, mi = 0.001, pvalue = 0.007, chr9p12\n",
      "Didn't pass (cutoff=0.0050): cat 323, mi = 0.001, pvalue = 0.014, chrxq28\n",
      "Didn't pass (cutoff=0.0050): cat 246, mi = 0.001, pvalue = 0.019, chr6p23\n",
      "Didn't pass (cutoff=0.0050): cat 4, mi = 0.001, pvalue = 0.022, chr10p15\n",
      "Didn't pass (cutoff=0.0050): cat 278, mi = 0.001, pvalue = 0.031, chr7q35\n",
      "Didn't pass (cutoff=0.0050): cat 272, mi = 0.001, pvalue = 0.030, chr7q21\n",
      "Didn't pass (cutoff=0.0050): cat 292, mi = 0.001, pvalue = 0.025, chr8q24\n",
      "Didn't pass (cutoff=0.0050): cat 86, mi = 0.001, pvalue = 0.027, chr17p12\n",
      "Didn't pass (cutoff=0.0050): cat 54, mi = 0.001, pvalue = 0.029, chr14q11\n",
      "Didn't pass (cutoff=0.0050): cat 209, mi = 0.001, pvalue = 0.043, chr4q13\n",
      "Didn't pass (cutoff=0.0050): cat 211, mi = 0.001, pvalue = 0.030, chr4q22\n",
      "Didn't pass (cutoff=0.0050): cat 287, mi = 0.001, pvalue = 0.057, chr8q12\n",
      "Didn't pass (cutoff=0.0050): cat 126, mi = 0.001, pvalue = 0.053, chr1q24\n",
      "Didn't pass (cutoff=0.0050): cat 284, mi = 0.001, pvalue = 0.050, chr8p23\n",
      "Didn't pass (cutoff=0.0050): cat 45, mi = 0.001, pvalue = 0.052, chr13q13\n",
      "Didn't pass (cutoff=0.0050): cat 152, mi = 0.001, pvalue = 0.053, chr2p14\n",
      "Didn't pass (cutoff=0.0050): cat 125, mi = 0.001, pvalue = 0.045, chr1q23\n",
      "Didn't pass (cutoff=0.0050): cat 253, mi = 0.001, pvalue = 0.035, chr6q15\n",
      "Didn't pass (cutoff=0.0050): cat 267, mi = 0.001, pvalue = 0.045, chr7p15\n",
      "Total number of categories evaluated = 29.\n",
      "Number of categories that passed the tests = 15\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM1_3.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM1_3.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 1, 11, 0.0538758005983764\n",
      "best merge is 7, 14, 0.0702496520333106\n",
      "best merge is 0, 4, 0.0721246108003979\n",
      "best merge is 0, 1, 0.114451719983197\n",
      "best merge is 2, 7, 0.163788325301813\n",
      "best merge is 0, 10, 0.201421975987701\n",
      "best merge is 6, 8, 0.237247471193889\n",
      "best merge is 0, 5, 0.435379057388642\n",
      "best merge is 9, 12, 0.494268533998615\n",
      "best merge is 0, 2, 0.530416829336527\n",
      "best merge is 3, 9, 0.553216593529693\n",
      "order classes\n",
      "9 - 0.556037848268313\n",
      "2 - 0.889678016977085\n",
      "2 - 1.85350700249481\n",
      "0 4 1 11 10 5 2 7 14\n",
      "3\n",
      "6 8\n",
      "9 12\n",
      "13\n",
      "Done.Start drawing\n",
      "84.94\t0.00\n",
      "chr1q32\n",
      "chr2q12\n",
      "chr14q31\n",
      "chr2p23\n",
      "chr2p13\n",
      "chr11p13\n",
      "chr21q22\n",
      "chr13q12\n",
      "chr17q25\n",
      "chr13q33\n",
      "chr13q14\n",
      "chr13q34\n",
      "chr13q32\n",
      "chr13q22\n",
      "chr6p22\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM1_3.csv_PAGE/filt_FC_LVM1_3.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM1_3.csv_PAGE/filt_FC_LVM1_3.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM1_3.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM1_3.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 1, 11, 0.0538758005983764\n",
      "best merge is 7, 14, 0.0702496520333106\n",
      "best merge is 0, 4, 0.0721246108003979\n",
      "best merge is 0, 1, 0.114451719983197\n",
      "best merge is 2, 7, 0.163788325301813\n",
      "best merge is 0, 10, 0.201421975987701\n",
      "best merge is 6, 8, 0.237247471193889\n",
      "best merge is 0, 5, 0.435379057388642\n",
      "best merge is 9, 12, 0.494268533998615\n",
      "best merge is 0, 2, 0.530416829336527\n",
      "best merge is 3, 9, 0.553216593529693\n",
      "order classes\n",
      "9 - 0.556037848268313\n",
      "2 - 0.889678016977085\n",
      "2 - 1.85350700249481\n",
      "0 4 1 11 10 5 2 7 14\n",
      "3\n",
      "6 8\n",
      "9 12\n",
      "13\n",
      "Done.-19.849187661879\t42\t0\n",
      "-19.6588179891688\t42\t0.301624676242053\n",
      "-19.5920649870497\t42\t0.380739345420297\n",
      "-19.5401459854015\t42\t0.435130680480339\n",
      "-19.4931716505769\t42\t0.484577348716741\n",
      "-19.4437249823405\t42\t0.529079350129503\n",
      "-19.3769719802213\t42\t0.583470685189545\n",
      "-19.2706616435131\t42\t0.662585354367789\n",
      "1.39804567930304\t42\t0.796091358606075\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM1_3.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM2_3.csv --species=human_ensembl_msigdb_c1 --exptype=continuous --ebins=9\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM1_4.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c1.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c1/human_ensembl_msigdb_c1.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM1_4.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 326\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t326\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 326\n",
      "cat 44, mi = 0.00307343, chr13q12\n",
      "cat 95, mi = 0.00246547, chr17q25\n",
      "cat 274, mi = 0.00231614, chr7q31\n",
      "cat 46, mi = 0.00207495, chr13q14\n",
      "cat 50, mi = 0.00195216, chr13q32\n",
      "cat 292, mi = 0.00182524, chr8q24\n",
      "cat 205, mi = 0.00165436, chr4p16\n",
      "cat 100, mi = 0.00151397, chr18q21\n",
      "cat 48, mi = 0.00135205, chr13q22\n",
      "cat 212, mi = 0.00132491, chr4q23\n",
      "cat 82, mi = 0.00124566, chr16q22\n",
      "cat 245, mi = 0.00124449, chr6p22\n",
      "cat 61, mi = 0.00124116, chr14q31\n",
      "cat 192, mi = 0.00124029, chr3q22\n",
      "cat 288, mi = 0.00120894, chr8q13\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 44, mi = 0.003, pvalue = 0.000, chr13q12\n",
      "Passed: cat 95, mi = 0.002, pvalue = 0.000, chr17q25\n",
      "Passed: cat 274, mi = 0.002, pvalue = 0.000, chr7q31\n",
      "Passed: cat 46, mi = 0.002, pvalue = 0.000, chr13q14\n",
      "Passed: cat 50, mi = 0.002, pvalue = 0.000, chr13q32\n",
      "Passed: cat 292, mi = 0.002, pvalue = 0.000, chr8q24\n",
      "Passed: cat 205, mi = 0.002, pvalue = 0.001, chr4p16\n",
      "Passed: cat 100, mi = 0.002, pvalue = 0.003, chr18q21\n",
      "Passed: cat 48, mi = 0.001, pvalue = 0.002, chr13q22\n",
      "Passed: cat 212, mi = 0.001, pvalue = 0.001, chr4q23\n",
      "Didn't pass (cutoff=0.0050): cat 82, mi = 0.001, pvalue = 0.011, chr16q22\n",
      "Didn't pass (cutoff=0.0050): cat 245, mi = 0.001, pvalue = 0.013, chr6p22\n",
      "Didn't pass (cutoff=0.0050): cat 61, mi = 0.001, pvalue = 0.011, chr14q31\n",
      "Didn't pass (cutoff=0.0050): cat 192, mi = 0.001, pvalue = 0.013, chr3q22\n",
      "Didn't pass (cutoff=0.0050): cat 288, mi = 0.001, pvalue = 0.018, chr8q13\n",
      "Didn't pass (cutoff=0.0050): cat 45, mi = 0.001, pvalue = 0.013, chr13q13\n",
      "Didn't pass (cutoff=0.0050): cat 51, mi = 0.001, pvalue = 0.006, chr13q33\n",
      "Didn't pass (cutoff=0.0050): cat 152, mi = 0.001, pvalue = 0.021, chr2p14\n",
      "Didn't pass (cutoff=0.0050): cat 89, mi = 0.001, pvalue = 0.018, chr17q11\n",
      "Didn't pass (cutoff=0.0050): cat 54, mi = 0.001, pvalue = 0.016, chr14q11\n",
      "Didn't pass (cutoff=0.0050): cat 4, mi = 0.001, pvalue = 0.024, chr10p15\n",
      "Didn't pass (cutoff=0.0050): cat 52, mi = 0.001, pvalue = 0.028, chr13q34\n",
      "Didn't pass (cutoff=0.0050): cat 25, mi = 0.001, pvalue = 0.031, chr11q22\n",
      "Didn't pass (cutoff=0.0050): cat 218, mi = 0.001, pvalue = 0.029, chr4q31\n",
      "Didn't pass (cutoff=0.0050): cat 159, mi = 0.001, pvalue = 0.032, chr2p25\n",
      "Didn't pass (cutoff=0.0050): cat 60, mi = 0.001, pvalue = 0.024, chr14q24\n",
      "Didn't pass (cutoff=0.0050): cat 3, mi = 0.001, pvalue = 0.025, chr10p14\n",
      "Didn't pass (cutoff=0.0050): cat 311, mi = 0.001, pvalue = 0.029, chrxp22\n",
      "Didn't pass (cutoff=0.0050): cat 138, mi = 0.001, pvalue = 0.043, chr20q12\n",
      "Didn't pass (cutoff=0.0050): cat 164, mi = 0.001, pvalue = 0.050, chr2q14\n",
      "Total number of categories evaluated = 20.\n",
      "Number of categories that passed the tests = 10\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM1_4.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM1_4.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 0, 5, 0.0652589056941481\n",
      "best merge is 2, 8, 0.0930610421364358\n",
      "best merge is 0, 3, 0.0948514275149299\n",
      "best merge is 2, 4, 0.155201253694471\n",
      "best merge is 0, 1, 0.289586575158948\n",
      "best merge is 0, 2, 0.549286374949268\n",
      "order classes\n",
      "4 - 0.351162964437766\n",
      "3 - 0.298861888537317\n",
      "2 8 4\n",
      "0 5 3 1\n",
      "6\n",
      "7\n",
      "9\n",
      "Done.Start drawing\n",
      "26.31\t0.00\n",
      "chr4p16\n",
      "chr18q21\n",
      "chr4q23\n",
      "chr17q25\n",
      "chr13q12\n",
      "chr8q24\n",
      "chr13q14\n",
      "chr7q31\n",
      "chr13q22\n",
      "chr13q32\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM1_4.csv_PAGE/filt_FC_LVM1_4.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM1_4.csv_PAGE/filt_FC_LVM1_4.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM1_4.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM1_4.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 0, 5, 0.0652589056941481\n",
      "best merge is 2, 8, 0.0930610421364358\n",
      "best merge is 0, 3, 0.0948514275149299\n",
      "best merge is 2, 4, 0.155201253694471\n",
      "best merge is 0, 1, 0.289586575158948\n",
      "best merge is 0, 2, 0.549286374949268\n",
      "order classes\n",
      "4 - 0.351162964437766\n",
      "3 - 0.298861888537317\n",
      "2 8 4\n",
      "0 5 3 1\n",
      "6\n",
      "7\n",
      "9\n",
      "Done.-19.5290763968073\t42\t0\n",
      "-18.9064994298746\t42\t0.941847206385405\n",
      "-18.6590649942987\t42\t1.24515393386545\n",
      "-18.4754846066135\t42\t1.43671607753706\n",
      "-18.3158494868871\t42\t1.61231470923603\n",
      "-18.1641961231471\t42\t1.75598631698974\n",
      "-17.9806157354618\t42\t1.91562143671608\n",
      "-17.709236031927\t42\t2.12314709236032\n",
      "2.22919042189282\t42\t2.45838084378563\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM1_4.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM2_4.csv --species=human_ensembl_msigdb_c1 --exptype=continuous --ebins=9\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM2_1.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c1.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c1/human_ensembl_msigdb_c1.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM2_1.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 326\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t326\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 326\n",
      "cat 46, mi = 0.00401895, chr13q14\n",
      "cat 44, mi = 0.00400952, chr13q12\n",
      "cat 311, mi = 0.00336362, chrxp22\n",
      "cat 95, mi = 0.00320702, chr17q25\n",
      "cat 52, mi = 0.00297577, chr13q34\n",
      "cat 143, mi = 0.00293450, chr21q22\n",
      "cat 309, mi = 0.00277301, chrxp11\n",
      "cat 93, mi = 0.00246891, chr17q23\n",
      "cat 292, mi = 0.00243761, chr8q24\n",
      "cat 45, mi = 0.00221357, chr13q13\n",
      "cat 50, mi = 0.00195786, chr13q32\n",
      "cat 129, mi = 0.00188065, chr1q32\n",
      "cat 48, mi = 0.00175782, chr13q22\n",
      "cat 315, mi = 0.00168429, chrxq13\n",
      "cat 274, mi = 0.00165982, chr7q31\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 46, mi = 0.004, pvalue = 0.000, chr13q14\n",
      "Passed: cat 44, mi = 0.004, pvalue = 0.000, chr13q12\n",
      "Passed: cat 311, mi = 0.003, pvalue = 0.000, chrxp22\n",
      "Passed: cat 95, mi = 0.003, pvalue = 0.000, chr17q25\n",
      "Passed: cat 52, mi = 0.003, pvalue = 0.000, chr13q34\n",
      "Passed: cat 143, mi = 0.003, pvalue = 0.000, chr21q22\n",
      "Passed: cat 309, mi = 0.003, pvalue = 0.000, chrxp11\n",
      "Passed: cat 93, mi = 0.002, pvalue = 0.000, chr17q23\n",
      "Passed: cat 292, mi = 0.002, pvalue = 0.000, chr8q24\n",
      "Passed: cat 45, mi = 0.002, pvalue = 0.000, chr13q13\n",
      "Passed: cat 50, mi = 0.002, pvalue = 0.000, chr13q32\n",
      "Passed: cat 129, mi = 0.002, pvalue = 0.000, chr1q32\n",
      "Passed: cat 48, mi = 0.002, pvalue = 0.000, chr13q22\n",
      "Passed: cat 315, mi = 0.002, pvalue = 0.001, chrxq13\n",
      "Passed: cat 274, mi = 0.002, pvalue = 0.001, chr7q31\n",
      "Passed: cat 3, mi = 0.002, pvalue = 0.000, chr10p14\n",
      "Passed: cat 89, mi = 0.002, pvalue = 0.001, chr17q11\n",
      "Passed: cat 146, mi = 0.002, pvalue = 0.002, chr22q12\n",
      "Passed: cat 245, mi = 0.002, pvalue = 0.002, chr6p22\n",
      "Passed: cat 273, mi = 0.002, pvalue = 0.002, chr7q22\n",
      "Passed: cat 213, mi = 0.001, pvalue = 0.002, chr4q24\n",
      "Passed: cat 219, mi = 0.001, pvalue = 0.003, chr4q32\n",
      "Didn't pass (cutoff=0.0050): cat 71, mi = 0.001, pvalue = 0.005, chr15q24\n",
      "Passed: cat 291, mi = 0.001, pvalue = 0.003, chr8q23\n",
      "Didn't pass (cutoff=0.0050): cat 222, mi = 0.001, pvalue = 0.007, chr4q35\n",
      "Didn't pass (cutoff=0.0050): cat 259, mi = 0.001, pvalue = 0.009, chr6q25\n",
      "Didn't pass (cutoff=0.0050): cat 254, mi = 0.001, pvalue = 0.007, chr6q16\n",
      "Didn't pass (cutoff=0.0050): cat 243, mi = 0.001, pvalue = 0.021, chr6p12\n",
      "Didn't pass (cutoff=0.0050): cat 79, mi = 0.001, pvalue = 0.025, chr16q12\n",
      "Didn't pass (cutoff=0.0050): cat 305, mi = 0.001, pvalue = 0.027, chr9q31\n",
      "Didn't pass (cutoff=0.0050): cat 132, mi = 0.001, pvalue = 0.005, chr1q43\n",
      "Didn't pass (cutoff=0.0050): cat 12, mi = 0.001, pvalue = 0.019, chr10q26\n",
      "Didn't pass (cutoff=0.0050): cat 33, mi = 0.001, pvalue = 0.016, chr12q\n",
      "Didn't pass (cutoff=0.0050): cat 258, mi = 0.001, pvalue = 0.034, chr6q24\n",
      "Didn't pass (cutoff=0.0050): cat 229, mi = 0.001, pvalue = 0.037, chr5q12\n",
      "Didn't pass (cutoff=0.0050): cat 16, mi = 0.001, pvalue = 0.039, chr11p13\n",
      "Didn't pass (cutoff=0.0050): cat 167, mi = 0.001, pvalue = 0.031, chr2q23\n",
      "Didn't pass (cutoff=0.0050): cat 210, mi = 0.001, pvalue = 0.036, chr4q21\n",
      "Didn't pass (cutoff=0.0050): cat 290, mi = 0.001, pvalue = 0.040, chr8q22\n",
      "Didn't pass (cutoff=0.0050): cat 298, mi = 0.001, pvalue = 0.034, chr9p22\n",
      "Didn't pass (cutoff=0.0050): cat 139, mi = 0.001, pvalue = 0.029, chr20q13\n",
      "Didn't pass (cutoff=0.0050): cat 186, mi = 0.001, pvalue = 0.025, chr3p26\n",
      "Didn't pass (cutoff=0.0050): cat 156, mi = 0.001, pvalue = 0.048, chr2p22\n",
      "Didn't pass (cutoff=0.0050): cat 169, mi = 0.001, pvalue = 0.043, chr2q31\n",
      "Total number of categories evaluated = 34.\n",
      "Number of categories that passed the tests = 23\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM2_1.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM2_1.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 10, 22, 0.0168620856321573\n",
      "best merge is 4, 12, 0.0265189093308913\n",
      "best merge is 1, 2, 0.0426712737128505\n",
      "best merge is 4, 10, 0.0546798892867017\n",
      "best merge is 0, 6, 0.0784652966060563\n",
      "best merge is 3, 16, 0.0900916126953564\n",
      "best merge is 5, 7, 0.0979264814974269\n",
      "best merge is 4, 14, 0.104329012298982\n",
      "best merge is 1, 9, 0.106974376866177\n",
      "best merge is 4, 18, 0.134677794244974\n",
      "best merge is 1, 13, 0.166655472703392\n",
      "best merge is 3, 5, 0.185605095428645\n",
      "best merge is 1, 3, 0.215011678422903\n",
      "best merge is 11, 21, 0.235579648344664\n",
      "best merge is 0, 1, 0.321483975315157\n",
      "best merge is 0, 8, 0.345999002681267\n",
      "best merge is 11, 17, 0.599368468361241\n",
      "best merge is 0, 4, 0.617596338130669\n",
      "best merge is 11, 15, 0.732736323296861\n",
      "order classes\n",
      "17 - 0.523607659539362\n",
      "3 - 2.44365047826254\n",
      "0 6 1 2 9 13 3 16 5 7 8 4 12 10 22 14 18\n",
      "11 21 17\n",
      "15\n",
      "19\n",
      "20\n",
      "Done.Start drawing\n",
      "70.09\t0.00\n",
      "chr1q32\n",
      "chr4q32\n",
      "chr10p14\n",
      "chr4q24\n",
      "chr22q12\n",
      "chr7q22\n",
      "chr13q14\n",
      "chrxp11\n",
      "chr13q12\n",
      "chrxp22\n",
      "chr13q13\n",
      "chrxq13\n",
      "chr17q25\n",
      "chr17q11\n",
      "chr21q22\n",
      "chr17q23\n",
      "chr8q24\n",
      "chr13q34\n",
      "chr13q22\n",
      "chr13q32\n",
      "chr8q23\n",
      "chr7q31\n",
      "chr6p22\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM2_1.csv_PAGE/filt_FC_LVM2_1.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM2_1.csv_PAGE/filt_FC_LVM2_1.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM2_1.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM2_1.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 10, 22, 0.0168620856321573\n",
      "best merge is 4, 12, 0.0265189093308913\n",
      "best merge is 1, 2, 0.0426712737128505\n",
      "best merge is 4, 10, 0.0546798892867017\n",
      "best merge is 0, 6, 0.0784652966060563\n",
      "best merge is 3, 16, 0.0900916126953564\n",
      "best merge is 5, 7, 0.0979264814974269\n",
      "best merge is 4, 14, 0.104329012298982\n",
      "best merge is 1, 9, 0.106974376866177\n",
      "best merge is 4, 18, 0.134677794244974\n",
      "best merge is 1, 13, 0.166655472703392\n",
      "best merge is 3, 5, 0.185605095428645\n",
      "best merge is 1, 3, 0.215011678422903\n",
      "best merge is 11, 21, 0.235579648344664\n",
      "best merge is 0, 1, 0.321483975315157\n",
      "best merge is 0, 8, 0.345999002681267\n",
      "best merge is 11, 17, 0.599368468361241\n",
      "best merge is 0, 4, 0.617596338130669\n",
      "best merge is 11, 15, 0.732736323296861\n",
      "order classes\n",
      "17 - 0.523607659539362\n",
      "3 - 2.44365047826254\n",
      "0 6 1 2 9 13 3 16 5 7 8 4 12 10 22 14 18\n",
      "11 21 17\n",
      "15\n",
      "19\n",
      "20\n",
      "Done.-19.784277357683\t42\t0\n",
      "-19.5056356113568\t42\t0.431445284634042\n",
      "-19.4037665858182\t42\t0.557283492652304\n",
      "-19.3318590383792\t42\t0.635183335711228\n",
      "-19.2689399343701\t42\t0.701098587530318\n",
      "-19.203024682551\t42\t0.761021543729491\n",
      "-19.1221286916821\t42\t0.832929091168498\n",
      "-18.9902981880439\t42\t0.922813525467256\n",
      "1.54829504922243\t42\t1.09659009844486\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM2_1.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM2_2.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c1.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c1/human_ensembl_msigdb_c1.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM2_2.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 326\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t326\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 326\n",
      "cat 46, mi = 0.00345719, chr13q14\n",
      "cat 52, mi = 0.00335124, chr13q34\n",
      "cat 245, mi = 0.00314955, chr6p22\n",
      "cat 95, mi = 0.00302953, chr17q25\n",
      "cat 44, mi = 0.00265646, chr13q12\n",
      "cat 309, mi = 0.00225646, chrxp11\n",
      "cat 274, mi = 0.00215341, chr7q31\n",
      "cat 50, mi = 0.00212361, chr13q32\n",
      "cat 259, mi = 0.00199858, chr6q25\n",
      "cat 48, mi = 0.00189012, chr13q22\n",
      "cat 292, mi = 0.00180923, chr8q24\n",
      "cat 209, mi = 0.00166554, chr4q13\n",
      "cat 129, mi = 0.00162017, chr1q32\n",
      "cat 104, mi = 0.00158145, chr19p12\n",
      "cat 210, mi = 0.00155403, chr4q21\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 46, mi = 0.003, pvalue = 0.000, chr13q14\n",
      "Passed: cat 52, mi = 0.003, pvalue = 0.000, chr13q34\n",
      "Passed: cat 245, mi = 0.003, pvalue = 0.000, chr6p22\n",
      "Passed: cat 95, mi = 0.003, pvalue = 0.000, chr17q25\n",
      "Passed: cat 44, mi = 0.003, pvalue = 0.000, chr13q12\n",
      "Passed: cat 309, mi = 0.002, pvalue = 0.000, chrxp11\n",
      "Passed: cat 274, mi = 0.002, pvalue = 0.000, chr7q31\n",
      "Passed: cat 50, mi = 0.002, pvalue = 0.000, chr13q32\n",
      "Passed: cat 259, mi = 0.002, pvalue = 0.000, chr6q25\n",
      "Passed: cat 48, mi = 0.002, pvalue = 0.000, chr13q22\n",
      "Passed: cat 292, mi = 0.002, pvalue = 0.000, chr8q24\n",
      "Passed: cat 209, mi = 0.002, pvalue = 0.000, chr4q13\n",
      "Passed: cat 129, mi = 0.002, pvalue = 0.001, chr1q32\n",
      "Passed: cat 104, mi = 0.002, pvalue = 0.001, chr19p12\n",
      "Passed: cat 210, mi = 0.002, pvalue = 0.002, chr4q21\n",
      "Passed: cat 291, mi = 0.002, pvalue = 0.001, chr8q23\n",
      "Passed: cat 277, mi = 0.001, pvalue = 0.003, chr7q34\n",
      "Passed: cat 143, mi = 0.001, pvalue = 0.003, chr21q22\n",
      "Passed: cat 69, mi = 0.001, pvalue = 0.004, chr15q22\n",
      "Passed: cat 1, mi = 0.001, pvalue = 0.004, chr10p12\n",
      "Passed: cat 214, mi = 0.001, pvalue = 0.003, chr4q25\n",
      "Passed: cat 203, mi = 0.001, pvalue = 0.003, chr4p14\n",
      "Didn't pass (cutoff=0.0050): cat 93, mi = 0.001, pvalue = 0.007, chr17q23\n",
      "Didn't pass (cutoff=0.0050): cat 311, mi = 0.001, pvalue = 0.005, chrxp22\n",
      "Passed: cat 0, mi = 0.001, pvalue = 0.004, chr10p11\n",
      "Didn't pass (cutoff=0.0050): cat 114, mi = 0.001, pvalue = 0.006, chr1p22\n",
      "Didn't pass (cutoff=0.0050): cat 303, mi = 0.001, pvalue = 0.011, chr9q21\n",
      "Didn't pass (cutoff=0.0050): cat 149, mi = 0.001, pvalue = 0.013, chr2p11\n",
      "Didn't pass (cutoff=0.0050): cat 182, mi = 0.001, pvalue = 0.011, chr3p22\n",
      "Didn't pass (cutoff=0.0050): cat 219, mi = 0.001, pvalue = 0.009, chr4q32\n",
      "Didn't pass (cutoff=0.0050): cat 222, mi = 0.001, pvalue = 0.017, chr4q35\n",
      "Didn't pass (cutoff=0.0050): cat 229, mi = 0.001, pvalue = 0.018, chr5q12\n",
      "Didn't pass (cutoff=0.0050): cat 89, mi = 0.001, pvalue = 0.012, chr17q11\n",
      "Didn't pass (cutoff=0.0050): cat 231, mi = 0.001, pvalue = 0.019, chr5q14\n",
      "Didn't pass (cutoff=0.0050): cat 161, mi = 0.001, pvalue = 0.022, chr2q11\n",
      "Passed: cat 194, mi = 0.001, pvalue = 0.004, chr3q24\n",
      "Didn't pass (cutoff=0.0050): cat 290, mi = 0.001, pvalue = 0.027, chr8q22\n",
      "Didn't pass (cutoff=0.0050): cat 232, mi = 0.001, pvalue = 0.025, chr5q15\n",
      "Didn't pass (cutoff=0.0050): cat 305, mi = 0.001, pvalue = 0.035, chr9q31\n",
      "Didn't pass (cutoff=0.0050): cat 216, mi = 0.001, pvalue = 0.010, chr4q27\n",
      "Didn't pass (cutoff=0.0050): cat 51, mi = 0.001, pvalue = 0.015, chr13q33\n",
      "Didn't pass (cutoff=0.0050): cat 94, mi = 0.001, pvalue = 0.036, chr17q24\n",
      "Didn't pass (cutoff=0.0050): cat 314, mi = 0.001, pvalue = 0.018, chrxq12\n",
      "Didn't pass (cutoff=0.0050): cat 119, mi = 0.001, pvalue = 0.033, chr1p35\n",
      "Didn't pass (cutoff=0.0050): cat 125, mi = 0.001, pvalue = 0.048, chr1q23\n",
      "Didn't pass (cutoff=0.0050): cat 248, mi = 0.001, pvalue = 0.052, chr6p25\n",
      "Didn't pass (cutoff=0.0050): cat 97, mi = 0.001, pvalue = 0.041, chr18p11\n",
      "Didn't pass (cutoff=0.0050): cat 321, mi = 0.001, pvalue = 0.060, chrxq26\n",
      "Didn't pass (cutoff=0.0050): cat 6, mi = 0.001, pvalue = 0.056, chr10q11\n",
      "Didn't pass (cutoff=0.0050): cat 319, mi = 0.001, pvalue = 0.062, chrxq24\n",
      "Didn't pass (cutoff=0.0050): cat 273, mi = 0.001, pvalue = 0.040, chr7q22\n",
      "Didn't pass (cutoff=0.0050): cat 90, mi = 0.001, pvalue = 0.059, chr17q12\n",
      "Didn't pass (cutoff=0.0050): cat 230, mi = 0.001, pvalue = 0.062, chr5q13\n",
      "Didn't pass (cutoff=0.0050): cat 186, mi = 0.001, pvalue = 0.039, chr3p26\n",
      "Didn't pass (cutoff=0.0050): cat 3, mi = 0.001, pvalue = 0.058, chr10p14\n",
      "Didn't pass (cutoff=0.0050): cat 102, mi = 0.001, pvalue = 0.062, chr18q23\n",
      "Total number of categories evaluated = 48.\n",
      "Number of categories that passed the tests = 24\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM2_2.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM2_2.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 7, 9, 0.0150488839914575\n",
      "best merge is 4, 15, 0.0166424323056471\n",
      "best merge is 1, 4, 0.0343672677345799\n",
      "best merge is 20, 21, 0.0565987488546376\n",
      "best merge is 0, 5, 0.0634656254103834\n",
      "best merge is 2, 20, 0.0961753444085028\n",
      "best merge is 0, 1, 0.112215679489007\n",
      "best merge is 6, 13, 0.116136719863866\n",
      "best merge is 11, 23, 0.123426132256667\n",
      "best merge is 2, 12, 0.144265640010121\n",
      "best merge is 0, 3, 0.179076644899359\n",
      "best merge is 6, 7, 0.240758589885062\n",
      "best merge is 8, 19, 0.244061836933605\n",
      "best merge is 0, 17, 0.284603250497215\n",
      "best merge is 8, 11, 0.397049944936974\n",
      "best merge is 0, 16, 0.43208168632581\n",
      "best merge is 0, 10, 0.486535603404703\n",
      "best merge is 14, 22, 0.5169694823766\n",
      "best merge is 0, 6, 0.566796653083594\n",
      "best merge is 2, 8, 0.627662524537321\n",
      "order classes\n",
      "13 - 0.695659073358708\n",
      "4 - 0.511309768276505\n",
      "4 - 1.46676581170363\n",
      "2 - 3.1018168942596\n",
      "2 20 21 12\n",
      "0 5 1 4 15 3 17 16 10 6 13 7 9\n",
      "8 19 11 23\n",
      "14 22\n",
      "18\n",
      "Done.Start drawing\n",
      "101.03\t0.00\n",
      "chr6p22\n",
      "chr4q25\n",
      "chr4p14\n",
      "chr1q32\n",
      "chr4q21\n",
      "chr6q25\n",
      "chr10p12\n",
      "chr4q13\n",
      "chr3q24\n",
      "chr10p11\n",
      "chr15q22\n",
      "chr8q24\n",
      "chr13q14\n",
      "chrxp11\n",
      "chr13q34\n",
      "chr13q12\n",
      "chr8q23\n",
      "chr17q25\n",
      "chr7q34\n",
      "chr21q22\n",
      "chr7q31\n",
      "chr19p12\n",
      "chr13q32\n",
      "chr13q22\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM2_2.csv_PAGE/filt_FC_LVM2_2.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM2_2.csv_PAGE/filt_FC_LVM2_2.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM2_2.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM2_2.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 7, 9, 0.0150488839914575\n",
      "best merge is 4, 15, 0.0166424323056471\n",
      "best merge is 1, 4, 0.0343672677345799\n",
      "best merge is 20, 21, 0.0565987488546376\n",
      "best merge is 0, 5, 0.0634656254103834\n",
      "best merge is 2, 20, 0.0961753444085028\n",
      "best merge is 0, 1, 0.112215679489007\n",
      "best merge is 6, 13, 0.116136719863866\n",
      "best merge is 11, 23, 0.123426132256667\n",
      "best merge is 2, 12, 0.144265640010121\n",
      "best merge is 0, 3, 0.179076644899359\n",
      "best merge is 6, 7, 0.240758589885062\n",
      "best merge is 8, 19, 0.244061836933605\n",
      "best merge is 0, 17, 0.284603250497215\n",
      "best merge is 8, 11, 0.397049944936974\n",
      "best merge is 0, 16, 0.43208168632581\n",
      "best merge is 0, 10, 0.486535603404703\n",
      "best merge is 14, 22, 0.5169694823766\n",
      "best merge is 0, 6, 0.566796653083594\n",
      "best merge is 2, 8, 0.627662524537321\n",
      "order classes\n",
      "13 - 0.695659073358708\n",
      "4 - 0.511309768276505\n",
      "4 - 1.46676581170363\n",
      "2 - 3.1018168942596\n",
      "2 20 21 12\n",
      "0 5 1 4 15 3 17 16 10 6 13 7 9\n",
      "8 19 11 23\n",
      "14 22\n",
      "18\n",
      "Done.-19.8544986637632\t42\t0\n",
      "-19.6570325645848\t42\t0.291002672473523\n",
      "-19.5759675343957\t42\t0.394932198356924\n",
      "-19.5219241809364\t42\t0.453132732851628\n",
      "-19.4720380085123\t42\t0.503018905275661\n",
      "-19.417994655053\t42\t0.552905077699693\n",
      "-19.3514797584876\t42\t0.611105612194398\n",
      "-19.2454716420865\t42\t0.685934870830446\n",
      "1.41156092249827\t42\t0.823121844996536\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM2_2.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM2_3.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c1.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c1/human_ensembl_msigdb_c1.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM2_3.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 326\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t326\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 326\n",
      "cat 52, mi = 0.00241953, chr13q34\n",
      "cat 129, mi = 0.00236779, chr1q32\n",
      "cat 46, mi = 0.00230763, chr13q14\n",
      "cat 245, mi = 0.00208958, chr6p22\n",
      "cat 50, mi = 0.00203377, chr13q32\n",
      "cat 44, mi = 0.00195734, chr13q12\n",
      "cat 95, mi = 0.00194245, chr17q25\n",
      "cat 143, mi = 0.00192563, chr21q22\n",
      "cat 292, mi = 0.00183382, chr8q24\n",
      "cat 311, mi = 0.00181326, chrxp22\n",
      "cat 274, mi = 0.00178555, chr7q31\n",
      "cat 290, mi = 0.00161741, chr8q22\n",
      "cat 208, mi = 0.00146407, chr4q12\n",
      "cat 277, mi = 0.00133619, chr7q34\n",
      "cat 309, mi = 0.00132191, chrxp11\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 52, mi = 0.002, pvalue = 0.000, chr13q34\n",
      "Passed: cat 129, mi = 0.002, pvalue = 0.000, chr1q32\n",
      "Passed: cat 46, mi = 0.002, pvalue = 0.000, chr13q14\n",
      "Passed: cat 245, mi = 0.002, pvalue = 0.000, chr6p22\n",
      "Passed: cat 50, mi = 0.002, pvalue = 0.000, chr13q32\n",
      "Passed: cat 44, mi = 0.002, pvalue = 0.000, chr13q12\n",
      "Passed: cat 95, mi = 0.002, pvalue = 0.000, chr17q25\n",
      "Passed: cat 143, mi = 0.002, pvalue = 0.000, chr21q22\n",
      "Passed: cat 292, mi = 0.002, pvalue = 0.000, chr8q24\n",
      "Passed: cat 311, mi = 0.002, pvalue = 0.000, chrxp22\n",
      "Passed: cat 274, mi = 0.002, pvalue = 0.000, chr7q31\n",
      "Passed: cat 290, mi = 0.002, pvalue = 0.001, chr8q22\n",
      "Passed: cat 208, mi = 0.001, pvalue = 0.002, chr4q12\n",
      "Didn't pass (cutoff=0.0050): cat 277, mi = 0.001, pvalue = 0.009, chr7q34\n",
      "Didn't pass (cutoff=0.0050): cat 309, mi = 0.001, pvalue = 0.007, chrxp11\n",
      "Didn't pass (cutoff=0.0050): cat 23, mi = 0.001, pvalue = 0.009, chr11q14\n",
      "Didn't pass (cutoff=0.0050): cat 25, mi = 0.001, pvalue = 0.011, chr11q22\n",
      "Passed: cat 47, mi = 0.001, pvalue = 0.000, chr13q21\n",
      "Didn't pass (cutoff=0.0050): cat 112, mi = 0.001, pvalue = 0.012, chr1p13\n",
      "Didn't pass (cutoff=0.0050): cat 38, mi = 0.001, pvalue = 0.010, chr12q15\n",
      "Didn't pass (cutoff=0.0050): cat 305, mi = 0.001, pvalue = 0.016, chr9q31\n",
      "Didn't pass (cutoff=0.0050): cat 97, mi = 0.001, pvalue = 0.016, chr18p11\n",
      "Didn't pass (cutoff=0.0050): cat 138, mi = 0.001, pvalue = 0.015, chr20q12\n",
      "Didn't pass (cutoff=0.0050): cat 231, mi = 0.001, pvalue = 0.018, chr5q14\n",
      "Didn't pass (cutoff=0.0050): cat 45, mi = 0.001, pvalue = 0.018, chr13q13\n",
      "Didn't pass (cutoff=0.0050): cat 59, mi = 0.001, pvalue = 0.022, chr14q23\n",
      "Didn't pass (cutoff=0.0050): cat 259, mi = 0.001, pvalue = 0.021, chr6q25\n",
      "Passed: cat 221, mi = 0.001, pvalue = 0.004, chr4q34\n",
      "Didn't pass (cutoff=0.0050): cat 56, mi = 0.001, pvalue = 0.021, chr14q13\n",
      "Didn't pass (cutoff=0.0050): cat 68, mi = 0.001, pvalue = 0.021, chr15q21\n",
      "Didn't pass (cutoff=0.0050): cat 71, mi = 0.001, pvalue = 0.023, chr15q24\n",
      "Didn't pass (cutoff=0.0050): cat 89, mi = 0.001, pvalue = 0.019, chr17q11\n",
      "Didn't pass (cutoff=0.0050): cat 0, mi = 0.001, pvalue = 0.026, chr10p11\n",
      "Didn't pass (cutoff=0.0050): cat 61, mi = 0.001, pvalue = 0.025, chr14q31\n",
      "Didn't pass (cutoff=0.0050): cat 217, mi = 0.001, pvalue = 0.027, chr4q28\n",
      "Didn't pass (cutoff=0.0050): cat 296, mi = 0.001, pvalue = 0.038, chr9p13\n",
      "Didn't pass (cutoff=0.0050): cat 192, mi = 0.001, pvalue = 0.048, chr3q22\n",
      "Didn't pass (cutoff=0.0050): cat 124, mi = 0.001, pvalue = 0.042, chr1q22\n",
      "Didn't pass (cutoff=0.0050): cat 295, mi = 0.001, pvalue = 0.027, chr9p12\n",
      "Didn't pass (cutoff=0.0050): cat 48, mi = 0.001, pvalue = 0.030, chr13q22\n",
      "Didn't pass (cutoff=0.0050): cat 104, mi = 0.001, pvalue = 0.055, chr19p12\n",
      "Didn't pass (cutoff=0.0050): cat 136, mi = 0.001, pvalue = 0.045, chr20p13\n",
      "Didn't pass (cutoff=0.0050): cat 145, mi = 0.001, pvalue = 0.037, chr22q11\n",
      "Didn't pass (cutoff=0.0050): cat 289, mi = 0.001, pvalue = 0.051, chr8q21\n",
      "Didn't pass (cutoff=0.0050): cat 232, mi = 0.001, pvalue = 0.052, chr5q15\n",
      "Didn't pass (cutoff=0.0050): cat 72, mi = 0.001, pvalue = 0.053, chr15q25\n",
      "Didn't pass (cutoff=0.0050): cat 197, mi = 0.001, pvalue = 0.062, chr3q27\n",
      "Didn't pass (cutoff=0.0050): cat 247, mi = 0.001, pvalue = 0.061, chr6p24\n",
      "Total number of categories evaluated = 38.\n",
      "Number of categories that passed the tests = 15\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM2_3.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM2_3.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 2, 5, 0.033087795124089\n",
      "best merge is 0, 2, 0.0636325749912166\n",
      "best merge is 7, 9, 0.0965639797078214\n",
      "best merge is 4, 13, 0.110235865947589\n",
      "best merge is 6, 8, 0.127575293594107\n",
      "best merge is 3, 14, 0.137931296965337\n",
      "best merge is 1, 12, 0.143291847653111\n",
      "best merge is 6, 7, 0.190347516452086\n",
      "best merge is 1, 3, 0.259442681765258\n",
      "best merge is 4, 10, 0.266889773873785\n",
      "best merge is 0, 6, 0.276642335391833\n",
      "order classes\n",
      "3 - 0.178169939007247\n",
      "4 - 0.618278377349757\n",
      "3 - 0.715572681883509\n",
      "4 - 0.46196687770794\n",
      "0 2 5\n",
      "1 12 3 14\n",
      "4 13 10\n",
      "6 8 7 9\n",
      "11\n",
      "Done.Start drawing\n",
      "69.14\t0.00\n",
      "chr1q32\n",
      "chr4q12\n",
      "chr6p22\n",
      "chr4q34\n",
      "chr17q25\n",
      "chr13q34\n",
      "chr13q14\n",
      "chr13q12\n",
      "chr8q24\n",
      "chr21q22\n",
      "chrxp22\n",
      "chr8q22\n",
      "chr13q32\n",
      "chr13q21\n",
      "chr7q31\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM2_3.csv_PAGE/filt_FC_LVM2_3.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM2_3.csv_PAGE/filt_FC_LVM2_3.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM2_3.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM2_3.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 2, 5, 0.033087795124089\n",
      "best merge is 0, 2, 0.0636325749912166\n",
      "best merge is 7, 9, 0.0965639797078214\n",
      "best merge is 4, 13, 0.110235865947589\n",
      "best merge is 6, 8, 0.127575293594107\n",
      "best merge is 3, 14, 0.137931296965337\n",
      "best merge is 1, 12, 0.143291847653111\n",
      "best merge is 6, 7, 0.190347516452086\n",
      "best merge is 1, 3, 0.259442681765258\n",
      "best merge is 4, 10, 0.266889773873785\n",
      "best merge is 0, 6, 0.276642335391833\n",
      "order classes\n",
      "3 - 0.178169939007247\n",
      "4 - 0.618278377349757\n",
      "3 - 0.715572681883509\n",
      "4 - 0.46196687770794\n",
      "0 2 5\n",
      "1 12 3 14\n",
      "4 13 10\n",
      "6 8 7 9\n",
      "11\n",
      "Done.-19.7843505929997\t42\t0\n",
      "-19.4715070870697\t42\t0.431298814000579\n",
      "-19.3166039919005\t42\t0.625687011859994\n",
      "-19.2133352617877\t42\t0.741105004339022\n",
      "-19.1191784784495\t42\t0.832224472085623\n",
      "-19.0189470639283\t42\t0.929418571015331\n",
      "-18.8974544402661\t42\t1.03268730112815\n",
      "-18.7000289268152\t42\t1.1724038183396\n",
      "1.71376916401504\t42\t1.42753832803008\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM2_3.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM2_5.csv --species=human_ensembl_msigdb_c1 --exptype=continuous --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_P_LVM1_1.csv --species=human_ensembl_msigdb_c1 --exptype=continuous --ebins=9\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM2_4.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c1.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c1/human_ensembl_msigdb_c1.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM2_4.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 326\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t326\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 326\n",
      "cat 52, mi = 0.00266591, chr13q34\n",
      "cat 143, mi = 0.00250248, chr21q22\n",
      "cat 50, mi = 0.00243947, chr13q32\n",
      "cat 95, mi = 0.00234765, chr17q25\n",
      "cat 44, mi = 0.00227233, chr13q12\n",
      "cat 46, mi = 0.00210428, chr13q14\n",
      "cat 229, mi = 0.00190372, chr5q12\n",
      "cat 51, mi = 0.00156263, chr13q33\n",
      "cat 245, mi = 0.00151442, chr6p22\n",
      "cat 127, mi = 0.00133958, chr1q25\n",
      "cat 64, mi = 0.00129270, chr15q11\n",
      "cat 292, mi = 0.00128545, chr8q24\n",
      "cat 69, mi = 0.00122756, chr15q22\n",
      "cat 71, mi = 0.00119936, chr15q24\n",
      "cat 248, mi = 0.00119898, chr6p25\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 52, mi = 0.003, pvalue = 0.000, chr13q34\n",
      "Passed: cat 143, mi = 0.003, pvalue = 0.000, chr21q22\n",
      "Passed: cat 50, mi = 0.002, pvalue = 0.000, chr13q32\n",
      "Passed: cat 95, mi = 0.002, pvalue = 0.000, chr17q25\n",
      "Passed: cat 44, mi = 0.002, pvalue = 0.000, chr13q12\n",
      "Passed: cat 46, mi = 0.002, pvalue = 0.000, chr13q14\n",
      "Passed: cat 229, mi = 0.002, pvalue = 0.000, chr5q12\n",
      "Passed: cat 51, mi = 0.002, pvalue = 0.000, chr13q33\n",
      "Passed: cat 245, mi = 0.002, pvalue = 0.002, chr6p22\n",
      "Didn't pass (cutoff=0.0050): cat 127, mi = 0.001, pvalue = 0.008, chr1q25\n",
      "Passed: cat 64, mi = 0.001, pvalue = 0.002, chr15q11\n",
      "Didn't pass (cutoff=0.0050): cat 292, mi = 0.001, pvalue = 0.006, chr8q24\n",
      "Didn't pass (cutoff=0.0050): cat 69, mi = 0.001, pvalue = 0.011, chr15q22\n",
      "Didn't pass (cutoff=0.0050): cat 71, mi = 0.001, pvalue = 0.016, chr15q24\n",
      "Didn't pass (cutoff=0.0050): cat 248, mi = 0.001, pvalue = 0.018, chr6p25\n",
      "Didn't pass (cutoff=0.0050): cat 139, mi = 0.001, pvalue = 0.012, chr20q13\n",
      "Didn't pass (cutoff=0.0050): cat 67, mi = 0.001, pvalue = 0.017, chr15q15\n",
      "Didn't pass (cutoff=0.0050): cat 204, mi = 0.001, pvalue = 0.023, chr4p15\n",
      "Didn't pass (cutoff=0.0050): cat 252, mi = 0.001, pvalue = 0.017, chr6q14\n",
      "Didn't pass (cutoff=0.0050): cat 274, mi = 0.001, pvalue = 0.022, chr7q31\n",
      "Didn't pass (cutoff=0.0050): cat 32, mi = 0.001, pvalue = 0.018, chr12p13\n",
      "Didn't pass (cutoff=0.0050): cat 66, mi = 0.001, pvalue = 0.026, chr15q14\n",
      "Didn't pass (cutoff=0.0050): cat 146, mi = 0.001, pvalue = 0.021, chr22q12\n",
      "Didn't pass (cutoff=0.0050): cat 211, mi = 0.001, pvalue = 0.014, chr4q22\n",
      "Didn't pass (cutoff=0.0050): cat 156, mi = 0.001, pvalue = 0.028, chr2p22\n",
      "Didn't pass (cutoff=0.0050): cat 168, mi = 0.001, pvalue = 0.036, chr2q24\n",
      "Didn't pass (cutoff=0.0050): cat 0, mi = 0.001, pvalue = 0.026, chr10p11\n",
      "Didn't pass (cutoff=0.0050): cat 163, mi = 0.001, pvalue = 0.038, chr2q13\n",
      "Didn't pass (cutoff=0.0050): cat 124, mi = 0.001, pvalue = 0.033, chr1q22\n",
      "Didn't pass (cutoff=0.0050): cat 162, mi = 0.001, pvalue = 0.034, chr2q12\n",
      "Didn't pass (cutoff=0.0050): cat 129, mi = 0.001, pvalue = 0.028, chr1q32\n",
      "Total number of categories evaluated = 22.\n",
      "Number of categories that passed the tests = 10\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM2_4.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM2_4.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 2, 7, 0.0103211388879434\n",
      "best merge is 0, 2, 0.0361219097532887\n",
      "best merge is 3, 5, 0.0552363566648463\n",
      "best merge is 3, 4, 0.168979328087658\n",
      "best merge is 1, 3, 0.178743807848313\n",
      "best merge is 0, 8, 0.179807091152235\n",
      "order classes\n",
      "3 - 0.0611592284403858\n",
      "4 - 0.290445761370345\n",
      "0 2 7\n",
      "1 3 5 4\n",
      "6\n",
      "8\n",
      "9\n",
      "Done.Start drawing\n",
      "100.30\t0.00\n",
      "chr15q11\n",
      "chr5q12\n",
      "chr21q22\n",
      "chr17q25\n",
      "chr13q14\n",
      "chr13q12\n",
      "chr13q34\n",
      "chr13q32\n",
      "chr13q33\n",
      "chr6p22\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM2_4.csv_PAGE/filt_FC_LVM2_4.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM2_4.csv_PAGE/filt_FC_LVM2_4.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM2_4.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM2_4.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 2, 7, 0.0103211388879434\n",
      "best merge is 0, 2, 0.0361219097532887\n",
      "best merge is 3, 5, 0.0552363566648463\n",
      "best merge is 3, 4, 0.168979328087658\n",
      "best merge is 1, 3, 0.178743807848313\n",
      "best merge is 0, 8, 0.179807091152235\n",
      "order classes\n",
      "3 - 0.0611592284403858\n",
      "4 - 0.290445761370345\n",
      "0 2 7\n",
      "1 3 5 4\n",
      "6\n",
      "8\n",
      "9\n",
      "Done.-19.8911266201396\t42\t0\n",
      "-19.7403788634098\t42\t0.217746759720837\n",
      "-19.6712861415753\t42\t0.301495513459621\n",
      "-19.621036889332\t42\t0.355932203389831\n",
      "-19.5728813559322\t42\t0.401994017946162\n",
      "-19.5163509471585\t42\t0.452243270189432\n",
      "-19.4451645064806\t42\t0.515054835493519\n",
      "-19.3321036889332\t42\t0.594616151545364\n",
      "1.37058823529411\t42\t0.741176470588235\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM2_4.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_P_LVM1_2.csv --species=human_ensembl_msigdb_c1 --exptype=continuous --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_P_LVM1_3.csv --species=human_ensembl_msigdb_c1 --exptype=continuous --ebins=9\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_P_LVM1_1.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c1.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c1/human_ensembl_msigdb_c1.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_P_LVM1_1.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 326\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t326\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 326\n",
      "cat 237, mi = 0.00125315, chr5q32\n",
      "cat 27, mi = 0.00123762, chr11q24\n",
      "cat 164, mi = 0.00122755, chr2q14\n",
      "cat 284, mi = 0.00120474, chr8p23\n",
      "cat 172, mi = 0.00109580, chr2q34\n",
      "cat 286, mi = 0.00109154, chr8q11\n",
      "cat 268, mi = 0.00108394, chr7p21\n",
      "cat 90, mi = 0.00106187, chr17q12\n",
      "cat 174, mi = 0.00104355, chr2q36\n",
      "cat 258, mi = 0.00104241, chr6q24\n",
      "cat 50, mi = 0.00098714, chr13q32\n",
      "cat 119, mi = 0.00097541, chr1p35\n",
      "cat 246, mi = 0.00096919, chr6p23\n",
      "cat 305, mi = 0.00096374, chr9q31\n",
      "cat 257, mi = 0.00093028, chr6q23\n",
      "\n",
      "RESULTS\n",
      "Didn't pass (cutoff=0.0050): cat 237, mi = 0.001, pvalue = 0.009, chr5q32\n",
      "Didn't pass (cutoff=0.0050): cat 27, mi = 0.001, pvalue = 0.015, chr11q24\n",
      "Didn't pass (cutoff=0.0050): cat 164, mi = 0.001, pvalue = 0.012, chr2q14\n",
      "Didn't pass (cutoff=0.0050): cat 284, mi = 0.001, pvalue = 0.015, chr8p23\n",
      "Didn't pass (cutoff=0.0050): cat 172, mi = 0.001, pvalue = 0.018, chr2q34\n",
      "Didn't pass (cutoff=0.0050): cat 286, mi = 0.001, pvalue = 0.017, chr8q11\n",
      "Didn't pass (cutoff=0.0050): cat 268, mi = 0.001, pvalue = 0.028, chr7p21\n",
      "Didn't pass (cutoff=0.0050): cat 90, mi = 0.001, pvalue = 0.034, chr17q12\n",
      "Didn't pass (cutoff=0.0050): cat 174, mi = 0.001, pvalue = 0.041, chr2q36\n",
      "Didn't pass (cutoff=0.0050): cat 258, mi = 0.001, pvalue = 0.042, chr6q24\n",
      "Didn't pass (cutoff=0.0050): cat 50, mi = 0.001, pvalue = 0.053, chr13q32\n",
      "Didn't pass (cutoff=0.0050): cat 119, mi = 0.001, pvalue = 0.045, chr1p35\n",
      "Didn't pass (cutoff=0.0050): cat 246, mi = 0.001, pvalue = 0.062, chr6p23\n",
      "Didn't pass (cutoff=0.0050): cat 305, mi = 0.001, pvalue = 0.067, chr9q31\n",
      "Didn't pass (cutoff=0.0050): cat 257, mi = 0.001, pvalue = 0.086, chr6q23\n",
      "Didn't pass (cutoff=0.0050): cat 155, mi = 0.001, pvalue = 0.082, chr2p21\n",
      "Didn't pass (cutoff=0.0050): cat 100, mi = 0.001, pvalue = 0.080, chr18q21\n",
      "Didn't pass (cutoff=0.0050): cat 32, mi = 0.001, pvalue = 0.064, chr12p13\n",
      "Didn't pass (cutoff=0.0050): cat 31, mi = 0.001, pvalue = 0.092, chr12p12\n",
      "Didn't pass (cutoff=0.0050): cat 73, mi = 0.001, pvalue = 0.077, chr15q26\n",
      "Total number of categories evaluated = 10.\n",
      "Number of categories that passed the tests = 0\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_P_LVM1_1.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_P_LVM1_1.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "280.55\t0.00\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_P_LVM1_1.csv_PAGE/filt_FC_P_LVM1_1.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_P_LVM1_1.csv_PAGE/filt_FC_P_LVM1_1.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_P_LVM1_1.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_P_LVM1_1.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_P_LVM1_1.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM2_5.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c1.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c1/human_ensembl_msigdb_c1.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM2_5.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 326\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t326\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 326\n",
      "cat 44, mi = 0.00363605, chr13q12\n",
      "cat 52, mi = 0.00276407, chr13q34\n",
      "cat 46, mi = 0.00272621, chr13q14\n",
      "cat 89, mi = 0.00263051, chr17q11\n",
      "cat 95, mi = 0.00243270, chr17q25\n",
      "cat 274, mi = 0.00239228, chr7q31\n",
      "cat 50, mi = 0.00232467, chr13q32\n",
      "cat 292, mi = 0.00209071, chr8q24\n",
      "cat 129, mi = 0.00190542, chr1q32\n",
      "cat 208, mi = 0.00156981, chr4q12\n",
      "cat 143, mi = 0.00155228, chr21q22\n",
      "cat 48, mi = 0.00152970, chr13q22\n",
      "cat 71, mi = 0.00149408, chr15q24\n",
      "cat 309, mi = 0.00145176, chrxp11\n",
      "cat 97, mi = 0.00143422, chr18p11\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 44, mi = 0.004, pvalue = 0.000, chr13q12\n",
      "Passed: cat 52, mi = 0.003, pvalue = 0.000, chr13q34\n",
      "Passed: cat 46, mi = 0.003, pvalue = 0.000, chr13q14\n",
      "Passed: cat 89, mi = 0.003, pvalue = 0.000, chr17q11\n",
      "Passed: cat 95, mi = 0.002, pvalue = 0.000, chr17q25\n",
      "Passed: cat 274, mi = 0.002, pvalue = 0.000, chr7q31\n",
      "Passed: cat 50, mi = 0.002, pvalue = 0.000, chr13q32\n",
      "Passed: cat 292, mi = 0.002, pvalue = 0.000, chr8q24\n",
      "Passed: cat 129, mi = 0.002, pvalue = 0.000, chr1q32\n",
      "Passed: cat 208, mi = 0.002, pvalue = 0.001, chr4q12\n",
      "Passed: cat 143, mi = 0.002, pvalue = 0.002, chr21q22\n",
      "Passed: cat 48, mi = 0.002, pvalue = 0.000, chr13q22\n",
      "Passed: cat 71, mi = 0.001, pvalue = 0.003, chr15q24\n",
      "Passed: cat 309, mi = 0.001, pvalue = 0.002, chrxp11\n",
      "Passed: cat 97, mi = 0.001, pvalue = 0.004, chr18p11\n",
      "Passed: cat 284, mi = 0.001, pvalue = 0.003, chr8p23\n",
      "Didn't pass (cutoff=0.0050): cat 104, mi = 0.001, pvalue = 0.005, chr19p12\n",
      "Passed: cat 311, mi = 0.001, pvalue = 0.005, chrxp22\n",
      "Passed: cat 232, mi = 0.001, pvalue = 0.004, chr5q15\n",
      "Didn't pass (cutoff=0.0050): cat 290, mi = 0.001, pvalue = 0.009, chr8q22\n",
      "Didn't pass (cutoff=0.0050): cat 92, mi = 0.001, pvalue = 0.011, chr17q22\n",
      "Didn't pass (cutoff=0.0050): cat 99, mi = 0.001, pvalue = 0.016, chr18q12\n",
      "Didn't pass (cutoff=0.0050): cat 231, mi = 0.001, pvalue = 0.021, chr5q14\n",
      "Didn't pass (cutoff=0.0050): cat 245, mi = 0.001, pvalue = 0.019, chr6p22\n",
      "Didn't pass (cutoff=0.0050): cat 133, mi = 0.001, pvalue = 0.020, chr1q44\n",
      "Didn't pass (cutoff=0.0050): cat 180, mi = 0.001, pvalue = 0.023, chr3p14\n",
      "Didn't pass (cutoff=0.0050): cat 224, mi = 0.001, pvalue = 0.027, chr5p13\n",
      "Didn't pass (cutoff=0.0050): cat 4, mi = 0.001, pvalue = 0.024, chr10p15\n",
      "Didn't pass (cutoff=0.0050): cat 51, mi = 0.001, pvalue = 0.007, chr13q33\n",
      "Didn't pass (cutoff=0.0050): cat 25, mi = 0.001, pvalue = 0.035, chr11q22\n",
      "Didn't pass (cutoff=0.0050): cat 142, mi = 0.001, pvalue = 0.022, chr21q21\n",
      "Didn't pass (cutoff=0.0050): cat 11, mi = 0.001, pvalue = 0.036, chr10q25\n",
      "Didn't pass (cutoff=0.0050): cat 294, mi = 0.001, pvalue = 0.029, chr9p11\n",
      "Didn't pass (cutoff=0.0050): cat 45, mi = 0.001, pvalue = 0.047, chr13q13\n",
      "Didn't pass (cutoff=0.0050): cat 82, mi = 0.001, pvalue = 0.035, chr16q22\n",
      "Didn't pass (cutoff=0.0050): cat 147, mi = 0.001, pvalue = 0.040, chr22q13\n",
      "Didn't pass (cutoff=0.0050): cat 305, mi = 0.001, pvalue = 0.059, chr9q31\n",
      "Didn't pass (cutoff=0.0050): cat 277, mi = 0.001, pvalue = 0.073, chr7q34\n",
      "Didn't pass (cutoff=0.0050): cat 248, mi = 0.001, pvalue = 0.074, chr6p25\n",
      "Total number of categories evaluated = 30.\n",
      "Number of categories that passed the tests = 18\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM2_5.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM2_5.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 0, 2, 0.0387592645442691\n",
      "best merge is 1, 6, 0.0674803154542827\n",
      "best merge is 0, 3, 0.0848726317872789\n",
      "best merge is 4, 16, 0.092561783962661\n",
      "best merge is 1, 17, 0.096851273440647\n",
      "best merge is 7, 10, 0.103481736533385\n",
      "best merge is 8, 9, 0.107210146947197\n",
      "best merge is 4, 7, 0.126614269405364\n",
      "best merge is 5, 11, 0.186889149558049\n",
      "best merge is 0, 4, 0.190269718277648\n",
      "best merge is 0, 13, 0.262797322680228\n",
      "best merge is 1, 5, 0.300783343597795\n",
      "best merge is 8, 12, 0.328976259324303\n",
      "best merge is 0, 1, 0.415011114647472\n",
      "order classes\n",
      "8 - 0.353940679991932\n",
      "5 - 0.648798357162354\n",
      "3 - 1.02021688746107\n",
      "0 2 3 4 16 7 10 13\n",
      "1 6 17 5 11\n",
      "8 9 12\n",
      "14\n",
      "15\n",
      "Done.Start drawing\n",
      "66.19\t0.00\n",
      "chr1q32\n",
      "chr4q12\n",
      "chr15q24\n",
      "chr18p11\n",
      "chr8p23\n",
      "chr8q24\n",
      "chr21q22\n",
      "chr13q12\n",
      "chr13q14\n",
      "chr17q11\n",
      "chr17q25\n",
      "chrxp22\n",
      "chrxp11\n",
      "chr13q34\n",
      "chr13q32\n",
      "chr5q15\n",
      "chr7q31\n",
      "chr13q22\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM2_5.csv_PAGE/filt_FC_LVM2_5.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM2_5.csv_PAGE/filt_FC_LVM2_5.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM2_5.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM2_5.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 0, 2, 0.0387592645442691\n",
      "best merge is 1, 6, 0.0674803154542827\n",
      "best merge is 0, 3, 0.0848726317872789\n",
      "best merge is 4, 16, 0.092561783962661\n",
      "best merge is 1, 17, 0.096851273440647\n",
      "best merge is 7, 10, 0.103481736533385\n",
      "best merge is 8, 9, 0.107210146947197\n",
      "best merge is 4, 7, 0.126614269405364\n",
      "best merge is 5, 11, 0.186889149558049\n",
      "best merge is 0, 4, 0.190269718277648\n",
      "best merge is 0, 13, 0.262797322680228\n",
      "best merge is 1, 5, 0.300783343597795\n",
      "best merge is 8, 12, 0.328976259324303\n",
      "best merge is 0, 1, 0.415011114647472\n",
      "order classes\n",
      "8 - 0.353940679991932\n",
      "5 - 0.648798357162354\n",
      "3 - 1.02021688746107\n",
      "0 2 3 4 16 7 10 13\n",
      "1 6 17 5 11\n",
      "8 9 12\n",
      "14\n",
      "15\n",
      "Done.-19.8477111346125\t42\t0\n",
      "-19.6224505212268\t42\t0.304577730775042\n",
      "-19.501888502795\t42\t0.450521226771416\n",
      "-19.4130533313189\t42\t0.545701767638616\n",
      "-19.330563529234\t42\t0.628191569723523\n",
      "-19.2480737271491\t42\t0.71068137180843\n",
      "-19.1528931862819\t42\t0.793171173893337\n",
      "-19.0196404290678\t42\t0.901042453542831\n",
      "1.52983834416075\t42\t1.0596766883215\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_LVM2_5.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_P_LVM1_4.csv --species=human_ensembl_msigdb_c1 --exptype=continuous --ebins=9\n",
      "done\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_P_LVM1_3.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c1.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c1/human_ensembl_msigdb_c1.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_P_LVM1_3.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 326\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t326\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 326\n",
      "cat 119, mi = 0.00170937, chr1p35\n",
      "cat 245, mi = 0.00142185, chr6p22\n",
      "cat 118, mi = 0.00137618, chr1p34\n",
      "cat 90, mi = 0.00135788, chr17q12\n",
      "cat 199, mi = 0.00134252, chr3q29\n",
      "cat 231, mi = 0.00128270, chr5q14\n",
      "cat 210, mi = 0.00121416, chr4q21\n",
      "cat 315, mi = 0.00118966, chrxq13\n",
      "cat 161, mi = 0.00109522, chr2q11\n",
      "cat 89, mi = 0.00104209, chr17q11\n",
      "cat 197, mi = 0.00103842, chr3q27\n",
      "cat 279, mi = 0.00103784, chr7q36\n",
      "cat 180, mi = 0.00103358, chr3p14\n",
      "cat 306, mi = 0.00103305, chr9q32\n",
      "cat 156, mi = 0.00102264, chr2p22\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 119, mi = 0.002, pvalue = 0.001, chr1p35\n",
      "Didn't pass (cutoff=0.0050): cat 245, mi = 0.001, pvalue = 0.005, chr6p22\n",
      "Passed: cat 118, mi = 0.001, pvalue = 0.004, chr1p34\n",
      "Didn't pass (cutoff=0.0050): cat 90, mi = 0.001, pvalue = 0.006, chr17q12\n",
      "Didn't pass (cutoff=0.0050): cat 199, mi = 0.001, pvalue = 0.009, chr3q29\n",
      "Didn't pass (cutoff=0.0050): cat 231, mi = 0.001, pvalue = 0.013, chr5q14\n",
      "Didn't pass (cutoff=0.0050): cat 210, mi = 0.001, pvalue = 0.013, chr4q21\n",
      "Didn't pass (cutoff=0.0050): cat 315, mi = 0.001, pvalue = 0.017, chrxq13\n",
      "Didn't pass (cutoff=0.0050): cat 161, mi = 0.001, pvalue = 0.029, chr2q11\n",
      "Didn't pass (cutoff=0.0050): cat 89, mi = 0.001, pvalue = 0.030, chr17q11\n",
      "Didn't pass (cutoff=0.0050): cat 197, mi = 0.001, pvalue = 0.049, chr3q27\n",
      "Didn't pass (cutoff=0.0050): cat 279, mi = 0.001, pvalue = 0.039, chr7q36\n",
      "Didn't pass (cutoff=0.0050): cat 180, mi = 0.001, pvalue = 0.049, chr3p14\n",
      "Didn't pass (cutoff=0.0050): cat 306, mi = 0.001, pvalue = 0.047, chr9q32\n",
      "Didn't pass (cutoff=0.0050): cat 156, mi = 0.001, pvalue = 0.046, chr2p22\n",
      "Didn't pass (cutoff=0.0050): cat 185, mi = 0.001, pvalue = 0.042, chr3p25\n",
      "Didn't pass (cutoff=0.0050): cat 232, mi = 0.001, pvalue = 0.049, chr5q15\n",
      "Didn't pass (cutoff=0.0050): cat 294, mi = 0.001, pvalue = 0.062, chr9p11\n",
      "Didn't pass (cutoff=0.0050): cat 215, mi = 0.001, pvalue = 0.035, chr4q26\n",
      "Didn't pass (cutoff=0.0050): cat 314, mi = 0.001, pvalue = 0.044, chrxq12\n",
      "Didn't pass (cutoff=0.0050): cat 212, mi = 0.001, pvalue = 0.056, chr4q23\n",
      "Didn't pass (cutoff=0.0050): cat 56, mi = 0.001, pvalue = 0.076, chr14q13\n",
      "Didn't pass (cutoff=0.0050): cat 81, mi = 0.001, pvalue = 0.074, chr16q21\n",
      "Total number of categories evaluated = 13.\n",
      "Number of categories that passed the tests = 2\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_P_LVM1_3.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_P_LVM1_3.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "11.42\t0.00\n",
      "chr1p35\n",
      "chr1p34\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_P_LVM1_3.csv_PAGE/filt_FC_P_LVM1_3.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_P_LVM1_3.csv_PAGE/filt_FC_P_LVM1_3.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_P_LVM1_3.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_P_LVM1_3.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-19.1725043782837\t42\t0\n",
      "-18.1059544658494\t42\t1.65499124343257\n",
      "-17.7381786339755\t42\t2.13309982486865\n",
      "-17.4807355516637\t42\t2.39054290718039\n",
      "-17.223292469352\t42\t2.64798598949212\n",
      "-16.9658493870403\t42\t2.90542907180385\n",
      "-16.6348511383538\t42\t3.16287215411559\n",
      "-16.0831873905429\t42\t3.56742556917688\n",
      "3.13309982486865\t42\t4.2661996497373\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_P_LVM1_3.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_P_LVM1_4.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c1.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c1/human_ensembl_msigdb_c1.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_P_LVM1_4.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 326\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t326\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 326\n",
      "cat 7, mi = 0.00123358, chr10q21\n",
      "cat 64, mi = 0.00122198, chr15q11\n",
      "cat 290, mi = 0.00119332, chr8q22\n",
      "cat 134, mi = 0.00117485, chr20p11\n",
      "cat 286, mi = 0.00112375, chr8q11\n",
      "cat 48, mi = 0.00109154, chr13q22\n",
      "cat 168, mi = 0.00108657, chr2q24\n",
      "cat 97, mi = 0.00104782, chr18p11\n",
      "cat 189, mi = 0.00102793, chr3q12\n",
      "cat 174, mi = 0.00101125, chr2q36\n",
      "cat 265, mi = 0.00099535, chr7p13\n",
      "cat 39, mi = 0.00098731, chr12q21\n",
      "cat 143, mi = 0.00098538, chr21q22\n",
      "cat 43, mi = 0.00095800, chr13q11\n",
      "cat 61, mi = 0.00095014, chr14q31\n",
      "\n",
      "RESULTS\n",
      "Didn't pass (cutoff=0.0050): cat 7, mi = 0.001, pvalue = 0.013, chr10q21\n",
      "Passed: cat 64, mi = 0.001, pvalue = 0.002, chr15q11\n",
      "Didn't pass (cutoff=0.0050): cat 290, mi = 0.001, pvalue = 0.015, chr8q22\n",
      "Didn't pass (cutoff=0.0050): cat 134, mi = 0.001, pvalue = 0.020, chr20p11\n",
      "Didn't pass (cutoff=0.0050): cat 286, mi = 0.001, pvalue = 0.013, chr8q11\n",
      "Didn't pass (cutoff=0.0050): cat 48, mi = 0.001, pvalue = 0.018, chr13q22\n",
      "Didn't pass (cutoff=0.0050): cat 168, mi = 0.001, pvalue = 0.039, chr2q24\n",
      "Didn't pass (cutoff=0.0050): cat 97, mi = 0.001, pvalue = 0.033, chr18p11\n",
      "Didn't pass (cutoff=0.0050): cat 189, mi = 0.001, pvalue = 0.048, chr3q12\n",
      "Didn't pass (cutoff=0.0050): cat 174, mi = 0.001, pvalue = 0.056, chr2q36\n",
      "Didn't pass (cutoff=0.0050): cat 265, mi = 0.001, pvalue = 0.062, chr7p13\n",
      "Didn't pass (cutoff=0.0050): cat 39, mi = 0.001, pvalue = 0.056, chr12q21\n",
      "Didn't pass (cutoff=0.0050): cat 143, mi = 0.001, pvalue = 0.041, chr21q22\n",
      "Didn't pass (cutoff=0.0050): cat 43, mi = 0.001, pvalue = 0.069, chr13q11\n",
      "Didn't pass (cutoff=0.0050): cat 61, mi = 0.001, pvalue = 0.079, chr14q31\n",
      "Didn't pass (cutoff=0.0050): cat 89, mi = 0.001, pvalue = 0.050, chr17q11\n",
      "Didn't pass (cutoff=0.0050): cat 145, mi = 0.001, pvalue = 0.050, chr22q11\n",
      "Didn't pass (cutoff=0.0050): cat 139, mi = 0.001, pvalue = 0.049, chr20q13\n",
      "Didn't pass (cutoff=0.0050): cat 245, mi = 0.001, pvalue = 0.066, chr6p22\n",
      "Didn't pass (cutoff=0.0050): cat 215, mi = 0.001, pvalue = 0.071, chr4q26\n",
      "Didn't pass (cutoff=0.0050): cat 248, mi = 0.001, pvalue = 0.084, chr6p25\n",
      "Didn't pass (cutoff=0.0050): cat 277, mi = 0.001, pvalue = 0.087, chr7q34\n",
      "Total number of categories evaluated = 12.\n",
      "Number of categories that passed the tests = 1\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_P_LVM1_4.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_P_LVM1_4.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "110.21\t0.00\n",
      "chr15q11\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_P_LVM1_4.csv_PAGE/filt_FC_P_LVM1_4.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_P_LVM1_4.csv_PAGE/filt_FC_P_LVM1_4.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_P_LVM1_4.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_P_LVM1_4.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-19.9980945467743\t42\t0\n",
      "-19.9923781870974\t42\t0.00381090645132021\n",
      "-19.9847563741947\t42\t0.0114327193539606\n",
      "-19.9790400145177\t42\t0.019054532256601\n",
      "-19.9733236548408\t42\t0.0228654387079212\n",
      "-19.9676072951638\t42\t0.0304872516105617\n",
      "-19.8361310225932\t42\t0.0342981580618819\n",
      "-19.4912439887488\t42\t0.293439796751656\n",
      "1.36203611287542\t42\t0.724072225750839\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_P_LVM1_4.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_P_LVM1_2.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c1.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c1/human_ensembl_msigdb_c1.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_P_LVM1_2.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 326\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t326\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 326\n",
      "cat 245, mi = 0.00185783, chr6p22\n",
      "cat 208, mi = 0.00119122, chr4q12\n",
      "cat 169, mi = 0.00115951, chr2q31\n",
      "cat 305, mi = 0.00111594, chr9q31\n",
      "cat 129, mi = 0.00110745, chr1q32\n",
      "cat 274, mi = 0.00109015, chr7q31\n",
      "cat 182, mi = 0.00107239, chr3p22\n",
      "cat 9, mi = 0.00107183, chr10q23\n",
      "cat 99, mi = 0.00106658, chr18q12\n",
      "cat 321, mi = 0.00104543, chrxq26\n",
      "cat 174, mi = 0.00104391, chr2q36\n",
      "cat 90, mi = 0.00102122, chr17q12\n",
      "cat 292, mi = 0.00101427, chr8q24\n",
      "cat 50, mi = 0.00098678, chr13q32\n",
      "cat 175, mi = 0.00097922, chr2q37\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 245, mi = 0.002, pvalue = 0.000, chr6p22\n",
      "Didn't pass (cutoff=0.0050): cat 208, mi = 0.001, pvalue = 0.016, chr4q12\n",
      "Didn't pass (cutoff=0.0050): cat 169, mi = 0.001, pvalue = 0.018, chr2q31\n",
      "Didn't pass (cutoff=0.0050): cat 305, mi = 0.001, pvalue = 0.028, chr9q31\n",
      "Didn't pass (cutoff=0.0050): cat 129, mi = 0.001, pvalue = 0.019, chr1q32\n",
      "Didn't pass (cutoff=0.0050): cat 274, mi = 0.001, pvalue = 0.026, chr7q31\n",
      "Didn't pass (cutoff=0.0050): cat 182, mi = 0.001, pvalue = 0.038, chr3p22\n",
      "Didn't pass (cutoff=0.0050): cat 9, mi = 0.001, pvalue = 0.031, chr10q23\n",
      "Didn't pass (cutoff=0.0050): cat 99, mi = 0.001, pvalue = 0.037, chr18q12\n",
      "Didn't pass (cutoff=0.0050): cat 321, mi = 0.001, pvalue = 0.039, chrxq26\n",
      "Didn't pass (cutoff=0.0050): cat 174, mi = 0.001, pvalue = 0.039, chr2q36\n",
      "Didn't pass (cutoff=0.0050): cat 90, mi = 0.001, pvalue = 0.045, chr17q12\n",
      "Didn't pass (cutoff=0.0050): cat 292, mi = 0.001, pvalue = 0.040, chr8q24\n",
      "Didn't pass (cutoff=0.0050): cat 50, mi = 0.001, pvalue = 0.055, chr13q32\n",
      "Didn't pass (cutoff=0.0050): cat 175, mi = 0.001, pvalue = 0.046, chr2q37\n",
      "Didn't pass (cutoff=0.0050): cat 298, mi = 0.001, pvalue = 0.061, chr9p22\n",
      "Didn't pass (cutoff=0.0050): cat 2, mi = 0.001, pvalue = 0.073, chr10p13\n",
      "Didn't pass (cutoff=0.0050): cat 286, mi = 0.001, pvalue = 0.069, chr8q11\n",
      "Didn't pass (cutoff=0.0050): cat 265, mi = 0.001, pvalue = 0.072, chr7p13\n",
      "Didn't pass (cutoff=0.0050): cat 149, mi = 0.001, pvalue = 0.066, chr2p11\n",
      "Didn't pass (cutoff=0.0050): cat 20, mi = 0.001, pvalue = 0.019, chr11q11\n",
      "Total number of categories evaluated = 11.\n",
      "Number of categories that passed the tests = 1\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_P_LVM1_2.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_P_LVM1_2.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "5.29\t0.00\n",
      "chr6p22\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_P_LVM1_2.csv_PAGE/filt_FC_P_LVM1_2.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_P_LVM1_2.csv_PAGE/filt_FC_P_LVM1_2.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_P_LVM1_2.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_P_LVM1_2.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-16.8638941398866\t42\t0\n",
      "-13.2117202268431\t42\t6.27221172022684\n",
      "-12.3780718336484\t42\t7.30434782608696\n",
      "-11.8223062381853\t42\t7.93950850661626\n",
      "-11.3856332703214\t42\t8.41587901701323\n",
      "-10.9092627599244\t42\t8.81285444234405\n",
      "-10.3137996219282\t42\t9.36862003780718\n",
      "-9.44045368620038\t42\t10.0037807183365\n",
      "6.55765595463138\t42\t11.1153119092628\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c1/fc_values/filt_FC_P_LVM1_2.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "sem --wait done\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "nothing to output for html page at /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl line 79.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "nohup sh run.sh iPAGE_JY.sh &"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
