{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# This is used for running data vs human_msigdb_h annotations located in: \n",
    "### /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_h"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import pickle\n",
    "import seaborn as sns\n",
    "import scipy.stats\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### get index of file names to loop through "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "cd fc_values\n",
    "ls FC* > filelist.txt\n",
    "cd .."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### get ensembl genes represented in iPAGE database"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash \n",
    "\n",
    "cp /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_h/human_ensembl_msigdb_h_index.txt human_ensembl_index.txt\n",
    "awk '{print $1}' human_ensembl_index.txt > ensembl_index.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### make sure that genes in file match the index "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "###################!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n",
    "###################REMEMBER TO SPECIFY INDEX USED IN THE IPAGE COMMAND!!!!!!!!\n",
    "###################!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n",
    "\n",
    "\n",
    "#####################\n",
    "#######read in data from logfc calculations\n",
    "#####################\n",
    "####define index file to loop through\n",
    "index_file = \"fc_values/filelist.txt\"\n",
    "####define matching ensg index\n",
    "GO_index_filename = \"ensembl_index.txt\"\n",
    "genes_with_GOs = []\n",
    "with open(GO_index_filename,'r') as rf:\n",
    "    for line in rf:\n",
    "        gene_name = line.split('\\n')[0]\n",
    "        genes_with_GOs.append(gene_name)\n",
    "genes_with_GOs_set = set(genes_with_GOs)\n",
    "####open file and loop through each line\n",
    "###\n",
    "iPAGE_sh = \"iPAGE_JY.sh\"\n",
    "cwd = os.getcwd()+\"/\"\n",
    "os.system(\"rm \"+iPAGE_sh)\n",
    "###\n",
    "with open(index_file,'r') as rf:\n",
    "    for line in rf:\n",
    "        #####do this to each file\n",
    "        ###read in file into pandas df\n",
    "        FC_file = \"fc_values/\"+line.split('\\n')[0]\n",
    "        FC_df = pd.read_csv(FC_file,sep=',',index_col=0)\n",
    "        FC_df.columns = ['log2FoldChange']\n",
    "        FC_df.index.names = ['Gene']\n",
    "        ###filter\n",
    "        FC_df = FC_df.loc[[x for x in FC_df.index if x in genes_with_GOs_set]]\n",
    "        ###save\n",
    "        save_file = \"fc_values/filt_\"+line.split('\\n')[0]\n",
    "        FC_df.to_csv(save_file,sep='\\t',header=True)\n",
    "        ###write final iPAGE command to an ipage .sh file\n",
    "        iPAGE_cmd = \"perl /nvme/bins/iPAGEv1.0/page.pl --expfile \"+cwd+save_file+\" --species=human_ensembl_msigdb_h --exptype=continuous --ebins=9 \\n\"\n",
    "        with open(iPAGE_sh, \"a\") as myfile:\n",
    "            myfile.write(iPAGE_cmd)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## then run the command from this folder:\n",
    "\n",
    "nohup sh run.sh iPAGE_JY.sh &"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM1_1.csv --species=human_ensembl_msigdb_h --exptype=continuous --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM1_2.csv --species=human_ensembl_msigdb_h --exptype=continuous --ebins=9\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM1_1.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_h.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_h/human_ensembl_msigdb_h.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM1_1.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 50\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t50\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 50\n",
      "cat 31, mi = 0.00402943, HALLMARK_MYC_TARGETS_V1\n",
      "cat 35, mi = 0.00364003, HALLMARK_OXIDATIVE_PHOSPHORYLATION\n",
      "cat 21, mi = 0.00316602, HALLMARK_HYPOXIA\n",
      "cat 28, mi = 0.00307932, HALLMARK_KRAS_SIGNALING_UP\n",
      "cat 14, mi = 0.00294816, HALLMARK_ESTROGEN_RESPONSE_EARLY\n",
      "cat 13, mi = 0.00254662, HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION\n",
      "cat 15, mi = 0.00221399, HALLMARK_ESTROGEN_RESPONSE_LATE\n",
      "cat 12, mi = 0.00198122, HALLMARK_E2F_TARGETS\n",
      "cat 22, mi = 0.00150992, HALLMARK_IL2_STAT5_SIGNALING\n",
      "cat 27, mi = 0.00136738, HALLMARK_KRAS_SIGNALING_DN\n",
      "cat 3, mi = 0.00128901, HALLMARK_ANGIOGENESIS\n",
      "cat 45, mi = 0.00120339, HALLMARK_UNFOLDED_PROTEIN_RESPONSE\n",
      "cat 4, mi = 0.00112512, HALLMARK_APICAL_JUNCTION\n",
      "cat 33, mi = 0.00111890, HALLMARK_MYOGENESIS\n",
      "cat 25, mi = 0.00098352, HALLMARK_INTERFERON_ALPHA_RESPONSE\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 31, mi = 0.004, pvalue = 0.000, HALLMARK_MYC_TARGETS_V1\n",
      "Passed: cat 35, mi = 0.004, pvalue = 0.000, HALLMARK_OXIDATIVE_PHOSPHORYLATION\n",
      "Passed: cat 21, mi = 0.003, pvalue = 0.000, HALLMARK_HYPOXIA\n",
      "Passed: cat 27, mi = 0.001, pvalue = 0.004, HALLMARK_KRAS_SIGNALING_DN\n",
      "Didn't pass (cutoff=0.0050): cat 25, mi = 0.001, pvalue = 0.041, HALLMARK_INTERFERON_ALPHA_RESPONSE\n",
      "Didn't pass (cutoff=0.0050): cat 40, mi = 0.001, pvalue = 0.079, HALLMARK_PROTEIN_SECRETION\n",
      "Didn't pass (cutoff=0.0050): cat 37, mi = 0.000, pvalue = 0.578, HALLMARK_PANCREAS_BETA_CELLS\n",
      "Number of categories that passed the tests = 4\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM1_1.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM1_1.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 0, 1, 0.090057818570299\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "Done.Start drawing\n",
      "23.77\t0.00\n",
      "HALLMARK_HYPOXIA\n",
      "HALLMARK_MYC_TARGETS_V1\n",
      "HALLMARK_OXIDATIVE_PHOSPHORYLATION\n",
      "HALLMARK_KRAS_SIGNALING_DN\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM1_1.csv_PAGE/filt_FC_LVM1_1.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM1_1.csv_PAGE/filt_FC_LVM1_1.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM1_1.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM1_1.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 0, 1, 0.090057818570299\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "Done.-19.2932267564156\t42\t0\n",
      "-18.4450988641144\t42\t1.4135464871687\n",
      "-18.2153975599495\t42\t1.69625578460244\n",
      "-18.056373580143\t42\t1.87294909549853\n",
      "-17.8973496003366\t42\t2.0143037442154\n",
      "-17.7206562894405\t42\t2.19099705511149\n",
      "-17.51745898191\t42\t2.36769036600757\n",
      "-17.2082456878418\t42\t2.59739167017249\n",
      "2.49305847707194\t42\t2.98611695414388\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM1_1.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM1_3.csv --species=human_ensembl_msigdb_h --exptype=continuous --ebins=9\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM1_2.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_h.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_h/human_ensembl_msigdb_h.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM1_2.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 50\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t50\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 50\n",
      "cat 17, mi = 0.00859296, HALLMARK_G2M_CHECKPOINT\n",
      "cat 35, mi = 0.00727555, HALLMARK_OXIDATIVE_PHOSPHORYLATION\n",
      "cat 31, mi = 0.00675726, HALLMARK_MYC_TARGETS_V1\n",
      "cat 29, mi = 0.00555860, HALLMARK_MITOTIC_SPINDLE\n",
      "cat 12, mi = 0.00414096, HALLMARK_E2F_TARGETS\n",
      "cat 13, mi = 0.00281477, HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION\n",
      "cat 30, mi = 0.00264392, HALLMARK_MTORC1_SIGNALING\n",
      "cat 28, mi = 0.00208604, HALLMARK_KRAS_SIGNALING_UP\n",
      "cat 21, mi = 0.00203456, HALLMARK_HYPOXIA\n",
      "cat 32, mi = 0.00187087, HALLMARK_MYC_TARGETS_V2\n",
      "cat 22, mi = 0.00178002, HALLMARK_IL2_STAT5_SIGNALING\n",
      "cat 44, mi = 0.00171004, HALLMARK_TNFA_SIGNALING_VIA_NFKB\n",
      "cat 14, mi = 0.00157140, HALLMARK_ESTROGEN_RESPONSE_EARLY\n",
      "cat 15, mi = 0.00145578, HALLMARK_ESTROGEN_RESPONSE_LATE\n",
      "cat 11, mi = 0.00143991, HALLMARK_DNA_REPAIR\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 17, mi = 0.009, pvalue = 0.000, HALLMARK_G2M_CHECKPOINT\n",
      "Passed: cat 35, mi = 0.007, pvalue = 0.000, HALLMARK_OXIDATIVE_PHOSPHORYLATION\n",
      "Passed: cat 13, mi = 0.003, pvalue = 0.000, HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION\n",
      "Passed: cat 30, mi = 0.003, pvalue = 0.000, HALLMARK_MTORC1_SIGNALING\n",
      "Passed: cat 11, mi = 0.001, pvalue = 0.003, HALLMARK_DNA_REPAIR\n",
      "Didn't pass (cutoff=0.0050): cat 27, mi = 0.001, pvalue = 0.010, HALLMARK_KRAS_SIGNALING_DN\n",
      "Didn't pass (cutoff=0.0050): cat 37, mi = 0.001, pvalue = 0.250, HALLMARK_PANCREAS_BETA_CELLS\n",
      "Number of categories that passed the tests = 5\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM1_2.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM1_2.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 0, 2, 0.144634987105933\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "4\n",
      "Done.Start drawing\n",
      "62.96\t0.00\n",
      "HALLMARK_G2M_CHECKPOINT\n",
      "HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION\n",
      "HALLMARK_DNA_REPAIR\n",
      "HALLMARK_OXIDATIVE_PHOSPHORYLATION\n",
      "HALLMARK_MTORC1_SIGNALING\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM1_2.csv_PAGE/filt_FC_LVM1_2.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM1_2.csv_PAGE/filt_FC_LVM1_2.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM1_2.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM1_2.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 0, 2, 0.144634987105933\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "4\n",
      "Done.-19.7198221092757\t42\t0\n",
      "-19.3562579415502\t42\t0.560355781448539\n",
      "-19.2195044472681\t42\t0.72712833545108\n",
      "-19.1227763659466\t42\t0.833862770012706\n",
      "-19.0360546378653\t42\t0.920584498094028\n",
      "-18.9459974587039\t42\t1.00730622617535\n",
      "-18.8359275730623\t42\t1.10069885641677\n",
      "-18.6558132147395\t42\t1.2274459974587\n",
      "1.73046378653113\t42\t1.46092757306226\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM1_2.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM1_4.csv --species=human_ensembl_msigdb_h --exptype=continuous --ebins=9\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM1_3.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_h.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_h/human_ensembl_msigdb_h.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM1_3.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 50\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t50\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 50\n",
      "cat 12, mi = 0.01451505, HALLMARK_E2F_TARGETS\n",
      "cat 17, mi = 0.00929300, HALLMARK_G2M_CHECKPOINT\n",
      "cat 31, mi = 0.00645276, HALLMARK_MYC_TARGETS_V1\n",
      "cat 35, mi = 0.00488353, HALLMARK_OXIDATIVE_PHOSPHORYLATION\n",
      "cat 13, mi = 0.00412606, HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION\n",
      "cat 21, mi = 0.00328139, HALLMARK_HYPOXIA\n",
      "cat 15, mi = 0.00260575, HALLMARK_ESTROGEN_RESPONSE_LATE\n",
      "cat 9, mi = 0.00218566, HALLMARK_COAGULATION\n",
      "cat 22, mi = 0.00216648, HALLMARK_IL2_STAT5_SIGNALING\n",
      "cat 29, mi = 0.00183171, HALLMARK_MITOTIC_SPINDLE\n",
      "cat 42, mi = 0.00175704, HALLMARK_SPERMATOGENESIS\n",
      "cat 24, mi = 0.00168529, HALLMARK_INFLAMMATORY_RESPONSE\n",
      "cat 10, mi = 0.00152896, HALLMARK_COMPLEMENT\n",
      "cat 44, mi = 0.00149798, HALLMARK_TNFA_SIGNALING_VIA_NFKB\n",
      "cat 43, mi = 0.00138563, HALLMARK_TGF_BETA_SIGNALING\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 12, mi = 0.015, pvalue = 0.000, HALLMARK_E2F_TARGETS\n",
      "Passed: cat 35, mi = 0.005, pvalue = 0.000, HALLMARK_OXIDATIVE_PHOSPHORYLATION\n",
      "Passed: cat 13, mi = 0.004, pvalue = 0.000, HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION\n",
      "Passed: cat 15, mi = 0.003, pvalue = 0.000, HALLMARK_ESTROGEN_RESPONSE_LATE\n",
      "Didn't pass (cutoff=0.0050): cat 27, mi = 0.001, pvalue = 0.006, HALLMARK_KRAS_SIGNALING_DN\n",
      "Didn't pass (cutoff=0.0050): cat 19, mi = 0.001, pvalue = 0.013, HALLMARK_HEDGEHOG_SIGNALING\n",
      "Didn't pass (cutoff=0.0050): cat 40, mi = 0.001, pvalue = 0.009, HALLMARK_PROTEIN_SECRETION\n",
      "Didn't pass (cutoff=0.0050): cat 25, mi = 0.001, pvalue = 0.025, HALLMARK_INTERFERON_ALPHA_RESPONSE\n",
      "Didn't pass (cutoff=0.0050): cat 26, mi = 0.001, pvalue = 0.056, HALLMARK_INTERFERON_GAMMA_RESPONSE\n",
      "Didn't pass (cutoff=0.0050): cat 37, mi = 0.001, pvalue = 0.142, HALLMARK_PANCREAS_BETA_CELLS\n",
      "Didn't pass (cutoff=0.0050): cat 20, mi = 0.001, pvalue = 0.139, HALLMARK_HEME_METABOLISM\n",
      "Number of categories that passed the tests = 4\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM1_3.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM1_3.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 0, 3, 0.199551099985471\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "Done.Start drawing\n",
      "84.94\t0.00\n",
      "HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION\n",
      "HALLMARK_OXIDATIVE_PHOSPHORYLATION\n",
      "HALLMARK_E2F_TARGETS\n",
      "HALLMARK_ESTROGEN_RESPONSE_LATE\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM1_3.csv_PAGE/filt_FC_LVM1_3.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM1_3.csv_PAGE/filt_FC_LVM1_3.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM1_3.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM1_3.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 0, 3, 0.199551099985471\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "Done.-19.849187661879\t42\t0\n",
      "-19.6588179891688\t42\t0.301624676242053\n",
      "-19.5920649870497\t42\t0.380739345420297\n",
      "-19.5401459854015\t42\t0.435130680480339\n",
      "-19.4931716505769\t42\t0.484577348716741\n",
      "-19.4437249823405\t42\t0.529079350129503\n",
      "-19.3769719802213\t42\t0.583470685189545\n",
      "-19.2706616435131\t42\t0.662585354367789\n",
      "1.39804567930304\t42\t0.796091358606075\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM1_3.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM2_1.csv --species=human_ensembl_msigdb_h --exptype=continuous --ebins=9\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM1_4.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_h.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_h/human_ensembl_msigdb_h.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM1_4.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 50\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t50\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 50\n",
      "cat 13, mi = 0.00395799, HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION\n",
      "cat 31, mi = 0.00388809, HALLMARK_MYC_TARGETS_V1\n",
      "cat 21, mi = 0.00346416, HALLMARK_HYPOXIA\n",
      "cat 35, mi = 0.00324975, HALLMARK_OXIDATIVE_PHOSPHORYLATION\n",
      "cat 44, mi = 0.00253590, HALLMARK_TNFA_SIGNALING_VIA_NFKB\n",
      "cat 30, mi = 0.00239465, HALLMARK_MTORC1_SIGNALING\n",
      "cat 45, mi = 0.00234484, HALLMARK_UNFOLDED_PROTEIN_RESPONSE\n",
      "cat 11, mi = 0.00207305, HALLMARK_DNA_REPAIR\n",
      "cat 14, mi = 0.00175443, HALLMARK_ESTROGEN_RESPONSE_EARLY\n",
      "cat 43, mi = 0.00164592, HALLMARK_TGF_BETA_SIGNALING\n",
      "cat 17, mi = 0.00154606, HALLMARK_G2M_CHECKPOINT\n",
      "cat 27, mi = 0.00153445, HALLMARK_KRAS_SIGNALING_DN\n",
      "cat 40, mi = 0.00141978, HALLMARK_PROTEIN_SECRETION\n",
      "cat 22, mi = 0.00127819, HALLMARK_IL2_STAT5_SIGNALING\n",
      "cat 24, mi = 0.00125099, HALLMARK_INFLAMMATORY_RESPONSE\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 13, mi = 0.004, pvalue = 0.000, HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION\n",
      "Passed: cat 31, mi = 0.004, pvalue = 0.000, HALLMARK_MYC_TARGETS_V1\n",
      "Passed: cat 14, mi = 0.002, pvalue = 0.000, HALLMARK_ESTROGEN_RESPONSE_EARLY\n",
      "Passed: cat 27, mi = 0.002, pvalue = 0.001, HALLMARK_KRAS_SIGNALING_DN\n",
      "Passed: cat 40, mi = 0.001, pvalue = 0.004, HALLMARK_PROTEIN_SECRETION\n",
      "Didn't pass (cutoff=0.0050): cat 16, mi = 0.001, pvalue = 0.018, HALLMARK_FATTY_ACID_METABOLISM\n",
      "Didn't pass (cutoff=0.0050): cat 41, mi = 0.001, pvalue = 0.082, HALLMARK_REACTIVE_OXIGEN_SPECIES_PATHWAY\n",
      "Didn't pass (cutoff=0.0050): cat 20, mi = 0.001, pvalue = 0.082, HALLMARK_HEME_METABOLISM\n",
      "Didn't pass (cutoff=0.0050): cat 37, mi = 0.001, pvalue = 0.402, HALLMARK_PANCREAS_BETA_CELLS\n",
      "Didn't pass (cutoff=0.0050): cat 26, mi = 0.000, pvalue = 0.671, HALLMARK_INTERFERON_GAMMA_RESPONSE\n",
      "Number of categories that passed the tests = 5\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM1_4.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM1_4.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 2, 3, 0.0707572637932846\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "4\n",
      "Done.Start drawing\n",
      "26.31\t0.00\n",
      "HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION\n",
      "HALLMARK_MYC_TARGETS_V1\n",
      "HALLMARK_PROTEIN_SECRETION\n",
      "HALLMARK_ESTROGEN_RESPONSE_EARLY\n",
      "HALLMARK_KRAS_SIGNALING_DN\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM1_4.csv_PAGE/filt_FC_LVM1_4.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM1_4.csv_PAGE/filt_FC_LVM1_4.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM1_4.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM1_4.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 2, 3, 0.0707572637932846\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "4\n",
      "Done.-19.5290763968073\t42\t0\n",
      "-18.9064994298746\t42\t0.941847206385405\n",
      "-18.6590649942987\t42\t1.24515393386545\n",
      "-18.4754846066135\t42\t1.43671607753706\n",
      "-18.3158494868871\t42\t1.61231470923603\n",
      "-18.1641961231471\t42\t1.75598631698974\n",
      "-17.9806157354618\t42\t1.91562143671608\n",
      "-17.709236031927\t42\t2.12314709236032\n",
      "2.22919042189282\t42\t2.45838084378563\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM1_4.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM2_2.csv --species=human_ensembl_msigdb_h --exptype=continuous --ebins=9\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM2_1.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_h.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_h/human_ensembl_msigdb_h.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM2_1.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 50\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t50\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 50\n",
      "cat 35, mi = 0.00579370, HALLMARK_OXIDATIVE_PHOSPHORYLATION\n",
      "cat 31, mi = 0.00576601, HALLMARK_MYC_TARGETS_V1\n",
      "cat 13, mi = 0.00354515, HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION\n",
      "cat 21, mi = 0.00311282, HALLMARK_HYPOXIA\n",
      "cat 15, mi = 0.00284614, HALLMARK_ESTROGEN_RESPONSE_LATE\n",
      "cat 14, mi = 0.00228339, HALLMARK_ESTROGEN_RESPONSE_EARLY\n",
      "cat 28, mi = 0.00219582, HALLMARK_KRAS_SIGNALING_UP\n",
      "cat 11, mi = 0.00204633, HALLMARK_DNA_REPAIR\n",
      "cat 27, mi = 0.00185440, HALLMARK_KRAS_SIGNALING_DN\n",
      "cat 33, mi = 0.00183134, HALLMARK_MYOGENESIS\n",
      "cat 19, mi = 0.00168289, HALLMARK_HEDGEHOG_SIGNALING\n",
      "cat 22, mi = 0.00168092, HALLMARK_IL2_STAT5_SIGNALING\n",
      "cat 42, mi = 0.00166832, HALLMARK_SPERMATOGENESIS\n",
      "cat 4, mi = 0.00135876, HALLMARK_APICAL_JUNCTION\n",
      "cat 24, mi = 0.00134864, HALLMARK_INFLAMMATORY_RESPONSE\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 35, mi = 0.006, pvalue = 0.000, HALLMARK_OXIDATIVE_PHOSPHORYLATION\n",
      "Passed: cat 31, mi = 0.006, pvalue = 0.000, HALLMARK_MYC_TARGETS_V1\n",
      "Passed: cat 13, mi = 0.004, pvalue = 0.000, HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION\n",
      "Passed: cat 15, mi = 0.003, pvalue = 0.000, HALLMARK_ESTROGEN_RESPONSE_LATE\n",
      "Passed: cat 27, mi = 0.002, pvalue = 0.000, HALLMARK_KRAS_SIGNALING_DN\n",
      "Passed: cat 19, mi = 0.002, pvalue = 0.001, HALLMARK_HEDGEHOG_SIGNALING\n",
      "Passed: cat 42, mi = 0.002, pvalue = 0.001, HALLMARK_SPERMATOGENESIS\n",
      "Didn't pass (cutoff=0.0050): cat 26, mi = 0.001, pvalue = 0.082, HALLMARK_INTERFERON_GAMMA_RESPONSE\n",
      "Didn't pass (cutoff=0.0050): cat 40, mi = 0.001, pvalue = 0.117, HALLMARK_PROTEIN_SECRETION\n",
      "Number of categories that passed the tests = 7\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM2_1.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM2_1.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 0, 1, 0.174509925218522\n",
      "best merge is 2, 5, 0.183257065402233\n",
      "best merge is 3, 4, 0.263893441316918\n",
      "order classes\n",
      "2 - 0.305392369132414\n",
      "2 - 0.320699864453908\n",
      "0 1\n",
      "2 5\n",
      "3\n",
      "4\n",
      "6\n",
      "Done.Start drawing\n",
      "70.09\t0.00\n",
      "HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION\n",
      "HALLMARK_HEDGEHOG_SIGNALING\n",
      "HALLMARK_OXIDATIVE_PHOSPHORYLATION\n",
      "HALLMARK_MYC_TARGETS_V1\n",
      "HALLMARK_ESTROGEN_RESPONSE_LATE\n",
      "HALLMARK_KRAS_SIGNALING_DN\n",
      "HALLMARK_SPERMATOGENESIS\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM2_1.csv_PAGE/filt_FC_LVM2_1.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM2_1.csv_PAGE/filt_FC_LVM2_1.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM2_1.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM2_1.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 0, 1, 0.174509925218522\n",
      "best merge is 2, 5, 0.183257065402233\n",
      "best merge is 3, 4, 0.263893441316918\n",
      "order classes\n",
      "2 - 0.305392369132414\n",
      "2 - 0.320699864453908\n",
      "0 1\n",
      "2 5\n",
      "3\n",
      "4\n",
      "6\n",
      "Done.-19.784277357683\t42\t0\n",
      "-19.5056356113568\t42\t0.431445284634042\n",
      "-19.4037665858182\t42\t0.557283492652304\n",
      "-19.3318590383792\t42\t0.635183335711228\n",
      "-19.2689399343701\t42\t0.701098587530318\n",
      "-19.203024682551\t42\t0.761021543729491\n",
      "-19.1221286916821\t42\t0.832929091168498\n",
      "-18.9902981880439\t42\t0.922813525467256\n",
      "1.54829504922243\t42\t1.09659009844486\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM2_1.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM2_3.csv --species=human_ensembl_msigdb_h --exptype=continuous --ebins=9\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM2_2.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_h.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_h/human_ensembl_msigdb_h.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM2_2.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 50\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t50\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 50\n",
      "cat 35, mi = 0.00568825, HALLMARK_OXIDATIVE_PHOSPHORYLATION\n",
      "cat 31, mi = 0.00409662, HALLMARK_MYC_TARGETS_V1\n",
      "cat 12, mi = 0.00330719, HALLMARK_E2F_TARGETS\n",
      "cat 14, mi = 0.00281423, HALLMARK_ESTROGEN_RESPONSE_EARLY\n",
      "cat 28, mi = 0.00264682, HALLMARK_KRAS_SIGNALING_UP\n",
      "cat 21, mi = 0.00263852, HALLMARK_HYPOXIA\n",
      "cat 27, mi = 0.00263243, HALLMARK_KRAS_SIGNALING_DN\n",
      "cat 13, mi = 0.00254945, HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION\n",
      "cat 17, mi = 0.00254541, HALLMARK_G2M_CHECKPOINT\n",
      "cat 15, mi = 0.00233855, HALLMARK_ESTROGEN_RESPONSE_LATE\n",
      "cat 22, mi = 0.00166764, HALLMARK_IL2_STAT5_SIGNALING\n",
      "cat 24, mi = 0.00144452, HALLMARK_INFLAMMATORY_RESPONSE\n",
      "cat 39, mi = 0.00136224, HALLMARK_PI3K_AKT_MTOR_SIGNALING\n",
      "cat 33, mi = 0.00122202, HALLMARK_MYOGENESIS\n",
      "cat 29, mi = 0.00120373, HALLMARK_MITOTIC_SPINDLE\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 35, mi = 0.006, pvalue = 0.000, HALLMARK_OXIDATIVE_PHOSPHORYLATION\n",
      "Passed: cat 31, mi = 0.004, pvalue = 0.000, HALLMARK_MYC_TARGETS_V1\n",
      "Passed: cat 14, mi = 0.003, pvalue = 0.000, HALLMARK_ESTROGEN_RESPONSE_EARLY\n",
      "Passed: cat 28, mi = 0.003, pvalue = 0.000, HALLMARK_KRAS_SIGNALING_UP\n",
      "Passed: cat 27, mi = 0.003, pvalue = 0.000, HALLMARK_KRAS_SIGNALING_DN\n",
      "Passed: cat 39, mi = 0.001, pvalue = 0.004, HALLMARK_PI3K_AKT_MTOR_SIGNALING\n",
      "Didn't pass (cutoff=0.0050): cat 20, mi = 0.001, pvalue = 0.055, HALLMARK_HEME_METABOLISM\n",
      "Didn't pass (cutoff=0.0050): cat 19, mi = 0.001, pvalue = 0.088, HALLMARK_HEDGEHOG_SIGNALING\n",
      "Didn't pass (cutoff=0.0050): cat 34, mi = 0.001, pvalue = 0.154, HALLMARK_NOTCH_SIGNALING\n",
      "Didn't pass (cutoff=0.0050): cat 37, mi = 0.000, pvalue = 0.558, HALLMARK_PANCREAS_BETA_CELLS\n",
      "Number of categories that passed the tests = 6\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM2_2.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM2_2.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 2, 4, 0.0428248126622739\n",
      "best merge is 2, 3, 0.0601115255477456\n",
      "order classes\n",
      "2 - 0.0642372189934108\n",
      "2 4\n",
      "1\n",
      "0\n",
      "3\n",
      "5\n",
      "Done.Start drawing\n",
      "101.03\t0.00\n",
      "HALLMARK_PI3K_AKT_MTOR_SIGNALING\n",
      "HALLMARK_OXIDATIVE_PHOSPHORYLATION\n",
      "HALLMARK_MYC_TARGETS_V1\n",
      "HALLMARK_ESTROGEN_RESPONSE_EARLY\n",
      "HALLMARK_KRAS_SIGNALING_DN\n",
      "HALLMARK_KRAS_SIGNALING_UP\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM2_2.csv_PAGE/filt_FC_LVM2_2.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM2_2.csv_PAGE/filt_FC_LVM2_2.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM2_2.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM2_2.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 2, 4, 0.0428248126622739\n",
      "best merge is 2, 3, 0.0601115255477456\n",
      "order classes\n",
      "2 - 0.0642372189934108\n",
      "2 4\n",
      "1\n",
      "0\n",
      "3\n",
      "5\n",
      "Done.-19.8544986637632\t42\t0\n",
      "-19.6570325645848\t42\t0.291002672473523\n",
      "-19.5759675343957\t42\t0.394932198356924\n",
      "-19.5219241809364\t42\t0.453132732851628\n",
      "-19.4720380085123\t42\t0.503018905275661\n",
      "-19.417994655053\t42\t0.552905077699693\n",
      "-19.3514797584876\t42\t0.611105612194398\n",
      "-19.2454716420865\t42\t0.685934870830446\n",
      "1.41156092249827\t42\t0.823121844996536\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM2_2.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM2_4.csv --species=human_ensembl_msigdb_h --exptype=continuous --ebins=9\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM2_3.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_h.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_h/human_ensembl_msigdb_h.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM2_3.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 50\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t50\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 50\n",
      "cat 17, mi = 0.00993515, HALLMARK_G2M_CHECKPOINT\n",
      "cat 35, mi = 0.00645828, HALLMARK_OXIDATIVE_PHOSPHORYLATION\n",
      "cat 31, mi = 0.00617122, HALLMARK_MYC_TARGETS_V1\n",
      "cat 29, mi = 0.00598755, HALLMARK_MITOTIC_SPINDLE\n",
      "cat 12, mi = 0.00481975, HALLMARK_E2F_TARGETS\n",
      "cat 13, mi = 0.00331614, HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION\n",
      "cat 28, mi = 0.00260967, HALLMARK_KRAS_SIGNALING_UP\n",
      "cat 11, mi = 0.00244020, HALLMARK_DNA_REPAIR\n",
      "cat 32, mi = 0.00234564, HALLMARK_MYC_TARGETS_V2\n",
      "cat 21, mi = 0.00202155, HALLMARK_HYPOXIA\n",
      "cat 27, mi = 0.00194724, HALLMARK_KRAS_SIGNALING_DN\n",
      "cat 4, mi = 0.00176981, HALLMARK_APICAL_JUNCTION\n",
      "cat 15, mi = 0.00174878, HALLMARK_ESTROGEN_RESPONSE_LATE\n",
      "cat 14, mi = 0.00158778, HALLMARK_ESTROGEN_RESPONSE_EARLY\n",
      "cat 39, mi = 0.00134164, HALLMARK_PI3K_AKT_MTOR_SIGNALING\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 17, mi = 0.010, pvalue = 0.000, HALLMARK_G2M_CHECKPOINT\n",
      "Passed: cat 35, mi = 0.006, pvalue = 0.000, HALLMARK_OXIDATIVE_PHOSPHORYLATION\n",
      "Passed: cat 13, mi = 0.003, pvalue = 0.000, HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION\n",
      "Passed: cat 11, mi = 0.002, pvalue = 0.000, HALLMARK_DNA_REPAIR\n",
      "Passed: cat 27, mi = 0.002, pvalue = 0.000, HALLMARK_KRAS_SIGNALING_DN\n",
      "Passed: cat 15, mi = 0.002, pvalue = 0.000, HALLMARK_ESTROGEN_RESPONSE_LATE\n",
      "Passed: cat 39, mi = 0.001, pvalue = 0.003, HALLMARK_PI3K_AKT_MTOR_SIGNALING\n",
      "Didn't pass (cutoff=0.0050): cat 25, mi = 0.001, pvalue = 0.006, HALLMARK_INTERFERON_ALPHA_RESPONSE\n",
      "Didn't pass (cutoff=0.0050): cat 20, mi = 0.001, pvalue = 0.086, HALLMARK_HEME_METABOLISM\n",
      "Didn't pass (cutoff=0.0050): cat 48, mi = 0.001, pvalue = 0.381, HALLMARK_WNT_BETA_CATENIN_SIGNALING\n",
      "Didn't pass (cutoff=0.0050): cat 37, mi = 0.000, pvalue = 0.878, HALLMARK_PANCREAS_BETA_CELLS\n",
      "Number of categories that passed the tests = 7\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM2_3.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM2_3.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 0, 2, 0.0566657163570179\n",
      "best merge is 4, 5, 0.0854781911279826\n",
      "best merge is 1, 3, 0.266205201646975\n",
      "order classes\n",
      "2 - 0.0991650036247813\n",
      "2 - 0.14958683447397\n",
      "0 2\n",
      "1\n",
      "3\n",
      "4 5\n",
      "6\n",
      "Done.Start drawing\n",
      "69.14\t0.00\n",
      "HALLMARK_G2M_CHECKPOINT\n",
      "HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION\n",
      "HALLMARK_PI3K_AKT_MTOR_SIGNALING\n",
      "HALLMARK_DNA_REPAIR\n",
      "HALLMARK_OXIDATIVE_PHOSPHORYLATION\n",
      "HALLMARK_KRAS_SIGNALING_DN\n",
      "HALLMARK_ESTROGEN_RESPONSE_LATE\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM2_3.csv_PAGE/filt_FC_LVM2_3.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM2_3.csv_PAGE/filt_FC_LVM2_3.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM2_3.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM2_3.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 0, 2, 0.0566657163570179\n",
      "best merge is 4, 5, 0.0854781911279826\n",
      "best merge is 1, 3, 0.266205201646975\n",
      "order classes\n",
      "2 - 0.0991650036247813\n",
      "2 - 0.14958683447397\n",
      "0 2\n",
      "1\n",
      "3\n",
      "4 5\n",
      "6\n",
      "Done.-19.7843505929997\t42\t0\n",
      "-19.4715070870697\t42\t0.431298814000579\n",
      "-19.3166039919005\t42\t0.625687011859994\n",
      "-19.2133352617877\t42\t0.741105004339022\n",
      "-19.1191784784495\t42\t0.832224472085623\n",
      "-19.0189470639283\t42\t0.929418571015331\n",
      "-18.8974544402661\t42\t1.03268730112815\n",
      "-18.7000289268152\t42\t1.1724038183396\n",
      "1.71376916401504\t42\t1.42753832803008\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM2_3.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM2_5.csv --species=human_ensembl_msigdb_h --exptype=continuous --ebins=9\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM2_4.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_h.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_h/human_ensembl_msigdb_h.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM2_4.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 50\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t50\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 50\n",
      "cat 12, mi = 0.01327034, HALLMARK_E2F_TARGETS\n",
      "cat 17, mi = 0.00797554, HALLMARK_G2M_CHECKPOINT\n",
      "cat 31, mi = 0.00722078, HALLMARK_MYC_TARGETS_V1\n",
      "cat 35, mi = 0.00629390, HALLMARK_OXIDATIVE_PHOSPHORYLATION\n",
      "cat 13, mi = 0.00283553, HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION\n",
      "cat 21, mi = 0.00270241, HALLMARK_HYPOXIA\n",
      "cat 15, mi = 0.00231761, HALLMARK_ESTROGEN_RESPONSE_LATE\n",
      "cat 9, mi = 0.00223698, HALLMARK_COAGULATION\n",
      "cat 29, mi = 0.00213774, HALLMARK_MITOTIC_SPINDLE\n",
      "cat 14, mi = 0.00211595, HALLMARK_ESTROGEN_RESPONSE_EARLY\n",
      "cat 32, mi = 0.00188744, HALLMARK_MYC_TARGETS_V2\n",
      "cat 22, mi = 0.00154069, HALLMARK_IL2_STAT5_SIGNALING\n",
      "cat 24, mi = 0.00132321, HALLMARK_INFLAMMATORY_RESPONSE\n",
      "cat 44, mi = 0.00129493, HALLMARK_TNFA_SIGNALING_VIA_NFKB\n",
      "cat 27, mi = 0.00127936, HALLMARK_KRAS_SIGNALING_DN\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 12, mi = 0.013, pvalue = 0.000, HALLMARK_E2F_TARGETS\n",
      "Passed: cat 35, mi = 0.006, pvalue = 0.000, HALLMARK_OXIDATIVE_PHOSPHORYLATION\n",
      "Passed: cat 13, mi = 0.003, pvalue = 0.000, HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION\n",
      "Passed: cat 15, mi = 0.002, pvalue = 0.000, HALLMARK_ESTROGEN_RESPONSE_LATE\n",
      "Didn't pass (cutoff=0.0050): cat 27, mi = 0.001, pvalue = 0.007, HALLMARK_KRAS_SIGNALING_DN\n",
      "Didn't pass (cutoff=0.0050): cat 19, mi = 0.001, pvalue = 0.011, HALLMARK_HEDGEHOG_SIGNALING\n",
      "Didn't pass (cutoff=0.0050): cat 40, mi = 0.001, pvalue = 0.013, HALLMARK_PROTEIN_SECRETION\n",
      "Didn't pass (cutoff=0.0050): cat 25, mi = 0.001, pvalue = 0.043, HALLMARK_INTERFERON_ALPHA_RESPONSE\n",
      "Didn't pass (cutoff=0.0050): cat 20, mi = 0.001, pvalue = 0.044, HALLMARK_HEME_METABOLISM\n",
      "Didn't pass (cutoff=0.0050): cat 5, mi = 0.001, pvalue = 0.144, HALLMARK_APICAL_SURFACE\n",
      "Didn't pass (cutoff=0.0050): cat 37, mi = 0.001, pvalue = 0.181, HALLMARK_PANCREAS_BETA_CELLS\n",
      "Didn't pass (cutoff=0.0050): cat 26, mi = 0.001, pvalue = 0.161, HALLMARK_INTERFERON_GAMMA_RESPONSE\n",
      "Didn't pass (cutoff=0.0050): cat 48, mi = 0.001, pvalue = 0.196, HALLMARK_WNT_BETA_CATENIN_SIGNALING\n",
      "Number of categories that passed the tests = 4\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM2_4.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM2_4.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 0, 3, 0.0776273099173319\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "Done.Start drawing\n",
      "100.30\t0.00\n",
      "HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION\n",
      "HALLMARK_OXIDATIVE_PHOSPHORYLATION\n",
      "HALLMARK_E2F_TARGETS\n",
      "HALLMARK_ESTROGEN_RESPONSE_LATE\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM2_4.csv_PAGE/filt_FC_LVM2_4.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM2_4.csv_PAGE/filt_FC_LVM2_4.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM2_4.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM2_4.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 0, 3, 0.0776273099173319\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "Done.-19.8911266201396\t42\t0\n",
      "-19.7403788634098\t42\t0.217746759720837\n",
      "-19.6712861415753\t42\t0.301495513459621\n",
      "-19.621036889332\t42\t0.355932203389831\n",
      "-19.5728813559322\t42\t0.401994017946162\n",
      "-19.5163509471585\t42\t0.452243270189432\n",
      "-19.4451645064806\t42\t0.515054835493519\n",
      "-19.3321036889332\t42\t0.594616151545364\n",
      "1.37058823529411\t42\t0.741176470588235\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM2_4.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_P_LVM1_1.csv --species=human_ensembl_msigdb_h --exptype=continuous --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_P_LVM1_2.csv --species=human_ensembl_msigdb_h --exptype=continuous --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_P_LVM1_3.csv --species=human_ensembl_msigdb_h --exptype=continuous --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_P_LVM1_4.csv --species=human_ensembl_msigdb_h --exptype=continuous --ebins=9\n",
      "done\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM2_5.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_h.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_h/human_ensembl_msigdb_h.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM2_5.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 50\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t50\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 50\n",
      "cat 31, mi = 0.00486083, HALLMARK_MYC_TARGETS_V1\n",
      "cat 35, mi = 0.00427599, HALLMARK_OXIDATIVE_PHOSPHORYLATION\n",
      "cat 21, mi = 0.00356645, HALLMARK_HYPOXIA\n",
      "cat 13, mi = 0.00310161, HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION\n",
      "cat 14, mi = 0.00275426, HALLMARK_ESTROGEN_RESPONSE_EARLY\n",
      "cat 28, mi = 0.00263490, HALLMARK_KRAS_SIGNALING_UP\n",
      "cat 15, mi = 0.00231143, HALLMARK_ESTROGEN_RESPONSE_LATE\n",
      "cat 11, mi = 0.00230817, HALLMARK_DNA_REPAIR\n",
      "cat 30, mi = 0.00226446, HALLMARK_MTORC1_SIGNALING\n",
      "cat 27, mi = 0.00218641, HALLMARK_KRAS_SIGNALING_DN\n",
      "cat 17, mi = 0.00208046, HALLMARK_G2M_CHECKPOINT\n",
      "cat 9, mi = 0.00178802, HALLMARK_COAGULATION\n",
      "cat 19, mi = 0.00174904, HALLMARK_HEDGEHOG_SIGNALING\n",
      "cat 32, mi = 0.00169142, HALLMARK_MYC_TARGETS_V2\n",
      "cat 12, mi = 0.00163933, HALLMARK_E2F_TARGETS\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 31, mi = 0.005, pvalue = 0.000, HALLMARK_MYC_TARGETS_V1\n",
      "Passed: cat 35, mi = 0.004, pvalue = 0.000, HALLMARK_OXIDATIVE_PHOSPHORYLATION\n",
      "Passed: cat 21, mi = 0.004, pvalue = 0.000, HALLMARK_HYPOXIA\n",
      "Passed: cat 11, mi = 0.002, pvalue = 0.000, HALLMARK_DNA_REPAIR\n",
      "Passed: cat 27, mi = 0.002, pvalue = 0.000, HALLMARK_KRAS_SIGNALING_DN\n",
      "Didn't pass (cutoff=0.0050): cat 40, mi = 0.001, pvalue = 0.066, HALLMARK_PROTEIN_SECRETION\n",
      "Didn't pass (cutoff=0.0050): cat 43, mi = 0.001, pvalue = 0.122, HALLMARK_TGF_BETA_SIGNALING\n",
      "Didn't pass (cutoff=0.0050): cat 37, mi = 0.000, pvalue = 0.735, HALLMARK_PANCREAS_BETA_CELLS\n",
      "Number of categories that passed the tests = 5\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM2_5.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM2_5.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 0, 1, 0.0812669693707017\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "4\n",
      "Done.Start drawing\n",
      "66.19\t0.00\n",
      "HALLMARK_HYPOXIA\n",
      "HALLMARK_MYC_TARGETS_V1\n",
      "HALLMARK_OXIDATIVE_PHOSPHORYLATION\n",
      "HALLMARK_DNA_REPAIR\n",
      "HALLMARK_KRAS_SIGNALING_DN\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM2_5.csv_PAGE/filt_FC_LVM2_5.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM2_5.csv_PAGE/filt_FC_LVM2_5.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM2_5.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM2_5.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 0, 1, 0.0812669693707017\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "4\n",
      "Done.-19.8477111346125\t42\t0\n",
      "-19.6224505212268\t42\t0.304577730775042\n",
      "-19.501888502795\t42\t0.450521226771416\n",
      "-19.4130533313189\t42\t0.545701767638616\n",
      "-19.330563529234\t42\t0.628191569723523\n",
      "-19.2480737271491\t42\t0.71068137180843\n",
      "-19.1528931862819\t42\t0.793171173893337\n",
      "-19.0196404290678\t42\t0.901042453542831\n",
      "1.52983834416075\t42\t1.0596766883215\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_LVM2_5.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_P_LVM1_3.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_h.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_h/human_ensembl_msigdb_h.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_P_LVM1_3.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 50\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t50\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 50\n",
      "cat 12, mi = 0.02671951, HALLMARK_E2F_TARGETS\n",
      "cat 17, mi = 0.01720703, HALLMARK_G2M_CHECKPOINT\n",
      "cat 31, mi = 0.00634820, HALLMARK_MYC_TARGETS_V1\n",
      "cat 35, mi = 0.00353193, HALLMARK_OXIDATIVE_PHOSPHORYLATION\n",
      "cat 42, mi = 0.00253083, HALLMARK_SPERMATOGENESIS\n",
      "cat 9, mi = 0.00223391, HALLMARK_COAGULATION\n",
      "cat 29, mi = 0.00211882, HALLMARK_MITOTIC_SPINDLE\n",
      "cat 20, mi = 0.00206074, HALLMARK_HEME_METABOLISM\n",
      "cat 28, mi = 0.00197040, HALLMARK_KRAS_SIGNALING_UP\n",
      "cat 32, mi = 0.00182222, HALLMARK_MYC_TARGETS_V2\n",
      "cat 30, mi = 0.00171181, HALLMARK_MTORC1_SIGNALING\n",
      "cat 10, mi = 0.00166513, HALLMARK_COMPLEMENT\n",
      "cat 26, mi = 0.00137783, HALLMARK_INTERFERON_GAMMA_RESPONSE\n",
      "cat 25, mi = 0.00136006, HALLMARK_INTERFERON_ALPHA_RESPONSE\n",
      "cat 44, mi = 0.00133881, HALLMARK_TNFA_SIGNALING_VIA_NFKB\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 12, mi = 0.027, pvalue = 0.000, HALLMARK_E2F_TARGETS\n",
      "Passed: cat 35, mi = 0.004, pvalue = 0.000, HALLMARK_OXIDATIVE_PHOSPHORYLATION\n",
      "Passed: cat 9, mi = 0.002, pvalue = 0.000, HALLMARK_COAGULATION\n",
      "Passed: cat 20, mi = 0.002, pvalue = 0.000, HALLMARK_HEME_METABOLISM\n",
      "Passed: cat 26, mi = 0.001, pvalue = 0.004, HALLMARK_INTERFERON_GAMMA_RESPONSE\n",
      "Didn't pass (cutoff=0.0050): cat 27, mi = 0.001, pvalue = 0.008, HALLMARK_KRAS_SIGNALING_DN\n",
      "Didn't pass (cutoff=0.0050): cat 40, mi = 0.001, pvalue = 0.020, HALLMARK_PROTEIN_SECRETION\n",
      "Didn't pass (cutoff=0.0050): cat 7, mi = 0.001, pvalue = 0.042, HALLMARK_BILE_ACID_METABOLISM\n",
      "Didn't pass (cutoff=0.0050): cat 34, mi = 0.001, pvalue = 0.134, HALLMARK_NOTCH_SIGNALING\n",
      "Didn't pass (cutoff=0.0050): cat 37, mi = 0.001, pvalue = 0.307, HALLMARK_PANCREAS_BETA_CELLS\n",
      "Didn't pass (cutoff=0.0050): cat 48, mi = 0.001, pvalue = 0.313, HALLMARK_WNT_BETA_CATENIN_SIGNALING\n",
      "Number of categories that passed the tests = 5\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_P_LVM1_3.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_P_LVM1_3.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 2, 4, 0.220733534843699\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "4\n",
      "Done.Start drawing\n",
      "11.42\t0.00\n",
      "HALLMARK_COAGULATION\n",
      "HALLMARK_INTERFERON_GAMMA_RESPONSE\n",
      "HALLMARK_HEME_METABOLISM\n",
      "HALLMARK_OXIDATIVE_PHOSPHORYLATION\n",
      "HALLMARK_E2F_TARGETS\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_P_LVM1_3.csv_PAGE/filt_FC_P_LVM1_3.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_P_LVM1_3.csv_PAGE/filt_FC_P_LVM1_3.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_P_LVM1_3.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_P_LVM1_3.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 2, 4, 0.220733534843699\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "4\n",
      "Done.-19.1725043782837\t42\t0\n",
      "-18.1059544658494\t42\t1.65499124343257\n",
      "-17.7381786339755\t42\t2.13309982486865\n",
      "-17.4807355516637\t42\t2.39054290718039\n",
      "-17.223292469352\t42\t2.64798598949212\n",
      "-16.9658493870403\t42\t2.90542907180385\n",
      "-16.6348511383538\t42\t3.16287215411559\n",
      "-16.0831873905429\t42\t3.56742556917688\n",
      "3.13309982486865\t42\t4.2661996497373\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_P_LVM1_3.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_P_LVM1_1.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_h.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_h/human_ensembl_msigdb_h.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_P_LVM1_1.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 50\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t50\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 50\n",
      "cat 31, mi = 0.01594744, HALLMARK_MYC_TARGETS_V1\n",
      "cat 35, mi = 0.00712526, HALLMARK_OXIDATIVE_PHOSPHORYLATION\n",
      "cat 12, mi = 0.00626337, HALLMARK_E2F_TARGETS\n",
      "cat 17, mi = 0.00375703, HALLMARK_G2M_CHECKPOINT\n",
      "cat 30, mi = 0.00314134, HALLMARK_MTORC1_SIGNALING\n",
      "cat 11, mi = 0.00260539, HALLMARK_DNA_REPAIR\n",
      "cat 45, mi = 0.00217824, HALLMARK_UNFOLDED_PROTEIN_RESPONSE\n",
      "cat 46, mi = 0.00164600, HALLMARK_UV_RESPONSE_DN\n",
      "cat 2, mi = 0.00131286, HALLMARK_ANDROGEN_RESPONSE\n",
      "cat 16, mi = 0.00120573, HALLMARK_FATTY_ACID_METABOLISM\n",
      "cat 0, mi = 0.00119267, HALLMARK_ADIPOGENESIS\n",
      "cat 32, mi = 0.00111981, HALLMARK_MYC_TARGETS_V2\n",
      "cat 14, mi = 0.00108704, HALLMARK_ESTROGEN_RESPONSE_EARLY\n",
      "cat 47, mi = 0.00100546, HALLMARK_UV_RESPONSE_UP\n",
      "cat 24, mi = 0.00098405, HALLMARK_INFLAMMATORY_RESPONSE\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 31, mi = 0.016, pvalue = 0.000, HALLMARK_MYC_TARGETS_V1\n",
      "Passed: cat 35, mi = 0.007, pvalue = 0.000, HALLMARK_OXIDATIVE_PHOSPHORYLATION\n",
      "Passed: cat 11, mi = 0.003, pvalue = 0.000, HALLMARK_DNA_REPAIR\n",
      "Passed: cat 46, mi = 0.002, pvalue = 0.001, HALLMARK_UV_RESPONSE_DN\n",
      "Didn't pass (cutoff=0.0050): cat 42, mi = 0.001, pvalue = 0.054, HALLMARK_SPERMATOGENESIS\n",
      "Didn't pass (cutoff=0.0050): cat 40, mi = 0.001, pvalue = 0.118, HALLMARK_PROTEIN_SECRETION\n",
      "Didn't pass (cutoff=0.0050): cat 25, mi = 0.001, pvalue = 0.172, HALLMARK_INTERFERON_ALPHA_RESPONSE\n",
      "Didn't pass (cutoff=0.0050): cat 23, mi = 0.001, pvalue = 0.198, HALLMARK_IL6_JAK_STAT3_SIGNALING\n",
      "Didn't pass (cutoff=0.0050): cat 26, mi = 0.001, pvalue = 0.216, HALLMARK_INTERFERON_GAMMA_RESPONSE\n",
      "Didn't pass (cutoff=0.0050): cat 15, mi = 0.001, pvalue = 0.367, HALLMARK_ESTROGEN_RESPONSE_LATE\n",
      "Didn't pass (cutoff=0.0050): cat 5, mi = 0.000, pvalue = 0.642, HALLMARK_APICAL_SURFACE\n",
      "Didn't pass (cutoff=0.0050): cat 37, mi = 0.000, pvalue = 0.710, HALLMARK_PANCREAS_BETA_CELLS\n",
      "Number of categories that passed the tests = 4\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_P_LVM1_1.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_P_LVM1_1.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 0, 1, 0.00347218802716864\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "Done.Start drawing\n",
      "280.55\t0.00\n",
      "HALLMARK_MYC_TARGETS_V1\n",
      "HALLMARK_OXIDATIVE_PHOSPHORYLATION\n",
      "HALLMARK_DNA_REPAIR\n",
      "HALLMARK_UV_RESPONSE_DN\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_P_LVM1_1.csv_PAGE/filt_FC_P_LVM1_1.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_P_LVM1_1.csv_PAGE/filt_FC_P_LVM1_1.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_P_LVM1_1.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_P_LVM1_1.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 0, 1, 0.00347218802716864\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "Done.-19.9985029406523\t42\t0\n",
      "-19.9962573516307\t42\t0.00299411869541971\n",
      "-19.9940117626092\t42\t0.00449117804312957\n",
      "-19.9917661735876\t42\t0.00748529673854928\n",
      "-19.989520584566\t42\t0.00898235608625913\n",
      "-19.9872749955445\t42\t0.0119764747816788\n",
      "-19.9169132062021\t42\t0.0134735341293887\n",
      "-19.7559793263233\t42\t0.152700053466405\n",
      "1.1676706469435\t42\t0.335341293887008\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_P_LVM1_1.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_P_LVM1_2.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_h.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_h/human_ensembl_msigdb_h.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_P_LVM1_2.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 50\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t50\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 50\n",
      "cat 17, mi = 0.01210989, HALLMARK_G2M_CHECKPOINT\n",
      "cat 35, mi = 0.00911778, HALLMARK_OXIDATIVE_PHOSPHORYLATION\n",
      "cat 29, mi = 0.00662460, HALLMARK_MITOTIC_SPINDLE\n",
      "cat 12, mi = 0.00648252, HALLMARK_E2F_TARGETS\n",
      "cat 32, mi = 0.00334699, HALLMARK_MYC_TARGETS_V2\n",
      "cat 31, mi = 0.00320227, HALLMARK_MYC_TARGETS_V1\n",
      "cat 42, mi = 0.00265366, HALLMARK_SPERMATOGENESIS\n",
      "cat 16, mi = 0.00194287, HALLMARK_FATTY_ACID_METABOLISM\n",
      "cat 27, mi = 0.00179634, HALLMARK_KRAS_SIGNALING_DN\n",
      "cat 23, mi = 0.00177970, HALLMARK_IL6_JAK_STAT3_SIGNALING\n",
      "cat 30, mi = 0.00137250, HALLMARK_MTORC1_SIGNALING\n",
      "cat 4, mi = 0.00128332, HALLMARK_APICAL_JUNCTION\n",
      "cat 15, mi = 0.00118795, HALLMARK_ESTROGEN_RESPONSE_LATE\n",
      "cat 43, mi = 0.00118230, HALLMARK_TGF_BETA_SIGNALING\n",
      "cat 20, mi = 0.00117882, HALLMARK_HEME_METABOLISM\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 17, mi = 0.012, pvalue = 0.000, HALLMARK_G2M_CHECKPOINT\n",
      "Passed: cat 35, mi = 0.009, pvalue = 0.000, HALLMARK_OXIDATIVE_PHOSPHORYLATION\n",
      "Passed: cat 32, mi = 0.003, pvalue = 0.000, HALLMARK_MYC_TARGETS_V2\n",
      "Passed: cat 27, mi = 0.002, pvalue = 0.000, HALLMARK_KRAS_SIGNALING_DN\n",
      "Passed: cat 23, mi = 0.002, pvalue = 0.001, HALLMARK_IL6_JAK_STAT3_SIGNALING\n",
      "Didn't pass (cutoff=0.0050): cat 4, mi = 0.001, pvalue = 0.006, HALLMARK_APICAL_JUNCTION\n",
      "Didn't pass (cutoff=0.0050): cat 20, mi = 0.001, pvalue = 0.014, HALLMARK_HEME_METABOLISM\n",
      "Didn't pass (cutoff=0.0050): cat 40, mi = 0.001, pvalue = 0.035, HALLMARK_PROTEIN_SECRETION\n",
      "Didn't pass (cutoff=0.0050): cat 33, mi = 0.001, pvalue = 0.037, HALLMARK_MYOGENESIS\n",
      "Didn't pass (cutoff=0.0050): cat 11, mi = 0.001, pvalue = 0.068, HALLMARK_DNA_REPAIR\n",
      "Didn't pass (cutoff=0.0050): cat 19, mi = 0.001, pvalue = 0.219, HALLMARK_HEDGEHOG_SIGNALING\n",
      "Didn't pass (cutoff=0.0050): cat 37, mi = 0.001, pvalue = 0.371, HALLMARK_PANCREAS_BETA_CELLS\n",
      "Didn't pass (cutoff=0.0050): cat 5, mi = 0.000, pvalue = 0.510, HALLMARK_APICAL_SURFACE\n",
      "Number of categories that passed the tests = 5\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_P_LVM1_2.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_P_LVM1_2.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 2, 3, 0.537939192601153\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "4\n",
      "Done.Start drawing\n",
      "5.29\t0.00\n",
      "HALLMARK_G2M_CHECKPOINT\n",
      "HALLMARK_IL6_JAK_STAT3_SIGNALING\n",
      "HALLMARK_OXIDATIVE_PHOSPHORYLATION\n",
      "HALLMARK_MYC_TARGETS_V2\n",
      "HALLMARK_KRAS_SIGNALING_DN\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_P_LVM1_2.csv_PAGE/filt_FC_P_LVM1_2.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_P_LVM1_2.csv_PAGE/filt_FC_P_LVM1_2.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_P_LVM1_2.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_P_LVM1_2.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 2, 3, 0.537939192601153\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "4\n",
      "Done.-16.8638941398866\t42\t0\n",
      "-13.2117202268431\t42\t6.27221172022684\n",
      "-12.3780718336484\t42\t7.30434782608696\n",
      "-11.8223062381853\t42\t7.93950850661626\n",
      "-11.3856332703214\t42\t8.41587901701323\n",
      "-10.9092627599244\t42\t8.81285444234405\n",
      "-10.3137996219282\t42\t9.36862003780718\n",
      "-9.44045368620038\t42\t10.0037807183365\n",
      "6.55765595463138\t42\t11.1153119092628\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_P_LVM1_2.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_P_LVM1_4.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_h.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_h/human_ensembl_msigdb_h.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_P_LVM1_4.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 50\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t50\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 50\n",
      "cat 31, mi = 0.01325592, HALLMARK_MYC_TARGETS_V1\n",
      "cat 35, mi = 0.01103035, HALLMARK_OXIDATIVE_PHOSPHORYLATION\n",
      "cat 30, mi = 0.00333845, HALLMARK_MTORC1_SIGNALING\n",
      "cat 12, mi = 0.00266843, HALLMARK_E2F_TARGETS\n",
      "cat 11, mi = 0.00222996, HALLMARK_DNA_REPAIR\n",
      "cat 17, mi = 0.00215795, HALLMARK_G2M_CHECKPOINT\n",
      "cat 27, mi = 0.00159562, HALLMARK_KRAS_SIGNALING_DN\n",
      "cat 39, mi = 0.00148593, HALLMARK_PI3K_AKT_MTOR_SIGNALING\n",
      "cat 36, mi = 0.00144163, HALLMARK_P53_PATHWAY\n",
      "cat 32, mi = 0.00139602, HALLMARK_MYC_TARGETS_V2\n",
      "cat 47, mi = 0.00129933, HALLMARK_UV_RESPONSE_UP\n",
      "cat 0, mi = 0.00128207, HALLMARK_ADIPOGENESIS\n",
      "cat 45, mi = 0.00125141, HALLMARK_UNFOLDED_PROTEIN_RESPONSE\n",
      "cat 41, mi = 0.00111675, HALLMARK_REACTIVE_OXIGEN_SPECIES_PATHWAY\n",
      "cat 23, mi = 0.00110297, HALLMARK_IL6_JAK_STAT3_SIGNALING\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 31, mi = 0.013, pvalue = 0.000, HALLMARK_MYC_TARGETS_V1\n",
      "Passed: cat 35, mi = 0.011, pvalue = 0.000, HALLMARK_OXIDATIVE_PHOSPHORYLATION\n",
      "Passed: cat 27, mi = 0.002, pvalue = 0.001, HALLMARK_KRAS_SIGNALING_DN\n",
      "Passed: cat 39, mi = 0.001, pvalue = 0.002, HALLMARK_PI3K_AKT_MTOR_SIGNALING\n",
      "Didn't pass (cutoff=0.0050): cat 8, mi = 0.001, pvalue = 0.072, HALLMARK_CHOLESTEROL_HOMEOSTASIS\n",
      "Didn't pass (cutoff=0.0050): cat 20, mi = 0.001, pvalue = 0.067, HALLMARK_HEME_METABOLISM\n",
      "Didn't pass (cutoff=0.0050): cat 34, mi = 0.001, pvalue = 0.234, HALLMARK_NOTCH_SIGNALING\n",
      "Didn't pass (cutoff=0.0050): cat 19, mi = 0.001, pvalue = 0.240, HALLMARK_HEDGEHOG_SIGNALING\n",
      "Didn't pass (cutoff=0.0050): cat 37, mi = 0.001, pvalue = 0.268, HALLMARK_PANCREAS_BETA_CELLS\n",
      "Didn't pass (cutoff=0.0050): cat 33, mi = 0.001, pvalue = 0.290, HALLMARK_MYOGENESIS\n",
      "Didn't pass (cutoff=0.0050): cat 15, mi = 0.000, pvalue = 0.476, HALLMARK_ESTROGEN_RESPONSE_LATE\n",
      "Didn't pass (cutoff=0.0050): cat 5, mi = 0.000, pvalue = 0.620, HALLMARK_APICAL_SURFACE\n",
      "Didn't pass (cutoff=0.0050): cat 3, mi = 0.000, pvalue = 0.796, HALLMARK_ANGIOGENESIS\n",
      "Number of categories that passed the tests = 4\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_P_LVM1_4.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_P_LVM1_4.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 0, 1, 0.0288504050490969\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "Done.Start drawing\n",
      "110.21\t0.00\n",
      "HALLMARK_MYC_TARGETS_V1\n",
      "HALLMARK_OXIDATIVE_PHOSPHORYLATION\n",
      "HALLMARK_PI3K_AKT_MTOR_SIGNALING\n",
      "HALLMARK_KRAS_SIGNALING_DN\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_P_LVM1_4.csv_PAGE/filt_FC_P_LVM1_4.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_P_LVM1_4.csv_PAGE/filt_FC_P_LVM1_4.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_P_LVM1_4.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_P_LVM1_4.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 0, 1, 0.0288504050490969\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "Done.-19.9980945467743\t42\t0\n",
      "-19.9923781870974\t42\t0.00381090645132021\n",
      "-19.9847563741947\t42\t0.0114327193539606\n",
      "-19.9790400145177\t42\t0.019054532256601\n",
      "-19.9733236548408\t42\t0.0228654387079212\n",
      "-19.9676072951638\t42\t0.0304872516105617\n",
      "-19.8361310225932\t42\t0.0342981580618819\n",
      "-19.4912439887488\t42\t0.293439796751656\n",
      "1.36203611287542\t42\t0.724072225750839\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_h/fc_values/filt_FC_P_LVM1_4.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "sem --wait done\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "nohup sh run.sh iPAGE_JY.sh &"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
