{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# This is used for running data vs human_msigdb_c3 annotations located in: \n",
    "### /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import pickle\n",
    "import seaborn as sns\n",
    "import scipy.stats\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### get index of file names to loop through "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "cd fc_values\n",
    "ls FC* > filelist.txt\n",
    "cd .."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### get ensembl genes represented in iPAGE database"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash \n",
    "\n",
    "cp /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c3/human_ensembl_msigdb_c3_index.txt human_ensembl_index.txt\n",
    "awk '{print $1}' human_ensembl_index.txt > ensembl_index.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### make sure that genes in file match the index "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "###################!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n",
    "###################REMEMBER TO SPECIFY INDEX USED IN THE IPAGE COMMAND!!!!!!!!\n",
    "###################!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n",
    "\n",
    "\n",
    "#####################\n",
    "#######read in data from logfc calculations\n",
    "#####################\n",
    "####define index file to loop through\n",
    "index_file = \"fc_values/filelist.txt\"\n",
    "####define matching ensg index\n",
    "GO_index_filename = \"ensembl_index.txt\"\n",
    "genes_with_GOs = []\n",
    "with open(GO_index_filename,'r') as rf:\n",
    "    for line in rf:\n",
    "        gene_name = line.split('\\n')[0]\n",
    "        genes_with_GOs.append(gene_name)\n",
    "genes_with_GOs_set = set(genes_with_GOs)\n",
    "####open file and loop through each line\n",
    "###\n",
    "iPAGE_sh = \"iPAGE_JY.sh\"\n",
    "cwd = os.getcwd()+\"/\"\n",
    "os.system(\"rm \"+iPAGE_sh)\n",
    "###\n",
    "with open(index_file,'r') as rf:\n",
    "    for line in rf:\n",
    "        #####do this to each file\n",
    "        ###read in file into pandas df\n",
    "        FC_file = \"fc_values/\"+line.split('\\n')[0]\n",
    "        FC_df = pd.read_csv(FC_file,sep=',',index_col=0)\n",
    "        FC_df.columns = ['log2FoldChange']\n",
    "        FC_df.index.names = ['Gene']\n",
    "        ###filter\n",
    "        FC_df = FC_df.loc[[x for x in FC_df.index if x in genes_with_GOs_set]]\n",
    "        ###save\n",
    "        save_file = \"fc_values/filt_\"+line.split('\\n')[0]\n",
    "        FC_df.to_csv(save_file,sep='\\t',header=True)\n",
    "        ###write final iPAGE command to an ipage .sh file\n",
    "        iPAGE_cmd = \"perl /nvme/bins/iPAGEv1.0/page.pl --expfile \"+cwd+save_file+\" --species=human_ensembl_msigdb_c3 --exptype=continuous --ebins=9 \\n\"\n",
    "        with open(iPAGE_sh, \"a\") as myfile:\n",
    "            myfile.write(iPAGE_cmd)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## then run the command from this folder:\n",
    "\n",
    "nohup sh run.sh iPAGE_JY.sh &"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM1_1.csv --species=human_ensembl_msigdb_c3 --exptype=continuous --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM1_2.csv --species=human_ensembl_msigdb_c3 --exptype=continuous --ebins=9\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM1_1.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c3.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c3/human_ensembl_msigdb_c3.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM1_1.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 836\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t836\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 836\n",
      "cat 643, mi = 0.00396501, SGCGSSAAA_E2F1DP2_01\n",
      "cat 559, mi = 0.00279993, NRSF_01\n",
      "cat 486, mi = 0.00235961, KTGGYRSGAA_UNKNOWN\n",
      "cat 159, mi = 0.00233491, CAGNWMCNNNGAC_UNKNOWN\n",
      "cat 176, mi = 0.00199553, CCAWWNAAGG_SRF_Q4\n",
      "cat 37, mi = 0.00187144, ACCTGTTG_UNKNOWN\n",
      "cat 817, mi = 0.00180345, YNTTTNNNANGCARM_UNKNOWN\n",
      "cat 128, mi = 0.00165646, ATGCTGG_MIR338\n",
      "cat 276, mi = 0.00161928, E2F_01\n",
      "cat 457, mi = 0.00158811, HNF3_Q6\n",
      "cat 314, mi = 0.00157457, FOXD3_01\n",
      "cat 821, mi = 0.00154819, YTAAYNGCT_UNKNOWN\n",
      "cat 41, mi = 0.00140150, ACTGAAA_MIR30A3P_MIR30E3P\n",
      "cat 350, mi = 0.00135743, GATA_Q6\n",
      "cat 507, mi = 0.00130718, MEIS1AHOXA9_01\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 643, mi = 0.004, pvalue = 0.000, SGCGSSAAA_E2F1DP2_01\n",
      "Passed: cat 559, mi = 0.003, pvalue = 0.000, NRSF_01\n",
      "Passed: cat 176, mi = 0.002, pvalue = 0.000, CCAWWNAAGG_SRF_Q4\n",
      "Passed: cat 37, mi = 0.002, pvalue = 0.000, ACCTGTTG_UNKNOWN\n",
      "Passed: cat 128, mi = 0.002, pvalue = 0.001, ATGCTGG_MIR338\n",
      "Passed: cat 350, mi = 0.001, pvalue = 0.004, GATA_Q6\n",
      "Didn't pass (cutoff=0.0050): cat 658, mi = 0.001, pvalue = 0.006, SREBP1_01\n",
      "Didn't pass (cutoff=0.0050): cat 374, mi = 0.001, pvalue = 0.011, GGARNTKYCCA_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 367, mi = 0.001, pvalue = 0.012, GCTCTTG_MIR335\n",
      "Didn't pass (cutoff=0.0050): cat 240, mi = 0.001, pvalue = 0.016, CTCAAGA_MIR526B\n",
      "Didn't pass (cutoff=0.0050): cat 523, mi = 0.001, pvalue = 0.012, MYC_Q2\n",
      "Didn't pass (cutoff=0.0050): cat 766, mi = 0.001, pvalue = 0.015, TMTCGCGANR_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 372, mi = 0.001, pvalue = 0.019, GGAANCGGAANY_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 363, mi = 0.001, pvalue = 0.029, GCGCTTT_MIR518B_MIR518C_MIR518D\n",
      "Didn't pass (cutoff=0.0050): cat 672, mi = 0.001, pvalue = 0.039, STAT3_01\n",
      "Didn't pass (cutoff=0.0050): cat 104, mi = 0.001, pvalue = 0.031, AR_01\n",
      "Didn't pass (cutoff=0.0050): cat 619, mi = 0.001, pvalue = 0.043, RNCTGNYNRNCTGNY_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 451, mi = 0.001, pvalue = 0.056, HMX1_01\n",
      "Didn't pass (cutoff=0.0050): cat 648, mi = 0.001, pvalue = 0.038, SNACANNNYSYAGA_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 157, mi = 0.001, pvalue = 0.053, CAGGTCC_MIR492\n",
      "Didn't pass (cutoff=0.0050): cat 44, mi = 0.001, pvalue = 0.048, ACTGTAG_MIR139\n",
      "Didn't pass (cutoff=0.0050): cat 250, mi = 0.001, pvalue = 0.048, CTGTTAC_MIR194\n",
      "Didn't pass (cutoff=0.0050): cat 232, mi = 0.001, pvalue = 0.060, CRGAARNNNNCGA_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 207, mi = 0.001, pvalue = 0.057, CGGAARNGGCNG_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 183, mi = 0.001, pvalue = 0.080, CCTGAGT_MIR510\n",
      "Didn't pass (cutoff=0.0050): cat 103, mi = 0.001, pvalue = 0.068, ARP1_01\n",
      "Total number of categories evaluated = 48.\n",
      "Number of categories that passed the tests = 6\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM1_1.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM1_1.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 0, 3, 0.0883361727106556\n",
      "best merge is 2, 5, 0.153175487639914\n",
      "order classes\n",
      "2 - 0.132504259065983\n",
      "0 3\n",
      "1\n",
      "2\n",
      "4\n",
      "5\n",
      "Done.Start drawing\n",
      "23.77\t0.00\n",
      "SGCGSSAAA_E2F1DP2_01\n",
      "ACCTGTTG_UNKNOWN\n",
      "CCAWWNAAGG_SRF_Q4\n",
      "GATA_Q6\n",
      "ATGCTGG_MIR338\n",
      "NRSF_01\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM1_1.csv_PAGE/filt_FC_LVM1_1.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM1_1.csv_PAGE/filt_FC_LVM1_1.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM1_1.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM1_1.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 0, 3, 0.0883361727106556\n",
      "best merge is 2, 5, 0.153175487639914\n",
      "order classes\n",
      "2 - 0.132504259065983\n",
      "0 3\n",
      "1\n",
      "2\n",
      "4\n",
      "5\n",
      "Done.-19.2932267564156\t42\t0\n",
      "-18.4450988641144\t42\t1.4135464871687\n",
      "-18.2153975599495\t42\t1.69625578460244\n",
      "-18.056373580143\t42\t1.87294909549853\n",
      "-17.8973496003366\t42\t2.0143037442154\n",
      "-17.7206562894405\t42\t2.19099705511149\n",
      "-17.51745898191\t42\t2.36769036600757\n",
      "-17.2082456878418\t42\t2.59739167017249\n",
      "2.49305847707194\t42\t2.98611695414388\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM1_1.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM1_3.csv --species=human_ensembl_msigdb_c3 --exptype=continuous --ebins=9\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM1_2.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c3.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c3/human_ensembl_msigdb_c3.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM1_2.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 836\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t836\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 836\n",
      "cat 109, mi = 0.00240812, ATAACCT_MIR154\n",
      "cat 559, mi = 0.00226419, NRSF_01\n",
      "cat 804, mi = 0.00215734, WYAAANNRNNNGCG_UNKNOWN\n",
      "cat 159, mi = 0.00213677, CAGNWMCNNNGAC_UNKNOWN\n",
      "cat 485, mi = 0.00212069, KRCTCNNNNMANAGC_UNKNOWN\n",
      "cat 483, mi = 0.00193427, KCCGNSWTTT_UNKNOWN\n",
      "cat 113, mi = 0.00193088, ATAGGAA_MIR202\n",
      "cat 365, mi = 0.00183468, GCGSCMNTTT_UNKNOWN\n",
      "cat 394, mi = 0.00178762, GGGNNTTTCC_NFKB_Q6_01\n",
      "cat 378, mi = 0.00171719, GGCACTT_MIR519E\n",
      "cat 449, mi = 0.00168667, HMEF2_Q6\n",
      "cat 150, mi = 0.00166804, CAGCACT_MIR5123P\n",
      "cat 627, mi = 0.00165442, RRCCGTTA_UNKNOWN\n",
      "cat 696, mi = 0.00165112, TAGAACC_MIR182\n",
      "cat 367, mi = 0.00164355, GCTCTTG_MIR335\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 109, mi = 0.002, pvalue = 0.000, ATAACCT_MIR154\n",
      "Passed: cat 559, mi = 0.002, pvalue = 0.000, NRSF_01\n",
      "Passed: cat 804, mi = 0.002, pvalue = 0.000, WYAAANNRNNNGCG_UNKNOWN\n",
      "Passed: cat 485, mi = 0.002, pvalue = 0.000, KRCTCNNNNMANAGC_UNKNOWN\n",
      "Passed: cat 113, mi = 0.002, pvalue = 0.000, ATAGGAA_MIR202\n",
      "Passed: cat 365, mi = 0.002, pvalue = 0.000, GCGSCMNTTT_UNKNOWN\n",
      "Passed: cat 394, mi = 0.002, pvalue = 0.000, GGGNNTTTCC_NFKB_Q6_01\n",
      "Passed: cat 150, mi = 0.002, pvalue = 0.001, CAGCACT_MIR5123P\n",
      "Passed: cat 627, mi = 0.002, pvalue = 0.001, RRCCGTTA_UNKNOWN\n",
      "Passed: cat 696, mi = 0.002, pvalue = 0.001, TAGAACC_MIR182\n",
      "Passed: cat 367, mi = 0.002, pvalue = 0.001, GCTCTTG_MIR335\n",
      "Passed: cat 515, mi = 0.002, pvalue = 0.001, MYAATNNNNNNNGGC_UNKNOWN\n",
      "Passed: cat 457, mi = 0.001, pvalue = 0.003, HNF3_Q6\n",
      "Didn't pass (cutoff=0.0050): cat 387, mi = 0.001, pvalue = 0.010, GGCNRNWCTTYS_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 715, mi = 0.001, pvalue = 0.012, TCCCCAC_MIR491\n",
      "Didn't pass (cutoff=0.0050): cat 173, mi = 0.001, pvalue = 0.018, CCANNAGRKGGC_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 372, mi = 0.001, pvalue = 0.033, GGAANCGGAANY_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 375, mi = 0.001, pvalue = 0.027, GGATCCG_MIR127\n",
      "Didn't pass (cutoff=0.0050): cat 507, mi = 0.001, pvalue = 0.061, MEIS1AHOXA9_01\n",
      "Didn't pass (cutoff=0.0050): cat 47, mi = 0.001, pvalue = 0.069, ACTWSNACTNY_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 637, mi = 0.001, pvalue = 0.072, RYTGCNWTGGNR_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 416, mi = 0.001, pvalue = 0.047, GTCGATC_MIR3695P\n",
      "Didn't pass (cutoff=0.0050): cat 363, mi = 0.001, pvalue = 0.204, GCGCTTT_MIR518B_MIR518C_MIR518D\n",
      "Didn't pass (cutoff=0.0050): cat 38, mi = 0.001, pvalue = 0.142, ACGCACA_MIR210\n",
      "Didn't pass (cutoff=0.0050): cat 411, mi = 0.001, pvalue = 0.282, GTAGGCA_MIR189\n",
      "Didn't pass (cutoff=0.0050): cat 39, mi = 0.001, pvalue = 0.212, ACTACCT_MIR196A_MIR196B\n",
      "Didn't pass (cutoff=0.0050): cat 434, mi = 0.001, pvalue = 0.259, GTTGNYNNRGNAAC_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 672, mi = 0.001, pvalue = 0.427, STAT3_01\n",
      "Didn't pass (cutoff=0.0050): cat 373, mi = 0.001, pvalue = 0.402, GGAMTNNNNNTCCY_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 59, mi = 0.000, pvalue = 0.485, AGGGCCA_MIR328\n",
      "Didn't pass: cat 383, mi = 0.000, pvalue = 1.000, GGCGGCA_MIR371\n",
      "Didn't pass (cutoff=0.0050): cat 641, mi = 0.000, pvalue = 0.742, SEF1_C\n",
      "Didn't pass (cutoff=0.0050): cat 602, mi = 0.000, pvalue = 0.657, PPARG_01\n",
      "Total number of categories evaluated = 285.\n",
      "Number of categories that passed the tests = 13\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM1_2.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM1_2.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 2, 8, 0.0559365022588242\n",
      "best merge is 11, 12, 0.187630305681003\n",
      "best merge is 2, 3, 0.21734070084481\n",
      "best merge is 7, 10, 0.230385188221248\n",
      "best merge is 2, 11, 0.289580540490236\n",
      "best merge is 5, 7, 0.330562714233605\n",
      "best merge is 4, 5, 0.37802898706591\n",
      "best merge is 2, 6, 0.405039425968727\n",
      "best merge is 4, 9, 0.525337186684369\n",
      "order classes\n",
      "6 - 0.534556218253598\n",
      "4 - 0.822899016016263\n",
      "2 8 3 11 12 6\n",
      "1\n",
      "0\n",
      "4 5 7 10\n",
      "9\n",
      "Done.Start drawing\n",
      "62.96\t0.00\n",
      "WYAAANNRNNNGCG_UNKNOWN\n",
      "RRCCGTTA_UNKNOWN\n",
      "KRCTCNNNNMANAGC_UNKNOWN\n",
      "MYAATNNNNNNNGGC_UNKNOWN\n",
      "HNF3_Q6\n",
      "GGGNNTTTCC_NFKB_Q6_01\n",
      "CAGCACT_MIR5123P\n",
      "ATAGGAA_MIR202\n",
      "ATAACCT_MIR154\n",
      "GCGSCMNTTT_UNKNOWN\n",
      "GCTCTTG_MIR335\n",
      "TAGAACC_MIR182\n",
      "NRSF_01\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM1_2.csv_PAGE/filt_FC_LVM1_2.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM1_2.csv_PAGE/filt_FC_LVM1_2.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM1_2.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM1_2.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 2, 8, 0.0559365022588242\n",
      "best merge is 11, 12, 0.187630305681003\n",
      "best merge is 2, 3, 0.21734070084481\n",
      "best merge is 7, 10, 0.230385188221248\n",
      "best merge is 2, 11, 0.289580540490236\n",
      "best merge is 5, 7, 0.330562714233605\n",
      "best merge is 4, 5, 0.37802898706591\n",
      "best merge is 2, 6, 0.405039425968727\n",
      "best merge is 4, 9, 0.525337186684369\n",
      "order classes\n",
      "6 - 0.534556218253598\n",
      "4 - 0.822899016016263\n",
      "2 8 3 11 12 6\n",
      "1\n",
      "0\n",
      "4 5 7 10\n",
      "9\n",
      "Done.-19.7198221092757\t42\t0\n",
      "-19.3562579415502\t42\t0.560355781448539\n",
      "-19.2195044472681\t42\t0.72712833545108\n",
      "-19.1227763659466\t42\t0.833862770012706\n",
      "-19.0360546378653\t42\t0.920584498094028\n",
      "-18.9459974587039\t42\t1.00730622617535\n",
      "-18.8359275730623\t42\t1.10069885641677\n",
      "-18.6558132147395\t42\t1.2274459974587\n",
      "1.73046378653113\t42\t1.46092757306226\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM1_2.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM1_4.csv --species=human_ensembl_msigdb_c3 --exptype=continuous --ebins=9\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM1_3.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c3.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c3/human_ensembl_msigdb_c3.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM1_3.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 836\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t836\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 836\n",
      "cat 485, mi = 0.00322139, KRCTCNNNNMANAGC_UNKNOWN\n",
      "cat 559, mi = 0.00250430, NRSF_01\n",
      "cat 159, mi = 0.00229893, CAGNWMCNNNGAC_UNKNOWN\n",
      "cat 643, mi = 0.00206450, SGCGSSAAA_E2F1DP2_01\n",
      "cat 766, mi = 0.00178575, TMTCGCGANR_UNKNOWN\n",
      "cat 150, mi = 0.00164827, CAGCACT_MIR5123P\n",
      "cat 443, mi = 0.00164345, HFH3_01\n",
      "cat 326, mi = 0.00162485, FREAC4_01\n",
      "cat 364, mi = 0.00159610, GCGNNANTTCC_UNKNOWN\n",
      "cat 786, mi = 0.00152461, TTTNNANAGCYR_UNKNOWN\n",
      "cat 507, mi = 0.00150804, MEIS1AHOXA9_01\n",
      "cat 314, mi = 0.00149784, FOXD3_01\n",
      "cat 444, mi = 0.00149595, HFH4_01\n",
      "cat 484, mi = 0.00144701, KMCATNNWGGA_UNKNOWN\n",
      "cat 37, mi = 0.00143197, ACCTGTTG_UNKNOWN\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 485, mi = 0.003, pvalue = 0.000, KRCTCNNNNMANAGC_UNKNOWN\n",
      "Passed: cat 559, mi = 0.003, pvalue = 0.000, NRSF_01\n",
      "Passed: cat 643, mi = 0.002, pvalue = 0.000, SGCGSSAAA_E2F1DP2_01\n",
      "Passed: cat 766, mi = 0.002, pvalue = 0.001, TMTCGCGANR_UNKNOWN\n",
      "Passed: cat 150, mi = 0.002, pvalue = 0.001, CAGCACT_MIR5123P\n",
      "Passed: cat 443, mi = 0.002, pvalue = 0.001, HFH3_01\n",
      "Passed: cat 364, mi = 0.002, pvalue = 0.001, GCGNNANTTCC_UNKNOWN\n",
      "Passed: cat 484, mi = 0.001, pvalue = 0.003, KMCATNNWGGA_UNKNOWN\n",
      "Passed: cat 713, mi = 0.001, pvalue = 0.004, TCCAGAT_MIR5165P\n",
      "Passed: cat 372, mi = 0.001, pvalue = 0.003, GGAANCGGAANY_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 600, mi = 0.001, pvalue = 0.016, PPARA_01\n",
      "Didn't pass (cutoff=0.0050): cat 781, mi = 0.001, pvalue = 0.013, TTGGGAG_MIR150\n",
      "Didn't pass (cutoff=0.0050): cat 377, mi = 0.001, pvalue = 0.034, GGCACAT_MIR455\n",
      "Didn't pass (cutoff=0.0050): cat 367, mi = 0.001, pvalue = 0.042, GCTCTTG_MIR335\n",
      "Didn't pass (cutoff=0.0050): cat 767, mi = 0.001, pvalue = 0.043, TNCATNTCCYR_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 7, mi = 0.001, pvalue = 0.065, AACATTC_MIR4093P\n",
      "Didn't pass (cutoff=0.0050): cat 727, mi = 0.001, pvalue = 0.098, TCTGATC_MIR383\n",
      "Didn't pass (cutoff=0.0050): cat 125, mi = 0.001, pvalue = 0.112, ATGCACG_MIR517B\n",
      "Didn't pass (cutoff=0.0050): cat 662, mi = 0.001, pvalue = 0.107, SRF_01\n",
      "Didn't pass (cutoff=0.0050): cat 698, mi = 0.001, pvalue = 0.106, TAGGTCA_MIR192_MIR215\n",
      "Didn't pass (cutoff=0.0050): cat 207, mi = 0.001, pvalue = 0.105, CGGAARNGGCNG_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 479, mi = 0.001, pvalue = 0.116, IRF2_01\n",
      "Didn't pass (cutoff=0.0050): cat 435, mi = 0.001, pvalue = 0.136, GTTNYYNNGGTNA_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 176, mi = 0.001, pvalue = 0.150, CCAWWNAAGG_SRF_Q4\n",
      "Didn't pass (cutoff=0.0050): cat 312, mi = 0.001, pvalue = 0.253, EVI1_06\n",
      "Didn't pass (cutoff=0.0050): cat 823, mi = 0.001, pvalue = 0.190, YTCCCRNNAGGY_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 451, mi = 0.001, pvalue = 0.286, HMX1_01\n",
      "Didn't pass (cutoff=0.0050): cat 588, mi = 0.001, pvalue = 0.316, PAX8_01\n",
      "Didn't pass (cutoff=0.0050): cat 669, mi = 0.001, pvalue = 0.251, STAT1_01\n",
      "Didn't pass (cutoff=0.0050): cat 612, mi = 0.001, pvalue = 0.244, RACTNNRTTTNC_UNKNOWN\n",
      "Total number of categories evaluated = 138.\n",
      "Number of categories that passed the tests = 10\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM1_3.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM1_3.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 0, 1, 0.0425800698637344\n",
      "best merge is 6, 9, 0.091902312180883\n",
      "best merge is 0, 2, 0.197665520063972\n",
      "best merge is 3, 6, 0.46866294928543\n",
      "best merge is 4, 7, 0.475743083387362\n",
      "best merge is 3, 8, 0.566283721785037\n",
      "order classes\n",
      "3 - 0.324378599993835\n",
      "3 - 0.762391267223513\n",
      "2 - 1.18935770846841\n",
      "0 1 2\n",
      "3 6 9\n",
      "4 7\n",
      "5\n",
      "8\n",
      "Done.Start drawing\n",
      "84.94\t0.00\n",
      "HFH3_01\n",
      "TCCAGAT_MIR5165P\n",
      "CAGCACT_MIR5123P\n",
      "KMCATNNWGGA_UNKNOWN\n",
      "GGAANCGGAANY_UNKNOWN\n",
      "GCGNNANTTCC_UNKNOWN\n",
      "TMTCGCGANR_UNKNOWN\n",
      "KRCTCNNNNMANAGC_UNKNOWN\n",
      "NRSF_01\n",
      "SGCGSSAAA_E2F1DP2_01\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM1_3.csv_PAGE/filt_FC_LVM1_3.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM1_3.csv_PAGE/filt_FC_LVM1_3.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM1_3.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM1_3.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 0, 1, 0.0425800698637344\n",
      "best merge is 6, 9, 0.091902312180883\n",
      "best merge is 0, 2, 0.197665520063972\n",
      "best merge is 3, 6, 0.46866294928543\n",
      "best merge is 4, 7, 0.475743083387362\n",
      "best merge is 3, 8, 0.566283721785037\n",
      "order classes\n",
      "3 - 0.324378599993835\n",
      "3 - 0.762391267223513\n",
      "2 - 1.18935770846841\n",
      "0 1 2\n",
      "3 6 9\n",
      "4 7\n",
      "5\n",
      "8\n",
      "Done.-19.849187661879\t42\t0\n",
      "-19.6588179891688\t42\t0.301624676242053\n",
      "-19.5920649870497\t42\t0.380739345420297\n",
      "-19.5401459854015\t42\t0.435130680480339\n",
      "-19.4931716505769\t42\t0.484577348716741\n",
      "-19.4437249823405\t42\t0.529079350129503\n",
      "-19.3769719802213\t42\t0.583470685189545\n",
      "-19.2706616435131\t42\t0.662585354367789\n",
      "1.39804567930304\t42\t0.796091358606075\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM1_3.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM2_1.csv --species=human_ensembl_msigdb_c3 --exptype=continuous --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM2_2.csv --species=human_ensembl_msigdb_c3 --exptype=continuous --ebins=9\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM1_4.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c3.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c3/human_ensembl_msigdb_c3.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM1_4.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 836\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t836\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 836\n",
      "cat 559, mi = 0.00217374, NRSF_01\n",
      "cat 483, mi = 0.00211274, KCCGNSWTTT_UNKNOWN\n",
      "cat 364, mi = 0.00202766, GCGNNANTTCC_UNKNOWN\n",
      "cat 766, mi = 0.00185840, TMTCGCGANR_UNKNOWN\n",
      "cat 159, mi = 0.00176909, CAGNWMCNNNGAC_UNKNOWN\n",
      "cat 485, mi = 0.00152139, KRCTCNNNNMANAGC_UNKNOWN\n",
      "cat 633, mi = 0.00151507, RYAAAKNNNNNNTTGW_UNKNOWN\n",
      "cat 365, mi = 0.00149412, GCGSCMNTTT_UNKNOWN\n",
      "cat 31, mi = 0.00142239, ACAWNRNSRCGG_UNKNOWN\n",
      "cat 401, mi = 0.00141417, GKCGCNNNNNNNTGAYG_UNKNOWN\n",
      "cat 501, mi = 0.00138852, MCAATNNNNNGCG_UNKNOWN\n",
      "cat 372, mi = 0.00133936, GGAANCGGAANY_UNKNOWN\n",
      "cat 314, mi = 0.00129790, FOXD3_01\n",
      "cat 249, mi = 0.00127606, CTGRYYYNATT_UNKNOWN\n",
      "cat 658, mi = 0.00126443, SREBP1_01\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 559, mi = 0.002, pvalue = 0.000, NRSF_01\n",
      "Passed: cat 483, mi = 0.002, pvalue = 0.000, KCCGNSWTTT_UNKNOWN\n",
      "Passed: cat 485, mi = 0.002, pvalue = 0.003, KRCTCNNNNMANAGC_UNKNOWN\n",
      "Passed: cat 633, mi = 0.002, pvalue = 0.002, RYAAAKNNNNNNTTGW_UNKNOWN\n",
      "Passed: cat 365, mi = 0.001, pvalue = 0.002, GCGSCMNTTT_UNKNOWN\n",
      "Passed: cat 401, mi = 0.001, pvalue = 0.003, GKCGCNNNNNNNTGAYG_UNKNOWN\n",
      "Passed: cat 501, mi = 0.001, pvalue = 0.004, MCAATNNNNNGCG_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 372, mi = 0.001, pvalue = 0.005, GGAANCGGAANY_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 249, mi = 0.001, pvalue = 0.013, CTGRYYYNATT_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 658, mi = 0.001, pvalue = 0.008, SREBP1_01\n",
      "Didn't pass (cutoff=0.0050): cat 449, mi = 0.001, pvalue = 0.014, HMEF2_Q6\n",
      "Didn't pass (cutoff=0.0050): cat 387, mi = 0.001, pvalue = 0.018, GGCNRNWCTTYS_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 619, mi = 0.001, pvalue = 0.021, RNCTGNYNRNCTGNY_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 412, mi = 0.001, pvalue = 0.018, GTATGAT_MIR154_MIR487\n",
      "Didn't pass (cutoff=0.0050): cat 602, mi = 0.001, pvalue = 0.026, PPARG_01\n",
      "Didn't pass (cutoff=0.0050): cat 621, mi = 0.001, pvalue = 0.034, RNTCANNRNNYNATTW_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 436, mi = 0.001, pvalue = 0.033, GTTRYCATRR_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 131, mi = 0.001, pvalue = 0.030, ATGTAGC_MIR221_MIR222\n",
      "Didn't pass (cutoff=0.0050): cat 743, mi = 0.001, pvalue = 0.032, TGASTMAGC_NFE2_01\n",
      "Didn't pass (cutoff=0.0050): cat 635, mi = 0.001, pvalue = 0.060, RYTAAWNNNTGAY_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 279, mi = 0.001, pvalue = 0.042, E2F_Q2\n",
      "Didn't pass (cutoff=0.0050): cat 407, mi = 0.001, pvalue = 0.066, GTAAACC_MIR2995P\n",
      "Didn't pass (cutoff=0.0050): cat 378, mi = 0.001, pvalue = 0.055, GGCACTT_MIR519E\n",
      "Didn't pass (cutoff=0.0050): cat 672, mi = 0.001, pvalue = 0.090, STAT3_01\n",
      "Didn't pass (cutoff=0.0050): cat 715, mi = 0.001, pvalue = 0.083, TCCCCAC_MIR491\n",
      "Didn't pass (cutoff=0.0050): cat 600, mi = 0.001, pvalue = 0.109, PPARA_01\n",
      "Didn't pass (cutoff=0.0050): cat 465, mi = 0.001, pvalue = 0.072, HOXA3_01\n",
      "Total number of categories evaluated = 64.\n",
      "Number of categories that passed the tests = 7\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM1_4.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM1_4.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 0, 2, 0.0710147814392328\n",
      "best merge is 1, 4, 0.248036342652587\n",
      "best merge is 5, 6, 0.402440873197888\n",
      "order classes\n",
      "2 - 0.124275867518657\n",
      "2 - 0.434063599642027\n",
      "0 2\n",
      "1 4\n",
      "3\n",
      "5\n",
      "6\n",
      "Done.Start drawing\n",
      "26.31\t0.00\n",
      "RYAAAKNNNNNNTTGW_UNKNOWN\n",
      "GKCGCNNNNNNNTGAYG_UNKNOWN\n",
      "MCAATNNNNNGCG_UNKNOWN\n",
      "KCCGNSWTTT_UNKNOWN\n",
      "GCGSCMNTTT_UNKNOWN\n",
      "NRSF_01\n",
      "KRCTCNNNNMANAGC_UNKNOWN\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM1_4.csv_PAGE/filt_FC_LVM1_4.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM1_4.csv_PAGE/filt_FC_LVM1_4.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM1_4.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM1_4.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 0, 2, 0.0710147814392328\n",
      "best merge is 1, 4, 0.248036342652587\n",
      "best merge is 5, 6, 0.402440873197888\n",
      "order classes\n",
      "2 - 0.124275867518657\n",
      "2 - 0.434063599642027\n",
      "0 2\n",
      "1 4\n",
      "3\n",
      "5\n",
      "6\n",
      "Done.-19.5290763968073\t42\t0\n",
      "-18.9064994298746\t42\t0.941847206385405\n",
      "-18.6590649942987\t42\t1.24515393386545\n",
      "-18.4754846066135\t42\t1.43671607753706\n",
      "-18.3158494868871\t42\t1.61231470923603\n",
      "-18.1641961231471\t42\t1.75598631698974\n",
      "-17.9806157354618\t42\t1.91562143671608\n",
      "-17.709236031927\t42\t2.12314709236032\n",
      "2.22919042189282\t42\t2.45838084378563\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM1_4.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM2_3.csv --species=human_ensembl_msigdb_c3 --exptype=continuous --ebins=9\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM2_1.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c3.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c3/human_ensembl_msigdb_c3.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM2_1.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 836\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t836\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 836\n",
      "cat 559, mi = 0.00346265, NRSF_01\n",
      "cat 485, mi = 0.00321931, KRCTCNNNNMANAGC_UNKNOWN\n",
      "cat 159, mi = 0.00226427, CAGNWMCNNNGAC_UNKNOWN\n",
      "cat 314, mi = 0.00213430, FOXD3_01\n",
      "cat 786, mi = 0.00208309, TTTNNANAGCYR_UNKNOWN\n",
      "cat 215, mi = 0.00200993, COMP1_01\n",
      "cat 50, mi = 0.00183829, AGCATTA_MIR155\n",
      "cat 66, mi = 0.00175959, AHRARNT_01\n",
      "cat 632, mi = 0.00172960, RTTTNNNYTGGM_UNKNOWN\n",
      "cat 350, mi = 0.00171546, GATA_Q6\n",
      "cat 658, mi = 0.00165088, SREBP1_01\n",
      "cat 483, mi = 0.00160995, KCCGNSWTTT_UNKNOWN\n",
      "cat 803, mi = 0.00153825, WWTAAGGC_UNKNOWN\n",
      "cat 210, mi = 0.00152174, CGTSACG_PAX3_B\n",
      "cat 4, mi = 0.00151981, AAANWWTGC_UNKNOWN\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 559, mi = 0.003, pvalue = 0.000, NRSF_01\n",
      "Passed: cat 485, mi = 0.003, pvalue = 0.000, KRCTCNNNNMANAGC_UNKNOWN\n",
      "Passed: cat 314, mi = 0.002, pvalue = 0.000, FOXD3_01\n",
      "Passed: cat 50, mi = 0.002, pvalue = 0.000, AGCATTA_MIR155\n",
      "Passed: cat 658, mi = 0.002, pvalue = 0.002, SREBP1_01\n",
      "Passed: cat 483, mi = 0.002, pvalue = 0.001, KCCGNSWTTT_UNKNOWN\n",
      "Passed: cat 515, mi = 0.001, pvalue = 0.005, MYAATNNNNNNNGGC_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 207, mi = 0.001, pvalue = 0.013, CGGAARNGGCNG_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 378, mi = 0.001, pvalue = 0.009, GGCACTT_MIR519E\n",
      "Didn't pass (cutoff=0.0050): cat 113, mi = 0.001, pvalue = 0.019, ATAGGAA_MIR202\n",
      "Didn't pass (cutoff=0.0050): cat 372, mi = 0.001, pvalue = 0.020, GGAANCGGAANY_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 125, mi = 0.001, pvalue = 0.027, ATGCACG_MIR517B\n",
      "Didn't pass (cutoff=0.0050): cat 717, mi = 0.001, pvalue = 0.021, TCCGTCC_MIR184\n",
      "Didn't pass (cutoff=0.0050): cat 13, mi = 0.001, pvalue = 0.024, AACYNNNNTTCCS_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 817, mi = 0.001, pvalue = 0.038, YNTTTNNNANGCARM_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 417, mi = 0.001, pvalue = 0.028, GTCNYYATGR_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 511, mi = 0.001, pvalue = 0.032, MIF1_01\n",
      "Didn't pass (cutoff=0.0050): cat 465, mi = 0.001, pvalue = 0.017, HOXA3_01\n",
      "Didn't pass (cutoff=0.0050): cat 117, mi = 0.001, pvalue = 0.061, ATCTTGC_MIR31\n",
      "Didn't pass (cutoff=0.0050): cat 411, mi = 0.001, pvalue = 0.092, GTAGGCA_MIR189\n",
      "Didn't pass (cutoff=0.0050): cat 146, mi = 0.001, pvalue = 0.094, CACGTTT_MIR302A\n",
      "Didn't pass (cutoff=0.0050): cat 705, mi = 0.001, pvalue = 0.079, TATCTGG_MIR488\n",
      "Didn't pass (cutoff=0.0050): cat 743, mi = 0.001, pvalue = 0.084, TGASTMAGC_NFE2_01\n",
      "Didn't pass (cutoff=0.0050): cat 370, mi = 0.001, pvalue = 0.097, GCTTGAA_MIR498\n",
      "Didn't pass (cutoff=0.0050): cat 602, mi = 0.001, pvalue = 0.137, PPARG_01\n",
      "Didn't pass (cutoff=0.0050): cat 61, mi = 0.001, pvalue = 0.117, AGTCAGC_MIR345\n",
      "Didn't pass (cutoff=0.0050): cat 52, mi = 0.001, pvalue = 0.149, AGCGCTT_MIR518F_MIR518E_MIR518A\n",
      "Total number of categories evaluated = 85.\n",
      "Number of categories that passed the tests = 7\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM2_1.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM2_1.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 3, 5, 0.113433759393443\n",
      "best merge is 0, 1, 0.120602868582455\n",
      "best merge is 3, 4, 0.524875778423181\n",
      "order classes\n",
      "2 - 0.211055020019296\n",
      "2 - 0.198509078938525\n",
      "3 5\n",
      "2\n",
      "0 1\n",
      "4\n",
      "6\n",
      "Done.Start drawing\n",
      "70.09\t0.00\n",
      "FOXD3_01\n",
      "AGCATTA_MIR155\n",
      "KCCGNSWTTT_UNKNOWN\n",
      "SREBP1_01\n",
      "MYAATNNNNNNNGGC_UNKNOWN\n",
      "NRSF_01\n",
      "KRCTCNNNNMANAGC_UNKNOWN\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM2_1.csv_PAGE/filt_FC_LVM2_1.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM2_1.csv_PAGE/filt_FC_LVM2_1.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM2_1.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM2_1.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 3, 5, 0.113433759393443\n",
      "best merge is 0, 1, 0.120602868582455\n",
      "best merge is 3, 4, 0.524875778423181\n",
      "order classes\n",
      "2 - 0.211055020019296\n",
      "2 - 0.198509078938525\n",
      "3 5\n",
      "2\n",
      "0 1\n",
      "4\n",
      "6\n",
      "Done.-19.784277357683\t42\t0\n",
      "-19.5056356113568\t42\t0.431445284634042\n",
      "-19.4037665858182\t42\t0.557283492652304\n",
      "-19.3318590383792\t42\t0.635183335711228\n",
      "-19.2689399343701\t42\t0.701098587530318\n",
      "-19.203024682551\t42\t0.761021543729491\n",
      "-19.1221286916821\t42\t0.832929091168498\n",
      "-18.9902981880439\t42\t0.922813525467256\n",
      "1.54829504922243\t42\t1.09659009844486\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM2_1.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM2_4.csv --species=human_ensembl_msigdb_c3 --exptype=continuous --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM2_5.csv --species=human_ensembl_msigdb_c3 --exptype=continuous --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_P_LVM1_1.csv --species=human_ensembl_msigdb_c3 --exptype=continuous --ebins=9\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM2_4.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c3.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c3/human_ensembl_msigdb_c3.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM2_4.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 836\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t836\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 836\n",
      "cat 485, mi = 0.00300977, KRCTCNNNNMANAGC_UNKNOWN\n",
      "cat 643, mi = 0.00287901, SGCGSSAAA_E2F1DP2_01\n",
      "cat 150, mi = 0.00228371, CAGCACT_MIR5123P\n",
      "cat 766, mi = 0.00200811, TMTCGCGANR_UNKNOWN\n",
      "cat 559, mi = 0.00196740, NRSF_01\n",
      "cat 4, mi = 0.00193726, AAANWWTGC_UNKNOWN\n",
      "cat 372, mi = 0.00188841, GGAANCGGAANY_UNKNOWN\n",
      "cat 159, mi = 0.00181174, CAGNWMCNNNGAC_UNKNOWN\n",
      "cat 131, mi = 0.00178750, ATGTAGC_MIR221_MIR222\n",
      "cat 132, mi = 0.00175482, ATGTCAC_MIR489\n",
      "cat 236, mi = 0.00171694, CTACTGT_MIR199A\n",
      "cat 44, mi = 0.00165999, ACTGTAG_MIR139\n",
      "cat 113, mi = 0.00165109, ATAGGAA_MIR202\n",
      "cat 337, mi = 0.00164372, GAGCCTG_MIR484\n",
      "cat 378, mi = 0.00156650, GGCACTT_MIR519E\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 485, mi = 0.003, pvalue = 0.000, KRCTCNNNNMANAGC_UNKNOWN\n",
      "Passed: cat 643, mi = 0.003, pvalue = 0.000, SGCGSSAAA_E2F1DP2_01\n",
      "Passed: cat 150, mi = 0.002, pvalue = 0.000, CAGCACT_MIR5123P\n",
      "Passed: cat 766, mi = 0.002, pvalue = 0.000, TMTCGCGANR_UNKNOWN\n",
      "Passed: cat 559, mi = 0.002, pvalue = 0.000, NRSF_01\n",
      "Passed: cat 4, mi = 0.002, pvalue = 0.000, AAANWWTGC_UNKNOWN\n",
      "Passed: cat 372, mi = 0.002, pvalue = 0.000, GGAANCGGAANY_UNKNOWN\n",
      "Passed: cat 132, mi = 0.002, pvalue = 0.000, ATGTCAC_MIR489\n",
      "Passed: cat 337, mi = 0.002, pvalue = 0.001, GAGCCTG_MIR484\n",
      "Passed: cat 743, mi = 0.002, pvalue = 0.002, TGASTMAGC_NFE2_01\n",
      "Didn't pass (cutoff=0.0050): cat 394, mi = 0.001, pvalue = 0.015, GGGNNTTTCC_NFKB_Q6_01\n",
      "Didn't pass (cutoff=0.0050): cat 387, mi = 0.001, pvalue = 0.022, GGCNRNWCTTYS_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 207, mi = 0.001, pvalue = 0.038, CGGAARNGGCNG_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 363, mi = 0.001, pvalue = 0.052, GCGCTTT_MIR518B_MIR518C_MIR518D\n",
      "Didn't pass (cutoff=0.0050): cat 125, mi = 0.001, pvalue = 0.055, ATGCACG_MIR517B\n",
      "Didn't pass (cutoff=0.0050): cat 747, mi = 0.001, pvalue = 0.062, TGCACGA_MIR517A_MIR517C\n",
      "Didn't pass (cutoff=0.0050): cat 502, mi = 0.001, pvalue = 0.068, MEF2_01\n",
      "Didn't pass (cutoff=0.0050): cat 395, mi = 0.001, pvalue = 0.070, GGGNRMNNYCAT_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 505, mi = 0.001, pvalue = 0.105, MEF2_04\n",
      "Didn't pass (cutoff=0.0050): cat 648, mi = 0.001, pvalue = 0.092, SNACANNNYSYAGA_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 411, mi = 0.001, pvalue = 0.151, GTAGGCA_MIR189\n",
      "Didn't pass (cutoff=0.0050): cat 416, mi = 0.001, pvalue = 0.045, GTCGATC_MIR3695P\n",
      "Didn't pass (cutoff=0.0050): cat 367, mi = 0.001, pvalue = 0.157, GCTCTTG_MIR335\n",
      "Didn't pass (cutoff=0.0050): cat 636, mi = 0.001, pvalue = 0.156, RYTGCNNRGNAAC_MIF1_01\n",
      "Didn't pass (cutoff=0.0050): cat 694, mi = 0.001, pvalue = 0.264, TACGGGT_MIR99A_MIR100_MIR99B\n",
      "Didn't pass (cutoff=0.0050): cat 691, mi = 0.001, pvalue = 0.185, TAAWWATAG_RSRFC4_Q2\n",
      "Didn't pass (cutoff=0.0050): cat 307, mi = 0.001, pvalue = 0.259, EVI1_01\n",
      "Didn't pass (cutoff=0.0050): cat 465, mi = 0.001, pvalue = 0.266, HOXA3_01\n",
      "Didn't pass (cutoff=0.0050): cat 622, mi = 0.001, pvalue = 0.269, ROAZ_01\n",
      "Didn't pass (cutoff=0.0050): cat 210, mi = 0.001, pvalue = 0.204, CGTSACG_PAX3_B\n",
      "Total number of categories evaluated = 153.\n",
      "Number of categories that passed the tests = 10\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM2_4.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM2_4.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 0, 4, 0.0186967036760203\n",
      "best merge is 0, 1, 0.0947695814525194\n",
      "best merge is 3, 6, 0.494605551353792\n",
      "best merge is 5, 8, 0.523326728932556\n",
      "best merge is 2, 7, 0.535747412919067\n",
      "best merge is 2, 3, 0.693044037874185\n",
      "order classes\n",
      "3 - 0.154248790060044\n",
      "2 - 1.33936853229767\n",
      "2 - 1.23651387838448\n",
      "2 - 1.30831682233139\n",
      "0 4 1\n",
      "2 7\n",
      "3 6\n",
      "5 8\n",
      "9\n",
      "Done.Start drawing\n",
      "100.30\t0.00\n",
      "AAANWWTGC_UNKNOWN\n",
      "GAGCCTG_MIR484\n",
      "TGASTMAGC_NFE2_01\n",
      "CAGCACT_MIR5123P\n",
      "ATGTCAC_MIR489\n",
      "GGAANCGGAANY_UNKNOWN\n",
      "TMTCGCGANR_UNKNOWN\n",
      "KRCTCNNNNMANAGC_UNKNOWN\n",
      "NRSF_01\n",
      "SGCGSSAAA_E2F1DP2_01\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM2_4.csv_PAGE/filt_FC_LVM2_4.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM2_4.csv_PAGE/filt_FC_LVM2_4.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM2_4.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM2_4.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 0, 4, 0.0186967036760203\n",
      "best merge is 0, 1, 0.0947695814525194\n",
      "best merge is 3, 6, 0.494605551353792\n",
      "best merge is 5, 8, 0.523326728932556\n",
      "best merge is 2, 7, 0.535747412919067\n",
      "best merge is 2, 3, 0.693044037874185\n",
      "order classes\n",
      "3 - 0.154248790060044\n",
      "2 - 1.33936853229767\n",
      "2 - 1.23651387838448\n",
      "2 - 1.30831682233139\n",
      "0 4 1\n",
      "2 7\n",
      "3 6\n",
      "5 8\n",
      "9\n",
      "Done.-19.8911266201396\t42\t0\n",
      "-19.7403788634098\t42\t0.217746759720837\n",
      "-19.6712861415753\t42\t0.301495513459621\n",
      "-19.621036889332\t42\t0.355932203389831\n",
      "-19.5728813559322\t42\t0.401994017946162\n",
      "-19.5163509471585\t42\t0.452243270189432\n",
      "-19.4451645064806\t42\t0.515054835493519\n",
      "-19.3321036889332\t42\t0.594616151545364\n",
      "1.37058823529411\t42\t0.741176470588235\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM2_4.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM2_2.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c3.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c3/human_ensembl_msigdb_c3.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM2_2.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 836\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t836\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 836\n",
      "cat 559, mi = 0.00364048, NRSF_01\n",
      "cat 485, mi = 0.00327192, KRCTCNNNNMANAGC_UNKNOWN\n",
      "cat 643, mi = 0.00313096, SGCGSSAAA_E2F1DP2_01\n",
      "cat 786, mi = 0.00262528, TTTNNANAGCYR_UNKNOWN\n",
      "cat 314, mi = 0.00252562, FOXD3_01\n",
      "cat 350, mi = 0.00246508, GATA_Q6\n",
      "cat 159, mi = 0.00221213, CAGNWMCNNNGAC_UNKNOWN\n",
      "cat 232, mi = 0.00217490, CRGAARNNNNCGA_UNKNOWN\n",
      "cat 372, mi = 0.00193131, GGAANCGGAANY_UNKNOWN\n",
      "cat 210, mi = 0.00188640, CGTSACG_PAX3_B\n",
      "cat 378, mi = 0.00175257, GGCACTT_MIR519E\n",
      "cat 4, mi = 0.00171042, AAANWWTGC_UNKNOWN\n",
      "cat 803, mi = 0.00170407, WWTAAGGC_UNKNOWN\n",
      "cat 150, mi = 0.00165174, CAGCACT_MIR5123P\n",
      "cat 124, mi = 0.00162020, ATGAAGG_MIR205\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 559, mi = 0.004, pvalue = 0.000, NRSF_01\n",
      "Passed: cat 485, mi = 0.003, pvalue = 0.000, KRCTCNNNNMANAGC_UNKNOWN\n",
      "Passed: cat 643, mi = 0.003, pvalue = 0.000, SGCGSSAAA_E2F1DP2_01\n",
      "Passed: cat 314, mi = 0.003, pvalue = 0.000, FOXD3_01\n",
      "Passed: cat 350, mi = 0.002, pvalue = 0.000, GATA_Q6\n",
      "Passed: cat 232, mi = 0.002, pvalue = 0.000, CRGAARNNNNCGA_UNKNOWN\n",
      "Passed: cat 372, mi = 0.002, pvalue = 0.000, GGAANCGGAANY_UNKNOWN\n",
      "Passed: cat 210, mi = 0.002, pvalue = 0.000, CGTSACG_PAX3_B\n",
      "Passed: cat 378, mi = 0.002, pvalue = 0.000, GGCACTT_MIR519E\n",
      "Passed: cat 303, mi = 0.002, pvalue = 0.001, ETF_Q6\n",
      "Passed: cat 766, mi = 0.002, pvalue = 0.002, TMTCGCGANR_UNKNOWN\n",
      "Passed: cat 505, mi = 0.001, pvalue = 0.004, MEF2_04\n",
      "Didn't pass (cutoff=0.0050): cat 689, mi = 0.001, pvalue = 0.007, TAATGTG_MIR323\n",
      "Didn't pass (cutoff=0.0050): cat 53, mi = 0.001, pvalue = 0.007, AGCTCCT_MIR28\n",
      "Didn't pass (cutoff=0.0050): cat 151, mi = 0.001, pvalue = 0.020, CAGCAGG_MIR370\n",
      "Didn't pass (cutoff=0.0050): cat 411, mi = 0.001, pvalue = 0.037, GTAGGCA_MIR189\n",
      "Didn't pass (cutoff=0.0050): cat 237, mi = 0.001, pvalue = 0.027, CTAGGAA_MIR384\n",
      "Didn't pass (cutoff=0.0050): cat 395, mi = 0.001, pvalue = 0.025, GGGNRMNNYCAT_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 44, mi = 0.001, pvalue = 0.026, ACTGTAG_MIR139\n",
      "Didn't pass (cutoff=0.0050): cat 26, mi = 0.001, pvalue = 0.032, ACACTCC_MIR122A\n",
      "Didn't pass (cutoff=0.0050): cat 428, mi = 0.001, pvalue = 0.038, GTGGTGA_MIR197\n",
      "Didn't pass (cutoff=0.0050): cat 367, mi = 0.001, pvalue = 0.051, GCTCTTG_MIR335\n",
      "Didn't pass (cutoff=0.0050): cat 715, mi = 0.001, pvalue = 0.086, TCCCCAC_MIR491\n",
      "Didn't pass (cutoff=0.0050): cat 612, mi = 0.001, pvalue = 0.076, RACTNNRTTTNC_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 440, mi = 0.001, pvalue = 0.084, HEN1_01\n",
      "Didn't pass (cutoff=0.0050): cat 351, mi = 0.001, pvalue = 0.107, GATGKMRGCG_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 648, mi = 0.001, pvalue = 0.119, SNACANNNYSYAGA_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 47, mi = 0.001, pvalue = 0.115, ACTWSNACTNY_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 354, mi = 0.001, pvalue = 0.195, GCAAGAC_MIR431\n",
      "Didn't pass (cutoff=0.0050): cat 393, mi = 0.001, pvalue = 0.171, GGGGCCC_MIR296\n",
      "Didn't pass (cutoff=0.0050): cat 125, mi = 0.001, pvalue = 0.269, ATGCACG_MIR517B\n",
      "Didn't pass (cutoff=0.0050): cat 56, mi = 0.001, pvalue = 0.181, AGGAGTG_MIR483\n",
      "Total number of categories evaluated = 120.\n",
      "Number of categories that passed the tests = 12\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM2_2.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM2_2.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 8, 9, 0.226752549274484\n",
      "best merge is 1, 4, 0.244360772314443\n",
      "best merge is 1, 3, 0.360388833773727\n",
      "best merge is 2, 8, 0.425556978251793\n",
      "best merge is 2, 7, 0.501929114174065\n",
      "best merge is 1, 11, 0.630661995925915\n",
      "best merge is 5, 10, 0.644025999267177\n",
      "best merge is 1, 2, 0.754603879180642\n",
      "order classes\n",
      "4 - 1.07142166036487\n",
      "4 - 0.968870193112599\n",
      "2 - 1.93207799780153\n",
      "2 8 9 7\n",
      "1 4 3 11\n",
      "0\n",
      "5 10\n",
      "6\n",
      "Done.Start drawing\n",
      "101.03\t0.00\n",
      "SGCGSSAAA_E2F1DP2_01\n",
      "KRCTCNNNNMANAGC_UNKNOWN\n",
      "GATA_Q6\n",
      "FOXD3_01\n",
      "MEF2_04\n",
      "GGCACTT_MIR519E\n",
      "ETF_Q6\n",
      "CGTSACG_PAX3_B\n",
      "CRGAARNNNNCGA_UNKNOWN\n",
      "GGAANCGGAANY_UNKNOWN\n",
      "TMTCGCGANR_UNKNOWN\n",
      "NRSF_01\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM2_2.csv_PAGE/filt_FC_LVM2_2.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM2_2.csv_PAGE/filt_FC_LVM2_2.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM2_2.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM2_2.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 8, 9, 0.226752549274484\n",
      "best merge is 1, 4, 0.244360772314443\n",
      "best merge is 1, 3, 0.360388833773727\n",
      "best merge is 2, 8, 0.425556978251793\n",
      "best merge is 2, 7, 0.501929114174065\n",
      "best merge is 1, 11, 0.630661995925915\n",
      "best merge is 5, 10, 0.644025999267177\n",
      "best merge is 1, 2, 0.754603879180642\n",
      "order classes\n",
      "4 - 1.07142166036487\n",
      "4 - 0.968870193112599\n",
      "2 - 1.93207799780153\n",
      "2 8 9 7\n",
      "1 4 3 11\n",
      "0\n",
      "5 10\n",
      "6\n",
      "Done.-19.8544986637632\t42\t0\n",
      "-19.6570325645848\t42\t0.291002672473523\n",
      "-19.5759675343957\t42\t0.394932198356924\n",
      "-19.5219241809364\t42\t0.453132732851628\n",
      "-19.4720380085123\t42\t0.503018905275661\n",
      "-19.417994655053\t42\t0.552905077699693\n",
      "-19.3514797584876\t42\t0.611105612194398\n",
      "-19.2454716420865\t42\t0.685934870830446\n",
      "1.41156092249827\t42\t0.823121844996536\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM2_2.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM2_3.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c3.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c3/human_ensembl_msigdb_c3.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM2_3.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 836\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t836\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 836\n",
      "cat 559, mi = 0.00319388, NRSF_01\n",
      "cat 303, mi = 0.00301498, ETF_Q6\n",
      "cat 378, mi = 0.00233894, GGCACTT_MIR519E\n",
      "cat 443, mi = 0.00215222, HFH3_01\n",
      "cat 4, mi = 0.00210353, AAANWWTGC_UNKNOWN\n",
      "cat 159, mi = 0.00209844, CAGNWMCNNNGAC_UNKNOWN\n",
      "cat 485, mi = 0.00209363, KRCTCNNNNMANAGC_UNKNOWN\n",
      "cat 113, mi = 0.00206598, ATAGGAA_MIR202\n",
      "cat 483, mi = 0.00198026, KCCGNSWTTT_UNKNOWN\n",
      "cat 372, mi = 0.00189251, GGAANCGGAANY_UNKNOWN\n",
      "cat 356, mi = 0.00185600, GCACCTT_MIR18A_MIR18B\n",
      "cat 129, mi = 0.00182663, ATGGYGGA_UNKNOWN\n",
      "cat 279, mi = 0.00180269, E2F_Q2\n",
      "cat 786, mi = 0.00179013, TTTNNANAGCYR_UNKNOWN\n",
      "cat 334, mi = 0.00176358, GACTGTT_MIR212_MIR132\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 559, mi = 0.003, pvalue = 0.000, NRSF_01\n",
      "Passed: cat 303, mi = 0.003, pvalue = 0.000, ETF_Q6\n",
      "Passed: cat 378, mi = 0.002, pvalue = 0.000, GGCACTT_MIR519E\n",
      "Passed: cat 443, mi = 0.002, pvalue = 0.000, HFH3_01\n",
      "Passed: cat 485, mi = 0.002, pvalue = 0.000, KRCTCNNNNMANAGC_UNKNOWN\n",
      "Passed: cat 483, mi = 0.002, pvalue = 0.000, KCCGNSWTTT_UNKNOWN\n",
      "Passed: cat 372, mi = 0.002, pvalue = 0.000, GGAANCGGAANY_UNKNOWN\n",
      "Passed: cat 388, mi = 0.002, pvalue = 0.000, GGGACCA_MIR133A_MIR133B\n",
      "Passed: cat 44, mi = 0.002, pvalue = 0.001, ACTGTAG_MIR139\n",
      "Passed: cat 602, mi = 0.002, pvalue = 0.002, PPARG_01\n",
      "Passed: cat 612, mi = 0.001, pvalue = 0.003, RACTNNRTTTNC_UNKNOWN\n",
      "Passed: cat 210, mi = 0.001, pvalue = 0.005, CGTSACG_PAX3_B\n",
      "Didn't pass (cutoff=0.0050): cat 181, mi = 0.001, pvalue = 0.006, CCCNNNNNNAAGWT_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 350, mi = 0.001, pvalue = 0.021, GATA_Q6\n",
      "Didn't pass (cutoff=0.0050): cat 715, mi = 0.001, pvalue = 0.033, TCCCCAC_MIR491\n",
      "Didn't pass (cutoff=0.0050): cat 32, mi = 0.001, pvalue = 0.036, ACAWYAAAG_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 363, mi = 0.001, pvalue = 0.049, GCGCTTT_MIR518B_MIR518C_MIR518D\n",
      "Didn't pass (cutoff=0.0050): cat 176, mi = 0.001, pvalue = 0.042, CCAWWNAAGG_SRF_Q4\n",
      "Didn't pass (cutoff=0.0050): cat 486, mi = 0.001, pvalue = 0.064, KTGGYRSGAA_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 705, mi = 0.001, pvalue = 0.077, TATCTGG_MIR488\n",
      "Didn't pass (cutoff=0.0050): cat 312, mi = 0.001, pvalue = 0.110, EVI1_06\n",
      "Didn't pass (cutoff=0.0050): cat 351, mi = 0.001, pvalue = 0.087, GATGKMRGCG_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 451, mi = 0.001, pvalue = 0.118, HMX1_01\n",
      "Didn't pass (cutoff=0.0050): cat 727, mi = 0.001, pvalue = 0.118, TCTGATC_MIR383\n",
      "Didn't pass (cutoff=0.0050): cat 387, mi = 0.001, pvalue = 0.139, GGCNRNWCTTYS_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 636, mi = 0.001, pvalue = 0.162, RYTGCNNRGNAAC_MIF1_01\n",
      "Didn't pass (cutoff=0.0050): cat 747, mi = 0.001, pvalue = 0.241, TGCACGA_MIR517A_MIR517C\n",
      "Didn't pass (cutoff=0.0050): cat 26, mi = 0.001, pvalue = 0.254, ACACTCC_MIR122A\n",
      "Didn't pass (cutoff=0.0050): cat 206, mi = 0.001, pvalue = 0.467, CGCTGCT_MIR503\n",
      "Didn't pass (cutoff=0.0050): cat 383, mi = 0.001, pvalue = 0.312, GGCGGCA_MIR371\n",
      "Didn't pass (cutoff=0.0050): cat 393, mi = 0.001, pvalue = 0.353, GGGGCCC_MIR296\n",
      "Didn't pass (cutoff=0.0050): cat 434, mi = 0.001, pvalue = 0.368, GTTGNYNNRGNAAC_UNKNOWN\n",
      "Total number of categories evaluated = 250.\n",
      "Number of categories that passed the tests = 12\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM2_3.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM2_3.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 2, 7, 0.122321874909666\n",
      "best merge is 3, 4, 0.123489679722653\n",
      "best merge is 5, 10, 0.146269765225051\n",
      "best merge is 8, 11, 0.172124079894913\n",
      "best merge is 1, 2, 0.178709786638688\n",
      "best merge is 6, 8, 0.497949235587863\n",
      "best merge is 5, 6, 0.559280645580211\n",
      "best merge is 1, 5, 0.697958453493235\n",
      "order classes\n",
      "3 - 0.426436842832926\n",
      "2 - 0.370469039167959\n",
      "5 - 0.896635428437175\n",
      "3 4\n",
      "1 2 7\n",
      "0\n",
      "5 10 6 8 11\n",
      "9\n",
      "Done.Start drawing\n",
      "69.14\t0.00\n",
      "HFH3_01\n",
      "KRCTCNNNNMANAGC_UNKNOWN\n",
      "ETF_Q6\n",
      "GGCACTT_MIR519E\n",
      "GGGACCA_MIR133A_MIR133B\n",
      "CGTSACG_PAX3_B\n",
      "ACTGTAG_MIR139\n",
      "GGAANCGGAANY_UNKNOWN\n",
      "PPARG_01\n",
      "KCCGNSWTTT_UNKNOWN\n",
      "RACTNNRTTTNC_UNKNOWN\n",
      "NRSF_01\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM2_3.csv_PAGE/filt_FC_LVM2_3.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM2_3.csv_PAGE/filt_FC_LVM2_3.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM2_3.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM2_3.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 2, 7, 0.122321874909666\n",
      "best merge is 3, 4, 0.123489679722653\n",
      "best merge is 5, 10, 0.146269765225051\n",
      "best merge is 8, 11, 0.172124079894913\n",
      "best merge is 1, 2, 0.178709786638688\n",
      "best merge is 6, 8, 0.497949235587863\n",
      "best merge is 5, 6, 0.559280645580211\n",
      "best merge is 1, 5, 0.697958453493235\n",
      "order classes\n",
      "3 - 0.426436842832926\n",
      "2 - 0.370469039167959\n",
      "5 - 0.896635428437175\n",
      "3 4\n",
      "1 2 7\n",
      "0\n",
      "5 10 6 8 11\n",
      "9\n",
      "Done.-19.7843505929997\t42\t0\n",
      "-19.4715070870697\t42\t0.431298814000579\n",
      "-19.3166039919005\t42\t0.625687011859994\n",
      "-19.2133352617877\t42\t0.741105004339022\n",
      "-19.1191784784495\t42\t0.832224472085623\n",
      "-19.0189470639283\t42\t0.929418571015331\n",
      "-18.8974544402661\t42\t1.03268730112815\n",
      "-18.7000289268152\t42\t1.1724038183396\n",
      "1.71376916401504\t42\t1.42753832803008\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM2_3.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM2_5.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c3.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c3/human_ensembl_msigdb_c3.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM2_5.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 836\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t836\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 836\n",
      "cat 559, mi = 0.00316257, NRSF_01\n",
      "cat 159, mi = 0.00263418, CAGNWMCNNNGAC_UNKNOWN\n",
      "cat 662, mi = 0.00216710, SRF_01\n",
      "cat 31, mi = 0.00209761, ACAWNRNSRCGG_UNKNOWN\n",
      "cat 176, mi = 0.00194963, CCAWWNAAGG_SRF_Q4\n",
      "cat 485, mi = 0.00188647, KRCTCNNNNMANAGC_UNKNOWN\n",
      "cat 766, mi = 0.00171462, TMTCGCGANR_UNKNOWN\n",
      "cat 364, mi = 0.00157599, GCGNNANTTCC_UNKNOWN\n",
      "cat 483, mi = 0.00157120, KCCGNSWTTT_UNKNOWN\n",
      "cat 372, mi = 0.00153843, GGAANCGGAANY_UNKNOWN\n",
      "cat 803, mi = 0.00141498, WWTAAGGC_UNKNOWN\n",
      "cat 727, mi = 0.00139594, TCTGATC_MIR383\n",
      "cat 443, mi = 0.00139430, HFH3_01\n",
      "cat 175, mi = 0.00137052, CCAWNWWNNNGGC_UNKNOWN\n",
      "cat 728, mi = 0.00135856, TCTGGAC_MIR198\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 559, mi = 0.003, pvalue = 0.000, NRSF_01\n",
      "Passed: cat 662, mi = 0.002, pvalue = 0.000, SRF_01\n",
      "Passed: cat 31, mi = 0.002, pvalue = 0.000, ACAWNRNSRCGG_UNKNOWN\n",
      "Passed: cat 485, mi = 0.002, pvalue = 0.000, KRCTCNNNNMANAGC_UNKNOWN\n",
      "Passed: cat 766, mi = 0.002, pvalue = 0.001, TMTCGCGANR_UNKNOWN\n",
      "Passed: cat 364, mi = 0.002, pvalue = 0.001, GCGNNANTTCC_UNKNOWN\n",
      "Passed: cat 372, mi = 0.002, pvalue = 0.001, GGAANCGGAANY_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 727, mi = 0.001, pvalue = 0.005, TCTGATC_MIR383\n",
      "Passed: cat 443, mi = 0.001, pvalue = 0.003, HFH3_01\n",
      "Passed: cat 728, mi = 0.001, pvalue = 0.005, TCTGGAC_MIR198\n",
      "Didn't pass (cutoff=0.0050): cat 44, mi = 0.001, pvalue = 0.006, ACTGTAG_MIR139\n",
      "Didn't pass (cutoff=0.0050): cat 696, mi = 0.001, pvalue = 0.010, TAGAACC_MIR182\n",
      "Didn't pass (cutoff=0.0050): cat 47, mi = 0.001, pvalue = 0.007, ACTWSNACTNY_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 124, mi = 0.001, pvalue = 0.006, ATGAAGG_MIR205\n",
      "Didn't pass (cutoff=0.0050): cat 669, mi = 0.001, pvalue = 0.013, STAT1_01\n",
      "Didn't pass (cutoff=0.0050): cat 588, mi = 0.001, pvalue = 0.021, PAX8_01\n",
      "Didn't pass (cutoff=0.0050): cat 50, mi = 0.001, pvalue = 0.012, AGCATTA_MIR155\n",
      "Didn't pass (cutoff=0.0050): cat 401, mi = 0.001, pvalue = 0.020, GKCGCNNNNNNNTGAYG_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 63, mi = 0.001, pvalue = 0.020, AGTCTTA_MIR499\n",
      "Didn't pass (cutoff=0.0050): cat 637, mi = 0.001, pvalue = 0.016, RYTGCNWTGGNR_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 61, mi = 0.001, pvalue = 0.032, AGTCAGC_MIR345\n",
      "Didn't pass (cutoff=0.0050): cat 658, mi = 0.001, pvalue = 0.025, SREBP1_01\n",
      "Didn't pass (cutoff=0.0050): cat 648, mi = 0.001, pvalue = 0.029, SNACANNNYSYAGA_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 365, mi = 0.001, pvalue = 0.033, GCGSCMNTTT_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 505, mi = 0.001, pvalue = 0.031, MEF2_04\n",
      "Didn't pass (cutoff=0.0050): cat 26, mi = 0.001, pvalue = 0.043, ACACTCC_MIR122A\n",
      "Didn't pass (cutoff=0.0050): cat 600, mi = 0.001, pvalue = 0.059, PPARA_01\n",
      "Didn't pass (cutoff=0.0050): cat 8, mi = 0.001, pvalue = 0.068, AACGGTT_MIR451\n",
      "Didn't pass (cutoff=0.0050): cat 249, mi = 0.001, pvalue = 0.060, CTGRYYYNATT_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 177, mi = 0.001, pvalue = 0.069, CCAWYNNGAAR_UNKNOWN\n",
      "Total number of categories evaluated = 55.\n",
      "Number of categories that passed the tests = 9\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM2_5.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM2_5.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 5, 6, 0.117122170712219\n",
      "best merge is 0, 3, 0.16681228068917\n",
      "best merge is 2, 5, 0.209234934818587\n",
      "best merge is 2, 4, 0.319952010509874\n",
      "best merge is 1, 7, 0.497124718301117\n",
      "order classes\n",
      "2 - 0.375327631550632\n",
      "4 - 0.420594770215973\n",
      "0 3\n",
      "1\n",
      "2 5 6 4\n",
      "7\n",
      "8\n",
      "Done.Start drawing\n",
      "66.19\t0.00\n",
      "SRF_01\n",
      "GGAANCGGAANY_UNKNOWN\n",
      "GCGNNANTTCC_UNKNOWN\n",
      "TMTCGCGANR_UNKNOWN\n",
      "HFH3_01\n",
      "TCTGGAC_MIR198\n",
      "ACAWNRNSRCGG_UNKNOWN\n",
      "NRSF_01\n",
      "KRCTCNNNNMANAGC_UNKNOWN\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM2_5.csv_PAGE/filt_FC_LVM2_5.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM2_5.csv_PAGE/filt_FC_LVM2_5.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM2_5.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM2_5.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 5, 6, 0.117122170712219\n",
      "best merge is 0, 3, 0.16681228068917\n",
      "best merge is 2, 5, 0.209234934818587\n",
      "best merge is 2, 4, 0.319952010509874\n",
      "best merge is 1, 7, 0.497124718301117\n",
      "order classes\n",
      "2 - 0.375327631550632\n",
      "4 - 0.420594770215973\n",
      "0 3\n",
      "1\n",
      "2 5 6 4\n",
      "7\n",
      "8\n",
      "Done.-19.8477111346125\t42\t0\n",
      "-19.6224505212268\t42\t0.304577730775042\n",
      "-19.501888502795\t42\t0.450521226771416\n",
      "-19.4130533313189\t42\t0.545701767638616\n",
      "-19.330563529234\t42\t0.628191569723523\n",
      "-19.2480737271491\t42\t0.71068137180843\n",
      "-19.1528931862819\t42\t0.793171173893337\n",
      "-19.0196404290678\t42\t0.901042453542831\n",
      "1.52983834416075\t42\t1.0596766883215\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_LVM2_5.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_P_LVM1_2.csv --species=human_ensembl_msigdb_c3 --exptype=continuous --ebins=9\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_P_LVM1_1.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c3.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c3/human_ensembl_msigdb_c3.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_P_LVM1_1.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 836\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t836\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 836\n",
      "cat 104, mi = 0.00193642, AR_01\n",
      "cat 372, mi = 0.00170786, GGAANCGGAANY_UNKNOWN\n",
      "cat 483, mi = 0.00164661, KCCGNSWTTT_UNKNOWN\n",
      "cat 643, mi = 0.00162484, SGCGSSAAA_E2F1DP2_01\n",
      "cat 251, mi = 0.00162319, CTGYNNCTYTAA_UNKNOWN\n",
      "cat 766, mi = 0.00162203, TMTCGCGANR_UNKNOWN\n",
      "cat 386, mi = 0.00147193, GGCNNMSMYNTTG_UNKNOWN\n",
      "cat 444, mi = 0.00140453, HFH4_01\n",
      "cat 501, mi = 0.00125422, MCAATNNNNNGCG_UNKNOWN\n",
      "cat 814, mi = 0.00123920, YGTCCTTGR_UNKNOWN\n",
      "cat 175, mi = 0.00117175, CCAWNWWNNNGGC_UNKNOWN\n",
      "cat 385, mi = 0.00114718, GGCNKCCATNK_UNKNOWN\n",
      "cat 773, mi = 0.00112689, TTCNRGNNNNTTC_HSF_Q6\n",
      "cat 820, mi = 0.00111856, YTAATTAA_LHX3_01\n",
      "cat 658, mi = 0.00110131, SREBP1_01\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 104, mi = 0.002, pvalue = 0.000, AR_01\n",
      "Passed: cat 372, mi = 0.002, pvalue = 0.001, GGAANCGGAANY_UNKNOWN\n",
      "Passed: cat 483, mi = 0.002, pvalue = 0.001, KCCGNSWTTT_UNKNOWN\n",
      "Passed: cat 251, mi = 0.002, pvalue = 0.001, CTGYNNCTYTAA_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 501, mi = 0.001, pvalue = 0.010, MCAATNNNNNGCG_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 814, mi = 0.001, pvalue = 0.009, YGTCCTTGR_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 658, mi = 0.001, pvalue = 0.019, SREBP1_01\n",
      "Didn't pass (cutoff=0.0050): cat 412, mi = 0.001, pvalue = 0.030, GTATGAT_MIR154_MIR487\n",
      "Didn't pass (cutoff=0.0050): cat 761, mi = 0.001, pvalue = 0.018, TGTGTGA_MIR377\n",
      "Didn't pass (cutoff=0.0050): cat 387, mi = 0.001, pvalue = 0.032, GGCNRNWCTTYS_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 597, mi = 0.001, pvalue = 0.038, POU3F2_01\n",
      "Didn't pass (cutoff=0.0050): cat 672, mi = 0.001, pvalue = 0.051, STAT3_01\n",
      "Didn't pass (cutoff=0.0050): cat 559, mi = 0.001, pvalue = 0.054, NRSF_01\n",
      "Didn't pass (cutoff=0.0050): cat 747, mi = 0.001, pvalue = 0.069, TGCACGA_MIR517A_MIR517C\n",
      "Didn't pass (cutoff=0.0050): cat 743, mi = 0.001, pvalue = 0.061, TGASTMAGC_NFE2_01\n",
      "Didn't pass (cutoff=0.0050): cat 117, mi = 0.001, pvalue = 0.072, ATCTTGC_MIR31\n",
      "Didn't pass (cutoff=0.0050): cat 803, mi = 0.001, pvalue = 0.063, WWTAAGGC_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 428, mi = 0.001, pvalue = 0.076, GTGGTGA_MIR197\n",
      "Didn't pass (cutoff=0.0050): cat 146, mi = 0.001, pvalue = 0.098, CACGTTT_MIR302A\n",
      "Didn't pass (cutoff=0.0050): cat 818, mi = 0.001, pvalue = 0.083, YRCCAKNNGNCGC_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 56, mi = 0.001, pvalue = 0.086, AGGAGTG_MIR483\n",
      "Didn't pass (cutoff=0.0050): cat 375, mi = 0.001, pvalue = 0.097, GGATCCG_MIR127\n",
      "Didn't pass (cutoff=0.0050): cat 23, mi = 0.001, pvalue = 0.112, ACAACCT_MIR453\n",
      "Didn't pass (cutoff=0.0050): cat 430, mi = 0.001, pvalue = 0.098, GTGTGAG_MIR342\n",
      "Total number of categories evaluated = 36.\n",
      "Number of categories that passed the tests = 4\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_P_LVM1_1.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_P_LVM1_1.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 1, 2, 0.166758985972487\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "Done.Start drawing\n",
      "280.55\t0.00\n",
      "GGAANCGGAANY_UNKNOWN\n",
      "KCCGNSWTTT_UNKNOWN\n",
      "AR_01\n",
      "CTGYNNCTYTAA_UNKNOWN\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_P_LVM1_1.csv_PAGE/filt_FC_P_LVM1_1.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_P_LVM1_1.csv_PAGE/filt_FC_P_LVM1_1.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_P_LVM1_1.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_P_LVM1_1.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 1, 2, 0.166758985972487\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "Done.-19.9985029406523\t42\t0\n",
      "-19.9962573516307\t42\t0.00299411869541971\n",
      "-19.9940117626092\t42\t0.00449117804312957\n",
      "-19.9917661735876\t42\t0.00748529673854928\n",
      "-19.989520584566\t42\t0.00898235608625913\n",
      "-19.9872749955445\t42\t0.0119764747816788\n",
      "-19.9169132062021\t42\t0.0134735341293887\n",
      "-19.7559793263233\t42\t0.152700053466405\n",
      "1.1676706469435\t42\t0.335341293887008\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_P_LVM1_1.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_P_LVM1_3.csv --species=human_ensembl_msigdb_c3 --exptype=continuous --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_P_LVM1_4.csv --species=human_ensembl_msigdb_c3 --exptype=continuous --ebins=9\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_P_LVM1_2.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c3.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c3/human_ensembl_msigdb_c3.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_P_LVM1_2.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 836\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t836\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 836\n",
      "cat 485, mi = 0.00309782, KRCTCNNNNMANAGC_UNKNOWN\n",
      "cat 627, mi = 0.00226682, RRCCGTTA_UNKNOWN\n",
      "cat 507, mi = 0.00214686, MEIS1AHOXA9_01\n",
      "cat 41, mi = 0.00205996, ACTGAAA_MIR30A3P_MIR30E3P\n",
      "cat 483, mi = 0.00202438, KCCGNSWTTT_UNKNOWN\n",
      "cat 35, mi = 0.00186527, ACCATTT_MIR522\n",
      "cat 786, mi = 0.00182718, TTTNNANAGCYR_UNKNOWN\n",
      "cat 372, mi = 0.00170321, GGAANCGGAANY_UNKNOWN\n",
      "cat 25, mi = 0.00169410, ACACTAC_MIR1423P\n",
      "cat 766, mi = 0.00167620, TMTCGCGANR_UNKNOWN\n",
      "cat 364, mi = 0.00159766, GCGNNANTTCC_UNKNOWN\n",
      "cat 236, mi = 0.00159748, CTACTGT_MIR199A\n",
      "cat 419, mi = 0.00157221, GTCTTCC_MIR7\n",
      "cat 761, mi = 0.00157172, TGTGTGA_MIR377\n",
      "cat 182, mi = 0.00154516, CCGNMNNTNACG_UNKNOWN\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 485, mi = 0.003, pvalue = 0.000, KRCTCNNNNMANAGC_UNKNOWN\n",
      "Passed: cat 627, mi = 0.002, pvalue = 0.000, RRCCGTTA_UNKNOWN\n",
      "Passed: cat 507, mi = 0.002, pvalue = 0.000, MEIS1AHOXA9_01\n",
      "Passed: cat 41, mi = 0.002, pvalue = 0.000, ACTGAAA_MIR30A3P_MIR30E3P\n",
      "Passed: cat 372, mi = 0.002, pvalue = 0.000, GGAANCGGAANY_UNKNOWN\n",
      "Passed: cat 25, mi = 0.002, pvalue = 0.000, ACACTAC_MIR1423P\n",
      "Passed: cat 766, mi = 0.002, pvalue = 0.001, TMTCGCGANR_UNKNOWN\n",
      "Passed: cat 364, mi = 0.002, pvalue = 0.001, GCGNNANTTCC_UNKNOWN\n",
      "Passed: cat 659, mi = 0.002, pvalue = 0.002, SREBP1_02\n",
      "Passed: cat 61, mi = 0.001, pvalue = 0.003, AGTCAGC_MIR345\n",
      "Didn't pass (cutoff=0.0050): cat 232, mi = 0.001, pvalue = 0.006, CRGAARNNNNCGA_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 56, mi = 0.001, pvalue = 0.014, AGGAGTG_MIR483\n",
      "Didn't pass (cutoff=0.0050): cat 804, mi = 0.001, pvalue = 0.013, WYAAANNRNNNGCG_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 367, mi = 0.001, pvalue = 0.017, GCTCTTG_MIR335\n",
      "Didn't pass (cutoff=0.0050): cat 117, mi = 0.001, pvalue = 0.024, ATCTTGC_MIR31\n",
      "Didn't pass (cutoff=0.0050): cat 177, mi = 0.001, pvalue = 0.023, CCAWYNNGAAR_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 388, mi = 0.001, pvalue = 0.029, GGGACCA_MIR133A_MIR133B\n",
      "Didn't pass (cutoff=0.0050): cat 354, mi = 0.001, pvalue = 0.074, GCAAGAC_MIR431\n",
      "Didn't pass (cutoff=0.0050): cat 67, mi = 0.001, pvalue = 0.075, AHRARNT_02\n",
      "Didn't pass (cutoff=0.0050): cat 602, mi = 0.001, pvalue = 0.081, PPARG_01\n",
      "Didn't pass (cutoff=0.0050): cat 58, mi = 0.001, pvalue = 0.071, AGGGCAG_MIR18A\n",
      "Didn't pass (cutoff=0.0050): cat 600, mi = 0.001, pvalue = 0.104, PPARA_01\n",
      "Didn't pass (cutoff=0.0050): cat 624, mi = 0.001, pvalue = 0.071, RORA2_01\n",
      "Didn't pass (cutoff=0.0050): cat 559, mi = 0.001, pvalue = 0.097, NRSF_01\n",
      "Didn't pass (cutoff=0.0050): cat 465, mi = 0.001, pvalue = 0.121, HOXA3_01\n",
      "Didn't pass (cutoff=0.0050): cat 401, mi = 0.001, pvalue = 0.096, GKCGCNNNNNNNTGAYG_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 338, mi = 0.001, pvalue = 0.095, GAGCTGG_MIR337\n",
      "Didn't pass (cutoff=0.0050): cat 436, mi = 0.001, pvalue = 0.119, GTTRYCATRR_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 411, mi = 0.001, pvalue = 0.175, GTAGGCA_MIR189\n",
      "Didn't pass (cutoff=0.0050): cat 157, mi = 0.001, pvalue = 0.175, CAGGTCC_MIR492\n",
      "Total number of categories evaluated = 133.\n",
      "Number of categories that passed the tests = 10\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_P_LVM1_2.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_P_LVM1_2.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 0, 1, 0.0801163707079364\n",
      "best merge is 0, 8, 0.379211516286958\n",
      "best merge is 5, 7, 0.395899308241715\n",
      "best merge is 4, 6, 0.404327287219873\n",
      "best merge is 0, 2, 0.425207146059016\n",
      "best merge is 3, 5, 0.55765228656525\n",
      "order classes\n",
      "4 - 0.660675262955906\n",
      "2 - 1.01081821804968\n",
      "2 - 0.989748270604288\n",
      "0 1 8 2\n",
      "3\n",
      "4 6\n",
      "5 7\n",
      "9\n",
      "Done.Start drawing\n",
      "5.29\t0.00\n",
      "KRCTCNNNNMANAGC_UNKNOWN\n",
      "RRCCGTTA_UNKNOWN\n",
      "MEIS1AHOXA9_01\n",
      "ACTGAAA_MIR30A3P_MIR30E3P\n",
      "ACACTAC_MIR1423P\n",
      "SREBP1_02\n",
      "AGTCAGC_MIR345\n",
      "GCGNNANTTCC_UNKNOWN\n",
      "GGAANCGGAANY_UNKNOWN\n",
      "TMTCGCGANR_UNKNOWN\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_P_LVM1_2.csv_PAGE/filt_FC_P_LVM1_2.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_P_LVM1_2.csv_PAGE/filt_FC_P_LVM1_2.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_P_LVM1_2.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_P_LVM1_2.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 0, 1, 0.0801163707079364\n",
      "best merge is 0, 8, 0.379211516286958\n",
      "best merge is 5, 7, 0.395899308241715\n",
      "best merge is 4, 6, 0.404327287219873\n",
      "best merge is 0, 2, 0.425207146059016\n",
      "best merge is 3, 5, 0.55765228656525\n",
      "order classes\n",
      "4 - 0.660675262955906\n",
      "2 - 1.01081821804968\n",
      "2 - 0.989748270604288\n",
      "0 1 8 2\n",
      "3\n",
      "4 6\n",
      "5 7\n",
      "9\n",
      "Done.-16.8638941398866\t42\t0\n",
      "-13.2117202268431\t42\t6.27221172022684\n",
      "-12.3780718336484\t42\t7.30434782608696\n",
      "-11.8223062381853\t42\t7.93950850661626\n",
      "-11.3856332703214\t42\t8.41587901701323\n",
      "-10.9092627599244\t42\t8.81285444234405\n",
      "-10.3137996219282\t42\t9.36862003780718\n",
      "-9.44045368620038\t42\t10.0037807183365\n",
      "6.55765595463138\t42\t11.1153119092628\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_P_LVM1_2.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_P_LVM1_3.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c3.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c3/human_ensembl_msigdb_c3.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_P_LVM1_3.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 836\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t836\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 836\n",
      "cat 643, mi = 0.00682746, SGCGSSAAA_E2F1DP2_01\n",
      "cat 485, mi = 0.00423587, KRCTCNNNNMANAGC_UNKNOWN\n",
      "cat 276, mi = 0.00285844, E2F_01\n",
      "cat 766, mi = 0.00259429, TMTCGCGANR_UNKNOWN\n",
      "cat 786, mi = 0.00209229, TTTNNANAGCYR_UNKNOWN\n",
      "cat 159, mi = 0.00170062, CAGNWMCNNNGAC_UNKNOWN\n",
      "cat 486, mi = 0.00159472, KTGGYRSGAA_UNKNOWN\n",
      "cat 326, mi = 0.00157950, FREAC4_01\n",
      "cat 372, mi = 0.00151728, GGAANCGGAANY_UNKNOWN\n",
      "cat 693, mi = 0.00149636, TACAATC_MIR508\n",
      "cat 151, mi = 0.00148300, CAGCAGG_MIR370\n",
      "cat 443, mi = 0.00144968, HFH3_01\n",
      "cat 35, mi = 0.00142527, ACCATTT_MIR522\n",
      "cat 479, mi = 0.00140482, IRF2_01\n",
      "cat 19, mi = 0.00139696, AAGTCCA_MIR422B_MIR422A\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 643, mi = 0.007, pvalue = 0.000, SGCGSSAAA_E2F1DP2_01\n",
      "Passed: cat 485, mi = 0.004, pvalue = 0.000, KRCTCNNNNMANAGC_UNKNOWN\n",
      "Passed: cat 766, mi = 0.003, pvalue = 0.000, TMTCGCGANR_UNKNOWN\n",
      "Passed: cat 159, mi = 0.002, pvalue = 0.001, CAGNWMCNNNGAC_UNKNOWN\n",
      "Passed: cat 326, mi = 0.002, pvalue = 0.001, FREAC4_01\n",
      "Passed: cat 372, mi = 0.002, pvalue = 0.002, GGAANCGGAANY_UNKNOWN\n",
      "Passed: cat 693, mi = 0.001, pvalue = 0.003, TACAATC_MIR508\n",
      "Passed: cat 151, mi = 0.001, pvalue = 0.002, CAGCAGG_MIR370\n",
      "Passed: cat 35, mi = 0.001, pvalue = 0.003, ACCATTT_MIR522\n",
      "Passed: cat 479, mi = 0.001, pvalue = 0.004, IRF2_01\n",
      "Didn't pass (cutoff=0.0050): cat 19, mi = 0.001, pvalue = 0.005, AAGTCCA_MIR422B_MIR422A\n",
      "Didn't pass (cutoff=0.0050): cat 662, mi = 0.001, pvalue = 0.011, SRF_01\n",
      "Didn't pass (cutoff=0.0050): cat 377, mi = 0.001, pvalue = 0.012, GGCACAT_MIR455\n",
      "Didn't pass (cutoff=0.0050): cat 132, mi = 0.001, pvalue = 0.017, ATGTCAC_MIR489\n",
      "Didn't pass (cutoff=0.0050): cat 610, mi = 0.001, pvalue = 0.018, RAAGNYNNCTTY_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 390, mi = 0.001, pvalue = 0.037, GGGATGC_MIR3245P\n",
      "Didn't pass (cutoff=0.0050): cat 207, mi = 0.001, pvalue = 0.040, CGGAARNGGCNG_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 18, mi = 0.001, pvalue = 0.057, AAGGGAT_MIR188\n",
      "Didn't pass (cutoff=0.0050): cat 140, mi = 0.001, pvalue = 0.068, CAAGGAT_MIR362\n",
      "Didn't pass (cutoff=0.0050): cat 387, mi = 0.001, pvalue = 0.064, GGCNRNWCTTYS_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 579, mi = 0.001, pvalue = 0.093, PAX3_01\n",
      "Didn't pass (cutoff=0.0050): cat 354, mi = 0.001, pvalue = 0.118, GCAAGAC_MIR431\n",
      "Didn't pass (cutoff=0.0050): cat 577, mi = 0.001, pvalue = 0.140, PAX2_01\n",
      "Didn't pass (cutoff=0.0050): cat 602, mi = 0.001, pvalue = 0.157, PPARG_01\n",
      "Didn't pass (cutoff=0.0050): cat 111, mi = 0.001, pvalue = 0.117, ATACCTC_MIR202\n",
      "Didn't pass (cutoff=0.0050): cat 355, mi = 0.001, pvalue = 0.161, GCAAGGA_MIR502\n",
      "Didn't pass (cutoff=0.0050): cat 208, mi = 0.001, pvalue = 0.145, CGGTGTG_MIR220\n",
      "Didn't pass (cutoff=0.0050): cat 363, mi = 0.001, pvalue = 0.377, GCGCTTT_MIR518B_MIR518C_MIR518D\n",
      "Didn't pass (cutoff=0.0050): cat 672, mi = 0.001, pvalue = 0.326, STAT3_01\n",
      "Didn't pass (cutoff=0.0050): cat 307, mi = 0.001, pvalue = 0.499, EVI1_01\n",
      "Total number of categories evaluated = 135.\n",
      "Number of categories that passed the tests = 10\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_P_LVM1_3.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_P_LVM1_3.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 0, 1, 0.037157524178132\n",
      "best merge is 0, 3, 0.177513007473138\n",
      "best merge is 2, 6, 0.194061017523217\n",
      "best merge is 5, 8, 0.220301997355147\n",
      "best merge is 2, 5, 0.403699528287405\n",
      "best merge is 4, 9, 0.7245534497801\n",
      "order classes\n",
      "3 - 0.290506325277339\n",
      "4 - 0.634112852508744\n",
      "0 1 3\n",
      "2 6 5 8\n",
      "4\n",
      "7\n",
      "9\n",
      "Done.Start drawing\n",
      "11.42\t0.00\n",
      "FREAC4_01\n",
      "GGAANCGGAANY_UNKNOWN\n",
      "TMTCGCGANR_UNKNOWN\n",
      "TACAATC_MIR508\n",
      "ACCATTT_MIR522\n",
      "IRF2_01\n",
      "CAGCAGG_MIR370\n",
      "SGCGSSAAA_E2F1DP2_01\n",
      "KRCTCNNNNMANAGC_UNKNOWN\n",
      "CAGNWMCNNNGAC_UNKNOWN\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_P_LVM1_3.csv_PAGE/filt_FC_P_LVM1_3.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_P_LVM1_3.csv_PAGE/filt_FC_P_LVM1_3.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_P_LVM1_3.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_P_LVM1_3.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 0, 1, 0.037157524178132\n",
      "best merge is 0, 3, 0.177513007473138\n",
      "best merge is 2, 6, 0.194061017523217\n",
      "best merge is 5, 8, 0.220301997355147\n",
      "best merge is 2, 5, 0.403699528287405\n",
      "best merge is 4, 9, 0.7245534497801\n",
      "order classes\n",
      "3 - 0.290506325277339\n",
      "4 - 0.634112852508744\n",
      "0 1 3\n",
      "2 6 5 8\n",
      "4\n",
      "7\n",
      "9\n",
      "Done.-19.1725043782837\t42\t0\n",
      "-18.1059544658494\t42\t1.65499124343257\n",
      "-17.7381786339755\t42\t2.13309982486865\n",
      "-17.4807355516637\t42\t2.39054290718039\n",
      "-17.223292469352\t42\t2.64798598949212\n",
      "-16.9658493870403\t42\t2.90542907180385\n",
      "-16.6348511383538\t42\t3.16287215411559\n",
      "-16.0831873905429\t42\t3.56742556917688\n",
      "3.13309982486865\t42\t4.2661996497373\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_P_LVM1_3.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "done\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_P_LVM1_4.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c3.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c3/human_ensembl_msigdb_c3.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_P_LVM1_4.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 836\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t836\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 836\n",
      "cat 372, mi = 0.00174487, GGAANCGGAANY_UNKNOWN\n",
      "cat 766, mi = 0.00165576, TMTCGCGANR_UNKNOWN\n",
      "cat 377, mi = 0.00159476, GGCACAT_MIR455\n",
      "cat 523, mi = 0.00154448, MYC_Q2\n",
      "cat 365, mi = 0.00150792, GCGSCMNTTT_UNKNOWN\n",
      "cat 292, mi = 0.00138425, EGR2_01\n",
      "cat 364, mi = 0.00134380, GCGNNANTTCC_UNKNOWN\n",
      "cat 806, mi = 0.00132431, YAATNANRNNNCAG_UNKNOWN\n",
      "cat 409, mi = 0.00127623, GTACAGG_MIR486\n",
      "cat 26, mi = 0.00124775, ACACTCC_MIR122A\n",
      "cat 440, mi = 0.00121566, HEN1_01\n",
      "cat 374, mi = 0.00121554, GGARNTKYCCA_UNKNOWN\n",
      "cat 13, mi = 0.00120211, AACYNNNNTTCCS_UNKNOWN\n",
      "cat 773, mi = 0.00116821, TTCNRGNNNNTTC_HSF_Q6\n",
      "cat 126, mi = 0.00114744, ATGCAGT_MIR217\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 372, mi = 0.002, pvalue = 0.001, GGAANCGGAANY_UNKNOWN\n",
      "Passed: cat 766, mi = 0.002, pvalue = 0.000, TMTCGCGANR_UNKNOWN\n",
      "Passed: cat 377, mi = 0.002, pvalue = 0.001, GGCACAT_MIR455\n",
      "Passed: cat 523, mi = 0.002, pvalue = 0.001, MYC_Q2\n",
      "Passed: cat 365, mi = 0.002, pvalue = 0.003, GCGSCMNTTT_UNKNOWN\n",
      "Passed: cat 364, mi = 0.001, pvalue = 0.005, GCGNNANTTCC_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 26, mi = 0.001, pvalue = 0.010, ACACTCC_MIR122A\n",
      "Didn't pass (cutoff=0.0050): cat 440, mi = 0.001, pvalue = 0.010, HEN1_01\n",
      "Didn't pass (cutoff=0.0050): cat 374, mi = 0.001, pvalue = 0.013, GGARNTKYCCA_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 13, mi = 0.001, pvalue = 0.011, AACYNNNNTTCCS_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 779, mi = 0.001, pvalue = 0.019, TTGCWCAAY_CEBPB_02\n",
      "Didn't pass (cutoff=0.0050): cat 559, mi = 0.001, pvalue = 0.039, NRSF_01\n",
      "Didn't pass (cutoff=0.0050): cat 170, mi = 0.001, pvalue = 0.032, CCACACA_MIR147\n",
      "Didn't pass (cutoff=0.0050): cat 579, mi = 0.001, pvalue = 0.043, PAX3_01\n",
      "Didn't pass (cutoff=0.0050): cat 781, mi = 0.001, pvalue = 0.040, TTGGGAG_MIR150\n",
      "Didn't pass (cutoff=0.0050): cat 693, mi = 0.001, pvalue = 0.054, TACAATC_MIR508\n",
      "Didn't pass (cutoff=0.0050): cat 417, mi = 0.001, pvalue = 0.049, GTCNYYATGR_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 485, mi = 0.001, pvalue = 0.065, KRCTCNNNNMANAGC_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 399, mi = 0.001, pvalue = 0.071, GGTGAAG_MIR412\n",
      "Didn't pass (cutoff=0.0050): cat 38, mi = 0.001, pvalue = 0.076, ACGCACA_MIR210\n",
      "Didn't pass (cutoff=0.0050): cat 31, mi = 0.001, pvalue = 0.099, ACAWNRNSRCGG_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 429, mi = 0.001, pvalue = 0.112, GTGTCAA_MIR514\n",
      "Didn't pass (cutoff=0.0050): cat 602, mi = 0.001, pvalue = 0.133, PPARG_01\n",
      "Didn't pass (cutoff=0.0050): cat 326, mi = 0.001, pvalue = 0.101, FREAC4_01\n",
      "Didn't pass (cutoff=0.0050): cat 171, mi = 0.001, pvalue = 0.106, CCAGGGG_MIR331\n",
      "Didn't pass (cutoff=0.0050): cat 449, mi = 0.001, pvalue = 0.124, HMEF2_Q6\n",
      "Total number of categories evaluated = 44.\n",
      "Number of categories that passed the tests = 6\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_P_LVM1_4.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_P_LVM1_4.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 3, 4, 0.085902226456758\n",
      "best merge is 0, 3, 0.296643258604529\n",
      "order classes\n",
      "2 - 0.128853339685137\n",
      "3 4\n",
      "1\n",
      "2\n",
      "0\n",
      "5\n",
      "Done.Start drawing\n",
      "110.21\t0.00\n",
      "GGCACAT_MIR455\n",
      "MYC_Q2\n",
      "GCGSCMNTTT_UNKNOWN\n",
      "GGAANCGGAANY_UNKNOWN\n",
      "GCGNNANTTCC_UNKNOWN\n",
      "TMTCGCGANR_UNKNOWN\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_P_LVM1_4.csv_PAGE/filt_FC_P_LVM1_4.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_P_LVM1_4.csv_PAGE/filt_FC_P_LVM1_4.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_P_LVM1_4.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_P_LVM1_4.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 3, 4, 0.085902226456758\n",
      "best merge is 0, 3, 0.296643258604529\n",
      "order classes\n",
      "2 - 0.128853339685137\n",
      "3 4\n",
      "1\n",
      "2\n",
      "0\n",
      "5\n",
      "Done.-19.9980945467743\t42\t0\n",
      "-19.9923781870974\t42\t0.00381090645132021\n",
      "-19.9847563741947\t42\t0.0114327193539606\n",
      "-19.9790400145177\t42\t0.019054532256601\n",
      "-19.9733236548408\t42\t0.0228654387079212\n",
      "-19.9676072951638\t42\t0.0304872516105617\n",
      "-19.8361310225932\t42\t0.0342981580618819\n",
      "-19.4912439887488\t42\t0.293439796751656\n",
      "1.36203611287542\t42\t0.724072225750839\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c3/fc_values/filt_FC_P_LVM1_4.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "sem --wait done\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "nohup sh run.sh iPAGE_JY.sh &"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
