{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# This is used for running data vs human_msigdb_c5 annotations located in: \n",
    "### /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import pickle\n",
    "import seaborn as sns\n",
    "import scipy.stats\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### get index of file names to loop through "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "cd fc_values\n",
    "ls FC* > filelist.txt\n",
    "cd .."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### get ensembl genes represented in iPAGE database"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash \n",
    "\n",
    "cp /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c5/human_ensembl_msigdb_c5_index.txt human_ensembl_index.txt\n",
    "awk '{print $1}' human_ensembl_index.txt > ensembl_index.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### make sure that genes in file match the index "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "###################!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n",
    "###################REMEMBER TO SPECIFY INDEX USED IN THE IPAGE COMMAND!!!!!!!!\n",
    "###################!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n",
    "\n",
    "\n",
    "#####################\n",
    "#######read in data from logfc calculations\n",
    "#####################\n",
    "####define index file to loop through\n",
    "index_file = \"fc_values/filelist.txt\"\n",
    "####define matching ensg index\n",
    "GO_index_filename = \"ensembl_index.txt\"\n",
    "genes_with_GOs = []\n",
    "with open(GO_index_filename,'r') as rf:\n",
    "    for line in rf:\n",
    "        gene_name = line.split('\\n')[0]\n",
    "        genes_with_GOs.append(gene_name)\n",
    "genes_with_GOs_set = set(genes_with_GOs)\n",
    "####open file and loop through each line\n",
    "###\n",
    "iPAGE_sh = \"iPAGE_JY.sh\"\n",
    "cwd = os.getcwd()+\"/\"\n",
    "os.system(\"rm \"+iPAGE_sh)\n",
    "###\n",
    "with open(index_file,'r') as rf:\n",
    "    for line in rf:\n",
    "        #####do this to each file\n",
    "        ###read in file into pandas df\n",
    "        FC_file = \"fc_values/\"+line.split('\\n')[0]\n",
    "        FC_df = pd.read_csv(FC_file,sep=',',index_col=0)\n",
    "        FC_df.columns = ['log2FoldChange']\n",
    "        FC_df.index.names = ['Gene']\n",
    "        ###filter\n",
    "        FC_df = FC_df.loc[[x for x in FC_df.index if x in genes_with_GOs_set]]\n",
    "        ###save\n",
    "        save_file = \"fc_values/filt_\"+line.split('\\n')[0]\n",
    "        FC_df.to_csv(save_file,sep='\\t',header=True)\n",
    "        ###write final iPAGE command to an ipage .sh file\n",
    "        iPAGE_cmd = \"perl /nvme/bins/iPAGEv1.0/page.pl --expfile \"+cwd+save_file+\" --species=human_ensembl_msigdb_c5 --exptype=continuous --ebins=9 \\n\"\n",
    "        with open(iPAGE_sh, \"a\") as myfile:\n",
    "            myfile.write(iPAGE_cmd)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## then run the command from this folder:\n",
    "\n",
    "nohup sh run.sh iPAGE_JY.sh &"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM1_1.csv --species=human_ensembl_msigdb_c5 --exptype=continuous --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM1_2.csv --species=human_ensembl_msigdb_c5 --exptype=continuous --ebins=9\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM1_1.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c5.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c5/human_ensembl_msigdb_c5.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM1_1.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 5917\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t5917\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 5917\n",
      "cat 5718, mi = 0.00908166, GO_TRANSLATIONAL_INITIATION\n",
      "cat 957, mi = 0.00768792, GO_CYTOSOLIC_RIBOSOME\n",
      "cat 1292, mi = 0.00753814, GO_ESTABLISHMENT_OF_PROTEIN_LOCALIZATION_TO_ENDOPLASMIC_RETICULUM\n",
      "cat 3995, mi = 0.00728464, GO_PROTEIN_LOCALIZATION_TO_ENDOPLASMIC_RETICULUM\n",
      "cat 2928, mi = 0.00724583, GO_NUCLEAR_TRANSCRIBED_MRNA_CATABOLIC_PROCESS_NONSENSE_MEDIATED_DECAY\n",
      "cat 5245, mi = 0.00701733, GO_RIBOSOMAL_SUBUNIT\n",
      "cat 2330, mi = 0.00688124, GO_MULTI_ORGANISM_METABOLIC_PROCESS\n",
      "cat 954, mi = 0.00543160, GO_CYTOSOLIC_LARGE_RIBOSOMAL_SUBUNIT\n",
      "cat 1742, mi = 0.00482901, GO_INNER_MITOCHONDRIAL_MEMBRANE_PROTEIN_COMPLEX\n",
      "cat 2219, mi = 0.00474518, GO_MITOCHONDRIAL_PROTEIN_COMPLEX\n",
      "cat 1889, mi = 0.00473827, GO_LARGE_RIBOSOMAL_SUBUNIT\n",
      "cat 5028, mi = 0.00441530, GO_RESPIRATORY_CHAIN\n",
      "cat 4041, mi = 0.00439074, GO_PROTEIN_TARGETING_TO_MEMBRANE\n",
      "cat 5499, mi = 0.00419922, GO_SPLICEOSOMAL_COMPLEX\n",
      "cat 2224, mi = 0.00399003, GO_MITOCHONDRIAL_TRANSLATION\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 5718, mi = 0.009, pvalue = 0.000, GO_TRANSLATIONAL_INITIATION\n",
      "Passed: cat 1742, mi = 0.005, pvalue = 0.000, GO_INNER_MITOCHONDRIAL_MEMBRANE_PROTEIN_COMPLEX\n",
      "Passed: cat 5499, mi = 0.004, pvalue = 0.000, GO_SPLICEOSOMAL_COMPLEX\n",
      "Passed: cat 2224, mi = 0.004, pvalue = 0.000, GO_MITOCHONDRIAL_TRANSLATION\n",
      "Passed: cat 1066, mi = 0.004, pvalue = 0.000, GO_DNA_DEPENDENT_DNA_REPLICATION\n",
      "Passed: cat 1670, mi = 0.003, pvalue = 0.000, GO_HORMONE_ACTIVITY\n",
      "Passed: cat 3734, mi = 0.003, pvalue = 0.000, GO_POSITIVE_REGULATION_OF_PROTEIN_MODIFICATION_BY_SMALL_PROTEIN_CONJUGATION_OR_REMOVAL\n",
      "Passed: cat 3073, mi = 0.002, pvalue = 0.000, GO_OUTER_MEMBRANE\n",
      "Passed: cat 536, mi = 0.002, pvalue = 0.000, GO_CELLULAR_RESPONSE_TO_AMINO_ACID_STARVATION\n",
      "Passed: cat 1925, mi = 0.002, pvalue = 0.000, GO_LIGAND_GATED_CHANNEL_ACTIVITY\n",
      "Passed: cat 2097, mi = 0.002, pvalue = 0.000, GO_MEMBRANE_BUDDING\n",
      "Passed: cat 1738, mi = 0.002, pvalue = 0.000, GO_INNER_EAR_MORPHOGENESIS\n",
      "Passed: cat 4711, mi = 0.002, pvalue = 0.000, GO_REGULATION_OF_ODONTOGENESIS\n",
      "Passed: cat 1994, mi = 0.002, pvalue = 0.001, GO_LYMPHOCYTE_MEDIATED_IMMUNITY\n",
      "Passed: cat 5181, mi = 0.002, pvalue = 0.002, GO_RESPONSE_TO_TYPE_I_INTERFERON\n",
      "Passed: cat 5023, mi = 0.002, pvalue = 0.001, GO_REPRESSING_TRANSCRIPTION_FACTOR_BINDING\n",
      "Passed: cat 5266, mi = 0.002, pvalue = 0.001, GO_RNA_MODIFICATION\n",
      "Passed: cat 5813, mi = 0.002, pvalue = 0.002, GO_UNFOLDED_PROTEIN_BINDING\n",
      "Passed: cat 3883, mi = 0.002, pvalue = 0.002, GO_PRESYNAPTIC_MEMBRANE\n",
      "Passed: cat 3323, mi = 0.002, pvalue = 0.001, GO_POLYAMINE_METABOLIC_PROCESS\n",
      "Passed: cat 4122, mi = 0.002, pvalue = 0.001, GO_RAC_PROTEIN_SIGNAL_TRANSDUCTION\n",
      "Passed: cat 1342, mi = 0.001, pvalue = 0.002, GO_EXTRACELLULAR_MATRIX_COMPONENT\n",
      "Passed: cat 1238, mi = 0.001, pvalue = 0.003, GO_EPHRIN_RECEPTOR_SIGNALING_PATHWAY\n",
      "Passed: cat 723, mi = 0.001, pvalue = 0.000, GO_CEREBELLAR_GRANULAR_LAYER_DEVELOPMENT\n",
      "Passed: cat 1654, mi = 0.001, pvalue = 0.003, GO_HISTONE_METHYLTRANSFERASE_COMPLEX\n",
      "Passed: cat 361, mi = 0.001, pvalue = 0.004, GO_BRANCHED_CHAIN_AMINO_ACID_METABOLIC_PROCESS\n",
      "Passed: cat 3599, mi = 0.001, pvalue = 0.001, GO_POSITIVE_REGULATION_OF_LIPID_CATABOLIC_PROCESS\n",
      "Passed: cat 1862, mi = 0.001, pvalue = 0.004, GO_KERATIN_FILAMENT\n",
      "Passed: cat 746, mi = 0.001, pvalue = 0.001, GO_CHEMOKINE_MEDIATED_SIGNALING_PATHWAY\n",
      "Didn't pass (cutoff=0.0050): cat 2266, mi = 0.001, pvalue = 0.006, GO_MONOCARBOXYLIC_ACID_BINDING\n",
      "Passed: cat 3960, mi = 0.001, pvalue = 0.005, GO_PROTEIN_HYDROXYLATION\n",
      "Didn't pass (cutoff=0.0050): cat 997, mi = 0.001, pvalue = 0.013, GO_DEOXYRIBOSE_PHOSPHATE_CATABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 710, mi = 0.001, pvalue = 0.009, GO_CENTRIOLE\n",
      "Didn't pass (cutoff=0.0050): cat 5406, mi = 0.001, pvalue = 0.015, GO_SKELETAL_MUSCLE_CELL_DIFFERENTIATION\n",
      "Didn't pass (cutoff=0.0050): cat 5218, mi = 0.001, pvalue = 0.019, GO_RETROMER_COMPLEX\n",
      "Didn't pass (cutoff=0.0050): cat 1923, mi = 0.001, pvalue = 0.016, GO_LIGAND_DEPENDENT_NUCLEAR_RECEPTOR_TRANSCRIPTION_COACTIVATOR_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 1458, mi = 0.001, pvalue = 0.010, GO_GENETIC_IMPRINTING\n",
      "Didn't pass (cutoff=0.0050): cat 1817, mi = 0.001, pvalue = 0.019, GO_INTRINSIC_COMPONENT_OF_GOLGI_MEMBRANE\n",
      "Didn't pass (cutoff=0.0050): cat 3181, mi = 0.001, pvalue = 0.009, GO_PERIPHERAL_NERVOUS_SYSTEM_NEURON_DIFFERENTIATION\n",
      "Didn't pass (cutoff=0.0050): cat 214, mi = 0.001, pvalue = 0.010, GO_APICOLATERAL_PLASMA_MEMBRANE\n",
      "Didn't pass (cutoff=0.0050): cat 5371, mi = 0.001, pvalue = 0.016, GO_SH3_DOMAIN_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 127, mi = 0.001, pvalue = 0.019, GO_ALPHA_TUBULIN_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 4102, mi = 0.001, pvalue = 0.017, GO_PYRIMIDINE_NUCLEOSIDE_MONOPHOSPHATE_METABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 5310, mi = 0.001, pvalue = 0.022, GO_R_SMAD_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 5731, mi = 0.001, pvalue = 0.016, GO_TRANSMEMBRANE_RECEPTOR_PROTEIN_SERINE_THREONINE_KINASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 2878, mi = 0.001, pvalue = 0.012, GO_NITRIC_OXIDE_METABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 4454, mi = 0.001, pvalue = 0.013, GO_REGULATION_OF_GENE_EXPRESSION_BY_GENETIC_IMPRINTING\n",
      "Didn't pass (cutoff=0.0050): cat 932, mi = 0.001, pvalue = 0.028, GO_CYTOPLASMIC_MICROTUBULE\n",
      "Didn't pass (cutoff=0.0050): cat 542, mi = 0.001, pvalue = 0.020, GO_CELLULAR_RESPONSE_TO_CADMIUM_ION\n",
      "Didn't pass (cutoff=0.0050): cat 3110, mi = 0.001, pvalue = 0.017, GO_OXIDOREDUCTASE_ACTIVITY_OXIDIZING_METAL_IONS\n",
      "Didn't pass (cutoff=0.0050): cat 2985, mi = 0.001, pvalue = 0.032, GO_NUCLEOTIDE_TRANSPORT\n",
      "Total number of categories evaluated = 418.\n",
      "Number of categories that passed the tests = 30\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM1_1.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM1_1.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 12, 23, 0.0052478442730709\n",
      "best merge is 5, 12, 0.0207763187412832\n",
      "best merge is 13, 21, 0.0444739757580203\n",
      "best merge is 1, 16, 0.0584737223168321\n",
      "best merge is 5, 27, 0.0829552139236666\n",
      "best merge is 19, 25, 0.0956511774867197\n",
      "best merge is 2, 3, 0.101732079185604\n",
      "best merge is 13, 28, 0.115450127865432\n",
      "best merge is 6, 10, 0.134401221478078\n",
      "best merge is 5, 9, 0.136186395805336\n",
      "best merge is 2, 17, 0.176289257391166\n",
      "best merge is 14, 20, 0.194284166320666\n",
      "best merge is 5, 18, 0.204394241918466\n",
      "best merge is 0, 1, 0.206681618878764\n",
      "best merge is 0, 2, 0.297564942398941\n",
      "best merge is 26, 29, 0.302273884730809\n",
      "best merge is 11, 13, 0.31701388310209\n",
      "best merge is 5, 11, 0.406790430618636\n",
      "best merge is 0, 6, 0.458498211295158\n",
      "best merge is 7, 24, 0.465404502904768\n",
      "best merge is 7, 26, 0.487066619909038\n",
      "best merge is 0, 19, 0.498057654927583\n",
      "best merge is 14, 22, 0.519322498998774\n",
      "best merge is 14, 15, 0.721340010616328\n",
      "best merge is 4, 5, 0.803497811398898\n",
      "best merge is 0, 14, 0.849027796071848\n",
      "order classes\n",
      "10 - 1.03831112693885\n",
      "11 - 0.941551340915328\n",
      "4 - 2.54619831306725\n",
      "4 - 3.18463987140675\n",
      "4 5 12 23 27 9 18 11 13 21 28\n",
      "0 1 16 2 3 17 6 10 19 25\n",
      "7 24 26 29\n",
      "8\n",
      "14 20 22 15\n",
      "Done.Start drawing\n",
      "23.77\t0.00\n",
      "GO_DNA_DEPENDENT_DNA_REPLICATION\n",
      "GO_INNER_EAR_MORPHOGENESIS\n",
      "GO_EXTRACELLULAR_MATRIX_COMPONENT\n",
      "GO_CHEMOKINE_MEDIATED_SIGNALING_PATHWAY\n",
      "GO_CELLULAR_RESPONSE_TO_AMINO_ACID_STARVATION\n",
      "GO_RAC_PROTEIN_SIGNAL_TRANSDUCTION\n",
      "GO_REPRESSING_TRANSCRIPTION_FACTOR_BINDING\n",
      "GO_OUTER_MEMBRANE\n",
      "GO_HISTONE_METHYLTRANSFERASE_COMPLEX\n",
      "GO_POSITIVE_REGULATION_OF_LIPID_CATABOLIC_PROCESS\n",
      "GO_PROTEIN_HYDROXYLATION\n",
      "GO_SPLICEOSOMAL_COMPLEX\n",
      "GO_POSITIVE_REGULATION_OF_PROTEIN_MODIFICATION_BY_SMALL_PROTEIN_CONJUGATION_OR_REMOVAL\n",
      "GO_MEMBRANE_BUDDING\n",
      "GO_RESPONSE_TO_TYPE_I_INTERFERON\n",
      "GO_TRANSLATIONAL_INITIATION\n",
      "GO_INNER_MITOCHONDRIAL_MEMBRANE_PROTEIN_COMPLEX\n",
      "GO_RNA_MODIFICATION\n",
      "GO_MITOCHONDRIAL_TRANSLATION\n",
      "GO_POLYAMINE_METABOLIC_PROCESS\n",
      "GO_BRANCHED_CHAIN_AMINO_ACID_METABOLIC_PROCESS\n",
      "GO_UNFOLDED_PROTEIN_BINDING\n",
      "GO_HORMONE_ACTIVITY\n",
      "GO_REGULATION_OF_ODONTOGENESIS\n",
      "GO_CEREBELLAR_GRANULAR_LAYER_DEVELOPMENT\n",
      "GO_KERATIN_FILAMENT\n",
      "GO_LIGAND_GATED_CHANNEL_ACTIVITY\n",
      "GO_PRESYNAPTIC_MEMBRANE\n",
      "GO_LYMPHOCYTE_MEDIATED_IMMUNITY\n",
      "GO_EPHRIN_RECEPTOR_SIGNALING_PATHWAY\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM1_1.csv_PAGE/filt_FC_LVM1_1.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM1_1.csv_PAGE/filt_FC_LVM1_1.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM1_1.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM1_1.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 12, 23, 0.0052478442730709\n",
      "best merge is 5, 12, 0.0207763187412832\n",
      "best merge is 13, 21, 0.0444739757580203\n",
      "best merge is 1, 16, 0.0584737223168321\n",
      "best merge is 5, 27, 0.0829552139236666\n",
      "best merge is 19, 25, 0.0956511774867197\n",
      "best merge is 2, 3, 0.101732079185604\n",
      "best merge is 13, 28, 0.115450127865432\n",
      "best merge is 6, 10, 0.134401221478078\n",
      "best merge is 5, 9, 0.136186395805336\n",
      "best merge is 2, 17, 0.176289257391166\n",
      "best merge is 14, 20, 0.194284166320666\n",
      "best merge is 5, 18, 0.204394241918466\n",
      "best merge is 0, 1, 0.206681618878764\n",
      "best merge is 0, 2, 0.297564942398941\n",
      "best merge is 26, 29, 0.302273884730809\n",
      "best merge is 11, 13, 0.31701388310209\n",
      "best merge is 5, 11, 0.406790430618636\n",
      "best merge is 0, 6, 0.458498211295158\n",
      "best merge is 7, 24, 0.465404502904768\n",
      "best merge is 7, 26, 0.487066619909038\n",
      "best merge is 0, 19, 0.498057654927583\n",
      "best merge is 14, 22, 0.519322498998774\n",
      "best merge is 14, 15, 0.721340010616328\n",
      "best merge is 4, 5, 0.803497811398898\n",
      "best merge is 0, 14, 0.849027796071848\n",
      "order classes\n",
      "10 - 1.03831112693885\n",
      "11 - 0.941551340915328\n",
      "4 - 2.54619831306725\n",
      "4 - 3.18463987140675\n",
      "4 5 12 23 27 9 18 11 13 21 28\n",
      "0 1 16 2 3 17 6 10 19 25\n",
      "7 24 26 29\n",
      "8\n",
      "14 20 22 15\n",
      "Done.-19.2932267564156\t42\t0\n",
      "-18.4450988641144\t42\t1.4135464871687\n",
      "-18.2153975599495\t42\t1.69625578460244\n",
      "-18.056373580143\t42\t1.87294909549853\n",
      "-17.8973496003366\t42\t2.0143037442154\n",
      "-17.7206562894405\t42\t2.19099705511149\n",
      "-17.51745898191\t42\t2.36769036600757\n",
      "-17.2082456878418\t42\t2.59739167017249\n",
      "2.49305847707194\t42\t2.98611695414388\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM1_1.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM1_3.csv --species=human_ensembl_msigdb_c5 --exptype=continuous --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM1_4.csv --species=human_ensembl_msigdb_c5 --exptype=continuous --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM2_1.csv --species=human_ensembl_msigdb_c5 --exptype=continuous --ebins=9\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM1_2.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c5.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c5/human_ensembl_msigdb_c5.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM1_2.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 5917\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t5917\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 5917\n",
      "cat 5718, mi = 0.01013899, GO_TRANSLATIONAL_INITIATION\n",
      "cat 5245, mi = 0.00953739, GO_RIBOSOMAL_SUBUNIT\n",
      "cat 957, mi = 0.00937893, GO_CYTOSOLIC_RIBOSOME\n",
      "cat 1292, mi = 0.00937749, GO_ESTABLISHMENT_OF_PROTEIN_LOCALIZATION_TO_ENDOPLASMIC_RETICULUM\n",
      "cat 3995, mi = 0.00920386, GO_PROTEIN_LOCALIZATION_TO_ENDOPLASMIC_RETICULUM\n",
      "cat 2928, mi = 0.00740830, GO_NUCLEAR_TRANSCRIBED_MRNA_CATABOLIC_PROCESS_NONSENSE_MEDIATED_DECAY\n",
      "cat 5402, mi = 0.00730854, GO_SISTER_CHROMATID_SEGREGATION\n",
      "cat 954, mi = 0.00705280, GO_CYTOSOLIC_LARGE_RIBOSOMAL_SUBUNIT\n",
      "cat 775, mi = 0.00696452, GO_CHROMOSOME_CENTROMERIC_REGION\n",
      "cat 1889, mi = 0.00668053, GO_LARGE_RIBOSOMAL_SUBUNIT\n",
      "cat 2219, mi = 0.00651516, GO_MITOCHONDRIAL_PROTEIN_COMPLEX\n",
      "cat 5401, mi = 0.00628640, GO_SISTER_CHROMATID_COHESION\n",
      "cat 1742, mi = 0.00617964, GO_INNER_MITOCHONDRIAL_MEMBRANE_PROTEIN_COMPLEX\n",
      "cat 4041, mi = 0.00611530, GO_PROTEIN_TARGETING_TO_MEMBRANE\n",
      "cat 2330, mi = 0.00590350, GO_MULTI_ORGANISM_METABOLIC_PROCESS\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 5718, mi = 0.010, pvalue = 0.000, GO_TRANSLATIONAL_INITIATION\n",
      "Passed: cat 5402, mi = 0.007, pvalue = 0.000, GO_SISTER_CHROMATID_SEGREGATION\n",
      "Passed: cat 2219, mi = 0.007, pvalue = 0.000, GO_MITOCHONDRIAL_PROTEIN_COMPLEX\n",
      "Passed: cat 5499, mi = 0.005, pvalue = 0.000, GO_SPLICEOSOMAL_COMPLEX\n",
      "Passed: cat 2224, mi = 0.004, pvalue = 0.000, GO_MITOCHONDRIAL_TRANSLATION\n",
      "Passed: cat 1680, mi = 0.003, pvalue = 0.000, GO_HUMORAL_IMMUNE_RESPONSE\n",
      "Passed: cat 5888, mi = 0.002, pvalue = 0.000, GO_VOLTAGE_GATED_POTASSIUM_CHANNEL_ACTIVITY\n",
      "Passed: cat 3910, mi = 0.002, pvalue = 0.000, GO_PROTEASOME_COMPLEX\n",
      "Passed: cat 5678, mi = 0.002, pvalue = 0.000, GO_TRANSCRIPTION_ELONGATION_FROM_RNA_POLYMERASE_II_PROMOTER\n",
      "Passed: cat 5266, mi = 0.002, pvalue = 0.000, GO_RNA_MODIFICATION\n",
      "Passed: cat 875, mi = 0.002, pvalue = 0.000, GO_CORE_PROMOTER_SEQUENCE_SPECIFIC_DNA_BINDING\n",
      "Passed: cat 640, mi = 0.002, pvalue = 0.000, GO_CELL_CHEMOTAXIS\n",
      "Passed: cat 4838, mi = 0.002, pvalue = 0.000, GO_REGULATION_OF_RESPONSE_TO_EXTRACELLULAR_STIMULUS\n",
      "Passed: cat 1397, mi = 0.002, pvalue = 0.000, GO_FIBROBLAST_GROWTH_FACTOR_RECEPTOR_BINDING\n",
      "Passed: cat 5802, mi = 0.002, pvalue = 0.000, GO_UBIQUITIN_LIKE_PROTEIN_CONJUGATING_ENZYME_ACTIVITY\n",
      "Passed: cat 3201, mi = 0.002, pvalue = 0.000, GO_PHOSPHATASE_COMPLEX\n",
      "Passed: cat 4228, mi = 0.002, pvalue = 0.000, GO_REGULATION_OF_CALCIUM_MEDIATED_SIGNALING\n",
      "Passed: cat 1700, mi = 0.002, pvalue = 0.000, GO_HYDRO_LYASE_ACTIVITY\n",
      "Passed: cat 2266, mi = 0.002, pvalue = 0.000, GO_MONOCARBOXYLIC_ACID_BINDING\n",
      "Passed: cat 25, mi = 0.002, pvalue = 0.001, GO_ACID_SECRETION\n",
      "Passed: cat 3114, mi = 0.002, pvalue = 0.000, GO_OXYGEN_BINDING\n",
      "Passed: cat 4903, mi = 0.002, pvalue = 0.001, GO_REGULATION_OF_SYNAPSE_ASSEMBLY\n",
      "Passed: cat 5268, mi = 0.002, pvalue = 0.001, GO_RNA_PHOSPHODIESTER_BOND_HYDROLYSIS_ENDONUCLEOLYTIC\n",
      "Passed: cat 4127, mi = 0.002, pvalue = 0.001, GO_RAS_PROTEIN_SIGNAL_TRANSDUCTION\n",
      "Passed: cat 2137, mi = 0.002, pvalue = 0.001, GO_METALLO_SULFUR_CLUSTER_ASSEMBLY\n",
      "Passed: cat 2807, mi = 0.002, pvalue = 0.002, GO_NEURAL_PRECURSOR_CELL_PROLIFERATION\n",
      "Passed: cat 1808, mi = 0.002, pvalue = 0.004, GO_INTRA_GOLGI_VESICLE_MEDIATED_TRANSPORT\n",
      "Passed: cat 3702, mi = 0.001, pvalue = 0.002, GO_POSITIVE_REGULATION_OF_PEPTIDYL_THREONINE_PHOSPHORYLATION\n",
      "Passed: cat 4924, mi = 0.001, pvalue = 0.003, GO_REGULATION_OF_TELOMERE_MAINTENANCE_VIA_TELOMERE_LENGTHENING\n",
      "Passed: cat 930, mi = 0.001, pvalue = 0.002, GO_CYTOPLASMIC_DYNEIN_COMPLEX\n",
      "Passed: cat 508, mi = 0.001, pvalue = 0.004, GO_CELLULAR_DEFENSE_RESPONSE\n",
      "Passed: cat 4011, mi = 0.001, pvalue = 0.001, GO_PROTEIN_NEDDYLATION\n",
      "Passed: cat 5571, mi = 0.001, pvalue = 0.005, GO_SWI_SNF_SUPERFAMILY_TYPE_COMPLEX\n",
      "Didn't pass (cutoff=0.0050): cat 536, mi = 0.001, pvalue = 0.007, GO_CELLULAR_RESPONSE_TO_AMINO_ACID_STARVATION\n",
      "Passed: cat 1946, mi = 0.001, pvalue = 0.001, GO_LIPID_PARTICLE_ORGANIZATION\n",
      "Didn't pass (cutoff=0.0050): cat 464, mi = 0.001, pvalue = 0.007, GO_CARGO_RECEPTOR_ACTIVITY\n",
      "Passed: cat 2106, mi = 0.001, pvalue = 0.004, GO_MEMBRANE_LIPID_BIOSYNTHETIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 693, mi = 0.001, pvalue = 0.007, GO_CELL_REDOX_HOMEOSTASIS\n",
      "Didn't pass (cutoff=0.0050): cat 1597, mi = 0.001, pvalue = 0.006, GO_HEAT_SHOCK_PROTEIN_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 1630, mi = 0.001, pvalue = 0.008, GO_HIPPO_SIGNALING\n",
      "Passed: cat 2254, mi = 0.001, pvalue = 0.001, GO_MODIFIED_AMINO_ACID_TRANSMEMBRANE_TRANSPORTER_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 2114, mi = 0.001, pvalue = 0.009, GO_MEMBRANE_PROTEIN_PROTEOLYSIS\n",
      "Didn't pass (cutoff=0.0050): cat 1862, mi = 0.001, pvalue = 0.008, GO_KERATIN_FILAMENT\n",
      "Didn't pass (cutoff=0.0050): cat 88, mi = 0.001, pvalue = 0.008, GO_ADENYLATE_CYCLASE_INHIBITING_G_PROTEIN_COUPLED_RECEPTOR_SIGNALING_PATHWAY\n",
      "Didn't pass (cutoff=0.0050): cat 5407, mi = 0.001, pvalue = 0.010, GO_SKELETAL_MUSCLE_CONTRACTION\n",
      "Didn't pass (cutoff=0.0050): cat 1779, mi = 0.001, pvalue = 0.011, GO_INTERMEDIATE_FILAMENT\n",
      "Didn't pass (cutoff=0.0050): cat 4161, mi = 0.001, pvalue = 0.014, GO_REGULATION_OF_ACTIN_FILAMENT_DEPOLYMERIZATION\n",
      "Didn't pass (cutoff=0.0050): cat 1752, mi = 0.001, pvalue = 0.006, GO_INOSITOL_PHOSPHATE_CATABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 2112, mi = 0.001, pvalue = 0.015, GO_MEMBRANE_PROTEIN_ECTODOMAIN_PROTEOLYSIS\n",
      "Didn't pass (cutoff=0.0050): cat 1511, mi = 0.001, pvalue = 0.007, GO_GLYCINE_METABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 210, mi = 0.001, pvalue = 0.012, GO_APICAL_JUNCTION_COMPLEX\n",
      "Didn't pass (cutoff=0.0050): cat 4089, mi = 0.001, pvalue = 0.015, GO_PYRIDINE_CONTAINING_COMPOUND_BIOSYNTHETIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 1483, mi = 0.001, pvalue = 0.015, GO_GLUCOSE_6_PHOSPHATE_METABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 1412, mi = 0.001, pvalue = 0.018, GO_FOLIC_ACID_CONTAINING_COMPOUND_METABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 5356, mi = 0.001, pvalue = 0.021, GO_SERINE_FAMILY_AMINO_ACID_METABOLIC_PROCESS\n",
      "Passed: cat 3922, mi = 0.001, pvalue = 0.003, GO_PROTEIN_BINDING_INVOLVED_IN_CELL_CELL_ADHESION\n",
      "Didn't pass (cutoff=0.0050): cat 5554, mi = 0.001, pvalue = 0.022, GO_SULFOTRANSFERASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 3943, mi = 0.001, pvalue = 0.027, GO_PROTEIN_DISULFIDE_OXIDOREDUCTASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 241, mi = 0.001, pvalue = 0.022, GO_ASPARTATE_FAMILY_AMINO_ACID_METABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 233, mi = 0.001, pvalue = 0.020, GO_ARP2_3_COMPLEX_MEDIATED_ACTIN_NUCLEATION\n",
      "Didn't pass (cutoff=0.0050): cat 657, mi = 0.001, pvalue = 0.025, GO_CELL_DIFFERENTIATION_INVOLVED_IN_EMBRYONIC_PLACENTA_DEVELOPMENT\n",
      "Didn't pass (cutoff=0.0050): cat 3011, mi = 0.001, pvalue = 0.017, GO_OMEGA_PEPTIDASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 1859, mi = 0.001, pvalue = 0.020, GO_KERATINIZATION\n",
      "Didn't pass (cutoff=0.0050): cat 1656, mi = 0.001, pvalue = 0.022, GO_HISTONE_MRNA_CATABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 2941, mi = 0.001, pvalue = 0.029, GO_NUCLEOBASE_CONTAINING_SMALL_MOLECULE_INTERCONVERSION\n",
      "Didn't pass (cutoff=0.0050): cat 2315, mi = 0.001, pvalue = 0.031, GO_MULTICELLULAR_ORGANISMAL_MOVEMENT\n",
      "Didn't pass (cutoff=0.0050): cat 4481, mi = 0.001, pvalue = 0.014, GO_REGULATION_OF_HAIR_FOLLICLE_DEVELOPMENT\n",
      "Didn't pass (cutoff=0.0050): cat 143, mi = 0.001, pvalue = 0.035, GO_AMINOPEPTIDASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 3185, mi = 0.001, pvalue = 0.024, GO_PEROXISOME_PROLIFERATOR_ACTIVATED_RECEPTOR_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 3804, mi = 0.001, pvalue = 0.039, GO_POSITIVE_REGULATION_OF_TOR_SIGNALING\n",
      "Didn't pass (cutoff=0.0050): cat 285, mi = 0.001, pvalue = 0.014, GO_AXONEMAL_DYNEIN_COMPLEX_ASSEMBLY\n",
      "Didn't pass (cutoff=0.0050): cat 1196, mi = 0.001, pvalue = 0.018, GO_ENDOPLASMIC_RETICULUM_CHAPERONE_COMPLEX\n",
      "Didn't pass (cutoff=0.0050): cat 4383, mi = 0.001, pvalue = 0.034, GO_REGULATION_OF_DNA_RECOMBINATION\n",
      "Didn't pass (cutoff=0.0050): cat 152, mi = 0.001, pvalue = 0.014, GO_AMINO_SUGAR_BIOSYNTHETIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 2048, mi = 0.001, pvalue = 0.015, GO_MALE_MEIOSIS_I\n",
      "Didn't pass (cutoff=0.0050): cat 1049, mi = 0.001, pvalue = 0.044, GO_DISULFIDE_OXIDOREDUCTASE_ACTIVITY\n",
      "Total number of categories evaluated = 683.\n",
      "Number of categories that passed the tests = 37\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM1_2.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM1_2.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 11, 20, 0.0186537847460208\n",
      "best merge is 7, 12, 0.0278861958208704\n",
      "best merge is 11, 13, 0.029873471542103\n",
      "best merge is 21, 36, 0.0389975847187496\n",
      "best merge is 11, 21, 0.0601389647910748\n",
      "best merge is 3, 7, 0.0651457837717133\n",
      "best merge is 5, 6, 0.0743952199802784\n",
      "best merge is 11, 19, 0.0842054963874696\n",
      "best merge is 0, 17, 0.109152107455912\n",
      "best merge is 14, 15, 0.12972979765357\n",
      "best merge is 0, 2, 0.149675543646246\n",
      "best merge is 5, 11, 0.162906798310565\n",
      "best merge is 27, 33, 0.17533431886939\n",
      "best merge is 3, 4, 0.178472009725652\n",
      "best merge is 1, 23, 0.191195099158667\n",
      "best merge is 8, 26, 0.20162017194658\n",
      "best merge is 24, 31, 0.240440962972062\n",
      "best merge is 9, 22, 0.240654265111348\n",
      "best merge is 10, 14, 0.266227705770157\n",
      "best merge is 3, 8, 0.331008340235601\n",
      "best merge is 10, 27, 0.352586794041688\n",
      "best merge is 5, 18, 0.375149099041071\n",
      "best merge is 0, 9, 0.422818687620827\n",
      "best merge is 29, 32, 0.444326413162644\n",
      "best merge is 5, 35, 0.468960147134373\n",
      "best merge is 1, 16, 0.481566344857805\n",
      "best merge is 3, 28, 0.539542579485109\n",
      "best merge is 1, 25, 0.574627847553499\n",
      "best merge is 3, 24, 0.644743245888819\n",
      "best merge is 3, 29, 0.658786514550419\n",
      "best merge is 1, 30, 0.702628579365791\n",
      "best merge is 0, 3, 0.785253900526703\n",
      "best merge is 1, 10, 0.830128047378323\n",
      "order classes\n",
      "16 - 1.35001040686422\n",
      "5 - 3.36772558420678\n",
      "10 - 0.751127416961354\n",
      "5 - 1.74819985304953\n",
      "5 6 11 20 13 21 36 19 18 35\n",
      "1 23 16 25 30\n",
      "0 17 2 9 22 3 7 12 4 8 26 28 24 31 29 32\n",
      "10 14 15 27 33\n",
      "34\n",
      "Done.Start drawing\n",
      "62.96\t0.00\n",
      "GO_SISTER_CHROMATID_SEGREGATION\n",
      "GO_RAS_PROTEIN_SIGNAL_TRANSDUCTION\n",
      "GO_REGULATION_OF_CALCIUM_MEDIATED_SIGNALING\n",
      "GO_NEURAL_PRECURSOR_CELL_PROLIFERATION\n",
      "GO_CELLULAR_DEFENSE_RESPONSE\n",
      "GO_MEMBRANE_LIPID_BIOSYNTHETIC_PROCESS\n",
      "GO_PROTEASOME_COMPLEX\n",
      "GO_CORE_PROMOTER_SEQUENCE_SPECIFIC_DNA_BINDING\n",
      "GO_UBIQUITIN_LIKE_PROTEIN_CONJUGATING_ENZYME_ACTIVITY\n",
      "GO_PHOSPHATASE_COMPLEX\n",
      "GO_POSITIVE_REGULATION_OF_PEPTIDYL_THREONINE_PHOSPHORYLATION\n",
      "GO_LIPID_PARTICLE_ORGANIZATION\n",
      "GO_TRANSCRIPTION_ELONGATION_FROM_RNA_POLYMERASE_II_PROMOTER\n",
      "GO_CYTOPLASMIC_DYNEIN_COMPLEX\n",
      "GO_SPLICEOSOMAL_COMPLEX\n",
      "GO_METALLO_SULFUR_CLUSTER_ASSEMBLY\n",
      "GO_PROTEIN_NEDDYLATION\n",
      "GO_SWI_SNF_SUPERFAMILY_TYPE_COMPLEX\n",
      "GO_RNA_MODIFICATION\n",
      "GO_REGULATION_OF_RESPONSE_TO_EXTRACELLULAR_STIMULUS\n",
      "GO_INTRA_GOLGI_VESICLE_MEDIATED_TRANSPORT\n",
      "GO_REGULATION_OF_TELOMERE_MAINTENANCE_VIA_TELOMERE_LENGTHENING\n",
      "GO_TRANSLATIONAL_INITIATION\n",
      "GO_HYDRO_LYASE_ACTIVITY\n",
      "GO_MITOCHONDRIAL_PROTEIN_COMPLEX\n",
      "GO_MITOCHONDRIAL_TRANSLATION\n",
      "GO_RNA_PHOSPHODIESTER_BOND_HYDROLYSIS_ENDONUCLEOLYTIC\n",
      "GO_HUMORAL_IMMUNE_RESPONSE\n",
      "GO_VOLTAGE_GATED_POTASSIUM_CHANNEL_ACTIVITY\n",
      "GO_CELL_CHEMOTAXIS\n",
      "GO_OXYGEN_BINDING\n",
      "GO_FIBROBLAST_GROWTH_FACTOR_RECEPTOR_BINDING\n",
      "GO_REGULATION_OF_SYNAPSE_ASSEMBLY\n",
      "GO_PROTEIN_BINDING_INVOLVED_IN_CELL_CELL_ADHESION\n",
      "GO_ACID_SECRETION\n",
      "GO_MONOCARBOXYLIC_ACID_BINDING\n",
      "GO_MODIFIED_AMINO_ACID_TRANSMEMBRANE_TRANSPORTER_ACTIVITY\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM1_2.csv_PAGE/filt_FC_LVM1_2.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM1_2.csv_PAGE/filt_FC_LVM1_2.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM1_2.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM1_2.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 11, 20, 0.0186537847460208\n",
      "best merge is 7, 12, 0.0278861958208704\n",
      "best merge is 11, 13, 0.029873471542103\n",
      "best merge is 21, 36, 0.0389975847187496\n",
      "best merge is 11, 21, 0.0601389647910748\n",
      "best merge is 3, 7, 0.0651457837717133\n",
      "best merge is 5, 6, 0.0743952199802784\n",
      "best merge is 11, 19, 0.0842054963874696\n",
      "best merge is 0, 17, 0.109152107455912\n",
      "best merge is 14, 15, 0.12972979765357\n",
      "best merge is 0, 2, 0.149675543646246\n",
      "best merge is 5, 11, 0.162906798310565\n",
      "best merge is 27, 33, 0.17533431886939\n",
      "best merge is 3, 4, 0.178472009725652\n",
      "best merge is 1, 23, 0.191195099158667\n",
      "best merge is 8, 26, 0.20162017194658\n",
      "best merge is 24, 31, 0.240440962972062\n",
      "best merge is 9, 22, 0.240654265111348\n",
      "best merge is 10, 14, 0.266227705770157\n",
      "best merge is 3, 8, 0.331008340235601\n",
      "best merge is 10, 27, 0.352586794041688\n",
      "best merge is 5, 18, 0.375149099041071\n",
      "best merge is 0, 9, 0.422818687620827\n",
      "best merge is 29, 32, 0.444326413162644\n",
      "best merge is 5, 35, 0.468960147134373\n",
      "best merge is 1, 16, 0.481566344857805\n",
      "best merge is 3, 28, 0.539542579485109\n",
      "best merge is 1, 25, 0.574627847553499\n",
      "best merge is 3, 24, 0.644743245888819\n",
      "best merge is 3, 29, 0.658786514550419\n",
      "best merge is 1, 30, 0.702628579365791\n",
      "best merge is 0, 3, 0.785253900526703\n",
      "best merge is 1, 10, 0.830128047378323\n",
      "order classes\n",
      "16 - 1.35001040686422\n",
      "5 - 3.36772558420678\n",
      "10 - 0.751127416961354\n",
      "5 - 1.74819985304953\n",
      "5 6 11 20 13 21 36 19 18 35\n",
      "1 23 16 25 30\n",
      "0 17 2 9 22 3 7 12 4 8 26 28 24 31 29 32\n",
      "10 14 15 27 33\n",
      "34\n",
      "Done.-19.7198221092757\t42\t0\n",
      "-19.3562579415502\t42\t0.560355781448539\n",
      "-19.2195044472681\t42\t0.72712833545108\n",
      "-19.1227763659466\t42\t0.833862770012706\n",
      "-19.0360546378653\t42\t0.920584498094028\n",
      "-18.9459974587039\t42\t1.00730622617535\n",
      "-18.8359275730623\t42\t1.10069885641677\n",
      "-18.6558132147395\t42\t1.2274459974587\n",
      "1.73046378653113\t42\t1.46092757306226\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM1_2.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM1_3.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c5.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c5/human_ensembl_msigdb_c5.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM1_3.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 5917\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t5917\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 5917\n",
      "cat 5245, mi = 0.00970964, GO_RIBOSOMAL_SUBUNIT\n",
      "cat 957, mi = 0.00802317, GO_CYTOSOLIC_RIBOSOME\n",
      "cat 2928, mi = 0.00792258, GO_NUCLEAR_TRANSCRIBED_MRNA_CATABOLIC_PROCESS_NONSENSE_MEDIATED_DECAY\n",
      "cat 1292, mi = 0.00787130, GO_ESTABLISHMENT_OF_PROTEIN_LOCALIZATION_TO_ENDOPLASMIC_RETICULUM\n",
      "cat 3995, mi = 0.00750510, GO_PROTEIN_LOCALIZATION_TO_ENDOPLASMIC_RETICULUM\n",
      "cat 854, mi = 0.00742650, GO_CONDENSED_CHROMOSOME\n",
      "cat 5718, mi = 0.00735425, GO_TRANSLATIONAL_INITIATION\n",
      "cat 1889, mi = 0.00671222, GO_LARGE_RIBOSOMAL_SUBUNIT\n",
      "cat 5402, mi = 0.00666214, GO_SISTER_CHROMATID_SEGREGATION\n",
      "cat 2330, mi = 0.00606449, GO_MULTI_ORGANISM_METABOLIC_PROCESS\n",
      "cat 4041, mi = 0.00573502, GO_PROTEIN_TARGETING_TO_MEMBRANE\n",
      "cat 1086, mi = 0.00526906, GO_DNA_PACKAGING\n",
      "cat 855, mi = 0.00523244, GO_CONDENSED_CHROMOSOME_CENTROMERIC_REGION\n",
      "cat 5401, mi = 0.00516008, GO_SISTER_CHROMATID_COHESION\n",
      "cat 5719, mi = 0.00499723, GO_TRANSLATIONAL_TERMINATION\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 5245, mi = 0.010, pvalue = 0.000, GO_RIBOSOMAL_SUBUNIT\n",
      "Passed: cat 854, mi = 0.007, pvalue = 0.000, GO_CONDENSED_CHROMOSOME\n",
      "Passed: cat 2217, mi = 0.005, pvalue = 0.000, GO_MITOCHONDRIAL_MEMBRANE_PART\n",
      "Passed: cat 1102, mi = 0.003, pvalue = 0.000, GO_DNA_STRAND_ELONGATION_INVOLVED_IN_DNA_REPLICATION\n",
      "Passed: cat 5062, mi = 0.003, pvalue = 0.000, GO_RESPONSE_TO_CORTICOSTEROID\n",
      "Passed: cat 2187, mi = 0.003, pvalue = 0.000, GO_MICROTUBULE_ORGANIZING_CENTER_PART\n",
      "Passed: cat 5499, mi = 0.003, pvalue = 0.000, GO_SPLICEOSOMAL_COMPLEX\n",
      "Passed: cat 5755, mi = 0.003, pvalue = 0.000, GO_TRNA_METABOLIC_PROCESS\n",
      "Passed: cat 3910, mi = 0.003, pvalue = 0.000, GO_PROTEASOME_COMPLEX\n",
      "Passed: cat 5678, mi = 0.002, pvalue = 0.000, GO_TRANSCRIPTION_ELONGATION_FROM_RNA_POLYMERASE_II_PROMOTER\n",
      "Passed: cat 2861, mi = 0.002, pvalue = 0.000, GO_NEUROTRANSMITTER_RECEPTOR_ACTIVITY\n",
      "Passed: cat 4349, mi = 0.002, pvalue = 0.000, GO_REGULATION_OF_CYTOKINE_BIOSYNTHETIC_PROCESS\n",
      "Passed: cat 1867, mi = 0.002, pvalue = 0.000, GO_KIDNEY_MORPHOGENESIS\n",
      "Passed: cat 25, mi = 0.002, pvalue = 0.000, GO_ACID_SECRETION\n",
      "Passed: cat 4272, mi = 0.002, pvalue = 0.000, GO_REGULATION_OF_CELLULAR_RESPONSE_TO_TRANSFORMING_GROWTH_FACTOR_BETA_STIMULUS\n",
      "Passed: cat 5568, mi = 0.002, pvalue = 0.000, GO_SUPEROXIDE_METABOLIC_PROCESS\n",
      "Passed: cat 836, mi = 0.002, pvalue = 0.000, GO_COFACTOR_BIOSYNTHETIC_PROCESS\n",
      "Passed: cat 2317, mi = 0.002, pvalue = 0.001, GO_MULTICELLULAR_ORGANISMAL_SIGNALING\n",
      "Passed: cat 5722, mi = 0.002, pvalue = 0.001, GO_TRANSLATION_INITIATION_FACTOR_ACTIVITY\n",
      "Passed: cat 1283, mi = 0.002, pvalue = 0.001, GO_ESTABLISHMENT_OF_EPITHELIAL_CELL_POLARITY\n",
      "Passed: cat 3317, mi = 0.002, pvalue = 0.002, GO_PLATELET_DERIVED_GROWTH_FACTOR_RECEPTOR_SIGNALING_PATHWAY\n",
      "Passed: cat 4818, mi = 0.002, pvalue = 0.002, GO_REGULATION_OF_RECEPTOR_INTERNALIZATION\n",
      "Passed: cat 879, mi = 0.001, pvalue = 0.002, GO_CORTICAL_ACTIN_CYTOSKELETON\n",
      "Passed: cat 4111, mi = 0.001, pvalue = 0.003, GO_PYRIMIDINE_RIBONUCLEOTIDE_METABOLIC_PROCESS\n",
      "Passed: cat 2087, mi = 0.001, pvalue = 0.002, GO_MEGAKARYOCYTE_DIFFERENTIATION\n",
      "Passed: cat 2966, mi = 0.001, pvalue = 0.004, GO_NUCLEOSOMAL_DNA_BINDING\n",
      "Passed: cat 5585, mi = 0.001, pvalue = 0.003, GO_SYNAPTIC_VESICLE_CYTOSKELETAL_TRANSPORT\n",
      "Passed: cat 5831, mi = 0.001, pvalue = 0.002, GO_VACUOLE_ORGANIZATION\n",
      "Passed: cat 5744, mi = 0.001, pvalue = 0.005, GO_TRANS_GOLGI_NETWORK_MEMBRANE\n",
      "Passed: cat 657, mi = 0.001, pvalue = 0.004, GO_CELL_DIFFERENTIATION_INVOLVED_IN_EMBRYONIC_PLACENTA_DEVELOPMENT\n",
      "Didn't pass (cutoff=0.0050): cat 926, mi = 0.001, pvalue = 0.005, GO_CYTOKINE_RECEPTOR_ACTIVITY\n",
      "Passed: cat 1735, mi = 0.001, pvalue = 0.002, GO_INNERVATION\n",
      "Passed: cat 4645, mi = 0.001, pvalue = 0.004, GO_REGULATION_OF_MRNA_CATABOLIC_PROCESS\n",
      "Passed: cat 2082, mi = 0.001, pvalue = 0.001, GO_MCM_COMPLEX\n",
      "Passed: cat 4082, mi = 0.001, pvalue = 0.003, GO_PURINE_NUCLEOBASE_BIOSYNTHETIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 5610, mi = 0.001, pvalue = 0.008, GO_TELOMERE_CAPPING\n",
      "Didn't pass (cutoff=0.0050): cat 1501, mi = 0.001, pvalue = 0.009, GO_GLUTATHIONE_METABOLIC_PROCESS\n",
      "Passed: cat 2887, mi = 0.001, pvalue = 0.004, GO_NONRIBOSOMAL_PEPTIDE_BIOSYNTHETIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 261, mi = 0.001, pvalue = 0.006, GO_ATP_DEPENDENT_MICROTUBULE_MOTOR_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 1043, mi = 0.001, pvalue = 0.012, GO_DIOXYGENASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 3258, mi = 0.001, pvalue = 0.010, GO_PHOSPHOLIPID_DEPHOSPHORYLATION\n",
      "Didn't pass (cutoff=0.0050): cat 5424, mi = 0.001, pvalue = 0.014, GO_SMALL_SUBUNIT_PROCESSOME\n",
      "Passed: cat 2745, mi = 0.001, pvalue = 0.003, GO_NEGATIVE_REGULATION_OF_SMOOTH_MUSCLE_CELL_MIGRATION\n",
      "Passed: cat 893, mi = 0.001, pvalue = 0.004, GO_CULLIN_FAMILY_PROTEIN_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 5872, mi = 0.001, pvalue = 0.010, GO_VIRUS_RECEPTOR_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 3184, mi = 0.001, pvalue = 0.022, GO_PEROXISOME_ORGANIZATION\n",
      "Didn't pass (cutoff=0.0050): cat 136, mi = 0.001, pvalue = 0.017, GO_AMINE_BIOSYNTHETIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 3152, mi = 0.001, pvalue = 0.020, GO_PEPTIDE_RECEPTOR_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 760, mi = 0.001, pvalue = 0.021, GO_CHONDROITIN_SULFATE_PROTEOGLYCAN_METABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 3238, mi = 0.001, pvalue = 0.029, GO_PHOSPHATIDYLINOSITOL_DEPHOSPHORYLATION\n",
      "Didn't pass (cutoff=0.0050): cat 4789, mi = 0.001, pvalue = 0.027, GO_REGULATION_OF_PROTEIN_LOCALIZATION_TO_CELL_SURFACE\n",
      "Didn't pass (cutoff=0.0050): cat 3242, mi = 0.001, pvalue = 0.017, GO_PHOSPHATIDYLINOSITOL_PHOSPHATE_5_PHOSPHATASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 4326, mi = 0.001, pvalue = 0.010, GO_REGULATION_OF_CHOLESTEROL_STORAGE\n",
      "Didn't pass (cutoff=0.0050): cat 2944, mi = 0.001, pvalue = 0.036, GO_NUCLEOCYTOPLASMIC_TRANSPORTER_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 3281, mi = 0.001, pvalue = 0.039, GO_PHOTORECEPTOR_OUTER_SEGMENT\n",
      "Didn't pass (cutoff=0.0050): cat 5435, mi = 0.001, pvalue = 0.041, GO_SNORNA_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 3086, mi = 0.001, pvalue = 0.025, GO_OXIDIZED_NAD_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 3208, mi = 0.001, pvalue = 0.023, GO_PHOSPHATIDATE_PHOSPHATASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 759, mi = 0.001, pvalue = 0.045, GO_CHONDROITIN_SULFATE_PROTEOGLYCAN_BIOSYNTHETIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 4921, mi = 0.001, pvalue = 0.042, GO_REGULATION_OF_TELOMERASE_RNA_LOCALIZATION_TO_CAJAL_BODY\n",
      "Didn't pass (cutoff=0.0050): cat 3116, mi = 0.001, pvalue = 0.047, GO_O_ACYLTRANSFERASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 2312, mi = 0.001, pvalue = 0.035, GO_MUCOPOLYSACCHARIDE_METABOLIC_PROCESS\n",
      "Passed: cat 1006, mi = 0.001, pvalue = 0.001, GO_DETECTION_OF_CHEMICAL_STIMULUS_INVOLVED_IN_SENSORY_PERCEPTION_OF_TASTE\n",
      "Passed: cat 1572, mi = 0.001, pvalue = 0.001, GO_G_PROTEIN_COUPLED_CHEMOATTRACTANT_RECEPTOR_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 3400, mi = 0.001, pvalue = 0.034, GO_POSITIVE_REGULATION_OF_CARDIAC_MUSCLE_CELL_PROLIFERATION\n",
      "Didn't pass (cutoff=0.0050): cat 1397, mi = 0.001, pvalue = 0.039, GO_FIBROBLAST_GROWTH_FACTOR_RECEPTOR_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 2780, mi = 0.001, pvalue = 0.038, GO_NEGATIVE_REGULATION_OF_TUMOR_NECROSIS_FACTOR_MEDIATED_SIGNALING_PATHWAY\n",
      "Didn't pass (cutoff=0.0050): cat 3323, mi = 0.001, pvalue = 0.054, GO_POLYAMINE_METABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 1200, mi = 0.001, pvalue = 0.058, GO_ENDOPLASMIC_RETICULUM_ORGANIZATION\n",
      "Didn't pass (cutoff=0.0050): cat 845, mi = 0.001, pvalue = 0.061, GO_COLLAGEN_TRIMER\n",
      "Didn't pass (cutoff=0.0050): cat 5527, mi = 0.001, pvalue = 0.040, GO_STEROL_METABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 3981, mi = 0.001, pvalue = 0.066, GO_PROTEIN_LIPID_COMPLEX\n",
      "Didn't pass (cutoff=0.0050): cat 2004, mi = 0.001, pvalue = 0.063, GO_LYSOPHOSPHOLIPID_ACYLTRANSFERASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 5851, mi = 0.001, pvalue = 0.063, GO_VENTRICULAR_SYSTEM_DEVELOPMENT\n",
      "Didn't pass (cutoff=0.0050): cat 757, mi = 0.001, pvalue = 0.058, GO_CHONDROITIN_SULFATE_BIOSYNTHETIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 1364, mi = 0.001, pvalue = 0.063, GO_FATTY_ACID_BETA_OXIDATION_USING_ACYL_COA_DEHYDROGENASE\n",
      "Didn't pass (cutoff=0.0050): cat 80, mi = 0.001, pvalue = 0.070, GO_ADA2_GCN5_ADA3_TRANSCRIPTION_ACTIVATOR_COMPLEX\n",
      "Didn't pass (cutoff=0.0050): cat 157, mi = 0.001, pvalue = 0.037, GO_AMMONIUM_TRANSMEMBRANE_TRANSPORT\n",
      "Didn't pass (cutoff=0.0050): cat 5133, mi = 0.001, pvalue = 0.065, GO_RESPONSE_TO_MISFOLDED_PROTEIN\n",
      "Didn't pass (cutoff=0.0050): cat 5855, mi = 0.001, pvalue = 0.052, GO_VESICLE_COATING\n",
      "Didn't pass (cutoff=0.0050): cat 5813, mi = 0.001, pvalue = 0.053, GO_UNFOLDED_PROTEIN_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 4125, mi = 0.001, pvalue = 0.067, GO_RAN_GTPASE_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 78, mi = 0.001, pvalue = 0.082, GO_ACYLGLYCEROL_O_ACYLTRANSFERASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 5358, mi = 0.001, pvalue = 0.045, GO_SERINE_TYPE_CARBOXYPEPTIDASE_ACTIVITY\n",
      "Total number of categories evaluated = 1028.\n",
      "Number of categories that passed the tests = 39\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM1_3.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM1_3.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 37, 38, 0\n",
      "best merge is 3, 5, 0.0188948603484824\n",
      "best merge is 6, 7, 0.0387915308459444\n",
      "best merge is 13, 30, 0.0563883752403964\n",
      "best merge is 1, 3, 0.0638397060608967\n",
      "best merge is 14, 35, 0.0853186660356534\n",
      "best merge is 8, 9, 0.0899606030584174\n",
      "best merge is 1, 13, 0.0982299363135472\n",
      "best merge is 2, 6, 0.102467101480649\n",
      "best merge is 4, 19, 0.118056971979675\n",
      "best merge is 8, 16, 0.119805583069857\n",
      "best merge is 14, 37, 0.124370850503396\n",
      "best merge is 26, 36, 0.138475868593457\n",
      "best merge is 0, 33, 0.169163473873044\n",
      "best merge is 11, 14, 0.185496561577424\n",
      "best merge is 2, 8, 0.211924277414762\n",
      "best merge is 17, 29, 0.219262489721119\n",
      "best merge is 11, 12, 0.220047799105543\n",
      "best merge is 1, 10, 0.234905701155728\n",
      "best merge is 18, 24, 0.248262003574209\n",
      "best merge is 15, 32, 0.286987311988544\n",
      "best merge is 4, 22, 0.302685787229969\n",
      "best merge is 0, 34, 0.302911851574634\n",
      "best merge is 1, 17, 0.310941916462459\n",
      "best merge is 18, 31, 0.338605470603046\n",
      "best merge is 11, 20, 0.385182929333578\n",
      "best merge is 0, 25, 0.393978783859255\n",
      "best merge is 4, 11, 0.447848243531644\n",
      "best merge is 27, 28, 0.50522168309916\n",
      "best merge is 15, 23, 0.532524070812801\n",
      "best merge is 2, 18, 0.575737840386758\n",
      "best merge is 2, 26, 0.710455040858508\n",
      "best merge is 21, 27, 0.732014346412227\n",
      "best merge is 0, 2, 0.804191255603292\n",
      "best merge is 0, 21, 0.93113431965367\n",
      "order classes\n",
      "15 - 1.47997489930441\n",
      "8 - 0.901484430027374\n",
      "10 - 1.1594205906264\n",
      "3 - 3.90588019932975\n",
      "3 - 5.68894553044599\n",
      "1 3 5 13 30 10 17 29\n",
      "0 33 34 25 2 6 7 8 9 16 18 24 31 26 36\n",
      "4 19 22 11 14 35 37 38 12 20\n",
      "15 32 23\n",
      "21 27 28\n",
      "Done.Start drawing\n",
      "84.94\t0.00\n",
      "GO_RESPONSE_TO_CORTICOSTEROID\n",
      "GO_ESTABLISHMENT_OF_EPITHELIAL_CELL_POLARITY\n",
      "GO_CORTICAL_ACTIN_CYTOSKELETON\n",
      "GO_REGULATION_OF_CYTOKINE_BIOSYNTHETIC_PROCESS\n",
      "GO_REGULATION_OF_CELLULAR_RESPONSE_TO_TRANSFORMING_GROWTH_FACTOR_BETA_STIMULUS\n",
      "GO_NEGATIVE_REGULATION_OF_SMOOTH_MUSCLE_CELL_MIGRATION\n",
      "GO_DETECTION_OF_CHEMICAL_STIMULUS_INVOLVED_IN_SENSORY_PERCEPTION_OF_TASTE\n",
      "GO_G_PROTEIN_COUPLED_CHEMOATTRACTANT_RECEPTOR_ACTIVITY\n",
      "GO_KIDNEY_MORPHOGENESIS\n",
      "GO_PLATELET_DERIVED_GROWTH_FACTOR_RECEPTOR_SIGNALING_PATHWAY\n",
      "GO_REGULATION_OF_RECEPTOR_INTERNALIZATION\n",
      "GO_VACUOLE_ORGANIZATION\n",
      "GO_TRANSLATION_INITIATION_FACTOR_ACTIVITY\n",
      "GO_MEGAKARYOCYTE_DIFFERENTIATION\n",
      "GO_TRANS_GOLGI_NETWORK_MEMBRANE\n",
      "GO_REGULATION_OF_MRNA_CATABOLIC_PROCESS\n",
      "GO_SYNAPTIC_VESICLE_CYTOSKELETAL_TRANSPORT\n",
      "GO_CULLIN_FAMILY_PROTEIN_BINDING\n",
      "GO_TRNA_METABOLIC_PROCESS\n",
      "GO_TRANSCRIPTION_ELONGATION_FROM_RNA_POLYMERASE_II_PROMOTER\n",
      "GO_RIBOSOMAL_SUBUNIT\n",
      "GO_PURINE_NUCLEOBASE_BIOSYNTHETIC_PROCESS\n",
      "GO_NONRIBOSOMAL_PEPTIDE_BIOSYNTHETIC_PROCESS\n",
      "GO_NUCLEOSOMAL_DNA_BINDING\n",
      "GO_PROTEASOME_COMPLEX\n",
      "GO_COFACTOR_BIOSYNTHETIC_PROCESS\n",
      "GO_MITOCHONDRIAL_MEMBRANE_PART\n",
      "GO_SPLICEOSOMAL_COMPLEX\n",
      "GO_SUPEROXIDE_METABOLIC_PROCESS\n",
      "GO_MCM_COMPLEX\n",
      "GO_PYRIMIDINE_RIBONUCLEOTIDE_METABOLIC_PROCESS\n",
      "GO_CONDENSED_CHROMOSOME\n",
      "GO_DNA_STRAND_ELONGATION_INVOLVED_IN_DNA_REPLICATION\n",
      "GO_MICROTUBULE_ORGANIZING_CENTER_PART\n",
      "GO_ACID_SECRETION\n",
      "GO_INNERVATION\n",
      "GO_NEUROTRANSMITTER_RECEPTOR_ACTIVITY\n",
      "GO_MULTICELLULAR_ORGANISMAL_SIGNALING\n",
      "GO_CELL_DIFFERENTIATION_INVOLVED_IN_EMBRYONIC_PLACENTA_DEVELOPMENT\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM1_3.csv_PAGE/filt_FC_LVM1_3.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM1_3.csv_PAGE/filt_FC_LVM1_3.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM1_3.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM1_3.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 37, 38, 0\n",
      "best merge is 3, 5, 0.0188948603484824\n",
      "best merge is 6, 7, 0.0387915308459444\n",
      "best merge is 13, 30, 0.0563883752403964\n",
      "best merge is 1, 3, 0.0638397060608967\n",
      "best merge is 14, 35, 0.0853186660356534\n",
      "best merge is 8, 9, 0.0899606030584174\n",
      "best merge is 1, 13, 0.0982299363135472\n",
      "best merge is 2, 6, 0.102467101480649\n",
      "best merge is 4, 19, 0.118056971979675\n",
      "best merge is 8, 16, 0.119805583069857\n",
      "best merge is 14, 37, 0.124370850503396\n",
      "best merge is 26, 36, 0.138475868593457\n",
      "best merge is 0, 33, 0.169163473873044\n",
      "best merge is 11, 14, 0.185496561577424\n",
      "best merge is 2, 8, 0.211924277414762\n",
      "best merge is 17, 29, 0.219262489721119\n",
      "best merge is 11, 12, 0.220047799105543\n",
      "best merge is 1, 10, 0.234905701155728\n",
      "best merge is 18, 24, 0.248262003574209\n",
      "best merge is 15, 32, 0.286987311988544\n",
      "best merge is 4, 22, 0.302685787229969\n",
      "best merge is 0, 34, 0.302911851574634\n",
      "best merge is 1, 17, 0.310941916462459\n",
      "best merge is 18, 31, 0.338605470603046\n",
      "best merge is 11, 20, 0.385182929333578\n",
      "best merge is 0, 25, 0.393978783859255\n",
      "best merge is 4, 11, 0.447848243531644\n",
      "best merge is 27, 28, 0.50522168309916\n",
      "best merge is 15, 23, 0.532524070812801\n",
      "best merge is 2, 18, 0.575737840386758\n",
      "best merge is 2, 26, 0.710455040858508\n",
      "best merge is 21, 27, 0.732014346412227\n",
      "best merge is 0, 2, 0.804191255603292\n",
      "best merge is 0, 21, 0.93113431965367\n",
      "order classes\n",
      "15 - 1.47997489930441\n",
      "8 - 0.901484430027374\n",
      "10 - 1.1594205906264\n",
      "3 - 3.90588019932975\n",
      "3 - 5.68894553044599\n",
      "1 3 5 13 30 10 17 29\n",
      "0 33 34 25 2 6 7 8 9 16 18 24 31 26 36\n",
      "4 19 22 11 14 35 37 38 12 20\n",
      "15 32 23\n",
      "21 27 28\n",
      "Done.-19.849187661879\t42\t0\n",
      "-19.6588179891688\t42\t0.301624676242053\n",
      "-19.5920649870497\t42\t0.380739345420297\n",
      "-19.5401459854015\t42\t0.435130680480339\n",
      "-19.4931716505769\t42\t0.484577348716741\n",
      "-19.4437249823405\t42\t0.529079350129503\n",
      "-19.3769719802213\t42\t0.583470685189545\n",
      "-19.2706616435131\t42\t0.662585354367789\n",
      "1.39804567930304\t42\t0.796091358606075\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM1_3.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM1_4.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c5.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c5/human_ensembl_msigdb_c5.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM1_4.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 5917\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t5917\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 5917\n",
      "cat 957, mi = 0.00794861, GO_CYTOSOLIC_RIBOSOME\n",
      "cat 1292, mi = 0.00792590, GO_ESTABLISHMENT_OF_PROTEIN_LOCALIZATION_TO_ENDOPLASMIC_RETICULUM\n",
      "cat 2330, mi = 0.00731835, GO_MULTI_ORGANISM_METABOLIC_PROCESS\n",
      "cat 5718, mi = 0.00692234, GO_TRANSLATIONAL_INITIATION\n",
      "cat 3995, mi = 0.00689173, GO_PROTEIN_LOCALIZATION_TO_ENDOPLASMIC_RETICULUM\n",
      "cat 2928, mi = 0.00678756, GO_NUCLEAR_TRANSCRIBED_MRNA_CATABOLIC_PROCESS_NONSENSE_MEDIATED_DECAY\n",
      "cat 4041, mi = 0.00563404, GO_PROTEIN_TARGETING_TO_MEMBRANE\n",
      "cat 954, mi = 0.00546837, GO_CYTOSOLIC_LARGE_RIBOSOMAL_SUBUNIT\n",
      "cat 5245, mi = 0.00527919, GO_RIBOSOMAL_SUBUNIT\n",
      "cat 5499, mi = 0.00465835, GO_SPLICEOSOMAL_COMPLEX\n",
      "cat 2219, mi = 0.00414434, GO_MITOCHONDRIAL_PROTEIN_COMPLEX\n",
      "cat 1889, mi = 0.00386908, GO_LARGE_RIBOSOMAL_SUBUNIT\n",
      "cat 3085, mi = 0.00365856, GO_OXIDATIVE_PHOSPHORYLATION\n",
      "cat 1742, mi = 0.00364092, GO_INNER_MITOCHONDRIAL_MEMBRANE_PROTEIN_COMPLEX\n",
      "cat 4852, mi = 0.00357067, GO_REGULATION_OF_RNA_STABILITY\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 957, mi = 0.008, pvalue = 0.000, GO_CYTOSOLIC_RIBOSOME\n",
      "Passed: cat 5499, mi = 0.005, pvalue = 0.000, GO_SPLICEOSOMAL_COMPLEX\n",
      "Passed: cat 2219, mi = 0.004, pvalue = 0.000, GO_MITOCHONDRIAL_PROTEIN_COMPLEX\n",
      "Passed: cat 4852, mi = 0.004, pvalue = 0.000, GO_REGULATION_OF_RNA_STABILITY\n",
      "Passed: cat 5719, mi = 0.003, pvalue = 0.000, GO_TRANSLATIONAL_TERMINATION\n",
      "Passed: cat 5755, mi = 0.003, pvalue = 0.000, GO_TRNA_METABOLIC_PROCESS\n",
      "Passed: cat 1342, mi = 0.002, pvalue = 0.000, GO_EXTRACELLULAR_MATRIX_COMPONENT\n",
      "Passed: cat 1323, mi = 0.002, pvalue = 0.000, GO_EXOCYTIC_VESICLE\n",
      "Passed: cat 1105, mi = 0.002, pvalue = 0.000, GO_DNA_TEMPLATED_TRANSCRIPTION_ELONGATION\n",
      "Passed: cat 5176, mi = 0.002, pvalue = 0.000, GO_RESPONSE_TO_TOPOLOGICALLY_INCORRECT_PROTEIN\n",
      "Passed: cat 1670, mi = 0.002, pvalue = 0.000, GO_HORMONE_ACTIVITY\n",
      "Passed: cat 2988, mi = 0.002, pvalue = 0.000, GO_NUCLEUS_ORGANIZATION\n",
      "Passed: cat 3865, mi = 0.002, pvalue = 0.000, GO_POTASSIUM_CHANNEL_ACTIVITY\n",
      "Passed: cat 2026, mi = 0.002, pvalue = 0.000, GO_MACROMOLECULE_DEACYLATION\n",
      "Passed: cat 5702, mi = 0.002, pvalue = 0.000, GO_TRANSFERASE_ACTIVITY_TRANSFERRING_HEXOSYL_GROUPS\n",
      "Passed: cat 2187, mi = 0.002, pvalue = 0.001, GO_MICROTUBULE_ORGANIZING_CENTER_PART\n",
      "Passed: cat 3792, mi = 0.002, pvalue = 0.001, GO_POSITIVE_REGULATION_OF_SYNAPSE_ASSEMBLY\n",
      "Passed: cat 4009, mi = 0.002, pvalue = 0.001, GO_PROTEIN_MODIFICATION_BY_SMALL_PROTEIN_REMOVAL\n",
      "Passed: cat 223, mi = 0.002, pvalue = 0.001, GO_APPENDAGE_DEVELOPMENT\n",
      "Passed: cat 171, mi = 0.002, pvalue = 0.001, GO_ANCHORED_COMPONENT_OF_MEMBRANE\n",
      "Passed: cat 993, mi = 0.002, pvalue = 0.002, GO_DEOXYRIBONUCLEASE_ACTIVITY\n",
      "Passed: cat 5786, mi = 0.002, pvalue = 0.000, GO_T_CELL_RECEPTOR_COMPLEX\n",
      "Passed: cat 3238, mi = 0.002, pvalue = 0.002, GO_PHOSPHATIDYLINOSITOL_DEPHOSPHORYLATION\n",
      "Passed: cat 1359, mi = 0.002, pvalue = 0.001, GO_E_BOX_BINDING\n",
      "Passed: cat 926, mi = 0.002, pvalue = 0.003, GO_CYTOKINE_RECEPTOR_ACTIVITY\n",
      "Passed: cat 1680, mi = 0.001, pvalue = 0.003, GO_HUMORAL_IMMUNE_RESPONSE\n",
      "Passed: cat 1735, mi = 0.001, pvalue = 0.001, GO_INNERVATION\n",
      "Passed: cat 5722, mi = 0.001, pvalue = 0.004, GO_TRANSLATION_INITIATION_FACTOR_ACTIVITY\n",
      "Passed: cat 5351, mi = 0.001, pvalue = 0.003, GO_SENSORY_PERCEPTION_OF_TASTE\n",
      "Passed: cat 4110, mi = 0.001, pvalue = 0.004, GO_PYRIMIDINE_RIBONUCLEOSIDE_TRIPHOSPHATE_METABOLIC_PROCESS\n",
      "Passed: cat 2131, mi = 0.001, pvalue = 0.002, GO_METALLOAMINOPEPTIDASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 642, mi = 0.001, pvalue = 0.007, GO_CELL_COMMUNICATION_INVOLVED_IN_CARDIAC_CONDUCTION\n",
      "Didn't pass (cutoff=0.0050): cat 879, mi = 0.001, pvalue = 0.009, GO_CORTICAL_ACTIN_CYTOSKELETON\n",
      "Didn't pass (cutoff=0.0050): cat 4881, mi = 0.001, pvalue = 0.008, GO_REGULATION_OF_SODIUM_ION_TRANSPORT\n",
      "Didn't pass (cutoff=0.0050): cat 880, mi = 0.001, pvalue = 0.008, GO_CORTICAL_CYTOSKELETON\n",
      "Didn't pass (cutoff=0.0050): cat 5854, mi = 0.001, pvalue = 0.010, GO_VESICLE_COAT\n",
      "Didn't pass (cutoff=0.0050): cat 1720, mi = 0.001, pvalue = 0.009, GO_IMPORT_INTO_CELL\n",
      "Didn't pass (cutoff=0.0050): cat 4118, mi = 0.001, pvalue = 0.011, GO_RAB_GUANYL_NUCLEOTIDE_EXCHANGE_FACTOR_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 41, mi = 0.001, pvalue = 0.010, GO_ACTIN_FILAMENT_BUNDLE\n",
      "Didn't pass (cutoff=0.0050): cat 930, mi = 0.001, pvalue = 0.006, GO_CYTOPLASMIC_DYNEIN_COMPLEX\n",
      "Didn't pass (cutoff=0.0050): cat 4921, mi = 0.001, pvalue = 0.006, GO_REGULATION_OF_TELOMERASE_RNA_LOCALIZATION_TO_CAJAL_BODY\n",
      "Didn't pass (cutoff=0.0050): cat 2000, mi = 0.001, pvalue = 0.008, GO_LYSINE_ACETYLATED_HISTONE_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 2370, mi = 0.001, pvalue = 0.011, GO_MYOSIN_COMPLEX\n",
      "Didn't pass (cutoff=0.0050): cat 126, mi = 0.001, pvalue = 0.006, GO_ALPHA_LINOLENIC_ACID_METABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 3996, mi = 0.001, pvalue = 0.012, GO_PROTEIN_LOCALIZATION_TO_GOLGI_APPARATUS\n",
      "Didn't pass (cutoff=0.0050): cat 1630, mi = 0.001, pvalue = 0.012, GO_HIPPO_SIGNALING\n",
      "Passed: cat 5893, mi = 0.001, pvalue = 0.004, GO_WASH_COMPLEX\n",
      "Didn't pass (cutoff=0.0050): cat 2064, mi = 0.001, pvalue = 0.006, GO_MAP_KINASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 4135, mi = 0.001, pvalue = 0.013, GO_RECEPTOR_CATABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 94, mi = 0.001, pvalue = 0.017, GO_ADIPOSE_TISSUE_DEVELOPMENT\n",
      "Didn't pass (cutoff=0.0050): cat 1542, mi = 0.001, pvalue = 0.020, GO_GOLGI_TO_ENDOSOME_TRANSPORT\n",
      "Didn't pass (cutoff=0.0050): cat 2751, mi = 0.001, pvalue = 0.020, GO_NEGATIVE_REGULATION_OF_STEM_CELL_DIFFERENTIATION\n",
      "Didn't pass (cutoff=0.0050): cat 866, mi = 0.001, pvalue = 0.018, GO_COPI_COATED_VESICLE_MEMBRANE\n",
      "Didn't pass (cutoff=0.0050): cat 1362, mi = 0.001, pvalue = 0.013, GO_FANCONI_ANAEMIA_NUCLEAR_COMPLEX\n",
      "Didn't pass (cutoff=0.0050): cat 5407, mi = 0.001, pvalue = 0.018, GO_SKELETAL_MUSCLE_CONTRACTION\n",
      "Didn't pass (cutoff=0.0050): cat 3350, mi = 0.001, pvalue = 0.021, GO_POSITIVE_REGULATION_OF_ACTIN_FILAMENT_BUNDLE_ASSEMBLY\n",
      "Didn't pass (cutoff=0.0050): cat 2152, mi = 0.001, pvalue = 0.022, GO_METHYLATED_HISTONE_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 2694, mi = 0.001, pvalue = 0.014, GO_NEGATIVE_REGULATION_OF_PRODUCTION_OF_MOLECULAR_MEDIATOR_OF_IMMUNE_RESPONSE\n",
      "Didn't pass (cutoff=0.0050): cat 3977, mi = 0.001, pvalue = 0.018, GO_PROTEIN_KINASE_C_ACTIVATING_G_PROTEIN_COUPLED_RECEPTOR_SIGNALING_PATHWAY\n",
      "Didn't pass (cutoff=0.0050): cat 2004, mi = 0.001, pvalue = 0.023, GO_LYSOPHOSPHOLIPID_ACYLTRANSFERASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 1391, mi = 0.001, pvalue = 0.022, GO_FERROUS_IRON_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 2379, mi = 0.001, pvalue = 0.028, GO_NADH_METABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 73, mi = 0.001, pvalue = 0.024, GO_ACTOMYOSIN\n",
      "Didn't pass (cutoff=0.0050): cat 2032, mi = 0.001, pvalue = 0.012, GO_MACROPHAGE_DIFFERENTIATION\n",
      "Didn't pass (cutoff=0.0050): cat 4888, mi = 0.001, pvalue = 0.017, GO_REGULATION_OF_STEM_CELL_POPULATION_MAINTENANCE\n",
      "Didn't pass (cutoff=0.0050): cat 3780, mi = 0.001, pvalue = 0.029, GO_POSITIVE_REGULATION_OF_SODIUM_ION_TRANSPORT\n",
      "Didn't pass (cutoff=0.0050): cat 2981, mi = 0.001, pvalue = 0.029, GO_NUCLEOTIDE_SUGAR_BIOSYNTHETIC_PROCESS\n",
      "Total number of categories evaluated = 443.\n",
      "Number of categories that passed the tests = 32\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM1_4.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM1_4.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 10, 16, 0.0172106464560686\n",
      "best merge is 0, 30, 0.0359600514921855\n",
      "best merge is 7, 10, 0.0435060007362639\n",
      "best merge is 8, 27, 0.0523061813671457\n",
      "best merge is 7, 12, 0.0786023313807679\n",
      "best merge is 1, 3, 0.0862872656739082\n",
      "best merge is 5, 11, 0.0897408325342012\n",
      "best merge is 5, 9, 0.103208730598105\n",
      "best merge is 7, 26, 0.109309325838089\n",
      "best merge is 5, 17, 0.144377577857543\n",
      "best merge is 1, 4, 0.162333703030433\n",
      "best merge is 6, 25, 0.170088119531302\n",
      "best merge is 7, 28, 0.186498339287317\n",
      "best merge is 5, 8, 0.186531794638039\n",
      "best merge is 21, 24, 0.207645626270453\n",
      "best merge is 1, 5, 0.254955053978456\n",
      "best merge is 6, 7, 0.283954272461822\n",
      "best merge is 21, 23, 0.347324060449251\n",
      "best merge is 0, 15, 0.350161346827471\n",
      "best merge is 2, 29, 0.371914445474033\n",
      "best merge is 14, 20, 0.37894826841231\n",
      "best merge is 13, 31, 0.379216084887375\n",
      "best merge is 6, 19, 0.505239583619725\n",
      "best merge is 1, 13, 0.517395667139871\n",
      "best merge is 6, 18, 0.595064648825666\n",
      "best merge is 1, 2, 0.632919004923524\n",
      "best merge is 0, 22, 0.636133066896005\n",
      "best merge is 0, 14, 0.701060068257336\n",
      "order classes\n",
      "4 - 2.64468194583514\n",
      "13 - 0.910695467821737\n",
      "10 - 0.939826978516596\n",
      "2 - 3.03158614729848\n",
      "3 - 2.13877036365975\n",
      "1 3 4 5 11 9 17 8 27 13 31 2 29\n",
      "0 30 15 22\n",
      "6 25 7 10 16 12 26 28 19 18\n",
      "14 20\n",
      "21 24 23\n",
      "Done.Start drawing\n",
      "26.31\t0.00\n",
      "GO_HUMORAL_IMMUNE_RESPONSE\n",
      "GO_APPENDAGE_DEVELOPMENT\n",
      "GO_T_CELL_RECEPTOR_COMPLEX\n",
      "GO_CYTOKINE_RECEPTOR_ACTIVITY\n",
      "GO_E_BOX_BINDING\n",
      "GO_DNA_TEMPLATED_TRANSCRIPTION_ELONGATION\n",
      "GO_TRANSLATION_INITIATION_FACTOR_ACTIVITY\n",
      "GO_MACROMOLECULE_DEACYLATION\n",
      "GO_WASH_COMPLEX\n",
      "GO_SPLICEOSOMAL_COMPLEX\n",
      "GO_REGULATION_OF_RNA_STABILITY\n",
      "GO_TRANSLATIONAL_TERMINATION\n",
      "GO_TRNA_METABOLIC_PROCESS\n",
      "GO_PROTEIN_MODIFICATION_BY_SMALL_PROTEIN_REMOVAL\n",
      "GO_PYRIMIDINE_RIBONUCLEOSIDE_TRIPHOSPHATE_METABOLIC_PROCESS\n",
      "GO_NUCLEUS_ORGANIZATION\n",
      "GO_RESPONSE_TO_TOPOLOGICALLY_INCORRECT_PROTEIN\n",
      "GO_MITOCHONDRIAL_PROTEIN_COMPLEX\n",
      "GO_TRANSFERASE_ACTIVITY_TRANSFERRING_HEXOSYL_GROUPS\n",
      "GO_DEOXYRIBONUCLEASE_ACTIVITY\n",
      "GO_CYTOSOLIC_RIBOSOME\n",
      "GO_METALLOAMINOPEPTIDASE_ACTIVITY\n",
      "GO_MICROTUBULE_ORGANIZING_CENTER_PART\n",
      "GO_PHOSPHATIDYLINOSITOL_DEPHOSPHORYLATION\n",
      "GO_EXTRACELLULAR_MATRIX_COMPONENT\n",
      "GO_EXOCYTIC_VESICLE\n",
      "GO_HORMONE_ACTIVITY\n",
      "GO_POSITIVE_REGULATION_OF_SYNAPSE_ASSEMBLY\n",
      "GO_POTASSIUM_CHANNEL_ACTIVITY\n",
      "GO_INNERVATION\n",
      "GO_SENSORY_PERCEPTION_OF_TASTE\n",
      "GO_ANCHORED_COMPONENT_OF_MEMBRANE\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM1_4.csv_PAGE/filt_FC_LVM1_4.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM1_4.csv_PAGE/filt_FC_LVM1_4.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM1_4.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM1_4.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 10, 16, 0.0172106464560686\n",
      "best merge is 0, 30, 0.0359600514921855\n",
      "best merge is 7, 10, 0.0435060007362639\n",
      "best merge is 8, 27, 0.0523061813671457\n",
      "best merge is 7, 12, 0.0786023313807679\n",
      "best merge is 1, 3, 0.0862872656739082\n",
      "best merge is 5, 11, 0.0897408325342012\n",
      "best merge is 5, 9, 0.103208730598105\n",
      "best merge is 7, 26, 0.109309325838089\n",
      "best merge is 5, 17, 0.144377577857543\n",
      "best merge is 1, 4, 0.162333703030433\n",
      "best merge is 6, 25, 0.170088119531302\n",
      "best merge is 7, 28, 0.186498339287317\n",
      "best merge is 5, 8, 0.186531794638039\n",
      "best merge is 21, 24, 0.207645626270453\n",
      "best merge is 1, 5, 0.254955053978456\n",
      "best merge is 6, 7, 0.283954272461822\n",
      "best merge is 21, 23, 0.347324060449251\n",
      "best merge is 0, 15, 0.350161346827471\n",
      "best merge is 2, 29, 0.371914445474033\n",
      "best merge is 14, 20, 0.37894826841231\n",
      "best merge is 13, 31, 0.379216084887375\n",
      "best merge is 6, 19, 0.505239583619725\n",
      "best merge is 1, 13, 0.517395667139871\n",
      "best merge is 6, 18, 0.595064648825666\n",
      "best merge is 1, 2, 0.632919004923524\n",
      "best merge is 0, 22, 0.636133066896005\n",
      "best merge is 0, 14, 0.701060068257336\n",
      "order classes\n",
      "4 - 2.64468194583514\n",
      "13 - 0.910695467821737\n",
      "10 - 0.939826978516596\n",
      "2 - 3.03158614729848\n",
      "3 - 2.13877036365975\n",
      "1 3 4 5 11 9 17 8 27 13 31 2 29\n",
      "0 30 15 22\n",
      "6 25 7 10 16 12 26 28 19 18\n",
      "14 20\n",
      "21 24 23\n",
      "Done.-19.5290763968073\t42\t0\n",
      "-18.9064994298746\t42\t0.941847206385405\n",
      "-18.6590649942987\t42\t1.24515393386545\n",
      "-18.4754846066135\t42\t1.43671607753706\n",
      "-18.3158494868871\t42\t1.61231470923603\n",
      "-18.1641961231471\t42\t1.75598631698974\n",
      "-17.9806157354618\t42\t1.91562143671608\n",
      "-17.709236031927\t42\t2.12314709236032\n",
      "2.22919042189282\t42\t2.45838084378563\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM1_4.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM2_2.csv --species=human_ensembl_msigdb_c5 --exptype=continuous --ebins=9\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM2_1.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c5.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c5/human_ensembl_msigdb_c5.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM2_1.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 5917\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t5917\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 5917\n",
      "cat 5718, mi = 0.00813532, GO_TRANSLATIONAL_INITIATION\n",
      "cat 957, mi = 0.00744551, GO_CYTOSOLIC_RIBOSOME\n",
      "cat 1292, mi = 0.00738183, GO_ESTABLISHMENT_OF_PROTEIN_LOCALIZATION_TO_ENDOPLASMIC_RETICULUM\n",
      "cat 5245, mi = 0.00725300, GO_RIBOSOMAL_SUBUNIT\n",
      "cat 3995, mi = 0.00665154, GO_PROTEIN_LOCALIZATION_TO_ENDOPLASMIC_RETICULUM\n",
      "cat 2928, mi = 0.00655146, GO_NUCLEAR_TRANSCRIBED_MRNA_CATABOLIC_PROCESS_NONSENSE_MEDIATED_DECAY\n",
      "cat 954, mi = 0.00632822, GO_CYTOSOLIC_LARGE_RIBOSOMAL_SUBUNIT\n",
      "cat 2330, mi = 0.00593174, GO_MULTI_ORGANISM_METABOLIC_PROCESS\n",
      "cat 5499, mi = 0.00561917, GO_SPLICEOSOMAL_COMPLEX\n",
      "cat 1889, mi = 0.00555507, GO_LARGE_RIBOSOMAL_SUBUNIT\n",
      "cat 2219, mi = 0.00451041, GO_MITOCHONDRIAL_PROTEIN_COMPLEX\n",
      "cat 1742, mi = 0.00413968, GO_INNER_MITOCHONDRIAL_MEMBRANE_PROTEIN_COMPLEX\n",
      "cat 2217, mi = 0.00411404, GO_MITOCHONDRIAL_MEMBRANE_PART\n",
      "cat 1087, mi = 0.00403324, GO_DNA_PACKAGING_COMPLEX\n",
      "cat 5028, mi = 0.00396562, GO_RESPIRATORY_CHAIN\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 5718, mi = 0.008, pvalue = 0.000, GO_TRANSLATIONAL_INITIATION\n",
      "Passed: cat 5499, mi = 0.006, pvalue = 0.000, GO_SPLICEOSOMAL_COMPLEX\n",
      "Passed: cat 2219, mi = 0.005, pvalue = 0.000, GO_MITOCHONDRIAL_PROTEIN_COMPLEX\n",
      "Passed: cat 1087, mi = 0.004, pvalue = 0.000, GO_DNA_PACKAGING_COMPLEX\n",
      "Passed: cat 2224, mi = 0.004, pvalue = 0.000, GO_MITOCHONDRIAL_TRANSLATION\n",
      "Passed: cat 1175, mi = 0.002, pvalue = 0.000, GO_ENDOCRINE_SYSTEM_DEVELOPMENT\n",
      "Passed: cat 3910, mi = 0.002, pvalue = 0.000, GO_PROTEASOME_COMPLEX\n",
      "Passed: cat 5495, mi = 0.002, pvalue = 0.000, GO_SPINDLE_MICROTUBULE\n",
      "Passed: cat 1323, mi = 0.002, pvalue = 0.000, GO_EXOCYTIC_VESICLE\n",
      "Passed: cat 1670, mi = 0.002, pvalue = 0.000, GO_HORMONE_ACTIVITY\n",
      "Passed: cat 671, mi = 0.002, pvalue = 0.000, GO_CELL_JUNCTION_ORGANIZATION\n",
      "Passed: cat 1862, mi = 0.002, pvalue = 0.000, GO_KERATIN_FILAMENT\n",
      "Passed: cat 5678, mi = 0.002, pvalue = 0.000, GO_TRANSCRIPTION_ELONGATION_FROM_RNA_POLYMERASE_II_PROMOTER\n",
      "Passed: cat 5885, mi = 0.002, pvalue = 0.000, GO_VOLTAGE_GATED_CATION_CHANNEL_ACTIVITY\n",
      "Passed: cat 5389, mi = 0.002, pvalue = 0.000, GO_SIN3_TYPE_COMPLEX\n",
      "Passed: cat 2328, mi = 0.002, pvalue = 0.001, GO_MULTI_ORGANISM_LOCALIZATION\n",
      "Passed: cat 5266, mi = 0.002, pvalue = 0.001, GO_RNA_MODIFICATION\n",
      "Passed: cat 1485, mi = 0.002, pvalue = 0.000, GO_GLUCOSE_CATABOLIC_PROCESS\n",
      "Passed: cat 356, mi = 0.002, pvalue = 0.001, GO_BONE_MORPHOGENESIS\n",
      "Passed: cat 4472, mi = 0.002, pvalue = 0.000, GO_REGULATION_OF_GRANULOCYTE_CHEMOTAXIS\n",
      "Passed: cat 4815, mi = 0.002, pvalue = 0.001, GO_REGULATION_OF_RECEPTOR_ACTIVITY\n",
      "Passed: cat 1099, mi = 0.002, pvalue = 0.002, GO_DNA_REPLICATION_INITIATION\n",
      "Passed: cat 4009, mi = 0.002, pvalue = 0.002, GO_PROTEIN_MODIFICATION_BY_SMALL_PROTEIN_REMOVAL\n",
      "Passed: cat 5214, mi = 0.002, pvalue = 0.003, GO_RETROGRADE_TRANSPORT_ENDOSOME_TO_GOLGI\n",
      "Passed: cat 3288, mi = 0.002, pvalue = 0.002, GO_PIGMENT_BIOSYNTHETIC_PROCESS\n",
      "Passed: cat 580, mi = 0.001, pvalue = 0.003, GO_CELLULAR_RESPONSE_TO_IONIZING_RADIATION\n",
      "Passed: cat 3792, mi = 0.001, pvalue = 0.004, GO_POSITIVE_REGULATION_OF_SYNAPSE_ASSEMBLY\n",
      "Passed: cat 5813, mi = 0.001, pvalue = 0.004, GO_UNFOLDED_PROTEIN_BINDING\n",
      "Passed: cat 5351, mi = 0.001, pvalue = 0.002, GO_SENSORY_PERCEPTION_OF_TASTE\n",
      "Passed: cat 3876, mi = 0.001, pvalue = 0.004, GO_PRENYLTRANSFERASE_ACTIVITY\n",
      "Passed: cat 5407, mi = 0.001, pvalue = 0.004, GO_SKELETAL_MUSCLE_CONTRACTION\n",
      "Passed: cat 1623, mi = 0.001, pvalue = 0.001, GO_HIGH_DENSITY_LIPOPROTEIN_PARTICLE\n",
      "Passed: cat 1521, mi = 0.001, pvalue = 0.003, GO_GLYCOSPHINGOLIPID_BIOSYNTHETIC_PROCESS\n",
      "Passed: cat 3939, mi = 0.001, pvalue = 0.004, GO_PROTEIN_DEPHOSPHORYLATION\n",
      "Passed: cat 967, mi = 0.001, pvalue = 0.004, GO_DEATH_RECEPTOR_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 5193, mi = 0.001, pvalue = 0.006, GO_RESPONSE_TO_XENOBIOTIC_STIMULUS\n",
      "Didn't pass (cutoff=0.0050): cat 2253, mi = 0.001, pvalue = 0.007, GO_MODIFIED_AMINO_ACID_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 436, mi = 0.001, pvalue = 0.009, GO_CARBOXYPEPTIDASE_ACTIVITY\n",
      "Passed: cat 796, mi = 0.001, pvalue = 0.005, GO_CILIUM_ORGANIZATION\n",
      "Didn't pass (cutoff=0.0050): cat 5663, mi = 0.001, pvalue = 0.007, GO_TRANSCRIPTIONALLY_ACTIVE_CHROMATIN\n",
      "Didn't pass (cutoff=0.0050): cat 4007, mi = 0.001, pvalue = 0.006, GO_PROTEIN_METHYLTRANSFERASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 693, mi = 0.001, pvalue = 0.008, GO_CELL_REDOX_HOMEOSTASIS\n",
      "Didn't pass (cutoff=0.0050): cat 1344, mi = 0.001, pvalue = 0.007, GO_EXTRACELLULAR_MATRIX_STRUCTURAL_CONSTITUENT\n",
      "Didn't pass (cutoff=0.0050): cat 865, mi = 0.001, pvalue = 0.012, GO_COPI_COATED_VESICLE\n",
      "Didn't pass (cutoff=0.0050): cat 3114, mi = 0.001, pvalue = 0.009, GO_OXYGEN_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 1832, mi = 0.001, pvalue = 0.017, GO_ION_ANTIPORTER_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 542, mi = 0.001, pvalue = 0.008, GO_CELLULAR_RESPONSE_TO_CADMIUM_ION\n",
      "Didn't pass (cutoff=0.0050): cat 205, mi = 0.001, pvalue = 0.024, GO_ANTIPORTER_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 750, mi = 0.001, pvalue = 0.015, GO_CHLORIDE_CHANNEL_COMPLEX\n",
      "Didn't pass (cutoff=0.0050): cat 3106, mi = 0.001, pvalue = 0.007, GO_OXIDOREDUCTASE_ACTIVITY_ACTING_ON_THE_CH_NH2_GROUP_OF_DONORS_OXYGEN_AS_ACCEPTOR\n",
      "Didn't pass (cutoff=0.0050): cat 3780, mi = 0.001, pvalue = 0.025, GO_POSITIVE_REGULATION_OF_SODIUM_ION_TRANSPORT\n",
      "Didn't pass (cutoff=0.0050): cat 5053, mi = 0.001, pvalue = 0.029, GO_RESPONSE_TO_CADMIUM_ION\n",
      "Didn't pass (cutoff=0.0050): cat 420, mi = 0.001, pvalue = 0.027, GO_CARBOHYDRATE_DERIVATIVE_TRANSPORT\n",
      "Didn't pass (cutoff=0.0050): cat 3821, mi = 0.001, pvalue = 0.012, GO_POSITIVE_REGULATION_OF_TRIGLYCERIDE_BIOSYNTHETIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 2113, mi = 0.001, pvalue = 0.027, GO_MEMBRANE_PROTEIN_INTRACELLULAR_DOMAIN_PROTEOLYSIS\n",
      "Didn't pass (cutoff=0.0050): cat 866, mi = 0.001, pvalue = 0.025, GO_COPI_COATED_VESICLE_MEMBRANE\n",
      "Didn't pass (cutoff=0.0050): cat 4853, mi = 0.001, pvalue = 0.029, GO_REGULATION_OF_RUFFLE_ASSEMBLY\n",
      "Didn't pass (cutoff=0.0050): cat 2500, mi = 0.001, pvalue = 0.028, GO_NEGATIVE_REGULATION_OF_CYTOKINE_SECRETION\n",
      "Didn't pass (cutoff=0.0050): cat 3943, mi = 0.001, pvalue = 0.043, GO_PROTEIN_DISULFIDE_OXIDOREDUCTASE_ACTIVITY\n",
      "Total number of categories evaluated = 690.\n",
      "Number of categories that passed the tests = 36\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM2_1.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM2_1.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 9, 19, 0.0497885101757138\n",
      "best merge is 17, 32, 0.0535131807236958\n",
      "best merge is 11, 26, 0.0592579102525866\n",
      "best merge is 9, 13, 0.0680848755068866\n",
      "best merge is 23, 24, 0.0739646309651801\n",
      "best merge is 1, 6, 0.0749364730862855\n",
      "best merge is 11, 28, 0.0856584299885653\n",
      "best merge is 4, 22, 0.0865015272437454\n",
      "best merge is 2, 15, 0.0907864441659283\n",
      "best merge is 3, 30, 0.102522080176738\n",
      "best merge is 12, 16, 0.120325064368138\n",
      "best merge is 7, 11, 0.127405699241305\n",
      "best merge is 4, 12, 0.139663951978998\n",
      "best merge is 1, 23, 0.154096931472596\n",
      "best merge is 18, 20, 0.158456363373663\n",
      "best merge is 7, 9, 0.179480580970745\n",
      "best merge is 17, 21, 0.194869605992454\n",
      "best merge is 2, 4, 0.218311428550886\n",
      "best merge is 1, 2, 0.248592296817199\n",
      "best merge is 7, 8, 0.256598362087411\n",
      "best merge is 0, 27, 0.271278352493554\n",
      "best merge is 3, 7, 0.27742418713816\n",
      "best merge is 5, 18, 0.279017784304245\n",
      "best merge is 14, 34, 0.37189327793842\n",
      "best merge is 5, 10, 0.378378464002336\n",
      "best merge is 14, 29, 0.392574262973135\n",
      "best merge is 5, 17, 0.462218815500549\n",
      "best merge is 0, 35, 0.497356518308632\n",
      "best merge is 1, 25, 0.528240644806187\n",
      "best merge is 3, 31, 0.584521901517423\n",
      "best merge is 0, 1, 0.64276030204899\n",
      "best merge is 5, 33, 0.745311587118626\n",
      "order classes\n",
      "14 - 0.973305446862984\n",
      "11 - 0.818735408112332\n",
      "7 - 1.64603030960558\n",
      "3 - 3.08544481035918\n",
      "3 30 7 11 26 28 9 19 13 8 31\n",
      "0 27 35 1 6 23 24 2 15 4 22 12 16 25\n",
      "5 18 20 10 17 32 21\n",
      "14 34 29\n",
      "33\n",
      "Done.Start drawing\n",
      "70.09\t0.00\n",
      "GO_ENDOCRINE_SYSTEM_DEVELOPMENT\n",
      "GO_BONE_MORPHOGENESIS\n",
      "GO_REGULATION_OF_RECEPTOR_ACTIVITY\n",
      "GO_CELL_JUNCTION_ORGANIZATION\n",
      "GO_GLYCOSPHINGOLIPID_BIOSYNTHETIC_PROCESS\n",
      "GO_DNA_REPLICATION_INITIATION\n",
      "GO_CELLULAR_RESPONSE_TO_IONIZING_RADIATION\n",
      "GO_GLUCOSE_CATABOLIC_PROCESS\n",
      "GO_PROTEIN_DEPHOSPHORYLATION\n",
      "GO_RNA_MODIFICATION\n",
      "GO_MITOCHONDRIAL_TRANSLATION\n",
      "GO_TRANSCRIPTION_ELONGATION_FROM_RNA_POLYMERASE_II_PROMOTER\n",
      "GO_SIN3_TYPE_COMPLEX\n",
      "GO_SPLICEOSOMAL_COMPLEX\n",
      "GO_PROTEASOME_COMPLEX\n",
      "GO_PROTEIN_MODIFICATION_BY_SMALL_PROTEIN_REMOVAL\n",
      "GO_DEATH_RECEPTOR_BINDING\n",
      "GO_PRENYLTRANSFERASE_ACTIVITY\n",
      "GO_UNFOLDED_PROTEIN_BINDING\n",
      "GO_RETROGRADE_TRANSPORT_ENDOSOME_TO_GOLGI\n",
      "GO_PIGMENT_BIOSYNTHETIC_PROCESS\n",
      "GO_MULTI_ORGANISM_LOCALIZATION\n",
      "GO_HIGH_DENSITY_LIPOPROTEIN_PARTICLE\n",
      "GO_TRANSLATIONAL_INITIATION\n",
      "GO_MITOCHONDRIAL_PROTEIN_COMPLEX\n",
      "GO_CILIUM_ORGANIZATION\n",
      "GO_DNA_PACKAGING_COMPLEX\n",
      "GO_SKELETAL_MUSCLE_CONTRACTION\n",
      "GO_SPINDLE_MICROTUBULE\n",
      "GO_KERATIN_FILAMENT\n",
      "GO_POSITIVE_REGULATION_OF_SYNAPSE_ASSEMBLY\n",
      "GO_SENSORY_PERCEPTION_OF_TASTE\n",
      "GO_HORMONE_ACTIVITY\n",
      "GO_REGULATION_OF_GRANULOCYTE_CHEMOTAXIS\n",
      "GO_VOLTAGE_GATED_CATION_CHANNEL_ACTIVITY\n",
      "GO_EXOCYTIC_VESICLE\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM2_1.csv_PAGE/filt_FC_LVM2_1.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM2_1.csv_PAGE/filt_FC_LVM2_1.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM2_1.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM2_1.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 9, 19, 0.0497885101757138\n",
      "best merge is 17, 32, 0.0535131807236958\n",
      "best merge is 11, 26, 0.0592579102525866\n",
      "best merge is 9, 13, 0.0680848755068866\n",
      "best merge is 23, 24, 0.0739646309651801\n",
      "best merge is 1, 6, 0.0749364730862855\n",
      "best merge is 11, 28, 0.0856584299885653\n",
      "best merge is 4, 22, 0.0865015272437454\n",
      "best merge is 2, 15, 0.0907864441659283\n",
      "best merge is 3, 30, 0.102522080176738\n",
      "best merge is 12, 16, 0.120325064368138\n",
      "best merge is 7, 11, 0.127405699241305\n",
      "best merge is 4, 12, 0.139663951978998\n",
      "best merge is 1, 23, 0.154096931472596\n",
      "best merge is 18, 20, 0.158456363373663\n",
      "best merge is 7, 9, 0.179480580970745\n",
      "best merge is 17, 21, 0.194869605992454\n",
      "best merge is 2, 4, 0.218311428550886\n",
      "best merge is 1, 2, 0.248592296817199\n",
      "best merge is 7, 8, 0.256598362087411\n",
      "best merge is 0, 27, 0.271278352493554\n",
      "best merge is 3, 7, 0.27742418713816\n",
      "best merge is 5, 18, 0.279017784304245\n",
      "best merge is 14, 34, 0.37189327793842\n",
      "best merge is 5, 10, 0.378378464002336\n",
      "best merge is 14, 29, 0.392574262973135\n",
      "best merge is 5, 17, 0.462218815500549\n",
      "best merge is 0, 35, 0.497356518308632\n",
      "best merge is 1, 25, 0.528240644806187\n",
      "best merge is 3, 31, 0.584521901517423\n",
      "best merge is 0, 1, 0.64276030204899\n",
      "best merge is 5, 33, 0.745311587118626\n",
      "order classes\n",
      "14 - 0.973305446862984\n",
      "11 - 0.818735408112332\n",
      "7 - 1.64603030960558\n",
      "3 - 3.08544481035918\n",
      "3 30 7 11 26 28 9 19 13 8 31\n",
      "0 27 35 1 6 23 24 2 15 4 22 12 16 25\n",
      "5 18 20 10 17 32 21\n",
      "14 34 29\n",
      "33\n",
      "Done.-19.784277357683\t42\t0\n",
      "-19.5056356113568\t42\t0.431445284634042\n",
      "-19.4037665858182\t42\t0.557283492652304\n",
      "-19.3318590383792\t42\t0.635183335711228\n",
      "-19.2689399343701\t42\t0.701098587530318\n",
      "-19.203024682551\t42\t0.761021543729491\n",
      "-19.1221286916821\t42\t0.832929091168498\n",
      "-18.9902981880439\t42\t0.922813525467256\n",
      "1.54829504922243\t42\t1.09659009844486\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM2_1.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM2_3.csv --species=human_ensembl_msigdb_c5 --exptype=continuous --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM2_4.csv --species=human_ensembl_msigdb_c5 --exptype=continuous --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM2_5.csv --species=human_ensembl_msigdb_c5 --exptype=continuous --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_P_LVM1_1.csv --species=human_ensembl_msigdb_c5 --exptype=continuous --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_P_LVM1_2.csv --species=human_ensembl_msigdb_c5 --exptype=continuous --ebins=9\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM2_2.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c5.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c5/human_ensembl_msigdb_c5.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM2_2.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 5917\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t5917\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 5917\n",
      "cat 5718, mi = 0.00871754, GO_TRANSLATIONAL_INITIATION\n",
      "cat 1292, mi = 0.00845772, GO_ESTABLISHMENT_OF_PROTEIN_LOCALIZATION_TO_ENDOPLASMIC_RETICULUM\n",
      "cat 957, mi = 0.00813247, GO_CYTOSOLIC_RIBOSOME\n",
      "cat 5245, mi = 0.00754704, GO_RIBOSOMAL_SUBUNIT\n",
      "cat 3995, mi = 0.00728309, GO_PROTEIN_LOCALIZATION_TO_ENDOPLASMIC_RETICULUM\n",
      "cat 954, mi = 0.00667897, GO_CYTOSOLIC_LARGE_RIBOSOMAL_SUBUNIT\n",
      "cat 2330, mi = 0.00635036, GO_MULTI_ORGANISM_METABOLIC_PROCESS\n",
      "cat 2928, mi = 0.00634111, GO_NUCLEAR_TRANSCRIBED_MRNA_CATABOLIC_PROCESS_NONSENSE_MEDIATED_DECAY\n",
      "cat 1889, mi = 0.00596370, GO_LARGE_RIBOSOMAL_SUBUNIT\n",
      "cat 2219, mi = 0.00538043, GO_MITOCHONDRIAL_PROTEIN_COMPLEX\n",
      "cat 5028, mi = 0.00507298, GO_RESPIRATORY_CHAIN\n",
      "cat 1742, mi = 0.00498218, GO_INNER_MITOCHONDRIAL_MEMBRANE_PROTEIN_COMPLEX\n",
      "cat 2217, mi = 0.00489426, GO_MITOCHONDRIAL_MEMBRANE_PART\n",
      "cat 4041, mi = 0.00453213, GO_PROTEIN_TARGETING_TO_MEMBRANE\n",
      "cat 1139, mi = 0.00430331, GO_ELECTRON_TRANSPORT_CHAIN\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 5718, mi = 0.009, pvalue = 0.000, GO_TRANSLATIONAL_INITIATION\n",
      "Passed: cat 2219, mi = 0.005, pvalue = 0.000, GO_MITOCHONDRIAL_PROTEIN_COMPLEX\n",
      "Passed: cat 5232, mi = 0.004, pvalue = 0.000, GO_RIBONUCLEOPROTEIN_COMPLEX_LOCALIZATION\n",
      "Passed: cat 1087, mi = 0.004, pvalue = 0.000, GO_DNA_PACKAGING_COMPLEX\n",
      "Passed: cat 5380, mi = 0.003, pvalue = 0.000, GO_SIGNAL_RELEASE\n",
      "Passed: cat 5755, mi = 0.003, pvalue = 0.000, GO_TRNA_METABOLIC_PROCESS\n",
      "Passed: cat 1925, mi = 0.003, pvalue = 0.000, GO_LIGAND_GATED_CHANNEL_ACTIVITY\n",
      "Passed: cat 5020, mi = 0.003, pvalue = 0.000, GO_REPLICATION_FORK\n",
      "Passed: cat 3379, mi = 0.003, pvalue = 0.000, GO_POSITIVE_REGULATION_OF_BLOOD_CIRCULATION\n",
      "Passed: cat 164, mi = 0.002, pvalue = 0.000, GO_ANAPHASE_PROMOTING_COMPLEX_DEPENDENT_CATABOLIC_PROCESS\n",
      "Passed: cat 5719, mi = 0.002, pvalue = 0.000, GO_TRANSLATIONAL_TERMINATION\n",
      "Passed: cat 1862, mi = 0.002, pvalue = 0.000, GO_KERATIN_FILAMENT\n",
      "Passed: cat 1542, mi = 0.002, pvalue = 0.000, GO_GOLGI_TO_ENDOSOME_TRANSPORT\n",
      "Passed: cat 5283, mi = 0.002, pvalue = 0.000, GO_RNA_POLYMERASE_II_TRANSCRIPTION_COREPRESSOR_ACTIVITY\n",
      "Passed: cat 5421, mi = 0.002, pvalue = 0.001, GO_SMALL_NUCLEAR_RIBONUCLEOPROTEIN_COMPLEX\n",
      "Passed: cat 3913, mi = 0.002, pvalue = 0.000, GO_PROTEIN_ACETYLATION\n",
      "Passed: cat 1133, mi = 0.002, pvalue = 0.000, GO_EAR_DEVELOPMENT\n",
      "Passed: cat 4790, mi = 0.002, pvalue = 0.000, GO_REGULATION_OF_PROTEIN_LOCALIZATION_TO_CHROMOSOME_TELOMERIC_REGION\n",
      "Passed: cat 214, mi = 0.002, pvalue = 0.000, GO_APICOLATERAL_PLASMA_MEMBRANE\n",
      "Passed: cat 1342, mi = 0.002, pvalue = 0.001, GO_EXTRACELLULAR_MATRIX_COMPONENT\n",
      "Passed: cat 4472, mi = 0.002, pvalue = 0.001, GO_REGULATION_OF_GRANULOCYTE_CHEMOTAXIS\n",
      "Passed: cat 3936, mi = 0.002, pvalue = 0.002, GO_PROTEIN_DEALKYLATION\n",
      "Passed: cat 2989, mi = 0.002, pvalue = 0.000, GO_N_ACETYLGLUCOSAMINE_METABOLIC_PROCESS\n",
      "Passed: cat 3876, mi = 0.002, pvalue = 0.001, GO_PRENYLTRANSFERASE_ACTIVITY\n",
      "Passed: cat 5386, mi = 0.002, pvalue = 0.002, GO_SIGNAL_TRANSDUCTION_IN_ABSENCE_OF_LIGAND\n",
      "Passed: cat 4838, mi = 0.002, pvalue = 0.001, GO_REGULATION_OF_RESPONSE_TO_EXTRACELLULAR_STIMULUS\n",
      "Passed: cat 736, mi = 0.002, pvalue = 0.002, GO_CHANNEL_REGULATOR_ACTIVITY\n",
      "Passed: cat 76, mi = 0.002, pvalue = 0.001, GO_ACUTE_PHASE_RESPONSE\n",
      "Passed: cat 5677, mi = 0.001, pvalue = 0.003, GO_TRANSCRIPTION_ELONGATION_FACTOR_COMPLEX\n",
      "Passed: cat 4983, mi = 0.001, pvalue = 0.004, GO_REGULATION_OF_VACUOLAR_TRANSPORT\n",
      "Passed: cat 1038, mi = 0.001, pvalue = 0.004, GO_DIGESTION\n",
      "Passed: cat 710, mi = 0.001, pvalue = 0.002, GO_CENTRIOLE\n",
      "Passed: cat 5150, mi = 0.001, pvalue = 0.003, GO_RESPONSE_TO_OXYGEN_RADICAL\n",
      "Passed: cat 5407, mi = 0.001, pvalue = 0.003, GO_SKELETAL_MUSCLE_CONTRACTION\n",
      "Passed: cat 1236, mi = 0.001, pvalue = 0.002, GO_EPHRIN_RECEPTOR_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 1667, mi = 0.001, pvalue = 0.006, GO_HOMOPHILIC_CELL_ADHESION_VIA_PLASMA_MEMBRANE_ADHESION_MOLECULES\n",
      "Passed: cat 1006, mi = 0.001, pvalue = 0.000, GO_DETECTION_OF_CHEMICAL_STIMULUS_INVOLVED_IN_SENSORY_PERCEPTION_OF_TASTE\n",
      "Didn't pass (cutoff=0.0050): cat 2381, mi = 0.001, pvalue = 0.008, GO_NADP_BINDING\n",
      "Passed: cat 876, mi = 0.001, pvalue = 0.003, GO_CORNIFIED_ENVELOPE\n",
      "Didn't pass (cutoff=0.0050): cat 5494, mi = 0.001, pvalue = 0.013, GO_SPINDLE_LOCALIZATION\n",
      "Didn't pass (cutoff=0.0050): cat 2136, mi = 0.001, pvalue = 0.009, GO_METALLOPEPTIDASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 1095, mi = 0.001, pvalue = 0.008, GO_DNA_REPLICATION_CHECKPOINT\n",
      "Didn't pass (cutoff=0.0050): cat 4521, mi = 0.001, pvalue = 0.009, GO_REGULATION_OF_INSULIN_LIKE_GROWTH_FACTOR_RECEPTOR_SIGNALING_PATHWAY\n",
      "Didn't pass (cutoff=0.0050): cat 4122, mi = 0.001, pvalue = 0.019, GO_RAC_PROTEIN_SIGNAL_TRANSDUCTION\n",
      "Didn't pass (cutoff=0.0050): cat 1239, mi = 0.001, pvalue = 0.017, GO_EPIBOLY\n",
      "Passed: cat 5297, mi = 0.001, pvalue = 0.002, GO_ROUNDABOUT_SIGNALING_PATHWAY\n",
      "Didn't pass (cutoff=0.0050): cat 2380, mi = 0.001, pvalue = 0.012, GO_NADPH_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 2792, mi = 0.001, pvalue = 0.019, GO_NEGATIVE_REGULATION_OF_VIRAL_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 4072, mi = 0.001, pvalue = 0.031, GO_PTERIDINE_CONTAINING_COMPOUND_BIOSYNTHETIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 3294, mi = 0.001, pvalue = 0.012, GO_PINOCYTOSIS\n",
      "Didn't pass (cutoff=0.0050): cat 2734, mi = 0.001, pvalue = 0.018, GO_NEGATIVE_REGULATION_OF_RHO_PROTEIN_SIGNAL_TRANSDUCTION\n",
      "Didn't pass (cutoff=0.0050): cat 3340, mi = 0.001, pvalue = 0.030, GO_POLY_PURINE_TRACT_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 2953, mi = 0.001, pvalue = 0.029, GO_NUCLEOSIDE_DIPHOSPHATE_KINASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 1401, mi = 0.001, pvalue = 0.011, GO_FILAMIN_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 78, mi = 0.001, pvalue = 0.036, GO_ACYLGLYCEROL_O_ACYLTRANSFERASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 5560, mi = 0.001, pvalue = 0.037, GO_SULFUR_COMPOUND_CATABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 121, mi = 0.001, pvalue = 0.029, GO_ALPHA_AMINO_ACID_BIOSYNTHETIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 172, mi = 0.001, pvalue = 0.036, GO_ANCHORED_COMPONENT_OF_PLASMA_MEMBRANE\n",
      "Didn't pass (cutoff=0.0050): cat 5193, mi = 0.001, pvalue = 0.035, GO_RESPONSE_TO_XENOBIOTIC_STIMULUS\n",
      "Didn't pass (cutoff=0.0050): cat 3114, mi = 0.001, pvalue = 0.028, GO_OXYGEN_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 3554, mi = 0.001, pvalue = 0.006, GO_POSITIVE_REGULATION_OF_INSULIN_LIKE_GROWTH_FACTOR_RECEPTOR_SIGNALING_PATHWAY\n",
      "Didn't pass (cutoff=0.0050): cat 1077, mi = 0.001, pvalue = 0.015, GO_DNA_INTEGRATION\n",
      "Didn't pass (cutoff=0.0050): cat 944, mi = 0.001, pvalue = 0.035, GO_CYTOPLASMIC_UBIQUITIN_LIGASE_COMPLEX\n",
      "Didn't pass (cutoff=0.0050): cat 335, mi = 0.001, pvalue = 0.029, GO_BIOTIN_METABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 436, mi = 0.001, pvalue = 0.049, GO_CARBOXYPEPTIDASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 1533, mi = 0.001, pvalue = 0.045, GO_GOLGI_ASSOCIATED_VESICLE_MEMBRANE\n",
      "Total number of categories evaluated = 738.\n",
      "Number of categories that passed the tests = 38\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM2_2.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM2_2.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 30, 35, 0.0237327413421468\n",
      "best merge is 18, 30, 0.0287208180521594\n",
      "best merge is 11, 34, 0.030728586875974\n",
      "best merge is 5, 10, 0.0418096724010468\n",
      "best merge is 18, 36, 0.0459093739216702\n",
      "best merge is 11, 18, 0.0633433541202821\n",
      "best merge is 0, 17, 0.0895625396965062\n",
      "best merge is 1, 9, 0.101869832133953\n",
      "best merge is 3, 37, 0.102107397231775\n",
      "best merge is 14, 15, 0.141606457038664\n",
      "best merge is 4, 11, 0.145177147729333\n",
      "best merge is 6, 20, 0.153717402568731\n",
      "best merge is 6, 33, 0.159821437392473\n",
      "best merge is 1, 5, 0.171624292603672\n",
      "best merge is 22, 24, 0.185932203834106\n",
      "best merge is 4, 6, 0.209370156275828\n",
      "best merge is 29, 31, 0.215959481593964\n",
      "best merge is 12, 29, 0.261244424551771\n",
      "best merge is 7, 22, 0.26508878669523\n",
      "best merge is 1, 14, 0.304199624666536\n",
      "best merge is 3, 13, 0.315169349435075\n",
      "best merge is 28, 32, 0.325550224513714\n",
      "best merge is 8, 27, 0.364066608278374\n",
      "best merge is 3, 19, 0.400937283364343\n",
      "best merge is 4, 16, 0.434237509907015\n",
      "best merge is 7, 12, 0.45227194669487\n",
      "best merge is 2, 25, 0.459950053977032\n",
      "best merge is 4, 8, 0.496070658505694\n",
      "best merge is 0, 1, 0.523201542442129\n",
      "best merge is 21, 28, 0.552384403978138\n",
      "best merge is 4, 26, 0.616123765784788\n",
      "best merge is 0, 21, 0.708462284915883\n",
      "best merge is 7, 23, 0.740826202012271\n",
      "best merge is 2, 7, 0.774331324033071\n",
      "order classes\n",
      "11 - 1.61060473606823\n",
      "2 - 4.36952551278181\n",
      "4 - 2.29811881110651\n",
      "14 - 0.834426254767796\n",
      "7 - 2.2090879762652\n",
      "4 11 34 18 30 35 36 6 20 33 16 8 27 26\n",
      "2 25\n",
      "3 37 13 19\n",
      "0 17 1 9 5 10 14 15 21 28 32\n",
      "7 22 24 12 29 31 23\n",
      "Done.Start drawing\n",
      "101.03\t0.00\n",
      "GO_POSITIVE_REGULATION_OF_BLOOD_CIRCULATION\n",
      "GO_CHANNEL_REGULATOR_ACTIVITY\n",
      "GO_DNA_PACKAGING_COMPLEX\n",
      "GO_ROUNDABOUT_SIGNALING_PATHWAY\n",
      "GO_RNA_POLYMERASE_II_TRANSCRIPTION_COREPRESSOR_ACTIVITY\n",
      "GO_EXTRACELLULAR_MATRIX_COMPONENT\n",
      "GO_REPLICATION_FORK\n",
      "GO_N_ACETYLGLUCOSAMINE_METABOLIC_PROCESS\n",
      "GO_SIGNAL_TRANSDUCTION_IN_ABSENCE_OF_LIGAND\n",
      "GO_REGULATION_OF_VACUOLAR_TRANSPORT\n",
      "GO_RIBONUCLEOPROTEIN_COMPLEX_LOCALIZATION\n",
      "GO_PROTEIN_DEALKYLATION\n",
      "GO_GOLGI_TO_ENDOSOME_TRANSPORT\n",
      "GO_CENTRIOLE\n",
      "GO_PRENYLTRANSFERASE_ACTIVITY\n",
      "GO_REGULATION_OF_RESPONSE_TO_EXTRACELLULAR_STIMULUS\n",
      "GO_SMALL_NUCLEAR_RIBONUCLEOPROTEIN_COMPLEX\n",
      "GO_PROTEIN_ACETYLATION\n",
      "GO_TRANSCRIPTION_ELONGATION_FACTOR_COMPLEX\n",
      "GO_REGULATION_OF_PROTEIN_LOCALIZATION_TO_CHROMOSOME_TELOMERIC_REGION\n",
      "GO_MITOCHONDRIAL_PROTEIN_COMPLEX\n",
      "GO_ANAPHASE_PROMOTING_COMPLEX_DEPENDENT_CATABOLIC_PROCESS\n",
      "GO_TRNA_METABOLIC_PROCESS\n",
      "GO_TRANSLATIONAL_TERMINATION\n",
      "GO_TRANSLATIONAL_INITIATION\n",
      "GO_RESPONSE_TO_OXYGEN_RADICAL\n",
      "GO_SIGNAL_RELEASE\n",
      "GO_KERATIN_FILAMENT\n",
      "GO_EPHRIN_RECEPTOR_ACTIVITY\n",
      "GO_APICOLATERAL_PLASMA_MEMBRANE\n",
      "GO_DIGESTION\n",
      "GO_DETECTION_OF_CHEMICAL_STIMULUS_INVOLVED_IN_SENSORY_PERCEPTION_OF_TASTE\n",
      "GO_CORNIFIED_ENVELOPE\n",
      "GO_LIGAND_GATED_CHANNEL_ACTIVITY\n",
      "GO_REGULATION_OF_GRANULOCYTE_CHEMOTAXIS\n",
      "GO_SKELETAL_MUSCLE_CONTRACTION\n",
      "GO_EAR_DEVELOPMENT\n",
      "GO_ACUTE_PHASE_RESPONSE\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM2_2.csv_PAGE/filt_FC_LVM2_2.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM2_2.csv_PAGE/filt_FC_LVM2_2.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM2_2.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM2_2.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 30, 35, 0.0237327413421468\n",
      "best merge is 18, 30, 0.0287208180521594\n",
      "best merge is 11, 34, 0.030728586875974\n",
      "best merge is 5, 10, 0.0418096724010468\n",
      "best merge is 18, 36, 0.0459093739216702\n",
      "best merge is 11, 18, 0.0633433541202821\n",
      "best merge is 0, 17, 0.0895625396965062\n",
      "best merge is 1, 9, 0.101869832133953\n",
      "best merge is 3, 37, 0.102107397231775\n",
      "best merge is 14, 15, 0.141606457038664\n",
      "best merge is 4, 11, 0.145177147729333\n",
      "best merge is 6, 20, 0.153717402568731\n",
      "best merge is 6, 33, 0.159821437392473\n",
      "best merge is 1, 5, 0.171624292603672\n",
      "best merge is 22, 24, 0.185932203834106\n",
      "best merge is 4, 6, 0.209370156275828\n",
      "best merge is 29, 31, 0.215959481593964\n",
      "best merge is 12, 29, 0.261244424551771\n",
      "best merge is 7, 22, 0.26508878669523\n",
      "best merge is 1, 14, 0.304199624666536\n",
      "best merge is 3, 13, 0.315169349435075\n",
      "best merge is 28, 32, 0.325550224513714\n",
      "best merge is 8, 27, 0.364066608278374\n",
      "best merge is 3, 19, 0.400937283364343\n",
      "best merge is 4, 16, 0.434237509907015\n",
      "best merge is 7, 12, 0.45227194669487\n",
      "best merge is 2, 25, 0.459950053977032\n",
      "best merge is 4, 8, 0.496070658505694\n",
      "best merge is 0, 1, 0.523201542442129\n",
      "best merge is 21, 28, 0.552384403978138\n",
      "best merge is 4, 26, 0.616123765784788\n",
      "best merge is 0, 21, 0.708462284915883\n",
      "best merge is 7, 23, 0.740826202012271\n",
      "best merge is 2, 7, 0.774331324033071\n",
      "order classes\n",
      "11 - 1.61060473606823\n",
      "2 - 4.36952551278181\n",
      "4 - 2.29811881110651\n",
      "14 - 0.834426254767796\n",
      "7 - 2.2090879762652\n",
      "4 11 34 18 30 35 36 6 20 33 16 8 27 26\n",
      "2 25\n",
      "3 37 13 19\n",
      "0 17 1 9 5 10 14 15 21 28 32\n",
      "7 22 24 12 29 31 23\n",
      "Done.-19.8544986637632\t42\t0\n",
      "-19.6570325645848\t42\t0.291002672473523\n",
      "-19.5759675343957\t42\t0.394932198356924\n",
      "-19.5219241809364\t42\t0.453132732851628\n",
      "-19.4720380085123\t42\t0.503018905275661\n",
      "-19.417994655053\t42\t0.552905077699693\n",
      "-19.3514797584876\t42\t0.611105612194398\n",
      "-19.2454716420865\t42\t0.685934870830446\n",
      "1.41156092249827\t42\t0.823121844996536\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM2_2.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_P_LVM1_3.csv --species=human_ensembl_msigdb_c5 --exptype=continuous --ebins=9\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM2_3.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c5.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c5/human_ensembl_msigdb_c5.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM2_3.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 5917\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t5917\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 5917\n",
      "cat 5245, mi = 0.00905724, GO_RIBOSOMAL_SUBUNIT\n",
      "cat 5718, mi = 0.00884809, GO_TRANSLATIONAL_INITIATION\n",
      "cat 957, mi = 0.00790769, GO_CYTOSOLIC_RIBOSOME\n",
      "cat 1292, mi = 0.00754507, GO_ESTABLISHMENT_OF_PROTEIN_LOCALIZATION_TO_ENDOPLASMIC_RETICULUM\n",
      "cat 3995, mi = 0.00739725, GO_PROTEIN_LOCALIZATION_TO_ENDOPLASMIC_RETICULUM\n",
      "cat 5402, mi = 0.00724541, GO_SISTER_CHROMATID_SEGREGATION\n",
      "cat 1889, mi = 0.00637768, GO_LARGE_RIBOSOMAL_SUBUNIT\n",
      "cat 2219, mi = 0.00637481, GO_MITOCHONDRIAL_PROTEIN_COMPLEX\n",
      "cat 2928, mi = 0.00621914, GO_NUCLEAR_TRANSCRIBED_MRNA_CATABOLIC_PROCESS_NONSENSE_MEDIATED_DECAY\n",
      "cat 5499, mi = 0.00619278, GO_SPLICEOSOMAL_COMPLEX\n",
      "cat 775, mi = 0.00612446, GO_CHROMOSOME_CENTROMERIC_REGION\n",
      "cat 954, mi = 0.00602189, GO_CYTOSOLIC_LARGE_RIBOSOMAL_SUBUNIT\n",
      "cat 1742, mi = 0.00555170, GO_INNER_MITOCHONDRIAL_MEMBRANE_PROTEIN_COMPLEX\n",
      "cat 2330, mi = 0.00544728, GO_MULTI_ORGANISM_METABOLIC_PROCESS\n",
      "cat 5401, mi = 0.00527955, GO_SISTER_CHROMATID_COHESION\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 5245, mi = 0.009, pvalue = 0.000, GO_RIBOSOMAL_SUBUNIT\n",
      "Passed: cat 5402, mi = 0.007, pvalue = 0.000, GO_SISTER_CHROMATID_SEGREGATION\n",
      "Passed: cat 2219, mi = 0.006, pvalue = 0.000, GO_MITOCHONDRIAL_PROTEIN_COMPLEX\n",
      "Passed: cat 5499, mi = 0.006, pvalue = 0.000, GO_SPLICEOSOMAL_COMPLEX\n",
      "Passed: cat 5755, mi = 0.004, pvalue = 0.000, GO_TRNA_METABOLIC_PROCESS\n",
      "Passed: cat 3910, mi = 0.003, pvalue = 0.000, GO_PROTEASOME_COMPLEX\n",
      "Passed: cat 783, mi = 0.003, pvalue = 0.000, GO_CHROMOSOME_TELOMERIC_REGION\n",
      "Passed: cat 5804, mi = 0.003, pvalue = 0.000, GO_UBIQUITIN_LIKE_PROTEIN_LIGASE_ACTIVITY\n",
      "Passed: cat 5885, mi = 0.003, pvalue = 0.000, GO_VOLTAGE_GATED_CATION_CHANNEL_ACTIVITY\n",
      "Passed: cat 5351, mi = 0.002, pvalue = 0.000, GO_SENSORY_PERCEPTION_OF_TASTE\n",
      "Passed: cat 462, mi = 0.002, pvalue = 0.000, GO_CARDIOCYTE_DIFFERENTIATION\n",
      "Passed: cat 171, mi = 0.002, pvalue = 0.000, GO_ANCHORED_COMPONENT_OF_MEMBRANE\n",
      "Passed: cat 5273, mi = 0.002, pvalue = 0.000, GO_RNA_POLYMERASE_COMPLEX\n",
      "Passed: cat 4321, mi = 0.002, pvalue = 0.000, GO_REGULATION_OF_CHEMOTAXIS\n",
      "Passed: cat 2863, mi = 0.002, pvalue = 0.000, GO_NEUROTRANSMITTER_TRANSPORT\n",
      "Passed: cat 5721, mi = 0.002, pvalue = 0.000, GO_TRANSLATION_FACTOR_ACTIVITY_RNA_BINDING\n",
      "Passed: cat 2967, mi = 0.002, pvalue = 0.000, GO_NUCLEOSOME_BINDING\n",
      "Passed: cat 710, mi = 0.002, pvalue = 0.000, GO_CENTRIOLE\n",
      "Passed: cat 2420, mi = 0.002, pvalue = 0.000, GO_NEGATIVE_REGULATION_OF_AUTOPHAGY\n",
      "Passed: cat 4023, mi = 0.002, pvalue = 0.000, GO_PROTEIN_PHOSPHATASE_TYPE_2A_COMPLEX\n",
      "Passed: cat 4227, mi = 0.002, pvalue = 0.000, GO_REGULATION_OF_CALCIUM_ION_TRANSPORT_INTO_CYTOSOL\n",
      "Passed: cat 1276, mi = 0.002, pvalue = 0.001, GO_ER_TO_GOLGI_VESICLE_MEDIATED_TRANSPORT\n",
      "Passed: cat 1670, mi = 0.002, pvalue = 0.001, GO_HORMONE_ACTIVITY\n",
      "Passed: cat 2006, mi = 0.002, pvalue = 0.001, GO_LYSOSOMAL_TRANSPORT\n",
      "Passed: cat 4009, mi = 0.002, pvalue = 0.001, GO_PROTEIN_MODIFICATION_BY_SMALL_PROTEIN_REMOVAL\n",
      "Passed: cat 836, mi = 0.002, pvalue = 0.000, GO_COFACTOR_BIOSYNTHETIC_PROCESS\n",
      "Passed: cat 908, mi = 0.002, pvalue = 0.001, GO_CYCLIN_DEPENDENT_PROTEIN_SERINE_THREONINE_KINASE_REGULATOR_ACTIVITY\n",
      "Passed: cat 1779, mi = 0.002, pvalue = 0.001, GO_INTERMEDIATE_FILAMENT\n",
      "Passed: cat 4127, mi = 0.002, pvalue = 0.001, GO_RAS_PROTEIN_SIGNAL_TRANSDUCTION\n",
      "Passed: cat 1403, mi = 0.002, pvalue = 0.001, GO_FILOPODIUM_ASSEMBLY\n",
      "Passed: cat 2608, mi = 0.001, pvalue = 0.002, GO_NEGATIVE_REGULATION_OF_LEUKOCYTE_MEDIATED_IMMUNITY\n",
      "Passed: cat 1342, mi = 0.001, pvalue = 0.004, GO_EXTRACELLULAR_MATRIX_COMPONENT\n",
      "Passed: cat 2744, mi = 0.001, pvalue = 0.003, GO_NEGATIVE_REGULATION_OF_SMOOTHENED_SIGNALING_PATHWAY\n",
      "Passed: cat 979, mi = 0.001, pvalue = 0.004, GO_DEMETHYLATION\n",
      "Didn't pass (cutoff=0.0050): cat 5424, mi = 0.001, pvalue = 0.005, GO_SMALL_SUBUNIT_PROCESSOME\n",
      "Didn't pass (cutoff=0.0050): cat 693, mi = 0.001, pvalue = 0.005, GO_CELL_REDOX_HOMEOSTASIS\n",
      "Passed: cat 1646, mi = 0.001, pvalue = 0.003, GO_HISTONE_H3_K9_MODIFICATION\n",
      "Passed: cat 1892, mi = 0.001, pvalue = 0.004, GO_LATERAL_PLASMA_MEMBRANE\n",
      "Didn't pass (cutoff=0.0050): cat 1630, mi = 0.001, pvalue = 0.006, GO_HIPPO_SIGNALING\n",
      "Didn't pass (cutoff=0.0050): cat 752, mi = 0.001, pvalue = 0.007, GO_CHLORIDE_TRANSPORT\n",
      "Passed: cat 247, mi = 0.001, pvalue = 0.002, GO_ASYMMETRIC_CELL_DIVISION\n",
      "Didn't pass (cutoff=0.0050): cat 474, mi = 0.001, pvalue = 0.006, GO_CATECHOLAMINE_METABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 1253, mi = 0.001, pvalue = 0.008, GO_EPITHELIAL_CELL_MORPHOGENESIS\n",
      "Didn't pass (cutoff=0.0050): cat 1522, mi = 0.001, pvalue = 0.006, GO_GLYCOSPHINGOLIPID_METABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 226, mi = 0.001, pvalue = 0.006, GO_ARACHIDONIC_ACID_METABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 1366, mi = 0.001, pvalue = 0.007, GO_FATTY_ACID_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 3857, mi = 0.001, pvalue = 0.008, GO_POSTTRANSCRIPTIONAL_GENE_SILENCING\n",
      "Didn't pass (cutoff=0.0050): cat 1944, mi = 0.001, pvalue = 0.007, GO_LIPID_OXIDATION\n",
      "Didn't pass (cutoff=0.0050): cat 73, mi = 0.001, pvalue = 0.008, GO_ACTOMYOSIN\n",
      "Didn't pass (cutoff=0.0050): cat 5407, mi = 0.001, pvalue = 0.006, GO_SKELETAL_MUSCLE_CONTRACTION\n",
      "Didn't pass (cutoff=0.0050): cat 1745, mi = 0.001, pvalue = 0.009, GO_INORGANIC_ANION_TRANSMEMBRANE_TRANSPORTER_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 1597, mi = 0.001, pvalue = 0.009, GO_HEAT_SHOCK_PROTEIN_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 3245, mi = 0.001, pvalue = 0.018, GO_PHOSPHATIDYLINOSITOL_PHOSPHATE_PHOSPHATASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 137, mi = 0.001, pvalue = 0.011, GO_AMINE_CATABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 1378, mi = 0.001, pvalue = 0.018, GO_FATTY_ACYL_COA_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 4921, mi = 0.001, pvalue = 0.011, GO_REGULATION_OF_TELOMERASE_RNA_LOCALIZATION_TO_CAJAL_BODY\n",
      "Didn't pass (cutoff=0.0050): cat 4705, mi = 0.001, pvalue = 0.011, GO_REGULATION_OF_NUCLEOBASE_CONTAINING_COMPOUND_TRANSPORT\n",
      "Didn't pass (cutoff=0.0050): cat 1804, mi = 0.001, pvalue = 0.019, GO_INTRAMOLECULAR_OXIDOREDUCTASE_ACTIVITY_TRANSPOSING_S_S_BONDS\n",
      "Didn't pass (cutoff=0.0050): cat 4421, mi = 0.001, pvalue = 0.009, GO_REGULATION_OF_ESTABLISHMENT_OF_PROTEIN_LOCALIZATION_TO_CHROMOSOME\n",
      "Didn't pass (cutoff=0.0050): cat 2268, mi = 0.001, pvalue = 0.013, GO_MONOCARBOXYLIC_ACID_CATABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 1196, mi = 0.001, pvalue = 0.006, GO_ENDOPLASMIC_RETICULUM_CHAPERONE_COMPLEX\n",
      "Total number of categories evaluated = 651.\n",
      "Number of categories that passed the tests = 37\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM2_3.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM2_3.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 8, 11, 0.0543595750224908\n",
      "best merge is 3, 5, 0.0723285609065479\n",
      "best merge is 9, 14, 0.0729010059151918\n",
      "best merge is 13, 31, 0.0796103440332552\n",
      "best merge is 1, 17, 0.0802344171206391\n",
      "best merge is 8, 22, 0.0821778585713872\n",
      "best merge is 19, 34, 0.0942754144376294\n",
      "best merge is 2, 15, 0.0972094231279408\n",
      "best merge is 13, 20, 0.12336753903273\n",
      "best merge is 4, 25, 0.132299488811531\n",
      "best merge is 3, 12, 0.133474788860803\n",
      "best merge is 0, 2, 0.164477181807477\n",
      "best merge is 18, 19, 0.168316121336415\n",
      "best merge is 16, 21, 0.178114616600014\n",
      "best merge is 13, 30, 0.18789768420248\n",
      "best merge is 27, 35, 0.202578078831664\n",
      "best merge is 6, 24, 0.203575637861372\n",
      "best merge is 9, 29, 0.212103088020484\n",
      "best merge is 3, 4, 0.238795974252642\n",
      "best merge is 7, 23, 0.259318431211006\n",
      "best merge is 1, 13, 0.291212015054233\n",
      "best merge is 7, 36, 0.317796585958196\n",
      "best merge is 8, 9, 0.341207167478574\n",
      "best merge is 3, 16, 0.346655861270476\n",
      "best merge is 1, 10, 0.373187093472878\n",
      "best merge is 6, 32, 0.387669851367297\n",
      "best merge is 8, 27, 0.388374387465453\n",
      "best merge is 3, 33, 0.427178259890601\n",
      "best merge is 3, 6, 0.479878740180581\n",
      "best merge is 26, 28, 0.541565639615857\n",
      "best merge is 0, 3, 0.566309078336024\n",
      "best merge is 1, 26, 0.606270894395853\n",
      "best merge is 0, 7, 0.717348532207939\n",
      "order classes\n",
      "14 - 1.0827527630725\n",
      "9 - 1.47882754504105\n",
      "3 - 2.45272069005287\n",
      "8 - 1.25016029400114\n",
      "3 - 1.18100617133978\n",
      "0 2 15 3 5 12 4 25 16 21 33 6 24 32\n",
      "1 17 13 31 20 30 10 26 28\n",
      "7 23 36\n",
      "8 11 22 9 14 29 27 35\n",
      "18 19 34\n",
      "Done.Start drawing\n",
      "69.14\t0.00\n",
      "GO_SISTER_CHROMATID_SEGREGATION\n",
      "GO_CENTRIOLE\n",
      "GO_REGULATION_OF_CHEMOTAXIS\n",
      "GO_EXTRACELLULAR_MATRIX_COMPONENT\n",
      "GO_REGULATION_OF_CALCIUM_ION_TRANSPORT_INTO_CYTOSOL\n",
      "GO_NEGATIVE_REGULATION_OF_LEUKOCYTE_MEDIATED_IMMUNITY\n",
      "GO_CARDIOCYTE_DIFFERENTIATION\n",
      "GO_RAS_PROTEIN_SIGNAL_TRANSDUCTION\n",
      "GO_NEGATIVE_REGULATION_OF_AUTOPHAGY\n",
      "GO_PROTEIN_PHOSPHATASE_TYPE_2A_COMPLEX\n",
      "GO_HISTONE_H3_K9_MODIFICATION\n",
      "GO_RNA_POLYMERASE_COMPLEX\n",
      "GO_UBIQUITIN_LIKE_PROTEIN_LIGASE_ACTIVITY\n",
      "GO_LYSOSOMAL_TRANSPORT\n",
      "GO_ASYMMETRIC_CELL_DIVISION\n",
      "GO_TRNA_METABOLIC_PROCESS\n",
      "GO_CHROMOSOME_TELOMERIC_REGION\n",
      "GO_PROTEIN_MODIFICATION_BY_SMALL_PROTEIN_REMOVAL\n",
      "GO_NEGATIVE_REGULATION_OF_SMOOTHENED_SIGNALING_PATHWAY\n",
      "GO_SPLICEOSOMAL_COMPLEX\n",
      "GO_COFACTOR_BIOSYNTHETIC_PROCESS\n",
      "GO_NUCLEOSOME_BINDING\n",
      "GO_ER_TO_GOLGI_VESICLE_MEDIATED_TRANSPORT\n",
      "GO_PROTEASOME_COMPLEX\n",
      "GO_DEMETHYLATION\n",
      "GO_CYCLIN_DEPENDENT_PROTEIN_SERINE_THREONINE_KINASE_REGULATOR_ACTIVITY\n",
      "GO_RIBOSOMAL_SUBUNIT\n",
      "GO_MITOCHONDRIAL_PROTEIN_COMPLEX\n",
      "GO_TRANSLATION_FACTOR_ACTIVITY_RNA_BINDING\n",
      "GO_VOLTAGE_GATED_CATION_CHANNEL_ACTIVITY\n",
      "GO_ANCHORED_COMPONENT_OF_MEMBRANE\n",
      "GO_HORMONE_ACTIVITY\n",
      "GO_SENSORY_PERCEPTION_OF_TASTE\n",
      "GO_NEUROTRANSMITTER_TRANSPORT\n",
      "GO_FILOPODIUM_ASSEMBLY\n",
      "GO_INTERMEDIATE_FILAMENT\n",
      "GO_LATERAL_PLASMA_MEMBRANE\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM2_3.csv_PAGE/filt_FC_LVM2_3.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM2_3.csv_PAGE/filt_FC_LVM2_3.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM2_3.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM2_3.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 8, 11, 0.0543595750224908\n",
      "best merge is 3, 5, 0.0723285609065479\n",
      "best merge is 9, 14, 0.0729010059151918\n",
      "best merge is 13, 31, 0.0796103440332552\n",
      "best merge is 1, 17, 0.0802344171206391\n",
      "best merge is 8, 22, 0.0821778585713872\n",
      "best merge is 19, 34, 0.0942754144376294\n",
      "best merge is 2, 15, 0.0972094231279408\n",
      "best merge is 13, 20, 0.12336753903273\n",
      "best merge is 4, 25, 0.132299488811531\n",
      "best merge is 3, 12, 0.133474788860803\n",
      "best merge is 0, 2, 0.164477181807477\n",
      "best merge is 18, 19, 0.168316121336415\n",
      "best merge is 16, 21, 0.178114616600014\n",
      "best merge is 13, 30, 0.18789768420248\n",
      "best merge is 27, 35, 0.202578078831664\n",
      "best merge is 6, 24, 0.203575637861372\n",
      "best merge is 9, 29, 0.212103088020484\n",
      "best merge is 3, 4, 0.238795974252642\n",
      "best merge is 7, 23, 0.259318431211006\n",
      "best merge is 1, 13, 0.291212015054233\n",
      "best merge is 7, 36, 0.317796585958196\n",
      "best merge is 8, 9, 0.341207167478574\n",
      "best merge is 3, 16, 0.346655861270476\n",
      "best merge is 1, 10, 0.373187093472878\n",
      "best merge is 6, 32, 0.387669851367297\n",
      "best merge is 8, 27, 0.388374387465453\n",
      "best merge is 3, 33, 0.427178259890601\n",
      "best merge is 3, 6, 0.479878740180581\n",
      "best merge is 26, 28, 0.541565639615857\n",
      "best merge is 0, 3, 0.566309078336024\n",
      "best merge is 1, 26, 0.606270894395853\n",
      "best merge is 0, 7, 0.717348532207939\n",
      "order classes\n",
      "14 - 1.0827527630725\n",
      "9 - 1.47882754504105\n",
      "3 - 2.45272069005287\n",
      "8 - 1.25016029400114\n",
      "3 - 1.18100617133978\n",
      "0 2 15 3 5 12 4 25 16 21 33 6 24 32\n",
      "1 17 13 31 20 30 10 26 28\n",
      "7 23 36\n",
      "8 11 22 9 14 29 27 35\n",
      "18 19 34\n",
      "Done.-19.7843505929997\t42\t0\n",
      "-19.4715070870697\t42\t0.431298814000579\n",
      "-19.3166039919005\t42\t0.625687011859994\n",
      "-19.2133352617877\t42\t0.741105004339022\n",
      "-19.1191784784495\t42\t0.832224472085623\n",
      "-19.0189470639283\t42\t0.929418571015331\n",
      "-18.8974544402661\t42\t1.03268730112815\n",
      "-18.7000289268152\t42\t1.1724038183396\n",
      "1.71376916401504\t42\t1.42753832803008\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM2_3.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_P_LVM1_4.csv --species=human_ensembl_msigdb_c5 --exptype=continuous --ebins=9\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM2_4.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c5.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c5/human_ensembl_msigdb_c5.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM2_4.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 5917\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t5917\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 5917\n",
      "cat 5245, mi = 0.00958229, GO_RIBOSOMAL_SUBUNIT\n",
      "cat 957, mi = 0.00724004, GO_CYTOSOLIC_RIBOSOME\n",
      "cat 854, mi = 0.00715585, GO_CONDENSED_CHROMOSOME\n",
      "cat 1292, mi = 0.00693944, GO_ESTABLISHMENT_OF_PROTEIN_LOCALIZATION_TO_ENDOPLASMIC_RETICULUM\n",
      "cat 3995, mi = 0.00673756, GO_PROTEIN_LOCALIZATION_TO_ENDOPLASMIC_RETICULUM\n",
      "cat 2928, mi = 0.00663915, GO_NUCLEAR_TRANSCRIBED_MRNA_CATABOLIC_PROCESS_NONSENSE_MEDIATED_DECAY\n",
      "cat 1889, mi = 0.00663704, GO_LARGE_RIBOSOMAL_SUBUNIT\n",
      "cat 5718, mi = 0.00640283, GO_TRANSLATIONAL_INITIATION\n",
      "cat 5402, mi = 0.00621453, GO_SISTER_CHROMATID_SEGREGATION\n",
      "cat 2330, mi = 0.00565905, GO_MULTI_ORGANISM_METABOLIC_PROCESS\n",
      "cat 954, mi = 0.00514497, GO_CYTOSOLIC_LARGE_RIBOSOMAL_SUBUNIT\n",
      "cat 2219, mi = 0.00501460, GO_MITOCHONDRIAL_PROTEIN_COMPLEX\n",
      "cat 855, mi = 0.00501090, GO_CONDENSED_CHROMOSOME_CENTROMERIC_REGION\n",
      "cat 1742, mi = 0.00500247, GO_INNER_MITOCHONDRIAL_MEMBRANE_PROTEIN_COMPLEX\n",
      "cat 5401, mi = 0.00484245, GO_SISTER_CHROMATID_COHESION\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 5245, mi = 0.010, pvalue = 0.000, GO_RIBOSOMAL_SUBUNIT\n",
      "Passed: cat 854, mi = 0.007, pvalue = 0.000, GO_CONDENSED_CHROMOSOME\n",
      "Passed: cat 2219, mi = 0.005, pvalue = 0.000, GO_MITOCHONDRIAL_PROTEIN_COMPLEX\n",
      "Passed: cat 5755, mi = 0.004, pvalue = 0.000, GO_TRNA_METABOLIC_PROCESS\n",
      "Passed: cat 5499, mi = 0.003, pvalue = 0.000, GO_SPLICEOSOMAL_COMPLEX\n",
      "Passed: cat 1099, mi = 0.003, pvalue = 0.000, GO_DNA_REPLICATION_INITIATION\n",
      "Passed: cat 4321, mi = 0.002, pvalue = 0.000, GO_REGULATION_OF_CHEMOTAXIS\n",
      "Passed: cat 3910, mi = 0.002, pvalue = 0.000, GO_PROTEASOME_COMPLEX\n",
      "Passed: cat 5831, mi = 0.002, pvalue = 0.000, GO_VACUOLE_ORGANIZATION\n",
      "Passed: cat 5229, mi = 0.002, pvalue = 0.000, GO_RIBONUCLEOPROTEIN_COMPLEX_BINDING\n",
      "Passed: cat 4009, mi = 0.002, pvalue = 0.000, GO_PROTEIN_MODIFICATION_BY_SMALL_PROTEIN_REMOVAL\n",
      "Passed: cat 4887, mi = 0.002, pvalue = 0.000, GO_REGULATION_OF_STEM_CELL_DIFFERENTIATION\n",
      "Passed: cat 324, mi = 0.002, pvalue = 0.000, GO_BICARBONATE_TRANSPORT\n",
      "Passed: cat 79, mi = 0.002, pvalue = 0.000, GO_ACYL_COA_DEHYDROGENASE_ACTIVITY\n",
      "Passed: cat 1105, mi = 0.002, pvalue = 0.000, GO_DNA_TEMPLATED_TRANSCRIPTION_ELONGATION\n",
      "Passed: cat 1862, mi = 0.002, pvalue = 0.000, GO_KERATIN_FILAMENT\n",
      "Passed: cat 464, mi = 0.002, pvalue = 0.001, GO_CARGO_RECEPTOR_ACTIVITY\n",
      "Passed: cat 710, mi = 0.002, pvalue = 0.001, GO_CENTRIOLE\n",
      "Passed: cat 3198, mi = 0.002, pvalue = 0.000, GO_PHENOL_CONTAINING_COMPOUND_METABOLIC_PROCESS\n",
      "Passed: cat 5399, mi = 0.002, pvalue = 0.000, GO_SINGLE_STRANDED_DNA_DEPENDENT_ATPASE_ACTIVITY\n",
      "Passed: cat 1766, mi = 0.002, pvalue = 0.000, GO_INSULIN_SECRETION\n",
      "Passed: cat 130, mi = 0.002, pvalue = 0.000, GO_AMELOGENESIS\n",
      "Passed: cat 2827, mi = 0.002, pvalue = 0.001, GO_NEURONAL_POSTSYNAPTIC_DENSITY\n",
      "Passed: cat 3203, mi = 0.002, pvalue = 0.001, GO_PHOSPHATASE_REGULATOR_ACTIVITY\n",
      "Passed: cat 2096, mi = 0.002, pvalue = 0.001, GO_MEMBRANE_BIOGENESIS\n",
      "Passed: cat 2550, mi = 0.002, pvalue = 0.002, GO_NEGATIVE_REGULATION_OF_GENE_EXPRESSION_EPIGENETIC\n",
      "Passed: cat 4421, mi = 0.002, pvalue = 0.001, GO_REGULATION_OF_ESTABLISHMENT_OF_PROTEIN_LOCALIZATION_TO_CHROMOSOME\n",
      "Passed: cat 5885, mi = 0.001, pvalue = 0.003, GO_VOLTAGE_GATED_CATION_CHANNEL_ACTIVITY\n",
      "Passed: cat 3458, mi = 0.001, pvalue = 0.003, GO_POSITIVE_REGULATION_OF_CYTOKINESIS\n",
      "Passed: cat 2746, mi = 0.001, pvalue = 0.003, GO_NEGATIVE_REGULATION_OF_SMOOTH_MUSCLE_CELL_PROLIFERATION\n",
      "Passed: cat 4893, mi = 0.001, pvalue = 0.003, GO_REGULATION_OF_STEROL_TRANSPORT\n",
      "Passed: cat 805, mi = 0.001, pvalue = 0.005, GO_CLATHRIN_ADAPTOR_COMPLEX\n",
      "Didn't pass (cutoff=0.0050): cat 4125, mi = 0.001, pvalue = 0.005, GO_RAN_GTPASE_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 3996, mi = 0.001, pvalue = 0.008, GO_PROTEIN_LOCALIZATION_TO_GOLGI_APPARATUS\n",
      "Passed: cat 3839, mi = 0.001, pvalue = 0.001, GO_POSITIVE_REGULATION_OF_VACUOLE_ORGANIZATION\n",
      "Didn't pass (cutoff=0.0050): cat 5112, mi = 0.001, pvalue = 0.005, GO_RESPONSE_TO_INTERLEUKIN_6\n",
      "Didn't pass (cutoff=0.0050): cat 2889, mi = 0.001, pvalue = 0.008, GO_NON_MEMBRANE_SPANNING_PROTEIN_TYROSINE_KINASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 912, mi = 0.001, pvalue = 0.010, GO_CYSTEINE_TYPE_ENDOPEPTIDASE_INHIBITOR_ACTIVITY\n",
      "Passed: cat 2061, mi = 0.001, pvalue = 0.004, GO_MANNOSIDASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 1338, mi = 0.001, pvalue = 0.006, GO_EXTRACELLULAR_LIGAND_GATED_ION_CHANNEL_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 544, mi = 0.001, pvalue = 0.011, GO_CELLULAR_RESPONSE_TO_CAMP\n",
      "Passed: cat 5786, mi = 0.001, pvalue = 0.005, GO_T_CELL_RECEPTOR_COMPLEX\n",
      "Didn't pass (cutoff=0.0050): cat 995, mi = 0.001, pvalue = 0.007, GO_DEOXYRIBONUCLEOTIDE_BIOSYNTHETIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 5874, mi = 0.001, pvalue = 0.012, GO_VITAMIN_BINDING\n",
      "Passed: cat 1196, mi = 0.001, pvalue = 0.002, GO_ENDOPLASMIC_RETICULUM_CHAPERONE_COMPLEX\n",
      "Passed: cat 911, mi = 0.001, pvalue = 0.003, GO_CYSTEINE_TYPE_ENDOPEPTIDASE_ACTIVITY_INVOLVED_IN_APOPTOTIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 5356, mi = 0.001, pvalue = 0.018, GO_SERINE_FAMILY_AMINO_ACID_METABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 1548, mi = 0.001, pvalue = 0.019, GO_GPI_ANCHOR_METABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 913, mi = 0.001, pvalue = 0.013, GO_CYSTEINE_TYPE_ENDOPEPTIDASE_INHIBITOR_ACTIVITY_INVOLVED_IN_APOPTOTIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 5181, mi = 0.001, pvalue = 0.017, GO_RESPONSE_TO_TYPE_I_INTERFERON\n",
      "Passed: cat 5161, mi = 0.001, pvalue = 0.003, GO_RESPONSE_TO_PROTOZOAN\n",
      "Didn't pass (cutoff=0.0050): cat 1817, mi = 0.001, pvalue = 0.026, GO_INTRINSIC_COMPONENT_OF_GOLGI_MEMBRANE\n",
      "Didn't pass (cutoff=0.0050): cat 3714, mi = 0.001, pvalue = 0.006, GO_POSITIVE_REGULATION_OF_PRI_MIRNA_TRANSCRIPTION_FROM_RNA_POLYMERASE_II_PROMOTER\n",
      "Didn't pass (cutoff=0.0050): cat 2944, mi = 0.001, pvalue = 0.028, GO_NUCLEOCYTOPLASMIC_TRANSPORTER_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 2959, mi = 0.001, pvalue = 0.024, GO_NUCLEOSIDE_PHOSPHATE_CATABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 5424, mi = 0.001, pvalue = 0.028, GO_SMALL_SUBUNIT_PROCESSOME\n",
      "Didn't pass (cutoff=0.0050): cat 2192, mi = 0.001, pvalue = 0.028, GO_MICROVILLUS\n",
      "Didn't pass (cutoff=0.0050): cat 121, mi = 0.001, pvalue = 0.024, GO_ALPHA_AMINO_ACID_BIOSYNTHETIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 2132, mi = 0.001, pvalue = 0.011, GO_METALLOCARBOXYPEPTIDASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 5663, mi = 0.001, pvalue = 0.030, GO_TRANSCRIPTIONALLY_ACTIVE_CHROMATIN\n",
      "Didn't pass (cutoff=0.0050): cat 2150, mi = 0.001, pvalue = 0.026, GO_METHIONINE_BIOSYNTHETIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 1534, mi = 0.001, pvalue = 0.025, GO_GOLGI_CISTERNA\n",
      "Didn't pass (cutoff=0.0050): cat 1618, mi = 0.001, pvalue = 0.039, GO_HETEROTRIMERIC_G_PROTEIN_COMPLEX\n",
      "Didn't pass (cutoff=0.0050): cat 426, mi = 0.001, pvalue = 0.041, GO_CARBOHYDRATE_PHOSPHORYLATION\n",
      "Didn't pass (cutoff=0.0050): cat 3216, mi = 0.001, pvalue = 0.014, GO_PHOSPHATIDYLETHANOLAMINE_ACYL_CHAIN_REMODELING\n",
      "Didn't pass (cutoff=0.0050): cat 3740, mi = 0.001, pvalue = 0.012, GO_POSITIVE_REGULATION_OF_PROTEIN_SUMOYLATION\n",
      "Didn't pass (cutoff=0.0050): cat 233, mi = 0.001, pvalue = 0.034, GO_ARP2_3_COMPLEX_MEDIATED_ACTIN_NUCLEATION\n",
      "Didn't pass (cutoff=0.0050): cat 642, mi = 0.001, pvalue = 0.041, GO_CELL_COMMUNICATION_INVOLVED_IN_CARDIAC_CONDUCTION\n",
      "Didn't pass (cutoff=0.0050): cat 2941, mi = 0.001, pvalue = 0.045, GO_NUCLEOBASE_CONTAINING_SMALL_MOLECULE_INTERCONVERSION\n",
      "Didn't pass (cutoff=0.0050): cat 4116, mi = 0.001, pvalue = 0.045, GO_QUINONE_METABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 423, mi = 0.001, pvalue = 0.040, GO_CARBOHYDRATE_KINASE_ACTIVITY\n",
      "Total number of categories evaluated = 721.\n",
      "Number of categories that passed the tests = 38\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM2_4.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM2_4.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 5, 19, 0.0330304404815581\n",
      "best merge is 1, 20, 0.0432840382508597\n",
      "best merge is 1, 5, 0.052863922223697\n",
      "best merge is 6, 12, 0.0797592252790553\n",
      "best merge is 3, 4, 0.0836515235706176\n",
      "best merge is 1, 25, 0.0864661444032986\n",
      "best merge is 11, 37, 0.0890272963954167\n",
      "best merge is 7, 14, 0.102032335348737\n",
      "best merge is 1, 15, 0.108527732736085\n",
      "best merge is 32, 36, 0.123325759474438\n",
      "best merge is 6, 18, 0.131422136550329\n",
      "best merge is 26, 35, 0.136319716709647\n",
      "best merge is 29, 34, 0.158142007136868\n",
      "best merge is 0, 9, 0.160586681371668\n",
      "best merge is 1, 28, 0.17063977117136\n",
      "best merge is 2, 3, 0.196574429204168\n",
      "best merge is 6, 16, 0.22893319037141\n",
      "best merge is 22, 27, 0.24041011802741\n",
      "best merge is 7, 23, 0.277285573624936\n",
      "best merge is 0, 2, 0.277501258861754\n",
      "best merge is 21, 24, 0.282827770684758\n",
      "best merge is 1, 17, 0.286904066462774\n",
      "best merge is 11, 29, 0.30426229277559\n",
      "best merge is 8, 32, 0.37651002591531\n",
      "best merge is 7, 10, 0.383917616435055\n",
      "best merge is 6, 11, 0.457866052828151\n",
      "best merge is 0, 26, 0.464768180915361\n",
      "best merge is 1, 22, 0.475366966363238\n",
      "best merge is 6, 30, 0.517032621875886\n",
      "best merge is 7, 13, 0.538863096679933\n",
      "best merge is 31, 33, 0.581192842937694\n",
      "best merge is 0, 7, 0.598042840546682\n",
      "best merge is 21, 31, 0.815642493535034\n",
      "best merge is 1, 6, 0.846836141374747\n",
      "order classes\n",
      "12 - 1.40660273454327\n",
      "10 - 0.9161588781581\n",
      "9 - 1.45491933016919\n",
      "3 - 2.46675117256238\n",
      "4 - 4.90032632296808\n",
      "1 20 5 19 25 15 28 17 22 27\n",
      "0 9 2 3 4 26 35 7 14 23 10 13\n",
      "6 12 18 16 11 37 29 34 30\n",
      "8 32 36\n",
      "21 24 31 33\n",
      "Done.Start drawing\n",
      "100.30\t0.00\n",
      "GO_BICARBONATE_TRANSPORT\n",
      "GO_PHENOL_CONTAINING_COMPOUND_METABOLIC_PROCESS\n",
      "GO_CARGO_RECEPTOR_ACTIVITY\n",
      "GO_REGULATION_OF_STEM_CELL_DIFFERENTIATION\n",
      "GO_RESPONSE_TO_PROTOZOAN\n",
      "GO_NEGATIVE_REGULATION_OF_SMOOTH_MUSCLE_CELL_PROLIFERATION\n",
      "GO_T_CELL_RECEPTOR_COMPLEX\n",
      "GO_REGULATION_OF_STEROL_TRANSPORT\n",
      "GO_AMELOGENESIS\n",
      "GO_MEMBRANE_BIOGENESIS\n",
      "GO_NEURONAL_POSTSYNAPTIC_DENSITY\n",
      "GO_PROTEIN_MODIFICATION_BY_SMALL_PROTEIN_REMOVAL\n",
      "GO_CYSTEINE_TYPE_ENDOPEPTIDASE_ACTIVITY_INVOLVED_IN_APOPTOTIC_PROCESS\n",
      "GO_MANNOSIDASE_ACTIVITY\n",
      "GO_PHOSPHATASE_REGULATOR_ACTIVITY\n",
      "GO_ACYL_COA_DEHYDROGENASE_ACTIVITY\n",
      "GO_VACUOLE_ORGANIZATION\n",
      "GO_POSITIVE_REGULATION_OF_VACUOLE_ORGANIZATION\n",
      "GO_TRNA_METABOLIC_PROCESS\n",
      "GO_CLATHRIN_ADAPTOR_COMPLEX\n",
      "GO_RIBOSOMAL_SUBUNIT\n",
      "GO_RIBONUCLEOPROTEIN_COMPLEX_BINDING\n",
      "GO_PROTEASOME_COMPLEX\n",
      "GO_DNA_TEMPLATED_TRANSCRIPTION_ELONGATION\n",
      "GO_MITOCHONDRIAL_PROTEIN_COMPLEX\n",
      "GO_SPLICEOSOMAL_COMPLEX\n",
      "GO_REGULATION_OF_ESTABLISHMENT_OF_PROTEIN_LOCALIZATION_TO_CHROMOSOME\n",
      "GO_ENDOPLASMIC_RETICULUM_CHAPERONE_COMPLEX\n",
      "GO_CONDENSED_CHROMOSOME\n",
      "GO_INSULIN_SECRETION\n",
      "GO_DNA_REPLICATION_INITIATION\n",
      "GO_SINGLE_STRANDED_DNA_DEPENDENT_ATPASE_ACTIVITY\n",
      "GO_NEGATIVE_REGULATION_OF_GENE_EXPRESSION_EPIGENETIC\n",
      "GO_KERATIN_FILAMENT\n",
      "GO_POSITIVE_REGULATION_OF_CYTOKINESIS\n",
      "GO_CENTRIOLE\n",
      "GO_VOLTAGE_GATED_CATION_CHANNEL_ACTIVITY\n",
      "GO_REGULATION_OF_CHEMOTAXIS\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM2_4.csv_PAGE/filt_FC_LVM2_4.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM2_4.csv_PAGE/filt_FC_LVM2_4.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM2_4.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM2_4.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 5, 19, 0.0330304404815581\n",
      "best merge is 1, 20, 0.0432840382508597\n",
      "best merge is 1, 5, 0.052863922223697\n",
      "best merge is 6, 12, 0.0797592252790553\n",
      "best merge is 3, 4, 0.0836515235706176\n",
      "best merge is 1, 25, 0.0864661444032986\n",
      "best merge is 11, 37, 0.0890272963954167\n",
      "best merge is 7, 14, 0.102032335348737\n",
      "best merge is 1, 15, 0.108527732736085\n",
      "best merge is 32, 36, 0.123325759474438\n",
      "best merge is 6, 18, 0.131422136550329\n",
      "best merge is 26, 35, 0.136319716709647\n",
      "best merge is 29, 34, 0.158142007136868\n",
      "best merge is 0, 9, 0.160586681371668\n",
      "best merge is 1, 28, 0.17063977117136\n",
      "best merge is 2, 3, 0.196574429204168\n",
      "best merge is 6, 16, 0.22893319037141\n",
      "best merge is 22, 27, 0.24041011802741\n",
      "best merge is 7, 23, 0.277285573624936\n",
      "best merge is 0, 2, 0.277501258861754\n",
      "best merge is 21, 24, 0.282827770684758\n",
      "best merge is 1, 17, 0.286904066462774\n",
      "best merge is 11, 29, 0.30426229277559\n",
      "best merge is 8, 32, 0.37651002591531\n",
      "best merge is 7, 10, 0.383917616435055\n",
      "best merge is 6, 11, 0.457866052828151\n",
      "best merge is 0, 26, 0.464768180915361\n",
      "best merge is 1, 22, 0.475366966363238\n",
      "best merge is 6, 30, 0.517032621875886\n",
      "best merge is 7, 13, 0.538863096679933\n",
      "best merge is 31, 33, 0.581192842937694\n",
      "best merge is 0, 7, 0.598042840546682\n",
      "best merge is 21, 31, 0.815642493535034\n",
      "best merge is 1, 6, 0.846836141374747\n",
      "order classes\n",
      "12 - 1.40660273454327\n",
      "10 - 0.9161588781581\n",
      "9 - 1.45491933016919\n",
      "3 - 2.46675117256238\n",
      "4 - 4.90032632296808\n",
      "1 20 5 19 25 15 28 17 22 27\n",
      "0 9 2 3 4 26 35 7 14 23 10 13\n",
      "6 12 18 16 11 37 29 34 30\n",
      "8 32 36\n",
      "21 24 31 33\n",
      "Done.-19.8911266201396\t42\t0\n",
      "-19.7403788634098\t42\t0.217746759720837\n",
      "-19.6712861415753\t42\t0.301495513459621\n",
      "-19.621036889332\t42\t0.355932203389831\n",
      "-19.5728813559322\t42\t0.401994017946162\n",
      "-19.5163509471585\t42\t0.452243270189432\n",
      "-19.4451645064806\t42\t0.515054835493519\n",
      "-19.3321036889332\t42\t0.594616151545364\n",
      "1.37058823529411\t42\t0.741176470588235\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM2_4.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "done\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_P_LVM1_1.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c5.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c5/human_ensembl_msigdb_c5.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_P_LVM1_1.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 5917\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t5917\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 5917\n",
      "cat 957, mi = 0.01592143, GO_CYTOSOLIC_RIBOSOME\n",
      "cat 5245, mi = 0.01468833, GO_RIBOSOMAL_SUBUNIT\n",
      "cat 5718, mi = 0.01418341, GO_TRANSLATIONAL_INITIATION\n",
      "cat 1292, mi = 0.01401355, GO_ESTABLISHMENT_OF_PROTEIN_LOCALIZATION_TO_ENDOPLASMIC_RETICULUM\n",
      "cat 3995, mi = 0.01333615, GO_PROTEIN_LOCALIZATION_TO_ENDOPLASMIC_RETICULUM\n",
      "cat 2928, mi = 0.01248244, GO_NUCLEAR_TRANSCRIBED_MRNA_CATABOLIC_PROCESS_NONSENSE_MEDIATED_DECAY\n",
      "cat 2330, mi = 0.00997471, GO_MULTI_ORGANISM_METABOLIC_PROCESS\n",
      "cat 4041, mi = 0.00969413, GO_PROTEIN_TARGETING_TO_MEMBRANE\n",
      "cat 1889, mi = 0.00967286, GO_LARGE_RIBOSOMAL_SUBUNIT\n",
      "cat 954, mi = 0.00967188, GO_CYTOSOLIC_LARGE_RIBOSOMAL_SUBUNIT\n",
      "cat 3085, mi = 0.00677040, GO_OXIDATIVE_PHOSPHORYLATION\n",
      "cat 2219, mi = 0.00674028, GO_MITOCHONDRIAL_PROTEIN_COMPLEX\n",
      "cat 1742, mi = 0.00633199, GO_INNER_MITOCHONDRIAL_MEMBRANE_PROTEIN_COMPLEX\n",
      "cat 2217, mi = 0.00616325, GO_MITOCHONDRIAL_MEMBRANE_PART\n",
      "cat 958, mi = 0.00591212, GO_CYTOSOLIC_SMALL_RIBOSOMAL_SUBUNIT\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 957, mi = 0.016, pvalue = 0.000, GO_CYTOSOLIC_RIBOSOME\n",
      "Passed: cat 3085, mi = 0.007, pvalue = 0.000, GO_OXIDATIVE_PHOSPHORYLATION\n",
      "Passed: cat 5499, mi = 0.004, pvalue = 0.000, GO_SPLICEOSOMAL_COMPLEX\n",
      "Passed: cat 5719, mi = 0.004, pvalue = 0.000, GO_TRANSLATIONAL_TERMINATION\n",
      "Passed: cat 5813, mi = 0.003, pvalue = 0.000, GO_UNFOLDED_PROTEIN_BINDING\n",
      "Passed: cat 1119, mi = 0.003, pvalue = 0.000, GO_DOUBLE_STRANDED_RNA_BINDING\n",
      "Passed: cat 2226, mi = 0.003, pvalue = 0.000, GO_MITOCHONDRIAL_TRANSPORT\n",
      "Passed: cat 164, mi = 0.002, pvalue = 0.000, GO_ANAPHASE_PROMOTING_COMPLEX_DEPENDENT_CATABOLIC_PROCESS\n",
      "Passed: cat 2967, mi = 0.002, pvalue = 0.000, GO_NUCLEOSOME_BINDING\n",
      "Passed: cat 783, mi = 0.002, pvalue = 0.000, GO_CHROMOSOME_TELOMERIC_REGION\n",
      "Passed: cat 1105, mi = 0.002, pvalue = 0.000, GO_DNA_TEMPLATED_TRANSCRIPTION_ELONGATION\n",
      "Passed: cat 345, mi = 0.002, pvalue = 0.001, GO_BLOOD_MICROPARTICLE\n",
      "Passed: cat 5466, mi = 0.002, pvalue = 0.001, GO_SOMITE_DEVELOPMENT\n",
      "Passed: cat 1698, mi = 0.002, pvalue = 0.001, GO_HYDROLASE_ACTIVITY_HYDROLYZING_N_GLYCOSYL_COMPOUNDS\n",
      "Passed: cat 228, mi = 0.002, pvalue = 0.001, GO_ARF_GUANYL_NUCLEOTIDE_EXCHANGE_FACTOR_ACTIVITY\n",
      "Passed: cat 1925, mi = 0.002, pvalue = 0.002, GO_LIGAND_GATED_CHANNEL_ACTIVITY\n",
      "Passed: cat 5222, mi = 0.002, pvalue = 0.002, GO_RHO_GUANYL_NUCLEOTIDE_EXCHANGE_FACTOR_ACTIVITY\n",
      "Passed: cat 5112, mi = 0.001, pvalue = 0.002, GO_RESPONSE_TO_INTERLEUKIN_6\n",
      "Passed: cat 3943, mi = 0.001, pvalue = 0.004, GO_PROTEIN_DISULFIDE_OXIDOREDUCTASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 4876, mi = 0.001, pvalue = 0.006, GO_REGULATION_OF_SMOOTH_MUSCLE_CELL_MIGRATION\n",
      "Didn't pass (cutoff=0.0050): cat 286, mi = 0.001, pvalue = 0.006, GO_AXONEME_ASSEMBLY\n",
      "Didn't pass (cutoff=0.0050): cat 2889, mi = 0.001, pvalue = 0.005, GO_NON_MEMBRANE_SPANNING_PROTEIN_TYROSINE_KINASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 5042, mi = 0.001, pvalue = 0.006, GO_RESPONSE_TO_AMMONIUM_ION\n",
      "Didn't pass (cutoff=0.0050): cat 1927, mi = 0.001, pvalue = 0.005, GO_LIGASE_ACTIVITY_FORMING_CARBON_NITROGEN_BONDS\n",
      "Didn't pass (cutoff=0.0050): cat 5910, mi = 0.001, pvalue = 0.009, GO_ZINC_II_ION_TRANSPORT\n",
      "Didn't pass (cutoff=0.0050): cat 2387, mi = 0.001, pvalue = 0.008, GO_NAD_METABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 2071, mi = 0.001, pvalue = 0.009, GO_MATERNAL_PLACENTA_DEVELOPMENT\n",
      "Didn't pass (cutoff=0.0050): cat 397, mi = 0.001, pvalue = 0.006, GO_CALCIUM_ION_REGULATED_EXOCYTOSIS\n",
      "Didn't pass (cutoff=0.0050): cat 5476, mi = 0.001, pvalue = 0.008, GO_SPERM_FLAGELLUM\n",
      "Didn't pass (cutoff=0.0050): cat 3882, mi = 0.001, pvalue = 0.008, GO_PRESYNAPTIC_ACTIVE_ZONE\n",
      "Didn't pass (cutoff=0.0050): cat 46, mi = 0.001, pvalue = 0.010, GO_ACTIN_MONOMER_BINDING\n",
      "Passed: cat 4727, mi = 0.001, pvalue = 0.003, GO_REGULATION_OF_OXIDATIVE_STRESS_INDUCED_NEURON_DEATH\n",
      "Didn't pass (cutoff=0.0050): cat 1621, mi = 0.001, pvalue = 0.010, GO_HEXOSE_CATABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 3171, mi = 0.001, pvalue = 0.011, GO_PEPTIDYL_TYROSINE_AUTOPHOSPHORYLATION\n",
      "Didn't pass (cutoff=0.0050): cat 3230, mi = 0.001, pvalue = 0.009, GO_PHOSPHATIDYLINOSITOL_3_PHOSPHATE_BIOSYNTHETIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 5562, mi = 0.001, pvalue = 0.005, GO_SULFUR_COMPOUND_TRANSMEMBRANE_TRANSPORTER_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 3954, mi = 0.001, pvalue = 0.006, GO_PROTEIN_HETEROTRIMERIZATION\n",
      "Didn't pass (cutoff=0.0050): cat 3384, mi = 0.001, pvalue = 0.011, GO_POSITIVE_REGULATION_OF_B_CELL_ACTIVATION\n",
      "Didn't pass (cutoff=0.0050): cat 3654, mi = 0.001, pvalue = 0.012, GO_POSITIVE_REGULATION_OF_MYELOID_LEUKOCYTE_DIFFERENTIATION\n",
      "Didn't pass (cutoff=0.0050): cat 1934, mi = 0.001, pvalue = 0.010, GO_LIPASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 5356, mi = 0.001, pvalue = 0.013, GO_SERINE_FAMILY_AMINO_ACID_METABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 3172, mi = 0.001, pvalue = 0.009, GO_PEPTIDYL_TYROSINE_MODIFICATION\n",
      "Didn't pass (cutoff=0.0050): cat 2948, mi = 0.001, pvalue = 0.010, GO_NUCLEOPHAGY\n",
      "Didn't pass (cutoff=0.0050): cat 4072, mi = 0.001, pvalue = 0.012, GO_PTERIDINE_CONTAINING_COMPOUND_BIOSYNTHETIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 1722, mi = 0.001, pvalue = 0.011, GO_IMP_METABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 2379, mi = 0.001, pvalue = 0.018, GO_NADH_METABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 2277, mi = 0.001, pvalue = 0.014, GO_MONOSACCHARIDE_CATABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 1876, mi = 0.001, pvalue = 0.008, GO_KINETOCHORE_ORGANIZATION\n",
      "Didn't pass (cutoff=0.0050): cat 493, mi = 0.001, pvalue = 0.016, GO_CELLULAR_AMINO_ACID_BIOSYNTHETIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 3266, mi = 0.001, pvalue = 0.018, GO_PHOSPHORIC_DIESTER_HYDROLASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 4050, mi = 0.001, pvalue = 0.012, GO_PROTEIN_TYROSINE_KINASE_ACTIVITY\n",
      "Passed: cat 5311, mi = 0.001, pvalue = 0.004, GO_S100_PROTEIN_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 5264, mi = 0.001, pvalue = 0.018, GO_RNA_METHYLATION\n",
      "Didn't pass (cutoff=0.0050): cat 521, mi = 0.001, pvalue = 0.019, GO_CELLULAR_MODIFIED_AMINO_ACID_BIOSYNTHETIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 2178, mi = 0.001, pvalue = 0.020, GO_MICROTUBULE_BUNDLE_FORMATION\n",
      "Didn't pass (cutoff=0.0050): cat 5290, mi = 0.001, pvalue = 0.016, GO_RNA_SPLICING_VIA_ENDONUCLEOLYTIC_CLEAVAGE_AND_LIGATION\n",
      "Didn't pass (cutoff=0.0050): cat 4987, mi = 0.001, pvalue = 0.022, GO_REGULATION_OF_VASCULAR_PERMEABILITY\n",
      "Didn't pass (cutoff=0.0050): cat 1817, mi = 0.001, pvalue = 0.021, GO_INTRINSIC_COMPONENT_OF_GOLGI_MEMBRANE\n",
      "Didn't pass (cutoff=0.0050): cat 537, mi = 0.001, pvalue = 0.023, GO_CELLULAR_RESPONSE_TO_AMINO_ACID_STIMULUS\n",
      "Didn't pass (cutoff=0.0050): cat 171, mi = 0.001, pvalue = 0.020, GO_ANCHORED_COMPONENT_OF_MEMBRANE\n",
      "Didn't pass (cutoff=0.0050): cat 233, mi = 0.001, pvalue = 0.021, GO_ARP2_3_COMPLEX_MEDIATED_ACTIN_NUCLEATION\n",
      "Didn't pass (cutoff=0.0050): cat 5265, mi = 0.001, pvalue = 0.023, GO_RNA_METHYLTRANSFERASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 1562, mi = 0.001, pvalue = 0.021, GO_GTP_DEPENDENT_PROTEIN_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 3387, mi = 0.001, pvalue = 0.025, GO_POSITIVE_REGULATION_OF_B_CELL_PROLIFERATION\n",
      "Didn't pass (cutoff=0.0050): cat 639, mi = 0.001, pvalue = 0.009, GO_CELL_CELL_SIGNALING_INVOLVED_IN_CARDIAC_CONDUCTION\n",
      "Didn't pass (cutoff=0.0050): cat 5563, mi = 0.001, pvalue = 0.026, GO_SULFUR_COMPOUND_TRANSPORT\n",
      "Didn't pass (cutoff=0.0050): cat 4363, mi = 0.001, pvalue = 0.028, GO_REGULATION_OF_DEFENSE_RESPONSE_TO_VIRUS_BY_VIRUS\n",
      "Didn't pass (cutoff=0.0050): cat 273, mi = 0.001, pvalue = 0.019, GO_AUTOPHAGOSOME\n",
      "Didn't pass (cutoff=0.0050): cat 3264, mi = 0.001, pvalue = 0.028, GO_PHOSPHOLIPID_TRANSPORTER_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 2107, mi = 0.001, pvalue = 0.021, GO_MEMBRANE_LIPID_CATABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 4030, mi = 0.001, pvalue = 0.025, GO_PROTEIN_SELF_ASSOCIATION\n",
      "Didn't pass (cutoff=0.0050): cat 121, mi = 0.001, pvalue = 0.026, GO_ALPHA_AMINO_ACID_BIOSYNTHETIC_PROCESS\n",
      "Total number of categories evaluated = 374.\n",
      "Number of categories that passed the tests = 21\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_P_LVM1_1.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_P_LVM1_1.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 0, 19, 0.0225457760632943\n",
      "best merge is 3, 6, 0.0237144723817216\n",
      "best merge is 4, 8, 0.0342590284092363\n",
      "best merge is 1, 20, 0.0437723753613956\n",
      "best merge is 2, 3, 0.0470177499704824\n",
      "best merge is 1, 2, 0.060717051037167\n",
      "best merge is 7, 18, 0.0868186730621429\n",
      "best merge is 0, 1, 0.101787345762315\n",
      "best merge is 4, 5, 0.172870964743101\n",
      "best merge is 9, 10, 0.218575579049074\n",
      "best merge is 4, 7, 0.223163169189095\n",
      "best merge is 12, 14, 0.24730830577383\n",
      "best merge is 4, 11, 0.325367499939421\n",
      "best merge is 0, 9, 0.361788026671163\n",
      "best merge is 15, 16, 0.453323972604056\n",
      "best merge is 13, 17, 0.554468948832852\n",
      "best merge is 0, 4, 0.612657441163064\n",
      "order classes\n",
      "9 - 0.394641669796479\n",
      "6 - 0.667457028347911\n",
      "2 - 1.29836860531261\n",
      "2 - 2.91096198137247\n",
      "2 - 2.3799508561713\n",
      "0 19 1 20 2 3 6 9 10\n",
      "4 8 5 7 18 11\n",
      "12 14\n",
      "13 17\n",
      "15 16\n",
      "Done.Start drawing\n",
      "280.55\t0.00\n",
      "GO_CHROMOSOME_TELOMERIC_REGION\n",
      "GO_CYTOSOLIC_RIBOSOME\n",
      "GO_REGULATION_OF_OXIDATIVE_STRESS_INDUCED_NEURON_DEATH\n",
      "GO_OXIDATIVE_PHOSPHORYLATION\n",
      "GO_S100_PROTEIN_BINDING\n",
      "GO_SPLICEOSOMAL_COMPLEX\n",
      "GO_TRANSLATIONAL_TERMINATION\n",
      "GO_MITOCHONDRIAL_TRANSPORT\n",
      "GO_DNA_TEMPLATED_TRANSCRIPTION_ELONGATION\n",
      "GO_UNFOLDED_PROTEIN_BINDING\n",
      "GO_NUCLEOSOME_BINDING\n",
      "GO_DOUBLE_STRANDED_RNA_BINDING\n",
      "GO_ANAPHASE_PROMOTING_COMPLEX_DEPENDENT_CATABOLIC_PROCESS\n",
      "GO_PROTEIN_DISULFIDE_OXIDOREDUCTASE_ACTIVITY\n",
      "GO_BLOOD_MICROPARTICLE\n",
      "GO_LIGAND_GATED_CHANNEL_ACTIVITY\n",
      "GO_SOMITE_DEVELOPMENT\n",
      "GO_ARF_GUANYL_NUCLEOTIDE_EXCHANGE_FACTOR_ACTIVITY\n",
      "GO_RHO_GUANYL_NUCLEOTIDE_EXCHANGE_FACTOR_ACTIVITY\n",
      "GO_HYDROLASE_ACTIVITY_HYDROLYZING_N_GLYCOSYL_COMPOUNDS\n",
      "GO_RESPONSE_TO_INTERLEUKIN_6\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_P_LVM1_1.csv_PAGE/filt_FC_P_LVM1_1.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_P_LVM1_1.csv_PAGE/filt_FC_P_LVM1_1.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_P_LVM1_1.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_P_LVM1_1.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 0, 19, 0.0225457760632943\n",
      "best merge is 3, 6, 0.0237144723817216\n",
      "best merge is 4, 8, 0.0342590284092363\n",
      "best merge is 1, 20, 0.0437723753613956\n",
      "best merge is 2, 3, 0.0470177499704824\n",
      "best merge is 1, 2, 0.060717051037167\n",
      "best merge is 7, 18, 0.0868186730621429\n",
      "best merge is 0, 1, 0.101787345762315\n",
      "best merge is 4, 5, 0.172870964743101\n",
      "best merge is 9, 10, 0.218575579049074\n",
      "best merge is 4, 7, 0.223163169189095\n",
      "best merge is 12, 14, 0.24730830577383\n",
      "best merge is 4, 11, 0.325367499939421\n",
      "best merge is 0, 9, 0.361788026671163\n",
      "best merge is 15, 16, 0.453323972604056\n",
      "best merge is 13, 17, 0.554468948832852\n",
      "best merge is 0, 4, 0.612657441163064\n",
      "order classes\n",
      "9 - 0.394641669796479\n",
      "6 - 0.667457028347911\n",
      "2 - 1.29836860531261\n",
      "2 - 2.91096198137247\n",
      "2 - 2.3799508561713\n",
      "0 19 1 20 2 3 6 9 10\n",
      "4 8 5 7 18 11\n",
      "12 14\n",
      "13 17\n",
      "15 16\n",
      "Done.-19.9985029406523\t42\t0\n",
      "-19.9962573516307\t42\t0.00299411869541971\n",
      "-19.9940117626092\t42\t0.00449117804312957\n",
      "-19.9917661735876\t42\t0.00748529673854928\n",
      "-19.989520584566\t42\t0.00898235608625913\n",
      "-19.9872749955445\t42\t0.0119764747816788\n",
      "-19.9169132062021\t42\t0.0134735341293887\n",
      "-19.7559793263233\t42\t0.152700053466405\n",
      "1.1676706469435\t42\t0.335341293887008\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_P_LVM1_1.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM2_5.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c5.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c5/human_ensembl_msigdb_c5.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM2_5.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 5917\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t5917\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 5917\n",
      "cat 5718, mi = 0.00659278, GO_TRANSLATIONAL_INITIATION\n",
      "cat 1292, mi = 0.00600124, GO_ESTABLISHMENT_OF_PROTEIN_LOCALIZATION_TO_ENDOPLASMIC_RETICULUM\n",
      "cat 957, mi = 0.00589126, GO_CYTOSOLIC_RIBOSOME\n",
      "cat 3995, mi = 0.00572510, GO_PROTEIN_LOCALIZATION_TO_ENDOPLASMIC_RETICULUM\n",
      "cat 5499, mi = 0.00571895, GO_SPLICEOSOMAL_COMPLEX\n",
      "cat 2928, mi = 0.00569840, GO_NUCLEAR_TRANSCRIBED_MRNA_CATABOLIC_PROCESS_NONSENSE_MEDIATED_DECAY\n",
      "cat 2330, mi = 0.00540292, GO_MULTI_ORGANISM_METABOLIC_PROCESS\n",
      "cat 5245, mi = 0.00498908, GO_RIBOSOMAL_SUBUNIT\n",
      "cat 2224, mi = 0.00459021, GO_MITOCHONDRIAL_TRANSLATION\n",
      "cat 4041, mi = 0.00404383, GO_PROTEIN_TARGETING_TO_MEMBRANE\n",
      "cat 954, mi = 0.00403130, GO_CYTOSOLIC_LARGE_RIBOSOMAL_SUBUNIT\n",
      "cat 5263, mi = 0.00400968, GO_RNA_LOCALIZATION\n",
      "cat 5719, mi = 0.00396888, GO_TRANSLATIONAL_TERMINATION\n",
      "cat 4852, mi = 0.00396856, GO_REGULATION_OF_RNA_STABILITY\n",
      "cat 5755, mi = 0.00392135, GO_TRNA_METABOLIC_PROCESS\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 5718, mi = 0.007, pvalue = 0.000, GO_TRANSLATIONAL_INITIATION\n",
      "Passed: cat 5499, mi = 0.006, pvalue = 0.000, GO_SPLICEOSOMAL_COMPLEX\n",
      "Passed: cat 2224, mi = 0.005, pvalue = 0.000, GO_MITOCHONDRIAL_TRANSLATION\n",
      "Passed: cat 4852, mi = 0.004, pvalue = 0.000, GO_REGULATION_OF_RNA_STABILITY\n",
      "Passed: cat 3085, mi = 0.003, pvalue = 0.000, GO_OXIDATIVE_PHOSPHORYLATION\n",
      "Passed: cat 1670, mi = 0.003, pvalue = 0.000, GO_HORMONE_ACTIVITY\n",
      "Passed: cat 3865, mi = 0.003, pvalue = 0.000, GO_POTASSIUM_CHANNEL_ACTIVITY\n",
      "Passed: cat 5402, mi = 0.003, pvalue = 0.000, GO_SISTER_CHROMATID_SEGREGATION\n",
      "Passed: cat 5759, mi = 0.003, pvalue = 0.000, GO_TRNA_PROCESSING\n",
      "Passed: cat 2244, mi = 0.002, pvalue = 0.000, GO_MITOTIC_RECOMBINATION\n",
      "Passed: cat 5429, mi = 0.002, pvalue = 0.000, GO_SMOOTH_MUSCLE_CELL_DIFFERENTIATION\n",
      "Passed: cat 5804, mi = 0.002, pvalue = 0.000, GO_UBIQUITIN_LIKE_PROTEIN_LIGASE_ACTIVITY\n",
      "Passed: cat 5678, mi = 0.002, pvalue = 0.000, GO_TRANSCRIPTION_ELONGATION_FROM_RNA_POLYMERASE_II_PROMOTER\n",
      "Passed: cat 4910, mi = 0.002, pvalue = 0.000, GO_REGULATION_OF_SYNAPTIC_TRANSMISSION_GLUTAMATERGIC\n",
      "Passed: cat 5729, mi = 0.002, pvalue = 0.000, GO_TRANSMEMBRANE_RECEPTOR_PROTEIN_KINASE_ACTIVITY\n",
      "Passed: cat 5813, mi = 0.002, pvalue = 0.000, GO_UNFOLDED_PROTEIN_BINDING\n",
      "Passed: cat 171, mi = 0.002, pvalue = 0.000, GO_ANCHORED_COMPONENT_OF_MEMBRANE\n",
      "Passed: cat 5575, mi = 0.002, pvalue = 0.001, GO_SYNAPSE_ASSEMBLY\n",
      "Passed: cat 2379, mi = 0.002, pvalue = 0.001, GO_NADH_METABOLIC_PROCESS\n",
      "Passed: cat 2226, mi = 0.002, pvalue = 0.001, GO_MITOCHONDRIAL_TRANSPORT\n",
      "Passed: cat 2321, mi = 0.002, pvalue = 0.001, GO_MULTICELLULAR_ORGANISM_METABOLIC_PROCESS\n",
      "Passed: cat 322, mi = 0.002, pvalue = 0.001, GO_BHLH_TRANSCRIPTION_FACTOR_BINDING\n",
      "Passed: cat 3351, mi = 0.002, pvalue = 0.000, GO_POSITIVE_REGULATION_OF_ACTIN_FILAMENT_DEPOLYMERIZATION\n",
      "Passed: cat 1862, mi = 0.002, pvalue = 0.001, GO_KERATIN_FILAMENT\n",
      "Passed: cat 3114, mi = 0.002, pvalue = 0.000, GO_OXYGEN_BINDING\n",
      "Passed: cat 2137, mi = 0.002, pvalue = 0.001, GO_METALLO_SULFUR_CLUSTER_ASSEMBLY\n",
      "Passed: cat 5424, mi = 0.002, pvalue = 0.003, GO_SMALL_SUBUNIT_PROCESSOME\n",
      "Passed: cat 390, mi = 0.002, pvalue = 0.002, GO_CALCIUM_DEPENDENT_PHOSPHOLIPID_BINDING\n",
      "Passed: cat 1653, mi = 0.001, pvalue = 0.003, GO_HISTONE_METHYLTRANSFERASE_ACTIVITY_H3_K4_SPECIFIC_\n",
      "Passed: cat 926, mi = 0.001, pvalue = 0.004, GO_CYTOKINE_RECEPTOR_ACTIVITY\n",
      "Passed: cat 3184, mi = 0.001, pvalue = 0.004, GO_PEROXISOME_ORGANIZATION\n",
      "Passed: cat 47, mi = 0.001, pvalue = 0.003, GO_ACTIN_MYOSIN_FILAMENT_SLIDING\n",
      "Passed: cat 5679, mi = 0.001, pvalue = 0.003, GO_TRANSCRIPTION_EXPORT_COMPLEX\n",
      "Passed: cat 4812, mi = 0.001, pvalue = 0.005, GO_REGULATION_OF_RAS_PROTEIN_SIGNAL_TRANSDUCTION\n",
      "Didn't pass (cutoff=0.0050): cat 5715, mi = 0.001, pvalue = 0.007, GO_TRANSITION_METAL_ION_TRANSMEMBRANE_TRANSPORTER_ACTIVITY\n",
      "Passed: cat 5811, mi = 0.001, pvalue = 0.005, GO_UDP_GLYCOSYLTRANSFERASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 93, mi = 0.001, pvalue = 0.007, GO_ADHERENS_JUNCTION_ORGANIZATION\n",
      "Didn't pass (cutoff=0.0050): cat 3372, mi = 0.001, pvalue = 0.009, GO_POSITIVE_REGULATION_OF_AUTOPHAGY\n",
      "Didn't pass (cutoff=0.0050): cat 3242, mi = 0.001, pvalue = 0.005, GO_PHOSPHATIDYLINOSITOL_PHOSPHATE_5_PHOSPHATASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 1453, mi = 0.001, pvalue = 0.015, GO_GDP_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 3335, mi = 0.001, pvalue = 0.014, GO_POLYUBIQUITIN_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 4470, mi = 0.001, pvalue = 0.006, GO_REGULATION_OF_GOLGI_ORGANIZATION\n",
      "Didn't pass (cutoff=0.0050): cat 791, mi = 0.001, pvalue = 0.015, GO_CILIARY_TIP\n",
      "Didn't pass (cutoff=0.0050): cat 464, mi = 0.001, pvalue = 0.018, GO_CARGO_RECEPTOR_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 1715, mi = 0.001, pvalue = 0.018, GO_IMMUNOGLOBULIN_PRODUCTION\n",
      "Didn't pass (cutoff=0.0050): cat 5786, mi = 0.001, pvalue = 0.008, GO_T_CELL_RECEPTOR_COMPLEX\n",
      "Didn't pass (cutoff=0.0050): cat 1832, mi = 0.001, pvalue = 0.016, GO_ION_ANTIPORTER_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 5373, mi = 0.001, pvalue = 0.011, GO_SHORT_CHAIN_FATTY_ACID_METABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 4841, mi = 0.001, pvalue = 0.015, GO_REGULATION_OF_RESPONSE_TO_OXIDATIVE_STRESS\n",
      "Didn't pass (cutoff=0.0050): cat 5425, mi = 0.001, pvalue = 0.012, GO_SMN_COMPLEX\n",
      "Didn't pass (cutoff=0.0050): cat 997, mi = 0.001, pvalue = 0.020, GO_DEOXYRIBOSE_PHOSPHATE_CATABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 4668, mi = 0.001, pvalue = 0.014, GO_REGULATION_OF_NATURAL_KILLER_CELL_ACTIVATION\n",
      "Didn't pass (cutoff=0.0050): cat 1322, mi = 0.001, pvalue = 0.015, GO_EXOCYST\n",
      "Passed: cat 3922, mi = 0.001, pvalue = 0.003, GO_PROTEIN_BINDING_INVOLVED_IN_CELL_CELL_ADHESION\n",
      "Didn't pass (cutoff=0.0050): cat 2131, mi = 0.001, pvalue = 0.009, GO_METALLOAMINOPEPTIDASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 504, mi = 0.001, pvalue = 0.023, GO_CELLULAR_COMPONENT_DISASSEMBLY_INVOLVED_IN_EXECUTION_PHASE_OF_APOPTOSIS\n",
      "Didn't pass (cutoff=0.0050): cat 4922, mi = 0.001, pvalue = 0.024, GO_REGULATION_OF_TELOMERE_CAPPING\n",
      "Didn't pass (cutoff=0.0050): cat 5279, mi = 0.001, pvalue = 0.022, GO_RNA_POLYMERASE_II_DISTAL_ENHANCER_SEQUENCE_SPECIFIC_DNA_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 1543, mi = 0.001, pvalue = 0.027, GO_GOLGI_TO_PLASMA_MEMBRANE_PROTEIN_TRANSPORT\n",
      "Didn't pass (cutoff=0.0050): cat 2254, mi = 0.001, pvalue = 0.011, GO_MODIFIED_AMINO_ACID_TRANSMEMBRANE_TRANSPORTER_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 1447, mi = 0.001, pvalue = 0.013, GO_GAP_JUNCTION\n",
      "Didn't pass (cutoff=0.0050): cat 5874, mi = 0.001, pvalue = 0.024, GO_VITAMIN_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 2952, mi = 0.001, pvalue = 0.016, GO_NUCLEOSIDE_DIPHOSPHATASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 4106, mi = 0.001, pvalue = 0.022, GO_PYRIMIDINE_NUCLEOTIDE_CATABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 4600, mi = 0.001, pvalue = 0.020, GO_REGULATION_OF_MACROPHAGE_DIFFERENTIATION\n",
      "Didn't pass (cutoff=0.0050): cat 2170, mi = 0.001, pvalue = 0.018, GO_MICROFILAMENT_MOTOR_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 1533, mi = 0.001, pvalue = 0.032, GO_GOLGI_ASSOCIATED_VESICLE_MEMBRANE\n",
      "Didn't pass (cutoff=0.0050): cat 1293, mi = 0.001, pvalue = 0.030, GO_ESTABLISHMENT_OF_PROTEIN_LOCALIZATION_TO_GOLGI\n",
      "Didn't pass (cutoff=0.0050): cat 1565, mi = 0.001, pvalue = 0.023, GO_GUANOSINE_CONTAINING_COMPOUND_BIOSYNTHETIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 1808, mi = 0.001, pvalue = 0.035, GO_INTRA_GOLGI_VESICLE_MEDIATED_TRANSPORT\n",
      "Didn't pass (cutoff=0.0050): cat 5571, mi = 0.001, pvalue = 0.025, GO_SWI_SNF_SUPERFAMILY_TYPE_COMPLEX\n",
      "Didn't pass (cutoff=0.0050): cat 4009, mi = 0.001, pvalue = 0.028, GO_PROTEIN_MODIFICATION_BY_SMALL_PROTEIN_REMOVAL\n",
      "Didn't pass (cutoff=0.0050): cat 3774, mi = 0.001, pvalue = 0.043, GO_POSITIVE_REGULATION_OF_SMOOTHENED_SIGNALING_PATHWAY\n",
      "Didn't pass (cutoff=0.0050): cat 2118, mi = 0.001, pvalue = 0.020, GO_MEMBRANE_TUBULATION\n",
      "Total number of categories evaluated = 630.\n",
      "Number of categories that passed the tests = 36\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM2_5.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM2_5.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 5, 6, 0.012448607755626\n",
      "best merge is 22, 24, 0.0452329608662904\n",
      "best merge is 11, 19, 0.0486189688207865\n",
      "best merge is 20, 31, 0.0680518286453722\n",
      "best merge is 1, 2, 0.0741427179978175\n",
      "best merge is 22, 27, 0.0791276616916116\n",
      "best merge is 8, 12, 0.0874162111999782\n",
      "best merge is 5, 35, 0.0888119533341077\n",
      "best merge is 10, 21, 0.0980555788087734\n",
      "best merge is 7, 18, 0.107200191987077\n",
      "best merge is 5, 22, 0.113380304309536\n",
      "best merge is 4, 11, 0.123727071249755\n",
      "best merge is 1, 8, 0.152677752867997\n",
      "best merge is 1, 3, 0.171383527028196\n",
      "best merge is 5, 17, 0.175911634786782\n",
      "best merge is 5, 16, 0.219136273809061\n",
      "best merge is 28, 30, 0.234749297542721\n",
      "best merge is 4, 15, 0.261264087365044\n",
      "best merge is 1, 4, 0.270699868086255\n",
      "best merge is 13, 33, 0.275159911584538\n",
      "best merge is 10, 20, 0.286443447818121\n",
      "best merge is 5, 13, 0.294485256682393\n",
      "best merge is 7, 26, 0.355041593393432\n",
      "best merge is 10, 29, 0.365216221635703\n",
      "best merge is 5, 14, 0.389543092884493\n",
      "best merge is 7, 32, 0.426484215423274\n",
      "best merge is 9, 28, 0.484770085012117\n",
      "best merge is 5, 23, 0.50324028925622\n",
      "best merge is 7, 25, 0.506383902928586\n",
      "best merge is 1, 7, 0.522692090033773\n",
      "best merge is 5, 34, 0.737837741911469\n",
      "best merge is 0, 9, 0.768108108556418\n",
      "order classes\n",
      "14 - 0.933981755830346\n",
      "13 - 0.893870276393351\n",
      "3 - 3.21143858017854\n",
      "5 - 1.5971017451152\n",
      "5 6 35 22 24 27 17 16 13 33 14 23 34\n",
      "1 2 8 12 3 4 11 19 15 7 18 26 32 25\n",
      "0\n",
      "9 28 30\n",
      "10 21 20 31 29\n",
      "Done.Start drawing\n",
      "66.19\t0.00\n",
      "GO_NADH_METABOLIC_PROCESS\n",
      "GO_SMALL_SUBUNIT_PROCESSOME\n",
      "GO_SMOOTH_MUSCLE_CELL_DIFFERENTIATION\n",
      "GO_BHLH_TRANSCRIPTION_FACTOR_BINDING\n",
      "GO_MULTICELLULAR_ORGANISM_METABOLIC_PROCESS\n",
      "GO_ACTIN_MYOSIN_FILAMENT_SLIDING\n",
      "GO_CYTOKINE_RECEPTOR_ACTIVITY\n",
      "GO_SISTER_CHROMATID_SEGREGATION\n",
      "GO_TRANSCRIPTION_EXPORT_COMPLEX\n",
      "GO_TRNA_PROCESSING\n",
      "GO_REGULATION_OF_RNA_STABILITY\n",
      "GO_MITOCHONDRIAL_TRANSPORT\n",
      "GO_METALLO_SULFUR_CLUSTER_ASSEMBLY\n",
      "GO_SPLICEOSOMAL_COMPLEX\n",
      "GO_MITOCHONDRIAL_TRANSLATION\n",
      "GO_TRANSCRIPTION_ELONGATION_FROM_RNA_POLYMERASE_II_PROMOTER\n",
      "GO_UBIQUITIN_LIKE_PROTEIN_LIGASE_ACTIVITY\n",
      "GO_UNFOLDED_PROTEIN_BINDING\n",
      "GO_HISTONE_METHYLTRANSFERASE_ACTIVITY_H3_K4_SPECIFIC_\n",
      "GO_OXIDATIVE_PHOSPHORYLATION\n",
      "GO_MITOTIC_RECOMBINATION\n",
      "GO_PEROXISOME_ORGANIZATION\n",
      "GO_TRANSLATIONAL_INITIATION\n",
      "GO_KERATIN_FILAMENT\n",
      "GO_UDP_GLYCOSYLTRANSFERASE_ACTIVITY\n",
      "GO_HORMONE_ACTIVITY\n",
      "GO_POTASSIUM_CHANNEL_ACTIVITY\n",
      "GO_PROTEIN_BINDING_INVOLVED_IN_CELL_CELL_ADHESION\n",
      "GO_POSITIVE_REGULATION_OF_ACTIN_FILAMENT_DEPOLYMERIZATION\n",
      "GO_OXYGEN_BINDING\n",
      "GO_CALCIUM_DEPENDENT_PHOSPHOLIPID_BINDING\n",
      "GO_SYNAPSE_ASSEMBLY\n",
      "GO_ANCHORED_COMPONENT_OF_MEMBRANE\n",
      "GO_REGULATION_OF_SYNAPTIC_TRANSMISSION_GLUTAMATERGIC\n",
      "GO_REGULATION_OF_RAS_PROTEIN_SIGNAL_TRANSDUCTION\n",
      "GO_TRANSMEMBRANE_RECEPTOR_PROTEIN_KINASE_ACTIVITY\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM2_5.csv_PAGE/filt_FC_LVM2_5.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM2_5.csv_PAGE/filt_FC_LVM2_5.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM2_5.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM2_5.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 5, 6, 0.012448607755626\n",
      "best merge is 22, 24, 0.0452329608662904\n",
      "best merge is 11, 19, 0.0486189688207865\n",
      "best merge is 20, 31, 0.0680518286453722\n",
      "best merge is 1, 2, 0.0741427179978175\n",
      "best merge is 22, 27, 0.0791276616916116\n",
      "best merge is 8, 12, 0.0874162111999782\n",
      "best merge is 5, 35, 0.0888119533341077\n",
      "best merge is 10, 21, 0.0980555788087734\n",
      "best merge is 7, 18, 0.107200191987077\n",
      "best merge is 5, 22, 0.113380304309536\n",
      "best merge is 4, 11, 0.123727071249755\n",
      "best merge is 1, 8, 0.152677752867997\n",
      "best merge is 1, 3, 0.171383527028196\n",
      "best merge is 5, 17, 0.175911634786782\n",
      "best merge is 5, 16, 0.219136273809061\n",
      "best merge is 28, 30, 0.234749297542721\n",
      "best merge is 4, 15, 0.261264087365044\n",
      "best merge is 1, 4, 0.270699868086255\n",
      "best merge is 13, 33, 0.275159911584538\n",
      "best merge is 10, 20, 0.286443447818121\n",
      "best merge is 5, 13, 0.294485256682393\n",
      "best merge is 7, 26, 0.355041593393432\n",
      "best merge is 10, 29, 0.365216221635703\n",
      "best merge is 5, 14, 0.389543092884493\n",
      "best merge is 7, 32, 0.426484215423274\n",
      "best merge is 9, 28, 0.484770085012117\n",
      "best merge is 5, 23, 0.50324028925622\n",
      "best merge is 7, 25, 0.506383902928586\n",
      "best merge is 1, 7, 0.522692090033773\n",
      "best merge is 5, 34, 0.737837741911469\n",
      "best merge is 0, 9, 0.768108108556418\n",
      "order classes\n",
      "14 - 0.933981755830346\n",
      "13 - 0.893870276393351\n",
      "3 - 3.21143858017854\n",
      "5 - 1.5971017451152\n",
      "5 6 35 22 24 27 17 16 13 33 14 23 34\n",
      "1 2 8 12 3 4 11 19 15 7 18 26 32 25\n",
      "0\n",
      "9 28 30\n",
      "10 21 20 31 29\n",
      "Done.-19.8477111346125\t42\t0\n",
      "-19.6224505212268\t42\t0.304577730775042\n",
      "-19.501888502795\t42\t0.450521226771416\n",
      "-19.4130533313189\t42\t0.545701767638616\n",
      "-19.330563529234\t42\t0.628191569723523\n",
      "-19.2480737271491\t42\t0.71068137180843\n",
      "-19.1528931862819\t42\t0.793171173893337\n",
      "-19.0196404290678\t42\t0.901042453542831\n",
      "1.52983834416075\t42\t1.0596766883215\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_LVM2_5.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_P_LVM1_2.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c5.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c5/human_ensembl_msigdb_c5.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_P_LVM1_2.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 5917\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t5917\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 5917\n",
      "cat 1292, mi = 0.01089400, GO_ESTABLISHMENT_OF_PROTEIN_LOCALIZATION_TO_ENDOPLASMIC_RETICULUM\n",
      "cat 3995, mi = 0.01035834, GO_PROTEIN_LOCALIZATION_TO_ENDOPLASMIC_RETICULUM\n",
      "cat 957, mi = 0.00995590, GO_CYTOSOLIC_RIBOSOME\n",
      "cat 5402, mi = 0.00911104, GO_SISTER_CHROMATID_SEGREGATION\n",
      "cat 2928, mi = 0.00878869, GO_NUCLEAR_TRANSCRIBED_MRNA_CATABOLIC_PROCESS_NONSENSE_MEDIATED_DECAY\n",
      "cat 5245, mi = 0.00876755, GO_RIBOSOMAL_SUBUNIT\n",
      "cat 5718, mi = 0.00873818, GO_TRANSLATIONAL_INITIATION\n",
      "cat 775, mi = 0.00869337, GO_CHROMOSOME_CENTROMERIC_REGION\n",
      "cat 4041, mi = 0.00789914, GO_PROTEIN_TARGETING_TO_MEMBRANE\n",
      "cat 854, mi = 0.00745763, GO_CONDENSED_CHROMOSOME\n",
      "cat 2330, mi = 0.00696272, GO_MULTI_ORGANISM_METABOLIC_PROCESS\n",
      "cat 5401, mi = 0.00687447, GO_SISTER_CHROMATID_COHESION\n",
      "cat 2219, mi = 0.00667020, GO_MITOCHONDRIAL_PROTEIN_COMPLEX\n",
      "cat 855, mi = 0.00663456, GO_CONDENSED_CHROMOSOME_CENTROMERIC_REGION\n",
      "cat 954, mi = 0.00659866, GO_CYTOSOLIC_LARGE_RIBOSOMAL_SUBUNIT\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 1292, mi = 0.011, pvalue = 0.000, GO_ESTABLISHMENT_OF_PROTEIN_LOCALIZATION_TO_ENDOPLASMIC_RETICULUM\n",
      "Passed: cat 5402, mi = 0.009, pvalue = 0.000, GO_SISTER_CHROMATID_SEGREGATION\n",
      "Passed: cat 2219, mi = 0.007, pvalue = 0.000, GO_MITOCHONDRIAL_PROTEIN_COMPLEX\n",
      "Passed: cat 5499, mi = 0.004, pvalue = 0.000, GO_SPLICEOSOMAL_COMPLEX\n",
      "Passed: cat 5755, mi = 0.003, pvalue = 0.000, GO_TRNA_METABOLIC_PROCESS\n",
      "Passed: cat 710, mi = 0.003, pvalue = 0.000, GO_CENTRIOLE\n",
      "Passed: cat 3910, mi = 0.003, pvalue = 0.000, GO_PROTEASOME_COMPLEX\n",
      "Passed: cat 5719, mi = 0.003, pvalue = 0.000, GO_TRANSLATIONAL_TERMINATION\n",
      "Passed: cat 2387, mi = 0.002, pvalue = 0.000, GO_NAD_METABOLIC_PROCESS\n",
      "Passed: cat 2082, mi = 0.002, pvalue = 0.000, GO_MCM_COMPLEX\n",
      "Passed: cat 5895, mi = 0.002, pvalue = 0.000, GO_WATER_SOLUBLE_VITAMIN_METABOLIC_PROCESS\n",
      "Passed: cat 3290, mi = 0.002, pvalue = 0.000, GO_PIGMENT_GRANULE\n",
      "Passed: cat 2537, mi = 0.002, pvalue = 0.000, GO_NEGATIVE_REGULATION_OF_ERBB_SIGNALING_PATHWAY\n",
      "Passed: cat 1944, mi = 0.002, pvalue = 0.000, GO_LIPID_OXIDATION\n",
      "Passed: cat 1630, mi = 0.002, pvalue = 0.000, GO_HIPPO_SIGNALING\n",
      "Passed: cat 3201, mi = 0.002, pvalue = 0.000, GO_PHOSPHATASE_COMPLEX\n",
      "Passed: cat 3865, mi = 0.002, pvalue = 0.000, GO_POTASSIUM_CHANNEL_ACTIVITY\n",
      "Passed: cat 226, mi = 0.002, pvalue = 0.000, GO_ARACHIDONIC_ACID_METABOLIC_PROCESS\n",
      "Passed: cat 1597, mi = 0.002, pvalue = 0.000, GO_HEAT_SHOCK_PROTEIN_BINDING\n",
      "Passed: cat 4527, mi = 0.002, pvalue = 0.001, GO_REGULATION_OF_INTERFERON_GAMMA_PRODUCTION\n",
      "Passed: cat 18, mi = 0.002, pvalue = 0.001, GO_ACETYLTRANSFERASE_COMPLEX\n",
      "Passed: cat 3849, mi = 0.002, pvalue = 0.001, GO_POSITIVE_REGULATION_OF_VIRAL_TRANSCRIPTION\n",
      "Passed: cat 4889, mi = 0.002, pvalue = 0.001, GO_REGULATION_OF_STEM_CELL_PROLIFERATION\n",
      "Passed: cat 2917, mi = 0.002, pvalue = 0.001, GO_NUCLEAR_NUCLEOSOME\n",
      "Passed: cat 865, mi = 0.002, pvalue = 0.001, GO_COPI_COATED_VESICLE\n",
      "Passed: cat 4423, mi = 0.002, pvalue = 0.002, GO_REGULATION_OF_ESTABLISHMENT_OF_PROTEIN_LOCALIZATION_TO_PLASMA_MEMBRANE\n",
      "Passed: cat 4157, mi = 0.002, pvalue = 0.001, GO_REGULATION_OF_ACTIN_CYTOSKELETON_REORGANIZATION\n",
      "Passed: cat 4924, mi = 0.002, pvalue = 0.001, GO_REGULATION_OF_TELOMERE_MAINTENANCE_VIA_TELOMERE_LENGTHENING\n",
      "Passed: cat 3099, mi = 0.001, pvalue = 0.004, GO_OXIDOREDUCTASE_ACTIVITY_ACTING_ON_PEROXIDE_AS_ACCEPTOR\n",
      "Passed: cat 4645, mi = 0.001, pvalue = 0.003, GO_REGULATION_OF_MRNA_CATABOLIC_PROCESS\n",
      "Passed: cat 3253, mi = 0.001, pvalue = 0.002, GO_PHOSPHOLIPASE_C_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 5435, mi = 0.001, pvalue = 0.005, GO_SNORNA_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 3996, mi = 0.001, pvalue = 0.006, GO_PROTEIN_LOCALIZATION_TO_GOLGI_APPARATUS\n",
      "Passed: cat 1565, mi = 0.001, pvalue = 0.004, GO_GUANOSINE_CONTAINING_COMPOUND_BIOSYNTHETIC_PROCESS\n",
      "Passed: cat 171, mi = 0.001, pvalue = 0.005, GO_ANCHORED_COMPONENT_OF_MEMBRANE\n",
      "Passed: cat 1095, mi = 0.001, pvalue = 0.003, GO_DNA_REPLICATION_CHECKPOINT\n",
      "Didn't pass (cutoff=0.0050): cat 3885, mi = 0.001, pvalue = 0.006, GO_PRE_AUTOPHAGOSOMAL_STRUCTURE\n",
      "Didn't pass (cutoff=0.0050): cat 5722, mi = 0.001, pvalue = 0.006, GO_TRANSLATION_INITIATION_FACTOR_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 2753, mi = 0.001, pvalue = 0.006, GO_NEGATIVE_REGULATION_OF_STEROID_METABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 1670, mi = 0.001, pvalue = 0.010, GO_HORMONE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 1745, mi = 0.001, pvalue = 0.009, GO_INORGANIC_ANION_TRANSMEMBRANE_TRANSPORTER_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 1603, mi = 0.001, pvalue = 0.011, GO_HEME_METABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 4184, mi = 0.001, pvalue = 0.011, GO_REGULATION_OF_ANION_TRANSMEMBRANE_TRANSPORT\n",
      "Didn't pass (cutoff=0.0050): cat 750, mi = 0.001, pvalue = 0.009, GO_CHLORIDE_CHANNEL_COMPLEX\n",
      "Didn't pass (cutoff=0.0050): cat 1746, mi = 0.001, pvalue = 0.010, GO_INORGANIC_ANION_TRANSPORT\n",
      "Didn't pass (cutoff=0.0050): cat 3138, mi = 0.001, pvalue = 0.006, GO_PENTOSE_PHOSPHATE_SHUNT\n",
      "Didn't pass (cutoff=0.0050): cat 183, mi = 0.001, pvalue = 0.011, GO_ANION_CHANNEL_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 1609, mi = 0.001, pvalue = 0.005, GO_HEPARAN_SULFATE_PROTEOGLYCAN_BIOSYNTHETIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 2906, mi = 0.001, pvalue = 0.011, GO_NUCLEAR_EUCHROMATIN\n",
      "Didn't pass (cutoff=0.0050): cat 752, mi = 0.001, pvalue = 0.013, GO_CHLORIDE_TRANSPORT\n",
      "Didn't pass (cutoff=0.0050): cat 4604, mi = 0.001, pvalue = 0.014, GO_REGULATION_OF_MAST_CELL_ACTIVATION\n",
      "Didn't pass (cutoff=0.0050): cat 4781, mi = 0.001, pvalue = 0.014, GO_REGULATION_OF_PROTEIN_HOMOOLIGOMERIZATION\n",
      "Didn't pass (cutoff=0.0050): cat 1653, mi = 0.001, pvalue = 0.015, GO_HISTONE_METHYLTRANSFERASE_ACTIVITY_H3_K4_SPECIFIC_\n",
      "Didn't pass (cutoff=0.0050): cat 3345, mi = 0.001, pvalue = 0.021, GO_PORPHYRIN_CONTAINING_COMPOUND_METABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 1314, mi = 0.001, pvalue = 0.013, GO_EUKARYOTIC_TRANSLATION_INITIATION_FACTOR_3_COMPLEX\n",
      "Didn't pass (cutoff=0.0050): cat 2874, mi = 0.001, pvalue = 0.013, GO_NEUTROPHIL_MEDIATED_IMMUNITY\n",
      "Total number of categories evaluated = 462.\n",
      "Number of categories that passed the tests = 34\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_P_LVM1_2.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_P_LVM1_2.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 1, 5, 0.0319792527458834\n",
      "best merge is 9, 17, 0.0460567900674674\n",
      "best merge is 9, 16, 0.0503830971590338\n",
      "best merge is 9, 32, 0.0762948950357095\n",
      "best merge is 1, 23, 0.0878802264778759\n",
      "best merge is 9, 33, 0.0933468835364737\n",
      "best merge is 4, 11, 0.108353089513683\n",
      "best merge is 8, 30, 0.110455399617598\n",
      "best merge is 15, 20, 0.123863933303954\n",
      "best merge is 4, 7, 0.128434235913246\n",
      "best merge is 6, 24, 0.149140600628221\n",
      "best merge is 2, 13, 0.228603309994102\n",
      "best merge is 3, 6, 0.238327704922372\n",
      "best merge is 12, 15, 0.265423383547825\n",
      "best merge is 8, 31, 0.270024275971239\n",
      "best merge is 18, 21, 0.279729424452115\n",
      "best merge is 10, 28, 0.313846186976021\n",
      "best merge is 12, 14, 0.321558129243234\n",
      "best merge is 1, 22, 0.331590428816679\n",
      "best merge is 2, 27, 0.337921138606875\n",
      "best merge is 12, 25, 0.340530299248559\n",
      "best merge is 3, 4, 0.349175235938869\n",
      "best merge is 0, 2, 0.368843290478048\n",
      "best merge is 0, 3, 0.510990017564452\n",
      "best merge is 19, 29, 0.524954449299593\n",
      "best merge is 8, 10, 0.539319694874844\n",
      "best merge is 12, 18, 0.576836815611792\n",
      "best merge is 9, 19, 0.771433021443848\n",
      "best merge is 8, 26, 0.798090831865219\n",
      "best merge is 0, 8, 0.877029657668204\n",
      "order classes\n",
      "10 - 1.25176647913072\n",
      "4 - 1.27766792916115\n",
      "6 - 2.57856077668317\n",
      "7 - 1.7819528600047\n",
      "7 - 1.79012554771768\n",
      "0 2 13 27 3 6 24 4 11 7\n",
      "1 5 23 22\n",
      "8 30 31 10 28 26\n",
      "9 17 16 32 33 19 29\n",
      "12 15 20 14 25 18 21\n",
      "Done.Start drawing\n",
      "5.29\t0.00\n",
      "GO_SISTER_CHROMATID_SEGREGATION\n",
      "GO_CENTRIOLE\n",
      "GO_NUCLEAR_NUCLEOSOME\n",
      "GO_REGULATION_OF_STEM_CELL_PROLIFERATION\n",
      "GO_REGULATION_OF_MRNA_CATABOLIC_PROCESS\n",
      "GO_HIPPO_SIGNALING\n",
      "GO_NEGATIVE_REGULATION_OF_ERBB_SIGNALING_PATHWAY\n",
      "GO_PHOSPHATASE_COMPLEX\n",
      "GO_ACETYLTRANSFERASE_COMPLEX\n",
      "GO_REGULATION_OF_ESTABLISHMENT_OF_PROTEIN_LOCALIZATION_TO_PLASMA_MEMBRANE\n",
      "GO_HEAT_SHOCK_PROTEIN_BINDING\n",
      "GO_POSITIVE_REGULATION_OF_VIRAL_TRANSCRIPTION\n",
      "GO_SPLICEOSOMAL_COMPLEX\n",
      "GO_COPI_COATED_VESICLE\n",
      "GO_ESTABLISHMENT_OF_PROTEIN_LOCALIZATION_TO_ENDOPLASMIC_RETICULUM\n",
      "GO_PROTEASOME_COMPLEX\n",
      "GO_TRNA_METABOLIC_PROCESS\n",
      "GO_PIGMENT_GRANULE\n",
      "GO_TRANSLATIONAL_TERMINATION\n",
      "GO_MITOCHONDRIAL_PROTEIN_COMPLEX\n",
      "GO_LIPID_OXIDATION\n",
      "GO_REGULATION_OF_TELOMERE_MAINTENANCE_VIA_TELOMERE_LENGTHENING\n",
      "GO_GUANOSINE_CONTAINING_COMPOUND_BIOSYNTHETIC_PROCESS\n",
      "GO_WATER_SOLUBLE_VITAMIN_METABOLIC_PROCESS\n",
      "GO_OXIDOREDUCTASE_ACTIVITY_ACTING_ON_PEROXIDE_AS_ACCEPTOR\n",
      "GO_NAD_METABOLIC_PROCESS\n",
      "GO_PHOSPHOLIPASE_C_ACTIVITY\n",
      "GO_REGULATION_OF_ACTIN_CYTOSKELETON_REORGANIZATION\n",
      "GO_REGULATION_OF_INTERFERON_GAMMA_PRODUCTION\n",
      "GO_MCM_COMPLEX\n",
      "GO_ARACHIDONIC_ACID_METABOLIC_PROCESS\n",
      "GO_POTASSIUM_CHANNEL_ACTIVITY\n",
      "GO_ANCHORED_COMPONENT_OF_MEMBRANE\n",
      "GO_DNA_REPLICATION_CHECKPOINT\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_P_LVM1_2.csv_PAGE/filt_FC_P_LVM1_2.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_P_LVM1_2.csv_PAGE/filt_FC_P_LVM1_2.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_P_LVM1_2.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_P_LVM1_2.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 1, 5, 0.0319792527458834\n",
      "best merge is 9, 17, 0.0460567900674674\n",
      "best merge is 9, 16, 0.0503830971590338\n",
      "best merge is 9, 32, 0.0762948950357095\n",
      "best merge is 1, 23, 0.0878802264778759\n",
      "best merge is 9, 33, 0.0933468835364737\n",
      "best merge is 4, 11, 0.108353089513683\n",
      "best merge is 8, 30, 0.110455399617598\n",
      "best merge is 15, 20, 0.123863933303954\n",
      "best merge is 4, 7, 0.128434235913246\n",
      "best merge is 6, 24, 0.149140600628221\n",
      "best merge is 2, 13, 0.228603309994102\n",
      "best merge is 3, 6, 0.238327704922372\n",
      "best merge is 12, 15, 0.265423383547825\n",
      "best merge is 8, 31, 0.270024275971239\n",
      "best merge is 18, 21, 0.279729424452115\n",
      "best merge is 10, 28, 0.313846186976021\n",
      "best merge is 12, 14, 0.321558129243234\n",
      "best merge is 1, 22, 0.331590428816679\n",
      "best merge is 2, 27, 0.337921138606875\n",
      "best merge is 12, 25, 0.340530299248559\n",
      "best merge is 3, 4, 0.349175235938869\n",
      "best merge is 0, 2, 0.368843290478048\n",
      "best merge is 0, 3, 0.510990017564452\n",
      "best merge is 19, 29, 0.524954449299593\n",
      "best merge is 8, 10, 0.539319694874844\n",
      "best merge is 12, 18, 0.576836815611792\n",
      "best merge is 9, 19, 0.771433021443848\n",
      "best merge is 8, 26, 0.798090831865219\n",
      "best merge is 0, 8, 0.877029657668204\n",
      "order classes\n",
      "10 - 1.25176647913072\n",
      "4 - 1.27766792916115\n",
      "6 - 2.57856077668317\n",
      "7 - 1.7819528600047\n",
      "7 - 1.79012554771768\n",
      "0 2 13 27 3 6 24 4 11 7\n",
      "1 5 23 22\n",
      "8 30 31 10 28 26\n",
      "9 17 16 32 33 19 29\n",
      "12 15 20 14 25 18 21\n",
      "Done.-16.8638941398866\t42\t0\n",
      "-13.2117202268431\t42\t6.27221172022684\n",
      "-12.3780718336484\t42\t7.30434782608696\n",
      "-11.8223062381853\t42\t7.93950850661626\n",
      "-11.3856332703214\t42\t8.41587901701323\n",
      "-10.9092627599244\t42\t8.81285444234405\n",
      "-10.3137996219282\t42\t9.36862003780718\n",
      "-9.44045368620038\t42\t10.0037807183365\n",
      "6.55765595463138\t42\t11.1153119092628\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_P_LVM1_2.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_P_LVM1_3.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c5.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c5/human_ensembl_msigdb_c5.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_P_LVM1_3.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 5917\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t5917\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 5917\n",
      "cat 854, mi = 0.01008938, GO_CONDENSED_CHROMOSOME\n",
      "cat 5402, mi = 0.01001329, GO_SISTER_CHROMATID_SEGREGATION\n",
      "cat 1292, mi = 0.00843798, GO_ESTABLISHMENT_OF_PROTEIN_LOCALIZATION_TO_ENDOPLASMIC_RETICULUM\n",
      "cat 5401, mi = 0.00818282, GO_SISTER_CHROMATID_COHESION\n",
      "cat 775, mi = 0.00817067, GO_CHROMOSOME_CENTROMERIC_REGION\n",
      "cat 3995, mi = 0.00769502, GO_PROTEIN_LOCALIZATION_TO_ENDOPLASMIC_RETICULUM\n",
      "cat 1066, mi = 0.00720258, GO_DNA_DEPENDENT_DNA_REPLICATION\n",
      "cat 855, mi = 0.00720197, GO_CONDENSED_CHROMOSOME_CENTROMERIC_REGION\n",
      "cat 1086, mi = 0.00700227, GO_DNA_PACKAGING\n",
      "cat 4041, mi = 0.00685785, GO_PROTEIN_TARGETING_TO_MEMBRANE\n",
      "cat 2089, mi = 0.00633167, GO_MEIOTIC_CELL_CYCLE\n",
      "cat 1057, mi = 0.00610081, GO_DNA_BIOSYNTHETIC_PROCESS\n",
      "cat 957, mi = 0.00606316, GO_CYTOSOLIC_RIBOSOME\n",
      "cat 1875, mi = 0.00590869, GO_KINETOCHORE\n",
      "cat 5245, mi = 0.00588124, GO_RIBOSOMAL_SUBUNIT\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 854, mi = 0.010, pvalue = 0.000, GO_CONDENSED_CHROMOSOME\n",
      "Passed: cat 1292, mi = 0.008, pvalue = 0.000, GO_ESTABLISHMENT_OF_PROTEIN_LOCALIZATION_TO_ENDOPLASMIC_RETICULUM\n",
      "Passed: cat 1066, mi = 0.007, pvalue = 0.000, GO_DNA_DEPENDENT_DNA_REPLICATION\n",
      "Passed: cat 5232, mi = 0.004, pvalue = 0.000, GO_RIBONUCLEOPROTEIN_COMPLEX_LOCALIZATION\n",
      "Passed: cat 5719, mi = 0.004, pvalue = 0.000, GO_TRANSLATIONAL_TERMINATION\n",
      "Passed: cat 710, mi = 0.003, pvalue = 0.000, GO_CENTRIOLE\n",
      "Passed: cat 2219, mi = 0.003, pvalue = 0.000, GO_MITOCHONDRIAL_PROTEIN_COMPLEX\n",
      "Passed: cat 4852, mi = 0.003, pvalue = 0.000, GO_REGULATION_OF_RNA_STABILITY\n",
      "Passed: cat 1808, mi = 0.002, pvalue = 0.000, GO_INTRA_GOLGI_VESICLE_MEDIATED_TRANSPORT\n",
      "Passed: cat 2156, mi = 0.002, pvalue = 0.000, GO_METHYLTRANSFERASE_COMPLEX\n",
      "Passed: cat 3458, mi = 0.002, pvalue = 0.000, GO_POSITIVE_REGULATION_OF_CYTOKINESIS\n",
      "Passed: cat 108, mi = 0.002, pvalue = 0.000, GO_ALCOHOL_BIOSYNTHETIC_PROCESS\n",
      "Passed: cat 2917, mi = 0.002, pvalue = 0.000, GO_NUCLEAR_NUCLEOSOME\n",
      "Passed: cat 4730, mi = 0.002, pvalue = 0.001, GO_REGULATION_OF_PATHWAY_RESTRICTED_SMAD_PROTEIN_PHOSPHORYLATION\n",
      "Passed: cat 2005, mi = 0.002, pvalue = 0.001, GO_LYSOSOMAL_LUMEN\n",
      "Passed: cat 390, mi = 0.002, pvalue = 0.000, GO_CALCIUM_DEPENDENT_PHOSPHOLIPID_BINDING\n",
      "Passed: cat 5109, mi = 0.002, pvalue = 0.000, GO_RESPONSE_TO_INTERFERON_GAMMA\n",
      "Passed: cat 3252, mi = 0.002, pvalue = 0.001, GO_PHOSPHOLIPASE_C_ACTIVATING_G_PROTEIN_COUPLED_RECEPTOR_SIGNALING_PATHWAY\n",
      "Passed: cat 5595, mi = 0.002, pvalue = 0.000, GO_S_ACYLTRANSFERASE_ACTIVITY\n",
      "Passed: cat 1631, mi = 0.002, pvalue = 0.001, GO_HISTONE_ACETYLTRANSFERASE_BINDING\n",
      "Passed: cat 297, mi = 0.002, pvalue = 0.001, GO_BAF_TYPE_COMPLEX\n",
      "Passed: cat 587, mi = 0.002, pvalue = 0.002, GO_CELLULAR_RESPONSE_TO_MECHANICAL_STIMULUS\n",
      "Passed: cat 125, mi = 0.002, pvalue = 0.001, GO_ALPHA_BETA_T_CELL_DIFFERENTIATION\n",
      "Passed: cat 4426, mi = 0.002, pvalue = 0.001, GO_REGULATION_OF_EXECUTION_PHASE_OF_APOPTOSIS\n",
      "Passed: cat 4012, mi = 0.001, pvalue = 0.004, GO_PROTEIN_N_LINKED_GLYCOSYLATION\n",
      "Passed: cat 1342, mi = 0.001, pvalue = 0.003, GO_EXTRACELLULAR_MATRIX_COMPONENT\n",
      "Passed: cat 5095, mi = 0.001, pvalue = 0.003, GO_RESPONSE_TO_HEAT\n",
      "Passed: cat 4636, mi = 0.001, pvalue = 0.003, GO_REGULATION_OF_MITOPHAGY\n",
      "Didn't pass (cutoff=0.0050): cat 2329, mi = 0.001, pvalue = 0.005, GO_MULTI_ORGANISM_MEMBRANE_ORGANIZATION\n",
      "Passed: cat 3371, mi = 0.001, pvalue = 0.004, GO_POSITIVE_REGULATION_OF_ATPASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 2133, mi = 0.001, pvalue = 0.005, GO_METALLOENDOPEPTIDASE_ACTIVITY\n",
      "Passed: cat 5760, mi = 0.001, pvalue = 0.002, GO_TRNA_SPECIFIC_RIBONUCLEASE_ACTIVITY\n",
      "Passed: cat 2136, mi = 0.001, pvalue = 0.004, GO_METALLOPEPTIDASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 3184, mi = 0.001, pvalue = 0.007, GO_PEROXISOME_ORGANIZATION\n",
      "Didn't pass (cutoff=0.0050): cat 2169, mi = 0.001, pvalue = 0.006, GO_MICROBODY_PART\n",
      "Didn't pass (cutoff=0.0050): cat 2288, mi = 0.001, pvalue = 0.010, GO_MORPHOGENESIS_OF_AN_EPITHELIAL_SHEET\n",
      "Didn't pass (cutoff=0.0050): cat 4072, mi = 0.001, pvalue = 0.006, GO_PTERIDINE_CONTAINING_COMPOUND_BIOSYNTHETIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 342, mi = 0.001, pvalue = 0.009, GO_BLOC_COMPLEX\n",
      "Didn't pass (cutoff=0.0050): cat 5830, mi = 0.001, pvalue = 0.010, GO_VACUOLE_FUSION\n",
      "Didn't pass (cutoff=0.0050): cat 5507, mi = 0.001, pvalue = 0.005, GO_STAGA_COMPLEX\n",
      "Didn't pass (cutoff=0.0050): cat 1845, mi = 0.001, pvalue = 0.005, GO_ISOPRENOID_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 1314, mi = 0.001, pvalue = 0.007, GO_EUKARYOTIC_TRANSLATION_INITIATION_FACTOR_3_COMPLEX\n",
      "Passed: cat 1447, mi = 0.001, pvalue = 0.003, GO_GAP_JUNCTION\n",
      "Didn't pass (cutoff=0.0050): cat 1386, mi = 0.001, pvalue = 0.013, GO_FEEDING_BEHAVIOR\n",
      "Didn't pass (cutoff=0.0050): cat 5745, mi = 0.001, pvalue = 0.012, GO_TRANS_GOLGI_NETWORK_TRANSPORT_VESICLE\n",
      "Passed: cat 353, mi = 0.001, pvalue = 0.001, GO_BONE_GROWTH\n",
      "Passed: cat 110, mi = 0.001, pvalue = 0.004, GO_ALCOHOL_DEHYDROGENASE_NADP_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 1617, mi = 0.001, pvalue = 0.014, GO_HETEROPHILIC_CELL_CELL_ADHESION_VIA_PLASMA_MEMBRANE_CELL_ADHESION_MOLECULES\n",
      "Didn't pass (cutoff=0.0050): cat 1619, mi = 0.001, pvalue = 0.011, GO_HETEROTYPIC_CELL_CELL_ADHESION\n",
      "Didn't pass (cutoff=0.0050): cat 4053, mi = 0.001, pvalue = 0.016, GO_PROTEIN_TYROSINE_SERINE_THREONINE_PHOSPHATASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 4052, mi = 0.001, pvalue = 0.012, GO_PROTEIN_TYROSINE_PHOSPHATASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 59, mi = 0.001, pvalue = 0.018, GO_ACTIVATION_OF_JUN_KINASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 477, mi = 0.001, pvalue = 0.014, GO_CATION_CATION_ANTIPORTER_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 1565, mi = 0.001, pvalue = 0.011, GO_GUANOSINE_CONTAINING_COMPOUND_BIOSYNTHETIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 1556, mi = 0.001, pvalue = 0.014, GO_GROWTH_FACTOR_RECEPTOR_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 4229, mi = 0.001, pvalue = 0.005, GO_REGULATION_OF_CAMP_DEPENDENT_PROTEIN_KINASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 3307, mi = 0.001, pvalue = 0.022, GO_PLATELET_AGGREGATION\n",
      "Didn't pass (cutoff=0.0050): cat 37, mi = 0.001, pvalue = 0.016, GO_ACTIN_FILAMENT\n",
      "Passed: cat 1048, mi = 0.001, pvalue = 0.002, GO_DISTAL_TUBULE_DEVELOPMENT\n",
      "Didn't pass (cutoff=0.0050): cat 3292, mi = 0.001, pvalue = 0.021, GO_PIGMENT_GRANULE_ORGANIZATION\n",
      "Didn't pass (cutoff=0.0050): cat 4105, mi = 0.001, pvalue = 0.022, GO_PYRIMIDINE_NUCLEOTIDE_BIOSYNTHETIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 2561, mi = 0.001, pvalue = 0.023, GO_NEGATIVE_REGULATION_OF_GTPASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 4092, mi = 0.001, pvalue = 0.024, GO_PYRIMIDINE_CONTAINING_COMPOUND_BIOSYNTHETIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 56, mi = 0.001, pvalue = 0.021, GO_ACTIVATION_OF_GTPASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 5888, mi = 0.001, pvalue = 0.030, GO_VOLTAGE_GATED_POTASSIUM_CHANNEL_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 3096, mi = 0.001, pvalue = 0.030, GO_OXIDOREDUCTASE_ACTIVITY_ACTING_ON_PAIRED_DONORS_WITH_INCORPORATION_OR_REDUCTION_OF_MOLECULAR_OXYGEN_2_OXOGLUTARATE_AS_ONE_DONOR_AND_INCORPORATION_OF_ONE_ATOM_EACH_OF_OXYGEN_INTO_BOTH_DONORS\n",
      "Didn't pass (cutoff=0.0050): cat 389, mi = 0.001, pvalue = 0.020, GO_CALCIUM_DEPENDENT_CYSTEINE_TYPE_ENDOPEPTIDASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 872, mi = 0.001, pvalue = 0.031, GO_CORECEPTOR_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 1377, mi = 0.001, pvalue = 0.006, GO_FATTY_ACID_TRANSPORTER_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 5622, mi = 0.001, pvalue = 0.020, GO_TETRAHYDROFOLATE_INTERCONVERSION\n",
      "Didn't pass (cutoff=0.0050): cat 3847, mi = 0.001, pvalue = 0.024, GO_POSITIVE_REGULATION_OF_VIRAL_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 525, mi = 0.001, pvalue = 0.034, GO_CELLULAR_PIGMENTATION\n",
      "Didn't pass (cutoff=0.0050): cat 5715, mi = 0.001, pvalue = 0.040, GO_TRANSITION_METAL_ION_TRANSMEMBRANE_TRANSPORTER_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 139, mi = 0.001, pvalue = 0.015, GO_AMINOACYL_TRNA_EDITING_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 2937, mi = 0.001, pvalue = 0.033, GO_NUCLEOBASE_BIOSYNTHETIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 5623, mi = 0.001, pvalue = 0.039, GO_TETRAHYDROFOLATE_METABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 41, mi = 0.001, pvalue = 0.036, GO_ACTIN_FILAMENT_BUNDLE\n",
      "Didn't pass (cutoff=0.0050): cat 4933, mi = 0.001, pvalue = 0.044, GO_REGULATION_OF_TRANSCRIPTION_ELONGATION_FROM_RNA_POLYMERASE_II_PROMOTER\n",
      "Didn't pass (cutoff=0.0050): cat 693, mi = 0.001, pvalue = 0.033, GO_CELL_REDOX_HOMEOSTASIS\n",
      "Total number of categories evaluated = 633.\n",
      "Number of categories that passed the tests = 35\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_P_LVM1_3.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_P_LVM1_3.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 0, 2, 0.0114404379355395\n",
      "best merge is 0, 5, 0.0219305004515488\n",
      "best merge is 0, 10, 0.034381217601871\n",
      "best merge is 16, 32, 0.0412712204738644\n",
      "best merge is 14, 16, 0.104975122925462\n",
      "best merge is 12, 17, 0.107250291955358\n",
      "best merge is 15, 22, 0.118176538577349\n",
      "best merge is 23, 28, 0.127508873506686\n",
      "best merge is 14, 34, 0.152007975186159\n",
      "best merge is 6, 7, 0.160276867721691\n",
      "best merge is 3, 9, 0.196936914521718\n",
      "best merge is 4, 23, 0.209243196524067\n",
      "best merge is 14, 15, 0.210678755783959\n",
      "best merge is 0, 12, 0.230233873108791\n",
      "best merge is 6, 20, 0.23404235753228\n",
      "best merge is 1, 21, 0.251061459849756\n",
      "best merge is 18, 27, 0.26215888830499\n",
      "best merge is 3, 6, 0.294543607513474\n",
      "best merge is 14, 31, 0.298820406801166\n",
      "best merge is 25, 30, 0.306630321650122\n",
      "best merge is 3, 18, 0.381033160240692\n",
      "best merge is 19, 24, 0.406552633998064\n",
      "best merge is 14, 25, 0.435096549732305\n",
      "best merge is 13, 33, 0.513945548364787\n",
      "best merge is 3, 4, 0.538402843733481\n",
      "best merge is 3, 11, 0.552868491084966\n",
      "best merge is 26, 29, 0.656739371911441\n",
      "best merge is 13, 14, 0.759442861847179\n",
      "best merge is 1, 8, 0.797759334789504\n",
      "best merge is 0, 26, 0.799219543486661\n",
      "best merge is 1, 19, 0.860047483586617\n",
      "order classes\n",
      "8 - 1.68915198897557\n",
      "3 - 4.78742996518568\n",
      "11 - 1.26463811947677\n",
      "11 - 1.30878823801194\n",
      "2 - 3.55733554748306\n",
      "3 9 6 7 20 18 27 4 23 28 11\n",
      "1 21 8\n",
      "0 2 5 10 12 17 26 29\n",
      "13 33 14 16 32 34 15 22 31 25 30\n",
      "19 24\n",
      "Done.Start drawing\n",
      "11.42\t0.00\n",
      "GO_LYSOSOMAL_LUMEN\n",
      "GO_RESPONSE_TO_INTERFERON_GAMMA\n",
      "GO_BONE_GROWTH\n",
      "GO_DISTAL_TUBULE_DEVELOPMENT\n",
      "GO_CALCIUM_DEPENDENT_PHOSPHOLIPID_BINDING\n",
      "GO_ALPHA_BETA_T_CELL_DIFFERENTIATION\n",
      "GO_GAP_JUNCTION\n",
      "GO_EXTRACELLULAR_MATRIX_COMPONENT\n",
      "GO_METALLOPEPTIDASE_ACTIVITY\n",
      "GO_REGULATION_OF_PATHWAY_RESTRICTED_SMAD_PROTEIN_PHOSPHORYLATION\n",
      "GO_PROTEIN_N_LINKED_GLYCOSYLATION\n",
      "GO_INTRA_GOLGI_VESICLE_MEDIATED_TRANSPORT\n",
      "GO_HISTONE_ACETYLTRANSFERASE_BINDING\n",
      "GO_ESTABLISHMENT_OF_PROTEIN_LOCALIZATION_TO_ENDOPLASMIC_RETICULUM\n",
      "GO_CELLULAR_RESPONSE_TO_MECHANICAL_STIMULUS\n",
      "GO_ALCOHOL_DEHYDROGENASE_NADP_ACTIVITY\n",
      "GO_TRANSLATIONAL_TERMINATION\n",
      "GO_POSITIVE_REGULATION_OF_ATPASE_ACTIVITY\n",
      "GO_RESPONSE_TO_HEAT\n",
      "GO_MITOCHONDRIAL_PROTEIN_COMPLEX\n",
      "GO_BAF_TYPE_COMPLEX\n",
      "GO_REGULATION_OF_EXECUTION_PHASE_OF_APOPTOSIS\n",
      "GO_ALCOHOL_BIOSYNTHETIC_PROCESS\n",
      "GO_RIBONUCLEOPROTEIN_COMPLEX_LOCALIZATION\n",
      "GO_METHYLTRANSFERASE_COMPLEX\n",
      "GO_REGULATION_OF_RNA_STABILITY\n",
      "GO_S_ACYLTRANSFERASE_ACTIVITY\n",
      "GO_REGULATION_OF_MITOPHAGY\n",
      "GO_TRNA_SPECIFIC_RIBONUCLEASE_ACTIVITY\n",
      "GO_CONDENSED_CHROMOSOME\n",
      "GO_DNA_DEPENDENT_DNA_REPLICATION\n",
      "GO_CENTRIOLE\n",
      "GO_POSITIVE_REGULATION_OF_CYTOKINESIS\n",
      "GO_NUCLEAR_NUCLEOSOME\n",
      "GO_PHOSPHOLIPASE_C_ACTIVATING_G_PROTEIN_COUPLED_RECEPTOR_SIGNALING_PATHWAY\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_P_LVM1_3.csv_PAGE/filt_FC_P_LVM1_3.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_P_LVM1_3.csv_PAGE/filt_FC_P_LVM1_3.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_P_LVM1_3.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_P_LVM1_3.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 0, 2, 0.0114404379355395\n",
      "best merge is 0, 5, 0.0219305004515488\n",
      "best merge is 0, 10, 0.034381217601871\n",
      "best merge is 16, 32, 0.0412712204738644\n",
      "best merge is 14, 16, 0.104975122925462\n",
      "best merge is 12, 17, 0.107250291955358\n",
      "best merge is 15, 22, 0.118176538577349\n",
      "best merge is 23, 28, 0.127508873506686\n",
      "best merge is 14, 34, 0.152007975186159\n",
      "best merge is 6, 7, 0.160276867721691\n",
      "best merge is 3, 9, 0.196936914521718\n",
      "best merge is 4, 23, 0.209243196524067\n",
      "best merge is 14, 15, 0.210678755783959\n",
      "best merge is 0, 12, 0.230233873108791\n",
      "best merge is 6, 20, 0.23404235753228\n",
      "best merge is 1, 21, 0.251061459849756\n",
      "best merge is 18, 27, 0.26215888830499\n",
      "best merge is 3, 6, 0.294543607513474\n",
      "best merge is 14, 31, 0.298820406801166\n",
      "best merge is 25, 30, 0.306630321650122\n",
      "best merge is 3, 18, 0.381033160240692\n",
      "best merge is 19, 24, 0.406552633998064\n",
      "best merge is 14, 25, 0.435096549732305\n",
      "best merge is 13, 33, 0.513945548364787\n",
      "best merge is 3, 4, 0.538402843733481\n",
      "best merge is 3, 11, 0.552868491084966\n",
      "best merge is 26, 29, 0.656739371911441\n",
      "best merge is 13, 14, 0.759442861847179\n",
      "best merge is 1, 8, 0.797759334789504\n",
      "best merge is 0, 26, 0.799219543486661\n",
      "best merge is 1, 19, 0.860047483586617\n",
      "order classes\n",
      "8 - 1.68915198897557\n",
      "3 - 4.78742996518568\n",
      "11 - 1.26463811947677\n",
      "11 - 1.30878823801194\n",
      "2 - 3.55733554748306\n",
      "3 9 6 7 20 18 27 4 23 28 11\n",
      "1 21 8\n",
      "0 2 5 10 12 17 26 29\n",
      "13 33 14 16 32 34 15 22 31 25 30\n",
      "19 24\n",
      "Done.-19.1725043782837\t42\t0\n",
      "-18.1059544658494\t42\t1.65499124343257\n",
      "-17.7381786339755\t42\t2.13309982486865\n",
      "-17.4807355516637\t42\t2.39054290718039\n",
      "-17.223292469352\t42\t2.64798598949212\n",
      "-16.9658493870403\t42\t2.90542907180385\n",
      "-16.6348511383538\t42\t3.16287215411559\n",
      "-16.0831873905429\t42\t3.56742556917688\n",
      "3.13309982486865\t42\t4.2661996497373\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_P_LVM1_3.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_P_LVM1_4.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c5.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c5/human_ensembl_msigdb_c5.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_P_LVM1_4.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 5917\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t5917\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 5917\n",
      "cat 5245, mi = 0.01359510, GO_RIBOSOMAL_SUBUNIT\n",
      "cat 1292, mi = 0.01347320, GO_ESTABLISHMENT_OF_PROTEIN_LOCALIZATION_TO_ENDOPLASMIC_RETICULUM\n",
      "cat 3995, mi = 0.01327650, GO_PROTEIN_LOCALIZATION_TO_ENDOPLASMIC_RETICULUM\n",
      "cat 5718, mi = 0.01321226, GO_TRANSLATIONAL_INITIATION\n",
      "cat 957, mi = 0.01176857, GO_CYTOSOLIC_RIBOSOME\n",
      "cat 2928, mi = 0.00983797, GO_NUCLEAR_TRANSCRIBED_MRNA_CATABOLIC_PROCESS_NONSENSE_MEDIATED_DECAY\n",
      "cat 2330, mi = 0.00971932, GO_MULTI_ORGANISM_METABOLIC_PROCESS\n",
      "cat 954, mi = 0.00854899, GO_CYTOSOLIC_LARGE_RIBOSOMAL_SUBUNIT\n",
      "cat 1889, mi = 0.00853365, GO_LARGE_RIBOSOMAL_SUBUNIT\n",
      "cat 4041, mi = 0.00824073, GO_PROTEIN_TARGETING_TO_MEMBRANE\n",
      "cat 1742, mi = 0.00709964, GO_INNER_MITOCHONDRIAL_MEMBRANE_PROTEIN_COMPLEX\n",
      "cat 2219, mi = 0.00626369, GO_MITOCHONDRIAL_PROTEIN_COMPLEX\n",
      "cat 5028, mi = 0.00602933, GO_RESPIRATORY_CHAIN\n",
      "cat 2217, mi = 0.00579070, GO_MITOCHONDRIAL_MEMBRANE_PART\n",
      "cat 3085, mi = 0.00534551, GO_OXIDATIVE_PHOSPHORYLATION\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 5245, mi = 0.014, pvalue = 0.000, GO_RIBOSOMAL_SUBUNIT\n",
      "Passed: cat 1742, mi = 0.007, pvalue = 0.000, GO_INNER_MITOCHONDRIAL_MEMBRANE_PROTEIN_COMPLEX\n",
      "Passed: cat 5499, mi = 0.005, pvalue = 0.000, GO_SPLICEOSOMAL_COMPLEX\n",
      "Passed: cat 4852, mi = 0.003, pvalue = 0.000, GO_REGULATION_OF_RNA_STABILITY\n",
      "Passed: cat 4921, mi = 0.002, pvalue = 0.000, GO_REGULATION_OF_TELOMERASE_RNA_LOCALIZATION_TO_CAJAL_BODY\n",
      "Passed: cat 3847, mi = 0.002, pvalue = 0.000, GO_POSITIVE_REGULATION_OF_VIRAL_PROCESS\n",
      "Passed: cat 4551, mi = 0.002, pvalue = 0.000, GO_REGULATION_OF_INTRINSIC_APOPTOTIC_SIGNALING_PATHWAY\n",
      "Passed: cat 2966, mi = 0.002, pvalue = 0.000, GO_NUCLEOSOMAL_DNA_BINDING\n",
      "Passed: cat 3633, mi = 0.002, pvalue = 0.000, GO_POSITIVE_REGULATION_OF_MITOTIC_NUCLEAR_DIVISION\n",
      "Passed: cat 3871, mi = 0.002, pvalue = 0.000, GO_POTASSIUM_ION_TRANSPORT\n",
      "Passed: cat 3374, mi = 0.002, pvalue = 0.001, GO_POSITIVE_REGULATION_OF_AXON_EXTENSION\n",
      "Passed: cat 5722, mi = 0.002, pvalue = 0.001, GO_TRANSLATION_INITIATION_FACTOR_ACTIVITY\n",
      "Passed: cat 144, mi = 0.002, pvalue = 0.000, GO_AMINO_ACID_ACTIVATION\n",
      "Passed: cat 3323, mi = 0.002, pvalue = 0.000, GO_POLYAMINE_METABOLIC_PROCESS\n",
      "Passed: cat 171, mi = 0.002, pvalue = 0.001, GO_ANCHORED_COMPONENT_OF_MEMBRANE\n",
      "Passed: cat 4118, mi = 0.002, pvalue = 0.002, GO_RAB_GUANYL_NUCLEOTIDE_EXCHANGE_FACTOR_ACTIVITY\n",
      "Passed: cat 1162, mi = 0.001, pvalue = 0.003, GO_EMBRYONIC_SKELETAL_SYSTEM_MORPHOGENESIS\n",
      "Passed: cat 149, mi = 0.001, pvalue = 0.003, GO_AMINO_ACID_TRANSMEMBRANE_TRANSPORT\n",
      "Passed: cat 5262, mi = 0.001, pvalue = 0.002, GO_RNA_INTERFERENCE\n",
      "Didn't pass (cutoff=0.0050): cat 2178, mi = 0.001, pvalue = 0.007, GO_MICROTUBULE_BUNDLE_FORMATION\n",
      "Passed: cat 339, mi = 0.001, pvalue = 0.005, GO_BLASTOCYST_GROWTH\n",
      "Passed: cat 1082, mi = 0.001, pvalue = 0.001, GO_DNA_METHYLATION_INVOLVED_IN_GAMETE_GENERATION\n",
      "Didn't pass (cutoff=0.0050): cat 4451, mi = 0.001, pvalue = 0.009, GO_REGULATION_OF_FILOPODIUM_ASSEMBLY\n",
      "Didn't pass (cutoff=0.0050): cat 984, mi = 0.001, pvalue = 0.010, GO_DENDRITE_MORPHOGENESIS\n",
      "Didn't pass (cutoff=0.0050): cat 4053, mi = 0.001, pvalue = 0.011, GO_PROTEIN_TYROSINE_SERINE_THREONINE_PHOSPHATASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 2384, mi = 0.001, pvalue = 0.010, GO_NAD_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 398, mi = 0.001, pvalue = 0.007, GO_CALCIUM_ION_REGULATED_EXOCYTOSIS_OF_NEUROTRANSMITTER\n",
      "Didn't pass (cutoff=0.0050): cat 2428, mi = 0.001, pvalue = 0.011, GO_NEGATIVE_REGULATION_OF_BLOOD_PRESSURE\n",
      "Passed: cat 4082, mi = 0.001, pvalue = 0.003, GO_PURINE_NUCLEOBASE_BIOSYNTHETIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 4846, mi = 0.001, pvalue = 0.005, GO_REGULATION_OF_RETINOIC_ACID_RECEPTOR_SIGNALING_PATHWAY\n",
      "Didn't pass (cutoff=0.0050): cat 3519, mi = 0.001, pvalue = 0.012, GO_POSITIVE_REGULATION_OF_FILOPODIUM_ASSEMBLY\n",
      "Didn't pass (cutoff=0.0050): cat 1663, mi = 0.001, pvalue = 0.013, GO_HOMEOSTASIS_OF_NUMBER_OF_CELLS_WITHIN_A_TISSUE\n",
      "Didn't pass (cutoff=0.0050): cat 5802, mi = 0.001, pvalue = 0.012, GO_UBIQUITIN_LIKE_PROTEIN_CONJUGATING_ENZYME_ACTIVITY\n",
      "Passed: cat 5846, mi = 0.001, pvalue = 0.004, GO_VENTRICULAR_CARDIAC_MUSCLE_CELL_DEVELOPMENT\n",
      "Passed: cat 3919, mi = 0.001, pvalue = 0.003, GO_PROTEIN_AUTOPROCESSING\n",
      "Didn't pass (cutoff=0.0050): cat 5538, mi = 0.001, pvalue = 0.013, GO_STRUCTURAL_CONSTITUENT_OF_CYTOSKELETON\n",
      "Didn't pass (cutoff=0.0050): cat 3197, mi = 0.001, pvalue = 0.008, GO_PHENOL_CONTAINING_COMPOUND_BIOSYNTHETIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 5373, mi = 0.001, pvalue = 0.007, GO_SHORT_CHAIN_FATTY_ACID_METABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 5823, mi = 0.001, pvalue = 0.010, GO_VACUOLAR_ACIDIFICATION\n",
      "Didn't pass (cutoff=0.0050): cat 286, mi = 0.001, pvalue = 0.019, GO_AXONEME_ASSEMBLY\n",
      "Didn't pass (cutoff=0.0050): cat 3239, mi = 0.001, pvalue = 0.018, GO_PHOSPHATIDYLINOSITOL_KINASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 1522, mi = 0.001, pvalue = 0.018, GO_GLYCOSPHINGOLIPID_METABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 750, mi = 0.001, pvalue = 0.017, GO_CHLORIDE_CHANNEL_COMPLEX\n",
      "Didn't pass (cutoff=0.0050): cat 1944, mi = 0.001, pvalue = 0.018, GO_LIPID_OXIDATION\n",
      "Passed: cat 5311, mi = 0.001, pvalue = 0.004, GO_S100_PROTEIN_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 1967, mi = 0.001, pvalue = 0.018, GO_LONG_CHAIN_FATTY_ACID_METABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 3709, mi = 0.001, pvalue = 0.006, GO_POSITIVE_REGULATION_OF_PHOSPHOPROTEIN_PHOSPHATASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 3821, mi = 0.001, pvalue = 0.009, GO_POSITIVE_REGULATION_OF_TRIGLYCERIDE_BIOSYNTHETIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 3325, mi = 0.001, pvalue = 0.018, GO_POLYOL_CATABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 5853, mi = 0.001, pvalue = 0.021, GO_VERY_LONG_CHAIN_FATTY_ACID_METABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 865, mi = 0.001, pvalue = 0.021, GO_COPI_COATED_VESICLE\n",
      "Didn't pass (cutoff=0.0050): cat 3225, mi = 0.001, pvalue = 0.021, GO_PHOSPHATIDYLINOSITOL_3_KINASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 3091, mi = 0.001, pvalue = 0.021, GO_OXIDOREDUCTASE_ACTIVITY_ACTING_ON_CH_OH_GROUP_OF_DONORS\n",
      "Didn't pass (cutoff=0.0050): cat 3109, mi = 0.001, pvalue = 0.019, GO_OXIDOREDUCTASE_ACTIVITY_ACTING_ON_THE_CH_OH_GROUP_OF_DONORS_NAD_OR_NADP_AS_ACCEPTOR\n",
      "Didn't pass (cutoff=0.0050): cat 1723, mi = 0.001, pvalue = 0.027, GO_INACTIVATION_OF_MAPK_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 1545, mi = 0.001, pvalue = 0.029, GO_GOLGI_TO_VACUOLE_TRANSPORT\n",
      "Didn't pass (cutoff=0.0050): cat 2767, mi = 0.001, pvalue = 0.030, GO_NEGATIVE_REGULATION_OF_TOLL_LIKE_RECEPTOR_SIGNALING_PATHWAY\n",
      "Didn't pass (cutoff=0.0050): cat 4023, mi = 0.001, pvalue = 0.033, GO_PROTEIN_PHOSPHATASE_TYPE_2A_COMPLEX\n",
      "Didn't pass (cutoff=0.0050): cat 1239, mi = 0.001, pvalue = 0.024, GO_EPIBOLY\n",
      "Didn't pass (cutoff=0.0050): cat 5868, mi = 0.001, pvalue = 0.019, GO_VIRAL_LATENCY\n",
      "Didn't pass (cutoff=0.0050): cat 1489, mi = 0.001, pvalue = 0.031, GO_GLUCOSYLTRANSFERASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 1049, mi = 0.001, pvalue = 0.031, GO_DISULFIDE_OXIDOREDUCTASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 3258, mi = 0.001, pvalue = 0.038, GO_PHOSPHOLIPID_DEPHOSPHORYLATION\n",
      "Didn't pass (cutoff=0.0050): cat 11, mi = 0.001, pvalue = 0.024, GO_5_3_EXONUCLEASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 3201, mi = 0.001, pvalue = 0.036, GO_PHOSPHATASE_COMPLEX\n",
      "Total number of categories evaluated = 448.\n",
      "Number of categories that passed the tests = 25\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_P_LVM1_4.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_P_LVM1_4.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 2, 4, 0.025511157602079\n",
      "best merge is 0, 11, 0.0280785732973895\n",
      "best merge is 1, 24, 0.0482674069810172\n",
      "best merge is 2, 3, 0.0577128535524087\n",
      "best merge is 2, 12, 0.0629545546499368\n",
      "best merge is 2, 6, 0.0796641610509358\n",
      "best merge is 2, 7, 0.127197035455499\n",
      "best merge is 1, 5, 0.135739447472175\n",
      "best merge is 0, 22, 0.17459426749547\n",
      "best merge is 1, 2, 0.191431876367012\n",
      "best merge is 0, 13, 0.192955096584292\n",
      "best merge is 1, 8, 0.255775497716702\n",
      "best merge is 14, 20, 0.260047793879526\n",
      "best merge is 1, 23, 0.328887259773629\n",
      "best merge is 16, 19, 0.34659283303849\n",
      "best merge is 10, 17, 0.415131800699739\n",
      "best merge is 18, 21, 0.463084557557279\n",
      "best merge is 0, 1, 0.523268193228559\n",
      "best merge is 10, 15, 0.711391542809042\n",
      "best merge is 0, 16, 0.774240717517063\n",
      "best merge is 10, 14, 0.871131857784171\n",
      "order classes\n",
      "17 - 0.59225078121824\n",
      "3 - 3.40354608577375\n",
      "2 - 1.62529871174704\n",
      "2 - 2.894278484733\n",
      "0 11 22 13 1 24 5 2 4 3 12 6 7 8 23 16 19\n",
      "9\n",
      "10 17 15\n",
      "14 20\n",
      "18 21\n",
      "Done.Start drawing\n",
      "110.21\t0.00\n",
      "GO_RNA_INTERFERENCE\n",
      "GO_PURINE_NUCLEOBASE_BIOSYNTHETIC_PROCESS\n",
      "GO_INNER_MITOCHONDRIAL_MEMBRANE_PROTEIN_COMPLEX\n",
      "GO_S100_PROTEIN_BINDING\n",
      "GO_POSITIVE_REGULATION_OF_VIRAL_PROCESS\n",
      "GO_SPLICEOSOMAL_COMPLEX\n",
      "GO_REGULATION_OF_TELOMERASE_RNA_LOCALIZATION_TO_CAJAL_BODY\n",
      "GO_REGULATION_OF_RNA_STABILITY\n",
      "GO_AMINO_ACID_ACTIVATION\n",
      "GO_REGULATION_OF_INTRINSIC_APOPTOTIC_SIGNALING_PATHWAY\n",
      "GO_NUCLEOSOMAL_DNA_BINDING\n",
      "GO_POSITIVE_REGULATION_OF_MITOTIC_NUCLEAR_DIVISION\n",
      "GO_PROTEIN_AUTOPROCESSING\n",
      "GO_RIBOSOMAL_SUBUNIT\n",
      "GO_TRANSLATION_INITIATION_FACTOR_ACTIVITY\n",
      "GO_VENTRICULAR_CARDIAC_MUSCLE_CELL_DEVELOPMENT\n",
      "GO_POLYAMINE_METABOLIC_PROCESS\n",
      "GO_BLASTOCYST_GROWTH\n",
      "GO_AMINO_ACID_TRANSMEMBRANE_TRANSPORT\n",
      "GO_RAB_GUANYL_NUCLEOTIDE_EXCHANGE_FACTOR_ACTIVITY\n",
      "GO_POTASSIUM_ION_TRANSPORT\n",
      "GO_ANCHORED_COMPONENT_OF_MEMBRANE\n",
      "GO_DNA_METHYLATION_INVOLVED_IN_GAMETE_GENERATION\n",
      "GO_POSITIVE_REGULATION_OF_AXON_EXTENSION\n",
      "GO_EMBRYONIC_SKELETAL_SYSTEM_MORPHOGENESIS\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_P_LVM1_4.csv_PAGE/filt_FC_P_LVM1_4.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_P_LVM1_4.csv_PAGE/filt_FC_P_LVM1_4.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_P_LVM1_4.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_P_LVM1_4.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 2, 4, 0.025511157602079\n",
      "best merge is 0, 11, 0.0280785732973895\n",
      "best merge is 1, 24, 0.0482674069810172\n",
      "best merge is 2, 3, 0.0577128535524087\n",
      "best merge is 2, 12, 0.0629545546499368\n",
      "best merge is 2, 6, 0.0796641610509358\n",
      "best merge is 2, 7, 0.127197035455499\n",
      "best merge is 1, 5, 0.135739447472175\n",
      "best merge is 0, 22, 0.17459426749547\n",
      "best merge is 1, 2, 0.191431876367012\n",
      "best merge is 0, 13, 0.192955096584292\n",
      "best merge is 1, 8, 0.255775497716702\n",
      "best merge is 14, 20, 0.260047793879526\n",
      "best merge is 1, 23, 0.328887259773629\n",
      "best merge is 16, 19, 0.34659283303849\n",
      "best merge is 10, 17, 0.415131800699739\n",
      "best merge is 18, 21, 0.463084557557279\n",
      "best merge is 0, 1, 0.523268193228559\n",
      "best merge is 10, 15, 0.711391542809042\n",
      "best merge is 0, 16, 0.774240717517063\n",
      "best merge is 10, 14, 0.871131857784171\n",
      "order classes\n",
      "17 - 0.59225078121824\n",
      "3 - 3.40354608577375\n",
      "2 - 1.62529871174704\n",
      "2 - 2.894278484733\n",
      "0 11 22 13 1 24 5 2 4 3 12 6 7 8 23 16 19\n",
      "9\n",
      "10 17 15\n",
      "14 20\n",
      "18 21\n",
      "Done.-19.9980945467743\t42\t0\n",
      "-19.9923781870974\t42\t0.00381090645132021\n",
      "-19.9847563741947\t42\t0.0114327193539606\n",
      "-19.9790400145177\t42\t0.019054532256601\n",
      "-19.9733236548408\t42\t0.0228654387079212\n",
      "-19.9676072951638\t42\t0.0304872516105617\n",
      "-19.8361310225932\t42\t0.0342981580618819\n",
      "-19.4912439887488\t42\t0.293439796751656\n",
      "1.36203611287542\t42\t0.724072225750839\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c5/fc_values/filt_FC_P_LVM1_4.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "sem --wait done\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "nohup sh run.sh iPAGE_JY.sh &"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
