{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# This is used for running data vs human_msigdb_c6 annotations located in: \n",
    "### /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c6"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import pickle\n",
    "import seaborn as sns\n",
    "import scipy.stats\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### get index of file names to loop through "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "cd fc_values\n",
    "ls FC* > filelist.txt\n",
    "cd .."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### get ensembl genes represented in iPAGE database"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash \n",
    "\n",
    "cp /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c6/human_ensembl_msigdb_c6_index.txt human_ensembl_index.txt\n",
    "awk '{print $1}' human_ensembl_index.txt > ensembl_index.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### make sure that genes in file match the index "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "###################!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n",
    "###################REMEMBER TO SPECIFY INDEX USED IN THE IPAGE COMMAND!!!!!!!!\n",
    "###################!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n",
    "\n",
    "\n",
    "#####################\n",
    "#######read in data from logfc calculations\n",
    "#####################\n",
    "####define index file to loop through\n",
    "index_file = \"fc_values/filelist.txt\"\n",
    "####define matching ensg index\n",
    "GO_index_filename = \"ensembl_index.txt\"\n",
    "genes_with_GOs = []\n",
    "with open(GO_index_filename,'r') as rf:\n",
    "    for line in rf:\n",
    "        gene_name = line.split('\\n')[0]\n",
    "        genes_with_GOs.append(gene_name)\n",
    "genes_with_GOs_set = set(genes_with_GOs)\n",
    "####open file and loop through each line\n",
    "###\n",
    "iPAGE_sh = \"iPAGE_JY.sh\"\n",
    "cwd = os.getcwd()+\"/\"\n",
    "os.system(\"rm \"+iPAGE_sh)\n",
    "###\n",
    "with open(index_file,'r') as rf:\n",
    "    for line in rf:\n",
    "        #####do this to each file\n",
    "        ###read in file into pandas df\n",
    "        FC_file = \"fc_values/\"+line.split('\\n')[0]\n",
    "        FC_df = pd.read_csv(FC_file,sep=',',index_col=0)\n",
    "        FC_df.columns = ['log2FoldChange']\n",
    "        FC_df.index.names = ['Gene']\n",
    "        ###filter\n",
    "        FC_df = FC_df.loc[[x for x in FC_df.index if x in genes_with_GOs_set]]\n",
    "        ###save\n",
    "        save_file = \"fc_values/filt_\"+line.split('\\n')[0]\n",
    "        FC_df.to_csv(save_file,sep='\\t',header=True)\n",
    "        ###write final iPAGE command to an ipage .sh file\n",
    "        iPAGE_cmd = \"perl /nvme/bins/iPAGEv1.0/page.pl --expfile \"+cwd+save_file+\" --species=human_ensembl_msigdb_c6 --exptype=continuous --ebins=9 \\n\"\n",
    "        with open(iPAGE_sh, \"a\") as myfile:\n",
    "            myfile.write(iPAGE_cmd)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## then run the command from this folder:\n",
    "\n",
    "nohup sh run.sh iPAGE_JY.sh &"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM1_1.csv --species=human_ensembl_msigdb_c6 --exptype=continuous --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM1_2.csv --species=human_ensembl_msigdb_c6 --exptype=continuous --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM1_3.csv --species=human_ensembl_msigdb_c6 --exptype=continuous --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM1_4.csv --species=human_ensembl_msigdb_c6 --exptype=continuous --ebins=9\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM1_1.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c6.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c6/human_ensembl_msigdb_c6.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM1_1.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 189\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t189\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 189\n",
      "cat 157, mi = 0.00315193, RB_P107_DN.V1_UP\n",
      "cat 73, mi = 0.00295861, HOXA9_DN.V1_DN\n",
      "cat 100, mi = 0.00276395, KRAS.KIDNEY_UP.V1_UP\n",
      "cat 108, mi = 0.00267493, LEF1_UP.V1_UP\n",
      "cat 86, mi = 0.00265578, KRAS.300_UP.V1_UP\n",
      "cat 102, mi = 0.00260756, KRAS.LUNG.BREAST_UP.V1_UP\n",
      "cat 21, mi = 0.00258864, BMI1_DN_MEL18_DN.V1_UP\n",
      "cat 46, mi = 0.00243240, DCA_UP.V1_UP\n",
      "cat 96, mi = 0.00232758, KRAS.BREAST_UP.V1_UP\n",
      "cat 26, mi = 0.00225891, CAHOY_NEURONAL\n",
      "cat 60, mi = 0.00224383, ESC_J1_UP_LATE.V1_UP\n",
      "cat 114, mi = 0.00223520, MEL18_DN.V1_UP\n",
      "cat 6, mi = 0.00200923, ATF2_S_UP.V1_DN\n",
      "cat 127, mi = 0.00196640, P53_DN.V1_UP\n",
      "cat 48, mi = 0.00194425, E2F1_UP.V1_UP\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 157, mi = 0.003, pvalue = 0.000, RB_P107_DN.V1_UP\n",
      "Passed: cat 100, mi = 0.003, pvalue = 0.000, KRAS.KIDNEY_UP.V1_UP\n",
      "Passed: cat 102, mi = 0.003, pvalue = 0.000, KRAS.LUNG.BREAST_UP.V1_UP\n",
      "Passed: cat 60, mi = 0.002, pvalue = 0.000, ESC_J1_UP_LATE.V1_UP\n",
      "Passed: cat 146, mi = 0.002, pvalue = 0.000, PTEN_DN.V1_DN\n",
      "Passed: cat 51, mi = 0.002, pvalue = 0.001, EGFR_UP.V1_DN\n",
      "Passed: cat 160, mi = 0.002, pvalue = 0.002, RELA_DN.V1_DN\n",
      "Passed: cat 136, mi = 0.002, pvalue = 0.003, PKCA_DN.V1_DN\n",
      "Passed: cat 112, mi = 0.001, pvalue = 0.002, MEK_UP.V1_UP\n",
      "Passed: cat 186, mi = 0.001, pvalue = 0.004, WNT_UP.V1_UP\n",
      "Passed: cat 29, mi = 0.001, pvalue = 0.004, CAMP_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 137, mi = 0.001, pvalue = 0.006, PKCA_DN.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 147, mi = 0.001, pvalue = 0.007, PTEN_DN.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 69, mi = 0.001, pvalue = 0.009, GLI1_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 85, mi = 0.001, pvalue = 0.012, KRAS.300_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 145, mi = 0.001, pvalue = 0.033, PRC2_SUZ12_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 35, mi = 0.001, pvalue = 0.039, CSR_EARLY_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 188, mi = 0.001, pvalue = 0.055, YAP1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 12, mi = 0.001, pvalue = 0.108, BCAT.100_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 164, mi = 0.001, pvalue = 0.097, SINGH_KRAS_DEPENDENCY_SIGNATURE_\n",
      "Didn't pass (cutoff=0.0050): cat 87, mi = 0.001, pvalue = 0.227, KRAS.50_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 170, mi = 0.001, pvalue = 0.138, SRC_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 14, mi = 0.001, pvalue = 0.224, BCAT_BILD_ET_AL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 16, mi = 0.001, pvalue = 0.242, BCAT_GDS748_DN\n",
      "Didn't pass (cutoff=0.0050): cat 15, mi = 0.001, pvalue = 0.269, BCAT_BILD_ET_AL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 133, mi = 0.000, pvalue = 0.494, PDGF_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 179, mi = 0.000, pvalue = 0.682, TBK1.DN.48HRS_DN\n",
      "Didn't pass: cat 89, mi = 0.000, pvalue = 1.000, KRAS.600.LUNG.BREAST_UP.V1_DN\n",
      "Didn't pass: cat 90, mi = 0.000, pvalue = 1.000, KRAS.600.LUNG.BREAST_UP.V1_UP\n",
      "Didn't pass: cat 91, mi = 0.000, pvalue = 1.000, KRAS.600_UP.V1_DN\n",
      "Didn't pass: cat 92, mi = 0.000, pvalue = 1.000, KRAS.600_UP.V1_UP\n",
      "Total number of categories evaluated = 170.\n",
      "Number of categories that passed the tests = 11\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM1_1.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM1_1.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 2, 6, 0.0704595512658449\n",
      "best merge is 1, 8, 0.104741657947894\n",
      "best merge is 4, 7, 0.157762988569799\n",
      "best merge is 1, 2, 0.195343529247353\n",
      "best merge is 4, 5, 0.19752666680688\n",
      "best merge is 1, 3, 0.243447727315615\n",
      "best merge is 0, 4, 0.298561182392196\n",
      "order classes\n",
      "5 - 0.339744457441948\n",
      "3 - 0.450442929186604\n",
      "1 8 2 6 3\n",
      "0\n",
      "4 7 5\n",
      "9\n",
      "10\n",
      "Done.Start drawing\n",
      "23.77\t0.00\n",
      "MEK_UP.V1_UP\n",
      "RB_P107_DN.V1_UP\n",
      "PTEN_DN.V1_DN\n",
      "PKCA_DN.V1_DN\n",
      "EGFR_UP.V1_DN\n",
      "CAMP_UP.V1_UP\n",
      "KRAS.KIDNEY_UP.V1_UP\n",
      "KRAS.LUNG.BREAST_UP.V1_UP\n",
      "RELA_DN.V1_DN\n",
      "ESC_J1_UP_LATE.V1_UP\n",
      "WNT_UP.V1_UP\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM1_1.csv_PAGE/filt_FC_LVM1_1.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM1_1.csv_PAGE/filt_FC_LVM1_1.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM1_1.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM1_1.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 2, 6, 0.0704595512658449\n",
      "best merge is 1, 8, 0.104741657947894\n",
      "best merge is 4, 7, 0.157762988569799\n",
      "best merge is 1, 2, 0.195343529247353\n",
      "best merge is 4, 5, 0.19752666680688\n",
      "best merge is 1, 3, 0.243447727315615\n",
      "best merge is 0, 4, 0.298561182392196\n",
      "order classes\n",
      "5 - 0.339744457441948\n",
      "3 - 0.450442929186604\n",
      "1 8 2 6 3\n",
      "0\n",
      "4 7 5\n",
      "9\n",
      "10\n",
      "Done.-19.2932267564156\t42\t0\n",
      "-18.4450988641144\t42\t1.4135464871687\n",
      "-18.2153975599495\t42\t1.69625578460244\n",
      "-18.056373580143\t42\t1.87294909549853\n",
      "-17.8973496003366\t42\t2.0143037442154\n",
      "-17.7206562894405\t42\t2.19099705511149\n",
      "-17.51745898191\t42\t2.36769036600757\n",
      "-17.2082456878418\t42\t2.59739167017249\n",
      "2.49305847707194\t42\t2.98611695414388\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM1_1.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM1_2.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c6.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c6/human_ensembl_msigdb_c6.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM1_2.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 189\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t189\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 189\n",
      "cat 102, mi = 0.00371761, KRAS.LUNG.BREAST_UP.V1_UP\n",
      "cat 30, mi = 0.00297067, CORDENONSI_YAP_CONSERVED_SIGNATURE\n",
      "cat 86, mi = 0.00278986, KRAS.300_UP.V1_UP\n",
      "cat 96, mi = 0.00277467, KRAS.BREAST_UP.V1_UP\n",
      "cat 127, mi = 0.00263525, P53_DN.V1_UP\n",
      "cat 99, mi = 0.00258838, KRAS.KIDNEY_UP.V1_DN\n",
      "cat 50, mi = 0.00253034, E2F3_UP.V1_UP\n",
      "cat 120, mi = 0.00247824, MYC_UP.V1_UP\n",
      "cat 82, mi = 0.00244092, JAK2_DN.V1_UP\n",
      "cat 108, mi = 0.00242111, LEF1_UP.V1_UP\n",
      "cat 26, mi = 0.00236075, CAHOY_NEURONAL\n",
      "cat 183, mi = 0.00232235, VEGF_A_UP.V1_DN\n",
      "cat 79, mi = 0.00225380, IL2_UP.V1_DN\n",
      "cat 104, mi = 0.00212378, KRAS.LUNG_UP.V1_UP\n",
      "cat 106, mi = 0.00208260, KRAS.PROSTATE_UP.V1_UP\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 102, mi = 0.004, pvalue = 0.000, KRAS.LUNG.BREAST_UP.V1_UP\n",
      "Passed: cat 30, mi = 0.003, pvalue = 0.000, CORDENONSI_YAP_CONSERVED_SIGNATURE\n",
      "Passed: cat 50, mi = 0.003, pvalue = 0.000, E2F3_UP.V1_UP\n",
      "Passed: cat 82, mi = 0.002, pvalue = 0.000, JAK2_DN.V1_UP\n",
      "Passed: cat 108, mi = 0.002, pvalue = 0.000, LEF1_UP.V1_UP\n",
      "Passed: cat 26, mi = 0.002, pvalue = 0.000, CAHOY_NEURONAL\n",
      "Passed: cat 183, mi = 0.002, pvalue = 0.000, VEGF_A_UP.V1_DN\n",
      "Passed: cat 79, mi = 0.002, pvalue = 0.000, IL2_UP.V1_DN\n",
      "Passed: cat 78, mi = 0.002, pvalue = 0.000, IL21_UP.V1_UP\n",
      "Passed: cat 164, mi = 0.002, pvalue = 0.000, SINGH_KRAS_DEPENDENCY_SIGNATURE_\n",
      "Passed: cat 105, mi = 0.002, pvalue = 0.001, KRAS.PROSTATE_UP.V1_DN\n",
      "Passed: cat 151, mi = 0.002, pvalue = 0.001, RAF_UP.V1_UP\n",
      "Passed: cat 136, mi = 0.002, pvalue = 0.002, PKCA_DN.V1_DN\n",
      "Passed: cat 41, mi = 0.001, pvalue = 0.002, CYCLIN_D1_KE_.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 166, mi = 0.001, pvalue = 0.006, SIRNA_EIF4GI_UP\n",
      "Didn't pass (cutoff=0.0050): cat 40, mi = 0.001, pvalue = 0.005, CTIP_DN.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 66, mi = 0.001, pvalue = 0.021, GCNP_SHH_UP_EARLY.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 62, mi = 0.001, pvalue = 0.049, ESC_V6.5_UP_EARLY.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 54, mi = 0.001, pvalue = 0.054, EIF4E_UP\n",
      "Didn't pass (cutoff=0.0050): cat 37, mi = 0.001, pvalue = 0.060, CSR_LATE_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 159, mi = 0.001, pvalue = 0.069, RB_P130_DN.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 81, mi = 0.001, pvalue = 0.088, JAK2_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 16, mi = 0.000, pvalue = 0.653, BCAT_GDS748_DN\n",
      "Didn't pass (cutoff=0.0050): cat 188, mi = 0.000, pvalue = 0.686, YAP1_UP\n",
      "Didn't pass: cat 89, mi = 0.000, pvalue = 1.000, KRAS.600.LUNG.BREAST_UP.V1_DN\n",
      "Didn't pass: cat 90, mi = 0.000, pvalue = 1.000, KRAS.600.LUNG.BREAST_UP.V1_UP\n",
      "Didn't pass: cat 91, mi = 0.000, pvalue = 1.000, KRAS.600_UP.V1_DN\n",
      "Didn't pass: cat 92, mi = 0.000, pvalue = 1.000, KRAS.600_UP.V1_UP\n",
      "Didn't pass: cat 121, mi = 0.000, pvalue = 1.000, NFE2L2.V2\n",
      "Didn't pass: cat 171, mi = 0.000, pvalue = 1.000, STK33_DN\n",
      "Didn't pass: cat 172, mi = 0.000, pvalue = 1.000, STK33_NOMO_DN\n",
      "Didn't pass: cat 173, mi = 0.000, pvalue = 1.000, STK33_NOMO_UP\n",
      "Didn't pass: cat 174, mi = 0.000, pvalue = 1.000, STK33_SKM_DN\n",
      "Didn't pass: cat 175, mi = 0.000, pvalue = 1.000, STK33_SKM_UP\n",
      "Total number of categories evaluated = 176.\n",
      "Number of categories that passed the tests = 14\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM1_2.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM1_2.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 0, 5, 0.0397062133017567\n",
      "best merge is 8, 9, 0.0738441639640562\n",
      "best merge is 0, 2, 0.0813159506421545\n",
      "best merge is 7, 12, 0.0851182393182212\n",
      "best merge is 0, 10, 0.0893322904213463\n",
      "best merge is 1, 6, 0.0983745629821988\n",
      "best merge is 0, 8, 0.129309804724733\n",
      "best merge is 3, 4, 0.148474711396084\n",
      "best merge is 3, 7, 0.187532868685188\n",
      "best merge is 3, 11, 0.205325637788487\n",
      "order classes\n",
      "6 - 0.204640798394151\n",
      "2 - 0.344310970437696\n",
      "4 - 0.430379436136588\n",
      "0 5 2 10 8 9\n",
      "1 6\n",
      "3 4 7 12\n",
      "11\n",
      "13\n",
      "Done.Start drawing\n",
      "62.96\t0.00\n",
      "CORDENONSI_YAP_CONSERVED_SIGNATURE\n",
      "VEGF_A_UP.V1_DN\n",
      "LEF1_UP.V1_UP\n",
      "PKCA_DN.V1_DN\n",
      "RAF_UP.V1_UP\n",
      "KRAS.LUNG.BREAST_UP.V1_UP\n",
      "CAHOY_NEURONAL\n",
      "E2F3_UP.V1_UP\n",
      "KRAS.PROSTATE_UP.V1_DN\n",
      "IL21_UP.V1_UP\n",
      "SINGH_KRAS_DEPENDENCY_SIGNATURE_\n",
      "JAK2_DN.V1_UP\n",
      "IL2_UP.V1_DN\n",
      "CYCLIN_D1_KE_.V1_DN\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM1_2.csv_PAGE/filt_FC_LVM1_2.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM1_2.csv_PAGE/filt_FC_LVM1_2.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM1_2.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM1_2.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 0, 5, 0.0397062133017567\n",
      "best merge is 8, 9, 0.0738441639640562\n",
      "best merge is 0, 2, 0.0813159506421545\n",
      "best merge is 7, 12, 0.0851182393182212\n",
      "best merge is 0, 10, 0.0893322904213463\n",
      "best merge is 1, 6, 0.0983745629821988\n",
      "best merge is 0, 8, 0.129309804724733\n",
      "best merge is 3, 4, 0.148474711396084\n",
      "best merge is 3, 7, 0.187532868685188\n",
      "best merge is 3, 11, 0.205325637788487\n",
      "order classes\n",
      "6 - 0.204640798394151\n",
      "2 - 0.344310970437696\n",
      "4 - 0.430379436136588\n",
      "0 5 2 10 8 9\n",
      "1 6\n",
      "3 4 7 12\n",
      "11\n",
      "13\n",
      "Done.-19.7198221092757\t42\t0\n",
      "-19.3562579415502\t42\t0.560355781448539\n",
      "-19.2195044472681\t42\t0.72712833545108\n",
      "-19.1227763659466\t42\t0.833862770012706\n",
      "-19.0360546378653\t42\t0.920584498094028\n",
      "-18.9459974587039\t42\t1.00730622617535\n",
      "-18.8359275730623\t42\t1.10069885641677\n",
      "-18.6558132147395\t42\t1.2274459974587\n",
      "1.73046378653113\t42\t1.46092757306226\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM1_2.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM1_3.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c6.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c6/human_ensembl_msigdb_c6.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM1_3.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 189\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t189\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 189\n",
      "cat 157, mi = 0.00490930, RB_P107_DN.V1_UP\n",
      "cat 143, mi = 0.00444803, PRC2_EZH2_UP.V1_UP\n",
      "cat 38, mi = 0.00317394, CSR_LATE_UP.V1_UP\n",
      "cat 100, mi = 0.00317317, KRAS.KIDNEY_UP.V1_UP\n",
      "cat 50, mi = 0.00303629, E2F3_UP.V1_UP\n",
      "cat 162, mi = 0.00290850, RPS14_DN.V1_DN\n",
      "cat 159, mi = 0.00274262, RB_P130_DN.V1_UP\n",
      "cat 86, mi = 0.00262925, KRAS.300_UP.V1_UP\n",
      "cat 107, mi = 0.00244873, LEF1_UP.V1_DN\n",
      "cat 21, mi = 0.00228699, BMI1_DN_MEL18_DN.V1_UP\n",
      "cat 37, mi = 0.00228501, CSR_LATE_UP.V1_DN\n",
      "cat 75, mi = 0.00227812, IL15_UP.V1_DN\n",
      "cat 127, mi = 0.00225316, P53_DN.V1_UP\n",
      "cat 108, mi = 0.00219253, LEF1_UP.V1_UP\n",
      "cat 126, mi = 0.00216552, P53_DN.V1_DN\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 157, mi = 0.005, pvalue = 0.000, RB_P107_DN.V1_UP\n",
      "Passed: cat 143, mi = 0.004, pvalue = 0.000, PRC2_EZH2_UP.V1_UP\n",
      "Passed: cat 100, mi = 0.003, pvalue = 0.000, KRAS.KIDNEY_UP.V1_UP\n",
      "Passed: cat 159, mi = 0.003, pvalue = 0.000, RB_P130_DN.V1_UP\n",
      "Passed: cat 21, mi = 0.002, pvalue = 0.000, BMI1_DN_MEL18_DN.V1_UP\n",
      "Passed: cat 37, mi = 0.002, pvalue = 0.000, CSR_LATE_UP.V1_DN\n",
      "Passed: cat 75, mi = 0.002, pvalue = 0.000, IL15_UP.V1_DN\n",
      "Passed: cat 146, mi = 0.002, pvalue = 0.000, PTEN_DN.V1_DN\n",
      "Passed: cat 137, mi = 0.002, pvalue = 0.000, PKCA_DN.V1_UP\n",
      "Passed: cat 30, mi = 0.002, pvalue = 0.000, CORDENONSI_YAP_CONSERVED_SIGNATURE\n",
      "Passed: cat 82, mi = 0.002, pvalue = 0.000, JAK2_DN.V1_UP\n",
      "Passed: cat 158, mi = 0.002, pvalue = 0.000, RB_P130_DN.V1_DN\n",
      "Passed: cat 101, mi = 0.002, pvalue = 0.001, KRAS.LUNG.BREAST_UP.V1_DN\n",
      "Passed: cat 25, mi = 0.001, pvalue = 0.003, CAHOY_ASTROGLIAL\n",
      "Passed: cat 29, mi = 0.001, pvalue = 0.004, CAMP_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 170, mi = 0.001, pvalue = 0.008, SRC_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 185, mi = 0.001, pvalue = 0.016, WNT_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 135, mi = 0.001, pvalue = 0.019, PIGF_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 129, mi = 0.001, pvalue = 0.078, P53_DN.V2_UP\n",
      "Didn't pass (cutoff=0.0050): cat 166, mi = 0.001, pvalue = 0.072, SIRNA_EIF4GI_UP\n",
      "Didn't pass (cutoff=0.0050): cat 10, mi = 0.001, pvalue = 0.143, ATM_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 64, mi = 0.001, pvalue = 0.134, ESC_V6.5_UP_LATE.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 33, mi = 0.001, pvalue = 0.166, CRX_NRL_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 120, mi = 0.001, pvalue = 0.174, MYC_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 81, mi = 0.001, pvalue = 0.316, JAK2_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 69, mi = 0.000, pvalue = 0.603, GLI1_UP.V1_DN\n",
      "Didn't pass: cat 89, mi = 0.000, pvalue = 1.000, KRAS.600.LUNG.BREAST_UP.V1_DN\n",
      "Didn't pass: cat 90, mi = 0.000, pvalue = 1.000, KRAS.600.LUNG.BREAST_UP.V1_UP\n",
      "Didn't pass: cat 91, mi = 0.000, pvalue = 1.000, KRAS.600_UP.V1_DN\n",
      "Didn't pass: cat 92, mi = 0.000, pvalue = 1.000, KRAS.600_UP.V1_UP\n",
      "Didn't pass: cat 121, mi = 0.000, pvalue = 1.000, NFE2L2.V2\n",
      "Didn't pass: cat 171, mi = 0.000, pvalue = 1.000, STK33_DN\n",
      "Didn't pass: cat 172, mi = 0.000, pvalue = 1.000, STK33_NOMO_DN\n",
      "Didn't pass: cat 173, mi = 0.000, pvalue = 1.000, STK33_NOMO_UP\n",
      "Didn't pass: cat 174, mi = 0.000, pvalue = 1.000, STK33_SKM_DN\n",
      "Total number of categories evaluated = 175.\n",
      "Number of categories that passed the tests = 15\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM1_3.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM1_3.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 6, 7, 0.0344745072653626\n",
      "best merge is 0, 1, 0.0380156741769754\n",
      "best merge is 6, 9, 0.0480330478678776\n",
      "best merge is 4, 10, 0.0751630167205578\n",
      "best merge is 4, 6, 0.138340658654408\n",
      "best merge is 0, 3, 0.167526702716793\n",
      "best merge is 2, 4, 0.183267157844805\n",
      "best merge is 2, 12, 0.249064087184627\n",
      "best merge is 2, 8, 0.317812705145878\n",
      "best merge is 11, 13, 0.359959873120458\n",
      "best merge is 0, 2, 0.579000008184961\n",
      "order classes\n",
      "3 - 0.41452119956729\n",
      "8 - 0.332294344921937\n",
      "2 - 1.34984952420172\n",
      "2 4 10 6 7 9 12 8\n",
      "0 1 3\n",
      "5\n",
      "11 13\n",
      "14\n",
      "Done.Start drawing\n",
      "84.94\t0.00\n",
      "BMI1_DN_MEL18_DN.V1_UP\n",
      "JAK2_DN.V1_UP\n",
      "IL15_UP.V1_DN\n",
      "PTEN_DN.V1_DN\n",
      "CORDENONSI_YAP_CONSERVED_SIGNATURE\n",
      "KRAS.LUNG.BREAST_UP.V1_DN\n",
      "CAHOY_ASTROGLIAL\n",
      "CSR_LATE_UP.V1_DN\n",
      "RB_P130_DN.V1_DN\n",
      "CAMP_UP.V1_UP\n",
      "KRAS.KIDNEY_UP.V1_UP\n",
      "PKCA_DN.V1_UP\n",
      "RB_P107_DN.V1_UP\n",
      "PRC2_EZH2_UP.V1_UP\n",
      "RB_P130_DN.V1_UP\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM1_3.csv_PAGE/filt_FC_LVM1_3.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM1_3.csv_PAGE/filt_FC_LVM1_3.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM1_3.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM1_3.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 6, 7, 0.0344745072653626\n",
      "best merge is 0, 1, 0.0380156741769754\n",
      "best merge is 6, 9, 0.0480330478678776\n",
      "best merge is 4, 10, 0.0751630167205578\n",
      "best merge is 4, 6, 0.138340658654408\n",
      "best merge is 0, 3, 0.167526702716793\n",
      "best merge is 2, 4, 0.183267157844805\n",
      "best merge is 2, 12, 0.249064087184627\n",
      "best merge is 2, 8, 0.317812705145878\n",
      "best merge is 11, 13, 0.359959873120458\n",
      "best merge is 0, 2, 0.579000008184961\n",
      "order classes\n",
      "3 - 0.41452119956729\n",
      "8 - 0.332294344921937\n",
      "2 - 1.34984952420172\n",
      "2 4 10 6 7 9 12 8\n",
      "0 1 3\n",
      "5\n",
      "11 13\n",
      "14\n",
      "Done.-19.849187661879\t42\t0\n",
      "-19.6588179891688\t42\t0.301624676242053\n",
      "-19.5920649870497\t42\t0.380739345420297\n",
      "-19.5401459854015\t42\t0.435130680480339\n",
      "-19.4931716505769\t42\t0.484577348716741\n",
      "-19.4437249823405\t42\t0.529079350129503\n",
      "-19.3769719802213\t42\t0.583470685189545\n",
      "-19.2706616435131\t42\t0.662585354367789\n",
      "1.39804567930304\t42\t0.796091358606075\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM1_3.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM2_1.csv --species=human_ensembl_msigdb_c6 --exptype=continuous --ebins=9\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM1_4.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c6.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c6/human_ensembl_msigdb_c6.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM1_4.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 189\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t189\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 189\n",
      "cat 30, mi = 0.00355253, CORDENONSI_YAP_CONSERVED_SIGNATURE\n",
      "cat 50, mi = 0.00342537, E2F3_UP.V1_UP\n",
      "cat 108, mi = 0.00314305, LEF1_UP.V1_UP\n",
      "cat 96, mi = 0.00278951, KRAS.BREAST_UP.V1_UP\n",
      "cat 102, mi = 0.00268136, KRAS.LUNG.BREAST_UP.V1_UP\n",
      "cat 100, mi = 0.00268097, KRAS.KIDNEY_UP.V1_UP\n",
      "cat 21, mi = 0.00266327, BMI1_DN_MEL18_DN.V1_UP\n",
      "cat 36, mi = 0.00250384, CSR_EARLY_UP.V1_UP\n",
      "cat 137, mi = 0.00249013, PKCA_DN.V1_UP\n",
      "cat 82, mi = 0.00241125, JAK2_DN.V1_UP\n",
      "cat 104, mi = 0.00226126, KRAS.LUNG_UP.V1_UP\n",
      "cat 77, mi = 0.00218496, IL21_UP.V1_DN\n",
      "cat 107, mi = 0.00213915, LEF1_UP.V1_DN\n",
      "cat 101, mi = 0.00208835, KRAS.LUNG.BREAST_UP.V1_DN\n",
      "cat 40, mi = 0.00208403, CTIP_DN.V1_UP\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 30, mi = 0.004, pvalue = 0.000, CORDENONSI_YAP_CONSERVED_SIGNATURE\n",
      "Passed: cat 50, mi = 0.003, pvalue = 0.000, E2F3_UP.V1_UP\n",
      "Passed: cat 108, mi = 0.003, pvalue = 0.000, LEF1_UP.V1_UP\n",
      "Passed: cat 96, mi = 0.003, pvalue = 0.000, KRAS.BREAST_UP.V1_UP\n",
      "Passed: cat 36, mi = 0.003, pvalue = 0.000, CSR_EARLY_UP.V1_UP\n",
      "Passed: cat 82, mi = 0.002, pvalue = 0.000, JAK2_DN.V1_UP\n",
      "Passed: cat 77, mi = 0.002, pvalue = 0.000, IL21_UP.V1_DN\n",
      "Passed: cat 107, mi = 0.002, pvalue = 0.000, LEF1_UP.V1_DN\n",
      "Passed: cat 40, mi = 0.002, pvalue = 0.000, CTIP_DN.V1_UP\n",
      "Passed: cat 78, mi = 0.002, pvalue = 0.000, IL21_UP.V1_UP\n",
      "Passed: cat 135, mi = 0.002, pvalue = 0.000, PIGF_UP.V1_UP\n",
      "Passed: cat 94, mi = 0.002, pvalue = 0.002, KRAS.AMP.LUNG_UP.V1_UP\n",
      "Passed: cat 158, mi = 0.002, pvalue = 0.002, RB_P130_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 164, mi = 0.001, pvalue = 0.009, SINGH_KRAS_DEPENDENCY_SIGNATURE_\n",
      "Didn't pass (cutoff=0.0050): cat 166, mi = 0.001, pvalue = 0.012, SIRNA_EIF4GI_UP\n",
      "Didn't pass (cutoff=0.0050): cat 186, mi = 0.001, pvalue = 0.019, WNT_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 159, mi = 0.001, pvalue = 0.029, RB_P130_DN.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 35, mi = 0.001, pvalue = 0.054, CSR_EARLY_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 1, mi = 0.001, pvalue = 0.071, AKT_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 42, mi = 0.001, pvalue = 0.083, CYCLIN_D1_KE_.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 68, mi = 0.001, pvalue = 0.096, GCNP_SHH_UP_LATE.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 131, mi = 0.001, pvalue = 0.127, PDGF_ERK_DN.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 179, mi = 0.001, pvalue = 0.341, TBK1.DN.48HRS_DN\n",
      "Didn't pass (cutoff=0.0050): cat 69, mi = 0.000, pvalue = 0.677, GLI1_UP.V1_DN\n",
      "Didn't pass: cat 89, mi = 0.000, pvalue = 1.000, KRAS.600.LUNG.BREAST_UP.V1_DN\n",
      "Didn't pass: cat 90, mi = 0.000, pvalue = 1.000, KRAS.600.LUNG.BREAST_UP.V1_UP\n",
      "Didn't pass: cat 91, mi = 0.000, pvalue = 1.000, KRAS.600_UP.V1_DN\n",
      "Didn't pass: cat 92, mi = 0.000, pvalue = 1.000, KRAS.600_UP.V1_UP\n",
      "Didn't pass: cat 121, mi = 0.000, pvalue = 1.000, NFE2L2.V2\n",
      "Didn't pass: cat 171, mi = 0.000, pvalue = 1.000, STK33_DN\n",
      "Didn't pass: cat 172, mi = 0.000, pvalue = 1.000, STK33_NOMO_DN\n",
      "Didn't pass: cat 173, mi = 0.000, pvalue = 1.000, STK33_NOMO_UP\n",
      "Didn't pass: cat 174, mi = 0.000, pvalue = 1.000, STK33_SKM_DN\n",
      "Total number of categories evaluated = 175.\n",
      "Number of categories that passed the tests = 13\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM1_4.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM1_4.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 3, 7, 0.0339428726887496\n",
      "best merge is 6, 8, 0.0447226164070905\n",
      "best merge is 3, 9, 0.0747358339417612\n",
      "best merge is 0, 2, 0.089087912069554\n",
      "best merge is 3, 6, 0.0973326310917161\n",
      "best merge is 1, 3, 0.101708890379258\n",
      "best merge is 4, 12, 0.14909640407335\n",
      "best merge is 1, 11, 0.150285958191277\n",
      "best merge is 1, 5, 0.273955264928581\n",
      "order classes\n",
      "2 - 0.289535714226051\n",
      "7 - 0.168461287926862\n",
      "2 - 0.484563313238389\n",
      "1 3 7 9 6 8 11\n",
      "0 2\n",
      "4 12\n",
      "5\n",
      "10\n",
      "Done.Start drawing\n",
      "26.31\t0.00\n",
      "CORDENONSI_YAP_CONSERVED_SIGNATURE\n",
      "LEF1_UP.V1_UP\n",
      "CSR_EARLY_UP.V1_UP\n",
      "RB_P130_DN.V1_DN\n",
      "PIGF_UP.V1_UP\n",
      "E2F3_UP.V1_UP\n",
      "KRAS.BREAST_UP.V1_UP\n",
      "LEF1_UP.V1_DN\n",
      "IL21_UP.V1_UP\n",
      "IL21_UP.V1_DN\n",
      "CTIP_DN.V1_UP\n",
      "KRAS.AMP.LUNG_UP.V1_UP\n",
      "JAK2_DN.V1_UP\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM1_4.csv_PAGE/filt_FC_LVM1_4.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM1_4.csv_PAGE/filt_FC_LVM1_4.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM1_4.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM1_4.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 3, 7, 0.0339428726887496\n",
      "best merge is 6, 8, 0.0447226164070905\n",
      "best merge is 3, 9, 0.0747358339417612\n",
      "best merge is 0, 2, 0.089087912069554\n",
      "best merge is 3, 6, 0.0973326310917161\n",
      "best merge is 1, 3, 0.101708890379258\n",
      "best merge is 4, 12, 0.14909640407335\n",
      "best merge is 1, 11, 0.150285958191277\n",
      "best merge is 1, 5, 0.273955264928581\n",
      "order classes\n",
      "2 - 0.289535714226051\n",
      "7 - 0.168461287926862\n",
      "2 - 0.484563313238389\n",
      "1 3 7 9 6 8 11\n",
      "0 2\n",
      "4 12\n",
      "5\n",
      "10\n",
      "Done.-19.5290763968073\t42\t0\n",
      "-18.9064994298746\t42\t0.941847206385405\n",
      "-18.6590649942987\t42\t1.24515393386545\n",
      "-18.4754846066135\t42\t1.43671607753706\n",
      "-18.3158494868871\t42\t1.61231470923603\n",
      "-18.1641961231471\t42\t1.75598631698974\n",
      "-17.9806157354618\t42\t1.91562143671608\n",
      "-17.709236031927\t42\t2.12314709236032\n",
      "2.22919042189282\t42\t2.45838084378563\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM1_4.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM2_2.csv --species=human_ensembl_msigdb_c6 --exptype=continuous --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM2_3.csv --species=human_ensembl_msigdb_c6 --exptype=continuous --ebins=9\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM2_1.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c6.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c6/human_ensembl_msigdb_c6.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM2_1.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 189\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t189\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 189\n",
      "cat 108, mi = 0.00461702, LEF1_UP.V1_UP\n",
      "cat 127, mi = 0.00391918, P53_DN.V1_UP\n",
      "cat 96, mi = 0.00350198, KRAS.BREAST_UP.V1_UP\n",
      "cat 102, mi = 0.00299807, KRAS.LUNG.BREAST_UP.V1_UP\n",
      "cat 86, mi = 0.00294437, KRAS.300_UP.V1_UP\n",
      "cat 126, mi = 0.00282807, P53_DN.V1_DN\n",
      "cat 144, mi = 0.00265056, PRC2_SUZ12_UP.V1_DN\n",
      "cat 107, mi = 0.00264757, LEF1_UP.V1_DN\n",
      "cat 105, mi = 0.00262897, KRAS.PROSTATE_UP.V1_DN\n",
      "cat 26, mi = 0.00248621, CAHOY_NEURONAL\n",
      "cat 147, mi = 0.00242479, PTEN_DN.V1_UP\n",
      "cat 100, mi = 0.00237654, KRAS.KIDNEY_UP.V1_UP\n",
      "cat 112, mi = 0.00234891, MEK_UP.V1_UP\n",
      "cat 37, mi = 0.00234506, CSR_LATE_UP.V1_DN\n",
      "cat 30, mi = 0.00227517, CORDENONSI_YAP_CONSERVED_SIGNATURE\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 108, mi = 0.005, pvalue = 0.000, LEF1_UP.V1_UP\n",
      "Passed: cat 127, mi = 0.004, pvalue = 0.000, P53_DN.V1_UP\n",
      "Passed: cat 96, mi = 0.004, pvalue = 0.000, KRAS.BREAST_UP.V1_UP\n",
      "Passed: cat 107, mi = 0.003, pvalue = 0.000, LEF1_UP.V1_DN\n",
      "Passed: cat 105, mi = 0.003, pvalue = 0.000, KRAS.PROSTATE_UP.V1_DN\n",
      "Passed: cat 26, mi = 0.002, pvalue = 0.000, CAHOY_NEURONAL\n",
      "Passed: cat 37, mi = 0.002, pvalue = 0.000, CSR_LATE_UP.V1_DN\n",
      "Passed: cat 186, mi = 0.002, pvalue = 0.000, WNT_UP.V1_UP\n",
      "Passed: cat 84, mi = 0.002, pvalue = 0.000, JNK_DN.V1_UP\n",
      "Passed: cat 36, mi = 0.002, pvalue = 0.000, CSR_EARLY_UP.V1_UP\n",
      "Passed: cat 48, mi = 0.002, pvalue = 0.000, E2F1_UP.V1_UP\n",
      "Passed: cat 82, mi = 0.002, pvalue = 0.001, JAK2_DN.V1_UP\n",
      "Passed: cat 29, mi = 0.002, pvalue = 0.000, CAMP_UP.V1_UP\n",
      "Passed: cat 27, mi = 0.002, pvalue = 0.002, CAHOY_OLIGODENDROCUTIC\n",
      "Passed: cat 164, mi = 0.001, pvalue = 0.002, SINGH_KRAS_DEPENDENCY_SIGNATURE_\n",
      "Passed: cat 135, mi = 0.001, pvalue = 0.003, PIGF_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 139, mi = 0.001, pvalue = 0.008, PRC1_BMI_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 75, mi = 0.001, pvalue = 0.019, IL15_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 119, mi = 0.001, pvalue = 0.024, MYC_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 16, mi = 0.001, pvalue = 0.074, BCAT_GDS748_DN\n",
      "Didn't pass (cutoff=0.0050): cat 165, mi = 0.001, pvalue = 0.086, SIRNA_EIF4GI_DN\n",
      "Didn't pass (cutoff=0.0050): cat 17, mi = 0.001, pvalue = 0.281, BCAT_GDS748_UP\n",
      "Didn't pass (cutoff=0.0050): cat 69, mi = 0.001, pvalue = 0.415, GLI1_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 24, mi = 0.001, pvalue = 0.386, CAHOY_ASTROCYTIC\n",
      "Didn't pass (cutoff=0.0050): cat 188, mi = 0.000, pvalue = 0.503, YAP1_UP\n",
      "Didn't pass: cat 89, mi = 0.000, pvalue = 1.000, KRAS.600.LUNG.BREAST_UP.V1_DN\n",
      "Didn't pass: cat 90, mi = 0.000, pvalue = 1.000, KRAS.600.LUNG.BREAST_UP.V1_UP\n",
      "Didn't pass: cat 91, mi = 0.000, pvalue = 1.000, KRAS.600_UP.V1_DN\n",
      "Didn't pass: cat 92, mi = 0.000, pvalue = 1.000, KRAS.600_UP.V1_UP\n",
      "Didn't pass: cat 121, mi = 0.000, pvalue = 1.000, NFE2L2.V2\n",
      "Didn't pass: cat 171, mi = 0.000, pvalue = 1.000, STK33_DN\n",
      "Didn't pass: cat 172, mi = 0.000, pvalue = 1.000, STK33_NOMO_DN\n",
      "Didn't pass: cat 173, mi = 0.000, pvalue = 1.000, STK33_NOMO_UP\n",
      "Didn't pass: cat 174, mi = 0.000, pvalue = 1.000, STK33_SKM_DN\n",
      "Didn't pass: cat 175, mi = 0.000, pvalue = 1.000, STK33_SKM_UP\n",
      "Didn't pass: cat 176, mi = 0.000, pvalue = 1.000, STK33_UP\n",
      "Total number of categories evaluated = 177.\n",
      "Number of categories that passed the tests = 16\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM2_1.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM2_1.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 2, 5, 0.0311976325066279\n",
      "best merge is 1, 3, 0.0818160975913228\n",
      "best merge is 2, 14, 0.0866643969772886\n",
      "best merge is 0, 6, 0.125316752486166\n",
      "best merge is 0, 8, 0.156109755825041\n",
      "best merge is 12, 15, 0.156487207079626\n",
      "best merge is 2, 7, 0.168233658123948\n",
      "best merge is 1, 2, 0.173083705430741\n",
      "best merge is 1, 11, 0.239547404047051\n",
      "best merge is 1, 4, 0.278677192929968\n",
      "best merge is 0, 1, 0.53828069729998\n",
      "best merge is 10, 13, 0.557955042569039\n",
      "order classes\n",
      "11 - 0.454876295501082\n",
      "2 - 0.625948828318503\n",
      "0 6 8 1 3 2 5 14 7 11 4\n",
      "9\n",
      "10\n",
      "12 15\n",
      "13\n",
      "Done.Start drawing\n",
      "70.09\t0.00\n",
      "LEF1_UP.V1_UP\n",
      "CSR_LATE_UP.V1_DN\n",
      "JNK_DN.V1_UP\n",
      "CSR_EARLY_UP.V1_UP\n",
      "E2F1_UP.V1_UP\n",
      "CAMP_UP.V1_UP\n",
      "PIGF_UP.V1_UP\n",
      "P53_DN.V1_UP\n",
      "LEF1_UP.V1_DN\n",
      "KRAS.BREAST_UP.V1_UP\n",
      "CAHOY_NEURONAL\n",
      "SINGH_KRAS_DEPENDENCY_SIGNATURE_\n",
      "WNT_UP.V1_UP\n",
      "JAK2_DN.V1_UP\n",
      "KRAS.PROSTATE_UP.V1_DN\n",
      "CAHOY_OLIGODENDROCUTIC\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM2_1.csv_PAGE/filt_FC_LVM2_1.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM2_1.csv_PAGE/filt_FC_LVM2_1.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM2_1.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM2_1.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 2, 5, 0.0311976325066279\n",
      "best merge is 1, 3, 0.0818160975913228\n",
      "best merge is 2, 14, 0.0866643969772886\n",
      "best merge is 0, 6, 0.125316752486166\n",
      "best merge is 0, 8, 0.156109755825041\n",
      "best merge is 12, 15, 0.156487207079626\n",
      "best merge is 2, 7, 0.168233658123948\n",
      "best merge is 1, 2, 0.173083705430741\n",
      "best merge is 1, 11, 0.239547404047051\n",
      "best merge is 1, 4, 0.278677192929968\n",
      "best merge is 0, 1, 0.53828069729998\n",
      "best merge is 10, 13, 0.557955042569039\n",
      "order classes\n",
      "11 - 0.454876295501082\n",
      "2 - 0.625948828318503\n",
      "0 6 8 1 3 2 5 14 7 11 4\n",
      "9\n",
      "10\n",
      "12 15\n",
      "13\n",
      "Done.-19.784277357683\t42\t0\n",
      "-19.5056356113568\t42\t0.431445284634042\n",
      "-19.4037665858182\t42\t0.557283492652304\n",
      "-19.3318590383792\t42\t0.635183335711228\n",
      "-19.2689399343701\t42\t0.701098587530318\n",
      "-19.203024682551\t42\t0.761021543729491\n",
      "-19.1221286916821\t42\t0.832929091168498\n",
      "-18.9902981880439\t42\t0.922813525467256\n",
      "1.54829504922243\t42\t1.09659009844486\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM2_1.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM2_2.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c6.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c6/human_ensembl_msigdb_c6.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM2_2.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 189\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t189\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 189\n",
      "cat 102, mi = 0.00389917, KRAS.LUNG.BREAST_UP.V1_UP\n",
      "cat 96, mi = 0.00381153, KRAS.BREAST_UP.V1_UP\n",
      "cat 26, mi = 0.00362337, CAHOY_NEURONAL\n",
      "cat 86, mi = 0.00336032, KRAS.300_UP.V1_UP\n",
      "cat 127, mi = 0.00330014, P53_DN.V1_UP\n",
      "cat 78, mi = 0.00299947, IL21_UP.V1_UP\n",
      "cat 167, mi = 0.00254462, SNF5_DN.V1_DN\n",
      "cat 108, mi = 0.00252478, LEF1_UP.V1_UP\n",
      "cat 100, mi = 0.00240055, KRAS.KIDNEY_UP.V1_UP\n",
      "cat 99, mi = 0.00224882, KRAS.KIDNEY_UP.V1_DN\n",
      "cat 48, mi = 0.00223542, E2F1_UP.V1_UP\n",
      "cat 13, mi = 0.00223503, BCAT.100_UP.V1_UP\n",
      "cat 30, mi = 0.00217806, CORDENONSI_YAP_CONSERVED_SIGNATURE\n",
      "cat 38, mi = 0.00214812, CSR_LATE_UP.V1_UP\n",
      "cat 73, mi = 0.00203216, HOXA9_DN.V1_DN\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 102, mi = 0.004, pvalue = 0.000, KRAS.LUNG.BREAST_UP.V1_UP\n",
      "Passed: cat 26, mi = 0.004, pvalue = 0.000, CAHOY_NEURONAL\n",
      "Passed: cat 127, mi = 0.003, pvalue = 0.000, P53_DN.V1_UP\n",
      "Passed: cat 78, mi = 0.003, pvalue = 0.000, IL21_UP.V1_UP\n",
      "Passed: cat 167, mi = 0.003, pvalue = 0.000, SNF5_DN.V1_DN\n",
      "Passed: cat 48, mi = 0.002, pvalue = 0.000, E2F1_UP.V1_UP\n",
      "Passed: cat 38, mi = 0.002, pvalue = 0.000, CSR_LATE_UP.V1_UP\n",
      "Passed: cat 73, mi = 0.002, pvalue = 0.000, HOXA9_DN.V1_DN\n",
      "Passed: cat 11, mi = 0.002, pvalue = 0.000, ATM_DN.V1_UP\n",
      "Passed: cat 82, mi = 0.002, pvalue = 0.000, JAK2_DN.V1_UP\n",
      "Passed: cat 105, mi = 0.002, pvalue = 0.000, KRAS.PROSTATE_UP.V1_DN\n",
      "Passed: cat 166, mi = 0.002, pvalue = 0.001, SIRNA_EIF4GI_UP\n",
      "Passed: cat 137, mi = 0.002, pvalue = 0.001, PKCA_DN.V1_UP\n",
      "Passed: cat 164, mi = 0.002, pvalue = 0.001, SINGH_KRAS_DEPENDENCY_SIGNATURE_\n",
      "Passed: cat 58, mi = 0.002, pvalue = 0.001, ESC_J1_UP_EARLY.V1_UP\n",
      "Passed: cat 186, mi = 0.002, pvalue = 0.001, WNT_UP.V1_UP\n",
      "Passed: cat 169, mi = 0.002, pvalue = 0.002, SRC_UP.V1_DN\n",
      "Passed: cat 3, mi = 0.002, pvalue = 0.002, AKT_UP_MTOR_DN.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 79, mi = 0.001, pvalue = 0.008, IL2_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 115, mi = 0.001, pvalue = 0.008, MTOR_UP.N4.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 69, mi = 0.001, pvalue = 0.059, GLI1_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 54, mi = 0.001, pvalue = 0.043, EIF4E_UP\n",
      "Didn't pass (cutoff=0.0050): cat 141, mi = 0.001, pvalue = 0.044, PRC2_EED_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 12, mi = 0.001, pvalue = 0.065, BCAT.100_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 133, mi = 0.001, pvalue = 0.093, PDGF_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 158, mi = 0.000, pvalue = 0.571, RB_P130_DN.V1_DN\n",
      "Didn't pass: cat 89, mi = 0.000, pvalue = 1.000, KRAS.600.LUNG.BREAST_UP.V1_DN\n",
      "Didn't pass: cat 90, mi = 0.000, pvalue = 1.000, KRAS.600.LUNG.BREAST_UP.V1_UP\n",
      "Didn't pass: cat 91, mi = 0.000, pvalue = 1.000, KRAS.600_UP.V1_DN\n",
      "Didn't pass: cat 92, mi = 0.000, pvalue = 1.000, KRAS.600_UP.V1_UP\n",
      "Didn't pass: cat 121, mi = 0.000, pvalue = 1.000, NFE2L2.V2\n",
      "Didn't pass: cat 171, mi = 0.000, pvalue = 1.000, STK33_DN\n",
      "Didn't pass: cat 172, mi = 0.000, pvalue = 1.000, STK33_NOMO_DN\n",
      "Didn't pass: cat 173, mi = 0.000, pvalue = 1.000, STK33_NOMO_UP\n",
      "Didn't pass: cat 174, mi = 0.000, pvalue = 1.000, STK33_SKM_DN\n",
      "Didn't pass: cat 175, mi = 0.000, pvalue = 1.000, STK33_SKM_UP\n",
      "Didn't pass: cat 176, mi = 0.000, pvalue = 1.000, STK33_UP\n",
      "Didn't pass: cat 177, mi = 0.000, pvalue = 1.000, TBK1.DF_DN\n",
      "Total number of categories evaluated = 178.\n",
      "Number of categories that passed the tests = 18\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM2_2.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM2_2.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 0, 1, 0.0349387060966658\n",
      "best merge is 0, 2, 0.0521366426977247\n",
      "best merge is 5, 6, 0.0775857372534496\n",
      "best merge is 4, 10, 0.0925645282055061\n",
      "best merge is 0, 13, 0.103656216135144\n",
      "best merge is 3, 12, 0.138729914195406\n",
      "best merge is 0, 4, 0.159801646800685\n",
      "best merge is 15, 16, 0.167221455073516\n",
      "best merge is 0, 8, 0.172535220054668\n",
      "best merge is 9, 15, 0.195561000277211\n",
      "best merge is 0, 3, 0.207081833896088\n",
      "best merge is 14, 17, 0.207600065947134\n",
      "best merge is 5, 7, 0.262074904998062\n",
      "best merge is 0, 9, 0.298012688626501\n",
      "order classes\n",
      "9 - 0.291073839584058\n",
      "3 - 0.802314062999431\n",
      "3 - 0.744457940837251\n",
      "2 - 0.934200296762104\n",
      "0 1 2 13 4 10 8 3 12\n",
      "5 6 7\n",
      "9 15 16\n",
      "11\n",
      "14 17\n",
      "Done.Start drawing\n",
      "101.03\t0.00\n",
      "E2F1_UP.V1_UP\n",
      "CSR_LATE_UP.V1_UP\n",
      "HOXA9_DN.V1_DN\n",
      "SRC_UP.V1_DN\n",
      "SIRNA_EIF4GI_UP\n",
      "KRAS.LUNG.BREAST_UP.V1_UP\n",
      "CAHOY_NEURONAL\n",
      "P53_DN.V1_UP\n",
      "SINGH_KRAS_DEPENDENCY_SIGNATURE_\n",
      "SNF5_DN.V1_DN\n",
      "KRAS.PROSTATE_UP.V1_DN\n",
      "ATM_DN.V1_UP\n",
      "IL21_UP.V1_UP\n",
      "PKCA_DN.V1_UP\n",
      "JAK2_DN.V1_UP\n",
      "WNT_UP.V1_UP\n",
      "ESC_J1_UP_EARLY.V1_UP\n",
      "AKT_UP_MTOR_DN.V1_UP\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM2_2.csv_PAGE/filt_FC_LVM2_2.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM2_2.csv_PAGE/filt_FC_LVM2_2.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM2_2.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM2_2.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 0, 1, 0.0349387060966658\n",
      "best merge is 0, 2, 0.0521366426977247\n",
      "best merge is 5, 6, 0.0775857372534496\n",
      "best merge is 4, 10, 0.0925645282055061\n",
      "best merge is 0, 13, 0.103656216135144\n",
      "best merge is 3, 12, 0.138729914195406\n",
      "best merge is 0, 4, 0.159801646800685\n",
      "best merge is 15, 16, 0.167221455073516\n",
      "best merge is 0, 8, 0.172535220054668\n",
      "best merge is 9, 15, 0.195561000277211\n",
      "best merge is 0, 3, 0.207081833896088\n",
      "best merge is 14, 17, 0.207600065947134\n",
      "best merge is 5, 7, 0.262074904998062\n",
      "best merge is 0, 9, 0.298012688626501\n",
      "order classes\n",
      "9 - 0.291073839584058\n",
      "3 - 0.802314062999431\n",
      "3 - 0.744457940837251\n",
      "2 - 0.934200296762104\n",
      "0 1 2 13 4 10 8 3 12\n",
      "5 6 7\n",
      "9 15 16\n",
      "11\n",
      "14 17\n",
      "Done.-19.8544986637632\t42\t0\n",
      "-19.6570325645848\t42\t0.291002672473523\n",
      "-19.5759675343957\t42\t0.394932198356924\n",
      "-19.5219241809364\t42\t0.453132732851628\n",
      "-19.4720380085123\t42\t0.503018905275661\n",
      "-19.417994655053\t42\t0.552905077699693\n",
      "-19.3514797584876\t42\t0.611105612194398\n",
      "-19.2454716420865\t42\t0.685934870830446\n",
      "1.41156092249827\t42\t0.823121844996536\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM2_2.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM2_4.csv --species=human_ensembl_msigdb_c6 --exptype=continuous --ebins=9\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM2_3.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c6.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c6/human_ensembl_msigdb_c6.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM2_3.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 189\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t189\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 189\n",
      "cat 30, mi = 0.00364248, CORDENONSI_YAP_CONSERVED_SIGNATURE\n",
      "cat 102, mi = 0.00331661, KRAS.LUNG.BREAST_UP.V1_UP\n",
      "cat 96, mi = 0.00326663, KRAS.BREAST_UP.V1_UP\n",
      "cat 183, mi = 0.00298534, VEGF_A_UP.V1_DN\n",
      "cat 120, mi = 0.00297453, MYC_UP.V1_UP\n",
      "cat 26, mi = 0.00288842, CAHOY_NEURONAL\n",
      "cat 86, mi = 0.00282530, KRAS.300_UP.V1_UP\n",
      "cat 108, mi = 0.00271769, LEF1_UP.V1_UP\n",
      "cat 104, mi = 0.00253566, KRAS.LUNG_UP.V1_UP\n",
      "cat 21, mi = 0.00246810, BMI1_DN_MEL18_DN.V1_UP\n",
      "cat 50, mi = 0.00241584, E2F3_UP.V1_UP\n",
      "cat 64, mi = 0.00223017, ESC_V6.5_UP_LATE.V1_UP\n",
      "cat 127, mi = 0.00222430, P53_DN.V1_UP\n",
      "cat 29, mi = 0.00219076, CAMP_UP.V1_UP\n",
      "cat 126, mi = 0.00216277, P53_DN.V1_DN\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 30, mi = 0.004, pvalue = 0.000, CORDENONSI_YAP_CONSERVED_SIGNATURE\n",
      "Passed: cat 102, mi = 0.003, pvalue = 0.000, KRAS.LUNG.BREAST_UP.V1_UP\n",
      "Passed: cat 183, mi = 0.003, pvalue = 0.000, VEGF_A_UP.V1_DN\n",
      "Passed: cat 120, mi = 0.003, pvalue = 0.000, MYC_UP.V1_UP\n",
      "Passed: cat 26, mi = 0.003, pvalue = 0.000, CAHOY_NEURONAL\n",
      "Passed: cat 108, mi = 0.003, pvalue = 0.000, LEF1_UP.V1_UP\n",
      "Passed: cat 64, mi = 0.002, pvalue = 0.000, ESC_V6.5_UP_LATE.V1_UP\n",
      "Passed: cat 29, mi = 0.002, pvalue = 0.000, CAMP_UP.V1_UP\n",
      "Passed: cat 105, mi = 0.002, pvalue = 0.000, KRAS.PROSTATE_UP.V1_DN\n",
      "Passed: cat 82, mi = 0.002, pvalue = 0.000, JAK2_DN.V1_UP\n",
      "Passed: cat 78, mi = 0.002, pvalue = 0.000, IL21_UP.V1_UP\n",
      "Passed: cat 132, mi = 0.002, pvalue = 0.001, PDGF_UP.V1_DN\n",
      "Passed: cat 145, mi = 0.001, pvalue = 0.004, PRC2_SUZ12_UP.V1_UP\n",
      "Passed: cat 164, mi = 0.001, pvalue = 0.002, SINGH_KRAS_DEPENDENCY_SIGNATURE_\n",
      "Didn't pass (cutoff=0.0050): cat 161, mi = 0.001, pvalue = 0.015, RELA_DN.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 111, mi = 0.001, pvalue = 0.019, MEK_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 75, mi = 0.001, pvalue = 0.022, IL15_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 109, mi = 0.001, pvalue = 0.024, LTE2_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 151, mi = 0.001, pvalue = 0.050, RAF_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 166, mi = 0.001, pvalue = 0.063, SIRNA_EIF4GI_UP\n",
      "Didn't pass (cutoff=0.0050): cat 188, mi = 0.001, pvalue = 0.082, YAP1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 156, mi = 0.001, pvalue = 0.087, RB_P107_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 118, mi = 0.001, pvalue = 0.117, MTOR_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 81, mi = 0.001, pvalue = 0.129, JAK2_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 12, mi = 0.001, pvalue = 0.397, BCAT.100_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 73, mi = 0.001, pvalue = 0.301, HOXA9_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 17, mi = 0.001, pvalue = 0.383, BCAT_GDS748_UP\n",
      "Didn't pass (cutoff=0.0050): cat 16, mi = 0.000, pvalue = 0.546, BCAT_GDS748_DN\n",
      "Didn't pass: cat 89, mi = 0.000, pvalue = 1.000, KRAS.600.LUNG.BREAST_UP.V1_DN\n",
      "Didn't pass: cat 90, mi = 0.000, pvalue = 1.000, KRAS.600.LUNG.BREAST_UP.V1_UP\n",
      "Didn't pass: cat 91, mi = 0.000, pvalue = 1.000, KRAS.600_UP.V1_DN\n",
      "Didn't pass: cat 92, mi = 0.000, pvalue = 1.000, KRAS.600_UP.V1_UP\n",
      "Didn't pass: cat 121, mi = 0.000, pvalue = 1.000, NFE2L2.V2\n",
      "Didn't pass: cat 171, mi = 0.000, pvalue = 1.000, STK33_DN\n",
      "Total number of categories evaluated = 172.\n",
      "Number of categories that passed the tests = 14\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM2_3.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM2_3.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 4, 10, 0.0310350236355267\n",
      "best merge is 1, 8, 0.0410378100399061\n",
      "best merge is 4, 13, 0.0474550186266127\n",
      "best merge is 0, 2, 0.0477037534260947\n",
      "best merge is 1, 9, 0.0723456076054774\n",
      "best merge is 5, 6, 0.0993813808660136\n",
      "best merge is 1, 4, 0.0996667321327186\n",
      "best merge is 1, 12, 0.181011611156492\n",
      "best merge is 5, 11, 0.182358117422295\n",
      "best merge is 1, 5, 0.388977980254074\n",
      "order classes\n",
      "2 - 0.166963136991331\n",
      "7 - 0.187326068756982\n",
      "3 - 0.481286416292477\n",
      "0 2\n",
      "1 8 9 4 10 13 12\n",
      "3\n",
      "5 6 11\n",
      "7\n",
      "Done.Start drawing\n",
      "69.14\t0.00\n",
      "CORDENONSI_YAP_CONSERVED_SIGNATURE\n",
      "VEGF_A_UP.V1_DN\n",
      "LEF1_UP.V1_UP\n",
      "ESC_V6.5_UP_LATE.V1_UP\n",
      "PDGF_UP.V1_DN\n",
      "CAMP_UP.V1_UP\n",
      "MYC_UP.V1_UP\n",
      "KRAS.LUNG.BREAST_UP.V1_UP\n",
      "KRAS.PROSTATE_UP.V1_DN\n",
      "JAK2_DN.V1_UP\n",
      "CAHOY_NEURONAL\n",
      "IL21_UP.V1_UP\n",
      "SINGH_KRAS_DEPENDENCY_SIGNATURE_\n",
      "PRC2_SUZ12_UP.V1_UP\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM2_3.csv_PAGE/filt_FC_LVM2_3.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM2_3.csv_PAGE/filt_FC_LVM2_3.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM2_3.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM2_3.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 4, 10, 0.0310350236355267\n",
      "best merge is 1, 8, 0.0410378100399061\n",
      "best merge is 4, 13, 0.0474550186266127\n",
      "best merge is 0, 2, 0.0477037534260947\n",
      "best merge is 1, 9, 0.0723456076054774\n",
      "best merge is 5, 6, 0.0993813808660136\n",
      "best merge is 1, 4, 0.0996667321327186\n",
      "best merge is 1, 12, 0.181011611156492\n",
      "best merge is 5, 11, 0.182358117422295\n",
      "best merge is 1, 5, 0.388977980254074\n",
      "order classes\n",
      "2 - 0.166963136991331\n",
      "7 - 0.187326068756982\n",
      "3 - 0.481286416292477\n",
      "0 2\n",
      "1 8 9 4 10 13 12\n",
      "3\n",
      "5 6 11\n",
      "7\n",
      "Done.-19.7843505929997\t42\t0\n",
      "-19.4715070870697\t42\t0.431298814000579\n",
      "-19.3166039919005\t42\t0.625687011859994\n",
      "-19.2133352617877\t42\t0.741105004339022\n",
      "-19.1191784784495\t42\t0.832224472085623\n",
      "-19.0189470639283\t42\t0.929418571015331\n",
      "-18.8974544402661\t42\t1.03268730112815\n",
      "-18.7000289268152\t42\t1.1724038183396\n",
      "1.71376916401504\t42\t1.42753832803008\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM2_3.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM2_5.csv --species=human_ensembl_msigdb_c6 --exptype=continuous --ebins=9\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM2_4.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c6.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c6/human_ensembl_msigdb_c6.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM2_4.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 189\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t189\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 189\n",
      "cat 143, mi = 0.00527508, PRC2_EZH2_UP.V1_UP\n",
      "cat 157, mi = 0.00490019, RB_P107_DN.V1_UP\n",
      "cat 50, mi = 0.00475068, E2F3_UP.V1_UP\n",
      "cat 162, mi = 0.00389718, RPS14_DN.V1_DN\n",
      "cat 38, mi = 0.00347621, CSR_LATE_UP.V1_UP\n",
      "cat 108, mi = 0.00315112, LEF1_UP.V1_UP\n",
      "cat 159, mi = 0.00266158, RB_P130_DN.V1_UP\n",
      "cat 86, mi = 0.00258291, KRAS.300_UP.V1_UP\n",
      "cat 75, mi = 0.00226018, IL15_UP.V1_DN\n",
      "cat 107, mi = 0.00221004, LEF1_UP.V1_DN\n",
      "cat 126, mi = 0.00218404, P53_DN.V1_DN\n",
      "cat 61, mi = 0.00204300, ESC_V6.5_UP_EARLY.V1_DN\n",
      "cat 60, mi = 0.00200454, ESC_J1_UP_LATE.V1_UP\n",
      "cat 96, mi = 0.00197622, KRAS.BREAST_UP.V1_UP\n",
      "cat 102, mi = 0.00186457, KRAS.LUNG.BREAST_UP.V1_UP\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 143, mi = 0.005, pvalue = 0.000, PRC2_EZH2_UP.V1_UP\n",
      "Passed: cat 157, mi = 0.005, pvalue = 0.000, RB_P107_DN.V1_UP\n",
      "Passed: cat 108, mi = 0.003, pvalue = 0.000, LEF1_UP.V1_UP\n",
      "Passed: cat 159, mi = 0.003, pvalue = 0.000, RB_P130_DN.V1_UP\n",
      "Passed: cat 75, mi = 0.002, pvalue = 0.000, IL15_UP.V1_DN\n",
      "Passed: cat 107, mi = 0.002, pvalue = 0.000, LEF1_UP.V1_DN\n",
      "Passed: cat 26, mi = 0.002, pvalue = 0.001, CAHOY_NEURONAL\n",
      "Passed: cat 186, mi = 0.002, pvalue = 0.002, WNT_UP.V1_UP\n",
      "Passed: cat 120, mi = 0.002, pvalue = 0.001, MYC_UP.V1_UP\n",
      "Passed: cat 58, mi = 0.002, pvalue = 0.001, ESC_J1_UP_EARLY.V1_UP\n",
      "Passed: cat 135, mi = 0.001, pvalue = 0.004, PIGF_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 99, mi = 0.001, pvalue = 0.008, KRAS.KIDNEY_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 158, mi = 0.001, pvalue = 0.006, RB_P130_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 82, mi = 0.001, pvalue = 0.012, JAK2_DN.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 119, mi = 0.001, pvalue = 0.014, MYC_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 10, mi = 0.001, pvalue = 0.022, ATM_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 84, mi = 0.001, pvalue = 0.018, JNK_DN.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 46, mi = 0.001, pvalue = 0.028, DCA_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 105, mi = 0.001, pvalue = 0.036, KRAS.PROSTATE_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 40, mi = 0.001, pvalue = 0.048, CTIP_DN.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 166, mi = 0.001, pvalue = 0.044, SIRNA_EIF4GI_UP\n",
      "Didn't pass (cutoff=0.0050): cat 151, mi = 0.001, pvalue = 0.049, RAF_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 152, mi = 0.001, pvalue = 0.057, RAPA_EARLY_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 17, mi = 0.001, pvalue = 0.085, BCAT_GDS748_UP\n",
      "Didn't pass (cutoff=0.0050): cat 164, mi = 0.001, pvalue = 0.092, SINGH_KRAS_DEPENDENCY_SIGNATURE_\n",
      "Didn't pass (cutoff=0.0050): cat 37, mi = 0.001, pvalue = 0.075, CSR_LATE_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 36, mi = 0.001, pvalue = 0.078, CSR_EARLY_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 136, mi = 0.001, pvalue = 0.098, PKCA_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 165, mi = 0.001, pvalue = 0.319, SIRNA_EIF4GI_DN\n",
      "Didn't pass (cutoff=0.0050): cat 24, mi = 0.001, pvalue = 0.349, CAHOY_ASTROCYTIC\n",
      "Didn't pass (cutoff=0.0050): cat 187, mi = 0.000, pvalue = 0.485, YAP1_DN\n",
      "Total number of categories evaluated = 138.\n",
      "Number of categories that passed the tests = 11\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM2_4.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM2_4.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 2, 4, 0.0136293734619334\n",
      "best merge is 0, 6, 0.0145141540208065\n",
      "best merge is 0, 1, 0.0285078296710344\n",
      "best merge is 2, 9, 0.0757731341175766\n",
      "best merge is 0, 3, 0.0839570841239415\n",
      "best merge is 0, 7, 0.136843956621631\n",
      "best merge is 2, 5, 0.242685247742107\n",
      "order classes\n",
      "5 - 0.153256733030935\n",
      "3 - 0.13458755990133\n",
      "2 4 9\n",
      "0 6 1 3 7\n",
      "5\n",
      "8\n",
      "10\n",
      "Done.Start drawing\n",
      "100.30\t0.00\n",
      "LEF1_UP.V1_UP\n",
      "IL15_UP.V1_DN\n",
      "ESC_J1_UP_EARLY.V1_UP\n",
      "LEF1_UP.V1_DN\n",
      "PIGF_UP.V1_UP\n",
      "MYC_UP.V1_UP\n",
      "PRC2_EZH2_UP.V1_UP\n",
      "CAHOY_NEURONAL\n",
      "RB_P107_DN.V1_UP\n",
      "RB_P130_DN.V1_UP\n",
      "WNT_UP.V1_UP\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM2_4.csv_PAGE/filt_FC_LVM2_4.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM2_4.csv_PAGE/filt_FC_LVM2_4.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM2_4.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM2_4.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 2, 4, 0.0136293734619334\n",
      "best merge is 0, 6, 0.0145141540208065\n",
      "best merge is 0, 1, 0.0285078296710344\n",
      "best merge is 2, 9, 0.0757731341175766\n",
      "best merge is 0, 3, 0.0839570841239415\n",
      "best merge is 0, 7, 0.136843956621631\n",
      "best merge is 2, 5, 0.242685247742107\n",
      "order classes\n",
      "5 - 0.153256733030935\n",
      "3 - 0.13458755990133\n",
      "2 4 9\n",
      "0 6 1 3 7\n",
      "5\n",
      "8\n",
      "10\n",
      "Done.-19.8911266201396\t42\t0\n",
      "-19.7403788634098\t42\t0.217746759720837\n",
      "-19.6712861415753\t42\t0.301495513459621\n",
      "-19.621036889332\t42\t0.355932203389831\n",
      "-19.5728813559322\t42\t0.401994017946162\n",
      "-19.5163509471585\t42\t0.452243270189432\n",
      "-19.4451645064806\t42\t0.515054835493519\n",
      "-19.3321036889332\t42\t0.594616151545364\n",
      "1.37058823529411\t42\t0.741176470588235\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM2_4.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_P_LVM1_1.csv --species=human_ensembl_msigdb_c6 --exptype=continuous --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_P_LVM1_2.csv --species=human_ensembl_msigdb_c6 --exptype=continuous --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_P_LVM1_3.csv --species=human_ensembl_msigdb_c6 --exptype=continuous --ebins=9\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_P_LVM1_1.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c6.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c6/human_ensembl_msigdb_c6.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_P_LVM1_1.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 189\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t189\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 189\n",
      "cat 38, mi = 0.00208316, CSR_LATE_UP.V1_UP\n",
      "cat 137, mi = 0.00206819, PKCA_DN.V1_UP\n",
      "cat 79, mi = 0.00175636, IL2_UP.V1_DN\n",
      "cat 149, mi = 0.00167958, PTEN_DN.V2_UP\n",
      "cat 166, mi = 0.00167696, SIRNA_EIF4GI_UP\n",
      "cat 52, mi = 0.00142260, EGFR_UP.V1_UP\n",
      "cat 19, mi = 0.00142132, BMI1_DN.V1_UP\n",
      "cat 99, mi = 0.00135751, KRAS.KIDNEY_UP.V1_DN\n",
      "cat 96, mi = 0.00135456, KRAS.BREAST_UP.V1_UP\n",
      "cat 39, mi = 0.00132555, CTIP_DN.V1_DN\n",
      "cat 139, mi = 0.00132493, PRC1_BMI_UP.V1_UP\n",
      "cat 105, mi = 0.00128193, KRAS.PROSTATE_UP.V1_DN\n",
      "cat 107, mi = 0.00127661, LEF1_UP.V1_DN\n",
      "cat 49, mi = 0.00126599, E2F3_UP.V1_DN\n",
      "cat 102, mi = 0.00126276, KRAS.LUNG.BREAST_UP.V1_UP\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 38, mi = 0.002, pvalue = 0.000, CSR_LATE_UP.V1_UP\n",
      "Passed: cat 137, mi = 0.002, pvalue = 0.000, PKCA_DN.V1_UP\n",
      "Passed: cat 79, mi = 0.002, pvalue = 0.001, IL2_UP.V1_DN\n",
      "Passed: cat 149, mi = 0.002, pvalue = 0.001, PTEN_DN.V2_UP\n",
      "Passed: cat 166, mi = 0.002, pvalue = 0.000, SIRNA_EIF4GI_UP\n",
      "Passed: cat 19, mi = 0.001, pvalue = 0.003, BMI1_DN.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 99, mi = 0.001, pvalue = 0.006, KRAS.KIDNEY_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 39, mi = 0.001, pvalue = 0.006, CTIP_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 105, mi = 0.001, pvalue = 0.010, KRAS.PROSTATE_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 85, mi = 0.001, pvalue = 0.013, KRAS.300_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 26, mi = 0.001, pvalue = 0.017, CAHOY_NEURONAL\n",
      "Didn't pass (cutoff=0.0050): cat 48, mi = 0.001, pvalue = 0.010, E2F1_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 29, mi = 0.001, pvalue = 0.013, CAMP_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 27, mi = 0.001, pvalue = 0.020, CAHOY_OLIGODENDROCUTIC\n",
      "Didn't pass (cutoff=0.0050): cat 158, mi = 0.001, pvalue = 0.018, RB_P130_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 62, mi = 0.001, pvalue = 0.017, ESC_V6.5_UP_EARLY.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 45, mi = 0.001, pvalue = 0.020, DCA_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 8, mi = 0.001, pvalue = 0.027, ATF2_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 30, mi = 0.001, pvalue = 0.036, CORDENONSI_YAP_CONSERVED_SIGNATURE\n",
      "Didn't pass (cutoff=0.0050): cat 82, mi = 0.001, pvalue = 0.030, JAK2_DN.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 136, mi = 0.001, pvalue = 0.049, PKCA_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 84, mi = 0.001, pvalue = 0.049, JNK_DN.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 87, mi = 0.001, pvalue = 0.086, KRAS.50_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 123, mi = 0.001, pvalue = 0.066, NOTCH_DN.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 71, mi = 0.001, pvalue = 0.103, HINATA_NFKB_IMMU_INF\n",
      "Didn't pass (cutoff=0.0050): cat 95, mi = 0.001, pvalue = 0.090, KRAS.BREAST_UP.V1_DN\n",
      "Total number of categories evaluated = 42.\n",
      "Number of categories that passed the tests = 6\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_P_LVM1_1.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_P_LVM1_1.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 0, 4, 0.207896965439562\n",
      "best merge is 2, 5, 0.36368854715102\n",
      "order classes\n",
      "2 - 0.311845448159343\n",
      "0 4\n",
      "1\n",
      "2\n",
      "3\n",
      "5\n",
      "Done.Start drawing\n",
      "280.55\t0.00\n",
      "SIRNA_EIF4GI_UP\n",
      "CSR_LATE_UP.V1_UP\n",
      "PKCA_DN.V1_UP\n",
      "IL2_UP.V1_DN\n",
      "BMI1_DN.V1_UP\n",
      "PTEN_DN.V2_UP\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_P_LVM1_1.csv_PAGE/filt_FC_P_LVM1_1.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_P_LVM1_1.csv_PAGE/filt_FC_P_LVM1_1.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_P_LVM1_1.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_P_LVM1_1.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 0, 4, 0.207896965439562\n",
      "best merge is 2, 5, 0.36368854715102\n",
      "order classes\n",
      "2 - 0.311845448159343\n",
      "0 4\n",
      "1\n",
      "2\n",
      "3\n",
      "5\n",
      "Done.-19.9985029406523\t42\t0\n",
      "-19.9962573516307\t42\t0.00299411869541971\n",
      "-19.9940117626092\t42\t0.00449117804312957\n",
      "-19.9917661735876\t42\t0.00748529673854928\n",
      "-19.989520584566\t42\t0.00898235608625913\n",
      "-19.9872749955445\t42\t0.0119764747816788\n",
      "-19.9169132062021\t42\t0.0134735341293887\n",
      "-19.7559793263233\t42\t0.152700053466405\n",
      "1.1676706469435\t42\t0.335341293887008\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_P_LVM1_1.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM2_5.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c6.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c6/human_ensembl_msigdb_c6.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM2_5.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 189\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t189\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 189\n",
      "cat 96, mi = 0.00383480, KRAS.BREAST_UP.V1_UP\n",
      "cat 108, mi = 0.00300209, LEF1_UP.V1_UP\n",
      "cat 100, mi = 0.00287222, KRAS.KIDNEY_UP.V1_UP\n",
      "cat 102, mi = 0.00286581, KRAS.LUNG.BREAST_UP.V1_UP\n",
      "cat 107, mi = 0.00285418, LEF1_UP.V1_DN\n",
      "cat 86, mi = 0.00282140, KRAS.300_UP.V1_UP\n",
      "cat 50, mi = 0.00278609, E2F3_UP.V1_UP\n",
      "cat 104, mi = 0.00274825, KRAS.LUNG_UP.V1_UP\n",
      "cat 85, mi = 0.00259658, KRAS.300_UP.V1_DN\n",
      "cat 183, mi = 0.00252486, VEGF_A_UP.V1_DN\n",
      "cat 127, mi = 0.00251717, P53_DN.V1_UP\n",
      "cat 77, mi = 0.00249134, IL21_UP.V1_DN\n",
      "cat 29, mi = 0.00246986, CAMP_UP.V1_UP\n",
      "cat 75, mi = 0.00238973, IL15_UP.V1_DN\n",
      "cat 21, mi = 0.00238862, BMI1_DN_MEL18_DN.V1_UP\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 96, mi = 0.004, pvalue = 0.000, KRAS.BREAST_UP.V1_UP\n",
      "Passed: cat 108, mi = 0.003, pvalue = 0.000, LEF1_UP.V1_UP\n",
      "Passed: cat 107, mi = 0.003, pvalue = 0.000, LEF1_UP.V1_DN\n",
      "Passed: cat 50, mi = 0.003, pvalue = 0.000, E2F3_UP.V1_UP\n",
      "Passed: cat 183, mi = 0.003, pvalue = 0.000, VEGF_A_UP.V1_DN\n",
      "Passed: cat 77, mi = 0.002, pvalue = 0.000, IL21_UP.V1_DN\n",
      "Passed: cat 29, mi = 0.002, pvalue = 0.000, CAMP_UP.V1_UP\n",
      "Passed: cat 36, mi = 0.002, pvalue = 0.000, CSR_EARLY_UP.V1_UP\n",
      "Passed: cat 30, mi = 0.002, pvalue = 0.000, CORDENONSI_YAP_CONSERVED_SIGNATURE\n",
      "Passed: cat 78, mi = 0.002, pvalue = 0.001, IL21_UP.V1_UP\n",
      "Passed: cat 84, mi = 0.002, pvalue = 0.001, JNK_DN.V1_UP\n",
      "Passed: cat 58, mi = 0.002, pvalue = 0.001, ESC_J1_UP_EARLY.V1_UP\n",
      "Passed: cat 25, mi = 0.001, pvalue = 0.003, CAHOY_ASTROGLIAL\n",
      "Passed: cat 82, mi = 0.001, pvalue = 0.004, JAK2_DN.V1_UP\n",
      "Passed: cat 158, mi = 0.001, pvalue = 0.003, RB_P130_DN.V1_DN\n",
      "Passed: cat 164, mi = 0.001, pvalue = 0.003, SINGH_KRAS_DEPENDENCY_SIGNATURE_\n",
      "Didn't pass (cutoff=0.0050): cat 145, mi = 0.001, pvalue = 0.005, PRC2_SUZ12_UP.V1_UP\n",
      "Passed: cat 166, mi = 0.001, pvalue = 0.004, SIRNA_EIF4GI_UP\n",
      "Didn't pass (cutoff=0.0050): cat 46, mi = 0.001, pvalue = 0.005, DCA_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 40, mi = 0.001, pvalue = 0.010, CTIP_DN.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 159, mi = 0.001, pvalue = 0.018, RB_P130_DN.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 35, mi = 0.001, pvalue = 0.027, CSR_EARLY_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 81, mi = 0.001, pvalue = 0.128, JAK2_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 24, mi = 0.001, pvalue = 0.159, CAHOY_ASTROCYTIC\n",
      "Didn't pass (cutoff=0.0050): cat 131, mi = 0.001, pvalue = 0.208, PDGF_ERK_DN.V1_UP\n",
      "Didn't pass: cat 89, mi = 0.000, pvalue = 1.000, KRAS.600.LUNG.BREAST_UP.V1_DN\n",
      "Didn't pass: cat 90, mi = 0.000, pvalue = 1.000, KRAS.600.LUNG.BREAST_UP.V1_UP\n",
      "Didn't pass: cat 91, mi = 0.000, pvalue = 1.000, KRAS.600_UP.V1_DN\n",
      "Didn't pass: cat 92, mi = 0.000, pvalue = 1.000, KRAS.600_UP.V1_UP\n",
      "Didn't pass: cat 121, mi = 0.000, pvalue = 1.000, NFE2L2.V2\n",
      "Didn't pass: cat 171, mi = 0.000, pvalue = 1.000, STK33_DN\n",
      "Didn't pass: cat 172, mi = 0.000, pvalue = 1.000, STK33_NOMO_DN\n",
      "Didn't pass: cat 173, mi = 0.000, pvalue = 1.000, STK33_NOMO_UP\n",
      "Didn't pass: cat 174, mi = 0.000, pvalue = 1.000, STK33_SKM_DN\n",
      "Didn't pass: cat 175, mi = 0.000, pvalue = 1.000, STK33_SKM_UP\n",
      "Didn't pass: cat 176, mi = 0.000, pvalue = 1.000, STK33_UP\n",
      "Didn't pass: cat 177, mi = 0.000, pvalue = 1.000, TBK1.DF_DN\n",
      "Didn't pass: cat 178, mi = 0.000, pvalue = 1.000, TBK1.DF_UP\n",
      "Number of categories that passed the tests = 17\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM2_5.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM2_5.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 0, 5, 0.029710106291887\n",
      "best merge is 0, 13, 0.052223774113437\n",
      "best merge is 0, 9, 0.0586167810081439\n",
      "best merge is 0, 15, 0.0763119125293667\n",
      "best merge is 0, 2, 0.0839113465096851\n",
      "best merge is 1, 12, 0.0860181853761297\n",
      "best merge is 3, 11, 0.111765183061765\n",
      "best merge is 1, 8, 0.184239573540686\n",
      "best merge is 3, 10, 0.202172409619377\n",
      "best merge is 4, 7, 0.301079728781471\n",
      "best merge is 0, 3, 0.309285773510184\n",
      "best merge is 4, 14, 0.423737618109998\n",
      "best merge is 4, 6, 0.529706592142452\n",
      "order classes\n",
      "9 - 0.331981144572783\n",
      "3 - 0.572329974205743\n",
      "3 - 1.44632847444629\n",
      "0 5 13 9 15 2 3 11 10\n",
      "1 12 8\n",
      "4 7 14\n",
      "6\n",
      "16\n",
      "Done.Start drawing\n",
      "66.19\t0.00\n",
      "LEF1_UP.V1_UP\n",
      "CAHOY_ASTROGLIAL\n",
      "CORDENONSI_YAP_CONSERVED_SIGNATURE\n",
      "CSR_EARLY_UP.V1_UP\n",
      "RB_P130_DN.V1_DN\n",
      "VEGF_A_UP.V1_DN\n",
      "SIRNA_EIF4GI_UP\n",
      "CAMP_UP.V1_UP\n",
      "KRAS.BREAST_UP.V1_UP\n",
      "IL21_UP.V1_DN\n",
      "JAK2_DN.V1_UP\n",
      "IL21_UP.V1_UP\n",
      "SINGH_KRAS_DEPENDENCY_SIGNATURE_\n",
      "LEF1_UP.V1_DN\n",
      "E2F3_UP.V1_UP\n",
      "ESC_J1_UP_EARLY.V1_UP\n",
      "JNK_DN.V1_UP\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM2_5.csv_PAGE/filt_FC_LVM2_5.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM2_5.csv_PAGE/filt_FC_LVM2_5.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM2_5.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM2_5.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 0, 5, 0.029710106291887\n",
      "best merge is 0, 13, 0.052223774113437\n",
      "best merge is 0, 9, 0.0586167810081439\n",
      "best merge is 0, 15, 0.0763119125293667\n",
      "best merge is 0, 2, 0.0839113465096851\n",
      "best merge is 1, 12, 0.0860181853761297\n",
      "best merge is 3, 11, 0.111765183061765\n",
      "best merge is 1, 8, 0.184239573540686\n",
      "best merge is 3, 10, 0.202172409619377\n",
      "best merge is 4, 7, 0.301079728781471\n",
      "best merge is 0, 3, 0.309285773510184\n",
      "best merge is 4, 14, 0.423737618109998\n",
      "best merge is 4, 6, 0.529706592142452\n",
      "order classes\n",
      "9 - 0.331981144572783\n",
      "3 - 0.572329974205743\n",
      "3 - 1.44632847444629\n",
      "0 5 13 9 15 2 3 11 10\n",
      "1 12 8\n",
      "4 7 14\n",
      "6\n",
      "16\n",
      "Done.-19.8477111346125\t42\t0\n",
      "-19.6224505212268\t42\t0.304577730775042\n",
      "-19.501888502795\t42\t0.450521226771416\n",
      "-19.4130533313189\t42\t0.545701767638616\n",
      "-19.330563529234\t42\t0.628191569723523\n",
      "-19.2480737271491\t42\t0.71068137180843\n",
      "-19.1528931862819\t42\t0.793171173893337\n",
      "-19.0196404290678\t42\t0.901042453542831\n",
      "1.52983834416075\t42\t1.0596766883215\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_LVM2_5.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_P_LVM1_2.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c6.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c6/human_ensembl_msigdb_c6.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_P_LVM1_2.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 189\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t189\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 189\n",
      "cat 120, mi = 0.00419676, MYC_UP.V1_UP\n",
      "cat 162, mi = 0.00371419, RPS14_DN.V1_DN\n",
      "cat 78, mi = 0.00224398, IL21_UP.V1_UP\n",
      "cat 85, mi = 0.00216488, KRAS.300_UP.V1_DN\n",
      "cat 26, mi = 0.00215715, CAHOY_NEURONAL\n",
      "cat 30, mi = 0.00213688, CORDENONSI_YAP_CONSERVED_SIGNATURE\n",
      "cat 148, mi = 0.00213346, PTEN_DN.V2_DN\n",
      "cat 96, mi = 0.00208053, KRAS.BREAST_UP.V1_UP\n",
      "cat 182, mi = 0.00195152, TGFB_UP.V1_UP\n",
      "cat 21, mi = 0.00194975, BMI1_DN_MEL18_DN.V1_UP\n",
      "cat 50, mi = 0.00182248, E2F3_UP.V1_UP\n",
      "cat 95, mi = 0.00180820, KRAS.BREAST_UP.V1_DN\n",
      "cat 157, mi = 0.00172247, RB_P107_DN.V1_UP\n",
      "cat 63, mi = 0.00165889, ESC_V6.5_UP_LATE.V1_DN\n",
      "cat 183, mi = 0.00164652, VEGF_A_UP.V1_DN\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 120, mi = 0.004, pvalue = 0.000, MYC_UP.V1_UP\n",
      "Passed: cat 78, mi = 0.002, pvalue = 0.000, IL21_UP.V1_UP\n",
      "Passed: cat 85, mi = 0.002, pvalue = 0.000, KRAS.300_UP.V1_DN\n",
      "Passed: cat 26, mi = 0.002, pvalue = 0.000, CAHOY_NEURONAL\n",
      "Passed: cat 30, mi = 0.002, pvalue = 0.000, CORDENONSI_YAP_CONSERVED_SIGNATURE\n",
      "Passed: cat 148, mi = 0.002, pvalue = 0.000, PTEN_DN.V2_DN\n",
      "Passed: cat 96, mi = 0.002, pvalue = 0.000, KRAS.BREAST_UP.V1_UP\n",
      "Passed: cat 157, mi = 0.002, pvalue = 0.000, RB_P107_DN.V1_UP\n",
      "Passed: cat 183, mi = 0.002, pvalue = 0.001, VEGF_A_UP.V1_DN\n",
      "Passed: cat 75, mi = 0.002, pvalue = 0.000, IL15_UP.V1_DN\n",
      "Passed: cat 137, mi = 0.002, pvalue = 0.002, PKCA_DN.V1_UP\n",
      "Passed: cat 29, mi = 0.002, pvalue = 0.001, CAMP_UP.V1_UP\n",
      "Passed: cat 128, mi = 0.001, pvalue = 0.003, P53_DN.V2_DN\n",
      "Didn't pass (cutoff=0.0050): cat 40, mi = 0.001, pvalue = 0.006, CTIP_DN.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 2, mi = 0.001, pvalue = 0.006, AKT_UP_MTOR_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 166, mi = 0.001, pvalue = 0.010, SIRNA_EIF4GI_UP\n",
      "Didn't pass (cutoff=0.0050): cat 186, mi = 0.001, pvalue = 0.014, WNT_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 84, mi = 0.001, pvalue = 0.019, JNK_DN.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 116, mi = 0.001, pvalue = 0.016, MTOR_UP.N4.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 158, mi = 0.001, pvalue = 0.020, RB_P130_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 154, mi = 0.001, pvalue = 0.025, RB_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 44, mi = 0.001, pvalue = 0.034, CYCLIN_D1_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 156, mi = 0.001, pvalue = 0.038, RB_P107_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 42, mi = 0.001, pvalue = 0.047, CYCLIN_D1_KE_.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 7, mi = 0.001, pvalue = 0.061, ATF2_S_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 36, mi = 0.001, pvalue = 0.060, CSR_EARLY_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 187, mi = 0.001, pvalue = 0.131, YAP1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 69, mi = 0.001, pvalue = 0.149, GLI1_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 82, mi = 0.001, pvalue = 0.185, JAK2_DN.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 16, mi = 0.001, pvalue = 0.255, BCAT_GDS748_DN\n",
      "Didn't pass (cutoff=0.0050): cat 24, mi = 0.001, pvalue = 0.359, CAHOY_ASTROCYTIC\n",
      "Didn't pass (cutoff=0.0050): cat 81, mi = 0.001, pvalue = 0.387, JAK2_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 188, mi = 0.001, pvalue = 0.439, YAP1_UP\n",
      "Total number of categories evaluated = 130.\n",
      "Number of categories that passed the tests = 13\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_P_LVM1_2.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_P_LVM1_2.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 6, 10, 0.02350654023237\n",
      "best merge is 4, 8, 0.085581919848409\n",
      "best merge is 3, 6, 0.13722207559154\n",
      "best merge is 5, 7, 0.144531904797778\n",
      "best merge is 1, 2, 0.145733683916453\n",
      "best merge is 1, 3, 0.204360616960497\n",
      "best merge is 1, 9, 0.270461707412182\n",
      "best merge is 5, 12, 0.326339736841585\n",
      "best merge is 0, 1, 0.461587821027129\n",
      "order classes\n",
      "6 - 0.363778110963198\n",
      "2 - 0.278141239507329\n",
      "3 - 0.7676850311298\n",
      "4 8\n",
      "1 2 3 6 10 9\n",
      "0\n",
      "5 7 12\n",
      "11\n",
      "Done.Start drawing\n",
      "5.29\t0.00\n",
      "CORDENONSI_YAP_CONSERVED_SIGNATURE\n",
      "VEGF_A_UP.V1_DN\n",
      "PTEN_DN.V2_DN\n",
      "RB_P107_DN.V1_UP\n",
      "P53_DN.V2_DN\n",
      "CAMP_UP.V1_UP\n",
      "MYC_UP.V1_UP\n",
      "IL21_UP.V1_UP\n",
      "KRAS.300_UP.V1_DN\n",
      "CAHOY_NEURONAL\n",
      "KRAS.BREAST_UP.V1_UP\n",
      "PKCA_DN.V1_UP\n",
      "IL15_UP.V1_DN\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_P_LVM1_2.csv_PAGE/filt_FC_P_LVM1_2.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_P_LVM1_2.csv_PAGE/filt_FC_P_LVM1_2.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_P_LVM1_2.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_P_LVM1_2.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 6, 10, 0.02350654023237\n",
      "best merge is 4, 8, 0.085581919848409\n",
      "best merge is 3, 6, 0.13722207559154\n",
      "best merge is 5, 7, 0.144531904797778\n",
      "best merge is 1, 2, 0.145733683916453\n",
      "best merge is 1, 3, 0.204360616960497\n",
      "best merge is 1, 9, 0.270461707412182\n",
      "best merge is 5, 12, 0.326339736841585\n",
      "best merge is 0, 1, 0.461587821027129\n",
      "order classes\n",
      "6 - 0.363778110963198\n",
      "2 - 0.278141239507329\n",
      "3 - 0.7676850311298\n",
      "4 8\n",
      "1 2 3 6 10 9\n",
      "0\n",
      "5 7 12\n",
      "11\n",
      "Done.-16.8638941398866\t42\t0\n",
      "-13.2117202268431\t42\t6.27221172022684\n",
      "-12.3780718336484\t42\t7.30434782608696\n",
      "-11.8223062381853\t42\t7.93950850661626\n",
      "-11.3856332703214\t42\t8.41587901701323\n",
      "-10.9092627599244\t42\t8.81285444234405\n",
      "-10.3137996219282\t42\t9.36862003780718\n",
      "-9.44045368620038\t42\t10.0037807183365\n",
      "6.55765595463138\t42\t11.1153119092628\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_P_LVM1_2.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_P_LVM1_4.csv --species=human_ensembl_msigdb_c6 --exptype=continuous --ebins=9\n",
      "done\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_P_LVM1_3.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c6.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c6/human_ensembl_msigdb_c6.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_P_LVM1_3.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 189\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t189\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 189\n",
      "cat 157, mi = 0.00922043, RB_P107_DN.V1_UP\n",
      "cat 38, mi = 0.00769883, CSR_LATE_UP.V1_UP\n",
      "cat 143, mi = 0.00540645, PRC2_EZH2_UP.V1_UP\n",
      "cat 48, mi = 0.00471631, E2F1_UP.V1_UP\n",
      "cat 162, mi = 0.00378741, RPS14_DN.V1_DN\n",
      "cat 50, mi = 0.00312479, E2F3_UP.V1_UP\n",
      "cat 107, mi = 0.00291154, LEF1_UP.V1_DN\n",
      "cat 68, mi = 0.00286331, GCNP_SHH_UP_LATE.V1_UP\n",
      "cat 147, mi = 0.00282915, PTEN_DN.V1_UP\n",
      "cat 79, mi = 0.00273443, IL2_UP.V1_DN\n",
      "cat 163, mi = 0.00272555, RPS14_DN.V1_UP\n",
      "cat 102, mi = 0.00250388, KRAS.LUNG.BREAST_UP.V1_UP\n",
      "cat 140, mi = 0.00232518, PRC2_EED_UP.V1_DN\n",
      "cat 115, mi = 0.00229306, MTOR_UP.N4.V1_DN\n",
      "cat 98, mi = 0.00228434, KRAS.DF.V1_UP\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 157, mi = 0.009, pvalue = 0.000, RB_P107_DN.V1_UP\n",
      "Passed: cat 38, mi = 0.008, pvalue = 0.000, CSR_LATE_UP.V1_UP\n",
      "Passed: cat 143, mi = 0.005, pvalue = 0.000, PRC2_EZH2_UP.V1_UP\n",
      "Passed: cat 107, mi = 0.003, pvalue = 0.000, LEF1_UP.V1_DN\n",
      "Passed: cat 147, mi = 0.003, pvalue = 0.000, PTEN_DN.V1_UP\n",
      "Passed: cat 79, mi = 0.003, pvalue = 0.000, IL2_UP.V1_DN\n",
      "Passed: cat 163, mi = 0.003, pvalue = 0.000, RPS14_DN.V1_UP\n",
      "Passed: cat 160, mi = 0.002, pvalue = 0.000, RELA_DN.V1_DN\n",
      "Passed: cat 58, mi = 0.002, pvalue = 0.001, ESC_J1_UP_EARLY.V1_UP\n",
      "Passed: cat 78, mi = 0.002, pvalue = 0.001, IL21_UP.V1_UP\n",
      "Passed: cat 135, mi = 0.002, pvalue = 0.001, PIGF_UP.V1_UP\n",
      "Passed: cat 105, mi = 0.001, pvalue = 0.002, KRAS.PROSTATE_UP.V1_DN\n",
      "Passed: cat 49, mi = 0.001, pvalue = 0.003, E2F3_UP.V1_DN\n",
      "Passed: cat 170, mi = 0.001, pvalue = 0.004, SRC_UP.V1_UP\n",
      "Passed: cat 0, mi = 0.001, pvalue = 0.004, AKT_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 158, mi = 0.001, pvalue = 0.018, RB_P130_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 179, mi = 0.001, pvalue = 0.023, TBK1.DN.48HRS_DN\n",
      "Didn't pass (cutoff=0.0050): cat 166, mi = 0.001, pvalue = 0.029, SIRNA_EIF4GI_UP\n",
      "Didn't pass (cutoff=0.0050): cat 129, mi = 0.001, pvalue = 0.044, P53_DN.V2_UP\n",
      "Didn't pass (cutoff=0.0050): cat 54, mi = 0.001, pvalue = 0.037, EIF4E_UP\n",
      "Didn't pass (cutoff=0.0050): cat 133, mi = 0.001, pvalue = 0.042, PDGF_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 33, mi = 0.001, pvalue = 0.140, CRX_NRL_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 131, mi = 0.001, pvalue = 0.180, PDGF_ERK_DN.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 69, mi = 0.001, pvalue = 0.376, GLI1_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 24, mi = 0.001, pvalue = 0.318, CAHOY_ASTROCYTIC\n",
      "Didn't pass: cat 89, mi = 0.000, pvalue = 1.000, KRAS.600.LUNG.BREAST_UP.V1_DN\n",
      "Didn't pass: cat 90, mi = 0.000, pvalue = 1.000, KRAS.600.LUNG.BREAST_UP.V1_UP\n",
      "Didn't pass: cat 91, mi = 0.000, pvalue = 1.000, KRAS.600_UP.V1_DN\n",
      "Didn't pass: cat 92, mi = 0.000, pvalue = 1.000, KRAS.600_UP.V1_UP\n",
      "Didn't pass: cat 121, mi = 0.000, pvalue = 1.000, NFE2L2.V2\n",
      "Didn't pass: cat 171, mi = 0.000, pvalue = 1.000, STK33_DN\n",
      "Didn't pass: cat 172, mi = 0.000, pvalue = 1.000, STK33_NOMO_DN\n",
      "Didn't pass: cat 173, mi = 0.000, pvalue = 1.000, STK33_NOMO_UP\n",
      "Didn't pass: cat 174, mi = 0.000, pvalue = 1.000, STK33_SKM_DN\n",
      "Didn't pass: cat 175, mi = 0.000, pvalue = 1.000, STK33_SKM_UP\n",
      "Total number of categories evaluated = 176.\n",
      "Number of categories that passed the tests = 15\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_P_LVM1_3.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_P_LVM1_3.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 0, 2, 0.0157568761596892\n",
      "best merge is 4, 5, 0.0188192529139801\n",
      "best merge is 0, 1, 0.0242016979313675\n",
      "best merge is 3, 7, 0.0763885796266458\n",
      "best merge is 3, 4, 0.103605302912938\n",
      "best merge is 3, 9, 0.178714621839014\n",
      "best merge is 6, 8, 0.250088956374544\n",
      "best merge is 3, 11, 0.33546559143648\n",
      "best merge is 6, 14, 0.423702106849584\n",
      "best merge is 3, 12, 0.502342680620733\n",
      "best merge is 0, 13, 0.613075624255809\n",
      "order classes\n",
      "3 - 0.0712891911360269\n",
      "7 - 0.517423169602498\n",
      "3 - 1.21943685563746\n",
      "0 2 1\n",
      "3 7 4 5 9 11 12\n",
      "6 8 14\n",
      "10\n",
      "13\n",
      "Done.Start drawing\n",
      "11.42\t0.00\n",
      "LEF1_UP.V1_DN\n",
      "RELA_DN.V1_DN\n",
      "PTEN_DN.V1_UP\n",
      "IL2_UP.V1_DN\n",
      "IL21_UP.V1_UP\n",
      "KRAS.PROSTATE_UP.V1_DN\n",
      "E2F3_UP.V1_DN\n",
      "RPS14_DN.V1_UP\n",
      "ESC_J1_UP_EARLY.V1_UP\n",
      "AKT_UP.V1_DN\n",
      "PIGF_UP.V1_UP\n",
      "SRC_UP.V1_UP\n",
      "RB_P107_DN.V1_UP\n",
      "PRC2_EZH2_UP.V1_UP\n",
      "CSR_LATE_UP.V1_UP\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_P_LVM1_3.csv_PAGE/filt_FC_P_LVM1_3.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_P_LVM1_3.csv_PAGE/filt_FC_P_LVM1_3.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_P_LVM1_3.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_P_LVM1_3.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 0, 2, 0.0157568761596892\n",
      "best merge is 4, 5, 0.0188192529139801\n",
      "best merge is 0, 1, 0.0242016979313675\n",
      "best merge is 3, 7, 0.0763885796266458\n",
      "best merge is 3, 4, 0.103605302912938\n",
      "best merge is 3, 9, 0.178714621839014\n",
      "best merge is 6, 8, 0.250088956374544\n",
      "best merge is 3, 11, 0.33546559143648\n",
      "best merge is 6, 14, 0.423702106849584\n",
      "best merge is 3, 12, 0.502342680620733\n",
      "best merge is 0, 13, 0.613075624255809\n",
      "order classes\n",
      "3 - 0.0712891911360269\n",
      "7 - 0.517423169602498\n",
      "3 - 1.21943685563746\n",
      "0 2 1\n",
      "3 7 4 5 9 11 12\n",
      "6 8 14\n",
      "10\n",
      "13\n",
      "Done.-19.1725043782837\t42\t0\n",
      "-18.1059544658494\t42\t1.65499124343257\n",
      "-17.7381786339755\t42\t2.13309982486865\n",
      "-17.4807355516637\t42\t2.39054290718039\n",
      "-17.223292469352\t42\t2.64798598949212\n",
      "-16.9658493870403\t42\t2.90542907180385\n",
      "-16.6348511383538\t42\t3.16287215411559\n",
      "-16.0831873905429\t42\t3.56742556917688\n",
      "3.13309982486865\t42\t4.2661996497373\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_P_LVM1_3.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_P_LVM1_4.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c6.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c6/human_ensembl_msigdb_c6.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_P_LVM1_4.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 189\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t189\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 189\n",
      "cat 140, mi = 0.00238057, PRC2_EED_UP.V1_DN\n",
      "cat 138, mi = 0.00222673, PRC1_BMI_UP.V1_DN\n",
      "cat 107, mi = 0.00198468, LEF1_UP.V1_DN\n",
      "cat 4, mi = 0.00192033, ALK_DN.V1_DN\n",
      "cat 83, mi = 0.00187589, JNK_DN.V1_DN\n",
      "cat 166, mi = 0.00176551, SIRNA_EIF4GI_UP\n",
      "cat 40, mi = 0.00157932, CTIP_DN.V1_UP\n",
      "cat 79, mi = 0.00154878, IL2_UP.V1_DN\n",
      "cat 75, mi = 0.00150931, IL15_UP.V1_DN\n",
      "cat 94, mi = 0.00147818, KRAS.AMP.LUNG_UP.V1_UP\n",
      "cat 50, mi = 0.00145380, E2F3_UP.V1_UP\n",
      "cat 184, mi = 0.00141693, VEGF_A_UP.V1_UP\n",
      "cat 169, mi = 0.00132585, SRC_UP.V1_DN\n",
      "cat 139, mi = 0.00131249, PRC1_BMI_UP.V1_UP\n",
      "cat 78, mi = 0.00128816, IL21_UP.V1_UP\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 140, mi = 0.002, pvalue = 0.000, PRC2_EED_UP.V1_DN\n",
      "Passed: cat 138, mi = 0.002, pvalue = 0.000, PRC1_BMI_UP.V1_DN\n",
      "Passed: cat 107, mi = 0.002, pvalue = 0.000, LEF1_UP.V1_DN\n",
      "Passed: cat 166, mi = 0.002, pvalue = 0.000, SIRNA_EIF4GI_UP\n",
      "Passed: cat 40, mi = 0.002, pvalue = 0.002, CTIP_DN.V1_UP\n",
      "Passed: cat 75, mi = 0.002, pvalue = 0.001, IL15_UP.V1_DN\n",
      "Passed: cat 94, mi = 0.001, pvalue = 0.003, KRAS.AMP.LUNG_UP.V1_UP\n",
      "Passed: cat 50, mi = 0.001, pvalue = 0.003, E2F3_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 169, mi = 0.001, pvalue = 0.006, SRC_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 156, mi = 0.001, pvalue = 0.009, RB_P107_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 38, mi = 0.001, pvalue = 0.008, CSR_LATE_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 82, mi = 0.001, pvalue = 0.015, JAK2_DN.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 137, mi = 0.001, pvalue = 0.023, PKCA_DN.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 85, mi = 0.001, pvalue = 0.082, KRAS.300_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 187, mi = 0.001, pvalue = 0.125, YAP1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 32, mi = 0.001, pvalue = 0.090, CRX_DN.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 54, mi = 0.001, pvalue = 0.142, EIF4E_UP\n",
      "Didn't pass (cutoff=0.0050): cat 25, mi = 0.001, pvalue = 0.196, CAHOY_ASTROGLIAL\n",
      "Didn't pass (cutoff=0.0050): cat 164, mi = 0.001, pvalue = 0.353, SINGH_KRAS_DEPENDENCY_SIGNATURE_\n",
      "Didn't pass (cutoff=0.0050): cat 34, mi = 0.001, pvalue = 0.246, CRX_NRL_DN.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 97, mi = 0.001, pvalue = 0.238, KRAS.DF.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 42, mi = 0.001, pvalue = 0.241, CYCLIN_D1_KE_.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 131, mi = 0.001, pvalue = 0.255, PDGF_ERK_DN.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 168, mi = 0.001, pvalue = 0.261, SNF5_DN.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 24, mi = 0.001, pvalue = 0.302, CAHOY_ASTROCYTIC\n",
      "Didn't pass (cutoff=0.0050): cat 124, mi = 0.001, pvalue = 0.314, NRL_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 122, mi = 0.001, pvalue = 0.338, NOTCH_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 44, mi = 0.001, pvalue = 0.345, CYCLIN_D1_UP.V1_UP\n",
      "Total number of categories evaluated = 94.\n",
      "Number of categories that passed the tests = 8\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_P_LVM1_4.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_P_LVM1_4.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 1, 2, 0.0857699009251283\n",
      "best merge is 5, 7, 0.230379445544389\n",
      "best merge is 1, 4, 0.249100422811788\n",
      "best merge is 0, 5, 0.320340787737368\n",
      "order classes\n",
      "3 - 0.346056738695528\n",
      "2 - 0.460758891088777\n",
      "1 2 4\n",
      "0\n",
      "3\n",
      "5 7\n",
      "6\n",
      "Done.Start drawing\n",
      "110.21\t0.00\n",
      "SIRNA_EIF4GI_UP\n",
      "E2F3_UP.V1_UP\n",
      "PRC2_EED_UP.V1_DN\n",
      "IL15_UP.V1_DN\n",
      "PRC1_BMI_UP.V1_DN\n",
      "KRAS.AMP.LUNG_UP.V1_UP\n",
      "LEF1_UP.V1_DN\n",
      "CTIP_DN.V1_UP\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_P_LVM1_4.csv_PAGE/filt_FC_P_LVM1_4.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_P_LVM1_4.csv_PAGE/filt_FC_P_LVM1_4.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_P_LVM1_4.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_P_LVM1_4.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 1, 2, 0.0857699009251283\n",
      "best merge is 5, 7, 0.230379445544389\n",
      "best merge is 1, 4, 0.249100422811788\n",
      "best merge is 0, 5, 0.320340787737368\n",
      "order classes\n",
      "3 - 0.346056738695528\n",
      "2 - 0.460758891088777\n",
      "1 2 4\n",
      "0\n",
      "3\n",
      "5 7\n",
      "6\n",
      "Done.-19.9980945467743\t42\t0\n",
      "-19.9923781870974\t42\t0.00381090645132021\n",
      "-19.9847563741947\t42\t0.0114327193539606\n",
      "-19.9790400145177\t42\t0.019054532256601\n",
      "-19.9733236548408\t42\t0.0228654387079212\n",
      "-19.9676072951638\t42\t0.0304872516105617\n",
      "-19.8361310225932\t42\t0.0342981580618819\n",
      "-19.4912439887488\t42\t0.293439796751656\n",
      "1.36203611287542\t42\t0.724072225750839\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c6/fc_values/filt_FC_P_LVM1_4.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "sem --wait done\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "nohup sh run.sh iPAGE_JY.sh &"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
