{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# This is used for running data vs human_msigdb_c1 annotations located in: \n",
    "### /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import pickle\n",
    "import seaborn as sns\n",
    "import scipy.stats\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### get index of file names to loop through "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "cd fc_values\n",
    "ls FC* > filelist.txt\n",
    "cd .."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### get ensembl genes represented in iPAGE database"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash \n",
    "\n",
    "cp /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_RBPs_hg19/human_ensembl_RBPs_hg19_index.txt human_ensembl_index.txt\n",
    "awk '{print $1}' human_ensembl_index.txt > ensembl_index.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### make sure that genes in file match the index "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "###################!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n",
    "###################REMEMBER TO SPECIFY INDEX USED IN THE IPAGE COMMAND!!!!!!!!\n",
    "###################!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n",
    "\n",
    "\n",
    "#####################\n",
    "#######read in data from logfc calculations\n",
    "#####################\n",
    "####define index file to loop through\n",
    "index_file = \"fc_values/filelist.txt\"\n",
    "####define matching ensg index\n",
    "GO_index_filename = \"ensembl_index.txt\"\n",
    "genes_with_GOs = []\n",
    "with open(GO_index_filename,'r') as rf:\n",
    "    for line in rf:\n",
    "        gene_name = line.split('\\n')[0]\n",
    "        genes_with_GOs.append(gene_name)\n",
    "genes_with_GOs_set = set(genes_with_GOs)\n",
    "####open file and loop through each line\n",
    "###\n",
    "iPAGE_sh = \"iPAGE_JY.sh\"\n",
    "cwd = os.getcwd()+\"/\"\n",
    "os.system(\"rm \"+iPAGE_sh)\n",
    "###\n",
    "with open(index_file,'r') as rf:\n",
    "    for line in rf:\n",
    "        #####do this to each file\n",
    "        ###read in file into pandas df\n",
    "        FC_file = \"fc_values/\"+line.split('\\n')[0]\n",
    "        FC_df = pd.read_csv(FC_file,sep=',',index_col=0)\n",
    "        FC_df.columns = ['log2FoldChange']\n",
    "        FC_df.index.names = ['Gene']\n",
    "        ###filter\n",
    "        FC_df = FC_df.loc[[x for x in FC_df.index if x in genes_with_GOs_set]]\n",
    "        ###save\n",
    "        save_file = \"fc_values/filt_\"+line.split('\\n')[0]\n",
    "        FC_df.to_csv(save_file,sep='\\t',header=True)\n",
    "        ###write final iPAGE command to an ipage .sh file\n",
    "        iPAGE_cmd = \"perl /nvme/bins/iPAGEv1.0/page.pl --expfile \"+cwd+save_file+\" --species=human_ensembl_RBPs_hg19 --exptype=continuous --max_p=0.1 --ebins=9 \\n\"\n",
    "        with open(iPAGE_sh, \"a\") as myfile:\n",
    "            myfile.write(iPAGE_cmd)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## then run the command from this folder:\n",
    "\n",
    "nohup sh run.sh iPAGE_JY.sh &"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM1_1.csv --species=human_ensembl_RBPs_hg19 --exptype=continuous --max_p=0.1 --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM1_2.csv --species=human_ensembl_RBPs_hg19 --exptype=continuous --max_p=0.1 --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM1_3.csv --species=human_ensembl_RBPs_hg19 --exptype=continuous --max_p=0.1 --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM1_4.csv --species=human_ensembl_RBPs_hg19 --exptype=continuous --max_p=0.1 --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM2_1.csv --species=human_ensembl_RBPs_hg19 --exptype=continuous --max_p=0.1 --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM2_2.csv --species=human_ensembl_RBPs_hg19 --exptype=continuous --max_p=0.1 --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM2_3.csv --species=human_ensembl_RBPs_hg19 --exptype=continuous --max_p=0.1 --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM2_4.csv --species=human_ensembl_RBPs_hg19 --exptype=continuous --max_p=0.1 --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM2_5.csv --species=human_ensembl_RBPs_hg19 --exptype=continuous --max_p=0.1 --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_P_LVM1_1.csv --species=human_ensembl_RBPs_hg19 --exptype=continuous --max_p=0.1 --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_P_LVM1_2.csv --species=human_ensembl_RBPs_hg19 --exptype=continuous --max_p=0.1 --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_P_LVM1_3.csv --species=human_ensembl_RBPs_hg19 --exptype=continuous --max_p=0.1 --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_P_LVM1_4.csv --species=human_ensembl_RBPs_hg19 --exptype=continuous --max_p=0.1 --ebins=9\n",
      "done\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM1_1.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_RBPs_hg19.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_RBPs_hg19/human_ensembl_RBPs_hg19.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM1_1.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 11408... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 168\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 11408 out of 11408 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 11408\t168\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1268 hits go_terms_num = 168\n",
      "cat 106, mi = 0.00130090, NSUN2\n",
      "cat 71, mi = 0.00085433, HLTF\n",
      "cat 111, mi = 0.00074064, PPIL4\n",
      "cat 73, mi = 0.00072467, HNRNPA2B1\n",
      "cat 133, mi = 0.00071983, SLBP\n",
      "cat 127, mi = 0.00041133, SBDS\n",
      "cat 52, mi = 0.00038081, EZH2\n",
      "cat 10, mi = 0.00027791, AUH\n",
      "cat 0, mi = 0.00000000, AARS\n",
      "cat 1, mi = 0.00000000, AGGF1\n",
      "cat 2, mi = 0.00000000, AGO1\n",
      "cat 3, mi = 0.00000000, AGO2\n",
      "cat 4, mi = 0.00000000, AGO3\n",
      "cat 5, mi = 0.00000000, AGO4\n",
      "cat 6, mi = 0.00000000, AKAP8L\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 106, mi = 0.001, pvalue = 0.009, NSUN2\n",
      "Didn't pass (cutoff=0.1000): cat 71, mi = 0.001, pvalue = 0.114, HLTF\n",
      "Didn't pass (cutoff=0.1000): cat 73, mi = 0.001, pvalue = 0.237, HNRNPA2B1\n",
      "Didn't pass (cutoff=0.1000): cat 52, mi = 0.000, pvalue = 0.309, EZH2\n",
      "Didn't pass: cat 0, mi = 0.000, pvalue = 1.000, AARS\n",
      "Didn't pass: cat 1, mi = 0.000, pvalue = 1.000, AGGF1\n",
      "Didn't pass: cat 2, mi = 0.000, pvalue = 1.000, AGO1\n",
      "Didn't pass: cat 3, mi = 0.000, pvalue = 1.000, AGO2\n",
      "Didn't pass: cat 4, mi = 0.000, pvalue = 1.000, AGO3\n",
      "Didn't pass: cat 5, mi = 0.000, pvalue = 1.000, AGO4\n",
      "Didn't pass: cat 6, mi = 0.000, pvalue = 1.000, AKAP8L\n",
      "Didn't pass: cat 7, mi = 0.000, pvalue = 1.000, ALKBH5\n",
      "Didn't pass: cat 8, mi = 0.000, pvalue = 1.000, ATXN2\n",
      "Didn't pass: cat 9, mi = 0.000, pvalue = 1.000, AUF1\n",
      "Didn't pass: cat 11, mi = 0.000, pvalue = 1.000, BCCIP\n",
      "Didn't pass: cat 12, mi = 0.000, pvalue = 1.000, BUD13\n",
      "Didn't pass: cat 13, mi = 0.000, pvalue = 1.000, C17orf85\n",
      "Didn't pass: cat 14, mi = 0.000, pvalue = 1.000, C22orf28\n",
      "Didn't pass: cat 15, mi = 0.000, pvalue = 1.000, CAPRIN1\n",
      "Didn't pass: cat 16, mi = 0.000, pvalue = 1.000, CDC40\n",
      "Didn't pass: cat 17, mi = 0.000, pvalue = 1.000, CF_Im25\n",
      "Total number of categories evaluated = 15.\n",
      "Number of categories that passed the tests = 1\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM1_1.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM1_1.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "23.77\t0.00\n",
      "NSUN2\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM1_1.csv_PAGE/filt_FC_LVM1_1.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM1_1.csv_PAGE/filt_FC_LVM1_1.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM1_1.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM1_1.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-19.275557425326\t42\t0\n",
      "-18.41859486748\t42\t1.44888514934792\n",
      "-18.2065628944047\t42\t1.71392511569205\n",
      "-18.0475389145982\t42\t1.87294909549853\n",
      "-17.8885149347918\t42\t2.03197307530501\n",
      "-17.7206562894405\t42\t2.19099705511149\n",
      "-17.51745898191\t42\t2.36769036600757\n",
      "-17.2082456878418\t42\t2.59739167017249\n",
      "2.49305847707194\t42\t2.98611695414388\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM1_1.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM2_4.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_RBPs_hg19.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_RBPs_hg19/human_ensembl_RBPs_hg19.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM2_4.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 11408... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 168\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 11408 out of 11408 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 11408\t168\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1268 hits go_terms_num = 168\n",
      "cat 133, mi = 0.00069993, SLBP\n",
      "cat 127, mi = 0.00058746, SBDS\n",
      "cat 106, mi = 0.00051374, NSUN2\n",
      "cat 111, mi = 0.00045058, PPIL4\n",
      "cat 52, mi = 0.00038081, EZH2\n",
      "cat 10, mi = 0.00031406, AUH\n",
      "cat 73, mi = 0.00024830, HNRNPA2B1\n",
      "cat 71, mi = 0.00017190, HLTF\n",
      "cat 0, mi = 0.00000000, AARS\n",
      "cat 1, mi = 0.00000000, AGGF1\n",
      "cat 2, mi = 0.00000000, AGO1\n",
      "cat 3, mi = 0.00000000, AGO2\n",
      "cat 4, mi = 0.00000000, AGO3\n",
      "cat 5, mi = 0.00000000, AGO4\n",
      "cat 6, mi = 0.00000000, AKAP8L\n",
      "\n",
      "RESULTS\n",
      "Didn't pass (cutoff=0.1000): cat 133, mi = 0.001, pvalue = 0.205, SLBP\n",
      "Didn't pass (cutoff=0.1000): cat 127, mi = 0.001, pvalue = 0.380, SBDS\n",
      "Didn't pass (cutoff=0.1000): cat 106, mi = 0.001, pvalue = 0.439, NSUN2\n",
      "Didn't pass (cutoff=0.1000): cat 111, mi = 0.000, pvalue = 0.534, PPIL4\n",
      "Didn't pass (cutoff=0.1000): cat 52, mi = 0.000, pvalue = 0.306, EZH2\n",
      "Didn't pass (cutoff=0.1000): cat 10, mi = 0.000, pvalue = 0.769, AUH\n",
      "Didn't pass (cutoff=0.1000): cat 73, mi = 0.000, pvalue = 0.892, HNRNPA2B1\n",
      "Didn't pass (cutoff=0.1000): cat 71, mi = 0.000, pvalue = 0.958, HLTF\n",
      "Didn't pass: cat 0, mi = 0.000, pvalue = 1.000, AARS\n",
      "Didn't pass: cat 1, mi = 0.000, pvalue = 1.000, AGGF1\n",
      "Didn't pass: cat 2, mi = 0.000, pvalue = 1.000, AGO1\n",
      "Didn't pass: cat 3, mi = 0.000, pvalue = 1.000, AGO2\n",
      "Didn't pass: cat 4, mi = 0.000, pvalue = 1.000, AGO3\n",
      "Didn't pass: cat 5, mi = 0.000, pvalue = 1.000, AGO4\n",
      "Didn't pass: cat 6, mi = 0.000, pvalue = 1.000, AKAP8L\n",
      "Didn't pass: cat 7, mi = 0.000, pvalue = 1.000, ALKBH5\n",
      "Didn't pass: cat 8, mi = 0.000, pvalue = 1.000, ATXN2\n",
      "Didn't pass: cat 9, mi = 0.000, pvalue = 1.000, AUF1\n",
      "Didn't pass: cat 11, mi = 0.000, pvalue = 1.000, BCCIP\n",
      "Didn't pass: cat 12, mi = 0.000, pvalue = 1.000, BUD13\n",
      "Total number of categories evaluated = 10.\n",
      "Number of categories that passed the tests = 0\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM2_4.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM2_4.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "100.30\t0.00\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM2_4.csv_PAGE/filt_FC_LVM2_4.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM2_4.csv_PAGE/filt_FC_LVM2_4.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM2_4.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM2_4.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM2_4.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM1_4.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_RBPs_hg19.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_RBPs_hg19/human_ensembl_RBPs_hg19.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM1_4.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 11408... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 168\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 11408 out of 11408 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 11408\t168\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1268 hits go_terms_num = 168\n",
      "cat 127, mi = 0.00186714, SBDS\n",
      "cat 73, mi = 0.00089571, HNRNPA2B1\n",
      "cat 10, mi = 0.00044026, AUH\n",
      "cat 106, mi = 0.00042721, NSUN2\n",
      "cat 111, mi = 0.00041659, PPIL4\n",
      "cat 52, mi = 0.00038041, EZH2\n",
      "cat 133, mi = 0.00034960, SLBP\n",
      "cat 71, mi = 0.00034143, HLTF\n",
      "cat 0, mi = 0.00000000, AARS\n",
      "cat 1, mi = 0.00000000, AGGF1\n",
      "cat 2, mi = 0.00000000, AGO1\n",
      "cat 3, mi = 0.00000000, AGO2\n",
      "cat 4, mi = 0.00000000, AGO3\n",
      "cat 5, mi = 0.00000000, AGO4\n",
      "cat 6, mi = 0.00000000, AKAP8L\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 127, mi = 0.002, pvalue = 0.000, SBDS\n",
      "Didn't pass (cutoff=0.1000): cat 73, mi = 0.001, pvalue = 0.119, HNRNPA2B1\n",
      "Didn't pass: cat 52, mi = 0.000, pvalue = 1.000, EZH2\n",
      "Didn't pass: cat 0, mi = 0.000, pvalue = 1.000, AARS\n",
      "Didn't pass: cat 1, mi = 0.000, pvalue = 1.000, AGGF1\n",
      "Didn't pass: cat 2, mi = 0.000, pvalue = 1.000, AGO1\n",
      "Didn't pass: cat 3, mi = 0.000, pvalue = 1.000, AGO2\n",
      "Didn't pass: cat 4, mi = 0.000, pvalue = 1.000, AGO3\n",
      "Didn't pass: cat 5, mi = 0.000, pvalue = 1.000, AGO4\n",
      "Didn't pass: cat 6, mi = 0.000, pvalue = 1.000, AKAP8L\n",
      "Didn't pass: cat 7, mi = 0.000, pvalue = 1.000, ALKBH5\n",
      "Didn't pass: cat 8, mi = 0.000, pvalue = 1.000, ATXN2\n",
      "Didn't pass: cat 9, mi = 0.000, pvalue = 1.000, AUF1\n",
      "Didn't pass: cat 11, mi = 0.000, pvalue = 1.000, BCCIP\n",
      "Didn't pass: cat 12, mi = 0.000, pvalue = 1.000, BUD13\n",
      "Didn't pass: cat 13, mi = 0.000, pvalue = 1.000, C17orf85\n",
      "Didn't pass: cat 14, mi = 0.000, pvalue = 1.000, C22orf28\n",
      "Didn't pass: cat 15, mi = 0.000, pvalue = 1.000, CAPRIN1\n",
      "Didn't pass: cat 16, mi = 0.000, pvalue = 1.000, CDC40\n",
      "Didn't pass: cat 17, mi = 0.000, pvalue = 1.000, CF_Im25\n",
      "Didn't pass: cat 18, mi = 0.000, pvalue = 1.000, CF_Im59\n",
      "Total number of categories evaluated = 16.\n",
      "Number of categories that passed the tests = 1\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM1_4.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM1_4.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "21.14\t0.00\n",
      "SBDS\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM1_4.csv_PAGE/filt_FC_LVM1_4.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM1_4.csv_PAGE/filt_FC_LVM1_4.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM1_4.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM1_4.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-19.4039735099338\t42\t0\n",
      "-18.6192052980132\t42\t1.19205298013245\n",
      "-18.3112582781457\t42\t1.56953642384106\n",
      "-18.0927152317881\t42\t1.80794701986755\n",
      "-17.8940397350993\t42\t2.00662251655629\n",
      "-17.6953642384106\t42\t2.20529801324503\n",
      "-17.476821192053\t42\t2.40397350993377\n",
      "-17.158940397351\t42\t2.64238410596026\n",
      "2.51986754966887\t42\t3.03973509933775\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM1_4.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_P_LVM1_3.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_RBPs_hg19.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_RBPs_hg19/human_ensembl_RBPs_hg19.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_P_LVM1_3.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 11408... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 168\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 11408 out of 11408 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 11408\t168\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1268 hits go_terms_num = 168\n",
      "cat 73, mi = 0.00064578, HNRNPA2B1\n",
      "cat 71, mi = 0.00055306, HLTF\n",
      "cat 10, mi = 0.00053768, AUH\n",
      "cat 133, mi = 0.00051887, SLBP\n",
      "cat 111, mi = 0.00049210, PPIL4\n",
      "cat 52, mi = 0.00038081, EZH2\n",
      "cat 106, mi = 0.00026841, NSUN2\n",
      "cat 127, mi = 0.00023094, SBDS\n",
      "cat 0, mi = 0.00000000, AARS\n",
      "cat 1, mi = 0.00000000, AGGF1\n",
      "cat 2, mi = 0.00000000, AGO1\n",
      "cat 3, mi = 0.00000000, AGO2\n",
      "cat 4, mi = 0.00000000, AGO3\n",
      "cat 5, mi = 0.00000000, AGO4\n",
      "cat 6, mi = 0.00000000, AKAP8L\n",
      "\n",
      "RESULTS\n",
      "Didn't pass (cutoff=0.1000): cat 73, mi = 0.001, pvalue = 0.328, HNRNPA2B1\n",
      "Didn't pass (cutoff=0.1000): cat 71, mi = 0.001, pvalue = 0.400, HLTF\n",
      "Didn't pass (cutoff=0.1000): cat 10, mi = 0.001, pvalue = 0.403, AUH\n",
      "Didn't pass (cutoff=0.1000): cat 133, mi = 0.001, pvalue = 0.422, SLBP\n",
      "Didn't pass (cutoff=0.1000): cat 111, mi = 0.000, pvalue = 0.472, PPIL4\n",
      "Didn't pass (cutoff=0.1000): cat 52, mi = 0.000, pvalue = 0.312, EZH2\n",
      "Didn't pass (cutoff=0.1000): cat 106, mi = 0.000, pvalue = 0.833, NSUN2\n",
      "Didn't pass (cutoff=0.1000): cat 127, mi = 0.000, pvalue = 0.898, SBDS\n",
      "Didn't pass: cat 0, mi = 0.000, pvalue = 1.000, AARS\n",
      "Didn't pass: cat 1, mi = 0.000, pvalue = 1.000, AGGF1\n",
      "Didn't pass: cat 2, mi = 0.000, pvalue = 1.000, AGO1\n",
      "Didn't pass: cat 3, mi = 0.000, pvalue = 1.000, AGO2\n",
      "Didn't pass: cat 4, mi = 0.000, pvalue = 1.000, AGO3\n",
      "Didn't pass: cat 5, mi = 0.000, pvalue = 1.000, AGO4\n",
      "Didn't pass: cat 6, mi = 0.000, pvalue = 1.000, AKAP8L\n",
      "Didn't pass: cat 7, mi = 0.000, pvalue = 1.000, ALKBH5\n",
      "Didn't pass: cat 8, mi = 0.000, pvalue = 1.000, ATXN2\n",
      "Didn't pass: cat 9, mi = 0.000, pvalue = 1.000, AUF1\n",
      "Didn't pass: cat 11, mi = 0.000, pvalue = 1.000, BCCIP\n",
      "Didn't pass: cat 12, mi = 0.000, pvalue = 1.000, BUD13\n",
      "Total number of categories evaluated = 10.\n",
      "Number of categories that passed the tests = 0\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_P_LVM1_3.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_P_LVM1_3.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "11.42\t0.00\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_P_LVM1_3.csv_PAGE/filt_FC_P_LVM1_3.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_P_LVM1_3.csv_PAGE/filt_FC_P_LVM1_3.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_P_LVM1_3.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_P_LVM1_3.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_P_LVM1_3.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM1_2.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_RBPs_hg19.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_RBPs_hg19/human_ensembl_RBPs_hg19.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM1_2.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 11408... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 168\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 11408 out of 11408 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 11408\t168\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1268 hits go_terms_num = 168\n",
      "cat 73, mi = 0.00110318, HNRNPA2B1\n",
      "cat 127, mi = 0.00104596, SBDS\n",
      "cat 133, mi = 0.00092050, SLBP\n",
      "cat 71, mi = 0.00081753, HLTF\n",
      "cat 111, mi = 0.00068369, PPIL4\n",
      "cat 106, mi = 0.00066267, NSUN2\n",
      "cat 52, mi = 0.00038081, EZH2\n",
      "cat 10, mi = 0.00027822, AUH\n",
      "cat 0, mi = 0.00000000, AARS\n",
      "cat 1, mi = 0.00000000, AGGF1\n",
      "cat 2, mi = 0.00000000, AGO1\n",
      "cat 3, mi = 0.00000000, AGO2\n",
      "cat 4, mi = 0.00000000, AGO3\n",
      "cat 5, mi = 0.00000000, AGO4\n",
      "cat 6, mi = 0.00000000, AKAP8L\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 73, mi = 0.001, pvalue = 0.043, HNRNPA2B1\n",
      "Passed: cat 127, mi = 0.001, pvalue = 0.056, SBDS\n",
      "Didn't pass (cutoff=0.1000): cat 71, mi = 0.001, pvalue = 0.133, HLTF\n",
      "Didn't pass (cutoff=0.1000): cat 52, mi = 0.000, pvalue = 0.304, EZH2\n",
      "Didn't pass: cat 0, mi = 0.000, pvalue = 1.000, AARS\n",
      "Didn't pass: cat 1, mi = 0.000, pvalue = 1.000, AGGF1\n",
      "Didn't pass: cat 2, mi = 0.000, pvalue = 1.000, AGO1\n",
      "Didn't pass: cat 3, mi = 0.000, pvalue = 1.000, AGO2\n",
      "Didn't pass: cat 4, mi = 0.000, pvalue = 1.000, AGO3\n",
      "Didn't pass: cat 5, mi = 0.000, pvalue = 1.000, AGO4\n",
      "Didn't pass: cat 6, mi = 0.000, pvalue = 1.000, AKAP8L\n",
      "Didn't pass: cat 7, mi = 0.000, pvalue = 1.000, ALKBH5\n",
      "Didn't pass: cat 8, mi = 0.000, pvalue = 1.000, ATXN2\n",
      "Didn't pass: cat 9, mi = 0.000, pvalue = 1.000, AUF1\n",
      "Didn't pass: cat 11, mi = 0.000, pvalue = 1.000, BCCIP\n",
      "Didn't pass: cat 12, mi = 0.000, pvalue = 1.000, BUD13\n",
      "Didn't pass: cat 13, mi = 0.000, pvalue = 1.000, C17orf85\n",
      "Didn't pass: cat 14, mi = 0.000, pvalue = 1.000, C22orf28\n",
      "Didn't pass: cat 15, mi = 0.000, pvalue = 1.000, CAPRIN1\n",
      "Didn't pass: cat 16, mi = 0.000, pvalue = 1.000, CDC40\n",
      "Didn't pass: cat 17, mi = 0.000, pvalue = 1.000, CF_Im25\n",
      "Didn't pass: cat 18, mi = 0.000, pvalue = 1.000, CF_Im59\n",
      "Total number of categories evaluated = 16.\n",
      "Number of categories that passed the tests = 2\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM1_2.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM1_2.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "62.96\t0.00\n",
      "SBDS\n",
      "HNRNPA2B1\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM1_2.csv_PAGE/filt_FC_LVM1_2.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM1_2.csv_PAGE/filt_FC_LVM1_2.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM1_2.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM1_2.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-19.7131512071156\t42\t0\n",
      "-19.34291613723\t42\t0.573697585768742\n",
      "-19.209498094028\t42\t0.740470139771283\n",
      "-19.1161054637865\t42\t0.840533672172808\n",
      "-19.0293837357052\t42\t0.92725540025413\n",
      "-18.9393265565438\t42\t1.01397712833545\n",
      "-18.8292566709022\t42\t1.10736975857687\n",
      "-18.6524777636595\t42\t1.23411689961881\n",
      "1.73046378653113\t42\t1.46092757306226\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM1_2.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM2_3.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_RBPs_hg19.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_RBPs_hg19/human_ensembl_RBPs_hg19.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM2_3.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 11408... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 168\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 11408 out of 11408 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 11408\t168\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1268 hits go_terms_num = 168\n",
      "cat 73, mi = 0.00108576, HNRNPA2B1\n",
      "cat 71, mi = 0.00079291, HLTF\n",
      "cat 133, mi = 0.00060262, SLBP\n",
      "cat 111, mi = 0.00059441, PPIL4\n",
      "cat 127, mi = 0.00058197, SBDS\n",
      "cat 10, mi = 0.00052551, AUH\n",
      "cat 106, mi = 0.00052255, NSUN2\n",
      "cat 52, mi = 0.00038081, EZH2\n",
      "cat 0, mi = 0.00000000, AARS\n",
      "cat 1, mi = 0.00000000, AGGF1\n",
      "cat 2, mi = 0.00000000, AGO1\n",
      "cat 3, mi = 0.00000000, AGO2\n",
      "cat 4, mi = 0.00000000, AGO3\n",
      "cat 5, mi = 0.00000000, AGO4\n",
      "cat 6, mi = 0.00000000, AKAP8L\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 73, mi = 0.001, pvalue = 0.044, HNRNPA2B1\n",
      "Didn't pass (cutoff=0.1000): cat 71, mi = 0.001, pvalue = 0.146, HLTF\n",
      "Didn't pass (cutoff=0.1000): cat 133, mi = 0.001, pvalue = 0.316, SLBP\n",
      "Didn't pass (cutoff=0.1000): cat 111, mi = 0.001, pvalue = 0.329, PPIL4\n",
      "Didn't pass (cutoff=0.1000): cat 127, mi = 0.001, pvalue = 0.399, SBDS\n",
      "Didn't pass (cutoff=0.1000): cat 10, mi = 0.001, pvalue = 0.411, AUH\n",
      "Didn't pass (cutoff=0.1000): cat 106, mi = 0.001, pvalue = 0.418, NSUN2\n",
      "Didn't pass (cutoff=0.1000): cat 52, mi = 0.000, pvalue = 0.311, EZH2\n",
      "Didn't pass: cat 0, mi = 0.000, pvalue = 1.000, AARS\n",
      "Didn't pass: cat 1, mi = 0.000, pvalue = 1.000, AGGF1\n",
      "Didn't pass: cat 2, mi = 0.000, pvalue = 1.000, AGO1\n",
      "Didn't pass: cat 3, mi = 0.000, pvalue = 1.000, AGO2\n",
      "Didn't pass: cat 4, mi = 0.000, pvalue = 1.000, AGO3\n",
      "Didn't pass: cat 5, mi = 0.000, pvalue = 1.000, AGO4\n",
      "Didn't pass: cat 6, mi = 0.000, pvalue = 1.000, AKAP8L\n",
      "Didn't pass: cat 7, mi = 0.000, pvalue = 1.000, ALKBH5\n",
      "Didn't pass: cat 8, mi = 0.000, pvalue = 1.000, ATXN2\n",
      "Didn't pass: cat 9, mi = 0.000, pvalue = 1.000, AUF1\n",
      "Didn't pass: cat 11, mi = 0.000, pvalue = 1.000, BCCIP\n",
      "Didn't pass: cat 12, mi = 0.000, pvalue = 1.000, BUD13\n",
      "Didn't pass: cat 13, mi = 0.000, pvalue = 1.000, C17orf85\n",
      "Total number of categories evaluated = 11.\n",
      "Number of categories that passed the tests = 1\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM2_3.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM2_3.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "79.76\t0.00\n",
      "HNRNPA2B1\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM2_3.csv_PAGE/filt_FC_LVM2_3.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM2_3.csv_PAGE/filt_FC_LVM2_3.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM2_3.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM2_3.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-19.8025325977934\t42\t0\n",
      "-19.5260782347041\t42\t0.39493480441324\n",
      "-19.3970661985958\t42\t0.552908726178536\n",
      "-19.3101805416249\t42\t0.65295887662989\n",
      "-19.2311935807422\t42\t0.726680040120361\n",
      "-19.1443079237713\t42\t0.810932798395186\n",
      "-19.0389919759278\t42\t0.900451354062186\n",
      "-18.8704864593781\t42\t1.02156469408225\n",
      "1.61873119358074\t42\t1.23746238716148\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM2_3.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM1_3.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_RBPs_hg19.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_RBPs_hg19/human_ensembl_RBPs_hg19.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM1_3.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 11408... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 168\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 11408 out of 11408 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 11408\t168\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1268 hits go_terms_num = 168\n",
      "cat 127, mi = 0.00116063, SBDS\n",
      "cat 133, mi = 0.00078573, SLBP\n",
      "cat 73, mi = 0.00068648, HNRNPA2B1\n",
      "cat 10, mi = 0.00048117, AUH\n",
      "cat 52, mi = 0.00038081, EZH2\n",
      "cat 106, mi = 0.00035713, NSUN2\n",
      "cat 71, mi = 0.00035029, HLTF\n",
      "cat 111, mi = 0.00032758, PPIL4\n",
      "cat 0, mi = 0.00000000, AARS\n",
      "cat 1, mi = 0.00000000, AGGF1\n",
      "cat 2, mi = 0.00000000, AGO1\n",
      "cat 3, mi = 0.00000000, AGO2\n",
      "cat 4, mi = 0.00000000, AGO3\n",
      "cat 5, mi = 0.00000000, AGO4\n",
      "cat 6, mi = 0.00000000, AKAP8L\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 127, mi = 0.001, pvalue = 0.033, SBDS\n",
      "Didn't pass (cutoff=0.1000): cat 73, mi = 0.001, pvalue = 0.269, HNRNPA2B1\n",
      "Didn't pass (cutoff=0.1000): cat 52, mi = 0.000, pvalue = 0.304, EZH2\n",
      "Didn't pass: cat 0, mi = 0.000, pvalue = 1.000, AARS\n",
      "Didn't pass: cat 1, mi = 0.000, pvalue = 1.000, AGGF1\n",
      "Didn't pass: cat 2, mi = 0.000, pvalue = 1.000, AGO1\n",
      "Didn't pass: cat 3, mi = 0.000, pvalue = 1.000, AGO2\n",
      "Didn't pass: cat 4, mi = 0.000, pvalue = 1.000, AGO3\n",
      "Didn't pass: cat 5, mi = 0.000, pvalue = 1.000, AGO4\n",
      "Didn't pass: cat 6, mi = 0.000, pvalue = 1.000, AKAP8L\n",
      "Didn't pass: cat 7, mi = 0.000, pvalue = 1.000, ALKBH5\n",
      "Didn't pass: cat 8, mi = 0.000, pvalue = 1.000, ATXN2\n",
      "Didn't pass: cat 9, mi = 0.000, pvalue = 1.000, AUF1\n",
      "Didn't pass: cat 11, mi = 0.000, pvalue = 1.000, BCCIP\n",
      "Didn't pass: cat 12, mi = 0.000, pvalue = 1.000, BUD13\n",
      "Didn't pass: cat 13, mi = 0.000, pvalue = 1.000, C17orf85\n",
      "Didn't pass: cat 14, mi = 0.000, pvalue = 1.000, C22orf28\n",
      "Didn't pass: cat 15, mi = 0.000, pvalue = 1.000, CAPRIN1\n",
      "Didn't pass: cat 16, mi = 0.000, pvalue = 1.000, CDC40\n",
      "Didn't pass: cat 17, mi = 0.000, pvalue = 1.000, CF_Im25\n",
      "Didn't pass: cat 18, mi = 0.000, pvalue = 1.000, CF_Im59\n",
      "Total number of categories evaluated = 16.\n",
      "Number of categories that passed the tests = 1\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM1_3.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM1_3.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "84.94\t0.00\n",
      "SBDS\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM1_3.csv_PAGE/filt_FC_LVM1_3.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM1_3.csv_PAGE/filt_FC_LVM1_3.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM1_3.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM1_3.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-19.8467153284672\t42\t0\n",
      "-19.656345655757\t42\t0.306569343065693\n",
      "-19.5920649870497\t42\t0.380739345420297\n",
      "-19.5401459854015\t42\t0.435130680480339\n",
      "-19.4931716505769\t42\t0.484577348716741\n",
      "-19.4437249823405\t42\t0.529079350129503\n",
      "-19.3794443136332\t42\t0.583470685189545\n",
      "-19.2756063103367\t42\t0.657640687544149\n",
      "1.39557334589122\t42\t0.791146691782435\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM1_3.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM2_1.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_RBPs_hg19.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_RBPs_hg19/human_ensembl_RBPs_hg19.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM2_1.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 11408... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 168\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 11408 out of 11408 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 11408\t168\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1268 hits go_terms_num = 168\n",
      "cat 127, mi = 0.00116684, SBDS\n",
      "cat 73, mi = 0.00108495, HNRNPA2B1\n",
      "cat 133, mi = 0.00085341, SLBP\n",
      "cat 111, mi = 0.00084854, PPIL4\n",
      "cat 71, mi = 0.00080666, HLTF\n",
      "cat 52, mi = 0.00038081, EZH2\n",
      "cat 106, mi = 0.00030305, NSUN2\n",
      "cat 10, mi = 0.00027400, AUH\n",
      "cat 0, mi = 0.00000000, AARS\n",
      "cat 1, mi = 0.00000000, AGGF1\n",
      "cat 2, mi = 0.00000000, AGO1\n",
      "cat 3, mi = 0.00000000, AGO2\n",
      "cat 4, mi = 0.00000000, AGO3\n",
      "cat 5, mi = 0.00000000, AGO4\n",
      "cat 6, mi = 0.00000000, AKAP8L\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 127, mi = 0.001, pvalue = 0.028, SBDS\n",
      "Passed: cat 73, mi = 0.001, pvalue = 0.047, HNRNPA2B1\n",
      "Didn't pass (cutoff=0.1000): cat 111, mi = 0.001, pvalue = 0.108, PPIL4\n",
      "Didn't pass (cutoff=0.1000): cat 71, mi = 0.001, pvalue = 0.135, HLTF\n",
      "Didn't pass (cutoff=0.1000): cat 52, mi = 0.000, pvalue = 0.300, EZH2\n",
      "Didn't pass: cat 0, mi = 0.000, pvalue = 1.000, AARS\n",
      "Didn't pass: cat 1, mi = 0.000, pvalue = 1.000, AGGF1\n",
      "Didn't pass: cat 2, mi = 0.000, pvalue = 1.000, AGO1\n",
      "Didn't pass: cat 3, mi = 0.000, pvalue = 1.000, AGO2\n",
      "Didn't pass: cat 4, mi = 0.000, pvalue = 1.000, AGO3\n",
      "Didn't pass: cat 5, mi = 0.000, pvalue = 1.000, AGO4\n",
      "Didn't pass: cat 6, mi = 0.000, pvalue = 1.000, AKAP8L\n",
      "Didn't pass: cat 7, mi = 0.000, pvalue = 1.000, ALKBH5\n",
      "Didn't pass: cat 8, mi = 0.000, pvalue = 1.000, ATXN2\n",
      "Didn't pass: cat 9, mi = 0.000, pvalue = 1.000, AUF1\n",
      "Didn't pass: cat 11, mi = 0.000, pvalue = 1.000, BCCIP\n",
      "Didn't pass: cat 12, mi = 0.000, pvalue = 1.000, BUD13\n",
      "Didn't pass: cat 13, mi = 0.000, pvalue = 1.000, C17orf85\n",
      "Didn't pass: cat 14, mi = 0.000, pvalue = 1.000, C22orf28\n",
      "Didn't pass: cat 15, mi = 0.000, pvalue = 1.000, CAPRIN1\n",
      "Didn't pass: cat 16, mi = 0.000, pvalue = 1.000, CDC40\n",
      "Didn't pass: cat 17, mi = 0.000, pvalue = 1.000, CF_Im25\n",
      "Total number of categories evaluated = 15.\n",
      "Number of categories that passed the tests = 2\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM2_1.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM2_1.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "123.78\t0.00\n",
      "SBDS\n",
      "HNRNPA2B1\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM2_1.csv_PAGE/filt_FC_LVM2_1.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM2_1.csv_PAGE/filt_FC_LVM2_1.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM2_1.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM2_1.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-19.8744546776539\t42\t0\n",
      "-19.7132816286961\t42\t0.251090644692196\n",
      "-19.6572952011634\t42\t0.322346097915657\n",
      "-19.6182743577315\t42\t0.363063499757635\n",
      "-19.5826466311197\t42\t0.400387784779447\n",
      "-19.5453223460979\t42\t0.434318952981095\n",
      "-19.5012118274358\t42\t0.475036354823073\n",
      "-19.4282598158022\t42\t0.522539990305381\n",
      "1.31047018904508\t42\t0.62094037809016\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM2_1.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM2_5.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_RBPs_hg19.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_RBPs_hg19/human_ensembl_RBPs_hg19.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM2_5.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 11408... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 168\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 11408 out of 11408 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 11408\t168\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1268 hits go_terms_num = 168\n",
      "cat 127, mi = 0.00145260, SBDS\n",
      "cat 73, mi = 0.00106171, HNRNPA2B1\n",
      "cat 10, mi = 0.00062035, AUH\n",
      "cat 106, mi = 0.00061797, NSUN2\n",
      "cat 133, mi = 0.00059037, SLBP\n",
      "cat 111, mi = 0.00055861, PPIL4\n",
      "cat 71, mi = 0.00045005, HLTF\n",
      "cat 52, mi = 0.00038081, EZH2\n",
      "cat 0, mi = 0.00000000, AARS\n",
      "cat 1, mi = 0.00000000, AGGF1\n",
      "cat 2, mi = 0.00000000, AGO1\n",
      "cat 3, mi = 0.00000000, AGO2\n",
      "cat 4, mi = 0.00000000, AGO3\n",
      "cat 5, mi = 0.00000000, AGO4\n",
      "cat 6, mi = 0.00000000, AKAP8L\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 127, mi = 0.001, pvalue = 0.007, SBDS\n",
      "Passed: cat 73, mi = 0.001, pvalue = 0.054, HNRNPA2B1\n",
      "Didn't pass (cutoff=0.1000): cat 52, mi = 0.000, pvalue = 0.305, EZH2\n",
      "Didn't pass: cat 0, mi = 0.000, pvalue = 1.000, AARS\n",
      "Didn't pass: cat 1, mi = 0.000, pvalue = 1.000, AGGF1\n",
      "Didn't pass: cat 2, mi = 0.000, pvalue = 1.000, AGO1\n",
      "Didn't pass: cat 3, mi = 0.000, pvalue = 1.000, AGO2\n",
      "Didn't pass: cat 4, mi = 0.000, pvalue = 1.000, AGO3\n",
      "Didn't pass: cat 5, mi = 0.000, pvalue = 1.000, AGO4\n",
      "Didn't pass: cat 6, mi = 0.000, pvalue = 1.000, AKAP8L\n",
      "Didn't pass: cat 7, mi = 0.000, pvalue = 1.000, ALKBH5\n",
      "Didn't pass: cat 8, mi = 0.000, pvalue = 1.000, ATXN2\n",
      "Didn't pass: cat 9, mi = 0.000, pvalue = 1.000, AUF1\n",
      "Didn't pass: cat 11, mi = 0.000, pvalue = 1.000, BCCIP\n",
      "Didn't pass: cat 12, mi = 0.000, pvalue = 1.000, BUD13\n",
      "Didn't pass: cat 13, mi = 0.000, pvalue = 1.000, C17orf85\n",
      "Didn't pass: cat 14, mi = 0.000, pvalue = 1.000, C22orf28\n",
      "Didn't pass: cat 15, mi = 0.000, pvalue = 1.000, CAPRIN1\n",
      "Didn't pass: cat 16, mi = 0.000, pvalue = 1.000, CDC40\n",
      "Didn't pass: cat 17, mi = 0.000, pvalue = 1.000, CF_Im25\n",
      "Didn't pass: cat 18, mi = 0.000, pvalue = 1.000, CF_Im59\n",
      "Didn't pass: cat 19, mi = 0.000, pvalue = 1.000, CF_Im68\n",
      "Total number of categories evaluated = 17.\n",
      "Number of categories that passed the tests = 2\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM2_5.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM2_5.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "80.82\t0.00\n",
      "SBDS\n",
      "HNRNPA2B1\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM2_5.csv_PAGE/filt_FC_LVM2_5.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM2_5.csv_PAGE/filt_FC_LVM2_5.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM2_5.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM2_5.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-19.8700816629547\t42\t0\n",
      "-19.6829992576095\t42\t0.259836674090572\n",
      "-19.586859688196\t42\t0.374164810690423\n",
      "-19.5141054194506\t42\t0.452115812917595\n",
      "-19.449146250928\t42\t0.519673348181143\n",
      "-19.3815887156644\t42\t0.582034149962881\n",
      "-19.3036377134373\t42\t0.654788418708241\n",
      "-19.1945063103192\t42\t0.737936154417223\n",
      "1.43652561247216\t42\t0.873051224944321\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM2_5.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_P_LVM1_1.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_RBPs_hg19.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_RBPs_hg19/human_ensembl_RBPs_hg19.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_P_LVM1_1.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 11408... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 168\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 11408 out of 11408 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 11408\t168\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1268 hits go_terms_num = 168\n",
      "cat 73, mi = 0.00195679, HNRNPA2B1\n",
      "cat 71, mi = 0.00100010, HLTF\n",
      "cat 10, mi = 0.00075347, AUH\n",
      "cat 106, mi = 0.00060439, NSUN2\n",
      "cat 127, mi = 0.00058694, SBDS\n",
      "cat 111, mi = 0.00038190, PPIL4\n",
      "cat 52, mi = 0.00038041, EZH2\n",
      "cat 133, mi = 0.00034848, SLBP\n",
      "cat 0, mi = 0.00000000, AARS\n",
      "cat 1, mi = 0.00000000, AGGF1\n",
      "cat 2, mi = 0.00000000, AGO1\n",
      "cat 3, mi = 0.00000000, AGO2\n",
      "cat 4, mi = 0.00000000, AGO3\n",
      "cat 5, mi = 0.00000000, AGO4\n",
      "cat 6, mi = 0.00000000, AKAP8L\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 73, mi = 0.002, pvalue = 0.000, HNRNPA2B1\n",
      "Passed: cat 71, mi = 0.001, pvalue = 0.056, HLTF\n",
      "Didn't pass (cutoff=0.1000): cat 106, mi = 0.001, pvalue = 0.306, NSUN2\n",
      "Didn't pass (cutoff=0.1000): cat 127, mi = 0.001, pvalue = 0.371, SBDS\n",
      "Didn't pass: cat 52, mi = 0.000, pvalue = 1.000, EZH2\n",
      "Didn't pass: cat 0, mi = 0.000, pvalue = 1.000, AARS\n",
      "Didn't pass: cat 1, mi = 0.000, pvalue = 1.000, AGGF1\n",
      "Didn't pass: cat 2, mi = 0.000, pvalue = 1.000, AGO1\n",
      "Didn't pass: cat 3, mi = 0.000, pvalue = 1.000, AGO2\n",
      "Didn't pass: cat 4, mi = 0.000, pvalue = 1.000, AGO3\n",
      "Didn't pass: cat 5, mi = 0.000, pvalue = 1.000, AGO4\n",
      "Didn't pass: cat 6, mi = 0.000, pvalue = 1.000, AKAP8L\n",
      "Didn't pass: cat 7, mi = 0.000, pvalue = 1.000, ALKBH5\n",
      "Didn't pass: cat 8, mi = 0.000, pvalue = 1.000, ATXN2\n",
      "Didn't pass: cat 9, mi = 0.000, pvalue = 1.000, AUF1\n",
      "Didn't pass: cat 11, mi = 0.000, pvalue = 1.000, BCCIP\n",
      "Didn't pass: cat 12, mi = 0.000, pvalue = 1.000, BUD13\n",
      "Didn't pass: cat 13, mi = 0.000, pvalue = 1.000, C17orf85\n",
      "Didn't pass: cat 14, mi = 0.000, pvalue = 1.000, C22orf28\n",
      "Didn't pass: cat 15, mi = 0.000, pvalue = 1.000, CAPRIN1\n",
      "Didn't pass: cat 16, mi = 0.000, pvalue = 1.000, CDC40\n",
      "Didn't pass: cat 17, mi = 0.000, pvalue = 1.000, CF_Im25\n",
      "Total number of categories evaluated = 15.\n",
      "Number of categories that passed the tests = 2\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_P_LVM1_1.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_P_LVM1_1.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "280.55\t0.00\n",
      "HNRNPA2B1\n",
      "HLTF\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_P_LVM1_1.csv_PAGE/filt_FC_P_LVM1_1.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_P_LVM1_1.csv_PAGE/filt_FC_P_LVM1_1.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_P_LVM1_1.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_P_LVM1_1.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-19.9985029406523\t42\t0\n",
      "-19.9962573516307\t42\t0.00299411869541971\n",
      "-19.9940117626092\t42\t0.00449117804312957\n",
      "-19.9917661735876\t42\t0.00748529673854928\n",
      "-19.989520584566\t42\t0.00898235608625913\n",
      "-19.9865264658706\t42\t0.0119764747816788\n",
      "-19.9131705578328\t42\t0.0149705934770986\n",
      "-19.7514881482802\t42\t0.158688290857245\n",
      "1.16916770629121\t42\t0.338335412582427\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_P_LVM1_1.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM2_2.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_RBPs_hg19.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_RBPs_hg19/human_ensembl_RBPs_hg19.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM2_2.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 11408... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 168\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 11408 out of 11408 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 11408\t168\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1268 hits go_terms_num = 168\n",
      "cat 111, mi = 0.00144227, PPIL4\n",
      "cat 127, mi = 0.00130203, SBDS\n",
      "cat 73, mi = 0.00107762, HNRNPA2B1\n",
      "cat 133, mi = 0.00103119, SLBP\n",
      "cat 71, mi = 0.00045767, HLTF\n",
      "cat 106, mi = 0.00038173, NSUN2\n",
      "cat 52, mi = 0.00038081, EZH2\n",
      "cat 10, mi = 0.00030031, AUH\n",
      "cat 0, mi = 0.00000000, AARS\n",
      "cat 1, mi = 0.00000000, AGGF1\n",
      "cat 2, mi = 0.00000000, AGO1\n",
      "cat 3, mi = 0.00000000, AGO2\n",
      "cat 4, mi = 0.00000000, AGO3\n",
      "cat 5, mi = 0.00000000, AGO4\n",
      "cat 6, mi = 0.00000000, AKAP8L\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 111, mi = 0.001, pvalue = 0.005, PPIL4\n",
      "Passed: cat 127, mi = 0.001, pvalue = 0.016, SBDS\n",
      "Passed: cat 73, mi = 0.001, pvalue = 0.048, HNRNPA2B1\n",
      "Didn't pass (cutoff=0.1000): cat 52, mi = 0.000, pvalue = 0.305, EZH2\n",
      "Didn't pass: cat 0, mi = 0.000, pvalue = 1.000, AARS\n",
      "Didn't pass: cat 1, mi = 0.000, pvalue = 1.000, AGGF1\n",
      "Didn't pass: cat 2, mi = 0.000, pvalue = 1.000, AGO1\n",
      "Didn't pass: cat 3, mi = 0.000, pvalue = 1.000, AGO2\n",
      "Didn't pass: cat 4, mi = 0.000, pvalue = 1.000, AGO3\n",
      "Didn't pass: cat 5, mi = 0.000, pvalue = 1.000, AGO4\n",
      "Didn't pass: cat 6, mi = 0.000, pvalue = 1.000, AKAP8L\n",
      "Didn't pass: cat 7, mi = 0.000, pvalue = 1.000, ALKBH5\n",
      "Didn't pass: cat 8, mi = 0.000, pvalue = 1.000, ATXN2\n",
      "Didn't pass: cat 9, mi = 0.000, pvalue = 1.000, AUF1\n",
      "Didn't pass: cat 11, mi = 0.000, pvalue = 1.000, BCCIP\n",
      "Didn't pass: cat 12, mi = 0.000, pvalue = 1.000, BUD13\n",
      "Didn't pass: cat 13, mi = 0.000, pvalue = 1.000, C17orf85\n",
      "Didn't pass: cat 14, mi = 0.000, pvalue = 1.000, C22orf28\n",
      "Didn't pass: cat 15, mi = 0.000, pvalue = 1.000, CAPRIN1\n",
      "Didn't pass: cat 16, mi = 0.000, pvalue = 1.000, CDC40\n",
      "Didn't pass: cat 17, mi = 0.000, pvalue = 1.000, CF_Im25\n",
      "Didn't pass: cat 18, mi = 0.000, pvalue = 1.000, CF_Im59\n",
      "Didn't pass: cat 19, mi = 0.000, pvalue = 1.000, CF_Im68\n",
      "Total number of categories evaluated = 17.\n",
      "Number of categories that passed the tests = 3\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM2_2.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM2_2.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 0, 2, 0.662199070335454\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "Done.Start drawing\n",
      "82.48\t0.00\n",
      "SBDS\n",
      "PPIL4\n",
      "HNRNPA2B1\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM2_2.csv_PAGE/filt_FC_LVM2_2.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM2_2.csv_PAGE/filt_FC_LVM2_2.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM2_2.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM2_2.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 0, 2, 0.662199070335454\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "Done.-19.8141367604268\t42\t0\n",
      "-19.5697138700291\t42\t0.37172647914646\n",
      "-19.475509214355\t42\t0.488845780795344\n",
      "-19.4093113482056\t42\t0.560135790494665\n",
      "-19.3482056256062\t42\t0.621241513094083\n",
      "-19.2820077594568\t42\t0.682347235693501\n",
      "-19.2005334626576\t42\t0.753637245392822\n",
      "-19.0732298739088\t42\t0.845295829291949\n",
      "1.5041222114452\t42\t1.0082444228904\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_LVM2_2.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_P_LVM1_2.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_RBPs_hg19.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_RBPs_hg19/human_ensembl_RBPs_hg19.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_P_LVM1_2.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 11408... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 168\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 11408 out of 11408 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 11408\t168\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1268 hits go_terms_num = 168\n",
      "cat 127, mi = 0.00141189, SBDS\n",
      "cat 73, mi = 0.00117342, HNRNPA2B1\n",
      "cat 10, mi = 0.00110474, AUH\n",
      "cat 111, mi = 0.00105087, PPIL4\n",
      "cat 133, mi = 0.00078649, SLBP\n",
      "cat 71, mi = 0.00062358, HLTF\n",
      "cat 52, mi = 0.00038041, EZH2\n",
      "cat 106, mi = 0.00033004, NSUN2\n",
      "cat 0, mi = 0.00000000, AARS\n",
      "cat 1, mi = 0.00000000, AGGF1\n",
      "cat 2, mi = 0.00000000, AGO1\n",
      "cat 3, mi = 0.00000000, AGO2\n",
      "cat 4, mi = 0.00000000, AGO3\n",
      "cat 5, mi = 0.00000000, AGO4\n",
      "cat 6, mi = 0.00000000, AKAP8L\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 127, mi = 0.001, pvalue = 0.007, SBDS\n",
      "Passed: cat 73, mi = 0.001, pvalue = 0.031, HNRNPA2B1\n",
      "Passed: cat 111, mi = 0.001, pvalue = 0.037, PPIL4\n",
      "Didn't pass: cat 52, mi = 0.000, pvalue = 1.000, EZH2\n",
      "Didn't pass: cat 0, mi = 0.000, pvalue = 1.000, AARS\n",
      "Didn't pass: cat 1, mi = 0.000, pvalue = 1.000, AGGF1\n",
      "Didn't pass: cat 2, mi = 0.000, pvalue = 1.000, AGO1\n",
      "Didn't pass: cat 3, mi = 0.000, pvalue = 1.000, AGO2\n",
      "Didn't pass: cat 4, mi = 0.000, pvalue = 1.000, AGO3\n",
      "Didn't pass: cat 5, mi = 0.000, pvalue = 1.000, AGO4\n",
      "Didn't pass: cat 6, mi = 0.000, pvalue = 1.000, AKAP8L\n",
      "Didn't pass: cat 7, mi = 0.000, pvalue = 1.000, ALKBH5\n",
      "Didn't pass: cat 8, mi = 0.000, pvalue = 1.000, ATXN2\n",
      "Didn't pass: cat 9, mi = 0.000, pvalue = 1.000, AUF1\n",
      "Didn't pass: cat 11, mi = 0.000, pvalue = 1.000, BCCIP\n",
      "Didn't pass: cat 12, mi = 0.000, pvalue = 1.000, BUD13\n",
      "Didn't pass: cat 13, mi = 0.000, pvalue = 1.000, C17orf85\n",
      "Didn't pass: cat 14, mi = 0.000, pvalue = 1.000, C22orf28\n",
      "Didn't pass: cat 15, mi = 0.000, pvalue = 1.000, CAPRIN1\n",
      "Didn't pass: cat 16, mi = 0.000, pvalue = 1.000, CDC40\n",
      "Didn't pass: cat 17, mi = 0.000, pvalue = 1.000, CF_Im25\n",
      "Didn't pass: cat 18, mi = 0.000, pvalue = 1.000, CF_Im59\n",
      "Didn't pass: cat 19, mi = 0.000, pvalue = 1.000, CF_Im68\n",
      "Total number of categories evaluated = 17.\n",
      "Number of categories that passed the tests = 3\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_P_LVM1_2.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_P_LVM1_2.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 1, 2, 0.315023316339086\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "Done.Start drawing\n",
      "5.29\t0.00\n",
      "SBDS\n",
      "HNRNPA2B1\n",
      "PPIL4\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_P_LVM1_2.csv_PAGE/filt_FC_P_LVM1_2.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_P_LVM1_2.csv_PAGE/filt_FC_P_LVM1_2.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_P_LVM1_2.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_P_LVM1_2.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 1, 2, 0.315023316339086\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "Done.-16.7844990548204\t42\t0\n",
      "-13.0926275992439\t42\t6.43100189035917\n",
      "-12.3383742911153\t42\t7.38374291115312\n",
      "-11.8223062381853\t42\t7.93950850661626\n",
      "-11.3856332703214\t42\t8.41587901701323\n",
      "-10.9092627599244\t42\t8.81285444234405\n",
      "-10.3137996219282\t42\t9.36862003780718\n",
      "-9.44045368620038\t42\t10.0037807183365\n",
      "6.55765595463138\t42\t11.1153119092628\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_P_LVM1_2.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_P_LVM1_4.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_RBPs_hg19.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_RBPs_hg19/human_ensembl_RBPs_hg19.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_P_LVM1_4.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 11408... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 168\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 11408 out of 11408 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 11408\t168\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1268 hits go_terms_num = 168\n",
      "cat 73, mi = 0.00122635, HNRNPA2B1\n",
      "cat 133, mi = 0.00114590, SLBP\n",
      "cat 106, mi = 0.00098057, NSUN2\n",
      "cat 111, mi = 0.00083896, PPIL4\n",
      "cat 10, mi = 0.00078278, AUH\n",
      "cat 71, mi = 0.00070308, HLTF\n",
      "cat 127, mi = 0.00055072, SBDS\n",
      "cat 52, mi = 0.00038041, EZH2\n",
      "cat 0, mi = 0.00000000, AARS\n",
      "cat 1, mi = 0.00000000, AGGF1\n",
      "cat 2, mi = 0.00000000, AGO1\n",
      "cat 3, mi = 0.00000000, AGO2\n",
      "cat 4, mi = 0.00000000, AGO3\n",
      "cat 5, mi = 0.00000000, AGO4\n",
      "cat 6, mi = 0.00000000, AKAP8L\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 73, mi = 0.001, pvalue = 0.022, HNRNPA2B1\n",
      "Passed: cat 133, mi = 0.001, pvalue = 0.020, SLBP\n",
      "Didn't pass: cat 52, mi = 0.000, pvalue = 1.000, EZH2\n",
      "Didn't pass: cat 0, mi = 0.000, pvalue = 1.000, AARS\n",
      "Didn't pass: cat 1, mi = 0.000, pvalue = 1.000, AGGF1\n",
      "Didn't pass: cat 2, mi = 0.000, pvalue = 1.000, AGO1\n",
      "Didn't pass: cat 3, mi = 0.000, pvalue = 1.000, AGO2\n",
      "Didn't pass: cat 4, mi = 0.000, pvalue = 1.000, AGO3\n",
      "Didn't pass: cat 5, mi = 0.000, pvalue = 1.000, AGO4\n",
      "Didn't pass: cat 6, mi = 0.000, pvalue = 1.000, AKAP8L\n",
      "Didn't pass: cat 7, mi = 0.000, pvalue = 1.000, ALKBH5\n",
      "Didn't pass: cat 8, mi = 0.000, pvalue = 1.000, ATXN2\n",
      "Didn't pass: cat 9, mi = 0.000, pvalue = 1.000, AUF1\n",
      "Didn't pass: cat 11, mi = 0.000, pvalue = 1.000, BCCIP\n",
      "Didn't pass: cat 12, mi = 0.000, pvalue = 1.000, BUD13\n",
      "Didn't pass: cat 13, mi = 0.000, pvalue = 1.000, C17orf85\n",
      "Didn't pass: cat 14, mi = 0.000, pvalue = 1.000, C22orf28\n",
      "Didn't pass: cat 15, mi = 0.000, pvalue = 1.000, CAPRIN1\n",
      "Didn't pass: cat 16, mi = 0.000, pvalue = 1.000, CDC40\n",
      "Didn't pass: cat 17, mi = 0.000, pvalue = 1.000, CF_Im25\n",
      "Didn't pass: cat 18, mi = 0.000, pvalue = 1.000, CF_Im59\n",
      "Didn't pass: cat 19, mi = 0.000, pvalue = 1.000, CF_Im68\n",
      "Total number of categories evaluated = 17.\n",
      "Number of categories that passed the tests = 2\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_P_LVM1_4.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_P_LVM1_4.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "110.21\t0.00\n",
      "HNRNPA2B1\n",
      "SLBP\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_P_LVM1_4.csv_PAGE/filt_FC_P_LVM1_4.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_P_LVM1_4.csv_PAGE/filt_FC_P_LVM1_4.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_P_LVM1_4.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_P_LVM1_4.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-19.9980945467743\t42\t0\n",
      "-19.9923781870974\t42\t0.00381090645132021\n",
      "-19.9847563741947\t42\t0.0114327193539606\n",
      "-19.9790400145177\t42\t0.019054532256601\n",
      "-19.9733236548408\t42\t0.0228654387079212\n",
      "-19.9676072951638\t42\t0.0304872516105617\n",
      "-19.8266037564649\t42\t0.0342981580618819\n",
      "-19.4760003629435\t42\t0.312494329008257\n",
      "1.3677524725524\t42\t0.7355049451048\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_RBPs_hg19/fc_values/filt_FC_P_LVM1_4.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "sem --wait done\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "nothing to output for html page at /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl line 79.\n",
      "nothing to output for html page at /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl line 79.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "nohup sh run.sh iPAGE_JY.sh &"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
