{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# This is used for running data vs human_msigdb_c4 annotations located in: \n",
    "### /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c4"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import pickle\n",
    "import seaborn as sns\n",
    "import scipy.stats\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### get index of file names to loop through "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "cd fc_values\n",
    "ls FC* > filelist.txt\n",
    "cd .."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### get ensembl genes represented in iPAGE database"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash \n",
    "\n",
    "cp /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c4/human_ensembl_msigdb_c4_index.txt human_ensembl_index.txt\n",
    "awk '{print $1}' human_ensembl_index.txt > ensembl_index.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### make sure that genes in file match the index "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "###################!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n",
    "###################REMEMBER TO SPECIFY INDEX USED IN THE IPAGE COMMAND!!!!!!!!\n",
    "###################!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n",
    "\n",
    "\n",
    "#####################\n",
    "#######read in data from logfc calculations\n",
    "#####################\n",
    "####define index file to loop through\n",
    "index_file = \"fc_values/filelist.txt\"\n",
    "####define matching ensg index\n",
    "GO_index_filename = \"ensembl_index.txt\"\n",
    "genes_with_GOs = []\n",
    "with open(GO_index_filename,'r') as rf:\n",
    "    for line in rf:\n",
    "        gene_name = line.split('\\n')[0]\n",
    "        genes_with_GOs.append(gene_name)\n",
    "genes_with_GOs_set = set(genes_with_GOs)\n",
    "####open file and loop through each line\n",
    "###\n",
    "iPAGE_sh = \"iPAGE_JY.sh\"\n",
    "cwd = os.getcwd()+\"/\"\n",
    "os.system(\"rm \"+iPAGE_sh)\n",
    "###\n",
    "with open(index_file,'r') as rf:\n",
    "    for line in rf:\n",
    "        #####do this to each file\n",
    "        ###read in file into pandas df\n",
    "        FC_file = \"fc_values/\"+line.split('\\n')[0]\n",
    "        FC_df = pd.read_csv(FC_file,sep=',',index_col=0)\n",
    "        FC_df.columns = ['log2FoldChange']\n",
    "        FC_df.index.names = ['Gene']\n",
    "        ###filter\n",
    "        FC_df = FC_df.loc[[x for x in FC_df.index if x in genes_with_GOs_set]]\n",
    "        ###save\n",
    "        save_file = \"fc_values/filt_\"+line.split('\\n')[0]\n",
    "        FC_df.to_csv(save_file,sep='\\t',header=True)\n",
    "        ###write final iPAGE command to an ipage .sh file\n",
    "        iPAGE_cmd = \"perl /nvme/bins/iPAGEv1.0/page.pl --expfile \"+cwd+save_file+\" --species=human_ensembl_msigdb_c4 --exptype=continuous --ebins=9 \\n\"\n",
    "        with open(iPAGE_sh, \"a\") as myfile:\n",
    "            myfile.write(iPAGE_cmd)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## then run the command from this folder:\n",
    "\n",
    "nohup sh run.sh iPAGE_JY.sh &"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM1_1.csv --species=human_ensembl_msigdb_c4 --exptype=continuous --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM1_2.csv --species=human_ensembl_msigdb_c4 --exptype=continuous --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM1_3.csv --species=human_ensembl_msigdb_c4 --exptype=continuous --ebins=9\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM1_1.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c4.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c4/human_ensembl_msigdb_c4.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM1_1.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 858\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t858\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 858\n",
      "cat 774, mi = 0.01125916, MORF_NPM1\n",
      "cat 667, mi = 0.00973429, MORF_ACTG1\n",
      "cat 772, mi = 0.00880374, MORF_NME2\n",
      "cat 131, mi = 0.00879342, GNF2_EIF3S6\n",
      "cat 847, mi = 0.00820652, MORF_TPT1\n",
      "cat 83, mi = 0.00765990, GCM_TPT1\n",
      "cat 56, mi = 0.00689069, GCM_NPM1\n",
      "cat 705, mi = 0.00685305, MORF_DAP3\n",
      "cat 715, mi = 0.00615633, MORF_EIF3S6\n",
      "cat 134, mi = 0.00568636, GNF2_FBL\n",
      "cat 217, mi = 0.00568204, GNF2_ST13\n",
      "cat 668, mi = 0.00562315, MORF_ANP32B\n",
      "cat 831, mi = 0.00496927, MORF_SKP1A\n",
      "cat 733, mi = 0.00493293, MORF_G22P1\n",
      "cat 6, mi = 0.00471576, GCM_ACTG1\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 774, mi = 0.011, pvalue = 0.000, MORF_NPM1\n",
      "Passed: cat 806, mi = 0.004, pvalue = 0.000, MORF_RAB1A\n",
      "Passed: cat 625, mi = 0.004, pvalue = 0.000, MODULE_62\n",
      "Passed: cat 421, mi = 0.004, pvalue = 0.000, MODULE_297\n",
      "Passed: cat 318, mi = 0.003, pvalue = 0.000, MODULE_180\n",
      "Passed: cat 50, mi = 0.003, pvalue = 0.000, GCM_MLL\n",
      "Passed: cat 262, mi = 0.003, pvalue = 0.000, MODULE_125\n",
      "Passed: cat 656, mi = 0.002, pvalue = 0.000, MODULE_91\n",
      "Passed: cat 158, mi = 0.002, pvalue = 0.000, GNF2_KPNB1\n",
      "Passed: cat 46, mi = 0.002, pvalue = 0.001, GCM_MAP1B\n",
      "Passed: cat 389, mi = 0.002, pvalue = 0.001, MODULE_257\n",
      "Passed: cat 657, mi = 0.002, pvalue = 0.001, MODULE_92\n",
      "Passed: cat 247, mi = 0.002, pvalue = 0.001, MODULE_111\n",
      "Passed: cat 106, mi = 0.002, pvalue = 0.001, GNF2_CD14\n",
      "Passed: cat 148, mi = 0.001, pvalue = 0.003, GNF2_HPX\n",
      "Didn't pass (cutoff=0.0050): cat 151, mi = 0.001, pvalue = 0.005, GNF2_IGFBP1\n",
      "Didn't pass (cutoff=0.0050): cat 115, mi = 0.001, pvalue = 0.011, GNF2_CDC27\n",
      "Didn't pass (cutoff=0.0050): cat 547, mi = 0.001, pvalue = 0.014, MODULE_456\n",
      "Didn't pass (cutoff=0.0050): cat 112, mi = 0.001, pvalue = 0.013, GNF2_CD97\n",
      "Didn't pass (cutoff=0.0050): cat 583, mi = 0.001, pvalue = 0.007, MODULE_512\n",
      "Didn't pass (cutoff=0.0050): cat 370, mi = 0.001, pvalue = 0.020, MODULE_237\n",
      "Didn't pass (cutoff=0.0050): cat 116, mi = 0.001, pvalue = 0.016, GNF2_CDH11\n",
      "Didn't pass (cutoff=0.0050): cat 126, mi = 0.001, pvalue = 0.028, GNF2_DDX5\n",
      "Didn't pass (cutoff=0.0050): cat 229, mi = 0.001, pvalue = 0.042, GNF2_TYK2\n",
      "Didn't pass (cutoff=0.0050): cat 10, mi = 0.001, pvalue = 0.045, GCM_AQP4\n",
      "Didn't pass (cutoff=0.0050): cat 222, mi = 0.001, pvalue = 0.042, GNF2_TM4SF2\n",
      "Didn't pass (cutoff=0.0050): cat 450, mi = 0.001, pvalue = 0.050, MODULE_329\n",
      "Didn't pass (cutoff=0.0050): cat 494, mi = 0.001, pvalue = 0.049, MODULE_387\n",
      "Didn't pass (cutoff=0.0050): cat 172, mi = 0.001, pvalue = 0.047, GNF2_MMP11\n",
      "Didn't pass (cutoff=0.0050): cat 419, mi = 0.001, pvalue = 0.036, MODULE_294\n",
      "Didn't pass (cutoff=0.0050): cat 177, mi = 0.001, pvalue = 0.037, GNF2_MYL2\n",
      "Didn't pass (cutoff=0.0050): cat 271, mi = 0.001, pvalue = 0.068, MODULE_133\n",
      "Didn't pass (cutoff=0.0050): cat 515, mi = 0.001, pvalue = 0.021, MODULE_412\n",
      "Didn't pass (cutoff=0.0050): cat 192, mi = 0.001, pvalue = 0.082, GNF2_RAB3A\n",
      "Didn't pass (cutoff=0.0050): cat 717, mi = 0.001, pvalue = 0.057, MORF_EIF4E\n",
      "Total number of categories evaluated = 276.\n",
      "Number of categories that passed the tests = 15\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM1_1.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM1_1.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 3, 4, 0.0152996283979816\n",
      "best merge is 1, 8, 0.0313024035114984\n",
      "best merge is 3, 11, 0.0527945083117771\n",
      "best merge is 3, 12, 0.085368504606294\n",
      "best merge is 0, 2, 0.16084294571994\n",
      "best merge is 5, 10, 0.167204686366881\n",
      "best merge is 1, 5, 0.250887157678902\n",
      "best merge is 3, 14, 0.277249633642279\n",
      "best merge is 1, 7, 0.415090526783593\n",
      "best merge is 3, 13, 0.542568845126714\n",
      "best merge is 0, 1, 0.598935532558175\n",
      "order classes\n",
      "2 - 0.603161046449777\n",
      "5 - 0.686980278654807\n",
      "6 - 0.583171794311542\n",
      "3 4 11 12 14 13\n",
      "1 8 5 10 7\n",
      "0 2\n",
      "6\n",
      "9\n",
      "Done.Start drawing\n",
      "23.77\t0.00\n",
      "MODULE_125\n",
      "GNF2_CD14\n",
      "MODULE_257\n",
      "GCM_MLL\n",
      "GCM_MAP1B\n",
      "MODULE_91\n",
      "MORF_RAB1A\n",
      "GNF2_KPNB1\n",
      "MORF_NPM1\n",
      "MODULE_62\n",
      "MODULE_297\n",
      "MODULE_180\n",
      "MODULE_92\n",
      "MODULE_111\n",
      "GNF2_HPX\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM1_1.csv_PAGE/filt_FC_LVM1_1.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM1_1.csv_PAGE/filt_FC_LVM1_1.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM1_1.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM1_1.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 3, 4, 0.0152996283979816\n",
      "best merge is 1, 8, 0.0313024035114984\n",
      "best merge is 3, 11, 0.0527945083117771\n",
      "best merge is 3, 12, 0.085368504606294\n",
      "best merge is 0, 2, 0.16084294571994\n",
      "best merge is 5, 10, 0.167204686366881\n",
      "best merge is 1, 5, 0.250887157678902\n",
      "best merge is 3, 14, 0.277249633642279\n",
      "best merge is 1, 7, 0.415090526783593\n",
      "best merge is 3, 13, 0.542568845126714\n",
      "best merge is 0, 1, 0.598935532558175\n",
      "order classes\n",
      "2 - 0.603161046449777\n",
      "5 - 0.686980278654807\n",
      "6 - 0.583171794311542\n",
      "3 4 11 12 14 13\n",
      "1 8 5 10 7\n",
      "0 2\n",
      "6\n",
      "9\n",
      "Done.-19.2932267564156\t42\t0\n",
      "-18.4450988641144\t42\t1.4135464871687\n",
      "-18.2153975599495\t42\t1.69625578460244\n",
      "-18.056373580143\t42\t1.87294909549853\n",
      "-17.8973496003366\t42\t2.0143037442154\n",
      "-17.7206562894405\t42\t2.19099705511149\n",
      "-17.51745898191\t42\t2.36769036600757\n",
      "-17.2082456878418\t42\t2.59739167017249\n",
      "2.49305847707194\t42\t2.98611695414388\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM1_1.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM1_2.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c4.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c4/human_ensembl_msigdb_c4.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM1_2.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 858\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t858\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 858\n",
      "cat 774, mi = 0.01237509, MORF_NPM1\n",
      "cat 114, mi = 0.01053003, GNF2_CDC20\n",
      "cat 105, mi = 0.01046176, GNF2_CCNB2\n",
      "cat 667, mi = 0.01012326, MORF_ACTG1\n",
      "cat 131, mi = 0.00992777, GNF2_EIF3S6\n",
      "cat 772, mi = 0.00955972, MORF_NME2\n",
      "cat 113, mi = 0.00904745, GNF2_CDC2\n",
      "cat 104, mi = 0.00886678, GNF2_CCNA2\n",
      "cat 847, mi = 0.00879267, MORF_TPT1\n",
      "cat 83, mi = 0.00873241, GCM_TPT1\n",
      "cat 56, mi = 0.00825770, GCM_NPM1\n",
      "cat 146, mi = 0.00795023, GNF2_HMMR\n",
      "cat 121, mi = 0.00774535, GNF2_CENPF\n",
      "cat 120, mi = 0.00748223, GNF2_CENPE\n",
      "cat 668, mi = 0.00656407, MORF_ANP32B\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 774, mi = 0.012, pvalue = 0.000, MORF_NPM1\n",
      "Passed: cat 114, mi = 0.011, pvalue = 0.000, GNF2_CDC20\n",
      "Passed: cat 625, mi = 0.006, pvalue = 0.000, MODULE_62\n",
      "Passed: cat 806, mi = 0.004, pvalue = 0.000, MORF_RAB1A\n",
      "Passed: cat 318, mi = 0.004, pvalue = 0.000, MODULE_180\n",
      "Passed: cat 180, mi = 0.003, pvalue = 0.000, GNF2_NS\n",
      "Passed: cat 67, mi = 0.003, pvalue = 0.000, GCM_RAB10\n",
      "Passed: cat 657, mi = 0.003, pvalue = 0.000, MODULE_92\n",
      "Passed: cat 421, mi = 0.003, pvalue = 0.000, MODULE_297\n",
      "Passed: cat 247, mi = 0.003, pvalue = 0.000, MODULE_111\n",
      "Passed: cat 158, mi = 0.003, pvalue = 0.000, GNF2_KPNB1\n",
      "Passed: cat 403, mi = 0.002, pvalue = 0.000, MODULE_277\n",
      "Passed: cat 29, mi = 0.002, pvalue = 0.001, GCM_DENR\n",
      "Passed: cat 151, mi = 0.002, pvalue = 0.000, GNF2_IGFBP1\n",
      "Passed: cat 126, mi = 0.001, pvalue = 0.003, GNF2_DDX5\n",
      "Didn't pass (cutoff=0.0050): cat 129, mi = 0.001, pvalue = 0.007, GNF2_DNM1\n",
      "Didn't pass (cutoff=0.0050): cat 442, mi = 0.001, pvalue = 0.007, MODULE_321\n",
      "Didn't pass (cutoff=0.0050): cat 719, mi = 0.001, pvalue = 0.008, MORF_ERCC2\n",
      "Didn't pass (cutoff=0.0050): cat 369, mi = 0.001, pvalue = 0.006, MODULE_236\n",
      "Didn't pass (cutoff=0.0050): cat 692, mi = 0.001, pvalue = 0.009, MORF_CD8A\n",
      "Didn't pass (cutoff=0.0050): cat 48, mi = 0.001, pvalue = 0.010, GCM_MAPK10\n",
      "Didn't pass (cutoff=0.0050): cat 425, mi = 0.001, pvalue = 0.015, MODULE_300\n",
      "Didn't pass (cutoff=0.0050): cat 642, mi = 0.001, pvalue = 0.015, MODULE_79\n",
      "Didn't pass (cutoff=0.0050): cat 519, mi = 0.001, pvalue = 0.015, MODULE_417\n",
      "Didn't pass (cutoff=0.0050): cat 558, mi = 0.001, pvalue = 0.013, MODULE_474\n",
      "Didn't pass (cutoff=0.0050): cat 419, mi = 0.001, pvalue = 0.007, MODULE_294\n",
      "Didn't pass (cutoff=0.0050): cat 462, mi = 0.001, pvalue = 0.020, MODULE_343\n",
      "Didn't pass (cutoff=0.0050): cat 159, mi = 0.001, pvalue = 0.023, GNF2_LCAT\n",
      "Didn't pass (cutoff=0.0050): cat 590, mi = 0.001, pvalue = 0.028, MODULE_524\n",
      "Didn't pass (cutoff=0.0050): cat 515, mi = 0.001, pvalue = 0.007, MODULE_412\n",
      "Didn't pass (cutoff=0.0050): cat 116, mi = 0.001, pvalue = 0.025, GNF2_CDH11\n",
      "Didn't pass (cutoff=0.0050): cat 165, mi = 0.001, pvalue = 0.031, GNF2_MCL1\n",
      "Didn't pass (cutoff=0.0050): cat 156, mi = 0.001, pvalue = 0.032, GNF2_JAK1\n",
      "Didn't pass (cutoff=0.0050): cat 417, mi = 0.001, pvalue = 0.023, MODULE_292\n",
      "Didn't pass (cutoff=0.0050): cat 848, mi = 0.001, pvalue = 0.036, MORF_TTN\n",
      "Total number of categories evaluated = 290.\n",
      "Number of categories that passed the tests = 15\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM1_2.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM1_2.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 4, 8, 0.00336497611208886\n",
      "best merge is 4, 13, 0.0334946421209826\n",
      "best merge is 2, 5, 0.0444166428897198\n",
      "best merge is 4, 9, 0.0488568554874267\n",
      "best merge is 4, 7, 0.0989590457366556\n",
      "best merge is 0, 2, 0.124365490407274\n",
      "best merge is 3, 6, 0.154544564309219\n",
      "best merge is 3, 12, 0.244425760167164\n",
      "best merge is 3, 14, 0.332202745886323\n",
      "best merge is 3, 10, 0.481657923825894\n",
      "best merge is 3, 11, 0.607578624443042\n",
      "order classes\n",
      "3 - 0.325719581893631\n",
      "5 - 0.855992644225462\n",
      "5 - 0.14706264234311\n",
      "4 8 13 9 7\n",
      "1\n",
      "3 6 12 14 10\n",
      "0 2 5\n",
      "11\n",
      "Done.Start drawing\n",
      "62.96\t0.00\n",
      "GNF2_CDC20\n",
      "GCM_RAB10\n",
      "MORF_RAB1A\n",
      "GCM_DENR\n",
      "MODULE_277\n",
      "GNF2_DDX5\n",
      "GNF2_KPNB1\n",
      "MORF_NPM1\n",
      "MODULE_62\n",
      "GNF2_NS\n",
      "MODULE_180\n",
      "MODULE_297\n",
      "GNF2_IGFBP1\n",
      "MODULE_111\n",
      "MODULE_92\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM1_2.csv_PAGE/filt_FC_LVM1_2.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM1_2.csv_PAGE/filt_FC_LVM1_2.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM1_2.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM1_2.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 4, 8, 0.00336497611208886\n",
      "best merge is 4, 13, 0.0334946421209826\n",
      "best merge is 2, 5, 0.0444166428897198\n",
      "best merge is 4, 9, 0.0488568554874267\n",
      "best merge is 4, 7, 0.0989590457366556\n",
      "best merge is 0, 2, 0.124365490407274\n",
      "best merge is 3, 6, 0.154544564309219\n",
      "best merge is 3, 12, 0.244425760167164\n",
      "best merge is 3, 14, 0.332202745886323\n",
      "best merge is 3, 10, 0.481657923825894\n",
      "best merge is 3, 11, 0.607578624443042\n",
      "order classes\n",
      "3 - 0.325719581893631\n",
      "5 - 0.855992644225462\n",
      "5 - 0.14706264234311\n",
      "4 8 13 9 7\n",
      "1\n",
      "3 6 12 14 10\n",
      "0 2 5\n",
      "11\n",
      "Done.-19.7198221092757\t42\t0\n",
      "-19.3562579415502\t42\t0.560355781448539\n",
      "-19.2195044472681\t42\t0.72712833545108\n",
      "-19.1227763659466\t42\t0.833862770012706\n",
      "-19.0360546378653\t42\t0.920584498094028\n",
      "-18.9459974587039\t42\t1.00730622617535\n",
      "-18.8359275730623\t42\t1.10069885641677\n",
      "-18.6558132147395\t42\t1.2274459974587\n",
      "1.73046378653113\t42\t1.46092757306226\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM1_2.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM1_4.csv --species=human_ensembl_msigdb_c4 --exptype=continuous --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM2_1.csv --species=human_ensembl_msigdb_c4 --exptype=continuous --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM2_2.csv --species=human_ensembl_msigdb_c4 --exptype=continuous --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM2_3.csv --species=human_ensembl_msigdb_c4 --exptype=continuous --ebins=9\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM1_3.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c4.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c4/human_ensembl_msigdb_c4.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM1_3.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 858\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t858\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 858\n",
      "cat 104, mi = 0.01437045, GNF2_CCNA2\n",
      "cat 113, mi = 0.01335098, GNF2_CDC2\n",
      "cat 121, mi = 0.01264711, GNF2_CENPF\n",
      "cat 114, mi = 0.01254780, GNF2_CDC20\n",
      "cat 105, mi = 0.01247954, GNF2_CCNB2\n",
      "cat 184, mi = 0.01227916, GNF2_PCNA\n",
      "cat 201, mi = 0.01166544, GNF2_RRM1\n",
      "cat 146, mi = 0.01084331, GNF2_HMMR\n",
      "cat 210, mi = 0.01053550, GNF2_SMC4L1\n",
      "cat 120, mi = 0.00999303, GNF2_CENPE\n",
      "cat 123, mi = 0.00883107, GNF2_CKS2\n",
      "cat 135, mi = 0.00853330, GNF2_FEN1\n",
      "cat 166, mi = 0.00840117, GNF2_MCM4\n",
      "cat 199, mi = 0.00830572, GNF2_RFC4\n",
      "cat 96, mi = 0.00826520, GNF2_BUB1B\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 104, mi = 0.014, pvalue = 0.000, GNF2_CCNA2\n",
      "Passed: cat 774, mi = 0.008, pvalue = 0.000, MORF_NPM1\n",
      "Passed: cat 806, mi = 0.004, pvalue = 0.000, MORF_RAB1A\n",
      "Passed: cat 289, mi = 0.004, pvalue = 0.000, MODULE_152\n",
      "Passed: cat 308, mi = 0.003, pvalue = 0.000, MODULE_171\n",
      "Passed: cat 50, mi = 0.003, pvalue = 0.000, GCM_MLL\n",
      "Passed: cat 129, mi = 0.002, pvalue = 0.000, GNF2_DNM1\n",
      "Passed: cat 126, mi = 0.002, pvalue = 0.000, GNF2_DDX5\n",
      "Passed: cat 97, mi = 0.002, pvalue = 0.000, GNF2_BUB3\n",
      "Passed: cat 472, mi = 0.002, pvalue = 0.000, MODULE_357\n",
      "Passed: cat 318, mi = 0.002, pvalue = 0.000, MODULE_180\n",
      "Passed: cat 381, mi = 0.002, pvalue = 0.000, MODULE_248\n",
      "Passed: cat 137, mi = 0.002, pvalue = 0.000, GNF2_FOS\n",
      "Passed: cat 657, mi = 0.002, pvalue = 0.001, MODULE_92\n",
      "Passed: cat 205, mi = 0.002, pvalue = 0.001, GNF2_SELL\n",
      "Passed: cat 363, mi = 0.001, pvalue = 0.000, MODULE_227\n",
      "Didn't pass (cutoff=0.0050): cat 521, mi = 0.001, pvalue = 0.006, MODULE_419\n",
      "Passed: cat 116, mi = 0.001, pvalue = 0.005, GNF2_CDH11\n",
      "Didn't pass (cutoff=0.0050): cat 590, mi = 0.001, pvalue = 0.008, MODULE_524\n",
      "Didn't pass (cutoff=0.0050): cat 748, mi = 0.001, pvalue = 0.007, MORF_ITGA2\n",
      "Passed: cat 327, mi = 0.001, pvalue = 0.002, MODULE_190\n",
      "Didn't pass (cutoff=0.0050): cat 759, mi = 0.001, pvalue = 0.008, MORF_MAP2K7\n",
      "Didn't pass (cutoff=0.0050): cat 32, mi = 0.001, pvalue = 0.013, GCM_DPF2\n",
      "Didn't pass (cutoff=0.0050): cat 609, mi = 0.001, pvalue = 0.011, MODULE_56\n",
      "Didn't pass (cutoff=0.0050): cat 804, mi = 0.001, pvalue = 0.018, MORF_PTPRR\n",
      "Didn't pass (cutoff=0.0050): cat 800, mi = 0.001, pvalue = 0.013, MORF_PTEN\n",
      "Didn't pass (cutoff=0.0050): cat 825, mi = 0.001, pvalue = 0.017, MORF_RFC5\n",
      "Didn't pass (cutoff=0.0050): cat 696, mi = 0.001, pvalue = 0.015, MORF_CDH4\n",
      "Didn't pass (cutoff=0.0050): cat 152, mi = 0.001, pvalue = 0.020, GNF2_IL2RB\n",
      "Didn't pass (cutoff=0.0050): cat 432, mi = 0.001, pvalue = 0.021, MODULE_310\n",
      "Didn't pass (cutoff=0.0050): cat 692, mi = 0.001, pvalue = 0.027, MORF_CD8A\n",
      "Didn't pass (cutoff=0.0050): cat 20, mi = 0.001, pvalue = 0.024, GCM_CDH5\n",
      "Didn't pass (cutoff=0.0050): cat 237, mi = 0.001, pvalue = 0.034, MODULE_102\n",
      "Didn't pass (cutoff=0.0050): cat 473, mi = 0.001, pvalue = 0.025, MODULE_358\n",
      "Didn't pass (cutoff=0.0050): cat 28, mi = 0.001, pvalue = 0.041, GCM_DEAF1\n",
      "Didn't pass (cutoff=0.0050): cat 3, mi = 0.001, pvalue = 0.047, CAR_MYST2\n",
      "Didn't pass (cutoff=0.0050): cat 169, mi = 0.001, pvalue = 0.045, GNF2_MLF1\n",
      "Didn't pass (cutoff=0.0050): cat 74, mi = 0.001, pvalue = 0.051, GCM_SIRT2\n",
      "Didn't pass (cutoff=0.0050): cat 717, mi = 0.001, pvalue = 0.038, MORF_EIF4E\n",
      "Didn't pass (cutoff=0.0050): cat 567, mi = 0.001, pvalue = 0.044, MODULE_489\n",
      "Didn't pass (cutoff=0.0050): cat 450, mi = 0.001, pvalue = 0.075, MODULE_329\n",
      "Total number of categories evaluated = 315.\n",
      "Number of categories that passed the tests = 18\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM1_3.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM1_3.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 9, 10, 0.028975075515473\n",
      "best merge is 1, 8, 0.0385726739754944\n",
      "best merge is 4, 17, 0.0529329866170813\n",
      "best merge is 0, 9, 0.0608940874338618\n",
      "best merge is 2, 5, 0.132869385579051\n",
      "best merge is 4, 13, 0.150882384645352\n",
      "best merge is 11, 12, 0.204792974188539\n",
      "best merge is 4, 16, 0.20745348410543\n",
      "best merge is 0, 6, 0.212315312978386\n",
      "best merge is 4, 11, 0.240149744216171\n",
      "best merge is 1, 3, 0.257737143586409\n",
      "best merge is 2, 7, 0.394768489326612\n",
      "best merge is 4, 14, 0.425489279494786\n",
      "best merge is 2, 15, 0.637634391222916\n",
      "order classes\n",
      "4 - 0.443086418991575\n",
      "3 - 0.738729281531082\n",
      "3 - 1.2298751523097\n",
      "7 - 0.593631059101998\n",
      "0 9 10 6\n",
      "1 8 3\n",
      "2 5 7\n",
      "4 17 13 16 11 12 14\n",
      "15\n",
      "Done.Start drawing\n",
      "84.94\t0.00\n",
      "MODULE_171\n",
      "MODULE_190\n",
      "MODULE_92\n",
      "GNF2_CDH11\n",
      "MODULE_248\n",
      "GNF2_FOS\n",
      "GNF2_SELL\n",
      "GNF2_DDX5\n",
      "MODULE_227\n",
      "MORF_RAB1A\n",
      "GCM_MLL\n",
      "MORF_NPM1\n",
      "GNF2_BUB3\n",
      "MODULE_152\n",
      "GNF2_CCNA2\n",
      "MODULE_357\n",
      "MODULE_180\n",
      "GNF2_DNM1\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM1_3.csv_PAGE/filt_FC_LVM1_3.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM1_3.csv_PAGE/filt_FC_LVM1_3.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM1_3.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM1_3.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 9, 10, 0.028975075515473\n",
      "best merge is 1, 8, 0.0385726739754944\n",
      "best merge is 4, 17, 0.0529329866170813\n",
      "best merge is 0, 9, 0.0608940874338618\n",
      "best merge is 2, 5, 0.132869385579051\n",
      "best merge is 4, 13, 0.150882384645352\n",
      "best merge is 11, 12, 0.204792974188539\n",
      "best merge is 4, 16, 0.20745348410543\n",
      "best merge is 0, 6, 0.212315312978386\n",
      "best merge is 4, 11, 0.240149744216171\n",
      "best merge is 1, 3, 0.257737143586409\n",
      "best merge is 2, 7, 0.394768489326612\n",
      "best merge is 4, 14, 0.425489279494786\n",
      "best merge is 2, 15, 0.637634391222916\n",
      "order classes\n",
      "4 - 0.443086418991575\n",
      "3 - 0.738729281531082\n",
      "3 - 1.2298751523097\n",
      "7 - 0.593631059101998\n",
      "0 9 10 6\n",
      "1 8 3\n",
      "2 5 7\n",
      "4 17 13 16 11 12 14\n",
      "15\n",
      "Done.-19.849187661879\t42\t0\n",
      "-19.6588179891688\t42\t0.301624676242053\n",
      "-19.5920649870497\t42\t0.380739345420297\n",
      "-19.5401459854015\t42\t0.435130680480339\n",
      "-19.4931716505769\t42\t0.484577348716741\n",
      "-19.4437249823405\t42\t0.529079350129503\n",
      "-19.3769719802213\t42\t0.583470685189545\n",
      "-19.2706616435131\t42\t0.662585354367789\n",
      "1.39804567930304\t42\t0.796091358606075\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM1_3.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM2_1.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c4.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c4/human_ensembl_msigdb_c4.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM2_1.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 858\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t858\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 858\n",
      "cat 774, mi = 0.00972082, MORF_NPM1\n",
      "cat 131, mi = 0.00946339, GNF2_EIF3S6\n",
      "cat 667, mi = 0.00852759, MORF_ACTG1\n",
      "cat 847, mi = 0.00794739, MORF_TPT1\n",
      "cat 772, mi = 0.00712728, MORF_NME2\n",
      "cat 668, mi = 0.00696218, MORF_ANP32B\n",
      "cat 56, mi = 0.00692513, GCM_NPM1\n",
      "cat 83, mi = 0.00691388, GCM_TPT1\n",
      "cat 811, mi = 0.00675859, MORF_RAD21\n",
      "cat 705, mi = 0.00670551, MORF_DAP3\n",
      "cat 831, mi = 0.00656586, MORF_SKP1A\n",
      "cat 783, mi = 0.00629461, MORF_PPP1CA\n",
      "cat 721, mi = 0.00622761, MORF_ERH\n",
      "cat 715, mi = 0.00603311, MORF_EIF3S6\n",
      "cat 114, mi = 0.00553065, GNF2_CDC20\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 774, mi = 0.010, pvalue = 0.000, MORF_NPM1\n",
      "Passed: cat 114, mi = 0.006, pvalue = 0.000, GNF2_CDC20\n",
      "Passed: cat 813, mi = 0.005, pvalue = 0.000, MORF_RAD23B\n",
      "Passed: cat 421, mi = 0.004, pvalue = 0.000, MODULE_297\n",
      "Passed: cat 318, mi = 0.004, pvalue = 0.000, MODULE_180\n",
      "Passed: cat 67, mi = 0.004, pvalue = 0.000, GCM_RAB10\n",
      "Passed: cat 158, mi = 0.003, pvalue = 0.000, GNF2_KPNB1\n",
      "Passed: cat 247, mi = 0.002, pvalue = 0.000, MODULE_111\n",
      "Passed: cat 657, mi = 0.002, pvalue = 0.000, MODULE_92\n",
      "Passed: cat 357, mi = 0.002, pvalue = 0.000, MODULE_22\n",
      "Passed: cat 162, mi = 0.002, pvalue = 0.000, GNF2_MAPT\n",
      "Passed: cat 846, mi = 0.002, pvalue = 0.000, MORF_TPR\n",
      "Passed: cat 126, mi = 0.002, pvalue = 0.000, GNF2_DDX5\n",
      "Passed: cat 29, mi = 0.002, pvalue = 0.001, GCM_DENR\n",
      "Passed: cat 177, mi = 0.002, pvalue = 0.000, GNF2_MYL2\n",
      "Passed: cat 642, mi = 0.002, pvalue = 0.002, MODULE_79\n",
      "Passed: cat 130, mi = 0.001, pvalue = 0.001, GNF2_EGFR\n",
      "Passed: cat 116, mi = 0.001, pvalue = 0.002, GNF2_CDH11\n",
      "Didn't pass (cutoff=0.0050): cat 590, mi = 0.001, pvalue = 0.007, MODULE_524\n",
      "Didn't pass (cutoff=0.0050): cat 99, mi = 0.001, pvalue = 0.007, GNF2_CASP1\n",
      "Passed: cat 653, mi = 0.001, pvalue = 0.003, MODULE_89\n",
      "Didn't pass (cutoff=0.0050): cat 188, mi = 0.001, pvalue = 0.008, GNF2_PTPN4\n",
      "Didn't pass (cutoff=0.0050): cat 662, mi = 0.001, pvalue = 0.006, MODULE_97\n",
      "Didn't pass (cutoff=0.0050): cat 476, mi = 0.001, pvalue = 0.009, MODULE_361\n",
      "Didn't pass (cutoff=0.0050): cat 377, mi = 0.001, pvalue = 0.007, MODULE_243\n",
      "Didn't pass (cutoff=0.0050): cat 411, mi = 0.001, pvalue = 0.013, MODULE_286\n",
      "Didn't pass (cutoff=0.0050): cat 609, mi = 0.001, pvalue = 0.007, MODULE_56\n",
      "Didn't pass (cutoff=0.0050): cat 502, mi = 0.001, pvalue = 0.014, MODULE_399\n",
      "Didn't pass (cutoff=0.0050): cat 119, mi = 0.001, pvalue = 0.011, GNF2_CEBPA\n",
      "Didn't pass (cutoff=0.0050): cat 111, mi = 0.001, pvalue = 0.010, GNF2_CD7\n",
      "Passed: cat 348, mi = 0.001, pvalue = 0.004, MODULE_21\n",
      "Didn't pass (cutoff=0.0050): cat 442, mi = 0.001, pvalue = 0.012, MODULE_321\n",
      "Didn't pass (cutoff=0.0050): cat 320, mi = 0.001, pvalue = 0.017, MODULE_182\n",
      "Didn't pass (cutoff=0.0050): cat 562, mi = 0.001, pvalue = 0.022, MODULE_481\n",
      "Didn't pass (cutoff=0.0050): cat 519, mi = 0.001, pvalue = 0.023, MODULE_417\n",
      "Didn't pass (cutoff=0.0050): cat 567, mi = 0.001, pvalue = 0.016, MODULE_489\n",
      "Didn't pass (cutoff=0.0050): cat 603, mi = 0.001, pvalue = 0.029, MODULE_544\n",
      "Didn't pass (cutoff=0.0050): cat 426, mi = 0.001, pvalue = 0.026, MODULE_301\n",
      "Didn't pass (cutoff=0.0050): cat 459, mi = 0.001, pvalue = 0.033, MODULE_340\n",
      "Didn't pass (cutoff=0.0050): cat 374, mi = 0.001, pvalue = 0.045, MODULE_240\n",
      "Didn't pass (cutoff=0.0050): cat 420, mi = 0.001, pvalue = 0.040, MODULE_295\n",
      "Didn't pass (cutoff=0.0050): cat 435, mi = 0.001, pvalue = 0.038, MODULE_313\n",
      "Didn't pass (cutoff=0.0050): cat 571, mi = 0.001, pvalue = 0.035, MODULE_493\n",
      "Didn't pass (cutoff=0.0050): cat 322, mi = 0.001, pvalue = 0.050, MODULE_184\n",
      "Didn't pass (cutoff=0.0050): cat 359, mi = 0.001, pvalue = 0.052, MODULE_221\n",
      "Didn't pass (cutoff=0.0050): cat 475, mi = 0.001, pvalue = 0.028, MODULE_360\n",
      "Didn't pass (cutoff=0.0050): cat 753, mi = 0.001, pvalue = 0.051, MORF_LCAT\n",
      "Didn't pass (cutoff=0.0050): cat 635, mi = 0.001, pvalue = 0.049, MODULE_71\n",
      "Didn't pass (cutoff=0.0050): cat 175, mi = 0.001, pvalue = 0.066, GNF2_MSN\n",
      "Didn't pass (cutoff=0.0050): cat 284, mi = 0.001, pvalue = 0.016, MODULE_148\n",
      "Didn't pass (cutoff=0.0050): cat 169, mi = 0.001, pvalue = 0.067, GNF2_MLF1\n",
      "Total number of categories evaluated = 312.\n",
      "Number of categories that passed the tests = 20\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM2_1.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM2_1.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 1, 4, 0.0157964636787757\n",
      "best merge is 1, 14, 0.0222453707487008\n",
      "best merge is 2, 11, 0.0280637733240884\n",
      "best merge is 1, 3, 0.0320301001739722\n",
      "best merge is 8, 16, 0.0771212789602927\n",
      "best merge is 8, 10, 0.100861509431174\n",
      "best merge is 15, 17, 0.14322764475213\n",
      "best merge is 1, 8, 0.171195117399346\n",
      "best merge is 5, 6, 0.194305607125101\n",
      "best merge is 7, 18, 0.221051177227887\n",
      "best merge is 2, 12, 0.226866921990783\n",
      "best merge is 2, 9, 0.26259702929608\n",
      "best merge is 2, 5, 0.348844974176164\n",
      "best merge is 1, 7, 0.505787747282733\n",
      "best merge is 2, 13, 0.522470988023535\n",
      "best merge is 2, 19, 0.60389404988589\n",
      "order classes\n",
      "9 - 0.537264353991717\n",
      "7 - 0.861746942725309\n",
      "2 - 0.71613822376065\n",
      "1 4 14 3 8 16 10 7 18\n",
      "0\n",
      "2 11 12 9 5 6 13\n",
      "15 17\n",
      "19\n",
      "Done.Start drawing\n",
      "70.09\t0.00\n",
      "MODULE_79\n",
      "GNF2_CDH11\n",
      "GCM_DENR\n",
      "MORF_TPR\n",
      "GCM_RAB10\n",
      "GNF2_KPNB1\n",
      "MODULE_21\n",
      "MORF_RAD23B\n",
      "GNF2_DDX5\n",
      "MODULE_22\n",
      "MODULE_89\n",
      "MORF_NPM1\n",
      "GNF2_CDC20\n",
      "MODULE_180\n",
      "GNF2_MYL2\n",
      "MODULE_297\n",
      "MODULE_92\n",
      "GNF2_EGFR\n",
      "GNF2_MAPT\n",
      "MODULE_111\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM2_1.csv_PAGE/filt_FC_LVM2_1.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM2_1.csv_PAGE/filt_FC_LVM2_1.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM2_1.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM2_1.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 1, 4, 0.0157964636787757\n",
      "best merge is 1, 14, 0.0222453707487008\n",
      "best merge is 2, 11, 0.0280637733240884\n",
      "best merge is 1, 3, 0.0320301001739722\n",
      "best merge is 8, 16, 0.0771212789602927\n",
      "best merge is 8, 10, 0.100861509431174\n",
      "best merge is 15, 17, 0.14322764475213\n",
      "best merge is 1, 8, 0.171195117399346\n",
      "best merge is 5, 6, 0.194305607125101\n",
      "best merge is 7, 18, 0.221051177227887\n",
      "best merge is 2, 12, 0.226866921990783\n",
      "best merge is 2, 9, 0.26259702929608\n",
      "best merge is 2, 5, 0.348844974176164\n",
      "best merge is 1, 7, 0.505787747282733\n",
      "best merge is 2, 13, 0.522470988023535\n",
      "best merge is 2, 19, 0.60389404988589\n",
      "order classes\n",
      "9 - 0.537264353991717\n",
      "7 - 0.861746942725309\n",
      "2 - 0.71613822376065\n",
      "1 4 14 3 8 16 10 7 18\n",
      "0\n",
      "2 11 12 9 5 6 13\n",
      "15 17\n",
      "19\n",
      "Done.-19.784277357683\t42\t0\n",
      "-19.5056356113568\t42\t0.431445284634042\n",
      "-19.4037665858182\t42\t0.557283492652304\n",
      "-19.3318590383792\t42\t0.635183335711228\n",
      "-19.2689399343701\t42\t0.701098587530318\n",
      "-19.203024682551\t42\t0.761021543729491\n",
      "-19.1221286916821\t42\t0.832929091168498\n",
      "-18.9902981880439\t42\t0.922813525467256\n",
      "1.54829504922243\t42\t1.09659009844486\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM2_1.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM1_4.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c4.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c4/human_ensembl_msigdb_c4.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM1_4.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 858\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t858\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 858\n",
      "cat 705, mi = 0.00763277, MORF_DAP3\n",
      "cat 831, mi = 0.00717723, MORF_SKP1A\n",
      "cat 131, mi = 0.00694120, GNF2_EIF3S6\n",
      "cat 847, mi = 0.00680069, MORF_TPT1\n",
      "cat 774, mi = 0.00651034, MORF_NPM1\n",
      "cat 733, mi = 0.00647360, MORF_G22P1\n",
      "cat 783, mi = 0.00640914, MORF_PPP1CA\n",
      "cat 83, mi = 0.00640171, GCM_TPT1\n",
      "cat 701, mi = 0.00638947, MORF_CTBP1\n",
      "cat 811, mi = 0.00634032, MORF_RAD21\n",
      "cat 797, mi = 0.00611293, MORF_PSMC1\n",
      "cat 813, mi = 0.00584317, MORF_RAD23B\n",
      "cat 806, mi = 0.00579497, MORF_RAB1A\n",
      "cat 667, mi = 0.00565946, MORF_ACTG1\n",
      "cat 794, mi = 0.00553990, MORF_PRKAR1A\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 705, mi = 0.008, pvalue = 0.000, MORF_DAP3\n",
      "Passed: cat 131, mi = 0.007, pvalue = 0.000, GNF2_EIF3S6\n",
      "Passed: cat 790, mi = 0.003, pvalue = 0.000, MORF_PPP6C\n",
      "Passed: cat 318, mi = 0.003, pvalue = 0.000, MODULE_180\n",
      "Passed: cat 421, mi = 0.003, pvalue = 0.000, MODULE_297\n",
      "Passed: cat 67, mi = 0.003, pvalue = 0.000, GCM_RAB10\n",
      "Passed: cat 132, mi = 0.003, pvalue = 0.000, GNF2_ELAC2\n",
      "Passed: cat 657, mi = 0.002, pvalue = 0.000, MODULE_92\n",
      "Passed: cat 162, mi = 0.002, pvalue = 0.000, GNF2_MAPT\n",
      "Passed: cat 294, mi = 0.002, pvalue = 0.000, MODULE_157\n",
      "Passed: cat 21, mi = 0.002, pvalue = 0.000, GCM_CHUK\n",
      "Passed: cat 442, mi = 0.002, pvalue = 0.001, MODULE_321\n",
      "Passed: cat 502, mi = 0.002, pvalue = 0.001, MODULE_399\n",
      "Passed: cat 151, mi = 0.002, pvalue = 0.000, GNF2_IGFBP1\n",
      "Didn't pass (cutoff=0.0050): cat 642, mi = 0.001, pvalue = 0.007, MODULE_79\n",
      "Didn't pass (cutoff=0.0050): cat 356, mi = 0.001, pvalue = 0.007, MODULE_219\n",
      "Didn't pass (cutoff=0.0050): cat 140, mi = 0.001, pvalue = 0.008, GNF2_GSTM1\n",
      "Didn't pass (cutoff=0.0050): cat 737, mi = 0.001, pvalue = 0.008, MORF_GSPT1\n",
      "Didn't pass (cutoff=0.0050): cat 307, mi = 0.001, pvalue = 0.009, MODULE_170\n",
      "Didn't pass (cutoff=0.0050): cat 262, mi = 0.001, pvalue = 0.012, MODULE_125\n",
      "Didn't pass (cutoff=0.0050): cat 265, mi = 0.001, pvalue = 0.011, MODULE_128\n",
      "Didn't pass (cutoff=0.0050): cat 484, mi = 0.001, pvalue = 0.014, MODULE_372\n",
      "Didn't pass (cutoff=0.0050): cat 159, mi = 0.001, pvalue = 0.014, GNF2_LCAT\n",
      "Didn't pass (cutoff=0.0050): cat 226, mi = 0.001, pvalue = 0.018, GNF2_TST\n",
      "Didn't pass (cutoff=0.0050): cat 148, mi = 0.001, pvalue = 0.019, GNF2_HPX\n",
      "Didn't pass (cutoff=0.0050): cat 389, mi = 0.001, pvalue = 0.016, MODULE_257\n",
      "Passed: cat 327, mi = 0.001, pvalue = 0.004, MODULE_190\n",
      "Didn't pass (cutoff=0.0050): cat 367, mi = 0.001, pvalue = 0.027, MODULE_234\n",
      "Didn't pass (cutoff=0.0050): cat 147, mi = 0.001, pvalue = 0.025, GNF2_HPN\n",
      "Didn't pass (cutoff=0.0050): cat 348, mi = 0.001, pvalue = 0.021, MODULE_21\n",
      "Didn't pass (cutoff=0.0050): cat 51, mi = 0.001, pvalue = 0.034, GCM_MSN\n",
      "Didn't pass (cutoff=0.0050): cat 459, mi = 0.001, pvalue = 0.031, MODULE_340\n",
      "Didn't pass (cutoff=0.0050): cat 823, mi = 0.001, pvalue = 0.025, MORF_RFC1\n",
      "Passed: cat 284, mi = 0.001, pvalue = 0.005, MODULE_148\n",
      "Didn't pass (cutoff=0.0050): cat 410, mi = 0.001, pvalue = 0.034, MODULE_285\n",
      "Didn't pass (cutoff=0.0050): cat 283, mi = 0.001, pvalue = 0.026, MODULE_147\n",
      "Didn't pass (cutoff=0.0050): cat 567, mi = 0.001, pvalue = 0.019, MODULE_489\n",
      "Didn't pass (cutoff=0.0050): cat 177, mi = 0.001, pvalue = 0.015, GNF2_MYL2\n",
      "Didn't pass (cutoff=0.0050): cat 854, mi = 0.001, pvalue = 0.042, MORF_WNT1\n",
      "Didn't pass (cutoff=0.0050): cat 471, mi = 0.001, pvalue = 0.031, MODULE_356\n",
      "Didn't pass (cutoff=0.0050): cat 331, mi = 0.001, pvalue = 0.029, MODULE_195\n",
      "Didn't pass (cutoff=0.0050): cat 746, mi = 0.001, pvalue = 0.032, MORF_IL4\n",
      "Didn't pass (cutoff=0.0050): cat 644, mi = 0.001, pvalue = 0.054, MODULE_80\n",
      "Didn't pass (cutoff=0.0050): cat 59, mi = 0.001, pvalue = 0.053, GCM_PPM1D\n",
      "Didn't pass (cutoff=0.0050): cat 119, mi = 0.001, pvalue = 0.055, GNF2_CEBPA\n",
      "Didn't pass (cutoff=0.0050): cat 115, mi = 0.001, pvalue = 0.053, GNF2_CDC27\n",
      "Didn't pass (cutoff=0.0050): cat 428, mi = 0.001, pvalue = 0.069, MODULE_305\n",
      "Didn't pass (cutoff=0.0050): cat 549, mi = 0.001, pvalue = 0.065, MODULE_458\n",
      "Didn't pass (cutoff=0.0050): cat 609, mi = 0.001, pvalue = 0.063, MODULE_56\n",
      "Didn't pass (cutoff=0.0050): cat 154, mi = 0.001, pvalue = 0.065, GNF2_ITGAL\n",
      "Didn't pass (cutoff=0.0050): cat 141, mi = 0.001, pvalue = 0.069, GNF2_H2AFX\n",
      "Didn't pass (cutoff=0.0050): cat 581, mi = 0.001, pvalue = 0.072, MODULE_51\n",
      "Didn't pass (cutoff=0.0050): cat 684, mi = 0.001, pvalue = 0.064, MORF_BUB1\n",
      "Didn't pass (cutoff=0.0050): cat 789, mi = 0.001, pvalue = 0.056, MORF_PPP5C\n",
      "Total number of categories evaluated = 286.\n",
      "Number of categories that passed the tests = 16\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM1_4.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM1_4.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 7, 13, 0.0247894462033673\n",
      "best merge is 4, 8, 0.0398240577638597\n",
      "best merge is 4, 9, 0.0538847308453447\n",
      "best merge is 3, 4, 0.0672806335586365\n",
      "best merge is 3, 7, 0.0761961232124531\n",
      "best merge is 0, 2, 0.089907854691983\n",
      "best merge is 11, 12, 0.136220118884992\n",
      "best merge is 0, 5, 0.196550522290092\n",
      "best merge is 0, 10, 0.328019363744152\n",
      "best merge is 11, 14, 0.340034668852278\n",
      "best merge is 3, 15, 0.371345352533281\n",
      "best merge is 0, 6, 0.420939005358068\n",
      "order classes\n",
      "4 - 0.733533495252312\n",
      "7 - 0.299649303065482\n",
      "3 - 0.967454170772797\n",
      "3 4 8 9 7 13 15\n",
      "1\n",
      "0 2 5 10\n",
      "6\n",
      "11 12 14\n",
      "Done.Start drawing\n",
      "26.31\t0.00\n",
      "MODULE_321\n",
      "MODULE_399\n",
      "MODULE_190\n",
      "MORF_DAP3\n",
      "MORF_PPP6C\n",
      "GNF2_ELAC2\n",
      "GCM_CHUK\n",
      "GCM_RAB10\n",
      "GNF2_EIF3S6\n",
      "MODULE_180\n",
      "MODULE_297\n",
      "GNF2_MAPT\n",
      "MODULE_157\n",
      "MODULE_92\n",
      "GNF2_IGFBP1\n",
      "MODULE_148\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM1_4.csv_PAGE/filt_FC_LVM1_4.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM1_4.csv_PAGE/filt_FC_LVM1_4.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM1_4.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM1_4.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 7, 13, 0.0247894462033673\n",
      "best merge is 4, 8, 0.0398240577638597\n",
      "best merge is 4, 9, 0.0538847308453447\n",
      "best merge is 3, 4, 0.0672806335586365\n",
      "best merge is 3, 7, 0.0761961232124531\n",
      "best merge is 0, 2, 0.089907854691983\n",
      "best merge is 11, 12, 0.136220118884992\n",
      "best merge is 0, 5, 0.196550522290092\n",
      "best merge is 0, 10, 0.328019363744152\n",
      "best merge is 11, 14, 0.340034668852278\n",
      "best merge is 3, 15, 0.371345352533281\n",
      "best merge is 0, 6, 0.420939005358068\n",
      "order classes\n",
      "4 - 0.733533495252312\n",
      "7 - 0.299649303065482\n",
      "3 - 0.967454170772797\n",
      "3 4 8 9 7 13 15\n",
      "1\n",
      "0 2 5 10\n",
      "6\n",
      "11 12 14\n",
      "Done.-19.5290763968073\t42\t0\n",
      "-18.9064994298746\t42\t0.941847206385405\n",
      "-18.6590649942987\t42\t1.24515393386545\n",
      "-18.4754846066135\t42\t1.43671607753706\n",
      "-18.3158494868871\t42\t1.61231470923603\n",
      "-18.1641961231471\t42\t1.75598631698974\n",
      "-17.9806157354618\t42\t1.91562143671608\n",
      "-17.709236031927\t42\t2.12314709236032\n",
      "2.22919042189282\t42\t2.45838084378563\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM1_4.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM2_2.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c4.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c4/human_ensembl_msigdb_c4.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM2_2.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 858\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t858\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 858\n",
      "cat 774, mi = 0.01079430, MORF_NPM1\n",
      "cat 667, mi = 0.00994122, MORF_ACTG1\n",
      "cat 772, mi = 0.00899582, MORF_NME2\n",
      "cat 131, mi = 0.00888110, GNF2_EIF3S6\n",
      "cat 847, mi = 0.00886568, MORF_TPT1\n",
      "cat 83, mi = 0.00784895, GCM_TPT1\n",
      "cat 56, mi = 0.00750909, GCM_NPM1\n",
      "cat 783, mi = 0.00671397, MORF_PPP1CA\n",
      "cat 705, mi = 0.00666627, MORF_DAP3\n",
      "cat 831, mi = 0.00635904, MORF_SKP1A\n",
      "cat 715, mi = 0.00578344, MORF_EIF3S6\n",
      "cat 811, mi = 0.00547245, MORF_RAD21\n",
      "cat 6, mi = 0.00536349, GCM_ACTG1\n",
      "cat 217, mi = 0.00535770, GNF2_ST13\n",
      "cat 721, mi = 0.00529723, MORF_ERH\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 774, mi = 0.011, pvalue = 0.000, MORF_NPM1\n",
      "Passed: cat 813, mi = 0.005, pvalue = 0.000, MORF_RAD23B\n",
      "Passed: cat 318, mi = 0.005, pvalue = 0.000, MODULE_180\n",
      "Passed: cat 421, mi = 0.003, pvalue = 0.000, MODULE_297\n",
      "Passed: cat 247, mi = 0.003, pvalue = 0.000, MODULE_111\n",
      "Passed: cat 67, mi = 0.003, pvalue = 0.000, GCM_RAB10\n",
      "Passed: cat 184, mi = 0.003, pvalue = 0.000, GNF2_PCNA\n",
      "Passed: cat 357, mi = 0.003, pvalue = 0.000, MODULE_22\n",
      "Passed: cat 657, mi = 0.002, pvalue = 0.000, MODULE_92\n",
      "Passed: cat 158, mi = 0.002, pvalue = 0.000, GNF2_KPNB1\n",
      "Passed: cat 162, mi = 0.002, pvalue = 0.000, GNF2_MAPT\n",
      "Passed: cat 177, mi = 0.002, pvalue = 0.000, GNF2_MYL2\n",
      "Passed: cat 126, mi = 0.002, pvalue = 0.001, GNF2_DDX5\n",
      "Passed: cat 392, mi = 0.002, pvalue = 0.001, MODULE_261\n",
      "Passed: cat 846, mi = 0.002, pvalue = 0.001, MORF_TPR\n",
      "Passed: cat 151, mi = 0.002, pvalue = 0.000, GNF2_IGFBP1\n",
      "Passed: cat 377, mi = 0.002, pvalue = 0.001, MODULE_243\n",
      "Passed: cat 115, mi = 0.002, pvalue = 0.001, GNF2_CDC27\n",
      "Passed: cat 544, mi = 0.002, pvalue = 0.002, MODULE_453\n",
      "Passed: cat 77, mi = 0.002, pvalue = 0.002, GCM_SMO\n",
      "Passed: cat 264, mi = 0.001, pvalue = 0.002, MODULE_127\n",
      "Passed: cat 442, mi = 0.001, pvalue = 0.003, MODULE_321\n",
      "Passed: cat 99, mi = 0.001, pvalue = 0.004, GNF2_CASP1\n",
      "Didn't pass (cutoff=0.0050): cat 175, mi = 0.001, pvalue = 0.006, GNF2_MSN\n",
      "Didn't pass (cutoff=0.0050): cat 148, mi = 0.001, pvalue = 0.016, GNF2_HPX\n",
      "Didn't pass (cutoff=0.0050): cat 244, mi = 0.001, pvalue = 0.014, MODULE_109\n",
      "Didn't pass (cutoff=0.0050): cat 116, mi = 0.001, pvalue = 0.011, GNF2_CDH11\n",
      "Didn't pass (cutoff=0.0050): cat 147, mi = 0.001, pvalue = 0.018, GNF2_HPN\n",
      "Didn't pass (cutoff=0.0050): cat 502, mi = 0.001, pvalue = 0.027, MODULE_399\n",
      "Passed: cat 284, mi = 0.001, pvalue = 0.003, MODULE_148\n",
      "Didn't pass (cutoff=0.0050): cat 13, mi = 0.001, pvalue = 0.028, GCM_BCL2L1\n",
      "Didn't pass (cutoff=0.0050): cat 610, mi = 0.001, pvalue = 0.022, MODULE_560\n",
      "Didn't pass (cutoff=0.0050): cat 340, mi = 0.001, pvalue = 0.042, MODULE_202\n",
      "Didn't pass (cutoff=0.0050): cat 156, mi = 0.001, pvalue = 0.051, GNF2_JAK1\n",
      "Didn't pass (cutoff=0.0050): cat 455, mi = 0.001, pvalue = 0.054, MODULE_333\n",
      "Didn't pass (cutoff=0.0050): cat 731, mi = 0.001, pvalue = 0.050, MORF_FRK\n",
      "Didn't pass (cutoff=0.0050): cat 747, mi = 0.001, pvalue = 0.054, MORF_IL9\n",
      "Didn't pass (cutoff=0.0050): cat 567, mi = 0.001, pvalue = 0.053, MODULE_489\n",
      "Didn't pass (cutoff=0.0050): cat 573, mi = 0.001, pvalue = 0.064, MODULE_497\n",
      "Didn't pass (cutoff=0.0050): cat 459, mi = 0.001, pvalue = 0.063, MODULE_340\n",
      "Didn't pass (cutoff=0.0050): cat 159, mi = 0.001, pvalue = 0.093, GNF2_LCAT\n",
      "Didn't pass (cutoff=0.0050): cat 453, mi = 0.001, pvalue = 0.074, MODULE_331\n",
      "Didn't pass (cutoff=0.0050): cat 680, mi = 0.001, pvalue = 0.093, MORF_BMI1\n",
      "Didn't pass (cutoff=0.0050): cat 519, mi = 0.001, pvalue = 0.177, MODULE_417\n",
      "Didn't pass (cutoff=0.0050): cat 570, mi = 0.001, pvalue = 0.150, MODULE_492\n",
      "Didn't pass (cutoff=0.0050): cat 140, mi = 0.001, pvalue = 0.142, GNF2_GSTM1\n",
      "Didn't pass (cutoff=0.0050): cat 507, mi = 0.001, pvalue = 0.083, MODULE_402\n",
      "Didn't pass (cutoff=0.0050): cat 226, mi = 0.001, pvalue = 0.148, GNF2_TST\n",
      "Didn't pass (cutoff=0.0050): cat 367, mi = 0.001, pvalue = 0.199, MODULE_234\n",
      "Didn't pass (cutoff=0.0050): cat 848, mi = 0.001, pvalue = 0.203, MORF_TTN\n",
      "Total number of categories evaluated = 367.\n",
      "Number of categories that passed the tests = 24\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM2_2.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM2_2.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 2, 3, 0.0246078509201917\n",
      "best merge is 1, 14, 0.0414212921004876\n",
      "best merge is 2, 23, 0.0420631844975333\n",
      "best merge is 4, 15, 0.0664088588750256\n",
      "best merge is 2, 11, 0.0701405798055674\n",
      "best merge is 2, 4, 0.0810504554549348\n",
      "best merge is 12, 18, 0.100204298074565\n",
      "best merge is 2, 8, 0.102489428717011\n",
      "best merge is 1, 5, 0.109531250689071\n",
      "best merge is 7, 16, 0.113877946638495\n",
      "best merge is 20, 21, 0.147840289354866\n",
      "best merge is 1, 9, 0.168120850310038\n",
      "best merge is 2, 10, 0.21312028002057\n",
      "best merge is 6, 20, 0.215098544278217\n",
      "best merge is 0, 7, 0.294563846042624\n",
      "best merge is 1, 13, 0.344308976836212\n",
      "best merge is 1, 12, 0.463376010472991\n",
      "best merge is 6, 22, 0.562695856468472\n",
      "best merge is 1, 19, 0.625132098800576\n",
      "best merge is 0, 17, 0.636080184921755\n",
      "order classes\n",
      "3 - 1.24978780217554\n",
      "8 - 1.0548723137027\n",
      "8 - 0.294445030714924\n",
      "4 - 1.69959371048754\n",
      "2 3 23 11 4 15 8 10\n",
      "1 14 5 9 13 12 18 19\n",
      "0 7 16\n",
      "6 20 21 22\n",
      "17\n",
      "Done.Start drawing\n",
      "101.03\t0.00\n",
      "GNF2_PCNA\n",
      "MODULE_127\n",
      "MODULE_321\n",
      "GNF2_CASP1\n",
      "GNF2_DDX5\n",
      "MODULE_453\n",
      "GCM_RAB10\n",
      "MORF_RAD23B\n",
      "MORF_TPR\n",
      "GNF2_CDC27\n",
      "GNF2_KPNB1\n",
      "GCM_SMO\n",
      "MODULE_22\n",
      "MODULE_243\n",
      "MODULE_261\n",
      "MORF_NPM1\n",
      "MODULE_180\n",
      "MODULE_297\n",
      "MODULE_148\n",
      "GNF2_MYL2\n",
      "MODULE_111\n",
      "GNF2_IGFBP1\n",
      "MODULE_92\n",
      "GNF2_MAPT\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM2_2.csv_PAGE/filt_FC_LVM2_2.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM2_2.csv_PAGE/filt_FC_LVM2_2.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM2_2.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM2_2.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 2, 3, 0.0246078509201917\n",
      "best merge is 1, 14, 0.0414212921004876\n",
      "best merge is 2, 23, 0.0420631844975333\n",
      "best merge is 4, 15, 0.0664088588750256\n",
      "best merge is 2, 11, 0.0701405798055674\n",
      "best merge is 2, 4, 0.0810504554549348\n",
      "best merge is 12, 18, 0.100204298074565\n",
      "best merge is 2, 8, 0.102489428717011\n",
      "best merge is 1, 5, 0.109531250689071\n",
      "best merge is 7, 16, 0.113877946638495\n",
      "best merge is 20, 21, 0.147840289354866\n",
      "best merge is 1, 9, 0.168120850310038\n",
      "best merge is 2, 10, 0.21312028002057\n",
      "best merge is 6, 20, 0.215098544278217\n",
      "best merge is 0, 7, 0.294563846042624\n",
      "best merge is 1, 13, 0.344308976836212\n",
      "best merge is 1, 12, 0.463376010472991\n",
      "best merge is 6, 22, 0.562695856468472\n",
      "best merge is 1, 19, 0.625132098800576\n",
      "best merge is 0, 17, 0.636080184921755\n",
      "order classes\n",
      "3 - 1.24978780217554\n",
      "8 - 1.0548723137027\n",
      "8 - 0.294445030714924\n",
      "4 - 1.69959371048754\n",
      "2 3 23 11 4 15 8 10\n",
      "1 14 5 9 13 12 18 19\n",
      "0 7 16\n",
      "6 20 21 22\n",
      "17\n",
      "Done.-19.8544986637632\t42\t0\n",
      "-19.6570325645848\t42\t0.291002672473523\n",
      "-19.5759675343957\t42\t0.394932198356924\n",
      "-19.5219241809364\t42\t0.453132732851628\n",
      "-19.4720380085123\t42\t0.503018905275661\n",
      "-19.417994655053\t42\t0.552905077699693\n",
      "-19.3514797584876\t42\t0.611105612194398\n",
      "-19.2454716420865\t42\t0.685934870830446\n",
      "1.41156092249827\t42\t0.823121844996536\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM2_2.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM2_4.csv --species=human_ensembl_msigdb_c4 --exptype=continuous --ebins=9\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM2_3.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c4.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c4/human_ensembl_msigdb_c4.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM2_3.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 858\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t858\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 858\n",
      "cat 105, mi = 0.01077756, GNF2_CCNB2\n",
      "cat 114, mi = 0.01072091, GNF2_CDC20\n",
      "cat 774, mi = 0.01034158, MORF_NPM1\n",
      "cat 104, mi = 0.00987647, GNF2_CCNA2\n",
      "cat 113, mi = 0.00947784, GNF2_CDC2\n",
      "cat 131, mi = 0.00930164, GNF2_EIF3S6\n",
      "cat 121, mi = 0.00841172, GNF2_CENPF\n",
      "cat 667, mi = 0.00803152, MORF_ACTG1\n",
      "cat 772, mi = 0.00796108, MORF_NME2\n",
      "cat 847, mi = 0.00783630, MORF_TPT1\n",
      "cat 83, mi = 0.00780459, GCM_TPT1\n",
      "cat 146, mi = 0.00761891, GNF2_HMMR\n",
      "cat 120, mi = 0.00756758, GNF2_CENPE\n",
      "cat 56, mi = 0.00714455, GCM_NPM1\n",
      "cat 184, mi = 0.00652824, GNF2_PCNA\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 105, mi = 0.011, pvalue = 0.000, GNF2_CCNB2\n",
      "Passed: cat 774, mi = 0.010, pvalue = 0.000, MORF_NPM1\n",
      "Passed: cat 806, mi = 0.006, pvalue = 0.000, MORF_RAB1A\n",
      "Passed: cat 318, mi = 0.005, pvalue = 0.000, MODULE_180\n",
      "Passed: cat 625, mi = 0.005, pvalue = 0.000, MODULE_62\n",
      "Passed: cat 67, mi = 0.005, pvalue = 0.000, GCM_RAB10\n",
      "Passed: cat 158, mi = 0.003, pvalue = 0.000, GNF2_KPNB1\n",
      "Passed: cat 421, mi = 0.003, pvalue = 0.000, MODULE_297\n",
      "Passed: cat 247, mi = 0.003, pvalue = 0.000, MODULE_111\n",
      "Passed: cat 129, mi = 0.002, pvalue = 0.000, GNF2_DNM1\n",
      "Passed: cat 657, mi = 0.002, pvalue = 0.000, MODULE_92\n",
      "Passed: cat 151, mi = 0.002, pvalue = 0.000, GNF2_IGFBP1\n",
      "Passed: cat 819, mi = 0.002, pvalue = 0.000, MORF_RAP1A\n",
      "Passed: cat 656, mi = 0.002, pvalue = 0.000, MODULE_91\n",
      "Passed: cat 13, mi = 0.002, pvalue = 0.000, GCM_BCL2L1\n",
      "Passed: cat 320, mi = 0.002, pvalue = 0.001, MODULE_182\n",
      "Passed: cat 442, mi = 0.002, pvalue = 0.000, MODULE_321\n",
      "Passed: cat 147, mi = 0.002, pvalue = 0.002, GNF2_HPN\n",
      "Passed: cat 126, mi = 0.001, pvalue = 0.003, GNF2_DDX5\n",
      "Passed: cat 177, mi = 0.001, pvalue = 0.000, GNF2_MYL2\n",
      "Didn't pass (cutoff=0.0050): cat 237, mi = 0.001, pvalue = 0.012, MODULE_102\n",
      "Didn't pass (cutoff=0.0050): cat 544, mi = 0.001, pvalue = 0.013, MODULE_453\n",
      "Didn't pass (cutoff=0.0050): cat 116, mi = 0.001, pvalue = 0.007, GNF2_CDH11\n",
      "Didn't pass (cutoff=0.0050): cat 364, mi = 0.001, pvalue = 0.016, MODULE_229\n",
      "Didn't pass (cutoff=0.0050): cat 303, mi = 0.001, pvalue = 0.017, MODULE_166\n",
      "Didn't pass (cutoff=0.0050): cat 389, mi = 0.001, pvalue = 0.016, MODULE_257\n",
      "Didn't pass (cutoff=0.0050): cat 531, mi = 0.001, pvalue = 0.023, MODULE_432\n",
      "Didn't pass (cutoff=0.0050): cat 99, mi = 0.001, pvalue = 0.026, GNF2_CASP1\n",
      "Didn't pass (cutoff=0.0050): cat 103, mi = 0.001, pvalue = 0.041, GNF2_CCNA1\n",
      "Didn't pass (cutoff=0.0050): cat 111, mi = 0.001, pvalue = 0.030, GNF2_CD7\n",
      "Didn't pass (cutoff=0.0050): cat 573, mi = 0.001, pvalue = 0.029, MODULE_497\n",
      "Didn't pass (cutoff=0.0050): cat 439, mi = 0.001, pvalue = 0.048, MODULE_318\n",
      "Didn't pass (cutoff=0.0050): cat 459, mi = 0.001, pvalue = 0.034, MODULE_340\n",
      "Didn't pass (cutoff=0.0050): cat 807, mi = 0.001, pvalue = 0.042, MORF_RAB3A\n",
      "Didn't pass (cutoff=0.0050): cat 453, mi = 0.001, pvalue = 0.032, MODULE_331\n",
      "Didn't pass (cutoff=0.0050): cat 188, mi = 0.001, pvalue = 0.047, GNF2_PTPN4\n",
      "Didn't pass (cutoff=0.0050): cat 189, mi = 0.001, pvalue = 0.053, GNF2_PTPN6\n",
      "Didn't pass (cutoff=0.0050): cat 264, mi = 0.001, pvalue = 0.039, MODULE_127\n",
      "Didn't pass (cutoff=0.0050): cat 183, mi = 0.001, pvalue = 0.051, GNF2_PCAF\n",
      "Didn't pass (cutoff=0.0050): cat 370, mi = 0.001, pvalue = 0.044, MODULE_237\n",
      "Total number of categories evaluated = 320.\n",
      "Number of categories that passed the tests = 20\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM2_3.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM2_3.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 7, 9, 0.00413847100516096\n",
      "best merge is 3, 7, 0.0160501449404617\n",
      "best merge is 3, 11, 0.0284477494988128\n",
      "best merge is 3, 19, 0.0609072621966528\n",
      "best merge is 3, 10, 0.0770935844956131\n",
      "best merge is 2, 18, 0.0923004899317038\n",
      "best merge is 3, 8, 0.103213723939634\n",
      "best merge is 2, 5, 0.147108679210706\n",
      "best merge is 1, 4, 0.157364596399615\n",
      "best merge is 0, 16, 0.193642175180322\n",
      "best merge is 3, 17, 0.228608768648332\n",
      "best merge is 2, 6, 0.244694043927602\n",
      "best merge is 1, 13, 0.295326874252682\n",
      "best merge is 2, 12, 0.299417465610178\n",
      "best merge is 2, 14, 0.413887320500086\n",
      "best merge is 2, 15, 0.478544895766401\n",
      "order classes\n",
      "2 - 0.968210875901609\n",
      "3 - 1.10817532578515\n",
      "6 - 0.812538230569826\n",
      "8 - 0.232048648814322\n",
      "3 7 9 11 19 10 8 17\n",
      "1 4 13\n",
      "2 18 5 6 12 14\n",
      "0 16\n",
      "15\n",
      "Done.Start drawing\n",
      "69.14\t0.00\n",
      "GNF2_CCNB2\n",
      "MODULE_321\n",
      "GNF2_DDX5\n",
      "MORF_RAP1A\n",
      "GCM_BCL2L1\n",
      "GCM_RAB10\n",
      "MODULE_182\n",
      "MORF_RAB1A\n",
      "GNF2_KPNB1\n",
      "MORF_NPM1\n",
      "MODULE_62\n",
      "MODULE_91\n",
      "MODULE_180\n",
      "MODULE_297\n",
      "GNF2_DNM1\n",
      "GNF2_IGFBP1\n",
      "GNF2_MYL2\n",
      "MODULE_92\n",
      "MODULE_111\n",
      "GNF2_HPN\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM2_3.csv_PAGE/filt_FC_LVM2_3.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM2_3.csv_PAGE/filt_FC_LVM2_3.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM2_3.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM2_3.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 7, 9, 0.00413847100516096\n",
      "best merge is 3, 7, 0.0160501449404617\n",
      "best merge is 3, 11, 0.0284477494988128\n",
      "best merge is 3, 19, 0.0609072621966528\n",
      "best merge is 3, 10, 0.0770935844956131\n",
      "best merge is 2, 18, 0.0923004899317038\n",
      "best merge is 3, 8, 0.103213723939634\n",
      "best merge is 2, 5, 0.147108679210706\n",
      "best merge is 1, 4, 0.157364596399615\n",
      "best merge is 0, 16, 0.193642175180322\n",
      "best merge is 3, 17, 0.228608768648332\n",
      "best merge is 2, 6, 0.244694043927602\n",
      "best merge is 1, 13, 0.295326874252682\n",
      "best merge is 2, 12, 0.299417465610178\n",
      "best merge is 2, 14, 0.413887320500086\n",
      "best merge is 2, 15, 0.478544895766401\n",
      "order classes\n",
      "2 - 0.968210875901609\n",
      "3 - 1.10817532578515\n",
      "6 - 0.812538230569826\n",
      "8 - 0.232048648814322\n",
      "3 7 9 11 19 10 8 17\n",
      "1 4 13\n",
      "2 18 5 6 12 14\n",
      "0 16\n",
      "15\n",
      "Done.-19.7843505929997\t42\t0\n",
      "-19.4715070870697\t42\t0.431298814000579\n",
      "-19.3166039919005\t42\t0.625687011859994\n",
      "-19.2133352617877\t42\t0.741105004339022\n",
      "-19.1191784784495\t42\t0.832224472085623\n",
      "-19.0189470639283\t42\t0.929418571015331\n",
      "-18.8974544402661\t42\t1.03268730112815\n",
      "-18.7000289268152\t42\t1.1724038183396\n",
      "1.71376916401504\t42\t1.42753832803008\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM2_3.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM2_5.csv --species=human_ensembl_msigdb_c4 --exptype=continuous --ebins=9\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM2_4.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c4.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c4/human_ensembl_msigdb_c4.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM2_4.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 858\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t858\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 858\n",
      "cat 104, mi = 0.01530259, GNF2_CCNA2\n",
      "cat 105, mi = 0.01377325, GNF2_CCNB2\n",
      "cat 121, mi = 0.01354535, GNF2_CENPF\n",
      "cat 113, mi = 0.01324634, GNF2_CDC2\n",
      "cat 114, mi = 0.01314230, GNF2_CDC20\n",
      "cat 184, mi = 0.01312406, GNF2_PCNA\n",
      "cat 201, mi = 0.01226256, GNF2_RRM1\n",
      "cat 146, mi = 0.01183330, GNF2_HMMR\n",
      "cat 210, mi = 0.01154118, GNF2_SMC4L1\n",
      "cat 166, mi = 0.01086212, GNF2_MCM4\n",
      "cat 135, mi = 0.01042539, GNF2_FEN1\n",
      "cat 774, mi = 0.01014049, MORF_NPM1\n",
      "cat 202, mi = 0.00999303, GNF2_RRM2\n",
      "cat 96, mi = 0.00979603, GNF2_BUB1B\n",
      "cat 123, mi = 0.00953527, GNF2_CKS2\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 104, mi = 0.015, pvalue = 0.000, GNF2_CCNA2\n",
      "Passed: cat 774, mi = 0.010, pvalue = 0.000, MORF_NPM1\n",
      "Passed: cat 813, mi = 0.004, pvalue = 0.000, MORF_RAD23B\n",
      "Passed: cat 421, mi = 0.003, pvalue = 0.000, MODULE_297\n",
      "Passed: cat 318, mi = 0.003, pvalue = 0.000, MODULE_180\n",
      "Passed: cat 67, mi = 0.003, pvalue = 0.000, GCM_RAB10\n",
      "Passed: cat 126, mi = 0.002, pvalue = 0.000, GNF2_DDX5\n",
      "Passed: cat 657, mi = 0.002, pvalue = 0.000, MODULE_92\n",
      "Passed: cat 694, mi = 0.002, pvalue = 0.000, MORF_CDC16\n",
      "Passed: cat 357, mi = 0.002, pvalue = 0.000, MODULE_22\n",
      "Passed: cat 149, mi = 0.002, pvalue = 0.000, GNF2_ICAM3\n",
      "Passed: cat 473, mi = 0.002, pvalue = 0.001, MODULE_358\n",
      "Passed: cat 642, mi = 0.002, pvalue = 0.001, MODULE_79\n",
      "Passed: cat 846, mi = 0.001, pvalue = 0.002, MORF_TPR\n",
      "Passed: cat 247, mi = 0.001, pvalue = 0.003, MODULE_111\n",
      "Passed: cat 237, mi = 0.001, pvalue = 0.002, MODULE_102\n",
      "Passed: cat 511, mi = 0.001, pvalue = 0.004, MODULE_408\n",
      "Passed: cat 222, mi = 0.001, pvalue = 0.005, GNF2_TM4SF2\n",
      "Didn't pass (cutoff=0.0050): cat 137, mi = 0.001, pvalue = 0.008, GNF2_FOS\n",
      "Didn't pass (cutoff=0.0050): cat 116, mi = 0.001, pvalue = 0.006, GNF2_CDH11\n",
      "Didn't pass (cutoff=0.0050): cat 244, mi = 0.001, pvalue = 0.007, MODULE_109\n",
      "Didn't pass (cutoff=0.0050): cat 571, mi = 0.001, pvalue = 0.011, MODULE_493\n",
      "Didn't pass (cutoff=0.0050): cat 453, mi = 0.001, pvalue = 0.012, MODULE_331\n",
      "Didn't pass (cutoff=0.0050): cat 322, mi = 0.001, pvalue = 0.018, MODULE_184\n",
      "Didn't pass (cutoff=0.0050): cat 359, mi = 0.001, pvalue = 0.026, MODULE_221\n",
      "Didn't pass (cutoff=0.0050): cat 419, mi = 0.001, pvalue = 0.017, MODULE_294\n",
      "Didn't pass (cutoff=0.0050): cat 363, mi = 0.001, pvalue = 0.017, MODULE_227\n",
      "Didn't pass (cutoff=0.0050): cat 118, mi = 0.001, pvalue = 0.028, GNF2_CDKN1C\n",
      "Didn't pass (cutoff=0.0050): cat 188, mi = 0.001, pvalue = 0.048, GNF2_PTPN4\n",
      "Didn't pass (cutoff=0.0050): cat 779, mi = 0.001, pvalue = 0.054, MORF_PDCD1\n",
      "Didn't pass (cutoff=0.0050): cat 379, mi = 0.001, pvalue = 0.063, MODULE_245\n",
      "Didn't pass (cutoff=0.0050): cat 157, mi = 0.001, pvalue = 0.072, GNF2_KISS1\n",
      "Didn't pass (cutoff=0.0050): cat 582, mi = 0.001, pvalue = 0.071, MODULE_511\n",
      "Didn't pass (cutoff=0.0050): cat 292, mi = 0.001, pvalue = 0.070, MODULE_155\n",
      "Didn't pass (cutoff=0.0050): cat 573, mi = 0.001, pvalue = 0.076, MODULE_497\n",
      "Didn't pass (cutoff=0.0050): cat 172, mi = 0.001, pvalue = 0.089, GNF2_MMP11\n",
      "Didn't pass (cutoff=0.0050): cat 495, mi = 0.001, pvalue = 0.105, MODULE_388\n",
      "Didn't pass (cutoff=0.0050): cat 368, mi = 0.001, pvalue = 0.074, MODULE_235\n",
      "Total number of categories evaluated = 331.\n",
      "Number of categories that passed the tests = 18\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM2_4.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM2_4.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 0, 4, 0.0205179566161131\n",
      "best merge is 0, 3, 0.0537049486028263\n",
      "best merge is 0, 17, 0.130172883040796\n",
      "best merge is 10, 12, 0.144985205741481\n",
      "best merge is 2, 13, 0.185334591575269\n",
      "best merge is 0, 14, 0.191315307710827\n",
      "best merge is 1, 15, 0.219773512861431\n",
      "best merge is 5, 6, 0.280412168579299\n",
      "best merge is 1, 8, 0.302470753091855\n",
      "best merge is 7, 10, 0.347476242096353\n",
      "best merge is 2, 9, 0.379721946613668\n",
      "best merge is 5, 16, 0.473891123145625\n",
      "best merge is 5, 11, 0.490417223918308\n",
      "best merge is 1, 2, 0.512001852146871\n",
      "order classes\n",
      "5 - 0.369707827330789\n",
      "3 - 1.09962002539352\n",
      "3 - 1.25970464640347\n",
      "4 - 1.51843842372683\n",
      "3 - 1.11991691991225\n",
      "0 4 3 17 14\n",
      "1 15 8\n",
      "2 13 9\n",
      "5 6 16 11\n",
      "7 10 12\n",
      "Done.Start drawing\n",
      "100.30\t0.00\n",
      "MODULE_92\n",
      "GNF2_ICAM3\n",
      "MODULE_79\n",
      "GCM_RAB10\n",
      "MODULE_408\n",
      "MORF_TPR\n",
      "GNF2_DDX5\n",
      "MODULE_358\n",
      "MORF_NPM1\n",
      "MORF_RAD23B\n",
      "MODULE_22\n",
      "MODULE_102\n",
      "MORF_CDC16\n",
      "GNF2_CCNA2\n",
      "MODULE_180\n",
      "MODULE_297\n",
      "GNF2_TM4SF2\n",
      "MODULE_111\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM2_4.csv_PAGE/filt_FC_LVM2_4.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM2_4.csv_PAGE/filt_FC_LVM2_4.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM2_4.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM2_4.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 0, 4, 0.0205179566161131\n",
      "best merge is 0, 3, 0.0537049486028263\n",
      "best merge is 0, 17, 0.130172883040796\n",
      "best merge is 10, 12, 0.144985205741481\n",
      "best merge is 2, 13, 0.185334591575269\n",
      "best merge is 0, 14, 0.191315307710827\n",
      "best merge is 1, 15, 0.219773512861431\n",
      "best merge is 5, 6, 0.280412168579299\n",
      "best merge is 1, 8, 0.302470753091855\n",
      "best merge is 7, 10, 0.347476242096353\n",
      "best merge is 2, 9, 0.379721946613668\n",
      "best merge is 5, 16, 0.473891123145625\n",
      "best merge is 5, 11, 0.490417223918308\n",
      "best merge is 1, 2, 0.512001852146871\n",
      "order classes\n",
      "5 - 0.369707827330789\n",
      "3 - 1.09962002539352\n",
      "3 - 1.25970464640347\n",
      "4 - 1.51843842372683\n",
      "3 - 1.11991691991225\n",
      "0 4 3 17 14\n",
      "1 15 8\n",
      "2 13 9\n",
      "5 6 16 11\n",
      "7 10 12\n",
      "Done.-19.8911266201396\t42\t0\n",
      "-19.7403788634098\t42\t0.217746759720837\n",
      "-19.6712861415753\t42\t0.301495513459621\n",
      "-19.621036889332\t42\t0.355932203389831\n",
      "-19.5728813559322\t42\t0.401994017946162\n",
      "-19.5163509471585\t42\t0.452243270189432\n",
      "-19.4451645064806\t42\t0.515054835493519\n",
      "-19.3321036889332\t42\t0.594616151545364\n",
      "1.37058823529411\t42\t0.741176470588235\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM2_4.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_P_LVM1_1.csv --species=human_ensembl_msigdb_c4 --exptype=continuous --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_P_LVM1_2.csv --species=human_ensembl_msigdb_c4 --exptype=continuous --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_P_LVM1_3.csv --species=human_ensembl_msigdb_c4 --exptype=continuous --ebins=9\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_P_LVM1_1.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c4.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c4/human_ensembl_msigdb_c4.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_P_LVM1_1.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 858\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t858\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 858\n",
      "cat 772, mi = 0.02337431, MORF_NME2\n",
      "cat 668, mi = 0.02291327, MORF_ANP32B\n",
      "cat 774, mi = 0.02290382, MORF_NPM1\n",
      "cat 667, mi = 0.01954792, MORF_ACTG1\n",
      "cat 783, mi = 0.01594008, MORF_PPP1CA\n",
      "cat 6, mi = 0.01565500, GCM_ACTG1\n",
      "cat 733, mi = 0.01506370, MORF_G22P1\n",
      "cat 831, mi = 0.01502231, MORF_SKP1A\n",
      "cat 811, mi = 0.01486655, MORF_RAD21\n",
      "cat 726, mi = 0.01478715, MORF_FBL\n",
      "cat 56, mi = 0.01455298, GCM_NPM1\n",
      "cat 847, mi = 0.01431994, MORF_TPT1\n",
      "cat 721, mi = 0.01431892, MORF_ERH\n",
      "cat 131, mi = 0.01429526, GNF2_EIF3S6\n",
      "cat 134, mi = 0.01358227, GNF2_FBL\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 772, mi = 0.023, pvalue = 0.000, MORF_NME2\n",
      "Passed: cat 738, mi = 0.011, pvalue = 0.000, MORF_HAT1\n",
      "Passed: cat 197, mi = 0.004, pvalue = 0.000, GNF2_RBBP6\n",
      "Passed: cat 292, mi = 0.003, pvalue = 0.000, MODULE_155\n",
      "Passed: cat 103, mi = 0.002, pvalue = 0.000, GNF2_CCNA1\n",
      "Passed: cat 372, mi = 0.002, pvalue = 0.001, MODULE_239\n",
      "Passed: cat 750, mi = 0.002, pvalue = 0.001, MORF_JAK3\n",
      "Passed: cat 516, mi = 0.002, pvalue = 0.001, MODULE_414\n",
      "Passed: cat 360, mi = 0.002, pvalue = 0.001, MODULE_222\n",
      "Passed: cat 162, mi = 0.001, pvalue = 0.004, GNF2_MAPT\n",
      "Didn't pass (cutoff=0.0050): cat 414, mi = 0.001, pvalue = 0.005, MODULE_289\n",
      "Didn't pass (cutoff=0.0050): cat 242, mi = 0.001, pvalue = 0.008, MODULE_107\n",
      "Didn't pass (cutoff=0.0050): cat 290, mi = 0.001, pvalue = 0.011, MODULE_153\n",
      "Didn't pass (cutoff=0.0050): cat 422, mi = 0.001, pvalue = 0.011, MODULE_298\n",
      "Didn't pass (cutoff=0.0050): cat 99, mi = 0.001, pvalue = 0.016, GNF2_CASP1\n",
      "Didn't pass (cutoff=0.0050): cat 362, mi = 0.001, pvalue = 0.019, MODULE_226\n",
      "Didn't pass (cutoff=0.0050): cat 98, mi = 0.001, pvalue = 0.025, GNF2_CARD15\n",
      "Didn't pass (cutoff=0.0050): cat 747, mi = 0.001, pvalue = 0.023, MORF_IL9\n",
      "Didn't pass (cutoff=0.0050): cat 77, mi = 0.001, pvalue = 0.021, GCM_SMO\n",
      "Didn't pass (cutoff=0.0050): cat 377, mi = 0.001, pvalue = 0.019, MODULE_243\n",
      "Didn't pass (cutoff=0.0050): cat 175, mi = 0.001, pvalue = 0.029, GNF2_MSN\n",
      "Didn't pass (cutoff=0.0050): cat 340, mi = 0.001, pvalue = 0.024, MODULE_202\n",
      "Didn't pass (cutoff=0.0050): cat 399, mi = 0.001, pvalue = 0.017, MODULE_272\n",
      "Didn't pass (cutoff=0.0050): cat 282, mi = 0.001, pvalue = 0.036, MODULE_146\n",
      "Didn't pass (cutoff=0.0050): cat 804, mi = 0.001, pvalue = 0.033, MORF_PTPRR\n",
      "Didn't pass (cutoff=0.0050): cat 308, mi = 0.001, pvalue = 0.028, MODULE_171\n",
      "Didn't pass (cutoff=0.0050): cat 521, mi = 0.001, pvalue = 0.039, MODULE_419\n",
      "Didn't pass (cutoff=0.0050): cat 390, mi = 0.001, pvalue = 0.043, MODULE_259\n",
      "Didn't pass (cutoff=0.0050): cat 610, mi = 0.001, pvalue = 0.020, MODULE_560\n",
      "Didn't pass (cutoff=0.0050): cat 165, mi = 0.001, pvalue = 0.037, GNF2_MCL1\n",
      "Total number of categories evaluated = 231.\n",
      "Number of categories that passed the tests = 10\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_P_LVM1_1.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_P_LVM1_1.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 0, 1, 0.0344916365039092\n",
      "best merge is 0, 7, 0.0742050408884882\n",
      "best merge is 2, 3, 0.0777598852248879\n",
      "best merge is 0, 8, 0.16504988877339\n",
      "best merge is 4, 9, 0.270585523710688\n",
      "best merge is 0, 5, 0.291684038164193\n",
      "order classes\n",
      "4 - 0.21189105768783\n",
      "2 - 0.19439971306222\n",
      "2 - 0.676463809276721\n",
      "2 3\n",
      "0 1 7 8\n",
      "4 9\n",
      "5\n",
      "6\n",
      "Done.Start drawing\n",
      "280.55\t0.00\n",
      "MORF_NME2\n",
      "MORF_HAT1\n",
      "MODULE_414\n",
      "MODULE_222\n",
      "MODULE_239\n",
      "GNF2_RBBP6\n",
      "MODULE_155\n",
      "MORF_JAK3\n",
      "GNF2_CCNA1\n",
      "GNF2_MAPT\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_P_LVM1_1.csv_PAGE/filt_FC_P_LVM1_1.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_P_LVM1_1.csv_PAGE/filt_FC_P_LVM1_1.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_P_LVM1_1.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_P_LVM1_1.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 0, 1, 0.0344916365039092\n",
      "best merge is 0, 7, 0.0742050408884882\n",
      "best merge is 2, 3, 0.0777598852248879\n",
      "best merge is 0, 8, 0.16504988877339\n",
      "best merge is 4, 9, 0.270585523710688\n",
      "best merge is 0, 5, 0.291684038164193\n",
      "order classes\n",
      "4 - 0.21189105768783\n",
      "2 - 0.19439971306222\n",
      "2 - 0.676463809276721\n",
      "2 3\n",
      "0 1 7 8\n",
      "4 9\n",
      "5\n",
      "6\n",
      "Done.-19.9985029406523\t42\t0\n",
      "-19.9962573516307\t42\t0.00299411869541971\n",
      "-19.9940117626092\t42\t0.00449117804312957\n",
      "-19.9917661735876\t42\t0.00748529673854928\n",
      "-19.989520584566\t42\t0.00898235608625913\n",
      "-19.9872749955445\t42\t0.0119764747816788\n",
      "-19.9169132062021\t42\t0.0134735341293887\n",
      "-19.7559793263233\t42\t0.152700053466405\n",
      "1.1676706469435\t42\t0.335341293887008\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_P_LVM1_1.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_P_LVM1_4.csv --species=human_ensembl_msigdb_c4 --exptype=continuous --ebins=9\n",
      "done\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM2_5.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c4.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c4/human_ensembl_msigdb_c4.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM2_5.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 858\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t858\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 858\n",
      "cat 705, mi = 0.00817169, MORF_DAP3\n",
      "cat 831, mi = 0.00766619, MORF_SKP1A\n",
      "cat 733, mi = 0.00690057, MORF_G22P1\n",
      "cat 131, mi = 0.00687575, GNF2_EIF3S6\n",
      "cat 701, mi = 0.00674373, MORF_CTBP1\n",
      "cat 783, mi = 0.00644827, MORF_PPP1CA\n",
      "cat 797, mi = 0.00616730, MORF_PSMC1\n",
      "cat 668, mi = 0.00602405, MORF_ANP32B\n",
      "cat 811, mi = 0.00594688, MORF_RAD21\n",
      "cat 738, mi = 0.00572452, MORF_HAT1\n",
      "cat 806, mi = 0.00561261, MORF_RAB1A\n",
      "cat 813, mi = 0.00556784, MORF_RAD23B\n",
      "cat 726, mi = 0.00554259, MORF_FBL\n",
      "cat 774, mi = 0.00551919, MORF_NPM1\n",
      "cat 785, mi = 0.00544802, MORF_PPP2CA\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 705, mi = 0.008, pvalue = 0.000, MORF_DAP3\n",
      "Passed: cat 131, mi = 0.007, pvalue = 0.000, GNF2_EIF3S6\n",
      "Passed: cat 318, mi = 0.004, pvalue = 0.000, MODULE_180\n",
      "Passed: cat 67, mi = 0.004, pvalue = 0.000, GCM_RAB10\n",
      "Passed: cat 421, mi = 0.003, pvalue = 0.000, MODULE_297\n",
      "Passed: cat 247, mi = 0.003, pvalue = 0.000, MODULE_111\n",
      "Passed: cat 114, mi = 0.003, pvalue = 0.000, GNF2_CDC20\n",
      "Passed: cat 29, mi = 0.003, pvalue = 0.000, GCM_DENR\n",
      "Passed: cat 790, mi = 0.003, pvalue = 0.000, MORF_PPP6C\n",
      "Passed: cat 132, mi = 0.002, pvalue = 0.000, GNF2_ELAC2\n",
      "Passed: cat 657, mi = 0.002, pvalue = 0.000, MODULE_92\n",
      "Passed: cat 650, mi = 0.002, pvalue = 0.000, MODULE_86\n",
      "Passed: cat 590, mi = 0.002, pvalue = 0.000, MODULE_524\n",
      "Passed: cat 737, mi = 0.002, pvalue = 0.000, MORF_GSPT1\n",
      "Passed: cat 642, mi = 0.002, pvalue = 0.000, MODULE_79\n",
      "Passed: cat 129, mi = 0.002, pvalue = 0.001, GNF2_DNM1\n",
      "Passed: cat 717, mi = 0.002, pvalue = 0.001, MORF_EIF4E\n",
      "Passed: cat 172, mi = 0.002, pvalue = 0.001, GNF2_MMP11\n",
      "Passed: cat 320, mi = 0.002, pvalue = 0.002, MODULE_182\n",
      "Passed: cat 495, mi = 0.001, pvalue = 0.001, MODULE_388\n",
      "Passed: cat 196, mi = 0.001, pvalue = 0.003, GNF2_RAP1B\n",
      "Passed: cat 519, mi = 0.001, pvalue = 0.004, MODULE_417\n",
      "Didn't pass (cutoff=0.0050): cat 410, mi = 0.001, pvalue = 0.009, MODULE_285\n",
      "Didn't pass (cutoff=0.0050): cat 502, mi = 0.001, pvalue = 0.009, MODULE_399\n",
      "Didn't pass (cutoff=0.0050): cat 51, mi = 0.001, pvalue = 0.027, GCM_MSN\n",
      "Passed: cat 284, mi = 0.001, pvalue = 0.002, MODULE_148\n",
      "Didn't pass (cutoff=0.0050): cat 237, mi = 0.001, pvalue = 0.029, MODULE_102\n",
      "Didn't pass (cutoff=0.0050): cat 566, mi = 0.001, pvalue = 0.047, MODULE_488\n",
      "Didn't pass (cutoff=0.0050): cat 573, mi = 0.001, pvalue = 0.048, MODULE_497\n",
      "Didn't pass (cutoff=0.0050): cat 499, mi = 0.001, pvalue = 0.050, MODULE_395\n",
      "Didn't pass (cutoff=0.0050): cat 479, mi = 0.001, pvalue = 0.069, MODULE_366\n",
      "Didn't pass (cutoff=0.0050): cat 465, mi = 0.001, pvalue = 0.067, MODULE_349\n",
      "Didn't pass (cutoff=0.0050): cat 607, mi = 0.001, pvalue = 0.076, MODULE_552\n",
      "Didn't pass (cutoff=0.0050): cat 183, mi = 0.001, pvalue = 0.082, GNF2_PCAF\n",
      "Didn't pass (cutoff=0.0050): cat 269, mi = 0.001, pvalue = 0.083, MODULE_131\n",
      "Didn't pass (cutoff=0.0050): cat 428, mi = 0.001, pvalue = 0.083, MODULE_305\n",
      "Didn't pass (cutoff=0.0050): cat 382, mi = 0.001, pvalue = 0.095, MODULE_249\n",
      "Didn't pass (cutoff=0.0050): cat 558, mi = 0.001, pvalue = 0.082, MODULE_474\n",
      "Didn't pass (cutoff=0.0050): cat 377, mi = 0.001, pvalue = 0.070, MODULE_243\n",
      "Didn't pass (cutoff=0.0050): cat 571, mi = 0.001, pvalue = 0.078, MODULE_493\n",
      "Didn't pass (cutoff=0.0050): cat 580, mi = 0.001, pvalue = 0.102, MODULE_509\n",
      "Didn't pass (cutoff=0.0050): cat 212, mi = 0.001, pvalue = 0.100, GNF2_SPI1\n",
      "Didn't pass (cutoff=0.0050): cat 364, mi = 0.001, pvalue = 0.129, MODULE_229\n",
      "Didn't pass (cutoff=0.0050): cat 610, mi = 0.001, pvalue = 0.131, MODULE_560\n",
      "Didn't pass (cutoff=0.0050): cat 99, mi = 0.001, pvalue = 0.120, GNF2_CASP1\n",
      "Didn't pass (cutoff=0.0050): cat 169, mi = 0.001, pvalue = 0.152, GNF2_MLF1\n",
      "Total number of categories evaluated = 397.\n",
      "Number of categories that passed the tests = 23\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM2_5.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM2_5.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 2, 4, 0.0119847535162795\n",
      "best merge is 2, 17, 0.0540579313258096\n",
      "best merge is 2, 15, 0.0559329509231561\n",
      "best merge is 5, 22, 0.0568758576344237\n",
      "best merge is 0, 8, 0.0786484966794915\n",
      "best merge is 6, 12, 0.0958589741574642\n",
      "best merge is 2, 10, 0.105945566174828\n",
      "best merge is 0, 3, 0.1133274664112\n",
      "best merge is 19, 20, 0.140650096236634\n",
      "best merge is 2, 5, 0.151801359836559\n",
      "best merge is 9, 16, 0.215420486914318\n",
      "best merge is 0, 7, 0.233619538252025\n",
      "best merge is 6, 14, 0.239802501342219\n",
      "best merge is 2, 21, 0.250180806654165\n",
      "best merge is 0, 18, 0.316154980855958\n",
      "best merge is 0, 9, 0.352370996799416\n",
      "best merge is 11, 13, 0.372262459409556\n",
      "best merge is 6, 19, 0.44743895610905\n",
      "best merge is 0, 11, 0.499382823738115\n",
      "order classes\n",
      "7 - 0.805994085828433\n",
      "8 - 0.362774403253767\n",
      "5 - 1.25147519398168\n",
      "2 - 2.14050914160495\n",
      "2 4 17 15 10 5 22 21\n",
      "1\n",
      "0 8 3 7 18 9 16\n",
      "6 12 14 19 20\n",
      "11 13\n",
      "Done.Start drawing\n",
      "66.19\t0.00\n",
      "GNF2_CDC20\n",
      "MODULE_524\n",
      "MORF_GSPT1\n",
      "MORF_PPP6C\n",
      "MODULE_79\n",
      "MODULE_388\n",
      "GNF2_RAP1B\n",
      "GNF2_ELAC2\n",
      "MORF_EIF4E\n",
      "MORF_DAP3\n",
      "GCM_RAB10\n",
      "GCM_DENR\n",
      "MODULE_86\n",
      "MODULE_182\n",
      "GNF2_EIF3S6\n",
      "MODULE_180\n",
      "MODULE_297\n",
      "GNF2_MMP11\n",
      "GNF2_DNM1\n",
      "MODULE_92\n",
      "MODULE_111\n",
      "MODULE_148\n",
      "MODULE_417\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM2_5.csv_PAGE/filt_FC_LVM2_5.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM2_5.csv_PAGE/filt_FC_LVM2_5.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM2_5.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM2_5.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 2, 4, 0.0119847535162795\n",
      "best merge is 2, 17, 0.0540579313258096\n",
      "best merge is 2, 15, 0.0559329509231561\n",
      "best merge is 5, 22, 0.0568758576344237\n",
      "best merge is 0, 8, 0.0786484966794915\n",
      "best merge is 6, 12, 0.0958589741574642\n",
      "best merge is 2, 10, 0.105945566174828\n",
      "best merge is 0, 3, 0.1133274664112\n",
      "best merge is 19, 20, 0.140650096236634\n",
      "best merge is 2, 5, 0.151801359836559\n",
      "best merge is 9, 16, 0.215420486914318\n",
      "best merge is 0, 7, 0.233619538252025\n",
      "best merge is 6, 14, 0.239802501342219\n",
      "best merge is 2, 21, 0.250180806654165\n",
      "best merge is 0, 18, 0.316154980855958\n",
      "best merge is 0, 9, 0.352370996799416\n",
      "best merge is 11, 13, 0.372262459409556\n",
      "best merge is 6, 19, 0.44743895610905\n",
      "best merge is 0, 11, 0.499382823738115\n",
      "order classes\n",
      "7 - 0.805994085828433\n",
      "8 - 0.362774403253767\n",
      "5 - 1.25147519398168\n",
      "2 - 2.14050914160495\n",
      "2 4 17 15 10 5 22 21\n",
      "1\n",
      "0 8 3 7 18 9 16\n",
      "6 12 14 19 20\n",
      "11 13\n",
      "Done.-19.8477111346125\t42\t0\n",
      "-19.6224505212268\t42\t0.304577730775042\n",
      "-19.501888502795\t42\t0.450521226771416\n",
      "-19.4130533313189\t42\t0.545701767638616\n",
      "-19.330563529234\t42\t0.628191569723523\n",
      "-19.2480737271491\t42\t0.71068137180843\n",
      "-19.1528931862819\t42\t0.793171173893337\n",
      "-19.0196404290678\t42\t0.901042453542831\n",
      "1.52983834416075\t42\t1.0596766883215\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_LVM2_5.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_P_LVM1_2.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c4.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c4/human_ensembl_msigdb_c4.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_P_LVM1_2.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 858\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t858\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 858\n",
      "cat 105, mi = 0.01160135, GNF2_CCNB2\n",
      "cat 114, mi = 0.01137060, GNF2_CDC20\n",
      "cat 113, mi = 0.00996510, GNF2_CDC2\n",
      "cat 131, mi = 0.00976724, GNF2_EIF3S6\n",
      "cat 104, mi = 0.00970787, GNF2_CCNA2\n",
      "cat 83, mi = 0.00854818, GCM_TPT1\n",
      "cat 121, mi = 0.00832009, GNF2_CENPF\n",
      "cat 774, mi = 0.00824567, MORF_NPM1\n",
      "cat 146, mi = 0.00824542, GNF2_HMMR\n",
      "cat 772, mi = 0.00803353, MORF_NME2\n",
      "cat 120, mi = 0.00794721, GNF2_CENPE\n",
      "cat 847, mi = 0.00782541, MORF_TPT1\n",
      "cat 56, mi = 0.00779233, GCM_NPM1\n",
      "cat 667, mi = 0.00767757, MORF_ACTG1\n",
      "cat 202, mi = 0.00728401, GNF2_RRM2\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 105, mi = 0.012, pvalue = 0.000, GNF2_CCNB2\n",
      "Passed: cat 131, mi = 0.010, pvalue = 0.000, GNF2_EIF3S6\n",
      "Passed: cat 289, mi = 0.006, pvalue = 0.000, MODULE_152\n",
      "Passed: cat 806, mi = 0.005, pvalue = 0.000, MORF_RAB1A\n",
      "Passed: cat 67, mi = 0.004, pvalue = 0.000, GCM_RAB10\n",
      "Passed: cat 372, mi = 0.003, pvalue = 0.000, MODULE_239\n",
      "Passed: cat 232, mi = 0.002, pvalue = 0.000, GNF2_XRCC5\n",
      "Passed: cat 437, mi = 0.002, pvalue = 0.000, MODULE_316\n",
      "Passed: cat 264, mi = 0.002, pvalue = 0.000, MODULE_127\n",
      "Passed: cat 403, mi = 0.002, pvalue = 0.001, MODULE_277\n",
      "Passed: cat 318, mi = 0.002, pvalue = 0.001, MODULE_180\n",
      "Passed: cat 411, mi = 0.002, pvalue = 0.001, MODULE_286\n",
      "Passed: cat 533, mi = 0.001, pvalue = 0.003, MODULE_436\n",
      "Passed: cat 103, mi = 0.001, pvalue = 0.002, GNF2_CCNA1\n",
      "Didn't pass (cutoff=0.0050): cat 294, mi = 0.001, pvalue = 0.005, MODULE_157\n",
      "Didn't pass (cutoff=0.0050): cat 21, mi = 0.001, pvalue = 0.005, GCM_CHUK\n",
      "Passed: cat 224, mi = 0.001, pvalue = 0.002, GNF2_TNFSF10\n",
      "Didn't pass (cutoff=0.0050): cat 698, mi = 0.001, pvalue = 0.007, MORF_CNTN1\n",
      "Didn't pass (cutoff=0.0050): cat 237, mi = 0.001, pvalue = 0.009, MODULE_102\n",
      "Didn't pass (cutoff=0.0050): cat 414, mi = 0.001, pvalue = 0.012, MODULE_289\n",
      "Didn't pass (cutoff=0.0050): cat 129, mi = 0.001, pvalue = 0.013, GNF2_DNM1\n",
      "Didn't pass (cutoff=0.0050): cat 590, mi = 0.001, pvalue = 0.017, MODULE_524\n",
      "Didn't pass (cutoff=0.0050): cat 822, mi = 0.001, pvalue = 0.016, MORF_REV3L\n",
      "Didn't pass (cutoff=0.0050): cat 206, mi = 0.001, pvalue = 0.024, GNF2_SERPINB5\n",
      "Didn't pass (cutoff=0.0050): cat 177, mi = 0.001, pvalue = 0.007, GNF2_MYL2\n",
      "Didn't pass (cutoff=0.0050): cat 547, mi = 0.001, pvalue = 0.020, MODULE_456\n",
      "Didn't pass (cutoff=0.0050): cat 159, mi = 0.001, pvalue = 0.026, GNF2_LCAT\n",
      "Didn't pass (cutoff=0.0050): cat 147, mi = 0.001, pvalue = 0.028, GNF2_HPN\n",
      "Didn't pass (cutoff=0.0050): cat 119, mi = 0.001, pvalue = 0.030, GNF2_CEBPA\n",
      "Didn't pass (cutoff=0.0050): cat 580, mi = 0.001, pvalue = 0.024, MODULE_509\n",
      "Didn't pass (cutoff=0.0050): cat 148, mi = 0.001, pvalue = 0.034, GNF2_HPX\n",
      "Didn't pass (cutoff=0.0050): cat 845, mi = 0.001, pvalue = 0.023, MORF_TNFRSF6\n",
      "Didn't pass (cutoff=0.0050): cat 612, mi = 0.001, pvalue = 0.027, MODULE_563\n",
      "Didn't pass (cutoff=0.0050): cat 510, mi = 0.001, pvalue = 0.012, MODULE_407\n",
      "Didn't pass (cutoff=0.0050): cat 109, mi = 0.001, pvalue = 0.052, GNF2_CD48\n",
      "Didn't pass (cutoff=0.0050): cat 323, mi = 0.001, pvalue = 0.026, MODULE_186\n",
      "Didn't pass (cutoff=0.0050): cat 724, mi = 0.001, pvalue = 0.059, MORF_ETV3\n",
      "Total number of categories evaluated = 258.\n",
      "Number of categories that passed the tests = 15\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_P_LVM1_2.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_P_LVM1_2.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 0, 13, 0.0350653797538616\n",
      "best merge is 0, 8, 0.0813344350412197\n",
      "best merge is 3, 5, 0.0820705362856463\n",
      "best merge is 1, 2, 0.114513799770944\n",
      "best merge is 7, 12, 0.195287643276966\n",
      "best merge is 3, 4, 0.20228495903341\n",
      "best merge is 3, 9, 0.244623931822784\n",
      "best merge is 7, 10, 0.272053139663013\n",
      "best merge is 1, 6, 0.315181817843317\n",
      "best merge is 7, 14, 0.340766806234218\n",
      "best merge is 1, 3, 0.596752159030243\n",
      "order classes\n",
      "3 - 0.219704722040335\n",
      "3 - 0.827641594952864\n",
      "4 - 0.572115117103508\n",
      "4 - 0.825794222487021\n",
      "0 13 8\n",
      "1 2 6\n",
      "3 5 4 9\n",
      "7 12 10 14\n",
      "11\n",
      "Done.Start drawing\n",
      "5.29\t0.00\n",
      "GNF2_CCNB2\n",
      "GNF2_CCNA1\n",
      "MODULE_127\n",
      "GCM_RAB10\n",
      "MORF_RAB1A\n",
      "MODULE_239\n",
      "MODULE_277\n",
      "GNF2_EIF3S6\n",
      "GNF2_XRCC5\n",
      "MODULE_152\n",
      "MODULE_286\n",
      "MODULE_316\n",
      "MODULE_436\n",
      "MODULE_180\n",
      "GNF2_TNFSF10\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_P_LVM1_2.csv_PAGE/filt_FC_P_LVM1_2.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_P_LVM1_2.csv_PAGE/filt_FC_P_LVM1_2.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_P_LVM1_2.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_P_LVM1_2.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 0, 13, 0.0350653797538616\n",
      "best merge is 0, 8, 0.0813344350412197\n",
      "best merge is 3, 5, 0.0820705362856463\n",
      "best merge is 1, 2, 0.114513799770944\n",
      "best merge is 7, 12, 0.195287643276966\n",
      "best merge is 3, 4, 0.20228495903341\n",
      "best merge is 3, 9, 0.244623931822784\n",
      "best merge is 7, 10, 0.272053139663013\n",
      "best merge is 1, 6, 0.315181817843317\n",
      "best merge is 7, 14, 0.340766806234218\n",
      "best merge is 1, 3, 0.596752159030243\n",
      "order classes\n",
      "3 - 0.219704722040335\n",
      "3 - 0.827641594952864\n",
      "4 - 0.572115117103508\n",
      "4 - 0.825794222487021\n",
      "0 13 8\n",
      "1 2 6\n",
      "3 5 4 9\n",
      "7 12 10 14\n",
      "11\n",
      "Done.-16.8638941398866\t42\t0\n",
      "-13.2117202268431\t42\t6.27221172022684\n",
      "-12.3780718336484\t42\t7.30434782608696\n",
      "-11.8223062381853\t42\t7.93950850661626\n",
      "-11.3856332703214\t42\t8.41587901701323\n",
      "-10.9092627599244\t42\t8.81285444234405\n",
      "-10.3137996219282\t42\t9.36862003780718\n",
      "-9.44045368620038\t42\t10.0037807183365\n",
      "6.55765595463138\t42\t11.1153119092628\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_P_LVM1_2.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_P_LVM1_4.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c4.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c4/human_ensembl_msigdb_c4.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_P_LVM1_4.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 858\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t858\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 858\n",
      "cat 774, mi = 0.01926170, MORF_NPM1\n",
      "cat 772, mi = 0.01828361, MORF_NME2\n",
      "cat 668, mi = 0.01807300, MORF_ANP32B\n",
      "cat 667, mi = 0.01709044, MORF_ACTG1\n",
      "cat 733, mi = 0.01493772, MORF_G22P1\n",
      "cat 131, mi = 0.01471423, GNF2_EIF3S6\n",
      "cat 705, mi = 0.01414026, MORF_DAP3\n",
      "cat 134, mi = 0.01409456, GNF2_FBL\n",
      "cat 726, mi = 0.01365477, MORF_FBL\n",
      "cat 56, mi = 0.01364062, GCM_NPM1\n",
      "cat 721, mi = 0.01354858, MORF_ERH\n",
      "cat 847, mi = 0.01337719, MORF_TPT1\n",
      "cat 783, mi = 0.01331282, MORF_PPP1CA\n",
      "cat 831, mi = 0.01289972, MORF_SKP1A\n",
      "cat 6, mi = 0.01270956, GCM_ACTG1\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 774, mi = 0.019, pvalue = 0.000, MORF_NPM1\n",
      "Passed: cat 797, mi = 0.011, pvalue = 0.000, MORF_PSMC1\n",
      "Passed: cat 670, mi = 0.008, pvalue = 0.000, MORF_AP3D1\n",
      "Passed: cat 181, mi = 0.006, pvalue = 0.000, GNF2_PA2G4\n",
      "Passed: cat 132, mi = 0.003, pvalue = 0.000, GNF2_ELAC2\n",
      "Passed: cat 126, mi = 0.002, pvalue = 0.000, GNF2_DDX5\n",
      "Passed: cat 240, mi = 0.002, pvalue = 0.001, MODULE_105\n",
      "Passed: cat 294, mi = 0.001, pvalue = 0.005, MODULE_157\n",
      "Passed: cat 340, mi = 0.001, pvalue = 0.002, MODULE_202\n",
      "Passed: cat 456, mi = 0.001, pvalue = 0.004, MODULE_334\n",
      "Didn't pass (cutoff=0.0050): cat 649, mi = 0.001, pvalue = 0.009, MODULE_85\n",
      "Didn't pass (cutoff=0.0050): cat 848, mi = 0.001, pvalue = 0.008, MORF_TTN\n",
      "Didn't pass (cutoff=0.0050): cat 789, mi = 0.001, pvalue = 0.011, MORF_PPP5C\n",
      "Didn't pass (cutoff=0.0050): cat 377, mi = 0.001, pvalue = 0.013, MODULE_243\n",
      "Didn't pass (cutoff=0.0050): cat 314, mi = 0.001, pvalue = 0.013, MODULE_177\n",
      "Didn't pass (cutoff=0.0050): cat 520, mi = 0.001, pvalue = 0.017, MODULE_418\n",
      "Didn't pass (cutoff=0.0050): cat 748, mi = 0.001, pvalue = 0.018, MORF_ITGA2\n",
      "Didn't pass (cutoff=0.0050): cat 33, mi = 0.001, pvalue = 0.019, GCM_ERBB2IP\n",
      "Didn't pass (cutoff=0.0050): cat 692, mi = 0.001, pvalue = 0.025, MORF_CD8A\n",
      "Didn't pass (cutoff=0.0050): cat 24, mi = 0.001, pvalue = 0.032, GCM_CSNK1D\n",
      "Didn't pass (cutoff=0.0050): cat 804, mi = 0.001, pvalue = 0.036, MORF_PTPRR\n",
      "Didn't pass (cutoff=0.0050): cat 136, mi = 0.001, pvalue = 0.044, GNF2_FGR\n",
      "Didn't pass (cutoff=0.0050): cat 318, mi = 0.001, pvalue = 0.027, MODULE_180\n",
      "Didn't pass (cutoff=0.0050): cat 549, mi = 0.001, pvalue = 0.043, MODULE_458\n",
      "Didn't pass (cutoff=0.0050): cat 335, mi = 0.001, pvalue = 0.046, MODULE_199\n",
      "Didn't pass (cutoff=0.0050): cat 547, mi = 0.001, pvalue = 0.032, MODULE_456\n",
      "Didn't pass (cutoff=0.0050): cat 187, mi = 0.001, pvalue = 0.058, GNF2_PRDX2\n",
      "Didn't pass (cutoff=0.0050): cat 219, mi = 0.001, pvalue = 0.061, GNF2_TAL1\n",
      "Didn't pass (cutoff=0.0050): cat 212, mi = 0.001, pvalue = 0.069, GNF2_SPI1\n",
      "Didn't pass (cutoff=0.0050): cat 546, mi = 0.001, pvalue = 0.072, MODULE_455\n",
      "Total number of categories evaluated = 219.\n",
      "Number of categories that passed the tests = 10\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_P_LVM1_4.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_P_LVM1_4.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 1, 3, 0.0191723273663442\n",
      "best merge is 1, 2, 0.0226534380035393\n",
      "best merge is 0, 4, 0.0331607305187859\n",
      "best merge is 1, 5, 0.0999609865639213\n",
      "best merge is 0, 1, 0.242270030798003\n",
      "best merge is 0, 6, 0.288372183951788\n",
      "order classes\n",
      "6 - 0.216266957404444\n",
      "0 4 1 3 2 5\n",
      "6\n",
      "7\n",
      "8\n",
      "9\n",
      "Done.Start drawing\n",
      "110.21\t0.00\n",
      "MODULE_334\n",
      "GNF2_ELAC2\n",
      "MORF_PSMC1\n",
      "GNF2_PA2G4\n",
      "MORF_AP3D1\n",
      "GNF2_DDX5\n",
      "MODULE_105\n",
      "MORF_NPM1\n",
      "MODULE_202\n",
      "MODULE_157\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_P_LVM1_4.csv_PAGE/filt_FC_P_LVM1_4.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_P_LVM1_4.csv_PAGE/filt_FC_P_LVM1_4.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_P_LVM1_4.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_P_LVM1_4.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 1, 3, 0.0191723273663442\n",
      "best merge is 1, 2, 0.0226534380035393\n",
      "best merge is 0, 4, 0.0331607305187859\n",
      "best merge is 1, 5, 0.0999609865639213\n",
      "best merge is 0, 1, 0.242270030798003\n",
      "best merge is 0, 6, 0.288372183951788\n",
      "order classes\n",
      "6 - 0.216266957404444\n",
      "0 4 1 3 2 5\n",
      "6\n",
      "7\n",
      "8\n",
      "9\n",
      "Done.-19.9980945467743\t42\t0\n",
      "-19.9923781870974\t42\t0.00381090645132021\n",
      "-19.9847563741947\t42\t0.0114327193539606\n",
      "-19.9790400145177\t42\t0.019054532256601\n",
      "-19.9733236548408\t42\t0.0228654387079212\n",
      "-19.9676072951638\t42\t0.0304872516105617\n",
      "-19.8361310225932\t42\t0.0342981580618819\n",
      "-19.4912439887488\t42\t0.293439796751656\n",
      "1.36203611287542\t42\t0.724072225750839\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_P_LVM1_4.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_P_LVM1_3.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_msigdb_c4.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_msigdb_c4/human_ensembl_msigdb_c4.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_P_LVM1_3.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 12086... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 858\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 12086 out of 12086 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 12086\t858\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1343 hits go_terms_num = 858\n",
      "cat 104, mi = 0.01775284, GNF2_CCNA2\n",
      "cat 184, mi = 0.01578305, GNF2_PCNA\n",
      "cat 113, mi = 0.01500342, GNF2_CDC2\n",
      "cat 121, mi = 0.01500342, GNF2_CENPF\n",
      "cat 201, mi = 0.01496874, GNF2_RRM1\n",
      "cat 105, mi = 0.01481300, GNF2_CCNB2\n",
      "cat 210, mi = 0.01471150, GNF2_SMC4L1\n",
      "cat 114, mi = 0.01454624, GNF2_CDC20\n",
      "cat 146, mi = 0.01241514, GNF2_HMMR\n",
      "cat 166, mi = 0.01239511, GNF2_MCM4\n",
      "cat 199, mi = 0.01180306, GNF2_RFC4\n",
      "cat 135, mi = 0.01146347, GNF2_FEN1\n",
      "cat 123, mi = 0.01103631, GNF2_CKS2\n",
      "cat 96, mi = 0.01100859, GNF2_BUB1B\n",
      "cat 120, mi = 0.01055475, GNF2_CENPE\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 104, mi = 0.018, pvalue = 0.000, GNF2_CCNA2\n",
      "Passed: cat 738, mi = 0.007, pvalue = 0.000, MORF_HAT1\n",
      "Passed: cat 772, mi = 0.005, pvalue = 0.000, MORF_NME2\n",
      "Passed: cat 533, mi = 0.003, pvalue = 0.000, MODULE_436\n",
      "Passed: cat 372, mi = 0.003, pvalue = 0.000, MODULE_239\n",
      "Passed: cat 788, mi = 0.002, pvalue = 0.000, MORF_PPP2R5E\n",
      "Passed: cat 186, mi = 0.002, pvalue = 0.000, GNF2_PPP6C\n",
      "Passed: cat 249, mi = 0.002, pvalue = 0.000, MODULE_113\n",
      "Passed: cat 23, mi = 0.002, pvalue = 0.000, GCM_CSNK1A1\n",
      "Passed: cat 347, mi = 0.002, pvalue = 0.000, MODULE_209\n",
      "Passed: cat 149, mi = 0.002, pvalue = 0.001, GNF2_ICAM3\n",
      "Passed: cat 454, mi = 0.002, pvalue = 0.001, MODULE_332\n",
      "Passed: cat 44, mi = 0.002, pvalue = 0.002, GCM_ING1\n",
      "Passed: cat 559, mi = 0.001, pvalue = 0.000, MODULE_478\n",
      "Didn't pass (cutoff=0.0050): cat 92, mi = 0.001, pvalue = 0.007, GNF2_ATM\n",
      "Didn't pass (cutoff=0.0050): cat 486, mi = 0.001, pvalue = 0.009, MODULE_375\n",
      "Didn't pass (cutoff=0.0050): cat 49, mi = 0.001, pvalue = 0.007, GCM_MAX\n",
      "Didn't pass (cutoff=0.0050): cat 823, mi = 0.001, pvalue = 0.005, MORF_RFC1\n",
      "Didn't pass (cutoff=0.0050): cat 431, mi = 0.001, pvalue = 0.009, MODULE_308\n",
      "Didn't pass (cutoff=0.0050): cat 259, mi = 0.001, pvalue = 0.010, MODULE_122\n",
      "Didn't pass (cutoff=0.0050): cat 192, mi = 0.001, pvalue = 0.012, GNF2_RAB3A\n",
      "Didn't pass (cutoff=0.0050): cat 137, mi = 0.001, pvalue = 0.015, GNF2_FOS\n",
      "Didn't pass (cutoff=0.0050): cat 641, mi = 0.001, pvalue = 0.014, MODULE_78\n",
      "Didn't pass (cutoff=0.0050): cat 98, mi = 0.001, pvalue = 0.019, GNF2_CARD15\n",
      "Didn't pass (cutoff=0.0050): cat 276, mi = 0.001, pvalue = 0.023, MODULE_138\n",
      "Didn't pass (cutoff=0.0050): cat 538, mi = 0.001, pvalue = 0.005, MODULE_444\n",
      "Didn't pass (cutoff=0.0050): cat 337, mi = 0.001, pvalue = 0.028, MODULE_20\n",
      "Didn't pass (cutoff=0.0050): cat 103, mi = 0.001, pvalue = 0.040, GNF2_CCNA1\n",
      "Didn't pass (cutoff=0.0050): cat 129, mi = 0.001, pvalue = 0.035, GNF2_DNM1\n",
      "Didn't pass (cutoff=0.0050): cat 412, mi = 0.001, pvalue = 0.053, MODULE_287\n",
      "Didn't pass (cutoff=0.0050): cat 571, mi = 0.001, pvalue = 0.045, MODULE_493\n",
      "Didn't pass (cutoff=0.0050): cat 437, mi = 0.001, pvalue = 0.070, MODULE_316\n",
      "Didn't pass (cutoff=0.0050): cat 477, mi = 0.001, pvalue = 0.045, MODULE_362\n",
      "Didn't pass (cutoff=0.0050): cat 278, mi = 0.001, pvalue = 0.022, MODULE_14\n",
      "Total number of categories evaluated = 289.\n",
      "Number of categories that passed the tests = 14\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_P_LVM1_3.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_P_LVM1_3.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 9, 10, 0.0334620237007557\n",
      "best merge is 3, 9, 0.0815849539999725\n",
      "best merge is 2, 5, 0.187174888973373\n",
      "best merge is 2, 4, 0.229847320255968\n",
      "best merge is 6, 8, 0.23429562168949\n",
      "best merge is 3, 7, 0.242794773106974\n",
      "best merge is 1, 12, 0.244458878175741\n",
      "best merge is 3, 13, 0.353064739827775\n",
      "best merge is 2, 6, 0.358732288417511\n",
      "best merge is 1, 2, 0.493476188173912\n",
      "order classes\n",
      "2 - 0.855606073615093\n",
      "5 - 0.67951718949629\n",
      "5 - 0.52354964711453\n",
      "3 9 10 7 13\n",
      "1 12\n",
      "2 5 4 6 8\n",
      "0\n",
      "11\n",
      "Done.Start drawing\n",
      "11.42\t0.00\n",
      "MODULE_436\n",
      "MODULE_209\n",
      "GNF2_ICAM3\n",
      "MODULE_113\n",
      "MODULE_478\n",
      "MODULE_239\n",
      "MORF_NME2\n",
      "MORF_PPP2R5E\n",
      "GNF2_PPP6C\n",
      "GCM_CSNK1A1\n",
      "MODULE_332\n",
      "MORF_HAT1\n",
      "GCM_ING1\n",
      "GNF2_CCNA2\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_P_LVM1_3.csv_PAGE/filt_FC_P_LVM1_3.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_P_LVM1_3.csv_PAGE/filt_FC_P_LVM1_3.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_P_LVM1_3.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_P_LVM1_3.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 9, 10, 0.0334620237007557\n",
      "best merge is 3, 9, 0.0815849539999725\n",
      "best merge is 2, 5, 0.187174888973373\n",
      "best merge is 2, 4, 0.229847320255968\n",
      "best merge is 6, 8, 0.23429562168949\n",
      "best merge is 3, 7, 0.242794773106974\n",
      "best merge is 1, 12, 0.244458878175741\n",
      "best merge is 3, 13, 0.353064739827775\n",
      "best merge is 2, 6, 0.358732288417511\n",
      "best merge is 1, 2, 0.493476188173912\n",
      "order classes\n",
      "2 - 0.855606073615093\n",
      "5 - 0.67951718949629\n",
      "5 - 0.52354964711453\n",
      "3 9 10 7 13\n",
      "1 12\n",
      "2 5 4 6 8\n",
      "0\n",
      "11\n",
      "Done.-19.1725043782837\t42\t0\n",
      "-18.1059544658494\t42\t1.65499124343257\n",
      "-17.7381786339755\t42\t2.13309982486865\n",
      "-17.4807355516637\t42\t2.39054290718039\n",
      "-17.223292469352\t42\t2.64798598949212\n",
      "-16.9658493870403\t42\t2.90542907180385\n",
      "-16.6348511383538\t42\t3.16287215411559\n",
      "-16.0831873905429\t42\t3.56742556917688\n",
      "3.13309982486865\t42\t4.2661996497373\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_msigdb_c4/fc_values/filt_FC_P_LVM1_3.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "sem --wait done\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "nohup sh run.sh iPAGE_JY.sh &"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
