{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# This is used for running data vs human_ensembl annotations located in: \n",
    "### /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import pickle\n",
    "import seaborn as sns\n",
    "import scipy.stats\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### get index of file names to loop through "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "cd fc_values\n",
    "ls FC* > filelist.txt\n",
    "cd .."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### get ensembl genes represented in iPAGE database"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash \n",
    "\n",
    "cp /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl/human_ensembl_index.txt human_ensembl_index.txt\n",
    "awk '{print $1}' human_ensembl_index.txt > ensembl_index.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### make sure that genes in file match the index "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "#####################\n",
    "#######read in data from logfc calculations\n",
    "#####################\n",
    "####define index file to loop through\n",
    "index_file = \"fc_values/filelist.txt\"\n",
    "####define matching ensg index\n",
    "GO_index_filename = \"ensembl_index.txt\"\n",
    "genes_with_GOs = []\n",
    "with open(GO_index_filename,'r') as rf:\n",
    "    for line in rf:\n",
    "        gene_name = line.split('\\n')[0]\n",
    "        genes_with_GOs.append(gene_name)\n",
    "genes_with_GOs_set = set(genes_with_GOs)\n",
    "####open file and loop through each line\n",
    "###\n",
    "iPAGE_sh = \"iPAGE_JY.sh\"\n",
    "cwd = os.getcwd()+\"/\"\n",
    "os.system(\"rm \"+iPAGE_sh)\n",
    "###\n",
    "with open(index_file,'r') as rf:\n",
    "    for line in rf:\n",
    "        #####do this to each file\n",
    "        ###read in file into pandas df\n",
    "        FC_file = \"fc_values/\"+line.split('\\n')[0]\n",
    "        FC_df = pd.read_csv(FC_file,sep=',',index_col=0)\n",
    "        FC_df.columns = ['log2FoldChange']\n",
    "        FC_df.index.names = ['Gene']\n",
    "        ###filter\n",
    "        FC_df = FC_df.loc[[x for x in FC_df.index if x in genes_with_GOs_set]]\n",
    "        ###save\n",
    "        save_file = \"fc_values/filt_\"+line.split('\\n')[0]\n",
    "        FC_df.to_csv(save_file,sep='\\t',header=True)\n",
    "        ###write final iPAGE command to an ipage .sh file\n",
    "        iPAGE_cmd = \"perl /nvme/bins/iPAGEv1.0/page.pl --expfile \"+cwd+save_file+\" --species=human_ensembl --exptype=continuous --ebins=9 \\n\"\n",
    "        with open(iPAGE_sh, \"a\") as myfile:\n",
    "            myfile.write(iPAGE_cmd)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## then run the command from this folder:\n",
    "\n",
    "nohup sh run.sh iPAGE_JY.sh &"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM1_1.csv --species=human_ensembl --exptype=continuous --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM1_2.csv --species=human_ensembl --exptype=continuous --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM1_3.csv --species=human_ensembl --exptype=continuous --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM1_4.csv --species=human_ensembl --exptype=continuous --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM2_1.csv --species=human_ensembl --exptype=continuous --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM2_2.csv --species=human_ensembl --exptype=continuous --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM2_3.csv --species=human_ensembl --exptype=continuous --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM2_4.csv --species=human_ensembl --exptype=continuous --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM2_5.csv --species=human_ensembl --exptype=continuous --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_P_LVM1_1.csv --species=human_ensembl --exptype=continuous --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_P_LVM1_2.csv --species=human_ensembl --exptype=continuous --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_P_LVM1_3.csv --species=human_ensembl --exptype=continuous --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_P_LVM1_4.csv --species=human_ensembl --exptype=continuous --ebins=9\n",
      "done\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_P_LVM1_1.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl/human_ensembl.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_P_LVM1_1.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 13290... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 17689\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 13290 out of 13290 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 13290\t17689\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1477 hits go_terms_num = 17689\n",
      "cat 692, mi = 0.01678378, structural constituent of ribosome\n",
      "cat 2546, mi = 0.01522372, SRP-dependent cotranslational protein targeting to membrane\n",
      "cat 836, mi = 0.01442542, translational initiation\n",
      "cat 2545, mi = 0.01258348, nuclear-transcribed mRNA catabolic process, nonsense-mediated decay\n",
      "cat 1498, mi = 0.01054573, viral transcription\n",
      "cat 2016, mi = 0.00795960, cytosolic large ribosomal subunit\n",
      "cat 4152, mi = 0.00677452, cytosolic small ribosomal subunit\n",
      "cat 7, mi = 0.00581698, respiratory chain\n",
      "cat 10, mi = 0.00466972, mitochondrial electron transport, NADH to ubiquinone\n",
      "cat 12, mi = 0.00437794, mitochondrial respiratory chain complex I\n",
      "cat 1872, mi = 0.00431486, spliceosomal complex\n",
      "cat 9, mi = 0.00423891, NADH dehydrogenase (ubiquinone) activity\n",
      "cat 694, mi = 0.00396985, mitochondrial translational termination\n",
      "cat 4159, mi = 0.00396328, small ribosomal subunit\n",
      "cat 6800, mi = 0.00383050, cytoplasmic translation\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 692, mi = 0.017, pvalue = 0.000, structural constituent of ribosome\n",
      "Passed: cat 7, mi = 0.006, pvalue = 0.000, respiratory chain\n",
      "Passed: cat 1872, mi = 0.004, pvalue = 0.000, spliceosomal complex\n",
      "Passed: cat 1064, mi = 0.003, pvalue = 0.000, proteasome complex\n",
      "Passed: cat 57, mi = 0.003, pvalue = 0.000, mitochondrial ATP synthesis coupled proton transport\n",
      "Passed: cat 893, mi = 0.003, pvalue = 0.000, protein folding\n",
      "Passed: cat 835, mi = 0.002, pvalue = 0.000, translation initiation factor activity\n",
      "Passed: cat 1680, mi = 0.002, pvalue = 0.000, nuclear chromosome, telomeric region\n",
      "Passed: cat 1956, mi = 0.002, pvalue = 0.000, Rab GTPase binding\n",
      "Passed: cat 944, mi = 0.002, pvalue = 0.000, transcription elongation from RNA polymerase II promoter\n",
      "Passed: cat 3156, mi = 0.002, pvalue = 0.000, myosin complex\n",
      "Passed: cat 904, mi = 0.002, pvalue = 0.000, response to virus\n",
      "Passed: cat 189, mi = 0.002, pvalue = 0.000, embryonic limb morphogenesis\n",
      "Passed: cat 2150, mi = 0.002, pvalue = 0.000, DNA methylation\n",
      "Passed: cat 3588, mi = 0.002, pvalue = 0.001, extracellular vesicle\n",
      "Passed: cat 574, mi = 0.001, pvalue = 0.001, intermediate filament cytoskeleton\n",
      "Passed: cat 1558, mi = 0.001, pvalue = 0.002, protein targeting to mitochondrion\n",
      "Passed: cat 2654, mi = 0.001, pvalue = 0.000, hydrolase activity, acting on carbon-nitrogen (but not peptide) bonds, in cyclic amidines\n",
      "Passed: cat 4661, mi = 0.001, pvalue = 0.003, cardiac conduction\n",
      "Passed: cat 4780, mi = 0.001, pvalue = 0.001, signal transducer, downstream of receptor, with serine/threonine kinase activity\n",
      "Passed: cat 1173, mi = 0.001, pvalue = 0.001, low-density lipoprotein receptor activity\n",
      "Passed: cat 6726, mi = 0.001, pvalue = 0.001, histone exchange\n",
      "Passed: cat 5638, mi = 0.001, pvalue = 0.003, decidualization\n",
      "Passed: cat 6919, mi = 0.001, pvalue = 0.001, calcium channel inhibitor activity\n",
      "Passed: cat 2333, mi = 0.001, pvalue = 0.005, retrograde vesicle-mediated transport, Golgi to ER\n",
      "Passed: cat 8112, mi = 0.001, pvalue = 0.004, motor neuron axon guidance\n",
      "Didn't pass (cutoff=0.0050): cat 6363, mi = 0.001, pvalue = 0.007, positive regulation of glucose import\n",
      "Didn't pass (cutoff=0.0050): cat 2879, mi = 0.001, pvalue = 0.005, cortical cytoskeleton\n",
      "Didn't pass (cutoff=0.0050): cat 2419, mi = 0.001, pvalue = 0.007, galactosyltransferase activity\n",
      "Didn't pass (cutoff=0.0050): cat 6464, mi = 0.001, pvalue = 0.006, K63-linked polyubiquitin binding\n",
      "Didn't pass (cutoff=0.0050): cat 5044, mi = 0.001, pvalue = 0.011, cellular amino acid biosynthetic process\n",
      "Didn't pass (cutoff=0.0050): cat 5786, mi = 0.001, pvalue = 0.009, carbohydrate phosphorylation\n",
      "Didn't pass (cutoff=0.0050): cat 5609, mi = 0.001, pvalue = 0.011, pre-autophagosomal structure membrane\n",
      "Didn't pass (cutoff=0.0050): cat 3178, mi = 0.001, pvalue = 0.013, protein K11-linked ubiquitination\n",
      "Didn't pass (cutoff=0.0050): cat 2695, mi = 0.001, pvalue = 0.009, synaptic vesicle\n",
      "Didn't pass (cutoff=0.0050): cat 2806, mi = 0.001, pvalue = 0.014, adherens junction organization\n",
      "Didn't pass (cutoff=0.0050): cat 2863, mi = 0.001, pvalue = 0.013, purine nucleotide biosynthetic process\n",
      "Didn't pass (cutoff=0.0050): cat 3048, mi = 0.001, pvalue = 0.007, aminoacyl-tRNA synthetase multienzyme complex\n",
      "Didn't pass (cutoff=0.0050): cat 7598, mi = 0.001, pvalue = 0.008, placenta blood vessel development\n",
      "Didn't pass (cutoff=0.0050): cat 862, mi = 0.001, pvalue = 0.006, sphingosine-1-phosphate signaling pathway\n",
      "Didn't pass (cutoff=0.0050): cat 987, mi = 0.001, pvalue = 0.005, leptin-mediated signaling pathway\n",
      "Didn't pass (cutoff=0.0050): cat 4528, mi = 0.001, pvalue = 0.006, acetylcholine-gated channel complex\n",
      "Didn't pass (cutoff=0.0050): cat 7927, mi = 0.001, pvalue = 0.007, cerebral cortex radially oriented cell migration\n",
      "Didn't pass (cutoff=0.0050): cat 9436, mi = 0.001, pvalue = 0.006, cotranslational protein targeting to membrane\n",
      "Didn't pass (cutoff=0.0050): cat 2465, mi = 0.001, pvalue = 0.027, establishment of protein localization\n",
      "Didn't pass (cutoff=0.0050): cat 3707, mi = 0.001, pvalue = 0.028, regulation of multicellular organism growth\n",
      "Total number of categories evaluated = 278.\n",
      "Number of categories that passed the tests = 26\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_P_LVM1_1.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_P_LVM1_1.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 1, 4, 0.0141428386765209\n",
      "best merge is 1, 2, 0.0181891030558878\n",
      "best merge is 0, 16, 0.0202532795338033\n",
      "best merge is 0, 1, 0.0276977850196122\n",
      "best merge is 0, 23, 0.0466816696839939\n",
      "best merge is 0, 21, 0.0762707184641387\n",
      "best merge is 3, 5, 0.0855030131616206\n",
      "best merge is 0, 13, 0.0890702748583888\n",
      "best merge is 6, 14, 0.0902547077667593\n",
      "best merge is 17, 18, 0.174117974331894\n",
      "best merge is 7, 9, 0.191208282168867\n",
      "best merge is 0, 6, 0.198106182834498\n",
      "best merge is 10, 15, 0.330019909989455\n",
      "best merge is 0, 24, 0.35306101894559\n",
      "best merge is 0, 3, 0.404923223576099\n",
      "best merge is 12, 19, 0.445577892228273\n",
      "best merge is 0, 7, 0.534321444344366\n",
      "best merge is 0, 11, 0.615101572348603\n",
      "best merge is 20, 22, 0.624979940928123\n",
      "best merge is 12, 25, 0.627193646732538\n",
      "best merge is 8, 12, 0.698077981422246\n",
      "best merge is 10, 17, 0.953138901375522\n",
      "order classes\n",
      "16 - 0.516013072559958\n",
      "4 - 3.08278679309257\n",
      "2 - 2.14512941493145\n",
      "2 - 1.13176683315731\n",
      "2 - 4.0623696160328\n",
      "0 16 1 4 2 23 21 13 6 14 24 3 5 7 9 11\n",
      "8 12 19 25\n",
      "10 15\n",
      "17 18\n",
      "20 22\n",
      "Done.Start drawing\n",
      "280.55\t0.00\n",
      "nuclear chromosome, telomeric region, GO:0000784\n",
      "Rab GTPase binding, GO:0017137\n",
      "low-density lipoprotein receptor activity, GO:0005041\n",
      "decidualization, GO:0046697\n",
      "structural constituent of ribosome, GO:0003735\n",
      "protein targeting to mitochondrion, GO:0006626\n",
      "respiratory chain, GO:0070469\n",
      "mitochondrial ATP synthesis coupled proton transport, GO:0042776\n",
      "spliceosomal complex, GO:0005681\n",
      "calcium channel inhibitor activity, GO:0019855\n",
      "histone exchange, GO:0043486\n",
      "DNA methylation, GO:0006306\n",
      "translation initiation factor activity, GO:0003743\n",
      "extracellular vesicle, GO:1903561\n",
      "retrograde vesicle-mediated transport, Golgi to ER, GO:0006890\n",
      "transcription elongation from RNA polymerase II promoter, GO:0006368\n",
      "proteasome complex, GO:0000502\n",
      "protein folding, GO:0006457\n",
      "response to virus, GO:0009615\n",
      "embryonic limb morphogenesis, GO:0030326\n",
      "motor neuron axon guidance, GO:0008045\n",
      "signal transducer, downstream of receptor, with serine/threonine kinase activity, GO:0004702\n",
      "hydrolase activity, acting on carbon-nitrogen (but not peptide) bonds, in cyclic amidines, GO:0016814\n",
      "cardiac conduction, GO:0061337\n",
      "myosin complex, GO:0016459\n",
      "intermediate filament cytoskeleton, GO:0045111\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_P_LVM1_1.csv_PAGE/filt_FC_P_LVM1_1.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_P_LVM1_1.csv_PAGE/filt_FC_P_LVM1_1.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_P_LVM1_1.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_P_LVM1_1.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 1, 4, 0.0141428386765209\n",
      "best merge is 1, 2, 0.0181891030558878\n",
      "best merge is 0, 16, 0.0202532795338033\n",
      "best merge is 0, 1, 0.0276977850196122\n",
      "best merge is 0, 23, 0.0466816696839939\n",
      "best merge is 0, 21, 0.0762707184641387\n",
      "best merge is 3, 5, 0.0855030131616206\n",
      "best merge is 0, 13, 0.0890702748583888\n",
      "best merge is 6, 14, 0.0902547077667593\n",
      "best merge is 17, 18, 0.174117974331894\n",
      "best merge is 7, 9, 0.191208282168867\n",
      "best merge is 0, 6, 0.198106182834498\n",
      "best merge is 10, 15, 0.330019909989455\n",
      "best merge is 0, 24, 0.35306101894559\n",
      "best merge is 0, 3, 0.404923223576099\n",
      "best merge is 12, 19, 0.445577892228273\n",
      "best merge is 0, 7, 0.534321444344366\n",
      "best merge is 0, 11, 0.615101572348603\n",
      "best merge is 20, 22, 0.624979940928123\n",
      "best merge is 12, 25, 0.627193646732538\n",
      "best merge is 8, 12, 0.698077981422246\n",
      "best merge is 10, 17, 0.953138901375522\n",
      "order classes\n",
      "16 - 0.516013072559958\n",
      "4 - 3.08278679309257\n",
      "2 - 2.14512941493145\n",
      "2 - 1.13176683315731\n",
      "2 - 4.0623696160328\n",
      "0 16 1 4 2 23 21 13 6 14 24 3 5 7 9 11\n",
      "8 12 19 25\n",
      "10 15\n",
      "17 18\n",
      "20 22\n",
      "Done.-20\t42\t0\n",
      "-20\t42\t0\n",
      "-19.9985029406523\t42\t0\n",
      "-19.994760292283\t42\t0.00299411869541971\n",
      "-19.9910176439137\t42\t0.00748529673854928\n",
      "-19.9880235252183\t42\t0.010479415433969\n",
      "-19.9221529139191\t42\t0.0134735341293887\n",
      "-19.7649616824096\t42\t0.142220638032436\n",
      "1.16392799857423\t42\t0.327855997148458\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_P_LVM1_1.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM1_1.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl/human_ensembl.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM1_1.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 13290... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 17689\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 13290 out of 13290 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 13290\t17689\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1477 hits go_terms_num = 17689\n",
      "cat 836, mi = 0.00883444, translational initiation\n",
      "cat 692, mi = 0.00776883, structural constituent of ribosome\n",
      "cat 2546, mi = 0.00772778, SRP-dependent cotranslational protein targeting to membrane\n",
      "cat 2545, mi = 0.00723093, nuclear-transcribed mRNA catabolic process, nonsense-mediated decay\n",
      "cat 1498, mi = 0.00717670, viral transcription\n",
      "cat 2016, mi = 0.00506557, cytosolic large ribosomal subunit\n",
      "cat 2023, mi = 0.00415847, DNA replication\n",
      "cat 7, mi = 0.00340661, respiratory chain\n",
      "cat 1872, mi = 0.00339228, spliceosomal complex\n",
      "cat 104, mi = 0.00313202, hormone activity\n",
      "cat 4152, mi = 0.00300731, cytosolic small ribosomal subunit\n",
      "cat 693, mi = 0.00300021, mitochondrial translational elongation\n",
      "cat 694, mi = 0.00291366, mitochondrial translational termination\n",
      "cat 4159, mi = 0.00285900, small ribosomal subunit\n",
      "cat 1247, mi = 0.00283055, ribosome biogenesis\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 836, mi = 0.009, pvalue = 0.000, translational initiation\n",
      "Passed: cat 2023, mi = 0.004, pvalue = 0.000, DNA replication\n",
      "Passed: cat 7, mi = 0.003, pvalue = 0.000, respiratory chain\n",
      "Passed: cat 1872, mi = 0.003, pvalue = 0.000, spliceosomal complex\n",
      "Passed: cat 104, mi = 0.003, pvalue = 0.000, hormone activity\n",
      "Passed: cat 693, mi = 0.003, pvalue = 0.000, mitochondrial translational elongation\n",
      "Passed: cat 893, mi = 0.003, pvalue = 0.000, protein folding\n",
      "Passed: cat 779, mi = 0.002, pvalue = 0.000, antigen processing and presentation of exogenous peptide antigen via MHC class I, TAP-dependent\n",
      "Passed: cat 2285, mi = 0.002, pvalue = 0.000, inner ear morphogenesis\n",
      "Passed: cat 6821, mi = 0.002, pvalue = 0.000, Rac protein signal transduction\n",
      "Passed: cat 1909, mi = 0.002, pvalue = 0.000, cytoskeleton organization\n",
      "Passed: cat 1118, mi = 0.002, pvalue = 0.000, aminoacyl-tRNA ligase activity\n",
      "Passed: cat 3157, mi = 0.002, pvalue = 0.000, muscle contraction\n",
      "Passed: cat 2789, mi = 0.002, pvalue = 0.000, presynaptic membrane\n",
      "Passed: cat 903, mi = 0.001, pvalue = 0.001, late endosome membrane\n",
      "Passed: cat 56, mi = 0.001, pvalue = 0.001, cristae formation\n",
      "Passed: cat 2237, mi = 0.001, pvalue = 0.002, cellular response to cadmium ion\n",
      "Passed: cat 1719, mi = 0.001, pvalue = 0.001, cellular response to oxidative stress\n",
      "Passed: cat 1363, mi = 0.001, pvalue = 0.001, chemokine-mediated signaling pathway\n",
      "Passed: cat 6078, mi = 0.001, pvalue = 0.002, canonical glycolysis\n",
      "Passed: cat 1204, mi = 0.001, pvalue = 0.004, anchored component of membrane\n",
      "Passed: cat 1487, mi = 0.001, pvalue = 0.004, nuclear envelope\n",
      "Passed: cat 2291, mi = 0.001, pvalue = 0.004, potassium ion transmembrane transport\n",
      "Didn't pass (cutoff=0.0050): cat 646, mi = 0.001, pvalue = 0.005, integral component of Golgi membrane\n",
      "Passed: cat 1449, mi = 0.001, pvalue = 0.004, lateral plasma membrane\n",
      "Passed: cat 3188, mi = 0.001, pvalue = 0.005, branched-chain amino acid catabolic process\n",
      "Didn't pass (cutoff=0.0050): cat 6976, mi = 0.001, pvalue = 0.005, preribosome, large subunit precursor\n",
      "Didn't pass (cutoff=0.0050): cat 752, mi = 0.001, pvalue = 0.006, stem cell population maintenance\n",
      "Didn't pass (cutoff=0.0050): cat 1504, mi = 0.001, pvalue = 0.008, N-acetyltransferase activity\n",
      "Passed: cat 8428, mi = 0.001, pvalue = 0.005, negative regulation of G1/S transition of mitotic cell cycle\n",
      "Didn't pass (cutoff=0.0050): cat 1604, mi = 0.001, pvalue = 0.007, pituitary gland development\n",
      "Didn't pass (cutoff=0.0050): cat 2340, mi = 0.001, pvalue = 0.011, learning\n",
      "Didn't pass (cutoff=0.0050): cat 3843, mi = 0.001, pvalue = 0.009, sphingolipid metabolic process\n",
      "Didn't pass (cutoff=0.0050): cat 154, mi = 0.001, pvalue = 0.006, Notch binding\n",
      "Didn't pass (cutoff=0.0050): cat 2434, mi = 0.001, pvalue = 0.005, acetyltransferase activity\n",
      "Didn't pass (cutoff=0.0050): cat 7508, mi = 0.001, pvalue = 0.005, regulation of sodium ion transport\n",
      "Didn't pass (cutoff=0.0050): cat 944, mi = 0.001, pvalue = 0.008, transcription elongation from RNA polymerase II promoter\n",
      "Didn't pass (cutoff=0.0050): cat 6379, mi = 0.001, pvalue = 0.009, vitamin D receptor binding\n",
      "Didn't pass (cutoff=0.0050): cat 6022, mi = 0.001, pvalue = 0.010, positive regulation of muscle cell differentiation\n",
      "Passed: cat 5781, mi = 0.001, pvalue = 0.004, mitotic recombination\n",
      "Didn't pass (cutoff=0.0050): cat 5423, mi = 0.001, pvalue = 0.009, COPI-coated vesicle membrane\n",
      "Didn't pass (cutoff=0.0050): cat 1865, mi = 0.001, pvalue = 0.017, nucleosome binding\n",
      "Didn't pass (cutoff=0.0050): cat 4821, mi = 0.001, pvalue = 0.011, polyamine biosynthetic process\n",
      "Didn't pass (cutoff=0.0050): cat 3993, mi = 0.001, pvalue = 0.016, ligand-dependent nuclear receptor transcription coactivator activity\n",
      "Didn't pass (cutoff=0.0050): cat 1249, mi = 0.001, pvalue = 0.021, small-subunit processome\n",
      "Didn't pass (cutoff=0.0050): cat 4332, mi = 0.001, pvalue = 0.024, mRNA polyadenylation\n",
      "Didn't pass (cutoff=0.0050): cat 1244, mi = 0.001, pvalue = 0.021, epidermis development\n",
      "Didn't pass (cutoff=0.0050): cat 3115, mi = 0.001, pvalue = 0.025, long-term memory\n",
      "Didn't pass (cutoff=0.0050): cat 4129, mi = 0.001, pvalue = 0.026, RNA polymerase II repressing transcription factor binding\n",
      "Didn't pass (cutoff=0.0050): cat 1981, mi = 0.001, pvalue = 0.024, tumor necrosis factor-activated receptor activity\n",
      "Didn't pass (cutoff=0.0050): cat 512, mi = 0.001, pvalue = 0.023, transcriptional repressor complex\n",
      "Didn't pass (cutoff=0.0050): cat 839, mi = 0.001, pvalue = 0.018, thiol-dependent ubiquitinyl hydrolase activity\n",
      "Didn't pass (cutoff=0.0050): cat 1741, mi = 0.001, pvalue = 0.021, regulation of gene expression, epigenetic\n",
      "Didn't pass (cutoff=0.0050): cat 5081, mi = 0.001, pvalue = 0.021, STAGA complex\n",
      "Didn't pass (cutoff=0.0050): cat 6378, mi = 0.001, pvalue = 0.025, intracellular steroid hormone receptor signaling pathway\n",
      "Didn't pass (cutoff=0.0050): cat 2794, mi = 0.001, pvalue = 0.012, voltage-gated calcium channel complex\n",
      "Didn't pass (cutoff=0.0050): cat 7704, mi = 0.001, pvalue = 0.014, determination of adult lifespan\n",
      "Didn't pass (cutoff=0.0050): cat 8946, mi = 0.001, pvalue = 0.022, ER membrane protein complex\n",
      "Didn't pass (cutoff=0.0050): cat 4047, mi = 0.001, pvalue = 0.031, extrinsic apoptotic signaling pathway\n",
      "Didn't pass (cutoff=0.0050): cat 4439, mi = 0.001, pvalue = 0.038, face morphogenesis\n",
      "Total number of categories evaluated = 306.\n",
      "Number of categories that passed the tests = 27\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM1_1.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM1_1.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 3, 5, 0.0953446075018112\n",
      "best merge is 16, 25, 0.0989790856401812\n",
      "best merge is 8, 18, 0.10105214351803\n",
      "best merge is 9, 17, 0.111196315348198\n",
      "best merge is 2, 7, 0.139051535579787\n",
      "best merge is 0, 24, 0.142000468700669\n",
      "best merge is 4, 22, 0.155659011707092\n",
      "best merge is 19, 26, 0.178526945423453\n",
      "best merge is 3, 6, 0.188855510596707\n",
      "best merge is 8, 20, 0.229313397898814\n",
      "best merge is 2, 15, 0.24761321197886\n",
      "best merge is 1, 12, 0.249429730280427\n",
      "best merge is 4, 13, 0.284148347916469\n",
      "best merge is 0, 2, 0.308738007054245\n",
      "best merge is 8, 23, 0.336950004719417\n",
      "best merge is 3, 14, 0.361717400756975\n",
      "best merge is 0, 3, 0.423151586810605\n",
      "best merge is 9, 21, 0.44226071893533\n",
      "best merge is 4, 8, 0.445894585424985\n",
      "best merge is 1, 10, 0.479348434434902\n",
      "best merge is 11, 19, 0.601520232507066\n",
      "best merge is 0, 9, 0.661209889776094\n",
      "best merge is 1, 4, 0.720061945995239\n",
      "order classes\n",
      "12 - 0.984322837966103\n",
      "3 - 2.41625319830046\n",
      "7 - 1.20362059198292\n",
      "3 - 2.76313482087517\n",
      "2 - 0.668108828071223\n",
      "16 25\n",
      "1 12 10\n",
      "4 22 13 8 18 20 23\n",
      "11 19 26\n",
      "0 24 2 7 15 3 5 6 14 9 17 21\n",
      "Done.Start drawing\n",
      "inf\t0.00\n",
      "DNA replication, GO:0006260\n",
      "muscle contraction, GO:0006936\n",
      "cytoskeleton organization, GO:0007010\n",
      "chemokine-mediated signaling pathway, GO:0070098\n",
      "anchored component of membrane, GO:0031225\n",
      "cellular response to cadmium ion, GO:0071276\n",
      "negative regulation of G1/S transition of mitotic cell cycle, GO:2000134\n",
      "aminoacyl-tRNA ligase activity, GO:0004812\n",
      "canonical glycolysis, GO:0061621\n",
      "mitotic recombination, GO:0006312\n",
      "protein folding, GO:0006457\n",
      "spliceosomal complex, GO:0005681\n",
      "antigen processing and presentation of exogenous peptide antigen via MHC class I, TAP-dependent, GO:0002479\n",
      "late endosome membrane, GO:0031902\n",
      "Rac protein signal transduction, GO:0016601\n",
      "cellular response to oxidative stress, GO:0034599\n",
      "nuclear envelope, GO:0005635\n",
      "translational initiation, GO:0006413\n",
      "branched-chain amino acid catabolic process, GO:0009083\n",
      "respiratory chain, GO:0070469\n",
      "cristae formation, GO:0042407\n",
      "mitochondrial translational elongation, GO:0070125\n",
      "hormone activity, GO:0005179\n",
      "potassium ion transmembrane transport, GO:0071805\n",
      "presynaptic membrane, GO:0042734\n",
      "inner ear morphogenesis, GO:0042472\n",
      "lateral plasma membrane, GO:0016328\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM1_1.csv_PAGE/filt_FC_LVM1_1.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM1_1.csv_PAGE/filt_FC_LVM1_1.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM1_1.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM1_1.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 3, 5, 0.0953446075018112\n",
      "best merge is 16, 25, 0.0989790856401812\n",
      "best merge is 8, 18, 0.10105214351803\n",
      "best merge is 9, 17, 0.111196315348198\n",
      "best merge is 2, 7, 0.139051535579787\n",
      "best merge is 0, 24, 0.142000468700669\n",
      "best merge is 4, 22, 0.155659011707092\n",
      "best merge is 19, 26, 0.178526945423453\n",
      "best merge is 3, 6, 0.188855510596707\n",
      "best merge is 8, 20, 0.229313397898814\n",
      "best merge is 2, 15, 0.24761321197886\n",
      "best merge is 1, 12, 0.249429730280427\n",
      "best merge is 4, 13, 0.284148347916469\n",
      "best merge is 0, 2, 0.308738007054245\n",
      "best merge is 8, 23, 0.336950004719417\n",
      "best merge is 3, 14, 0.361717400756975\n",
      "best merge is 0, 3, 0.423151586810605\n",
      "best merge is 9, 21, 0.44226071893533\n",
      "best merge is 4, 8, 0.445894585424985\n",
      "best merge is 1, 10, 0.479348434434902\n",
      "best merge is 11, 19, 0.601520232507066\n",
      "best merge is 0, 9, 0.661209889776094\n",
      "best merge is 1, 4, 0.720061945995239\n",
      "order classes\n",
      "12 - 0.984322837966103\n",
      "3 - 2.41625319830046\n",
      "7 - 1.20362059198292\n",
      "3 - 2.76313482087517\n",
      "2 - 0.668108828071223\n",
      "16 25\n",
      "1 12 10\n",
      "4 22 13 8 18 20 23\n",
      "11 19 26\n",
      "0 24 2 7 15 3 5 6 14 9 17 21\n",
      "Done.-20\t42\t0\n",
      "-20\t42\t0\n",
      "-20\t42\t0\n",
      "-20\t42\t0\n",
      "-20\t42\t0\n",
      "-20\t42\t0\n",
      "-20\t42\t0\n",
      "-20\t42\t0\n",
      "NaN\t42\t0\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM1_1.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM1_3.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl/human_ensembl.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM1_3.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 13290... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 17689\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 13290 out of 13290 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 13290\t17689\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1477 hits go_terms_num = 17689\n",
      "cat 692, mi = 0.01002594, structural constituent of ribosome\n",
      "cat 2546, mi = 0.00874512, SRP-dependent cotranslational protein targeting to membrane\n",
      "cat 2545, mi = 0.00836239, nuclear-transcribed mRNA catabolic process, nonsense-mediated decay\n",
      "cat 1498, mi = 0.00830799, viral transcription\n",
      "cat 836, mi = 0.00802902, translational initiation\n",
      "cat 2016, mi = 0.00485097, cytosolic large ribosomal subunit\n",
      "cat 2023, mi = 0.00466659, DNA replication\n",
      "cat 694, mi = 0.00427647, mitochondrial translational termination\n",
      "cat 693, mi = 0.00409894, mitochondrial translational elongation\n",
      "cat 4152, mi = 0.00409792, cytosolic small ribosomal subunit\n",
      "cat 2121, mi = 0.00407699, sister chromatid cohesion\n",
      "cat 2990, mi = 0.00367585, chromosome, centromeric region\n",
      "cat 4583, mi = 0.00343494, condensed chromosome kinetochore\n",
      "cat 4159, mi = 0.00335396, small ribosomal subunit\n",
      "cat 696, mi = 0.00323328, mitochondrial large ribosomal subunit\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 692, mi = 0.010, pvalue = 0.000, structural constituent of ribosome\n",
      "Passed: cat 2023, mi = 0.005, pvalue = 0.000, DNA replication\n",
      "Passed: cat 2121, mi = 0.004, pvalue = 0.000, sister chromatid cohesion\n",
      "Passed: cat 2431, mi = 0.003, pvalue = 0.000, nucleosome\n",
      "Passed: cat 7, mi = 0.003, pvalue = 0.000, respiratory chain\n",
      "Passed: cat 1872, mi = 0.002, pvalue = 0.000, spliceosomal complex\n",
      "Passed: cat 1256, mi = 0.002, pvalue = 0.000, tRNA processing\n",
      "Passed: cat 552, mi = 0.002, pvalue = 0.000, condensed chromosome\n",
      "Passed: cat 1079, mi = 0.002, pvalue = 0.000, Wnt signaling pathway, planar cell polarity pathway\n",
      "Passed: cat 2536, mi = 0.002, pvalue = 0.000, embryo development\n",
      "Passed: cat 944, mi = 0.002, pvalue = 0.000, transcription elongation from RNA polymerase II promoter\n",
      "Passed: cat 931, mi = 0.002, pvalue = 0.000, cell fate commitment\n",
      "Passed: cat 104, mi = 0.002, pvalue = 0.000, hormone activity\n",
      "Passed: cat 2997, mi = 0.002, pvalue = 0.000, positive regulation of epithelial cell migration\n",
      "Passed: cat 2783, mi = 0.002, pvalue = 0.000, ubiquitin ligase complex\n",
      "Passed: cat 835, mi = 0.002, pvalue = 0.000, translation initiation factor activity\n",
      "Passed: cat 1912, mi = 0.002, pvalue = 0.000, retrograde transport, endosome to Golgi\n",
      "Passed: cat 328, mi = 0.002, pvalue = 0.000, cell redox homeostasis\n",
      "Passed: cat 4037, mi = 0.002, pvalue = 0.001, regulation of exocytosis\n",
      "Passed: cat 7057, mi = 0.002, pvalue = 0.000, transforming growth factor beta binding\n",
      "Passed: cat 2101, mi = 0.002, pvalue = 0.000, centriole\n",
      "Passed: cat 1843, mi = 0.002, pvalue = 0.001, steroid hormone receptor activity\n",
      "Passed: cat 1249, mi = 0.001, pvalue = 0.001, small-subunit processome\n",
      "Passed: cat 4516, mi = 0.001, pvalue = 0.000, response to lipid\n",
      "Passed: cat 91, mi = 0.001, pvalue = 0.001, integral component of mitochondrial inner membrane\n",
      "Passed: cat 629, mi = 0.001, pvalue = 0.002, regulation of ion transmembrane transport\n",
      "Passed: cat 617, mi = 0.001, pvalue = 0.002, basement membrane\n",
      "Passed: cat 6662, mi = 0.001, pvalue = 0.000, myoblast fusion\n",
      "Passed: cat 58, mi = 0.001, pvalue = 0.003, mitochondrial proton-transporting ATP synthase complex\n",
      "Passed: cat 2998, mi = 0.001, pvalue = 0.003, cell motility\n",
      "Passed: cat 3233, mi = 0.001, pvalue = 0.004, anterograde synaptic vesicle transport\n",
      "Passed: cat 2403, mi = 0.001, pvalue = 0.004, axon extension\n",
      "Passed: cat 8616, mi = 0.001, pvalue = 0.001, regulation of attachment of spindle microtubules to kinetochore\n",
      "Passed: cat 2174, mi = 0.001, pvalue = 0.001, Lys48-specific deubiquitinase activity\n",
      "Passed: cat 209, mi = 0.001, pvalue = 0.004, negative regulation of protein kinase activity\n",
      "Passed: cat 2747, mi = 0.001, pvalue = 0.002, 3'-UTR-mediated mRNA stabilization\n",
      "Didn't pass (cutoff=0.0050): cat 8359, mi = 0.001, pvalue = 0.005, positive regulation of receptor internalization\n",
      "Didn't pass (cutoff=0.0050): cat 4004, mi = 0.001, pvalue = 0.006, phosphatidylinositol 3-kinase binding\n",
      "Didn't pass (cutoff=0.0050): cat 2930, mi = 0.001, pvalue = 0.006, platelet-derived growth factor receptor signaling pathway\n",
      "Didn't pass (cutoff=0.0050): cat 2863, mi = 0.001, pvalue = 0.008, purine nucleotide biosynthetic process\n",
      "Didn't pass (cutoff=0.0050): cat 3790, mi = 0.001, pvalue = 0.006, carboxypeptidase activity\n",
      "Passed: cat 3426, mi = 0.001, pvalue = 0.003, male meiosis I\n",
      "Didn't pass (cutoff=0.0050): cat 4668, mi = 0.001, pvalue = 0.009, azurophil granule membrane\n",
      "Didn't pass (cutoff=0.0050): cat 887, mi = 0.001, pvalue = 0.013, regulation of endocytosis\n",
      "Didn't pass (cutoff=0.0050): cat 777, mi = 0.001, pvalue = 0.009, integral component of endoplasmic reticulum membrane\n",
      "Didn't pass (cutoff=0.0050): cat 3161, mi = 0.001, pvalue = 0.006, regulation of mitochondrion organization\n",
      "Didn't pass (cutoff=0.0050): cat 286, mi = 0.001, pvalue = 0.014, positive regulation of T cell proliferation\n",
      "Didn't pass (cutoff=0.0050): cat 517, mi = 0.001, pvalue = 0.011, phospholipase A2 activity\n",
      "Didn't pass (cutoff=0.0050): cat 4766, mi = 0.001, pvalue = 0.012, bicarbonate transport\n",
      "Passed: cat 6246, mi = 0.001, pvalue = 0.001, cellular response to copper ion\n",
      "Passed: cat 10477, mi = 0.001, pvalue = 0.002, calcium-transporting ATPase activity\n",
      "Didn't pass (cutoff=0.0050): cat 2278, mi = 0.001, pvalue = 0.012, phosphatidylinositol dephosphorylation\n",
      "Didn't pass (cutoff=0.0050): cat 1363, mi = 0.001, pvalue = 0.008, chemokine-mediated signaling pathway\n",
      "Didn't pass (cutoff=0.0050): cat 217, mi = 0.001, pvalue = 0.012, dioxygenase activity\n",
      "Didn't pass (cutoff=0.0050): cat 5674, mi = 0.001, pvalue = 0.016, negative regulation of nucleic acid-templated transcription\n",
      "Didn't pass (cutoff=0.0050): cat 392, mi = 0.001, pvalue = 0.016, positive regulation of protein dephosphorylation\n",
      "Didn't pass (cutoff=0.0050): cat 5268, mi = 0.001, pvalue = 0.007, T cell receptor complex\n",
      "Didn't pass (cutoff=0.0050): cat 2952, mi = 0.001, pvalue = 0.009, insulin receptor substrate binding\n",
      "Didn't pass (cutoff=0.0050): cat 7965, mi = 0.001, pvalue = 0.011, RNA methyltransferase activity\n",
      "Didn't pass (cutoff=0.0050): cat 5945, mi = 0.001, pvalue = 0.014, positive regulation of mitochondrial fission\n",
      "Didn't pass (cutoff=0.0050): cat 892, mi = 0.001, pvalue = 0.019, chaperone-mediated protein folding\n",
      "Didn't pass (cutoff=0.0050): cat 5007, mi = 0.001, pvalue = 0.006, nuclear export signal receptor activity\n",
      "Didn't pass (cutoff=0.0050): cat 911, mi = 0.001, pvalue = 0.006, response to interferon-alpha\n",
      "Didn't pass (cutoff=0.0050): cat 4519, mi = 0.001, pvalue = 0.007, extracellular ligand-gated ion channel activity\n",
      "Didn't pass (cutoff=0.0050): cat 2851, mi = 0.001, pvalue = 0.023, oxidoreductase activity, acting on paired donors, with incorporation or reduction of molecular oxygen\n",
      "Didn't pass (cutoff=0.0050): cat 5087, mi = 0.001, pvalue = 0.021, negative regulation of GTPase activity\n",
      "Didn't pass (cutoff=0.0050): cat 1975, mi = 0.001, pvalue = 0.015, megakaryocyte development\n",
      "Didn't pass (cutoff=0.0050): cat 4333, mi = 0.001, pvalue = 0.022, mRNA cleavage and polyadenylation specificity factor complex\n",
      "Didn't pass (cutoff=0.0050): cat 6417, mi = 0.001, pvalue = 0.012, adrenal gland development\n",
      "Didn't pass (cutoff=0.0050): cat 3103, mi = 0.001, pvalue = 0.008, positive regulation of transcription of nuclear large rRNA transcript from RNA polymerase I promoter\n",
      "Didn't pass (cutoff=0.0050): cat 8591, mi = 0.001, pvalue = 0.015, positive regulation of extrinsic apoptotic signaling pathway in absence of ligand\n",
      "Total number of categories evaluated = 388.\n",
      "Number of categories that passed the tests = 39\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM1_3.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM1_3.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 11, 25, 0.0194095855007946\n",
      "best merge is 3, 32, 0.0263588653948414\n",
      "best merge is 3, 36, 0.0331567282012901\n",
      "best merge is 1, 2, 0.0496836140434038\n",
      "best merge is 3, 12, 0.0556230658832917\n",
      "best merge is 5, 6, 0.0565411227967931\n",
      "best merge is 22, 24, 0.0567183192778472\n",
      "best merge is 1, 7, 0.0946450820301963\n",
      "best merge is 0, 22, 0.0994123341579953\n",
      "best merge is 3, 18, 0.130642639875474\n",
      "best merge is 8, 10, 0.140698783329262\n",
      "best merge is 13, 19, 0.151689287897204\n",
      "best merge is 1, 3, 0.153225784296014\n",
      "best merge is 30, 35, 0.1610585681489\n",
      "best merge is 4, 28, 0.164637694440289\n",
      "best merge is 16, 34, 0.187628351889985\n",
      "best merge is 8, 15, 0.200845075702481\n",
      "best merge is 13, 27, 0.217174621578178\n",
      "best merge is 9, 29, 0.235343849311634\n",
      "best merge is 5, 8, 0.242180973008948\n",
      "best merge is 1, 11, 0.243344152959043\n",
      "best merge is 0, 4, 0.247766673860443\n",
      "best merge is 14, 30, 0.333405320337547\n",
      "best merge is 13, 26, 0.364458267316398\n",
      "best merge is 0, 5, 0.393380267589565\n",
      "best merge is 14, 33, 0.417267140038861\n",
      "best merge is 1, 20, 0.420741096684092\n",
      "best merge is 16, 17, 0.425608109000167\n",
      "best merge is 13, 21, 0.480920998141742\n",
      "best merge is 9, 14, 0.530109461874617\n",
      "best merge is 0, 31, 0.647124660696111\n",
      "best merge is 1, 23, 0.65966527511305\n",
      "best merge is 13, 37, 0.712133877220785\n",
      "best merge is 0, 16, 0.732981499261758\n",
      "best merge is 0, 38, 0.789135414400262\n",
      "order classes\n",
      "14 - 1.29313380069781\n",
      "12 - 0.843921977140628\n",
      "6 - 2.36740478478441\n",
      "6 - 2.58691575699326\n",
      "1 2 7 3 32 36 12 18 11 25 20 23\n",
      "0 22 24 4 28 5 6 8 10 15 31 16 34 17\n",
      "9 29 14 30 35 33\n",
      "13 19 27 26 21 37\n",
      "38\n",
      "Done.Start drawing\n",
      "inf\t0.00\n",
      "cell motility, GO:0048870\n",
      "positive regulation of epithelial cell migration, GO:0010634\n",
      "transforming growth factor beta binding, GO:0050431\n",
      "myoblast fusion, GO:0007520\n",
      "basement membrane, GO:0005604\n",
      "response to lipid, GO:0033993\n",
      "retrograde transport, endosome to Golgi, GO:0042147\n",
      "negative regulation of protein kinase activity, GO:0006469\n",
      "steroid hormone receptor activity, GO:0003707\n",
      "translation initiation factor activity, GO:0003743\n",
      "embryo development, GO:0009790\n",
      "ubiquitin ligase complex, GO:0000151\n",
      "3'-UTR-mediated mRNA stabilization, GO:0070935\n",
      "Lys48-specific deubiquitinase activity, GO:1990380\n",
      "cellular response to copper ion, GO:0071280\n",
      "spliceosomal complex, GO:0005681\n",
      "anterograde synaptic vesicle transport, GO:0048490\n",
      "calcium-transporting ATPase activity, GO:0005388\n",
      "respiratory chain, GO:0070469\n",
      "tRNA processing, GO:0008033\n",
      "axon extension, GO:0048675\n",
      "structural constituent of ribosome, GO:0003735\n",
      "small-subunit processome, GO:0032040\n",
      "integral component of mitochondrial inner membrane, GO:0031305\n",
      "Wnt signaling pathway, planar cell polarity pathway, GO:0060071\n",
      "transcription elongation from RNA polymerase II promoter, GO:0006368\n",
      "cell redox homeostasis, GO:0045454\n",
      "centriole, GO:0005814\n",
      "mitochondrial proton-transporting ATP synthase complex, GO:0005753\n",
      "DNA replication, GO:0006260\n",
      "sister chromatid cohesion, GO:0007062\n",
      "condensed chromosome, GO:0000793\n",
      "nucleosome, GO:0000786\n",
      "regulation of attachment of spindle microtubules to kinetochore, GO:0051988\n",
      "male meiosis I, GO:0007141\n",
      "hormone activity, GO:0005179\n",
      "regulation of exocytosis, GO:0017157\n",
      "cell fate commitment, GO:0045165\n",
      "regulation of ion transmembrane transport, GO:0034765\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM1_3.csv_PAGE/filt_FC_LVM1_3.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM1_3.csv_PAGE/filt_FC_LVM1_3.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM1_3.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM1_3.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 11, 25, 0.0194095855007946\n",
      "best merge is 3, 32, 0.0263588653948414\n",
      "best merge is 3, 36, 0.0331567282012901\n",
      "best merge is 1, 2, 0.0496836140434038\n",
      "best merge is 3, 12, 0.0556230658832917\n",
      "best merge is 5, 6, 0.0565411227967931\n",
      "best merge is 22, 24, 0.0567183192778472\n",
      "best merge is 1, 7, 0.0946450820301963\n",
      "best merge is 0, 22, 0.0994123341579953\n",
      "best merge is 3, 18, 0.130642639875474\n",
      "best merge is 8, 10, 0.140698783329262\n",
      "best merge is 13, 19, 0.151689287897204\n",
      "best merge is 1, 3, 0.153225784296014\n",
      "best merge is 30, 35, 0.1610585681489\n",
      "best merge is 4, 28, 0.164637694440289\n",
      "best merge is 16, 34, 0.187628351889985\n",
      "best merge is 8, 15, 0.200845075702481\n",
      "best merge is 13, 27, 0.217174621578178\n",
      "best merge is 9, 29, 0.235343849311634\n",
      "best merge is 5, 8, 0.242180973008948\n",
      "best merge is 1, 11, 0.243344152959043\n",
      "best merge is 0, 4, 0.247766673860443\n",
      "best merge is 14, 30, 0.333405320337547\n",
      "best merge is 13, 26, 0.364458267316398\n",
      "best merge is 0, 5, 0.393380267589565\n",
      "best merge is 14, 33, 0.417267140038861\n",
      "best merge is 1, 20, 0.420741096684092\n",
      "best merge is 16, 17, 0.425608109000167\n",
      "best merge is 13, 21, 0.480920998141742\n",
      "best merge is 9, 14, 0.530109461874617\n",
      "best merge is 0, 31, 0.647124660696111\n",
      "best merge is 1, 23, 0.65966527511305\n",
      "best merge is 13, 37, 0.712133877220785\n",
      "best merge is 0, 16, 0.732981499261758\n",
      "best merge is 0, 38, 0.789135414400262\n",
      "order classes\n",
      "14 - 1.29313380069781\n",
      "12 - 0.843921977140628\n",
      "6 - 2.36740478478441\n",
      "6 - 2.58691575699326\n",
      "1 2 7 3 32 36 12 18 11 25 20 23\n",
      "0 22 24 4 28 5 6 8 10 15 31 16 34 17\n",
      "9 29 14 30 35 33\n",
      "13 19 27 26 21 37\n",
      "38\n",
      "Done.-20\t42\t0\n",
      "-20\t42\t0\n",
      "-20\t42\t0\n",
      "-20\t42\t0\n",
      "-20\t42\t0\n",
      "-20\t42\t0\n",
      "-20\t42\t0\n",
      "-20\t42\t0\n",
      "NaN\t42\t0\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM1_3.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM1_4.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl/human_ensembl.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM1_4.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 13290... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 17689\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 13290 out of 13290 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 13290\t17689\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1477 hits go_terms_num = 17689\n",
      "cat 1498, mi = 0.00920373, viral transcription\n",
      "cat 2546, mi = 0.00901496, SRP-dependent cotranslational protein targeting to membrane\n",
      "cat 836, mi = 0.00767527, translational initiation\n",
      "cat 2545, mi = 0.00748110, nuclear-transcribed mRNA catabolic process, nonsense-mediated decay\n",
      "cat 692, mi = 0.00597230, structural constituent of ribosome\n",
      "cat 2016, mi = 0.00558356, cytosolic large ribosomal subunit\n",
      "cat 1872, mi = 0.00481018, spliceosomal complex\n",
      "cat 1076, mi = 0.00369530, regulation of mRNA stability\n",
      "cat 694, mi = 0.00353351, mitochondrial translational termination\n",
      "cat 4152, mi = 0.00338570, cytosolic small ribosomal subunit\n",
      "cat 693, mi = 0.00326564, mitochondrial translational elongation\n",
      "cat 1070, mi = 0.00305602, regulation of cellular amino acid metabolic process\n",
      "cat 1064, mi = 0.00299804, proteasome complex\n",
      "cat 682, mi = 0.00296086, regulation of translation\n",
      "cat 1876, mi = 0.00286406, catalytic step 2 spliceosome\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 1498, mi = 0.009, pvalue = 0.000, viral transcription\n",
      "Passed: cat 1872, mi = 0.005, pvalue = 0.000, spliceosomal complex\n",
      "Passed: cat 1076, mi = 0.004, pvalue = 0.000, regulation of mRNA stability\n",
      "Passed: cat 694, mi = 0.004, pvalue = 0.000, mitochondrial translational termination\n",
      "Passed: cat 1247, mi = 0.003, pvalue = 0.000, ribosome biogenesis\n",
      "Passed: cat 944, mi = 0.003, pvalue = 0.000, transcription elongation from RNA polymerase II promoter\n",
      "Passed: cat 7, mi = 0.002, pvalue = 0.000, respiratory chain\n",
      "Passed: cat 104, mi = 0.002, pvalue = 0.000, hormone activity\n",
      "Passed: cat 2291, mi = 0.002, pvalue = 0.000, potassium ion transmembrane transport\n",
      "Passed: cat 2759, mi = 0.002, pvalue = 0.000, cornification\n",
      "Passed: cat 766, mi = 0.002, pvalue = 0.000, calcium ion-regulated exocytosis of neurotransmitter\n",
      "Passed: cat 56, mi = 0.002, pvalue = 0.000, cristae formation\n",
      "Passed: cat 3903, mi = 0.002, pvalue = 0.000, histone deacetylation\n",
      "Passed: cat 3047, mi = 0.002, pvalue = 0.000, tRNA binding\n",
      "Passed: cat 777, mi = 0.002, pvalue = 0.000, integral component of endoplasmic reticulum membrane\n",
      "Passed: cat 3022, mi = 0.002, pvalue = 0.000, endodeoxyribonuclease activity\n",
      "Passed: cat 803, mi = 0.002, pvalue = 0.001, chaperone binding\n",
      "Passed: cat 922, mi = 0.002, pvalue = 0.001, E-box binding\n",
      "Passed: cat 1164, mi = 0.002, pvalue = 0.001, retromer complex\n",
      "Passed: cat 3250, mi = 0.001, pvalue = 0.001, positive regulation of synapse assembly\n",
      "Passed: cat 1204, mi = 0.001, pvalue = 0.001, anchored component of membrane\n",
      "Passed: cat 855, mi = 0.001, pvalue = 0.002, chemotaxis\n",
      "Passed: cat 1286, mi = 0.001, pvalue = 0.001, midbody\n",
      "Passed: cat 157, mi = 0.001, pvalue = 0.002, heparin binding\n",
      "Passed: cat 4424, mi = 0.001, pvalue = 0.001, regulation of canonical Wnt signaling pathway\n",
      "Passed: cat 4525, mi = 0.001, pvalue = 0.001, skeletal muscle contraction\n",
      "Passed: cat 839, mi = 0.001, pvalue = 0.002, thiol-dependent ubiquitinyl hydrolase activity\n",
      "Passed: cat 2431, mi = 0.001, pvalue = 0.003, nucleosome\n",
      "Passed: cat 5784, mi = 0.001, pvalue = 0.001, Fanconi anaemia nuclear complex\n",
      "Passed: cat 6022, mi = 0.001, pvalue = 0.003, positive regulation of muscle cell differentiation\n",
      "Passed: cat 883, mi = 0.001, pvalue = 0.004, protein serine/threonine phosphatase activity\n",
      "Passed: cat 5268, mi = 0.001, pvalue = 0.001, T cell receptor complex\n",
      "Passed: cat 4073, mi = 0.001, pvalue = 0.004, mesoderm development\n",
      "Passed: cat 5423, mi = 0.001, pvalue = 0.003, COPI-coated vesicle membrane\n",
      "Didn't pass (cutoff=0.0050): cat 2638, mi = 0.001, pvalue = 0.005, heterochromatin\n",
      "Didn't pass (cutoff=0.0050): cat 1162, mi = 0.001, pvalue = 0.008, protein localization to plasma membrane\n",
      "Didn't pass (cutoff=0.0050): cat 328, mi = 0.001, pvalue = 0.010, cell redox homeostasis\n",
      "Didn't pass (cutoff=0.0050): cat 835, mi = 0.001, pvalue = 0.009, translation initiation factor activity\n",
      "Didn't pass (cutoff=0.0050): cat 2745, mi = 0.001, pvalue = 0.011, response to interleukin-1\n",
      "Didn't pass (cutoff=0.0050): cat 1628, mi = 0.001, pvalue = 0.016, nucleotide metabolic process\n",
      "Didn't pass (cutoff=0.0050): cat 305, mi = 0.001, pvalue = 0.014, cell growth\n",
      "Didn't pass (cutoff=0.0050): cat 1673, mi = 0.001, pvalue = 0.016, protein complex scaffold\n",
      "Didn't pass (cutoff=0.0050): cat 1851, mi = 0.001, pvalue = 0.015, amino acid transmembrane transporter activity\n",
      "Didn't pass (cutoff=0.0050): cat 586, mi = 0.001, pvalue = 0.011, metalloaminopeptidase activity\n",
      "Didn't pass (cutoff=0.0050): cat 8209, mi = 0.001, pvalue = 0.016, peroxisomal matrix\n",
      "Didn't pass (cutoff=0.0050): cat 3003, mi = 0.001, pvalue = 0.014, double-stranded RNA binding\n",
      "Didn't pass (cutoff=0.0050): cat 3274, mi = 0.001, pvalue = 0.019, protein palmitoylation\n",
      "Didn't pass (cutoff=0.0050): cat 2117, mi = 0.001, pvalue = 0.019, BLOC-1 complex\n",
      "Didn't pass (cutoff=0.0050): cat 8112, mi = 0.001, pvalue = 0.013, motor neuron axon guidance\n",
      "Didn't pass (cutoff=0.0050): cat 11441, mi = 0.001, pvalue = 0.009, small protein activating enzyme activity\n",
      "Didn't pass (cutoff=0.0050): cat 5004, mi = 0.001, pvalue = 0.020, Ino80 complex\n",
      "Didn't pass (cutoff=0.0050): cat 1175, mi = 0.001, pvalue = 0.017, guanyl-nucleotide exchange factor activity\n",
      "Didn't pass (cutoff=0.0050): cat 2422, mi = 0.001, pvalue = 0.010, UDP-galactose:beta-N-acetylglucosamine beta-1,3-galactosyltransferase activity\n",
      "Didn't pass (cutoff=0.0050): cat 8181, mi = 0.001, pvalue = 0.010, peptidyl-arginine N-methylation\n",
      "Total number of categories evaluated = 282.\n",
      "Number of categories that passed the tests = 34\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM1_4.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM1_4.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 6, 13, 0.0212290401670396\n",
      "best merge is 7, 10, 0.0267935884206895\n",
      "best merge is 7, 19, 0.0485229790787881\n",
      "best merge is 5, 22, 0.0528614406087065\n",
      "best merge is 7, 8, 0.0644317980720561\n",
      "best merge is 4, 5, 0.0699981356004145\n",
      "best merge is 15, 28, 0.0752028304725948\n",
      "best merge is 1, 3, 0.0768402520945901\n",
      "best merge is 2, 4, 0.115622057959342\n",
      "best merge is 7, 27, 0.138840790697391\n",
      "best merge is 17, 21, 0.158778195975294\n",
      "best merge is 1, 2, 0.179705389627894\n",
      "best merge is 9, 24, 0.225900645605303\n",
      "best merge is 17, 23, 0.251145268417021\n",
      "best merge is 1, 16, 0.269405539322657\n",
      "best merge is 0, 29, 0.29504261815546\n",
      "best merge is 6, 11, 0.31767037620237\n",
      "best merge is 1, 18, 0.349162370070158\n",
      "best merge is 7, 20, 0.349844388290626\n",
      "best merge is 12, 26, 0.401718024623143\n",
      "best merge is 7, 25, 0.406243839581266\n",
      "best merge is 17, 31, 0.407095013365635\n",
      "best merge is 1, 14, 0.457437438141674\n",
      "best merge is 1, 12, 0.474467108629412\n",
      "best merge is 6, 33, 0.561054941965059\n",
      "best merge is 1, 30, 0.593901155560946\n",
      "best merge is 1, 6, 0.612892821752197\n",
      "best merge is 7, 9, 0.629824574034954\n",
      "best merge is 17, 32, 0.655206066354719\n",
      "best merge is 7, 17, 0.79200692423298\n",
      "order classes\n",
      "2 - 2.50786225432141\n",
      "16 - 0.946420303497108\n",
      "9 - 1.31547145129509\n",
      "2 - 0.639224059017056\n",
      "5 - 2.44972885284886\n",
      "15 28\n",
      "1 3 2 4 5 22 16 18 14 12 26 30 6 13 11 33\n",
      "7 10 19 8 27 20 25 9 24\n",
      "0 29\n",
      "17 21 23 31 32\n",
      "Done.Start drawing\n",
      "inf\t0.00\n",
      "E-box binding, GO:0070888\n",
      "chemotaxis, GO:0006935\n",
      "heparin binding, GO:0008201\n",
      "T cell receptor complex, GO:0042101\n",
      "regulation of mRNA stability, GO:0043488\n",
      "transcription elongation from RNA polymerase II promoter, GO:0006368\n",
      "midbody, GO:0030496\n",
      "chaperone binding, GO:0051087\n",
      "protein serine/threonine phosphatase activity, GO:0004722\n",
      "mesoderm development, GO:0007498\n",
      "spliceosomal complex, GO:0005681\n",
      "ribosome biogenesis, GO:0042254\n",
      "retromer complex, GO:0030904\n",
      "cristae formation, GO:0042407\n",
      "mitochondrial translational termination, GO:0070126\n",
      "integral component of endoplasmic reticulum membrane, GO:0030176\n",
      "COPI-coated vesicle membrane, GO:0030663\n",
      "histone deacetylation, GO:0016575\n",
      "thiol-dependent ubiquitinyl hydrolase activity, GO:0036459\n",
      "respiratory chain, GO:0070469\n",
      "tRNA binding, GO:0000049\n",
      "viral transcription, GO:0019083\n",
      "positive regulation of muscle cell differentiation, GO:0051149\n",
      "endodeoxyribonuclease activity, GO:0004520\n",
      "Fanconi anaemia nuclear complex, GO:0043240\n",
      "cornification, GO:0070268\n",
      "regulation of canonical Wnt signaling pathway, GO:0060828\n",
      "hormone activity, GO:0005179\n",
      "calcium ion-regulated exocytosis of neurotransmitter, GO:0048791\n",
      "positive regulation of synapse assembly, GO:0051965\n",
      "potassium ion transmembrane transport, GO:0071805\n",
      "nucleosome, GO:0000786\n",
      "anchored component of membrane, GO:0031225\n",
      "skeletal muscle contraction, GO:0003009\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM1_4.csv_PAGE/filt_FC_LVM1_4.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM1_4.csv_PAGE/filt_FC_LVM1_4.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM1_4.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM1_4.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 6, 13, 0.0212290401670396\n",
      "best merge is 7, 10, 0.0267935884206895\n",
      "best merge is 7, 19, 0.0485229790787881\n",
      "best merge is 5, 22, 0.0528614406087065\n",
      "best merge is 7, 8, 0.0644317980720561\n",
      "best merge is 4, 5, 0.0699981356004145\n",
      "best merge is 15, 28, 0.0752028304725948\n",
      "best merge is 1, 3, 0.0768402520945901\n",
      "best merge is 2, 4, 0.115622057959342\n",
      "best merge is 7, 27, 0.138840790697391\n",
      "best merge is 17, 21, 0.158778195975294\n",
      "best merge is 1, 2, 0.179705389627894\n",
      "best merge is 9, 24, 0.225900645605303\n",
      "best merge is 17, 23, 0.251145268417021\n",
      "best merge is 1, 16, 0.269405539322657\n",
      "best merge is 0, 29, 0.29504261815546\n",
      "best merge is 6, 11, 0.31767037620237\n",
      "best merge is 1, 18, 0.349162370070158\n",
      "best merge is 7, 20, 0.349844388290626\n",
      "best merge is 12, 26, 0.401718024623143\n",
      "best merge is 7, 25, 0.406243839581266\n",
      "best merge is 17, 31, 0.407095013365635\n",
      "best merge is 1, 14, 0.457437438141674\n",
      "best merge is 1, 12, 0.474467108629412\n",
      "best merge is 6, 33, 0.561054941965059\n",
      "best merge is 1, 30, 0.593901155560946\n",
      "best merge is 1, 6, 0.612892821752197\n",
      "best merge is 7, 9, 0.629824574034954\n",
      "best merge is 17, 32, 0.655206066354719\n",
      "best merge is 7, 17, 0.79200692423298\n",
      "order classes\n",
      "2 - 2.50786225432141\n",
      "16 - 0.946420303497108\n",
      "9 - 1.31547145129509\n",
      "2 - 0.639224059017056\n",
      "5 - 2.44972885284886\n",
      "15 28\n",
      "1 3 2 4 5 22 16 18 14 12 26 30 6 13 11 33\n",
      "7 10 19 8 27 20 25 9 24\n",
      "0 29\n",
      "17 21 23 31 32\n",
      "Done.-20\t42\t0\n",
      "-20\t42\t0\n",
      "-20\t42\t0\n",
      "-20\t42\t0\n",
      "-20\t42\t0\n",
      "-20\t42\t0\n",
      "-20\t42\t0\n",
      "-20\t42\t0\n",
      "NaN\t42\t0\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM1_4.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM2_5.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl/human_ensembl.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM2_5.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 13290... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 17689\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 13290 out of 13290 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 13290\t17689\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1477 hits go_terms_num = 17689\n",
      "cat 2546, mi = 0.00757328, SRP-dependent cotranslational protein targeting to membrane\n",
      "cat 1498, mi = 0.00713970, viral transcription\n",
      "cat 836, mi = 0.00660107, translational initiation\n",
      "cat 692, mi = 0.00625858, structural constituent of ribosome\n",
      "cat 2545, mi = 0.00567844, nuclear-transcribed mRNA catabolic process, nonsense-mediated decay\n",
      "cat 1872, mi = 0.00509627, spliceosomal complex\n",
      "cat 693, mi = 0.00437142, mitochondrial translational elongation\n",
      "cat 694, mi = 0.00435549, mitochondrial translational termination\n",
      "cat 2016, mi = 0.00405049, cytosolic large ribosomal subunit\n",
      "cat 1876, mi = 0.00378788, catalytic step 2 spliceosome\n",
      "cat 1076, mi = 0.00338300, regulation of mRNA stability\n",
      "cat 1072, mi = 0.00327727, anaphase-promoting complex-dependent catabolic process\n",
      "cat 1070, mi = 0.00323148, regulation of cellular amino acid metabolic process\n",
      "cat 1078, mi = 0.00314430, positive regulation of ubiquitin-protein ligase activity involved in regulation of mitotic cell cycle transition\n",
      "cat 1080, mi = 0.00306499, regulation of transcription from RNA polymerase II promoter in response to hypoxia\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 2546, mi = 0.008, pvalue = 0.000, SRP-dependent cotranslational protein targeting to membrane\n",
      "Passed: cat 1872, mi = 0.005, pvalue = 0.000, spliceosomal complex\n",
      "Passed: cat 693, mi = 0.004, pvalue = 0.000, mitochondrial translational elongation\n",
      "Passed: cat 1076, mi = 0.003, pvalue = 0.000, regulation of mRNA stability\n",
      "Passed: cat 2291, mi = 0.003, pvalue = 0.000, potassium ion transmembrane transport\n",
      "Passed: cat 104, mi = 0.003, pvalue = 0.000, hormone activity\n",
      "Passed: cat 7, mi = 0.003, pvalue = 0.000, respiratory chain\n",
      "Passed: cat 1256, mi = 0.002, pvalue = 0.000, tRNA processing\n",
      "Passed: cat 1034, mi = 0.002, pvalue = 0.000, unfolded protein binding\n",
      "Passed: cat 2759, mi = 0.002, pvalue = 0.000, cornification\n",
      "Passed: cat 1555, mi = 0.002, pvalue = 0.000, macroautophagy\n",
      "Passed: cat 2990, mi = 0.002, pvalue = 0.000, chromosome, centromeric region\n",
      "Passed: cat 4854, mi = 0.002, pvalue = 0.000, intrinsic apoptotic signaling pathway\n",
      "Passed: cat 57, mi = 0.002, pvalue = 0.000, mitochondrial ATP synthesis coupled proton transport\n",
      "Passed: cat 944, mi = 0.002, pvalue = 0.000, transcription elongation from RNA polymerase II promoter\n",
      "Passed: cat 3250, mi = 0.002, pvalue = 0.000, positive regulation of synapse assembly\n",
      "Passed: cat 2023, mi = 0.002, pvalue = 0.000, DNA replication\n",
      "Passed: cat 777, mi = 0.002, pvalue = 0.000, integral component of endoplasmic reticulum membrane\n",
      "Passed: cat 5627, mi = 0.002, pvalue = 0.000, synaptic vesicle exocytosis\n",
      "Passed: cat 1009, mi = 0.002, pvalue = 0.001, cellular response to retinoic acid\n",
      "Passed: cat 580, mi = 0.002, pvalue = 0.000, cargo loading into COPII-coated vesicle\n",
      "Passed: cat 1261, mi = 0.001, pvalue = 0.001, transcriptional activator activity, RNA polymerase II transcription regulatory region sequence-specific binding\n",
      "Passed: cat 835, mi = 0.001, pvalue = 0.001, translation initiation factor activity\n",
      "Passed: cat 5495, mi = 0.001, pvalue = 0.001, polyubiquitin binding\n",
      "Passed: cat 3809, mi = 0.001, pvalue = 0.001, ubiquitin conjugating enzyme activity\n",
      "Passed: cat 5144, mi = 0.001, pvalue = 0.000, oxygen binding\n",
      "Passed: cat 1204, mi = 0.001, pvalue = 0.004, anchored component of membrane\n",
      "Passed: cat 3521, mi = 0.001, pvalue = 0.003, actin filament organization\n",
      "Passed: cat 1963, mi = 0.001, pvalue = 0.003, Golgi cisterna membrane\n",
      "Passed: cat 1206, mi = 0.001, pvalue = 0.003, Ras guanyl-nucleotide exchange factor activity\n",
      "Passed: cat 4784, mi = 0.001, pvalue = 0.003, cell death\n",
      "Passed: cat 3801, mi = 0.001, pvalue = 0.001, transcription export complex\n",
      "Passed: cat 2446, mi = 0.001, pvalue = 0.003, histone H3-K4 methylation\n",
      "Passed: cat 1249, mi = 0.001, pvalue = 0.004, small-subunit processome\n",
      "Passed: cat 328, mi = 0.001, pvalue = 0.005, cell redox homeostasis\n",
      "Passed: cat 1363, mi = 0.001, pvalue = 0.002, chemokine-mediated signaling pathway\n",
      "Passed: cat 678, mi = 0.001, pvalue = 0.003, chemoattractant activity\n",
      "Passed: cat 2422, mi = 0.001, pvalue = 0.001, UDP-galactose:beta-N-acetylglucosamine beta-1,3-galactosyltransferase activity\n",
      "Passed: cat 6374, mi = 0.001, pvalue = 0.003, nucleobase-containing small molecule catabolic process\n",
      "Passed: cat 1409, mi = 0.001, pvalue = 0.004, ubiquitin-ubiquitin ligase activity\n",
      "Passed: cat 8202, mi = 0.001, pvalue = 0.004, establishment of epithelial cell polarity\n",
      "Didn't pass (cutoff=0.0050): cat 3833, mi = 0.001, pvalue = 0.005, JAK-STAT cascade\n",
      "Didn't pass (cutoff=0.0050): cat 6986, mi = 0.001, pvalue = 0.005, peroxisome organization\n",
      "Passed: cat 749, mi = 0.001, pvalue = 0.003, trophectodermal cell differentiation\n",
      "Didn't pass (cutoff=0.0050): cat 6078, mi = 0.001, pvalue = 0.006, canonical glycolysis\n",
      "Didn't pass (cutoff=0.0050): cat 4706, mi = 0.001, pvalue = 0.010, regulation of Rho protein signal transduction\n",
      "Didn't pass (cutoff=0.0050): cat 2703, mi = 0.001, pvalue = 0.008, iron-sulfur cluster assembly\n",
      "Passed: cat 6662, mi = 0.001, pvalue = 0.001, myoblast fusion\n",
      "Didn't pass (cutoff=0.0050): cat 841, mi = 0.001, pvalue = 0.010, thiol-dependent ubiquitin-specific protease activity\n",
      "Didn't pass (cutoff=0.0050): cat 2132, mi = 0.001, pvalue = 0.014, protein K48-linked deubiquitination\n",
      "Didn't pass (cutoff=0.0050): cat 1461, mi = 0.001, pvalue = 0.017, hemostasis\n",
      "Didn't pass (cutoff=0.0050): cat 839, mi = 0.001, pvalue = 0.013, thiol-dependent ubiquitinyl hydrolase activity\n",
      "Didn't pass (cutoff=0.0050): cat 4525, mi = 0.001, pvalue = 0.010, skeletal muscle contraction\n",
      "Didn't pass (cutoff=0.0050): cat 2122, mi = 0.001, pvalue = 0.015, protein transporter activity\n",
      "Didn't pass (cutoff=0.0050): cat 1864, mi = 0.001, pvalue = 0.019, ATP-dependent chromatin remodeling\n",
      "Passed: cat 2090, mi = 0.001, pvalue = 0.002, MICOS complex\n",
      "Passed: cat 2280, mi = 0.001, pvalue = 0.003, phosphatidylinositol-4,5-bisphosphate 5-phosphatase activity\n",
      "Didn't pass (cutoff=0.0050): cat 6519, mi = 0.001, pvalue = 0.010, DNA N-glycosylase activity\n",
      "Didn't pass (cutoff=0.0050): cat 1777, mi = 0.001, pvalue = 0.018, mitochondrial respiratory chain complex IV assembly\n",
      "Didn't pass (cutoff=0.0050): cat 3790, mi = 0.001, pvalue = 0.021, carboxypeptidase activity\n",
      "Didn't pass (cutoff=0.0050): cat 1470, mi = 0.001, pvalue = 0.021, regulation of protein stability\n",
      "Didn't pass (cutoff=0.0050): cat 5940, mi = 0.001, pvalue = 0.008, positive regulation of protein sumoylation\n",
      "Didn't pass (cutoff=0.0050): cat 620, mi = 0.001, pvalue = 0.029, phosphoric diester hydrolase activity\n",
      "Didn't pass (cutoff=0.0050): cat 4988, mi = 0.001, pvalue = 0.028, spindle assembly\n",
      "Didn't pass (cutoff=0.0050): cat 2249, mi = 0.001, pvalue = 0.028, ligand-dependent nuclear receptor binding\n",
      "Didn't pass (cutoff=0.0050): cat 586, mi = 0.001, pvalue = 0.021, metalloaminopeptidase activity\n",
      "Didn't pass (cutoff=0.0050): cat 3048, mi = 0.001, pvalue = 0.021, aminoacyl-tRNA synthetase multienzyme complex\n",
      "Didn't pass (cutoff=0.0050): cat 2207, mi = 0.001, pvalue = 0.030, cholesterol binding\n",
      "Didn't pass (cutoff=0.0050): cat 4519, mi = 0.001, pvalue = 0.015, extracellular ligand-gated ion channel activity\n",
      "Didn't pass (cutoff=0.0050): cat 7965, mi = 0.001, pvalue = 0.020, RNA methyltransferase activity\n",
      "Didn't pass (cutoff=0.0050): cat 9498, mi = 0.001, pvalue = 0.019, beta-catenin destruction complex\n",
      "Didn't pass (cutoff=0.0050): cat 5032, mi = 0.001, pvalue = 0.027, Golgi-associated vesicle membrane\n",
      "Didn't pass (cutoff=0.0050): cat 2133, mi = 0.001, pvalue = 0.034, proline-rich region binding\n",
      "Didn't pass (cutoff=0.0050): cat 3660, mi = 0.001, pvalue = 0.039, chloride channel activity\n",
      "Didn't pass (cutoff=0.0050): cat 2460, mi = 0.001, pvalue = 0.024, microvillus assembly\n",
      "Didn't pass (cutoff=0.0050): cat 2584, mi = 0.001, pvalue = 0.039, axon cytoplasm\n",
      "Didn't pass (cutoff=0.0050): cat 1413, mi = 0.001, pvalue = 0.029, phospholipase C-activating G-protein coupled receptor signaling pathway\n",
      "Total number of categories evaluated = 440.\n",
      "Number of categories that passed the tests = 45\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM2_5.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM2_5.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 9, 37, 0.0335562312036271\n",
      "best merge is 4, 5, 0.0343703008032655\n",
      "best merge is 18, 25, 0.035262737114545\n",
      "best merge is 36, 38, 0.082152450615631\n",
      "best merge is 4, 9, 0.0847672308025855\n",
      "best merge is 15, 29, 0.0940126161636187\n",
      "best merge is 1, 14, 0.0940774137920207\n",
      "best merge is 20, 35, 0.0964473675471328\n",
      "best merge is 7, 10, 0.101194288531179\n",
      "best merge is 31, 41, 0.10190950370795\n",
      "best merge is 20, 21, 0.115427536394873\n",
      "best merge is 24, 31, 0.127192963757413\n",
      "best merge is 7, 11, 0.128684560327247\n",
      "best merge is 2, 17, 0.137241953329755\n",
      "best merge is 8, 28, 0.13968966990604\n",
      "best merge is 2, 23, 0.145374209040404\n",
      "best merge is 4, 18, 0.147339443964953\n",
      "best merge is 6, 16, 0.153265066720333\n",
      "best merge is 40, 44, 0.169437167105213\n",
      "best merge is 1, 3, 0.176874988579549\n",
      "best merge is 4, 15, 0.178433244763834\n",
      "best merge is 1, 7, 0.199916335766136\n",
      "best merge is 20, 30, 0.220403420656214\n",
      "best merge is 12, 39, 0.235823329553961\n",
      "best merge is 4, 26, 0.244048045800234\n",
      "best merge is 6, 32, 0.261101550341653\n",
      "best merge is 13, 22, 0.275681698889556\n",
      "best merge is 33, 34, 0.308070539318402\n",
      "best merge is 4, 19, 0.309767567851787\n",
      "best merge is 8, 24, 0.315340455317391\n",
      "best merge is 4, 36, 0.335205732691748\n",
      "best merge is 20, 42, 0.352877721801041\n",
      "best merge is 6, 13, 0.36010076736281\n",
      "best merge is 1, 33, 0.361771857217326\n",
      "best merge is 1, 12, 0.439184901288117\n",
      "best merge is 20, 27, 0.469581080087647\n",
      "best merge is 0, 6, 0.476320169958183\n",
      "best merge is 1, 2, 0.54889078895284\n",
      "best merge is 1, 8, 0.618410353355873\n",
      "best merge is 0, 43, 0.795498319902653\n",
      "best merge is 0, 1, 0.806004596285424\n",
      "order classes\n",
      "7 - 2.69379321157703\n",
      "18 - 1.14193159319881\n",
      "12 - 0.801184629109352\n",
      "6 - 1.97830374581163\n",
      "2 - 1.90616812993364\n",
      "4 5 9 37 18 25 15 29 26 19 36 38\n",
      "1 14 3 7 10 11 33 34 12 39 2 17 23 8 28 24 31 41\n",
      "0 6 16 32 13 22 43\n",
      "20 35 21 30 42 27\n",
      "40 44\n",
      "Done.Start drawing\n",
      "272.13\t0.00\n",
      "cargo loading into COPII-coated vesicle, GO:0090110\n",
      "chemokine-mediated signaling pathway, GO:0070098\n",
      "transcriptional activator activity, RNA polymerase II transcription regulatory region sequence-specific binding, GO:0001228\n",
      "cell death, GO:0008219\n",
      "myoblast fusion, GO:0007520\n",
      "actin filament organization, GO:0007015\n",
      "unfolded protein binding, GO:0051082\n",
      "ubiquitin conjugating enzyme activity, GO:0061631\n",
      "transcription export complex, GO:0000346\n",
      "trophectodermal cell differentiation, GO:0001829\n",
      "regulation of mRNA stability, GO:0043488\n",
      "macroautophagy, GO:0016236\n",
      "chromosome, centromeric region, GO:0000775\n",
      "cell redox homeostasis, GO:0045454\n",
      "intrinsic apoptotic signaling pathway, GO:0097193\n",
      "ubiquitin-ubiquitin ligase activity, GO:0034450\n",
      "spliceosomal complex, GO:0005681\n",
      "small-subunit processome, GO:0032040\n",
      "mitochondrial translational elongation, GO:0070125\n",
      "integral component of endoplasmic reticulum membrane, GO:0030176\n",
      "polyubiquitin binding, GO:0031593\n",
      "translation initiation factor activity, GO:0003743\n",
      "transcription elongation from RNA polymerase II promoter, GO:0006368\n",
      "tRNA processing, GO:0008033\n",
      "Golgi cisterna membrane, GO:0032580\n",
      "MICOS complex, GO:0061617\n",
      "respiratory chain, GO:0070469\n",
      "histone H3-K4 methylation, GO:0051568\n",
      "mitochondrial ATP synthesis coupled proton transport, GO:0042776\n",
      "SRP-dependent cotranslational protein targeting to membrane, GO:0006614\n",
      "DNA replication, GO:0006260\n",
      "establishment of epithelial cell polarity, GO:0090162\n",
      "phosphatidylinositol-4,5-bisphosphate 5-phosphatase activity, GO:0004439\n",
      "potassium ion transmembrane transport, GO:0071805\n",
      "hormone activity, GO:0005179\n",
      "cornification, GO:0070268\n",
      "UDP-galactose:beta-N-acetylglucosamine beta-1,3-galactosyltransferase activity, GO:0008499\n",
      "synaptic vesicle exocytosis, GO:0016079\n",
      "oxygen binding, GO:0019825\n",
      "positive regulation of synapse assembly, GO:0051965\n",
      "Ras guanyl-nucleotide exchange factor activity, GO:0005088\n",
      "anchored component of membrane, GO:0031225\n",
      "cellular response to retinoic acid, GO:0071300\n",
      "chemoattractant activity, GO:0042056\n",
      "nucleobase-containing small molecule catabolic process, GO:0034656\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM2_5.csv_PAGE/filt_FC_LVM2_5.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM2_5.csv_PAGE/filt_FC_LVM2_5.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM2_5.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM2_5.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 9, 37, 0.0335562312036271\n",
      "best merge is 4, 5, 0.0343703008032655\n",
      "best merge is 18, 25, 0.035262737114545\n",
      "best merge is 36, 38, 0.082152450615631\n",
      "best merge is 4, 9, 0.0847672308025855\n",
      "best merge is 15, 29, 0.0940126161636187\n",
      "best merge is 1, 14, 0.0940774137920207\n",
      "best merge is 20, 35, 0.0964473675471328\n",
      "best merge is 7, 10, 0.101194288531179\n",
      "best merge is 31, 41, 0.10190950370795\n",
      "best merge is 20, 21, 0.115427536394873\n",
      "best merge is 24, 31, 0.127192963757413\n",
      "best merge is 7, 11, 0.128684560327247\n",
      "best merge is 2, 17, 0.137241953329755\n",
      "best merge is 8, 28, 0.13968966990604\n",
      "best merge is 2, 23, 0.145374209040404\n",
      "best merge is 4, 18, 0.147339443964953\n",
      "best merge is 6, 16, 0.153265066720333\n",
      "best merge is 40, 44, 0.169437167105213\n",
      "best merge is 1, 3, 0.176874988579549\n",
      "best merge is 4, 15, 0.178433244763834\n",
      "best merge is 1, 7, 0.199916335766136\n",
      "best merge is 20, 30, 0.220403420656214\n",
      "best merge is 12, 39, 0.235823329553961\n",
      "best merge is 4, 26, 0.244048045800234\n",
      "best merge is 6, 32, 0.261101550341653\n",
      "best merge is 13, 22, 0.275681698889556\n",
      "best merge is 33, 34, 0.308070539318402\n",
      "best merge is 4, 19, 0.309767567851787\n",
      "best merge is 8, 24, 0.315340455317391\n",
      "best merge is 4, 36, 0.335205732691748\n",
      "best merge is 20, 42, 0.352877721801041\n",
      "best merge is 6, 13, 0.36010076736281\n",
      "best merge is 1, 33, 0.361771857217326\n",
      "best merge is 1, 12, 0.439184901288117\n",
      "best merge is 20, 27, 0.469581080087647\n",
      "best merge is 0, 6, 0.476320169958183\n",
      "best merge is 1, 2, 0.54889078895284\n",
      "best merge is 1, 8, 0.618410353355873\n",
      "best merge is 0, 43, 0.795498319902653\n",
      "best merge is 0, 1, 0.806004596285424\n",
      "order classes\n",
      "7 - 2.69379321157703\n",
      "18 - 1.14193159319881\n",
      "12 - 0.801184629109352\n",
      "6 - 1.97830374581163\n",
      "2 - 1.90616812993364\n",
      "4 5 9 37 18 25 15 29 26 19 36 38\n",
      "1 14 3 7 10 11 33 34 12 39 2 17 23 8 28 24 31 41\n",
      "0 6 16 32 13 22 43\n",
      "20 35 21 30 42 27\n",
      "40 44\n",
      "Done.-19.9645022599493\t42\t0\n",
      "-19.9104839598721\t42\t0.0709954801014221\n",
      "-19.8803880498291\t42\t0.108036600154338\n",
      "-19.8580090397972\t42\t0.13118730018741\n",
      "-19.8379450997685\t42\t0.152794620218278\n",
      "-19.8178811597398\t42\t0.171315180244736\n",
      "-19.7939587697057\t42\t0.192922500275604\n",
      "-19.7600044096571\t42\t0.219159960313086\n",
      "1.13041561018631\t42\t0.260831220372616\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM2_5.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM2_2.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl/human_ensembl.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM2_2.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 13290... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 17689\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 13290 out of 13290 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 13290\t17689\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1477 hits go_terms_num = 17689\n",
      "cat 2546, mi = 0.00885161, SRP-dependent cotranslational protein targeting to membrane\n",
      "cat 836, mi = 0.00813001, translational initiation\n",
      "cat 692, mi = 0.00812601, structural constituent of ribosome\n",
      "cat 1498, mi = 0.00646110, viral transcription\n",
      "cat 2545, mi = 0.00619239, nuclear-transcribed mRNA catabolic process, nonsense-mediated decay\n",
      "cat 2016, mi = 0.00580601, cytosolic large ribosomal subunit\n",
      "cat 1872, mi = 0.00422527, spliceosomal complex\n",
      "cat 7, mi = 0.00413582, respiratory chain\n",
      "cat 1491, mi = 0.00362303, mRNA export from nucleus\n",
      "cat 2431, mi = 0.00332274, nucleosome\n",
      "cat 11, mi = 0.00325671, mitochondrial respiratory chain complex I assembly\n",
      "cat 2023, mi = 0.00301397, DNA replication\n",
      "cat 10, mi = 0.00299887, mitochondrial electron transport, NADH to ubiquinone\n",
      "cat 9, mi = 0.00298962, NADH dehydrogenase (ubiquinone) activity\n",
      "cat 12, mi = 0.00291564, mitochondrial respiratory chain complex I\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 2546, mi = 0.009, pvalue = 0.000, SRP-dependent cotranslational protein targeting to membrane\n",
      "Passed: cat 1872, mi = 0.004, pvalue = 0.000, spliceosomal complex\n",
      "Passed: cat 7, mi = 0.004, pvalue = 0.000, respiratory chain\n",
      "Passed: cat 2431, mi = 0.003, pvalue = 0.000, nucleosome\n",
      "Passed: cat 2023, mi = 0.003, pvalue = 0.000, DNA replication\n",
      "Passed: cat 2759, mi = 0.003, pvalue = 0.000, cornification\n",
      "Passed: cat 1064, mi = 0.003, pvalue = 0.000, proteasome complex\n",
      "Passed: cat 2990, mi = 0.003, pvalue = 0.000, chromosome, centromeric region\n",
      "Passed: cat 693, mi = 0.003, pvalue = 0.000, mitochondrial translational elongation\n",
      "Passed: cat 313, mi = 0.002, pvalue = 0.000, response to lipopolysaccharide\n",
      "Passed: cat 556, mi = 0.002, pvalue = 0.000, secretory granule\n",
      "Passed: cat 3047, mi = 0.002, pvalue = 0.000, tRNA binding\n",
      "Passed: cat 903, mi = 0.002, pvalue = 0.000, late endosome membrane\n",
      "Passed: cat 1034, mi = 0.002, pvalue = 0.000, unfolded protein binding\n",
      "Passed: cat 1976, mi = 0.002, pvalue = 0.000, PcG protein complex\n",
      "Passed: cat 3348, mi = 0.002, pvalue = 0.000, global genome nucleotide-excision repair\n",
      "Passed: cat 1238, mi = 0.002, pvalue = 0.000, bicellular tight junction\n",
      "Passed: cat 682, mi = 0.002, pvalue = 0.000, regulation of translation\n",
      "Passed: cat 2291, mi = 0.002, pvalue = 0.001, potassium ion transmembrane transport\n",
      "Passed: cat 328, mi = 0.002, pvalue = 0.001, cell redox homeostasis\n",
      "Passed: cat 56, mi = 0.002, pvalue = 0.001, cristae formation\n",
      "Passed: cat 1445, mi = 0.002, pvalue = 0.001, thymus development\n",
      "Passed: cat 7241, mi = 0.002, pvalue = 0.000, N-acetylglucosamine metabolic process\n",
      "Passed: cat 1947, mi = 0.002, pvalue = 0.000, Golgi to endosome transport\n",
      "Passed: cat 2448, mi = 0.001, pvalue = 0.001, histone methyltransferase activity (H3-K4 specific)\n",
      "Passed: cat 752, mi = 0.001, pvalue = 0.002, stem cell population maintenance\n",
      "Passed: cat 2368, mi = 0.001, pvalue = 0.001, ephrin receptor signaling pathway\n",
      "Passed: cat 439, mi = 0.001, pvalue = 0.002, histone acetyltransferase binding\n",
      "Passed: cat 847, mi = 0.001, pvalue = 0.002, digestion\n",
      "Passed: cat 3248, mi = 0.001, pvalue = 0.003, synaptic vesicle endocytosis\n",
      "Passed: cat 1280, mi = 0.001, pvalue = 0.004, protein K48-linked ubiquitination\n",
      "Passed: cat 1715, mi = 0.001, pvalue = 0.005, DNA-directed 5'-3' RNA polymerase activity\n",
      "Didn't pass (cutoff=0.0050): cat 330, mi = 0.001, pvalue = 0.005, positive regulation of neuron differentiation\n",
      "Passed: cat 7012, mi = 0.001, pvalue = 0.001, positive regulation of blood vessel diameter\n",
      "Didn't pass (cutoff=0.0050): cat 110, mi = 0.001, pvalue = 0.006, regulation of MAPK cascade\n",
      "Didn't pass (cutoff=0.0050): cat 2919, mi = 0.001, pvalue = 0.007, histone demethylase activity\n",
      "Passed: cat 3783, mi = 0.001, pvalue = 0.003, fibroblast growth factor receptor binding\n",
      "Didn't pass (cutoff=0.0050): cat 2835, mi = 0.001, pvalue = 0.005, adenylate cyclase-modulating G-protein coupled receptor signaling pathway\n",
      "Didn't pass (cutoff=0.0050): cat 5323, mi = 0.001, pvalue = 0.009, cortical actin cytoskeleton\n",
      "Didn't pass (cutoff=0.0050): cat 1864, mi = 0.001, pvalue = 0.009, ATP-dependent chromatin remodeling\n",
      "Didn't pass (cutoff=0.0050): cat 2130, mi = 0.001, pvalue = 0.011, protein K63-linked deubiquitination\n",
      "Didn't pass (cutoff=0.0050): cat 6105, mi = 0.001, pvalue = 0.007, NADPH binding\n",
      "Didn't pass (cutoff=0.0050): cat 269, mi = 0.001, pvalue = 0.006, acute-phase response\n",
      "Didn't pass (cutoff=0.0050): cat 3745, mi = 0.001, pvalue = 0.009, positive regulation of peptidyl-threonine phosphorylation\n",
      "Passed: cat 6113, mi = 0.001, pvalue = 0.001, detection of chemical stimulus involved in sensory perception of bitter taste\n",
      "Passed: cat 4468, mi = 0.001, pvalue = 0.001, cellular component organization\n",
      "Passed: cat 4855, mi = 0.001, pvalue = 0.004, peptidyl-threonine dephosphorylation\n",
      "Didn't pass (cutoff=0.0050): cat 837, mi = 0.001, pvalue = 0.015, eukaryotic translation initiation factor 3 complex\n",
      "Didn't pass (cutoff=0.0050): cat 2316, mi = 0.001, pvalue = 0.015, flagellated sperm motility\n",
      "Didn't pass (cutoff=0.0050): cat 4090, mi = 0.001, pvalue = 0.018, protein destabilization\n",
      "Didn't pass (cutoff=0.0050): cat 3450, mi = 0.001, pvalue = 0.012, signal peptide processing\n",
      "Didn't pass (cutoff=0.0050): cat 1777, mi = 0.001, pvalue = 0.012, mitochondrial respiratory chain complex IV assembly\n",
      "Didn't pass (cutoff=0.0050): cat 698, mi = 0.001, pvalue = 0.017, tricarboxylic acid cycle\n",
      "Didn't pass (cutoff=0.0050): cat 3188, mi = 0.001, pvalue = 0.020, branched-chain amino acid catabolic process\n",
      "Didn't pass (cutoff=0.0050): cat 4112, mi = 0.001, pvalue = 0.018, oxidoreductase activity, acting on the CH-CH group of donors\n",
      "Didn't pass (cutoff=0.0050): cat 1678, mi = 0.001, pvalue = 0.018, regulation of TOR signaling\n",
      "Didn't pass (cutoff=0.0050): cat 2101, mi = 0.001, pvalue = 0.012, centriole\n",
      "Didn't pass (cutoff=0.0050): cat 2869, mi = 0.001, pvalue = 0.018, nucleoside metabolic process\n",
      "Didn't pass (cutoff=0.0050): cat 841, mi = 0.001, pvalue = 0.016, thiol-dependent ubiquitin-specific protease activity\n",
      "Didn't pass (cutoff=0.0050): cat 1847, mi = 0.001, pvalue = 0.021, protein homotetramerization\n",
      "Didn't pass (cutoff=0.0050): cat 6014, mi = 0.001, pvalue = 0.025, pericentriolar material\n",
      "Didn't pass (cutoff=0.0050): cat 408, mi = 0.001, pvalue = 0.018, protein glycosylation\n",
      "Didn't pass (cutoff=0.0050): cat 2839, mi = 0.001, pvalue = 0.006, regulation of rhodopsin mediated signaling pathway\n",
      "Didn't pass (cutoff=0.0050): cat 3423, mi = 0.001, pvalue = 0.005, telomere maintenance via recombination\n",
      "Didn't pass (cutoff=0.0050): cat 7064, mi = 0.001, pvalue = 0.006, prenyltransferase activity\n",
      "Didn't pass (cutoff=0.0050): cat 4132, mi = 0.001, pvalue = 0.017, biotin metabolic process\n",
      "Didn't pass (cutoff=0.0050): cat 413, mi = 0.001, pvalue = 0.028, cis-Golgi network\n",
      "Total number of categories evaluated = 325.\n",
      "Number of categories that passed the tests = 37\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM2_2.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM2_2.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 28, 34, 0.0120193062150217\n",
      "best merge is 5, 28, 0.0194923194653187\n",
      "best merge is 5, 33, 0.0577980993033501\n",
      "best merge is 19, 25, 0.072499389715498\n",
      "best merge is 23, 24, 0.0830764397208315\n",
      "best merge is 6, 8, 0.102297104814218\n",
      "best merge is 22, 35, 0.118248760496892\n",
      "best merge is 7, 17, 0.127847949464887\n",
      "best merge is 2, 20, 0.141107716699597\n",
      "best merge is 22, 36, 0.188331919533933\n",
      "best merge is 0, 2, 0.19486298825993\n",
      "best merge is 1, 7, 0.199380408240193\n",
      "best merge is 6, 13, 0.206726570532072\n",
      "best merge is 10, 32, 0.211552493889282\n",
      "best merge is 6, 11, 0.243510888858156\n",
      "best merge is 5, 18, 0.244715139356179\n",
      "best merge is 4, 22, 0.269237694595906\n",
      "best merge is 26, 31, 0.292389908176471\n",
      "best merge is 12, 19, 0.297369092939432\n",
      "best merge is 0, 6, 0.303443505931295\n",
      "best merge is 14, 30, 0.308815764235999\n",
      "best merge is 12, 15, 0.361656150651123\n",
      "best merge is 4, 21, 0.385120914399955\n",
      "best merge is 5, 9, 0.397040837327347\n",
      "best merge is 3, 27, 0.401266977202325\n",
      "best merge is 1, 12, 0.425281056827883\n",
      "best merge is 4, 23, 0.539389387921116\n",
      "best merge is 5, 16, 0.54511188665337\n",
      "best merge is 0, 1, 0.575772525729755\n",
      "best merge is 0, 26, 0.621745912188222\n",
      "best merge is 3, 10, 0.643964820458685\n",
      "best merge is 0, 14, 0.791972417201288\n",
      "best merge is 5, 29, 0.813101221979554\n",
      "order classes\n",
      "18 - 1.07049639014226\n",
      "4 - 3.68690980807109\n",
      "7 - 1.79500260883801\n",
      "7 - 1.39351585632876\n",
      "0 2 20 6 8 13 11 1 7 17 12 19 25 15 26 31 14 30\n",
      "3 27 10 32\n",
      "4 22 35 36 21 23 24\n",
      "5 28 34 33 18 9 16\n",
      "29\n",
      "Done.Start drawing\n",
      "101.03\t0.00\n",
      "nucleosome, GO:0000786\n",
      "histone acetyltransferase binding, GO:0035035\n",
      "positive regulation of blood vessel diameter, GO:0097755\n",
      "DNA replication, GO:0006260\n",
      "N-acetylglucosamine metabolic process, GO:0006044\n",
      "cellular component organization, GO:0016043\n",
      "peptidyl-threonine dephosphorylation, GO:0035970\n",
      "thymus development, GO:0048538\n",
      "PcG protein complex, GO:0031519\n",
      "protein K48-linked ubiquitination, GO:0070936\n",
      "chromosome, centromeric region, GO:0000775\n",
      "global genome nucleotide-excision repair, GO:0070911\n",
      "Golgi to endosome transport, GO:0006895\n",
      "histone methyltransferase activity (H3-K4 specific), GO:0042800\n",
      "regulation of translation, GO:0006417\n",
      "late endosome membrane, GO:0031902\n",
      "stem cell population maintenance, GO:0019827\n",
      "ephrin receptor signaling pathway, GO:0048013\n",
      "spliceosomal complex, GO:0005681\n",
      "DNA-directed 5'-3' RNA polymerase activity, GO:0003899\n",
      "SRP-dependent cotranslational protein targeting to membrane, GO:0006614\n",
      "respiratory chain, GO:0070469\n",
      "cristae formation, GO:0042407\n",
      "proteasome complex, GO:0000502\n",
      "mitochondrial translational elongation, GO:0070125\n",
      "unfolded protein binding, GO:0051082\n",
      "synaptic vesicle endocytosis, GO:0048488\n",
      "tRNA binding, GO:0000049\n",
      "cell redox homeostasis, GO:0045454\n",
      "secretory granule, GO:0030141\n",
      "cornification, GO:0070268\n",
      "digestion, GO:0007586\n",
      "detection of chemical stimulus involved in sensory perception of bitter taste, GO:0001580\n",
      "fibroblast growth factor receptor binding, GO:0005104\n",
      "potassium ion transmembrane transport, GO:0071805\n",
      "response to lipopolysaccharide, GO:0032496\n",
      "bicellular tight junction, GO:0005923\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM2_2.csv_PAGE/filt_FC_LVM2_2.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM2_2.csv_PAGE/filt_FC_LVM2_2.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM2_2.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM2_2.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 28, 34, 0.0120193062150217\n",
      "best merge is 5, 28, 0.0194923194653187\n",
      "best merge is 5, 33, 0.0577980993033501\n",
      "best merge is 19, 25, 0.072499389715498\n",
      "best merge is 23, 24, 0.0830764397208315\n",
      "best merge is 6, 8, 0.102297104814218\n",
      "best merge is 22, 35, 0.118248760496892\n",
      "best merge is 7, 17, 0.127847949464887\n",
      "best merge is 2, 20, 0.141107716699597\n",
      "best merge is 22, 36, 0.188331919533933\n",
      "best merge is 0, 2, 0.19486298825993\n",
      "best merge is 1, 7, 0.199380408240193\n",
      "best merge is 6, 13, 0.206726570532072\n",
      "best merge is 10, 32, 0.211552493889282\n",
      "best merge is 6, 11, 0.243510888858156\n",
      "best merge is 5, 18, 0.244715139356179\n",
      "best merge is 4, 22, 0.269237694595906\n",
      "best merge is 26, 31, 0.292389908176471\n",
      "best merge is 12, 19, 0.297369092939432\n",
      "best merge is 0, 6, 0.303443505931295\n",
      "best merge is 14, 30, 0.308815764235999\n",
      "best merge is 12, 15, 0.361656150651123\n",
      "best merge is 4, 21, 0.385120914399955\n",
      "best merge is 5, 9, 0.397040837327347\n",
      "best merge is 3, 27, 0.401266977202325\n",
      "best merge is 1, 12, 0.425281056827883\n",
      "best merge is 4, 23, 0.539389387921116\n",
      "best merge is 5, 16, 0.54511188665337\n",
      "best merge is 0, 1, 0.575772525729755\n",
      "best merge is 0, 26, 0.621745912188222\n",
      "best merge is 3, 10, 0.643964820458685\n",
      "best merge is 0, 14, 0.791972417201288\n",
      "best merge is 5, 29, 0.813101221979554\n",
      "order classes\n",
      "18 - 1.07049639014226\n",
      "4 - 3.68690980807109\n",
      "7 - 1.79500260883801\n",
      "7 - 1.39351585632876\n",
      "0 2 20 6 8 13 11 1 7 17 12 19 25 15 26 31 14 30\n",
      "3 27 10 32\n",
      "4 22 35 36 21 23 24\n",
      "5 28 34 33 18 9 16\n",
      "29\n",
      "Done.-19.8586558447986\t42\t0\n",
      "-19.6674255171731\t42\t0.282688310402851\n",
      "-19.5863604869841\t42\t0.382460655250916\n",
      "-19.5281599524894\t42\t0.444818370780956\n",
      "-19.4761951895477\t42\t0.498861724240325\n",
      "-19.4221518360883\t42\t0.548747896664357\n",
      "-19.3535583490052\t42\t0.606948431159062\n",
      "-19.2433930515688\t42\t0.685934870830446\n",
      "1.41363951301594\t42\t0.827279026031872\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM2_2.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_P_LVM1_2.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl/human_ensembl.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_P_LVM1_2.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 13290... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 17689\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 13290 out of 13290 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 13290\t17689\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1477 hits go_terms_num = 17689\n",
      "cat 2546, mi = 0.01290346, SRP-dependent cotranslational protein targeting to membrane\n",
      "cat 692, mi = 0.01108132, structural constituent of ribosome\n",
      "cat 836, mi = 0.01061314, translational initiation\n",
      "cat 2545, mi = 0.01045079, nuclear-transcribed mRNA catabolic process, nonsense-mediated decay\n",
      "cat 1498, mi = 0.00972424, viral transcription\n",
      "cat 2121, mi = 0.00685957, sister chromatid cohesion\n",
      "cat 2990, mi = 0.00663311, chromosome, centromeric region\n",
      "cat 2016, mi = 0.00581688, cytosolic large ribosomal subunit\n",
      "cat 2585, mi = 0.00543512, kinetochore\n",
      "cat 4152, mi = 0.00484561, cytosolic small ribosomal subunit\n",
      "cat 4583, mi = 0.00472238, condensed chromosome kinetochore\n",
      "cat 1308, mi = 0.00462187, spindle pole\n",
      "cat 7, mi = 0.00460825, respiratory chain\n",
      "cat 394, mi = 0.00444013, chromosome segregation\n",
      "cat 12, mi = 0.00395704, mitochondrial respiratory chain complex I\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 2546, mi = 0.013, pvalue = 0.000, SRP-dependent cotranslational protein targeting to membrane\n",
      "Passed: cat 2121, mi = 0.007, pvalue = 0.000, sister chromatid cohesion\n",
      "Passed: cat 7, mi = 0.005, pvalue = 0.000, respiratory chain\n",
      "Passed: cat 2431, mi = 0.004, pvalue = 0.000, nucleosome\n",
      "Passed: cat 1872, mi = 0.003, pvalue = 0.000, spliceosomal complex\n",
      "Passed: cat 694, mi = 0.003, pvalue = 0.000, mitochondrial translational termination\n",
      "Passed: cat 2101, mi = 0.003, pvalue = 0.000, centriole\n",
      "Passed: cat 2013, mi = 0.003, pvalue = 0.000, endosomal transport\n",
      "Passed: cat 890, mi = 0.002, pvalue = 0.000, melanosome\n",
      "Passed: cat 1064, mi = 0.002, pvalue = 0.000, proteasome complex\n",
      "Passed: cat 176, mi = 0.002, pvalue = 0.000, transcription factor complex\n",
      "Passed: cat 3180, mi = 0.002, pvalue = 0.000, lyase activity\n",
      "Passed: cat 57, mi = 0.002, pvalue = 0.000, mitochondrial ATP synthesis coupled proton transport\n",
      "Passed: cat 2295, mi = 0.002, pvalue = 0.000, potassium channel activity\n",
      "Passed: cat 6415, mi = 0.002, pvalue = 0.000, chromosome condensation\n",
      "Passed: cat 1256, mi = 0.002, pvalue = 0.000, tRNA processing\n",
      "Passed: cat 6876, mi = 0.002, pvalue = 0.000, MCM complex\n",
      "Passed: cat 5305, mi = 0.002, pvalue = 0.000, positive regulation of viral transcription\n",
      "Passed: cat 298, mi = 0.002, pvalue = 0.001, regulation of cell shape\n",
      "Passed: cat 4784, mi = 0.002, pvalue = 0.000, cell death\n",
      "Passed: cat 835, mi = 0.002, pvalue = 0.001, translation initiation factor activity\n",
      "Passed: cat 5637, mi = 0.002, pvalue = 0.000, mitotic nuclear envelope reassembly\n",
      "Passed: cat 817, mi = 0.002, pvalue = 0.001, autophagy\n",
      "Passed: cat 5083, mi = 0.002, pvalue = 0.000, NuA4 histone acetyltransferase complex\n",
      "Passed: cat 1204, mi = 0.001, pvalue = 0.002, anchored component of membrane\n",
      "Passed: cat 1716, mi = 0.001, pvalue = 0.000, cellular oxidant detoxification\n",
      "Passed: cat 84, mi = 0.001, pvalue = 0.001, animal organ regeneration\n",
      "Passed: cat 2638, mi = 0.001, pvalue = 0.002, heterochromatin\n",
      "Passed: cat 2371, mi = 0.001, pvalue = 0.002, positive regulation of cytokinesis\n",
      "Passed: cat 1160, mi = 0.001, pvalue = 0.003, Rab guanyl-nucleotide exchange factor activity\n",
      "Passed: cat 3920, mi = 0.001, pvalue = 0.001, SUMO transferase activity\n",
      "Passed: cat 2448, mi = 0.001, pvalue = 0.002, histone methyltransferase activity (H3-K4 specific)\n",
      "Didn't pass (cutoff=0.0050): cat 2359, mi = 0.001, pvalue = 0.005, negative regulation of phosphatase activity\n",
      "Passed: cat 3533, mi = 0.001, pvalue = 0.001, cilium movement\n",
      "Passed: cat 6976, mi = 0.001, pvalue = 0.004, preribosome, large subunit precursor\n",
      "Didn't pass (cutoff=0.0050): cat 3195, mi = 0.001, pvalue = 0.005, translation elongation factor activity\n",
      "Didn't pass (cutoff=0.0050): cat 2797, mi = 0.001, pvalue = 0.006, chloride transmembrane transport\n",
      "Didn't pass (cutoff=0.0050): cat 1628, mi = 0.001, pvalue = 0.006, nucleotide metabolic process\n",
      "Passed: cat 1304, mi = 0.001, pvalue = 0.005, phosphatidylinositol biosynthetic process\n",
      "Passed: cat 3698, mi = 0.001, pvalue = 0.002, NAD metabolic process\n",
      "Passed: cat 412, mi = 0.001, pvalue = 0.002, heparan sulfate proteoglycan biosynthetic process\n",
      "Passed: cat 3951, mi = 0.001, pvalue = 0.003, pentose-phosphate shunt\n",
      "Passed: cat 1685, mi = 0.001, pvalue = 0.001, contractile fiber\n",
      "Didn't pass (cutoff=0.0050): cat 104, mi = 0.001, pvalue = 0.008, hormone activity\n",
      "Didn't pass (cutoff=0.0050): cat 91, mi = 0.001, pvalue = 0.008, integral component of mitochondrial inner membrane\n",
      "Didn't pass (cutoff=0.0050): cat 5423, mi = 0.001, pvalue = 0.005, COPI-coated vesicle membrane\n",
      "Didn't pass (cutoff=0.0050): cat 5849, mi = 0.001, pvalue = 0.010, positive regulation of ATPase activity\n",
      "Didn't pass (cutoff=0.0050): cat 9427, mi = 0.001, pvalue = 0.006, magnesium ion transmembrane transporter activity\n",
      "Didn't pass (cutoff=0.0050): cat 3326, mi = 0.001, pvalue = 0.006, positive regulation of mRNA catabolic process\n",
      "Passed: cat 3470, mi = 0.001, pvalue = 0.005, positive regulation of nuclear-transcribed mRNA catabolic process, deadenylation-dependent decay\n",
      "Passed: cat 3765, mi = 0.001, pvalue = 0.005, positive regulation of release of sequestered calcium ion into cytosol\n",
      "Didn't pass (cutoff=0.0050): cat 1017, mi = 0.001, pvalue = 0.011, positive regulation of TOR signaling\n",
      "Passed: cat 6969, mi = 0.001, pvalue = 0.004, 3'-phosphoadenosine 5'-phosphosulfate metabolic process\n",
      "Didn't pass (cutoff=0.0050): cat 2798, mi = 0.001, pvalue = 0.015, chloride transport\n",
      "Didn't pass (cutoff=0.0050): cat 3660, mi = 0.001, pvalue = 0.014, chloride channel activity\n",
      "Didn't pass (cutoff=0.0050): cat 3950, mi = 0.001, pvalue = 0.016, response to testosterone\n",
      "Passed: cat 749, mi = 0.001, pvalue = 0.005, trophectodermal cell differentiation\n",
      "Didn't pass (cutoff=0.0050): cat 5107, mi = 0.001, pvalue = 0.009, GTPase inhibitor activity\n",
      "Didn't pass (cutoff=0.0050): cat 8112, mi = 0.001, pvalue = 0.013, motor neuron axon guidance\n",
      "Didn't pass (cutoff=0.0050): cat 9657, mi = 0.001, pvalue = 0.013, magnesium ion transmembrane transport\n",
      "Didn't pass (cutoff=0.0050): cat 3787, mi = 0.001, pvalue = 0.023, embryonic skeletal system development\n",
      "Didn't pass (cutoff=0.0050): cat 6079, mi = 0.001, pvalue = 0.014, positive regulation of glycogen biosynthetic process\n",
      "Didn't pass (cutoff=0.0050): cat 279, mi = 0.001, pvalue = 0.017, defense response to virus\n",
      "Didn't pass (cutoff=0.0050): cat 1250, mi = 0.001, pvalue = 0.030, snoRNA binding\n",
      "Didn't pass (cutoff=0.0050): cat 1206, mi = 0.001, pvalue = 0.023, Ras guanyl-nucleotide exchange factor activity\n",
      "Didn't pass (cutoff=0.0050): cat 7651, mi = 0.001, pvalue = 0.014, positive regulation of dendrite development\n",
      "Didn't pass (cutoff=0.0050): cat 6637, mi = 0.001, pvalue = 0.020, regulation of mRNA splicing, via spliceosome\n",
      "Didn't pass (cutoff=0.0050): cat 3661, mi = 0.001, pvalue = 0.029, chloride channel complex\n",
      "Didn't pass (cutoff=0.0050): cat 3297, mi = 0.001, pvalue = 0.031, syntaxin-1 binding\n",
      "Didn't pass (cutoff=0.0050): cat 765, mi = 0.001, pvalue = 0.026, regulation of calcium ion-dependent exocytosis\n",
      "Didn't pass (cutoff=0.0050): cat 3659, mi = 0.001, pvalue = 0.023, regulation of anion transmembrane transport\n",
      "Didn't pass (cutoff=0.0050): cat 480, mi = 0.001, pvalue = 0.010, negative thymic T cell selection\n",
      "Didn't pass (cutoff=0.0050): cat 5225, mi = 0.001, pvalue = 0.010, macrophage activation\n",
      "Didn't pass (cutoff=0.0050): cat 4537, mi = 0.001, pvalue = 0.037, negative regulation of Notch signaling pathway\n",
      "Didn't pass (cutoff=0.0050): cat 2538, mi = 0.001, pvalue = 0.029, negative regulation of protein binding\n",
      "Didn't pass (cutoff=0.0050): cat 2345, mi = 0.001, pvalue = 0.031, isoprenoid biosynthetic process\n",
      "Didn't pass (cutoff=0.0050): cat 2622, mi = 0.001, pvalue = 0.032, positive regulation of dendritic spine morphogenesis\n",
      "Total number of categories evaluated = 459.\n",
      "Number of categories that passed the tests = 43\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_P_LVM1_2.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_P_LVM1_2.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 3, 14, 0.0116347442111547\n",
      "best merge is 16, 24, 0.012278943258604\n",
      "best merge is 1, 3, 0.0371144017301236\n",
      "best merge is 1, 6, 0.0487746070481396\n",
      "best merge is 0, 37, 0.0665256147729932\n",
      "best merge is 1, 28, 0.078141962737292\n",
      "best merge is 5, 15, 0.0862836797122402\n",
      "best merge is 16, 32, 0.0932818358751525\n",
      "best merge is 33, 35, 0.0965027270830539\n",
      "best merge is 19, 30, 0.106555355338929\n",
      "best merge is 18, 23, 0.11204205149186\n",
      "best merge is 5, 9, 0.131884485601158\n",
      "best merge is 13, 16, 0.133031652764161\n",
      "best merge is 18, 19, 0.141088871040473\n",
      "best merge is 2, 12, 0.150270411475183\n",
      "best merge is 4, 7, 0.157741344023316\n",
      "best merge is 5, 8, 0.167963954320121\n",
      "best merge is 33, 40, 0.168898200596302\n",
      "best merge is 11, 20, 0.209412853487564\n",
      "best merge is 33, 41, 0.237887743484811\n",
      "best merge is 4, 17, 0.253163313919973\n",
      "best merge is 10, 29, 0.260218387887959\n",
      "best merge is 39, 42, 0.274364564819742\n",
      "best merge is 36, 38, 0.283366332965327\n",
      "best merge is 2, 5, 0.297286853706197\n",
      "best merge is 11, 25, 0.305622546233865\n",
      "best merge is 18, 21, 0.307117165237916\n",
      "best merge is 31, 39, 0.352309170766005\n",
      "best merge is 10, 27, 0.357618702360194\n",
      "best merge is 26, 34, 0.368104814653853\n",
      "best merge is 4, 22, 0.409788954248417\n",
      "best merge is 0, 2, 0.462811904136997\n",
      "best merge is 4, 18, 0.533930905633316\n",
      "best merge is 0, 11, 0.539108562016371\n",
      "best merge is 4, 26, 0.659006501150251\n",
      "best merge is 31, 36, 0.687442629197174\n",
      "best merge is 4, 10, 0.779345208458652\n",
      "best merge is 0, 33, 0.918960853674379\n",
      "best merge is 13, 31, 0.974523160723452\n",
      "order classes\n",
      "15 - 1.63958761937138\n",
      "5 - 0.374791591198312\n",
      "14 - 1.67698570865293\n",
      "4 - 0.803478614123747\n",
      "5 - 3.70625944997609\n",
      "1 3 14 6 28\n",
      "0 37 2 12 5 15 9 8 11 20 25 33 35 40 41\n",
      "4 7 17 22 18 23 19 30 21 26 34 10 29 27\n",
      "13 16 24 32\n",
      "31 39 42 36 38\n",
      "Done.Start drawing\n",
      "9.27\t0.00\n",
      "sister chromatid cohesion, GO:0007062\n",
      "nucleosome, GO:0000786\n",
      "chromosome condensation, GO:0030261\n",
      "centriole, GO:0005814\n",
      "positive regulation of cytokinesis, GO:0032467\n",
      "histone methyltransferase activity (H3-K4 specific), GO:0042800\n",
      "positive regulation of nuclear-transcribed mRNA catabolic process, deadenylation-dependent decay, GO:1900153\n",
      "trophectodermal cell differentiation, GO:0001829\n",
      "heparan sulfate proteoglycan biosynthetic process, GO:0015012\n",
      "autophagy, GO:0006914\n",
      "regulation of cell shape, GO:0008360\n",
      "NuA4 histone acetyltransferase complex, GO:0035267\n",
      "cell death, GO:0008219\n",
      "SUMO transferase activity, GO:0019789\n",
      "mitotic nuclear envelope reassembly, GO:0007084\n",
      "melanosome, GO:0042470\n",
      "spliceosomal complex, GO:0005681\n",
      "endosomal transport, GO:0016197\n",
      "positive regulation of viral transcription, GO:0050434\n",
      "respiratory chain, GO:0070469\n",
      "phosphatidylinositol biosynthetic process, GO:0006661\n",
      "transcription factor complex, GO:0005667\n",
      "Rab guanyl-nucleotide exchange factor activity, GO:0017112\n",
      "heterochromatin, GO:0000792\n",
      "SRP-dependent cotranslational protein targeting to membrane, GO:0006614\n",
      "pentose-phosphate shunt, GO:0006098\n",
      "mitochondrial translational termination, GO:0070126\n",
      "tRNA processing, GO:0008033\n",
      "proteasome complex, GO:0000502\n",
      "mitochondrial ATP synthesis coupled proton transport, GO:0042776\n",
      "lyase activity, GO:0016829\n",
      "translation initiation factor activity, GO:0003743\n",
      "cellular oxidant detoxification, GO:0098869\n",
      "animal organ regeneration, GO:0031100\n",
      "preribosome, large subunit precursor, GO:0030687\n",
      "NAD metabolic process, GO:0019674\n",
      "positive regulation of release of sequestered calcium ion into cytosol, GO:0051281\n",
      "3'-phosphoadenosine 5'-phosphosulfate metabolic process, GO:0050427\n",
      "potassium channel activity, GO:0005267\n",
      "MCM complex, GO:0042555\n",
      "anchored component of membrane, GO:0031225\n",
      "cilium movement, GO:0003341\n",
      "contractile fiber, GO:0043292\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_P_LVM1_2.csv_PAGE/filt_FC_P_LVM1_2.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_P_LVM1_2.csv_PAGE/filt_FC_P_LVM1_2.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_P_LVM1_2.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_P_LVM1_2.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 3, 14, 0.0116347442111547\n",
      "best merge is 16, 24, 0.012278943258604\n",
      "best merge is 1, 3, 0.0371144017301236\n",
      "best merge is 1, 6, 0.0487746070481396\n",
      "best merge is 0, 37, 0.0665256147729932\n",
      "best merge is 1, 28, 0.078141962737292\n",
      "best merge is 5, 15, 0.0862836797122402\n",
      "best merge is 16, 32, 0.0932818358751525\n",
      "best merge is 33, 35, 0.0965027270830539\n",
      "best merge is 19, 30, 0.106555355338929\n",
      "best merge is 18, 23, 0.11204205149186\n",
      "best merge is 5, 9, 0.131884485601158\n",
      "best merge is 13, 16, 0.133031652764161\n",
      "best merge is 18, 19, 0.141088871040473\n",
      "best merge is 2, 12, 0.150270411475183\n",
      "best merge is 4, 7, 0.157741344023316\n",
      "best merge is 5, 8, 0.167963954320121\n",
      "best merge is 33, 40, 0.168898200596302\n",
      "best merge is 11, 20, 0.209412853487564\n",
      "best merge is 33, 41, 0.237887743484811\n",
      "best merge is 4, 17, 0.253163313919973\n",
      "best merge is 10, 29, 0.260218387887959\n",
      "best merge is 39, 42, 0.274364564819742\n",
      "best merge is 36, 38, 0.283366332965327\n",
      "best merge is 2, 5, 0.297286853706197\n",
      "best merge is 11, 25, 0.305622546233865\n",
      "best merge is 18, 21, 0.307117165237916\n",
      "best merge is 31, 39, 0.352309170766005\n",
      "best merge is 10, 27, 0.357618702360194\n",
      "best merge is 26, 34, 0.368104814653853\n",
      "best merge is 4, 22, 0.409788954248417\n",
      "best merge is 0, 2, 0.462811904136997\n",
      "best merge is 4, 18, 0.533930905633316\n",
      "best merge is 0, 11, 0.539108562016371\n",
      "best merge is 4, 26, 0.659006501150251\n",
      "best merge is 31, 36, 0.687442629197174\n",
      "best merge is 4, 10, 0.779345208458652\n",
      "best merge is 0, 33, 0.918960853674379\n",
      "best merge is 13, 31, 0.974523160723452\n",
      "order classes\n",
      "15 - 1.63958761937138\n",
      "5 - 0.374791591198312\n",
      "14 - 1.67698570865293\n",
      "4 - 0.803478614123747\n",
      "5 - 3.70625944997609\n",
      "1 3 14 6 28\n",
      "0 37 2 12 5 15 9 8 11 20 25 33 35 40 41\n",
      "4 7 17 22 18 23 19 30 21 26 34 10 29 27\n",
      "13 16 24 32\n",
      "31 39 42 36 38\n",
      "Done.-18.2556634304207\t42\t0\n",
      "-16.1941747572816\t42\t3.48867313915858\n",
      "-15.6957928802589\t42\t4.12297734627832\n",
      "-15.378640776699\t42\t4.48543689320388\n",
      "-15.1067961165049\t42\t4.75728155339806\n",
      "-14.8349514563107\t42\t5.02912621359223\n",
      "-14.495145631068\t42\t5.30097087378641\n",
      "-13.9514563106796\t42\t5.70873786407767\n",
      "4.19417475728155\t42\t6.38834951456311\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_P_LVM1_2.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_P_LVM1_4.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl/human_ensembl.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_P_LVM1_4.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 13290... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 17689\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 13290 out of 13290 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 13290\t17689\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1477 hits go_terms_num = 17689\n",
      "cat 692, mi = 0.01528542, structural constituent of ribosome\n",
      "cat 2546, mi = 0.01314320, SRP-dependent cotranslational protein targeting to membrane\n",
      "cat 836, mi = 0.01254410, translational initiation\n",
      "cat 1498, mi = 0.00920937, viral transcription\n",
      "cat 2545, mi = 0.00912583, nuclear-transcribed mRNA catabolic process, nonsense-mediated decay\n",
      "cat 2016, mi = 0.00719912, cytosolic large ribosomal subunit\n",
      "cat 7, mi = 0.00633845, respiratory chain\n",
      "cat 9, mi = 0.00481711, NADH dehydrogenase (ubiquinone) activity\n",
      "cat 1876, mi = 0.00460201, catalytic step 2 spliceosome\n",
      "cat 1872, mi = 0.00435425, spliceosomal complex\n",
      "cat 12, mi = 0.00426586, mitochondrial respiratory chain complex I\n",
      "cat 10, mi = 0.00425614, mitochondrial electron transport, NADH to ubiquinone\n",
      "cat 4152, mi = 0.00424993, cytosolic small ribosomal subunit\n",
      "cat 11, mi = 0.00402692, mitochondrial respiratory chain complex I assembly\n",
      "cat 4159, mi = 0.00387084, small ribosomal subunit\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 692, mi = 0.015, pvalue = 0.000, structural constituent of ribosome\n",
      "Passed: cat 7, mi = 0.006, pvalue = 0.000, respiratory chain\n",
      "Passed: cat 1876, mi = 0.005, pvalue = 0.000, catalytic step 2 spliceosome\n",
      "Passed: cat 1064, mi = 0.004, pvalue = 0.000, proteasome complex\n",
      "Passed: cat 54, mi = 0.003, pvalue = 0.000, ATP biosynthetic process\n",
      "Passed: cat 893, mi = 0.003, pvalue = 0.000, protein folding\n",
      "Passed: cat 835, mi = 0.002, pvalue = 0.000, translation initiation factor activity\n",
      "Passed: cat 1558, mi = 0.002, pvalue = 0.000, protein targeting to mitochondrion\n",
      "Passed: cat 3632, mi = 0.002, pvalue = 0.000, glycosphingolipid metabolic process\n",
      "Passed: cat 3048, mi = 0.002, pvalue = 0.000, aminoacyl-tRNA synthetase multienzyme complex\n",
      "Passed: cat 944, mi = 0.002, pvalue = 0.000, transcription elongation from RNA polymerase II promoter\n",
      "Passed: cat 1956, mi = 0.002, pvalue = 0.000, Rab GTPase binding\n",
      "Passed: cat 2431, mi = 0.002, pvalue = 0.000, nucleosome\n",
      "Passed: cat 2600, mi = 0.002, pvalue = 0.000, microtubule-based process\n",
      "Passed: cat 3156, mi = 0.002, pvalue = 0.000, myosin complex\n",
      "Passed: cat 6976, mi = 0.002, pvalue = 0.000, preribosome, large subunit precursor\n",
      "Passed: cat 1134, mi = 0.002, pvalue = 0.000, intracellular estrogen receptor signaling pathway\n",
      "Passed: cat 1985, mi = 0.001, pvalue = 0.002, interferon-gamma-mediated signaling pathway\n",
      "Passed: cat 1906, mi = 0.001, pvalue = 0.000, lipoprotein metabolic process\n",
      "Passed: cat 1858, mi = 0.001, pvalue = 0.003, cellular response to mechanical stimulus\n",
      "Passed: cat 2291, mi = 0.001, pvalue = 0.002, potassium ion transmembrane transport\n",
      "Passed: cat 1196, mi = 0.001, pvalue = 0.002, canonical Wnt signaling pathway\n",
      "Passed: cat 1358, mi = 0.001, pvalue = 0.003, lipid particle\n",
      "Passed: cat 2948, mi = 0.001, pvalue = 0.002, Rho guanyl-nucleotide exchange factor activity\n",
      "Didn't pass (cutoff=0.0050): cat 1864, mi = 0.001, pvalue = 0.007, ATP-dependent chromatin remodeling\n",
      "Didn't pass (cutoff=0.0050): cat 1701, mi = 0.001, pvalue = 0.007, phagocytic vesicle\n",
      "Passed: cat 2252, mi = 0.001, pvalue = 0.003, outflow tract septum morphogenesis\n",
      "Didn't pass (cutoff=0.0050): cat 884, mi = 0.001, pvalue = 0.007, regulation of nucleic acid-templated transcription\n",
      "Didn't pass (cutoff=0.0050): cat 698, mi = 0.001, pvalue = 0.009, tricarboxylic acid cycle\n",
      "Didn't pass (cutoff=0.0050): cat 1719, mi = 0.001, pvalue = 0.007, cellular response to oxidative stress\n",
      "Didn't pass (cutoff=0.0050): cat 5952, mi = 0.001, pvalue = 0.008, cellular response to gamma radiation\n",
      "Passed: cat 10485, mi = 0.001, pvalue = 0.002, small molecule metabolic process\n",
      "Didn't pass (cutoff=0.0050): cat 1243, mi = 0.001, pvalue = 0.009, COPII vesicle coating\n",
      "Didn't pass (cutoff=0.0050): cat 8205, mi = 0.001, pvalue = 0.007, cofactor binding\n",
      "Didn't pass (cutoff=0.0050): cat 1318, mi = 0.001, pvalue = 0.012, COP9 signalosome\n",
      "Didn't pass (cutoff=0.0050): cat 387, mi = 0.001, pvalue = 0.013, neuron apoptotic process\n",
      "Didn't pass (cutoff=0.0050): cat 855, mi = 0.001, pvalue = 0.010, chemotaxis\n",
      "Passed: cat 6534, mi = 0.001, pvalue = 0.003, Arp2/3 protein complex\n",
      "Passed: cat 1587, mi = 0.001, pvalue = 0.003, resolution of meiotic recombination intermediates\n",
      "Didn't pass (cutoff=0.0050): cat 4523, mi = 0.001, pvalue = 0.006, ligand-gated ion channel activity\n",
      "Didn't pass (cutoff=0.0050): cat 1272, mi = 0.001, pvalue = 0.013, protein phosphatase regulator activity\n",
      "Didn't pass (cutoff=0.0050): cat 4479, mi = 0.001, pvalue = 0.012, tubulin binding\n",
      "Didn't pass (cutoff=0.0050): cat 1288, mi = 0.001, pvalue = 0.015, regulation of mitochondrial membrane potential\n",
      "Didn't pass (cutoff=0.0050): cat 221, mi = 0.001, pvalue = 0.011, response to zinc ion\n",
      "Passed: cat 2968, mi = 0.001, pvalue = 0.003, L-alpha-amino acid transmembrane transport\n",
      "Didn't pass (cutoff=0.0050): cat 1203, mi = 0.001, pvalue = 0.014, negative regulation of peptidase activity\n",
      "Didn't pass (cutoff=0.0050): cat 2560, mi = 0.001, pvalue = 0.009, apicolateral plasma membrane\n",
      "Didn't pass (cutoff=0.0050): cat 4225, mi = 0.001, pvalue = 0.007, Cul4A-RING E3 ubiquitin ligase complex\n",
      "Didn't pass (cutoff=0.0050): cat 1202, mi = 0.001, pvalue = 0.016, peptidase inhibitor activity\n",
      "Didn't pass (cutoff=0.0050): cat 2759, mi = 0.001, pvalue = 0.018, cornification\n",
      "Didn't pass (cutoff=0.0050): cat 1054, mi = 0.001, pvalue = 0.012, leukocyte cell-cell adhesion\n",
      "Didn't pass (cutoff=0.0050): cat 2185, mi = 0.001, pvalue = 0.012, ER overload response\n",
      "Passed: cat 8755, mi = 0.001, pvalue = 0.002, tetrahydrofolate interconversion\n",
      "Passed: cat 6205, mi = 0.001, pvalue = 0.003, peroxiredoxin activity\n",
      "Didn't pass (cutoff=0.0050): cat 4822, mi = 0.001, pvalue = 0.006, polyamine metabolic process\n",
      "Didn't pass (cutoff=0.0050): cat 5666, mi = 0.001, pvalue = 0.005, establishment of integrated proviral latency\n",
      "Didn't pass (cutoff=0.0050): cat 7387, mi = 0.001, pvalue = 0.010, protein deneddylation\n",
      "Didn't pass (cutoff=0.0050): cat 579, mi = 0.001, pvalue = 0.013, COPII vesicle coat\n",
      "Didn't pass (cutoff=0.0050): cat 8710, mi = 0.001, pvalue = 0.024, positive regulation of DNA repair\n",
      "Didn't pass (cutoff=0.0050): cat 1473, mi = 0.001, pvalue = 0.020, protein localization\n",
      "Didn't pass (cutoff=0.0050): cat 6874, mi = 0.001, pvalue = 0.021, cysteine-type endopeptidase inhibitor activity\n",
      "Didn't pass (cutoff=0.0050): cat 4766, mi = 0.001, pvalue = 0.028, bicarbonate transport\n",
      "Didn't pass (cutoff=0.0050): cat 1110, mi = 0.001, pvalue = 0.030, excitatory postsynaptic potential\n",
      "Passed: cat 2356, mi = 0.001, pvalue = 0.002, acetylcholine receptor activity\n",
      "Passed: cat 4688, mi = 0.001, pvalue = 0.002, DNA replication factor C complex\n",
      "Didn't pass (cutoff=0.0050): cat 1363, mi = 0.001, pvalue = 0.020, chemokine-mediated signaling pathway\n",
      "Didn't pass (cutoff=0.0050): cat 4969, mi = 0.001, pvalue = 0.029, sarcomere organization\n",
      "Didn't pass (cutoff=0.0050): cat 4821, mi = 0.001, pvalue = 0.016, polyamine biosynthetic process\n",
      "Didn't pass (cutoff=0.0050): cat 2024, mi = 0.001, pvalue = 0.026, retinoic acid receptor binding\n",
      "Didn't pass (cutoff=0.0050): cat 252, mi = 0.001, pvalue = 0.025, transmembrane receptor protein tyrosine kinase signaling pathway\n",
      "Didn't pass (cutoff=0.0050): cat 5637, mi = 0.001, pvalue = 0.011, mitotic nuclear envelope reassembly\n",
      "Didn't pass (cutoff=0.0050): cat 2174, mi = 0.001, pvalue = 0.016, Lys48-specific deubiquitinase activity\n",
      "Didn't pass (cutoff=0.0050): cat 4671, mi = 0.001, pvalue = 0.028, multivesicular body\n",
      "Didn't pass (cutoff=0.0050): cat 5898, mi = 0.001, pvalue = 0.027, Ras GTPase binding\n",
      "Didn't pass (cutoff=0.0050): cat 5512, mi = 0.001, pvalue = 0.016, AP-3 adaptor complex\n",
      "Didn't pass (cutoff=0.0050): cat 555, mi = 0.001, pvalue = 0.034, negative regulation of JAK-STAT cascade\n",
      "Didn't pass (cutoff=0.0050): cat 4283, mi = 0.001, pvalue = 0.033, glutamine metabolic process\n",
      "Didn't pass (cutoff=0.0050): cat 1413, mi = 0.001, pvalue = 0.026, phospholipase C-activating G-protein coupled receptor signaling pathway\n",
      "Didn't pass (cutoff=0.0050): cat 3300, mi = 0.001, pvalue = 0.023, protein tyrosine phosphatase activity\n",
      "Didn't pass (cutoff=0.0050): cat 3303, mi = 0.001, pvalue = 0.026, peptidyl-tyrosine dephosphorylation\n",
      "Didn't pass (cutoff=0.0050): cat 925, mi = 0.001, pvalue = 0.039, nucleocytoplasmic transport\n",
      "Didn't pass (cutoff=0.0050): cat 12516, mi = 0.001, pvalue = 0.017, protein K11-linked deubiquitination\n",
      "Didn't pass (cutoff=0.0050): cat 6986, mi = 0.001, pvalue = 0.036, peroxisome organization\n",
      "Didn't pass (cutoff=0.0050): cat 6464, mi = 0.001, pvalue = 0.037, K63-linked polyubiquitin binding\n",
      "Didn't pass (cutoff=0.0050): cat 394, mi = 0.001, pvalue = 0.029, chromosome segregation\n",
      "Total number of categories evaluated = 369.\n",
      "Number of categories that passed the tests = 33\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_P_LVM1_4.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_P_LVM1_4.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 0, 4, 0.0114015614763894\n",
      "best merge is 2, 9, 0.0165363321484688\n",
      "best merge is 26, 30, 0.0226894015014851\n",
      "best merge is 3, 5, 0.0254236861000421\n",
      "best merge is 2, 25, 0.030745632092878\n",
      "best merge is 0, 26, 0.0382929808715401\n",
      "best merge is 2, 3, 0.0527398806476267\n",
      "best merge is 27, 31, 0.059908185310906\n",
      "best merge is 1, 6, 0.0642751899944938\n",
      "best merge is 7, 32, 0.0888097117444903\n",
      "best merge is 2, 16, 0.110923604172998\n",
      "best merge is 2, 7, 0.130000783171775\n",
      "best merge is 0, 2, 0.165784055293612\n",
      "best merge is 12, 19, 0.185670664649812\n",
      "best merge is 14, 27, 0.188496516259435\n",
      "best merge is 0, 29, 0.230395320565808\n",
      "best merge is 1, 13, 0.314612819618946\n",
      "best merge is 20, 23, 0.324493329386669\n",
      "best merge is 0, 10, 0.332889943070504\n",
      "best merge is 14, 28, 0.35025130947618\n",
      "best merge is 11, 24, 0.356679620775393\n",
      "best merge is 0, 22, 0.410591379711351\n",
      "best merge is 1, 12, 0.470709930578182\n",
      "best merge is 8, 15, 0.502988867411486\n",
      "best merge is 17, 20, 0.51257638304245\n",
      "best merge is 0, 1, 0.68048619569619\n",
      "best merge is 11, 21, 0.724314808950238\n",
      "best merge is 11, 17, 0.799540756710563\n",
      "best merge is 0, 8, 0.909079794028974\n",
      "order classes\n",
      "20 - 0.62449339037317\n",
      "2 - 4.14965815614476\n",
      "6 - 3.16275121083243\n",
      "4 - 1.53414436957889\n",
      "0 4 26 30 2 9 25 3 5 16 7 32 29 10 22 1 6 13 12 19\n",
      "8 15\n",
      "11 24 21 17 20 23\n",
      "14 27 31 28\n",
      "18\n",
      "Done.Start drawing\n",
      "110.21\t0.00\n",
      "glycosphingolipid metabolic process, GO:0006687\n",
      "preribosome, large subunit precursor, GO:0030687\n",
      "structural constituent of ribosome, GO:0003735\n",
      "ATP biosynthetic process, GO:0006754\n",
      "Arp2/3 protein complex, GO:0005885\n",
      "peroxiredoxin activity, GO:0051920\n",
      "catalytic step 2 spliceosome, GO:0071013\n",
      "aminoacyl-tRNA synthetase multienzyme complex, GO:0017101\n",
      "small molecule metabolic process, GO:0044281\n",
      "proteasome complex, GO:0000502\n",
      "protein folding, GO:0006457\n",
      "intracellular estrogen receptor signaling pathway, GO:0030520\n",
      "protein targeting to mitochondrion, GO:0006626\n",
      "DNA replication factor C complex, GO:0005663\n",
      "tetrahydrofolate interconversion, GO:0035999\n",
      "transcription elongation from RNA polymerase II promoter, GO:0006368\n",
      "lipid particle, GO:0005811\n",
      "respiratory chain, GO:0070469\n",
      "translation initiation factor activity, GO:0003743\n",
      "microtubule-based process, GO:0007017\n",
      "nucleosome, GO:0000786\n",
      "cellular response to mechanical stimulus, GO:0071260\n",
      "canonical Wnt signaling pathway, GO:0060070\n",
      "interferon-gamma-mediated signaling pathway, GO:0060333\n",
      "potassium ion transmembrane transport, GO:0071805\n",
      "Rho guanyl-nucleotide exchange factor activity, GO:0005089\n",
      "Rab GTPase binding, GO:0017137\n",
      "outflow tract septum morphogenesis, GO:0003148\n",
      "lipoprotein metabolic process, GO:0042157\n",
      "myosin complex, GO:0016459\n",
      "resolution of meiotic recombination intermediates, GO:0000712\n",
      "acetylcholine receptor activity, GO:0015464\n",
      "L-alpha-amino acid transmembrane transport, GO:1902475\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_P_LVM1_4.csv_PAGE/filt_FC_P_LVM1_4.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_P_LVM1_4.csv_PAGE/filt_FC_P_LVM1_4.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_P_LVM1_4.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_P_LVM1_4.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 0, 4, 0.0114015614763894\n",
      "best merge is 2, 9, 0.0165363321484688\n",
      "best merge is 26, 30, 0.0226894015014851\n",
      "best merge is 3, 5, 0.0254236861000421\n",
      "best merge is 2, 25, 0.030745632092878\n",
      "best merge is 0, 26, 0.0382929808715401\n",
      "best merge is 2, 3, 0.0527398806476267\n",
      "best merge is 27, 31, 0.059908185310906\n",
      "best merge is 1, 6, 0.0642751899944938\n",
      "best merge is 7, 32, 0.0888097117444903\n",
      "best merge is 2, 16, 0.110923604172998\n",
      "best merge is 2, 7, 0.130000783171775\n",
      "best merge is 0, 2, 0.165784055293612\n",
      "best merge is 12, 19, 0.185670664649812\n",
      "best merge is 14, 27, 0.188496516259435\n",
      "best merge is 0, 29, 0.230395320565808\n",
      "best merge is 1, 13, 0.314612819618946\n",
      "best merge is 20, 23, 0.324493329386669\n",
      "best merge is 0, 10, 0.332889943070504\n",
      "best merge is 14, 28, 0.35025130947618\n",
      "best merge is 11, 24, 0.356679620775393\n",
      "best merge is 0, 22, 0.410591379711351\n",
      "best merge is 1, 12, 0.470709930578182\n",
      "best merge is 8, 15, 0.502988867411486\n",
      "best merge is 17, 20, 0.51257638304245\n",
      "best merge is 0, 1, 0.68048619569619\n",
      "best merge is 11, 21, 0.724314808950238\n",
      "best merge is 11, 17, 0.799540756710563\n",
      "best merge is 0, 8, 0.909079794028974\n",
      "order classes\n",
      "20 - 0.62449339037317\n",
      "2 - 4.14965815614476\n",
      "6 - 3.16275121083243\n",
      "4 - 1.53414436957889\n",
      "0 4 26 30 2 9 25 3 5 16 7 32 29 10 22 1 6 13 12 19\n",
      "8 15\n",
      "11 24 21 17 20 23\n",
      "14 27 31 28\n",
      "18\n",
      "Done.-20\t42\t0\n",
      "-20\t42\t0\n",
      "-19.9961890935487\t42\t0\n",
      "-19.988567280646\t42\t0.00762181290264041\n",
      "-19.9809454677434\t42\t0.0152436258052808\n",
      "-19.9714182016151\t42\t0.0228654387079212\n",
      "-19.8494691951729\t42\t0.0342981580618819\n",
      "-19.5141094274567\t42\t0.266763451592414\n",
      "1.35250884674712\t42\t0.705017693494238\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_P_LVM1_4.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_P_LVM1_3.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl/human_ensembl.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_P_LVM1_3.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 13290... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 17689\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 13290 out of 13290 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 13290\t17689\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1477 hits go_terms_num = 17689\n",
      "cat 2023, mi = 0.00992249, DNA replication\n",
      "cat 2546, mi = 0.00856772, SRP-dependent cotranslational protein targeting to membrane\n",
      "cat 2990, mi = 0.00810647, chromosome, centromeric region\n",
      "cat 692, mi = 0.00758722, structural constituent of ribosome\n",
      "cat 2121, mi = 0.00723262, sister chromatid cohesion\n",
      "cat 836, mi = 0.00618431, translational initiation\n",
      "cat 1498, mi = 0.00597925, viral transcription\n",
      "cat 4583, mi = 0.00585539, condensed chromosome kinetochore\n",
      "cat 2545, mi = 0.00552218, nuclear-transcribed mRNA catabolic process, nonsense-mediated decay\n",
      "cat 2585, mi = 0.00526053, kinetochore\n",
      "cat 1528, mi = 0.00481837, DNA synthesis involved in DNA repair\n",
      "cat 394, mi = 0.00445214, chromosome segregation\n",
      "cat 1268, mi = 0.00422294, mitotic cell cycle\n",
      "cat 2642, mi = 0.00411273, DNA replication initiation\n",
      "cat 1308, mi = 0.00400436, spindle pole\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 2023, mi = 0.010, pvalue = 0.000, DNA replication\n",
      "Passed: cat 2546, mi = 0.009, pvalue = 0.000, SRP-dependent cotranslational protein targeting to membrane\n",
      "Passed: cat 2990, mi = 0.008, pvalue = 0.000, chromosome, centromeric region\n",
      "Passed: cat 2431, mi = 0.004, pvalue = 0.000, nucleosome\n",
      "Passed: cat 693, mi = 0.003, pvalue = 0.000, mitochondrial translational elongation\n",
      "Passed: cat 3799, mi = 0.003, pvalue = 0.000, termination of RNA polymerase II transcription\n",
      "Passed: cat 1080, mi = 0.003, pvalue = 0.000, regulation of transcription from RNA polymerase II promoter in response to hypoxia\n",
      "Passed: cat 2101, mi = 0.003, pvalue = 0.000, centriole\n",
      "Passed: cat 6415, mi = 0.002, pvalue = 0.000, chromosome condensation\n",
      "Passed: cat 5956, mi = 0.002, pvalue = 0.000, four-way junction DNA binding\n",
      "Passed: cat 1886, mi = 0.002, pvalue = 0.000, mitochondrial intermembrane space\n",
      "Passed: cat 2371, mi = 0.002, pvalue = 0.000, positive regulation of cytokinesis\n",
      "Passed: cat 2783, mi = 0.002, pvalue = 0.000, ubiquitin ligase complex\n",
      "Passed: cat 887, mi = 0.002, pvalue = 0.000, regulation of endocytosis\n",
      "Passed: cat 5706, mi = 0.002, pvalue = 0.000, peroxisome\n",
      "Passed: cat 1175, mi = 0.002, pvalue = 0.000, guanyl-nucleotide exchange factor activity\n",
      "Passed: cat 5005, mi = 0.002, pvalue = 0.000, Ran GTPase binding\n",
      "Passed: cat 2874, mi = 0.002, pvalue = 0.001, RNA metabolic process\n",
      "Passed: cat 761, mi = 0.002, pvalue = 0.001, calcium-dependent phospholipid binding\n",
      "Passed: cat 353, mi = 0.002, pvalue = 0.000, cellular response to interleukin-1\n",
      "Passed: cat 1679, mi = 0.002, pvalue = 0.000, late endosome\n",
      "Passed: cat 108, mi = 0.001, pvalue = 0.001, positive regulation of pathway-restricted SMAD protein phosphorylation\n",
      "Passed: cat 1671, mi = 0.001, pvalue = 0.001, specific granule membrane\n",
      "Passed: cat 2979, mi = 0.001, pvalue = 0.001, melanosome organization\n",
      "Passed: cat 550, mi = 0.001, pvalue = 0.002, chromatin remodeling\n",
      "Passed: cat 4110, mi = 0.001, pvalue = 0.003, sterol biosynthetic process\n",
      "Passed: cat 27, mi = 0.001, pvalue = 0.002, cytochrome-c oxidase activity\n",
      "Passed: cat 3273, mi = 0.001, pvalue = 0.002, palmitoyltransferase activity\n",
      "Passed: cat 54, mi = 0.001, pvalue = 0.003, ATP biosynthetic process\n",
      "Passed: cat 2759, mi = 0.001, pvalue = 0.003, cornification\n",
      "Passed: cat 6658, mi = 0.001, pvalue = 0.003, skeletal muscle tissue development\n",
      "Passed: cat 516, mi = 0.001, pvalue = 0.003, lipid catabolic process\n",
      "Passed: cat 4241, mi = 0.001, pvalue = 0.002, positive regulation of histone H3-K4 methylation\n",
      "Passed: cat 1242, mi = 0.001, pvalue = 0.003, ER to Golgi vesicle-mediated transport\n",
      "Passed: cat 1050, mi = 0.001, pvalue = 0.004, substrate adhesion-dependent cell spreading\n",
      "Passed: cat 1985, mi = 0.001, pvalue = 0.005, interferon-gamma-mediated signaling pathway\n",
      "Didn't pass (cutoff=0.0050): cat 884, mi = 0.001, pvalue = 0.005, regulation of nucleic acid-templated transcription\n",
      "Passed: cat 3920, mi = 0.001, pvalue = 0.004, SUMO transferase activity\n",
      "Passed: cat 1097, mi = 0.001, pvalue = 0.005, neuropeptide signaling pathway\n",
      "Didn't pass (cutoff=0.0050): cat 3814, mi = 0.001, pvalue = 0.006, apoptotic mitochondrial changes\n",
      "Didn't pass (cutoff=0.0050): cat 1475, mi = 0.001, pvalue = 0.007, positive regulation of B cell proliferation\n",
      "Passed: cat 10485, mi = 0.001, pvalue = 0.002, small molecule metabolic process\n",
      "Passed: cat 4056, mi = 0.001, pvalue = 0.002, positive regulation of autophagosome assembly\n",
      "Didn't pass (cutoff=0.0050): cat 2659, mi = 0.001, pvalue = 0.011, ribonucleoprotein complex binding\n",
      "Didn't pass (cutoff=0.0050): cat 2869, mi = 0.001, pvalue = 0.007, nucleoside metabolic process\n",
      "Didn't pass (cutoff=0.0050): cat 3300, mi = 0.001, pvalue = 0.007, protein tyrosine phosphatase activity\n",
      "Didn't pass (cutoff=0.0050): cat 3303, mi = 0.001, pvalue = 0.007, peptidyl-tyrosine dephosphorylation\n",
      "Passed: cat 8252, mi = 0.001, pvalue = 0.002, actin crosslink formation\n",
      "Passed: cat 5189, mi = 0.001, pvalue = 0.003, positive regulation of cell motility\n",
      "Didn't pass (cutoff=0.0050): cat 837, mi = 0.001, pvalue = 0.008, eukaryotic translation initiation factor 3 complex\n",
      "Didn't pass (cutoff=0.0050): cat 3302, mi = 0.001, pvalue = 0.011, phosphatase activity\n",
      "Didn't pass (cutoff=0.0050): cat 2985, mi = 0.001, pvalue = 0.007, chaperone-mediated protein complex assembly\n",
      "Didn't pass (cutoff=0.0050): cat 5638, mi = 0.001, pvalue = 0.012, decidualization\n",
      "Didn't pass (cutoff=0.0050): cat 5204, mi = 0.001, pvalue = 0.011, protein targeting to Golgi\n",
      "Didn't pass (cutoff=0.0050): cat 630, mi = 0.001, pvalue = 0.015, sodium ion transport\n",
      "Didn't pass (cutoff=0.0050): cat 4661, mi = 0.001, pvalue = 0.018, cardiac conduction\n",
      "Didn't pass (cutoff=0.0050): cat 3500, mi = 0.001, pvalue = 0.015, transcription factor activity, RNA polymerase II core promoter proximal region sequence-specific binding\n",
      "Didn't pass (cutoff=0.0050): cat 1742, mi = 0.001, pvalue = 0.007, vasodilation\n",
      "Didn't pass (cutoff=0.0050): cat 2583, mi = 0.001, pvalue = 0.019, response to cAMP\n",
      "Didn't pass (cutoff=0.0050): cat 3790, mi = 0.001, pvalue = 0.016, carboxypeptidase activity\n",
      "Didn't pass (cutoff=0.0050): cat 2295, mi = 0.001, pvalue = 0.020, potassium channel activity\n",
      "Passed: cat 5416, mi = 0.001, pvalue = 0.003, clathrin-coated endocytic vesicle\n",
      "Didn't pass (cutoff=0.0050): cat 881, mi = 0.001, pvalue = 0.013, protein dephosphorylation\n",
      "Didn't pass (cutoff=0.0050): cat 1584, mi = 0.001, pvalue = 0.024, response to metal ion\n",
      "Didn't pass (cutoff=0.0050): cat 9001, mi = 0.001, pvalue = 0.011, negative regulation of protein localization to cell surface\n",
      "Didn't pass (cutoff=0.0050): cat 1781, mi = 0.001, pvalue = 0.015, fatty-acyl-CoA biosynthetic process\n",
      "Passed: cat 6113, mi = 0.001, pvalue = 0.001, detection of chemical stimulus involved in sensory perception of bitter taste\n",
      "Passed: cat 7071, mi = 0.001, pvalue = 0.005, protein anchor\n",
      "Didn't pass (cutoff=0.0050): cat 5284, mi = 0.001, pvalue = 0.019, intraciliary retrograde transport\n",
      "Didn't pass (cutoff=0.0050): cat 1256, mi = 0.001, pvalue = 0.024, tRNA processing\n",
      "Didn't pass (cutoff=0.0050): cat 1241, mi = 0.001, pvalue = 0.034, collagen catabolic process\n",
      "Didn't pass (cutoff=0.0050): cat 1645, mi = 0.001, pvalue = 0.026, metalloendopeptidase activity\n",
      "Didn't pass (cutoff=0.0050): cat 1249, mi = 0.001, pvalue = 0.032, small-subunit processome\n",
      "Didn't pass (cutoff=0.0050): cat 4770, mi = 0.001, pvalue = 0.041, translation factor activity, RNA binding\n",
      "Didn't pass (cutoff=0.0050): cat 5860, mi = 0.001, pvalue = 0.013, response to pH\n",
      "Didn't pass (cutoff=0.0050): cat 5485, mi = 0.001, pvalue = 0.021, regulation of DNA damage checkpoint\n",
      "Didn't pass (cutoff=0.0050): cat 7024, mi = 0.001, pvalue = 0.020, double-stranded telomeric DNA binding\n",
      "Didn't pass (cutoff=0.0050): cat 1850, mi = 0.001, pvalue = 0.038, symporter activity\n",
      "Passed: cat 2657, mi = 0.001, pvalue = 0.000, cytidine deaminase activity\n",
      "Passed: cat 5665, mi = 0.001, pvalue = 0.001, nuclear transport\n",
      "Didn't pass (cutoff=0.0050): cat 2498, mi = 0.001, pvalue = 0.042, protein N-linked glycosylation\n",
      "Didn't pass (cutoff=0.0050): cat 858, mi = 0.001, pvalue = 0.031, intrinsic component of plasma membrane\n",
      "Didn't pass (cutoff=0.0050): cat 1176, mi = 0.001, pvalue = 0.040, cholesterol homeostasis\n",
      "Didn't pass (cutoff=0.0050): cat 3570, mi = 0.001, pvalue = 0.034, steroid binding\n",
      "Didn't pass (cutoff=0.0050): cat 3059, mi = 0.001, pvalue = 0.035, regulation of protein secretion\n",
      "Didn't pass (cutoff=0.0050): cat 3301, mi = 0.001, pvalue = 0.042, protein tyrosine/serine/threonine phosphatase activity\n",
      "Passed: cat 6371, mi = 0.001, pvalue = 0.004, connexin complex\n",
      "Didn't pass (cutoff=0.0050): cat 3034, mi = 0.001, pvalue = 0.038, cytoplasmic stress granule\n",
      "Didn't pass (cutoff=0.0050): cat 2462, mi = 0.001, pvalue = 0.042, regulation of JNK cascade\n",
      "Passed: cat 12354, mi = 0.001, pvalue = 0.001, detection of virus\n",
      "Didn't pass (cutoff=0.0050): cat 8495, mi = 0.001, pvalue = 0.055, nucleotide biosynthetic process\n",
      "Didn't pass (cutoff=0.0050): cat 2292, mi = 0.001, pvalue = 0.054, voltage-gated potassium channel activity\n",
      "Didn't pass (cutoff=0.0050): cat 3751, mi = 0.001, pvalue = 0.058, nuclear-transcribed mRNA poly(A) tail shortening\n",
      "Didn't pass (cutoff=0.0050): cat 5849, mi = 0.001, pvalue = 0.059, positive regulation of ATPase activity\n",
      "Didn't pass (cutoff=0.0050): cat 6626, mi = 0.001, pvalue = 0.058, MAP kinase kinase kinase activity\n",
      "Didn't pass (cutoff=0.0050): cat 3734, mi = 0.001, pvalue = 0.021, lipid transporter activity\n",
      "Didn't pass (cutoff=0.0050): cat 892, mi = 0.001, pvalue = 0.052, chaperone-mediated protein folding\n",
      "Didn't pass (cutoff=0.0050): cat 1407, mi = 0.001, pvalue = 0.033, defense response to fungus\n",
      "Didn't pass (cutoff=0.0050): cat 5245, mi = 0.001, pvalue = 0.036, NAD biosynthesis via nicotinamide riboside salvage pathway\n",
      "Didn't pass (cutoff=0.0050): cat 8992, mi = 0.001, pvalue = 0.046, MAP kinase kinase kinase kinase activity\n",
      "Didn't pass (cutoff=0.0050): cat 5044, mi = 0.001, pvalue = 0.053, cellular amino acid biosynthetic process\n",
      "Didn't pass (cutoff=0.0050): cat 2080, mi = 0.001, pvalue = 0.053, cochlea development\n",
      "Didn't pass (cutoff=0.0050): cat 8208, mi = 0.001, pvalue = 0.066, cellular nitrogen compound metabolic process\n",
      "Didn't pass (cutoff=0.0050): cat 1120, mi = 0.001, pvalue = 0.059, aminoacyl-tRNA editing activity\n",
      "Didn't pass (cutoff=0.0050): cat 946, mi = 0.001, pvalue = 0.064, transcription factor TFIID complex\n",
      "Didn't pass (cutoff=0.0050): cat 6034, mi = 0.001, pvalue = 0.039, positive regulation of cholesterol efflux\n",
      "Didn't pass (cutoff=0.0050): cat 1747, mi = 0.001, pvalue = 0.067, skeletal muscle fiber development\n",
      "Didn't pass (cutoff=0.0050): cat 3304, mi = 0.001, pvalue = 0.065, MAP kinase tyrosine/serine/threonine phosphatase activity\n",
      "Didn't pass (cutoff=0.0050): cat 8523, mi = 0.001, pvalue = 0.068, regulation of mitochondrial membrane permeability\n",
      "Didn't pass (cutoff=0.0050): cat 3880, mi = 0.001, pvalue = 0.018, skin morphogenesis\n",
      "Total number of categories evaluated = 637.\n",
      "Number of categories that passed the tests = 49\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_P_LVM1_3.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_P_LVM1_3.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 0, 9, 0.00997035037282401\n",
      "best merge is 0, 8, 0.0130654632781347\n",
      "best merge is 45, 48, 0.015841149529119\n",
      "best merge is 7, 11, 0.0175005046173147\n",
      "best merge is 0, 2, 0.0181569689479181\n",
      "best merge is 0, 7, 0.0435227210632454\n",
      "best merge is 40, 41, 0.051765196537765\n",
      "best merge is 3, 43, 0.0588183392108779\n",
      "best merge is 31, 45, 0.0656037023861643\n",
      "best merge is 13, 30, 0.0688893918671465\n",
      "best merge is 4, 17, 0.0994854725660661\n",
      "best merge is 31, 35, 0.100824843350846\n",
      "best merge is 1, 26, 0.102805499252939\n",
      "best merge is 0, 3, 0.112072954393484\n",
      "best merge is 27, 46, 0.12530335607606\n",
      "best merge is 18, 47, 0.132822496690348\n",
      "best merge is 18, 31, 0.160922362442612\n",
      "best merge is 40, 44, 0.183753155451165\n",
      "best merge is 5, 16, 0.188873534367533\n",
      "best merge is 4, 10, 0.191239552346984\n",
      "best merge is 15, 20, 0.198309374694115\n",
      "best merge is 25, 38, 0.204812268025303\n",
      "best merge is 0, 37, 0.240388357492805\n",
      "best merge is 13, 24, 0.274369946693661\n",
      "best merge is 4, 6, 0.277892196955194\n",
      "best merge is 23, 40, 0.290626780502856\n",
      "best merge is 14, 22, 0.292551753966095\n",
      "best merge is 12, 29, 0.296838438580594\n",
      "best merge is 5, 27, 0.314473629488861\n",
      "best merge is 4, 28, 0.316670440106862\n",
      "best merge is 15, 34, 0.339864923378783\n",
      "best merge is 1, 36, 0.359186142705103\n",
      "best merge is 14, 33, 0.366544707314872\n",
      "best merge is 23, 25, 0.399426241074981\n",
      "best merge is 13, 39, 0.443477202582162\n",
      "best merge is 13, 14, 0.49480545684418\n",
      "best merge is 0, 32, 0.536124614991009\n",
      "best merge is 1, 12, 0.557576494963018\n",
      "best merge is 4, 23, 0.60088990518505\n",
      "best merge is 18, 42, 0.642425124053325\n",
      "best merge is 1, 15, 0.677710193070886\n",
      "best merge is 1, 13, 0.704789695774086\n",
      "best merge is 4, 5, 0.769051671946854\n",
      "best merge is 18, 19, 0.806137842160717\n",
      "best merge is 18, 21, 0.881161162051695\n",
      "order classes\n",
      "10 - 0.843601311081653\n",
      "15 - 1.85508215292623\n",
      "15 - 1.76285997990116\n",
      "8 - 2.17576080422453\n",
      "0 9 8 2 7 11 3 43 37 32\n",
      "1 26 36 12 29 15 20 34 13 30 24 39 14 22 33\n",
      "4 17 10 6 28 23 40 41 44 25 38 5 16 27 46\n",
      "18 47 31 45 48 35 42 19\n",
      "21\n",
      "Done.Start drawing\n",
      "15.99\t0.00\n",
      "specific granule membrane, GO:0035579\n",
      "calcium-dependent phospholipid binding, GO:0005544\n",
      "connexin complex, GO:0005922\n",
      "lipid catabolic process, GO:0016042\n",
      "cytidine deaminase activity, GO:0004126\n",
      "detection of virus, GO:0009597\n",
      "interferon-gamma-mediated signaling pathway, GO:0060333\n",
      "positive regulation of pathway-restricted SMAD protein phosphorylation, GO:0010862\n",
      "ubiquitin ligase complex, GO:0000151\n",
      "cornification, GO:0070268\n",
      "regulation of endocytosis, GO:0030100\n",
      "skeletal muscle tissue development, GO:0007519\n",
      "chromatin remodeling, GO:0006338\n",
      "positive regulation of autophagosome assembly, GO:2000786\n",
      "peroxisome, GO:0005777\n",
      "ER to Golgi vesicle-mediated transport, GO:0006888\n",
      "SUMO transferase activity, GO:0019789\n",
      "guanyl-nucleotide exchange factor activity, GO:0005085\n",
      "late endosome, GO:0005770\n",
      "substrate adhesion-dependent cell spreading, GO:0034446\n",
      "cellular response to interleukin-1, GO:0071347\n",
      "SRP-dependent cotranslational protein targeting to membrane, GO:0006614\n",
      "cytochrome-c oxidase activity, GO:0004129\n",
      "positive regulation of histone H3-K4 methylation, GO:0051571\n",
      "RNA metabolic process, GO:0016070\n",
      "mitochondrial intermembrane space, GO:0005758\n",
      "ATP biosynthetic process, GO:0006754\n",
      "clathrin-coated endocytic vesicle, GO:0045334\n",
      "mitochondrial translational elongation, GO:0070125\n",
      "regulation of transcription from RNA polymerase II promoter in response to hypoxia, GO:0061418\n",
      "melanosome organization, GO:0032438\n",
      "actin crosslink formation, GO:0051764\n",
      "positive regulation of cell motility, GO:2000147\n",
      "protein anchor, GO:0043495\n",
      "sterol biosynthetic process, GO:0016126\n",
      "small molecule metabolic process, GO:0044281\n",
      "termination of RNA polymerase II transcription, GO:0006369\n",
      "Ran GTPase binding, GO:0008536\n",
      "palmitoyltransferase activity, GO:0016409\n",
      "nuclear transport, GO:0051169\n",
      "DNA replication, GO:0006260\n",
      "four-way junction DNA binding, GO:0000400\n",
      "chromosome condensation, GO:0030261\n",
      "chromosome, centromeric region, GO:0000775\n",
      "centriole, GO:0005814\n",
      "positive regulation of cytokinesis, GO:0032467\n",
      "nucleosome, GO:0000786\n",
      "detection of chemical stimulus involved in sensory perception of bitter taste, GO:0001580\n",
      "neuropeptide signaling pathway, GO:0007218\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_P_LVM1_3.csv_PAGE/filt_FC_P_LVM1_3.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_P_LVM1_3.csv_PAGE/filt_FC_P_LVM1_3.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_P_LVM1_3.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_P_LVM1_3.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 0, 9, 0.00997035037282401\n",
      "best merge is 0, 8, 0.0130654632781347\n",
      "best merge is 45, 48, 0.015841149529119\n",
      "best merge is 7, 11, 0.0175005046173147\n",
      "best merge is 0, 2, 0.0181569689479181\n",
      "best merge is 0, 7, 0.0435227210632454\n",
      "best merge is 40, 41, 0.051765196537765\n",
      "best merge is 3, 43, 0.0588183392108779\n",
      "best merge is 31, 45, 0.0656037023861643\n",
      "best merge is 13, 30, 0.0688893918671465\n",
      "best merge is 4, 17, 0.0994854725660661\n",
      "best merge is 31, 35, 0.100824843350846\n",
      "best merge is 1, 26, 0.102805499252939\n",
      "best merge is 0, 3, 0.112072954393484\n",
      "best merge is 27, 46, 0.12530335607606\n",
      "best merge is 18, 47, 0.132822496690348\n",
      "best merge is 18, 31, 0.160922362442612\n",
      "best merge is 40, 44, 0.183753155451165\n",
      "best merge is 5, 16, 0.188873534367533\n",
      "best merge is 4, 10, 0.191239552346984\n",
      "best merge is 15, 20, 0.198309374694115\n",
      "best merge is 25, 38, 0.204812268025303\n",
      "best merge is 0, 37, 0.240388357492805\n",
      "best merge is 13, 24, 0.274369946693661\n",
      "best merge is 4, 6, 0.277892196955194\n",
      "best merge is 23, 40, 0.290626780502856\n",
      "best merge is 14, 22, 0.292551753966095\n",
      "best merge is 12, 29, 0.296838438580594\n",
      "best merge is 5, 27, 0.314473629488861\n",
      "best merge is 4, 28, 0.316670440106862\n",
      "best merge is 15, 34, 0.339864923378783\n",
      "best merge is 1, 36, 0.359186142705103\n",
      "best merge is 14, 33, 0.366544707314872\n",
      "best merge is 23, 25, 0.399426241074981\n",
      "best merge is 13, 39, 0.443477202582162\n",
      "best merge is 13, 14, 0.49480545684418\n",
      "best merge is 0, 32, 0.536124614991009\n",
      "best merge is 1, 12, 0.557576494963018\n",
      "best merge is 4, 23, 0.60088990518505\n",
      "best merge is 18, 42, 0.642425124053325\n",
      "best merge is 1, 15, 0.677710193070886\n",
      "best merge is 1, 13, 0.704789695774086\n",
      "best merge is 4, 5, 0.769051671946854\n",
      "best merge is 18, 19, 0.806137842160717\n",
      "best merge is 18, 21, 0.881161162051695\n",
      "order classes\n",
      "10 - 0.843601311081653\n",
      "15 - 1.85508215292623\n",
      "15 - 1.76285997990116\n",
      "8 - 2.17576080422453\n",
      "0 9 8 2 7 11 3 43 37 32\n",
      "1 26 36 12 29 15 20 34 13 30 24 39 14 22 33\n",
      "4 17 10 6 28 23 40 41 44 25 38 5 16 27 46\n",
      "18 47 31 45 48 35 42 19\n",
      "21\n",
      "Done.-19.4090056285178\t42\t0\n",
      "-18.6604127579737\t42\t1.18198874296435\n",
      "-18.3977485928705\t42\t1.49718574108818\n",
      "-18.2007504690432\t42\t1.70731707317073\n",
      "-18.0168855534709\t42\t1.89118198874296\n",
      "-17.8198874296435\t42\t2.0750469043152\n",
      "-17.5703564727955\t42\t2.28517823639775\n",
      "-17.1763602251407\t42\t2.57410881801126\n",
      "2.53658536585366\t42\t3.07317073170732\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_P_LVM1_3.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM2_4.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl/human_ensembl.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM2_4.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 13290... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 17689\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 13290 out of 13290 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 13290\t17689\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1477 hits go_terms_num = 17689\n",
      "cat 692, mi = 0.00962421, structural constituent of ribosome\n",
      "cat 1498, mi = 0.00821396, viral transcription\n",
      "cat 2546, mi = 0.00785731, SRP-dependent cotranslational protein targeting to membrane\n",
      "cat 2545, mi = 0.00679434, nuclear-transcribed mRNA catabolic process, nonsense-mediated decay\n",
      "cat 836, mi = 0.00645981, translational initiation\n",
      "cat 2023, mi = 0.00460698, DNA replication\n",
      "cat 2016, mi = 0.00451307, cytosolic large ribosomal subunit\n",
      "cat 2990, mi = 0.00380905, chromosome, centromeric region\n",
      "cat 4583, mi = 0.00375449, condensed chromosome kinetochore\n",
      "cat 2121, mi = 0.00371844, sister chromatid cohesion\n",
      "cat 1872, mi = 0.00367236, spliceosomal complex\n",
      "cat 2585, mi = 0.00322454, kinetochore\n",
      "cat 694, mi = 0.00312549, mitochondrial translational termination\n",
      "cat 4159, mi = 0.00310340, small ribosomal subunit\n",
      "cat 693, mi = 0.00304692, mitochondrial translational elongation\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 692, mi = 0.010, pvalue = 0.000, structural constituent of ribosome\n",
      "Passed: cat 2023, mi = 0.005, pvalue = 0.000, DNA replication\n",
      "Passed: cat 2990, mi = 0.004, pvalue = 0.000, chromosome, centromeric region\n",
      "Passed: cat 1872, mi = 0.004, pvalue = 0.000, spliceosomal complex\n",
      "Passed: cat 11, mi = 0.003, pvalue = 0.000, mitochondrial respiratory chain complex I assembly\n",
      "Passed: cat 2431, mi = 0.003, pvalue = 0.000, nucleosome\n",
      "Passed: cat 1076, mi = 0.002, pvalue = 0.000, regulation of mRNA stability\n",
      "Passed: cat 679, mi = 0.002, pvalue = 0.000, positive chemotaxis\n",
      "Passed: cat 25, mi = 0.002, pvalue = 0.000, hydrogen ion transmembrane transport\n",
      "Passed: cat 198, mi = 0.002, pvalue = 0.000, endocytosis\n",
      "Passed: cat 3109, mi = 0.002, pvalue = 0.000, ribosome binding\n",
      "Passed: cat 1256, mi = 0.002, pvalue = 0.000, tRNA processing\n",
      "Passed: cat 975, mi = 0.002, pvalue = 0.000, negative regulation of autophagy\n",
      "Passed: cat 698, mi = 0.002, pvalue = 0.001, tricarboxylic acid cycle\n",
      "Passed: cat 6415, mi = 0.002, pvalue = 0.000, chromosome condensation\n",
      "Passed: cat 1280, mi = 0.002, pvalue = 0.000, protein K48-linked ubiquitination\n",
      "Passed: cat 5005, mi = 0.002, pvalue = 0.000, Ran GTPase binding\n",
      "Passed: cat 1839, mi = 0.002, pvalue = 0.000, keratin filament\n",
      "Passed: cat 8336, mi = 0.002, pvalue = 0.000, acyl-CoA dehydrogenase activity\n",
      "Passed: cat 1533, mi = 0.001, pvalue = 0.002, transcription factor activity, RNA polymerase II distal enhancer sequence-specific binding\n",
      "Passed: cat 943, mi = 0.001, pvalue = 0.001, insulin secretion\n",
      "Passed: cat 57, mi = 0.001, pvalue = 0.001, mitochondrial ATP synthesis coupled proton transport\n",
      "Passed: cat 1514, mi = 0.001, pvalue = 0.002, voltage-gated potassium channel complex\n",
      "Passed: cat 1249, mi = 0.001, pvalue = 0.001, small-subunit processome\n",
      "Passed: cat 931, mi = 0.001, pvalue = 0.002, cell fate commitment\n",
      "Passed: cat 4766, mi = 0.001, pvalue = 0.002, bicarbonate transport\n",
      "Passed: cat 4483, mi = 0.001, pvalue = 0.001, gene silencing by miRNA\n",
      "Passed: cat 4131, mi = 0.001, pvalue = 0.002, transport vesicle membrane\n",
      "Passed: cat 839, mi = 0.001, pvalue = 0.003, thiol-dependent ubiquitinyl hydrolase activity\n",
      "Passed: cat 5666, mi = 0.001, pvalue = 0.000, establishment of integrated proviral latency\n",
      "Passed: cat 2703, mi = 0.001, pvalue = 0.002, iron-sulfur cluster assembly\n",
      "Passed: cat 2711, mi = 0.001, pvalue = 0.002, centriole replication\n",
      "Passed: cat 2958, mi = 0.001, pvalue = 0.000, mitochondrion transport along microtubule\n",
      "Passed: cat 1167, mi = 0.001, pvalue = 0.002, scavenger receptor activity\n",
      "Passed: cat 847, mi = 0.001, pvalue = 0.002, digestion\n",
      "Passed: cat 2506, mi = 0.001, pvalue = 0.000, enamel mineralization\n",
      "Passed: cat 527, mi = 0.001, pvalue = 0.001, phosphatidylethanolamine acyl-chain remodeling\n",
      "Didn't pass (cutoff=0.0050): cat 944, mi = 0.001, pvalue = 0.005, transcription elongation from RNA polymerase II promoter\n",
      "Didn't pass (cutoff=0.0050): cat 2311, mi = 0.001, pvalue = 0.006, transcription elongation from RNA polymerase I promoter\n",
      "Didn't pass (cutoff=0.0050): cat 4737, mi = 0.001, pvalue = 0.006, negative regulation of smooth muscle cell proliferation\n",
      "Didn't pass (cutoff=0.0050): cat 5468, mi = 0.001, pvalue = 0.008, mitophagy\n",
      "Didn't pass (cutoff=0.0050): cat 2027, mi = 0.001, pvalue = 0.009, thyroid hormone receptor binding\n",
      "Didn't pass (cutoff=0.0050): cat 4872, mi = 0.001, pvalue = 0.006, blood vessel morphogenesis\n",
      "Didn't pass (cutoff=0.0050): cat 2310, mi = 0.001, pvalue = 0.007, transcription initiation from RNA polymerase I promoter\n",
      "Didn't pass (cutoff=0.0050): cat 2312, mi = 0.001, pvalue = 0.009, termination of RNA polymerase I transcription\n",
      "Passed: cat 5781, mi = 0.001, pvalue = 0.002, mitotic recombination\n",
      "Didn't pass (cutoff=0.0050): cat 2046, mi = 0.001, pvalue = 0.009, SCF ubiquitin ligase complex\n",
      "Didn't pass (cutoff=0.0050): cat 4622, mi = 0.001, pvalue = 0.010, intraciliary transport involved in cilium assembly\n",
      "Didn't pass (cutoff=0.0050): cat 5469, mi = 0.001, pvalue = 0.010, nucleophagy\n",
      "Didn't pass (cutoff=0.0050): cat 3963, mi = 0.001, pvalue = 0.009, positive regulation of telomerase RNA localization to Cajal body\n",
      "Passed: cat 6662, mi = 0.001, pvalue = 0.000, myoblast fusion\n",
      "Didn't pass (cutoff=0.0050): cat 1934, mi = 0.001, pvalue = 0.014, regulation of cholesterol biosynthetic process\n",
      "Passed: cat 3524, mi = 0.001, pvalue = 0.003, positive regulation of endoplasmic reticulum stress-induced intrinsic apoptotic signaling pathway\n",
      "Didn't pass (cutoff=0.0050): cat 6916, mi = 0.001, pvalue = 0.013, biosynthetic process\n",
      "Didn't pass (cutoff=0.0050): cat 7019, mi = 0.001, pvalue = 0.012, RNA polymerase III activity\n",
      "Didn't pass (cutoff=0.0050): cat 1654, mi = 0.001, pvalue = 0.011, negative regulation of cell migration\n",
      "Didn't pass (cutoff=0.0050): cat 243, mi = 0.001, pvalue = 0.013, non-membrane spanning protein tyrosine kinase activity\n",
      "Didn't pass (cutoff=0.0050): cat 4942, mi = 0.001, pvalue = 0.005, ESCRT III complex disassembly\n",
      "Passed: cat 6738, mi = 0.001, pvalue = 0.005, VCP-NPL4-UFD1 AAA ATPase complex\n",
      "Didn't pass (cutoff=0.0050): cat 1904, mi = 0.001, pvalue = 0.012, cell communication\n",
      "Didn't pass (cutoff=0.0050): cat 6330, mi = 0.001, pvalue = 0.013, positive regulation of ubiquitin-protein transferase activity\n",
      "Passed: cat 2422, mi = 0.001, pvalue = 0.004, UDP-galactose:beta-N-acetylglucosamine beta-1,3-galactosyltransferase activity\n",
      "Passed: cat 3962, mi = 0.001, pvalue = 0.004, positive regulation of establishment of protein localization to telomere\n",
      "Didn't pass (cutoff=0.0050): cat 8115, mi = 0.001, pvalue = 0.016, cortical actin cytoskeleton organization\n",
      "Didn't pass (cutoff=0.0050): cat 7057, mi = 0.001, pvalue = 0.009, transforming growth factor beta binding\n",
      "Didn't pass (cutoff=0.0050): cat 4936, mi = 0.001, pvalue = 0.015, multivesicular body assembly\n",
      "Didn't pass (cutoff=0.0050): cat 250, mi = 0.001, pvalue = 0.018, peptidyl-tyrosine autophosphorylation\n",
      "Didn't pass (cutoff=0.0050): cat 2430, mi = 0.001, pvalue = 0.006, Golgi cisterna\n",
      "Didn't pass (cutoff=0.0050): cat 4056, mi = 0.001, pvalue = 0.006, positive regulation of autophagosome assembly\n",
      "Didn't pass (cutoff=0.0050): cat 8561, mi = 0.001, pvalue = 0.005, sterol transporter activity\n",
      "Didn't pass (cutoff=0.0050): cat 5180, mi = 0.001, pvalue = 0.010, positive regulation of protein ubiquitination involved in ubiquitin-dependent protein catabolic process\n",
      "Didn't pass (cutoff=0.0050): cat 484, mi = 0.001, pvalue = 0.013, negative regulation of smoothened signaling pathway\n",
      "Didn't pass (cutoff=0.0050): cat 1731, mi = 0.001, pvalue = 0.015, sensory perception of sound\n",
      "Passed: cat 7795, mi = 0.001, pvalue = 0.002, microtubule depolymerization\n",
      "Passed: cat 9215, mi = 0.001, pvalue = 0.004, platelet dense tubular network membrane\n",
      "Passed: cat 1830, mi = 0.001, pvalue = 0.004, glycerophospholipid metabolic process\n",
      "Didn't pass (cutoff=0.0050): cat 8703, mi = 0.001, pvalue = 0.006, sterol transport\n",
      "Didn't pass (cutoff=0.0050): cat 430, mi = 0.001, pvalue = 0.020, ciliary tip\n",
      "Didn't pass (cutoff=0.0050): cat 7020, mi = 0.001, pvalue = 0.021, DNA-directed RNA polymerase III complex\n",
      "Didn't pass (cutoff=0.0050): cat 1204, mi = 0.001, pvalue = 0.021, anchored component of membrane\n",
      "Didn't pass (cutoff=0.0050): cat 2313, mi = 0.001, pvalue = 0.015, DNA-directed RNA polymerase I complex\n",
      "Didn't pass (cutoff=0.0050): cat 6987, mi = 0.001, pvalue = 0.010, protein import into peroxisome matrix\n",
      "Didn't pass (cutoff=0.0050): cat 4972, mi = 0.001, pvalue = 0.023, positive regulation of dendrite morphogenesis\n",
      "Didn't pass (cutoff=0.0050): cat 9034, mi = 0.001, pvalue = 0.022, histone H4-K16 acetylation\n",
      "Didn't pass (cutoff=0.0050): cat 945, mi = 0.001, pvalue = 0.019, snRNA transcription from RNA polymerase II promoter\n",
      "Didn't pass (cutoff=0.0050): cat 10696, mi = 0.001, pvalue = 0.007, bleb assembly\n",
      "Didn't pass (cutoff=0.0050): cat 8275, mi = 0.001, pvalue = 0.009, negative regulation of axon regeneration\n",
      "Didn't pass (cutoff=0.0050): cat 1050, mi = 0.001, pvalue = 0.032, substrate adhesion-dependent cell spreading\n",
      "Didn't pass (cutoff=0.0050): cat 10198, mi = 0.001, pvalue = 0.019, nuclear export\n",
      "Didn't pass (cutoff=0.0050): cat 254, mi = 0.001, pvalue = 0.027, positive regulation of tyrosine phosphorylation of STAT protein\n",
      "Didn't pass (cutoff=0.0050): cat 4625, mi = 0.001, pvalue = 0.031, intraciliary transport\n",
      "Didn't pass (cutoff=0.0050): cat 4798, mi = 0.001, pvalue = 0.034, negative regulation of myeloid cell differentiation\n",
      "Didn't pass (cutoff=0.0050): cat 784, mi = 0.001, pvalue = 0.039, antigen processing and presentation of peptide antigen via MHC class I\n",
      "Didn't pass (cutoff=0.0050): cat 541, mi = 0.001, pvalue = 0.040, lysine-acetylated histone binding\n",
      "Didn't pass (cutoff=0.0050): cat 1976, mi = 0.001, pvalue = 0.036, PcG protein complex\n",
      "Didn't pass (cutoff=0.0050): cat 1054, mi = 0.001, pvalue = 0.026, leukocyte cell-cell adhesion\n",
      "Total number of categories evaluated = 411.\n",
      "Number of categories that passed the tests = 46\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM2_4.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM2_4.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 36, 38, 0.0114818845272013\n",
      "best merge is 1, 2, 0.0154630928706109\n",
      "best merge is 1, 14, 0.0236678891222665\n",
      "best merge is 5, 20, 0.0277184680558796\n",
      "best merge is 19, 39, 0.0289023456826394\n",
      "best merge is 1, 43, 0.0295649152443752\n",
      "best merge is 8, 16, 0.0346836119042708\n",
      "best merge is 1, 37, 0.0395193860517402\n",
      "best merge is 18, 40, 0.0472843444809505\n",
      "best merge is 5, 34, 0.0479019535565334\n",
      "best merge is 1, 5, 0.0540617505272285\n",
      "best merge is 26, 30, 0.0769408604790596\n",
      "best merge is 6, 11, 0.079051560486292\n",
      "best merge is 1, 22, 0.07961076734739\n",
      "best merge is 4, 10, 0.0809075697245372\n",
      "best merge is 21, 42, 0.0936804268702705\n",
      "best merge is 1, 17, 0.0984794347720586\n",
      "best merge is 7, 24, 0.101411812337971\n",
      "best merge is 19, 35, 0.109607384640818\n",
      "best merge is 18, 26, 0.110314194408699\n",
      "best merge is 31, 41, 0.114720241793752\n",
      "best merge is 1, 7, 0.120815387404715\n",
      "best merge is 3, 6, 0.130458506870461\n",
      "best merge is 21, 29, 0.163327698917331\n",
      "best merge is 25, 33, 0.174359846931564\n",
      "best merge is 0, 8, 0.185032161484902\n",
      "best merge is 13, 23, 0.185936840011947\n",
      "best merge is 3, 13, 0.205619383023074\n",
      "best merge is 0, 4, 0.206538036861202\n",
      "best merge is 18, 45, 0.24241615556928\n",
      "best merge is 9, 28, 0.259265797350603\n",
      "best merge is 1, 31, 0.265516086680478\n",
      "best merge is 19, 36, 0.298724568502696\n",
      "best merge is 12, 15, 0.324128545584079\n",
      "best merge is 0, 3, 0.355763184587935\n",
      "best merge is 32, 44, 0.386658934688177\n",
      "best merge is 25, 27, 0.461218260474916\n",
      "best merge is 0, 21, 0.535627250152289\n",
      "best merge is 1, 25, 0.612269679399524\n",
      "best merge is 9, 12, 0.686793705678832\n",
      "best merge is 0, 18, 0.764989959094357\n",
      "best merge is 9, 19, 0.833117227438448\n",
      "order classes\n",
      "18 - 1.26400983782357\n",
      "17 - 0.72252057929153\n",
      "4 - 4.78769317562189\n",
      "5 - 1.51023255813363\n",
      "2 - 4.44657774891404\n",
      "1 2 14 43 37 5 20 34 22 17 7 24 31 41 25 33 27\n",
      "0 8 16 4 10 3 6 11 13 23 21 42 29 18 40 26 30 45\n",
      "9 28 12 15\n",
      "19 39 35 36 38\n",
      "32 44\n",
      "Done.Start drawing\n",
      "100.30\t0.00\n",
      "bicarbonate transport, GO:0015701\n",
      "scavenger receptor activity, GO:0005044\n",
      "transport vesicle membrane, GO:0030658\n",
      "phosphatidylethanolamine acyl-chain remodeling, GO:0036152\n",
      "myoblast fusion, GO:0007520\n",
      "transcription factor activity, RNA polymerase II distal enhancer sequence-specific binding, GO:0003705\n",
      "positive regulation of endoplasmic reticulum stress-induced intrinsic apoptotic signaling pathway, GO:1902237\n",
      "enamel mineralization, GO:0070166\n",
      "negative regulation of autophagy, GO:0010507\n",
      "protein K48-linked ubiquitination, GO:0070936\n",
      "endocytosis, GO:0006897\n",
      "thiol-dependent ubiquitinyl hydrolase activity, GO:0036459\n",
      "platelet dense tubular network membrane, GO:0031095\n",
      "acyl-CoA dehydrogenase activity, GO:0003995\n",
      "VCP-NPL4-UFD1 AAA ATPase complex, GO:0034098\n",
      "gene silencing by miRNA, GO:0035195\n",
      "iron-sulfur cluster assembly, GO:0016226\n",
      "glycerophospholipid metabolic process, GO:0006650\n",
      "structural constituent of ribosome, GO:0003735\n",
      "hydrogen ion transmembrane transport, GO:1902600\n",
      "Ran GTPase binding, GO:0008536\n",
      "mitochondrial respiratory chain complex I assembly, GO:0032981\n",
      "ribosome binding, GO:0043022\n",
      "regulation of mRNA stability, GO:0043488\n",
      "tRNA processing, GO:0008033\n",
      "tricarboxylic acid cycle, GO:0006099\n",
      "small-subunit processome, GO:0032040\n",
      "mitochondrion transport along microtubule, GO:0047497\n",
      "spliceosomal complex, GO:0005681\n",
      "mitochondrial ATP synthesis coupled proton transport, GO:0042776\n",
      "positive regulation of establishment of protein localization to telomere, GO:1904851\n",
      "establishment of integrated proviral latency, GO:0075713\n",
      "DNA replication, GO:0006260\n",
      "chromosome, centromeric region, GO:0000775\n",
      "chromosome condensation, GO:0030261\n",
      "microtubule depolymerization, GO:0007019\n",
      "mitotic recombination, GO:0006312\n",
      "nucleosome, GO:0000786\n",
      "insulin secretion, GO:0030073\n",
      "digestion, GO:0007586\n",
      "voltage-gated potassium channel complex, GO:0008076\n",
      "keratin filament, GO:0045095\n",
      "positive chemotaxis, GO:0050918\n",
      "cell fate commitment, GO:0045165\n",
      "centriole replication, GO:0007099\n",
      "UDP-galactose:beta-N-acetylglucosamine beta-1,3-galactosyltransferase activity, GO:0008499\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM2_4.csv_PAGE/filt_FC_LVM2_4.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM2_4.csv_PAGE/filt_FC_LVM2_4.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM2_4.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM2_4.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 36, 38, 0.0114818845272013\n",
      "best merge is 1, 2, 0.0154630928706109\n",
      "best merge is 1, 14, 0.0236678891222665\n",
      "best merge is 5, 20, 0.0277184680558796\n",
      "best merge is 19, 39, 0.0289023456826394\n",
      "best merge is 1, 43, 0.0295649152443752\n",
      "best merge is 8, 16, 0.0346836119042708\n",
      "best merge is 1, 37, 0.0395193860517402\n",
      "best merge is 18, 40, 0.0472843444809505\n",
      "best merge is 5, 34, 0.0479019535565334\n",
      "best merge is 1, 5, 0.0540617505272285\n",
      "best merge is 26, 30, 0.0769408604790596\n",
      "best merge is 6, 11, 0.079051560486292\n",
      "best merge is 1, 22, 0.07961076734739\n",
      "best merge is 4, 10, 0.0809075697245372\n",
      "best merge is 21, 42, 0.0936804268702705\n",
      "best merge is 1, 17, 0.0984794347720586\n",
      "best merge is 7, 24, 0.101411812337971\n",
      "best merge is 19, 35, 0.109607384640818\n",
      "best merge is 18, 26, 0.110314194408699\n",
      "best merge is 31, 41, 0.114720241793752\n",
      "best merge is 1, 7, 0.120815387404715\n",
      "best merge is 3, 6, 0.130458506870461\n",
      "best merge is 21, 29, 0.163327698917331\n",
      "best merge is 25, 33, 0.174359846931564\n",
      "best merge is 0, 8, 0.185032161484902\n",
      "best merge is 13, 23, 0.185936840011947\n",
      "best merge is 3, 13, 0.205619383023074\n",
      "best merge is 0, 4, 0.206538036861202\n",
      "best merge is 18, 45, 0.24241615556928\n",
      "best merge is 9, 28, 0.259265797350603\n",
      "best merge is 1, 31, 0.265516086680478\n",
      "best merge is 19, 36, 0.298724568502696\n",
      "best merge is 12, 15, 0.324128545584079\n",
      "best merge is 0, 3, 0.355763184587935\n",
      "best merge is 32, 44, 0.386658934688177\n",
      "best merge is 25, 27, 0.461218260474916\n",
      "best merge is 0, 21, 0.535627250152289\n",
      "best merge is 1, 25, 0.612269679399524\n",
      "best merge is 9, 12, 0.686793705678832\n",
      "best merge is 0, 18, 0.764989959094357\n",
      "best merge is 9, 19, 0.833117227438448\n",
      "order classes\n",
      "18 - 1.26400983782357\n",
      "17 - 0.72252057929153\n",
      "4 - 4.78769317562189\n",
      "5 - 1.51023255813363\n",
      "2 - 4.44657774891404\n",
      "1 2 14 43 37 5 20 34 22 17 7 24 31 41 25 33 27\n",
      "0 8 16 4 10 3 6 11 13 23 21 42 29 18 40 26 30 45\n",
      "9 28 12 15\n",
      "19 39 35 36 38\n",
      "32 44\n",
      "Done.-19.8953140578265\t42\t0\n",
      "-19.7466600199402\t42\t0.209371884346959\n",
      "-19.6754735792622\t42\t0.297308075772682\n",
      "-19.6231306081755\t42\t0.351744765702891\n",
      "-19.5728813559322\t42\t0.401994017946162\n",
      "-19.5163509471585\t42\t0.452243270189432\n",
      "-19.4430707876371\t42\t0.515054835493519\n",
      "-19.3279162512463\t42\t0.598803589232303\n",
      "1.37268195413758\t42\t0.745363908275175\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM2_4.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM2_3.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl/human_ensembl.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM2_3.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 13290... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 17689\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 13290 out of 13290 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 13290\t17689\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1477 hits go_terms_num = 17689\n",
      "cat 692, mi = 0.01027657, structural constituent of ribosome\n",
      "cat 836, mi = 0.00897460, translational initiation\n",
      "cat 2546, mi = 0.00837438, SRP-dependent cotranslational protein targeting to membrane\n",
      "cat 1498, mi = 0.00648310, viral transcription\n",
      "cat 2545, mi = 0.00632543, nuclear-transcribed mRNA catabolic process, nonsense-mediated decay\n",
      "cat 2990, mi = 0.00593695, chromosome, centromeric region\n",
      "cat 1872, mi = 0.00583618, spliceosomal complex\n",
      "cat 2016, mi = 0.00568106, cytosolic large ribosomal subunit\n",
      "cat 7, mi = 0.00520141, respiratory chain\n",
      "cat 2121, mi = 0.00517224, sister chromatid cohesion\n",
      "cat 2585, mi = 0.00468187, kinetochore\n",
      "cat 4583, mi = 0.00433640, condensed chromosome kinetochore\n",
      "cat 693, mi = 0.00400408, mitochondrial translational elongation\n",
      "cat 394, mi = 0.00393490, chromosome segregation\n",
      "cat 1491, mi = 0.00384897, mRNA export from nucleus\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 692, mi = 0.010, pvalue = 0.000, structural constituent of ribosome\n",
      "Passed: cat 2990, mi = 0.006, pvalue = 0.000, chromosome, centromeric region\n",
      "Passed: cat 1872, mi = 0.006, pvalue = 0.000, spliceosomal complex\n",
      "Passed: cat 7, mi = 0.005, pvalue = 0.000, respiratory chain\n",
      "Passed: cat 2431, mi = 0.003, pvalue = 0.000, nucleosome\n",
      "Passed: cat 1256, mi = 0.003, pvalue = 0.000, tRNA processing\n",
      "Passed: cat 1064, mi = 0.002, pvalue = 0.000, proteasome complex\n",
      "Passed: cat 2732, mi = 0.002, pvalue = 0.000, protein ubiquitination involved in ubiquitin-dependent protein catabolic process\n",
      "Passed: cat 2759, mi = 0.002, pvalue = 0.000, cornification\n",
      "Passed: cat 1242, mi = 0.002, pvalue = 0.000, ER to Golgi vesicle-mediated transport\n",
      "Passed: cat 944, mi = 0.002, pvalue = 0.000, transcription elongation from RNA polymerase II promoter\n",
      "Passed: cat 1204, mi = 0.002, pvalue = 0.000, anchored component of membrane\n",
      "Passed: cat 4131, mi = 0.002, pvalue = 0.000, transport vesicle membrane\n",
      "Passed: cat 2291, mi = 0.002, pvalue = 0.000, potassium ion transmembrane transport\n",
      "Passed: cat 835, mi = 0.002, pvalue = 0.000, translation initiation factor activity\n",
      "Passed: cat 5165, mi = 0.002, pvalue = 0.000, mitotic cytokinesis\n",
      "Passed: cat 550, mi = 0.002, pvalue = 0.000, chromatin remodeling\n",
      "Passed: cat 1688, mi = 0.002, pvalue = 0.000, positive regulation of endothelial cell proliferation\n",
      "Passed: cat 967, mi = 0.002, pvalue = 0.000, fatty acid beta-oxidation\n",
      "Passed: cat 56, mi = 0.002, pvalue = 0.000, cristae formation\n",
      "Passed: cat 3226, mi = 0.002, pvalue = 0.000, gene silencing by RNA\n",
      "Passed: cat 4092, mi = 0.002, pvalue = 0.001, cellular response to glucose starvation\n",
      "Passed: cat 1164, mi = 0.002, pvalue = 0.000, retromer complex\n",
      "Passed: cat 2101, mi = 0.002, pvalue = 0.000, centriole\n",
      "Passed: cat 847, mi = 0.002, pvalue = 0.000, digestion\n",
      "Passed: cat 1110, mi = 0.002, pvalue = 0.001, excitatory postsynaptic potential\n",
      "Passed: cat 1249, mi = 0.002, pvalue = 0.001, small-subunit processome\n",
      "Passed: cat 4784, mi = 0.001, pvalue = 0.001, cell death\n",
      "Passed: cat 3653, mi = 0.001, pvalue = 0.001, proton-transporting ATPase activity, rotational mechanism\n",
      "Passed: cat 1860, mi = 0.001, pvalue = 0.001, glutamate secretion\n",
      "Passed: cat 217, mi = 0.001, pvalue = 0.001, dioxygenase activity\n",
      "Passed: cat 328, mi = 0.001, pvalue = 0.001, cell redox homeostasis\n",
      "Passed: cat 3783, mi = 0.001, pvalue = 0.001, fibroblast growth factor receptor binding\n",
      "Passed: cat 6383, mi = 0.001, pvalue = 0.000, protein desumoylation\n",
      "Passed: cat 2023, mi = 0.001, pvalue = 0.002, DNA replication\n",
      "Passed: cat 1709, mi = 0.001, pvalue = 0.002, B cell receptor signaling pathway\n",
      "Passed: cat 6415, mi = 0.001, pvalue = 0.001, chromosome condensation\n",
      "Passed: cat 7121, mi = 0.001, pvalue = 0.001, lamin binding\n",
      "Passed: cat 584, mi = 0.001, pvalue = 0.005, aminopeptidase activity\n",
      "Passed: cat 2006, mi = 0.001, pvalue = 0.004, lamellipodium\n",
      "Didn't pass (cutoff=0.0050): cat 3660, mi = 0.001, pvalue = 0.005, chloride channel activity\n",
      "Passed: cat 3533, mi = 0.001, pvalue = 0.003, cilium movement\n",
      "Didn't pass (cutoff=0.0050): cat 3993, mi = 0.001, pvalue = 0.007, ligand-dependent nuclear receptor transcription coactivator activity\n",
      "Passed: cat 6700, mi = 0.001, pvalue = 0.002, positive regulation of mitochondrial outer membrane permeabilization involved in apoptotic signaling pathway\n",
      "Didn't pass (cutoff=0.0050): cat 3181, mi = 0.001, pvalue = 0.010, pyridoxal phosphate binding\n",
      "Passed: cat 136, mi = 0.001, pvalue = 0.004, gap junction\n",
      "Didn't pass (cutoff=0.0050): cat 2251, mi = 0.001, pvalue = 0.010, promoter-specific chromatin binding\n",
      "Didn't pass (cutoff=0.0050): cat 2448, mi = 0.001, pvalue = 0.006, histone methyltransferase activity (H3-K4 specific)\n",
      "Didn't pass (cutoff=0.0050): cat 3551, mi = 0.001, pvalue = 0.007, retrograde protein transport, ER to cytosol\n",
      "Didn't pass (cutoff=0.0050): cat 4469, mi = 0.001, pvalue = 0.012, Rho GTPase binding\n",
      "Didn't pass (cutoff=0.0050): cat 2131, mi = 0.001, pvalue = 0.010, regulation of mitotic cell cycle\n",
      "Passed: cat 2653, mi = 0.001, pvalue = 0.001, negative regulation of smoothened signaling pathway involved in dorsal/ventral neural tube patterning\n",
      "Didn't pass (cutoff=0.0050): cat 724, mi = 0.001, pvalue = 0.014, RNA polymerase II activating transcription factor binding\n",
      "Didn't pass (cutoff=0.0050): cat 2446, mi = 0.001, pvalue = 0.013, histone H3-K4 methylation\n",
      "Passed: cat 3423, mi = 0.001, pvalue = 0.004, telomere maintenance via recombination\n",
      "Didn't pass (cutoff=0.0050): cat 10900, mi = 0.001, pvalue = 0.006, porphyrin-containing compound biosynthetic process\n",
      "Didn't pass (cutoff=0.0050): cat 3048, mi = 0.001, pvalue = 0.007, aminoacyl-tRNA synthetase multienzyme complex\n",
      "Didn't pass (cutoff=0.0050): cat 3843, mi = 0.001, pvalue = 0.018, sphingolipid metabolic process\n",
      "Didn't pass (cutoff=0.0050): cat 5005, mi = 0.001, pvalue = 0.017, Ran GTPase binding\n",
      "Didn't pass (cutoff=0.0050): cat 6078, mi = 0.001, pvalue = 0.015, canonical glycolysis\n",
      "Didn't pass (cutoff=0.0050): cat 5144, mi = 0.001, pvalue = 0.009, oxygen binding\n",
      "Didn't pass (cutoff=0.0050): cat 2509, mi = 0.001, pvalue = 0.014, snRNA processing\n",
      "Passed: cat 1543, mi = 0.001, pvalue = 0.002, complement activation\n",
      "Didn't pass (cutoff=0.0050): cat 2810, mi = 0.001, pvalue = 0.006, positive regulation of insulin-like growth factor receptor signaling pathway\n",
      "Didn't pass (cutoff=0.0050): cat 237, mi = 0.001, pvalue = 0.024, xenobiotic metabolic process\n",
      "Didn't pass (cutoff=0.0050): cat 665, mi = 0.001, pvalue = 0.013, transcriptional repressor activity, RNA polymerase II core promoter proximal region sequence-specific binding\n",
      "Didn't pass (cutoff=0.0050): cat 2452, mi = 0.001, pvalue = 0.022, glycolytic process\n",
      "Didn't pass (cutoff=0.0050): cat 6976, mi = 0.001, pvalue = 0.022, preribosome, large subunit precursor\n",
      "Didn't pass (cutoff=0.0050): cat 2869, mi = 0.001, pvalue = 0.019, nucleoside metabolic process\n",
      "Didn't pass (cutoff=0.0050): cat 1261, mi = 0.001, pvalue = 0.020, transcriptional activator activity, RNA polymerase II transcription regulatory region sequence-specific binding\n",
      "Didn't pass (cutoff=0.0050): cat 5197, mi = 0.001, pvalue = 0.032, NLS-bearing protein import into nucleus\n",
      "Didn't pass (cutoff=0.0050): cat 7017, mi = 0.001, pvalue = 0.007, BORC complex\n",
      "Didn't pass (cutoff=0.0050): cat 8180, mi = 0.001, pvalue = 0.021, protein-arginine N-methyltransferase activity\n",
      "Didn't pass (cutoff=0.0050): cat 5032, mi = 0.001, pvalue = 0.028, Golgi-associated vesicle membrane\n",
      "Didn't pass (cutoff=0.0050): cat 5572, mi = 0.001, pvalue = 0.035, protein import into nucleus, translocation\n",
      "Passed: cat 2752, mi = 0.001, pvalue = 0.004, phosphoric ester hydrolase activity\n",
      "Didn't pass (cutoff=0.0050): cat 2793, mi = 0.001, pvalue = 0.037, voltage-gated calcium channel activity\n",
      "Didn't pass (cutoff=0.0050): cat 7604, mi = 0.001, pvalue = 0.020, transcription from mitochondrial promoter\n",
      "Didn't pass (cutoff=0.0050): cat 8181, mi = 0.001, pvalue = 0.024, peptidyl-arginine N-methylation\n",
      "Didn't pass (cutoff=0.0050): cat 2444, mi = 0.001, pvalue = 0.041, histone-lysine N-methyltransferase activity\n",
      "Passed: cat 2421, mi = 0.001, pvalue = 0.001, N-acetyllactosaminide beta-1,3-N-acetylglucosaminyltransferase activity\n",
      "Passed: cat 3749, mi = 0.001, pvalue = 0.002, N-acylphosphatidylethanolamine-specific phospholipase D activity\n",
      "Didn't pass (cutoff=0.0050): cat 2523, mi = 0.001, pvalue = 0.027, cell recognition\n",
      "Didn't pass (cutoff=0.0050): cat 7221, mi = 0.001, pvalue = 0.030, phosphatidylethanolamine biosynthetic process\n",
      "Didn't pass (cutoff=0.0050): cat 3321, mi = 0.001, pvalue = 0.035, positive regulation of activated T cell proliferation\n",
      "Didn't pass (cutoff=0.0050): cat 2394, mi = 0.001, pvalue = 0.043, positive regulation of Wnt signaling pathway\n",
      "Didn't pass (cutoff=0.0050): cat 6909, mi = 0.001, pvalue = 0.032, cholesterol efflux\n",
      "Didn't pass (cutoff=0.0050): cat 1977, mi = 0.001, pvalue = 0.033, PRC1 complex\n",
      "Didn't pass (cutoff=0.0050): cat 7932, mi = 0.001, pvalue = 0.034, mitochondrial genome maintenance\n",
      "Didn't pass (cutoff=0.0050): cat 2208, mi = 0.001, pvalue = 0.016, cholesterol transporter activity\n",
      "Didn't pass (cutoff=0.0050): cat 476, mi = 0.001, pvalue = 0.043, camera-type eye development\n",
      "Didn't pass (cutoff=0.0050): cat 4766, mi = 0.001, pvalue = 0.041, bicarbonate transport\n",
      "Didn't pass (cutoff=0.0050): cat 2700, mi = 0.001, pvalue = 0.042, kinesin binding\n",
      "Didn't pass (cutoff=0.0050): cat 2443, mi = 0.001, pvalue = 0.050, histone methyltransferase complex\n",
      "Passed: cat 5011, mi = 0.001, pvalue = 0.001, chemokine receptor activity\n",
      "Didn't pass (cutoff=0.0050): cat 4326, mi = 0.001, pvalue = 0.006, cardiac myofibril\n",
      "Didn't pass (cutoff=0.0050): cat 10784, mi = 0.001, pvalue = 0.006, oxygen transporter activity\n",
      "Didn't pass (cutoff=0.0050): cat 12039, mi = 0.001, pvalue = 0.007, AMP metabolic process\n",
      "Didn't pass (cutoff=0.0050): cat 4044, mi = 0.001, pvalue = 0.043, activation of NF-kappaB-inducing kinase activity\n",
      "Didn't pass (cutoff=0.0050): cat 4235, mi = 0.001, pvalue = 0.044, regulation of transforming growth factor beta receptor signaling pathway\n",
      "Didn't pass (cutoff=0.0050): cat 2449, mi = 0.001, pvalue = 0.036, Set1C/COMPASS complex\n",
      "Didn't pass (cutoff=0.0050): cat 1777, mi = 0.001, pvalue = 0.055, mitochondrial respiratory chain complex IV assembly\n",
      "Didn't pass (cutoff=0.0050): cat 4796, mi = 0.001, pvalue = 0.056, stress-activated protein kinase signaling cascade\n",
      "Didn't pass (cutoff=0.0050): cat 2794, mi = 0.001, pvalue = 0.034, voltage-gated calcium channel complex\n",
      "Didn't pass (cutoff=0.0050): cat 2863, mi = 0.001, pvalue = 0.066, purine nucleotide biosynthetic process\n",
      "Didn't pass (cutoff=0.0050): cat 951, mi = 0.001, pvalue = 0.059, collagen binding\n",
      "Didn't pass (cutoff=0.0050): cat 2495, mi = 0.001, pvalue = 0.076, dolichol-linked oligosaccharide biosynthetic process\n",
      "Didn't pass (cutoff=0.0050): cat 1579, mi = 0.001, pvalue = 0.037, nucleobase-containing compound kinase activity\n",
      "Didn't pass (cutoff=0.0050): cat 3962, mi = 0.001, pvalue = 0.040, positive regulation of establishment of protein localization to telomere\n",
      "Didn't pass (cutoff=0.0050): cat 3010, mi = 0.001, pvalue = 0.050, negative regulation of phosphoprotein phosphatase activity\n",
      "Didn't pass (cutoff=0.0050): cat 5071, mi = 0.001, pvalue = 0.054, regulation of heart contraction\n",
      "Didn't pass (cutoff=0.0050): cat 97, mi = 0.001, pvalue = 0.065, iron ion transport\n",
      "Didn't pass (cutoff=0.0050): cat 10800, mi = 0.001, pvalue = 0.009, positive regulation of ceramide biosynthetic process\n",
      "Didn't pass (cutoff=0.0050): cat 10940, mi = 0.001, pvalue = 0.008, SRP-dependent cotranslational protein targeting to membrane, translocation\n",
      "Didn't pass (cutoff=0.0050): cat 1035, mi = 0.001, pvalue = 0.019, hemoglobin complex\n",
      "Total number of categories evaluated = 784.\n",
      "Number of categories that passed the tests = 50\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM2_3.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM2_3.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 32, 45, 0.0102082321105719\n",
      "best merge is 33, 43, 0.0185001426195398\n",
      "best merge is 32, 47, 0.0200427733778665\n",
      "best merge is 4, 27, 0.0249981961957534\n",
      "best merge is 8, 40, 0.0251667956080327\n",
      "best merge is 11, 12, 0.0340223838032224\n",
      "best merge is 0, 26, 0.0459079549766338\n",
      "best merge is 4, 36, 0.0491597853261663\n",
      "best merge is 8, 32, 0.0589268813789742\n",
      "best merge is 37, 49, 0.0644435123027568\n",
      "best merge is 2, 9, 0.0669153474091928\n",
      "best merge is 0, 14, 0.0710599097438748\n",
      "best merge is 1, 7, 0.0729977009453524\n",
      "best merge is 4, 23, 0.0799686744284023\n",
      "best merge is 11, 24, 0.0807747171069564\n",
      "best merge is 8, 29, 0.0967836732403536\n",
      "best merge is 18, 48, 0.104902634161975\n",
      "best merge is 3, 19, 0.108110073754311\n",
      "best merge is 10, 34, 0.130566894597224\n",
      "best merge is 4, 37, 0.131069633608047\n",
      "best merge is 18, 28, 0.13378821869083\n",
      "best merge is 1, 15, 0.143141809358295\n",
      "best merge is 8, 11, 0.147291153432462\n",
      "best merge is 16, 22, 0.185019828286605\n",
      "best merge is 0, 18, 0.189894224413076\n",
      "best merge is 1, 4, 0.190834848045795\n",
      "best merge is 8, 25, 0.191537239539164\n",
      "best merge is 3, 6, 0.2038354095404\n",
      "best merge is 5, 31, 0.214442795501345\n",
      "best merge is 20, 44, 0.255406063510833\n",
      "best merge is 1, 17, 0.274105398572364\n",
      "best merge is 13, 35, 0.285943524821111\n",
      "best merge is 2, 10, 0.28736460110737\n",
      "best merge is 39, 41, 0.301500531083476\n",
      "best merge is 16, 20, 0.329292129191589\n",
      "best merge is 2, 5, 0.346433629687105\n",
      "best merge is 30, 46, 0.408374941500498\n",
      "best merge is 0, 21, 0.420340670473898\n",
      "best merge is 1, 42, 0.428901184336763\n",
      "best merge is 2, 3, 0.446287467512673\n",
      "best merge is 8, 13, 0.471011927682222\n",
      "best merge is 30, 39, 0.494572364571125\n",
      "best merge is 0, 2, 0.586945949011768\n",
      "best merge is 16, 33, 0.692743744708341\n",
      "best merge is 0, 38, 0.749635464943653\n",
      "best merge is 1, 30, 0.886387079172003\n",
      "order classes\n",
      "17 - 1.35650659277236\n",
      "11 - 0.903568248407632\n",
      "12 - 0.879037618233185\n",
      "6 - 3.38798356891207\n",
      "4 - 4.20025770448199\n",
      "8 40 32 45 47 29 11 12 24 25 13 35\n",
      "1 7 15 4 27 36 23 37 49 17 42\n",
      "0 26 14 18 48 28 21 2 9 10 34 5 31 3 19 6 38\n",
      "16 22 20 44 33 43\n",
      "30 46 39 41\n",
      "Done.Start drawing\n",
      "184.76\t0.00\n",
      "B cell receptor signaling pathway, GO:0050853\n",
      "chromosome, centromeric region, GO:0000775\n",
      "protein ubiquitination involved in ubiquitin-dependent protein catabolic process, GO:0042787\n",
      "mitotic cytokinesis, GO:0000281\n",
      "nucleosome, GO:0000786\n",
      "cell death, GO:0008219\n",
      "chromosome condensation, GO:0030261\n",
      "centriole, GO:0005814\n",
      "lamin binding, GO:0005521\n",
      "chemokine receptor activity, GO:0004950\n",
      "positive regulation of endothelial cell proliferation, GO:0001938\n",
      "gap junction, GO:0005921\n",
      "dioxygenase activity, GO:0051213\n",
      "lamellipodium, GO:0030027\n",
      "positive regulation of mitochondrial outer membrane permeabilization involved in apoptotic signaling pathway, GO:1901030\n",
      "ER to Golgi vesicle-mediated transport, GO:0006888\n",
      "chromatin remodeling, GO:0006338\n",
      "retromer complex, GO:0030904\n",
      "gene silencing by RNA, GO:0031047\n",
      "telomere maintenance via recombination, GO:0000722\n",
      "transcription elongation from RNA polymerase II promoter, GO:0006368\n",
      "protein desumoylation, GO:0016926\n",
      "negative regulation of smoothened signaling pathway involved in dorsal/ventral neural tube patterning, GO:1901621\n",
      "spliceosomal complex, GO:0005681\n",
      "cell redox homeostasis, GO:0045454\n",
      "respiratory chain, GO:0070469\n",
      "cristae formation, GO:0042407\n",
      "proteasome complex, GO:0000502\n",
      "phosphoric ester hydrolase activity, GO:0042578\n",
      "structural constituent of ribosome, GO:0003735\n",
      "small-subunit processome, GO:0032040\n",
      "translation initiation factor activity, GO:0003743\n",
      "fatty acid beta-oxidation, GO:0006635\n",
      "N-acylphosphatidylethanolamine-specific phospholipase D activity, GO:0070290\n",
      "proton-transporting ATPase activity, rotational mechanism, GO:0046961\n",
      "cellular response to glucose starvation, GO:0042149\n",
      "DNA replication, GO:0006260\n",
      "tRNA processing, GO:0008033\n",
      "aminopeptidase activity, GO:0004177\n",
      "cornification, GO:0070268\n",
      "cilium movement, GO:0003341\n",
      "fibroblast growth factor receptor binding, GO:0005104\n",
      "complement activation, GO:0006956\n",
      "N-acetyllactosaminide beta-1,3-N-acetylglucosaminyltransferase activity, GO:0008532\n",
      "glutamate secretion, GO:0014047\n",
      "anchored component of membrane, GO:0031225\n",
      "transport vesicle membrane, GO:0030658\n",
      "digestion, GO:0007586\n",
      "excitatory postsynaptic potential, GO:0060079\n",
      "potassium ion transmembrane transport, GO:0071805\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM2_3.csv_PAGE/filt_FC_LVM2_3.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM2_3.csv_PAGE/filt_FC_LVM2_3.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM2_3.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM2_3.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 32, 45, 0.0102082321105719\n",
      "best merge is 33, 43, 0.0185001426195398\n",
      "best merge is 32, 47, 0.0200427733778665\n",
      "best merge is 4, 27, 0.0249981961957534\n",
      "best merge is 8, 40, 0.0251667956080327\n",
      "best merge is 11, 12, 0.0340223838032224\n",
      "best merge is 0, 26, 0.0459079549766338\n",
      "best merge is 4, 36, 0.0491597853261663\n",
      "best merge is 8, 32, 0.0589268813789742\n",
      "best merge is 37, 49, 0.0644435123027568\n",
      "best merge is 2, 9, 0.0669153474091928\n",
      "best merge is 0, 14, 0.0710599097438748\n",
      "best merge is 1, 7, 0.0729977009453524\n",
      "best merge is 4, 23, 0.0799686744284023\n",
      "best merge is 11, 24, 0.0807747171069564\n",
      "best merge is 8, 29, 0.0967836732403536\n",
      "best merge is 18, 48, 0.104902634161975\n",
      "best merge is 3, 19, 0.108110073754311\n",
      "best merge is 10, 34, 0.130566894597224\n",
      "best merge is 4, 37, 0.131069633608047\n",
      "best merge is 18, 28, 0.13378821869083\n",
      "best merge is 1, 15, 0.143141809358295\n",
      "best merge is 8, 11, 0.147291153432462\n",
      "best merge is 16, 22, 0.185019828286605\n",
      "best merge is 0, 18, 0.189894224413076\n",
      "best merge is 1, 4, 0.190834848045795\n",
      "best merge is 8, 25, 0.191537239539164\n",
      "best merge is 3, 6, 0.2038354095404\n",
      "best merge is 5, 31, 0.214442795501345\n",
      "best merge is 20, 44, 0.255406063510833\n",
      "best merge is 1, 17, 0.274105398572364\n",
      "best merge is 13, 35, 0.285943524821111\n",
      "best merge is 2, 10, 0.28736460110737\n",
      "best merge is 39, 41, 0.301500531083476\n",
      "best merge is 16, 20, 0.329292129191589\n",
      "best merge is 2, 5, 0.346433629687105\n",
      "best merge is 30, 46, 0.408374941500498\n",
      "best merge is 0, 21, 0.420340670473898\n",
      "best merge is 1, 42, 0.428901184336763\n",
      "best merge is 2, 3, 0.446287467512673\n",
      "best merge is 8, 13, 0.471011927682222\n",
      "best merge is 30, 39, 0.494572364571125\n",
      "best merge is 0, 2, 0.586945949011768\n",
      "best merge is 16, 33, 0.692743744708341\n",
      "best merge is 0, 38, 0.749635464943653\n",
      "best merge is 1, 30, 0.886387079172003\n",
      "order classes\n",
      "17 - 1.35650659277236\n",
      "11 - 0.903568248407632\n",
      "12 - 0.879037618233185\n",
      "6 - 3.38798356891207\n",
      "4 - 4.20025770448199\n",
      "8 40 32 45 47 29 11 12 24 25 13 35\n",
      "1 7 15 4 27 36 23 37 49 17 42\n",
      "0 26 14 18 48 28 21 2 9 10 34 5 31 3 19 6 38\n",
      "16 22 20 44 33 43\n",
      "30 46 39 41\n",
      "Done.-19.9238471530634\t42\t0\n",
      "-19.8101861874865\t42\t0.152305693873133\n",
      "-19.7499458757307\t42\t0.227321931153929\n",
      "-19.7101645377787\t42\t0.272786317384715\n",
      "-19.6749296384499\t42\t0.306884607057805\n",
      "-19.6362849101537\t42\t0.343256116042433\n",
      "-19.5896839142672\t42\t0.384174063650141\n",
      "-19.5146676769864\t42\t0.436458107815545\n",
      "1.26710326910587\t42\t0.534206538211734\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM2_3.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM1_2.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl/human_ensembl.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM1_2.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 13290... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 17689\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 13290 out of 13290 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 13290\t17689\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1477 hits go_terms_num = 17689\n",
      "cat 836, mi = 0.01007429, translational initiation\n",
      "cat 692, mi = 0.00967462, structural constituent of ribosome\n",
      "cat 2546, mi = 0.00965863, SRP-dependent cotranslational protein targeting to membrane\n",
      "cat 2545, mi = 0.00752418, nuclear-transcribed mRNA catabolic process, nonsense-mediated decay\n",
      "cat 1498, mi = 0.00738788, viral transcription\n",
      "cat 2990, mi = 0.00619431, chromosome, centromeric region\n",
      "cat 2121, mi = 0.00592148, sister chromatid cohesion\n",
      "cat 2016, mi = 0.00589846, cytosolic large ribosomal subunit\n",
      "cat 2585, mi = 0.00478890, kinetochore\n",
      "cat 394, mi = 0.00423539, chromosome segregation\n",
      "cat 4583, mi = 0.00423019, condensed chromosome kinetochore\n",
      "cat 7, mi = 0.00421024, respiratory chain\n",
      "cat 1872, mi = 0.00404968, spliceosomal complex\n",
      "cat 4152, mi = 0.00374437, cytosolic small ribosomal subunit\n",
      "cat 1247, mi = 0.00372733, ribosome biogenesis\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 836, mi = 0.010, pvalue = 0.000, translational initiation\n",
      "Passed: cat 2990, mi = 0.006, pvalue = 0.000, chromosome, centromeric region\n",
      "Passed: cat 7, mi = 0.004, pvalue = 0.000, respiratory chain\n",
      "Passed: cat 1872, mi = 0.004, pvalue = 0.000, spliceosomal complex\n",
      "Passed: cat 1247, mi = 0.004, pvalue = 0.000, ribosome biogenesis\n",
      "Passed: cat 1274, mi = 0.003, pvalue = 0.000, G2/M transition of mitotic cell cycle\n",
      "Passed: cat 693, mi = 0.003, pvalue = 0.000, mitochondrial translational elongation\n",
      "Passed: cat 2431, mi = 0.003, pvalue = 0.000, nucleosome\n",
      "Passed: cat 944, mi = 0.002, pvalue = 0.000, transcription elongation from RNA polymerase II promoter\n",
      "Passed: cat 1064, mi = 0.002, pvalue = 0.000, proteasome complex\n",
      "Passed: cat 1119, mi = 0.002, pvalue = 0.000, tRNA aminoacylation for protein translation\n",
      "Passed: cat 103, mi = 0.002, pvalue = 0.000, growth factor activity\n",
      "Passed: cat 58, mi = 0.002, pvalue = 0.000, mitochondrial proton-transporting ATP synthase complex\n",
      "Passed: cat 1236, mi = 0.002, pvalue = 0.000, transcriptional repressor activity, RNA polymerase II transcription regulatory region sequence-specific binding\n",
      "Passed: cat 745, mi = 0.002, pvalue = 0.000, activation of MAPKK activity\n",
      "Passed: cat 2291, mi = 0.002, pvalue = 0.000, potassium ion transmembrane transport\n",
      "Passed: cat 1280, mi = 0.002, pvalue = 0.001, protein K48-linked ubiquitination\n",
      "Passed: cat 1253, mi = 0.002, pvalue = 0.000, heterotrimeric G-protein complex\n",
      "Passed: cat 5165, mi = 0.002, pvalue = 0.000, mitotic cytokinesis\n",
      "Passed: cat 2332, mi = 0.002, pvalue = 0.000, intra-Golgi vesicle-mediated transport\n",
      "Passed: cat 1218, mi = 0.002, pvalue = 0.001, regulation of macroautophagy\n",
      "Passed: cat 1449, mi = 0.002, pvalue = 0.000, lateral plasma membrane\n",
      "Passed: cat 5144, mi = 0.002, pvalue = 0.000, oxygen binding\n",
      "Passed: cat 3966, mi = 0.001, pvalue = 0.000, chaperonin-containing T-complex\n",
      "Passed: cat 6314, mi = 0.001, pvalue = 0.001, tRNA methylation\n",
      "Passed: cat 1050, mi = 0.001, pvalue = 0.002, substrate adhesion-dependent cell spreading\n",
      "Passed: cat 541, mi = 0.001, pvalue = 0.001, lysine-acetylated histone binding\n",
      "Passed: cat 328, mi = 0.001, pvalue = 0.001, cell redox homeostasis\n",
      "Passed: cat 766, mi = 0.001, pvalue = 0.000, calcium ion-regulated exocytosis of neurotransmitter\n",
      "Passed: cat 6415, mi = 0.001, pvalue = 0.001, chromosome condensation\n",
      "Passed: cat 2248, mi = 0.001, pvalue = 0.003, bHLH transcription factor binding\n",
      "Passed: cat 3064, mi = 0.001, pvalue = 0.004, cellular defense response\n",
      "Passed: cat 5850, mi = 0.001, pvalue = 0.005, ATPase activator activity\n",
      "Passed: cat 6078, mi = 0.001, pvalue = 0.003, canonical glycolysis\n",
      "Passed: cat 9804, mi = 0.001, pvalue = 0.002, protein neddylation\n",
      "Passed: cat 2229, mi = 0.001, pvalue = 0.002, regulation of neuronal synaptic plasticity\n",
      "Passed: cat 297, mi = 0.001, pvalue = 0.004, humoral immune response\n",
      "Didn't pass (cutoff=0.0050): cat 1404, mi = 0.001, pvalue = 0.006, sphingolipid biosynthetic process\n",
      "Passed: cat 5584, mi = 0.001, pvalue = 0.002, positive regulation of cilium assembly\n",
      "Passed: cat 1359, mi = 0.001, pvalue = 0.002, chemokine activity\n",
      "Didn't pass (cutoff=0.0050): cat 584, mi = 0.001, pvalue = 0.007, aminopeptidase activity\n",
      "Didn't pass (cutoff=0.0050): cat 847, mi = 0.001, pvalue = 0.005, digestion\n",
      "Didn't pass (cutoff=0.0050): cat 3188, mi = 0.001, pvalue = 0.009, branched-chain amino acid catabolic process\n",
      "Didn't pass (cutoff=0.0050): cat 3036, mi = 0.001, pvalue = 0.006, pseudouridine synthase activity\n",
      "Didn't pass (cutoff=0.0050): cat 3038, mi = 0.001, pvalue = 0.007, RNA modification\n",
      "Didn't pass (cutoff=0.0050): cat 1355, mi = 0.001, pvalue = 0.005, lipid particle organization\n",
      "Didn't pass (cutoff=0.0050): cat 2480, mi = 0.001, pvalue = 0.013, GPI anchor biosynthetic process\n",
      "Passed: cat 13636, mi = 0.001, pvalue = 0.001, beta-galactoside (CMP) alpha-2,3-sialyltransferase activity\n",
      "Didn't pass (cutoff=0.0050): cat 6105, mi = 0.001, pvalue = 0.010, NADPH binding\n",
      "Didn't pass (cutoff=0.0050): cat 1204, mi = 0.001, pvalue = 0.011, anchored component of membrane\n",
      "Didn't pass (cutoff=0.0050): cat 1130, mi = 0.001, pvalue = 0.013, estrogen receptor binding\n",
      "Didn't pass (cutoff=0.0050): cat 3755, mi = 0.001, pvalue = 0.011, mannosylation\n",
      "Didn't pass (cutoff=0.0050): cat 520, mi = 0.001, pvalue = 0.013, antimicrobial humoral response\n",
      "Didn't pass (cutoff=0.0050): cat 6464, mi = 0.001, pvalue = 0.016, K63-linked polyubiquitin binding\n",
      "Didn't pass (cutoff=0.0050): cat 967, mi = 0.001, pvalue = 0.017, fatty acid beta-oxidation\n",
      "Didn't pass (cutoff=0.0050): cat 4854, mi = 0.001, pvalue = 0.020, intrinsic apoptotic signaling pathway\n",
      "Didn't pass (cutoff=0.0050): cat 3533, mi = 0.001, pvalue = 0.010, cilium movement\n",
      "Didn't pass (cutoff=0.0050): cat 7913, mi = 0.001, pvalue = 0.013, negative regulation of endothelial cell migration\n",
      "Passed: cat 10696, mi = 0.001, pvalue = 0.003, bleb assembly\n",
      "Passed: cat 7151, mi = 0.001, pvalue = 0.004, anion transmembrane transporter activity\n",
      "Passed: cat 4729, mi = 0.001, pvalue = 0.004, protein autoprocessing\n",
      "Didn't pass (cutoff=0.0050): cat 8097, mi = 0.001, pvalue = 0.006, inositol phosphate dephosphorylation\n",
      "Didn't pass (cutoff=0.0050): cat 2214, mi = 0.001, pvalue = 0.023, response to ionizing radiation\n",
      "Passed: cat 4511, mi = 0.001, pvalue = 0.004, positive regulation of triglyceride biosynthetic process\n",
      "Didn't pass (cutoff=0.0050): cat 4639, mi = 0.001, pvalue = 0.025, vasculature development\n",
      "Didn't pass (cutoff=0.0050): cat 6381, mi = 0.001, pvalue = 0.022, filopodium assembly\n",
      "Didn't pass (cutoff=0.0050): cat 4928, mi = 0.001, pvalue = 0.022, clathrin coat assembly\n",
      "Didn't pass (cutoff=0.0050): cat 1839, mi = 0.001, pvalue = 0.030, keratin filament\n",
      "Didn't pass (cutoff=0.0050): cat 4525, mi = 0.001, pvalue = 0.020, skeletal muscle contraction\n",
      "Didn't pass (cutoff=0.0050): cat 7275, mi = 0.001, pvalue = 0.027, 2-oxoglutarate metabolic process\n",
      "Didn't pass (cutoff=0.0050): cat 1628, mi = 0.001, pvalue = 0.037, nucleotide metabolic process\n",
      "Didn't pass (cutoff=0.0050): cat 2522, mi = 0.001, pvalue = 0.031, heterophilic cell-cell adhesion via plasma membrane cell adhesion molecules\n",
      "Didn't pass (cutoff=0.0050): cat 1530, mi = 0.001, pvalue = 0.033, cellular response to ionizing radiation\n",
      "Didn't pass (cutoff=0.0050): cat 5083, mi = 0.001, pvalue = 0.038, NuA4 histone acetyltransferase complex\n",
      "Didn't pass (cutoff=0.0050): cat 4112, mi = 0.001, pvalue = 0.036, oxidoreductase activity, acting on the CH-CH group of donors\n",
      "Didn't pass (cutoff=0.0050): cat 6188, mi = 0.001, pvalue = 0.023, nuclear inclusion body\n",
      "Didn't pass (cutoff=0.0050): cat 10477, mi = 0.001, pvalue = 0.010, calcium-transporting ATPase activity\n",
      "Didn't pass (cutoff=0.0050): cat 10148, mi = 0.001, pvalue = 0.014, tRNA-splicing ligase complex\n",
      "Didn't pass (cutoff=0.0050): cat 8709, mi = 0.001, pvalue = 0.016, enoyl-CoA hydratase activity\n",
      "Didn't pass (cutoff=0.0050): cat 711, mi = 0.001, pvalue = 0.039, regulation of actin filament polymerization\n",
      "Didn't pass (cutoff=0.0050): cat 4226, mi = 0.001, pvalue = 0.034, RNA phosphodiester bond hydrolysis, endonucleolytic\n",
      "Didn't pass (cutoff=0.0050): cat 3426, mi = 0.001, pvalue = 0.015, male meiosis I\n",
      "Passed: cat 2891, mi = 0.001, pvalue = 0.002, ATP-dependent peptidase activity\n",
      "Didn't pass (cutoff=0.0050): cat 356, mi = 0.001, pvalue = 0.048, cellular response to dexamethasone stimulus\n",
      "Didn't pass (cutoff=0.0050): cat 5032, mi = 0.001, pvalue = 0.041, Golgi-associated vesicle membrane\n",
      "Didn't pass (cutoff=0.0050): cat 8715, mi = 0.001, pvalue = 0.047, Sin3 complex\n",
      "Didn't pass (cutoff=0.0050): cat 586, mi = 0.001, pvalue = 0.041, metalloaminopeptidase activity\n",
      "Didn't pass (cutoff=0.0050): cat 1040, mi = 0.001, pvalue = 0.012, phospholipid dephosphorylation\n",
      "Didn't pass (cutoff=0.0050): cat 4735, mi = 0.001, pvalue = 0.014, cGMP biosynthetic process\n",
      "Didn't pass (cutoff=0.0050): cat 2631, mi = 0.001, pvalue = 0.042, mannosyltransferase activity\n",
      "Didn't pass (cutoff=0.0050): cat 1124, mi = 0.001, pvalue = 0.046, sulfotransferase activity\n",
      "Didn't pass (cutoff=0.0050): cat 7387, mi = 0.001, pvalue = 0.028, protein deneddylation\n",
      "Didn't pass (cutoff=0.0050): cat 1433, mi = 0.001, pvalue = 0.033, protein localization to Golgi apparatus\n",
      "Didn't pass (cutoff=0.0050): cat 2975, mi = 0.001, pvalue = 0.045, clathrin adaptor complex\n",
      "Didn't pass (cutoff=0.0050): cat 6703, mi = 0.001, pvalue = 0.044, methyl-CpG binding\n",
      "Didn't pass (cutoff=0.0050): cat 5468, mi = 0.001, pvalue = 0.048, mitophagy\n",
      "Didn't pass (cutoff=0.0050): cat 3181, mi = 0.001, pvalue = 0.051, pyridoxal phosphate binding\n",
      "Didn't pass (cutoff=0.0050): cat 3584, mi = 0.001, pvalue = 0.032, basal part of cell\n",
      "Didn't pass (cutoff=0.0050): cat 5082, mi = 0.001, pvalue = 0.053, histone H2A acetylation\n",
      "Didn't pass (cutoff=0.0050): cat 4370, mi = 0.001, pvalue = 0.020, positive regulation by host of viral genome replication\n",
      "Didn't pass (cutoff=0.0050): cat 11735, mi = 0.001, pvalue = 0.021, UDP-N-acetylglucosamine biosynthetic process\n",
      "Didn't pass (cutoff=0.0050): cat 2598, mi = 0.001, pvalue = 0.021, acrosome assembly\n",
      "Didn't pass (cutoff=0.0050): cat 2980, mi = 0.001, pvalue = 0.026, endosome to melanosome transport\n",
      "Total number of categories evaluated = 583.\n",
      "Number of categories that passed the tests = 45\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM1_2.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM1_2.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 1, 5, 0.0241821713874512\n",
      "best merge is 34, 44, 0.0259284642658515\n",
      "best merge is 11, 22, 0.0287561349129096\n",
      "best merge is 3, 8, 0.0332657337476269\n",
      "best merge is 37, 39, 0.0399532325672559\n",
      "best merge is 11, 38, 0.047279586733445\n",
      "best merge is 15, 36, 0.058110275130059\n",
      "best merge is 7, 29, 0.0591299694040878\n",
      "best merge is 2, 12, 0.0701080350772391\n",
      "best merge is 14, 37, 0.0708074068190298\n",
      "best merge is 4, 6, 0.0927278650681466\n",
      "best merge is 1, 7, 0.10286834427697\n",
      "best merge is 18, 43, 0.113432408112677\n",
      "best merge is 23, 24, 0.114523357710201\n",
      "best merge is 9, 20, 0.132647944760765\n",
      "best merge is 19, 23, 0.161015626139022\n",
      "best merge is 2, 4, 0.164281400170624\n",
      "best merge is 11, 35, 0.181105607894887\n",
      "best merge is 11, 15, 0.192927492643403\n",
      "best merge is 1, 18, 0.196066931917371\n",
      "best merge is 11, 21, 0.213128208284619\n",
      "best merge is 0, 2, 0.221179478893128\n",
      "best merge is 3, 9, 0.222651156502212\n",
      "best merge is 33, 34, 0.256482886647162\n",
      "best merge is 0, 27, 0.273577857964275\n",
      "best merge is 16, 25, 0.287489951243281\n",
      "best merge is 11, 28, 0.287551797754661\n",
      "best merge is 1, 40, 0.293080149466346\n",
      "best merge is 14, 17, 0.314405978635227\n",
      "best merge is 0, 10, 0.329862625020596\n",
      "best merge is 32, 33, 0.338062289914848\n",
      "best merge is 19, 41, 0.341206016490227\n",
      "best merge is 26, 30, 0.354311721384622\n",
      "best merge is 1, 13, 0.359235331921763\n",
      "best merge is 0, 42, 0.378316299059134\n",
      "best merge is 16, 26, 0.426055231867137\n",
      "best merge is 0, 3, 0.545877612620525\n",
      "best merge is 14, 31, 0.556223932766205\n",
      "best merge is 0, 19, 0.700897738728821\n",
      "best merge is 1, 16, 0.763606879310282\n",
      "best merge is 0, 32, 0.828965058911211\n",
      "order classes\n",
      "16 - 1.3475352079814\n",
      "12 - 1.7307981426889\n",
      "8 - 0.977307571113062\n",
      "5 - 2.41177083348659\n",
      "4 - 2.18402030714726\n",
      "11 22 38 35 15 36 21 28\n",
      "1 5 7 29 18 43 40 13 16 25 26 30\n",
      "0 2 12 4 6 27 10 42 3 8 9 20 19 23 24 41\n",
      "14 37 39 17 31\n",
      "32 33 34 44\n",
      "Done.Start drawing\n",
      "62.96\t0.00\n",
      "chromosome, centromeric region, GO:0000775\n",
      "G2/M transition of mitotic cell cycle, GO:0000086\n",
      "nucleosome, GO:0000786\n",
      "chromosome condensation, GO:0030261\n",
      "mitotic cytokinesis, GO:0000281\n",
      "positive regulation of triglyceride biosynthetic process, GO:0010867\n",
      "bleb assembly, GO:0032060\n",
      "transcriptional repressor activity, RNA polymerase II transcription regulatory region sequence-specific binding, GO:0001227\n",
      "bHLH transcription factor binding, GO:0043425\n",
      "activation of MAPKK activity, GO:0000186\n",
      "positive regulation of cilium assembly, GO:0045724\n",
      "beta-galactoside (CMP) alpha-2,3-sialyltransferase activity, GO:0003836\n",
      "heterotrimeric G-protein complex, GO:0005834\n",
      "cellular defense response, GO:0006968\n",
      "protein K48-linked ubiquitination, GO:0070936\n",
      "substrate adhesion-dependent cell spreading, GO:0034446\n",
      "regulation of macroautophagy, GO:0016241\n",
      "lysine-acetylated histone binding, GO:0070577\n",
      "transcription elongation from RNA polymerase II promoter, GO:0006368\n",
      "spliceosomal complex, GO:0005681\n",
      "ATPase activator activity, GO:0001671\n",
      "canonical glycolysis, GO:0061621\n",
      "protein neddylation, GO:0045116\n",
      "ATP-dependent peptidase activity, GO:0004176\n",
      "proteasome complex, GO:0000502\n",
      "intra-Golgi vesicle-mediated transport, GO:0006891\n",
      "chaperonin-containing T-complex, GO:0005832\n",
      "tRNA methylation, GO:0030488\n",
      "anion transmembrane transporter activity, GO:0008509\n",
      "translational initiation, GO:0006413\n",
      "respiratory chain, GO:0070469\n",
      "mitochondrial proton-transporting ATP synthase complex, GO:0005753\n",
      "ribosome biogenesis, GO:0042254\n",
      "mitochondrial translational elongation, GO:0070125\n",
      "tRNA aminoacylation for protein translation, GO:0006418\n",
      "protein autoprocessing, GO:0016540\n",
      "cell redox homeostasis, GO:0045454\n",
      "growth factor activity, GO:0008083\n",
      "oxygen binding, GO:0019825\n",
      "chemokine activity, GO:0008009\n",
      "regulation of neuronal synaptic plasticity, GO:0048168\n",
      "potassium ion transmembrane transport, GO:0071805\n",
      "humoral immune response, GO:0006959\n",
      "lateral plasma membrane, GO:0016328\n",
      "calcium ion-regulated exocytosis of neurotransmitter, GO:0048791\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM1_2.csv_PAGE/filt_FC_LVM1_2.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM1_2.csv_PAGE/filt_FC_LVM1_2.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM1_2.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM1_2.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 1, 5, 0.0241821713874512\n",
      "best merge is 34, 44, 0.0259284642658515\n",
      "best merge is 11, 22, 0.0287561349129096\n",
      "best merge is 3, 8, 0.0332657337476269\n",
      "best merge is 37, 39, 0.0399532325672559\n",
      "best merge is 11, 38, 0.047279586733445\n",
      "best merge is 15, 36, 0.058110275130059\n",
      "best merge is 7, 29, 0.0591299694040878\n",
      "best merge is 2, 12, 0.0701080350772391\n",
      "best merge is 14, 37, 0.0708074068190298\n",
      "best merge is 4, 6, 0.0927278650681466\n",
      "best merge is 1, 7, 0.10286834427697\n",
      "best merge is 18, 43, 0.113432408112677\n",
      "best merge is 23, 24, 0.114523357710201\n",
      "best merge is 9, 20, 0.132647944760765\n",
      "best merge is 19, 23, 0.161015626139022\n",
      "best merge is 2, 4, 0.164281400170624\n",
      "best merge is 11, 35, 0.181105607894887\n",
      "best merge is 11, 15, 0.192927492643403\n",
      "best merge is 1, 18, 0.196066931917371\n",
      "best merge is 11, 21, 0.213128208284619\n",
      "best merge is 0, 2, 0.221179478893128\n",
      "best merge is 3, 9, 0.222651156502212\n",
      "best merge is 33, 34, 0.256482886647162\n",
      "best merge is 0, 27, 0.273577857964275\n",
      "best merge is 16, 25, 0.287489951243281\n",
      "best merge is 11, 28, 0.287551797754661\n",
      "best merge is 1, 40, 0.293080149466346\n",
      "best merge is 14, 17, 0.314405978635227\n",
      "best merge is 0, 10, 0.329862625020596\n",
      "best merge is 32, 33, 0.338062289914848\n",
      "best merge is 19, 41, 0.341206016490227\n",
      "best merge is 26, 30, 0.354311721384622\n",
      "best merge is 1, 13, 0.359235331921763\n",
      "best merge is 0, 42, 0.378316299059134\n",
      "best merge is 16, 26, 0.426055231867137\n",
      "best merge is 0, 3, 0.545877612620525\n",
      "best merge is 14, 31, 0.556223932766205\n",
      "best merge is 0, 19, 0.700897738728821\n",
      "best merge is 1, 16, 0.763606879310282\n",
      "best merge is 0, 32, 0.828965058911211\n",
      "order classes\n",
      "16 - 1.3475352079814\n",
      "12 - 1.7307981426889\n",
      "8 - 0.977307571113062\n",
      "5 - 2.41177083348659\n",
      "4 - 2.18402030714726\n",
      "11 22 38 35 15 36 21 28\n",
      "1 5 7 29 18 43 40 13 16 25 26 30\n",
      "0 2 12 4 6 27 10 42 3 8 9 20 19 23 24 41\n",
      "14 37 39 17 31\n",
      "32 33 34 44\n",
      "Done.-19.7331639135959\t42\t0\n",
      "-19.3762706480305\t42\t0.533672172808132\n",
      "-19.2328462515883\t42\t0.713786531130877\n",
      "-19.1327827191868\t42\t0.820520965692503\n",
      "-19.0427255400254\t42\t0.913913595933926\n",
      "-18.949332909784\t42\t1.00063532401525\n",
      "-18.8359275730623\t42\t1.10069885641677\n",
      "-18.6558132147395\t42\t1.2274459974587\n",
      "1.73046378653113\t42\t1.46092757306226\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM1_2.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM2_1.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl/human_ensembl.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM2_1.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 13290... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 17689\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 13290 out of 13290 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 13290\t17689\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1477 hits go_terms_num = 17689\n",
      "cat 2546, mi = 0.00758298, SRP-dependent cotranslational protein targeting to membrane\n",
      "cat 1498, mi = 0.00729400, viral transcription\n",
      "cat 836, mi = 0.00727822, translational initiation\n",
      "cat 692, mi = 0.00720361, structural constituent of ribosome\n",
      "cat 2545, mi = 0.00660981, nuclear-transcribed mRNA catabolic process, nonsense-mediated decay\n",
      "cat 2016, mi = 0.00566926, cytosolic large ribosomal subunit\n",
      "cat 1872, mi = 0.00517945, spliceosomal complex\n",
      "cat 7, mi = 0.00324729, respiratory chain\n",
      "cat 1876, mi = 0.00311250, catalytic step 2 spliceosome\n",
      "cat 1491, mi = 0.00305078, mRNA export from nucleus\n",
      "cat 694, mi = 0.00291836, mitochondrial translational termination\n",
      "cat 693, mi = 0.00288551, mitochondrial translational elongation\n",
      "cat 2424, mi = 0.00280691, mRNA transport\n",
      "cat 2431, mi = 0.00272770, nucleosome\n",
      "cat 6800, mi = 0.00263283, cytoplasmic translation\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 2546, mi = 0.008, pvalue = 0.000, SRP-dependent cotranslational protein targeting to membrane\n",
      "Passed: cat 1872, mi = 0.005, pvalue = 0.000, spliceosomal complex\n",
      "Passed: cat 7, mi = 0.003, pvalue = 0.000, respiratory chain\n",
      "Passed: cat 694, mi = 0.003, pvalue = 0.000, mitochondrial translational termination\n",
      "Passed: cat 2431, mi = 0.003, pvalue = 0.000, nucleosome\n",
      "Passed: cat 1074, mi = 0.002, pvalue = 0.000, NIK/NF-kappaB signaling\n",
      "Passed: cat 2759, mi = 0.002, pvalue = 0.000, cornification\n",
      "Passed: cat 944, mi = 0.002, pvalue = 0.000, transcription elongation from RNA polymerase II promoter\n",
      "Passed: cat 682, mi = 0.002, pvalue = 0.000, regulation of translation\n",
      "Passed: cat 104, mi = 0.002, pvalue = 0.000, hormone activity\n",
      "Passed: cat 5114, mi = 0.002, pvalue = 0.000, spindle organization\n",
      "Passed: cat 855, mi = 0.002, pvalue = 0.000, chemotaxis\n",
      "Passed: cat 2291, mi = 0.002, pvalue = 0.000, potassium ion transmembrane transport\n",
      "Passed: cat 1924, mi = 0.002, pvalue = 0.000, phosphatidylserine binding\n",
      "Passed: cat 5627, mi = 0.002, pvalue = 0.000, synaptic vesicle exocytosis\n",
      "Passed: cat 1488, mi = 0.002, pvalue = 0.000, nuclear pore\n",
      "Passed: cat 839, mi = 0.002, pvalue = 0.001, thiol-dependent ubiquitinyl hydrolase activity\n",
      "Passed: cat 931, mi = 0.002, pvalue = 0.001, cell fate commitment\n",
      "Passed: cat 2171, mi = 0.002, pvalue = 0.001, IRE1-mediated unfolded protein response\n",
      "Passed: cat 3521, mi = 0.002, pvalue = 0.001, actin filament organization\n",
      "Passed: cat 6347, mi = 0.001, pvalue = 0.001, regulation of reactive oxygen species metabolic process\n",
      "Passed: cat 2340, mi = 0.001, pvalue = 0.001, learning\n",
      "Passed: cat 1904, mi = 0.001, pvalue = 0.001, cell communication\n",
      "Passed: cat 1132, mi = 0.001, pvalue = 0.002, substantia nigra development\n",
      "Passed: cat 1883, mi = 0.001, pvalue = 0.001, mitochondrial calcium ion transmembrane transport\n",
      "Passed: cat 410, mi = 0.001, pvalue = 0.001, glycosaminoglycan metabolic process\n",
      "Passed: cat 546, mi = 0.001, pvalue = 0.002, double-strand break repair\n",
      "Passed: cat 825, mi = 0.001, pvalue = 0.003, GDP binding\n",
      "Passed: cat 6662, mi = 0.001, pvalue = 0.000, myoblast fusion\n",
      "Passed: cat 1164, mi = 0.001, pvalue = 0.002, retromer complex\n",
      "Passed: cat 400, mi = 0.001, pvalue = 0.003, cilium assembly\n",
      "Passed: cat 9817, mi = 0.001, pvalue = 0.000, negative regulation of organ growth\n",
      "Passed: cat 237, mi = 0.001, pvalue = 0.004, xenobiotic metabolic process\n",
      "Passed: cat 6078, mi = 0.001, pvalue = 0.004, canonical glycolysis\n",
      "Passed: cat 1089, mi = 0.001, pvalue = 0.004, methyltransferase activity\n",
      "Didn't pass (cutoff=0.0050): cat 3660, mi = 0.001, pvalue = 0.009, chloride channel activity\n",
      "Didn't pass (cutoff=0.0050): cat 882, mi = 0.001, pvalue = 0.006, phosphoprotein phosphatase activity\n",
      "Didn't pass (cutoff=0.0050): cat 143, mi = 0.001, pvalue = 0.011, negative regulation of inflammatory response\n",
      "Didn't pass (cutoff=0.0050): cat 2642, mi = 0.001, pvalue = 0.011, DNA replication initiation\n",
      "Didn't pass (cutoff=0.0050): cat 5849, mi = 0.001, pvalue = 0.012, positive regulation of ATPase activity\n",
      "Didn't pass (cutoff=0.0050): cat 3745, mi = 0.001, pvalue = 0.008, positive regulation of peptidyl-threonine phosphorylation\n",
      "Passed: cat 6113, mi = 0.001, pvalue = 0.001, detection of chemical stimulus involved in sensory perception of bitter taste\n",
      "Passed: cat 4989, mi = 0.001, pvalue = 0.003, HAUS complex\n",
      "Passed: cat 2467, mi = 0.001, pvalue = 0.003, sialyltransferase activity\n",
      "Didn't pass (cutoff=0.0050): cat 3790, mi = 0.001, pvalue = 0.013, carboxypeptidase activity\n",
      "Didn't pass (cutoff=0.0050): cat 1417, mi = 0.001, pvalue = 0.009, activation of adenylate cyclase activity\n",
      "Didn't pass (cutoff=0.0050): cat 881, mi = 0.001, pvalue = 0.011, protein dephosphorylation\n",
      "Didn't pass (cutoff=0.0050): cat 2004, mi = 0.001, pvalue = 0.009, dephosphorylation\n",
      "Didn't pass (cutoff=0.0050): cat 4525, mi = 0.001, pvalue = 0.008, skeletal muscle contraction\n",
      "Didn't pass (cutoff=0.0050): cat 2829, mi = 0.001, pvalue = 0.016, telomere maintenance via semi-conservative replication\n",
      "Didn't pass (cutoff=0.0050): cat 4936, mi = 0.001, pvalue = 0.020, multivesicular body assembly\n",
      "Didn't pass (cutoff=0.0050): cat 1864, mi = 0.001, pvalue = 0.020, ATP-dependent chromatin remodeling\n",
      "Didn't pass (cutoff=0.0050): cat 2863, mi = 0.001, pvalue = 0.018, purine nucleotide biosynthetic process\n",
      "Didn't pass (cutoff=0.0050): cat 12013, mi = 0.001, pvalue = 0.008, low-density lipoprotein particle receptor catabolic process\n",
      "Didn't pass (cutoff=0.0050): cat 1240, mi = 0.001, pvalue = 0.021, collagen trimer\n",
      "Didn't pass (cutoff=0.0050): cat 4784, mi = 0.001, pvalue = 0.028, cell death\n",
      "Didn't pass (cutoff=0.0050): cat 3886, mi = 0.001, pvalue = 0.024, regulation of cell morphogenesis\n",
      "Didn't pass (cutoff=0.0050): cat 5678, mi = 0.001, pvalue = 0.007, lens morphogenesis in camera-type eye\n",
      "Didn't pass (cutoff=0.0050): cat 4045, mi = 0.001, pvalue = 0.029, extrinsic apoptotic signaling pathway via death domain receptors\n",
      "Didn't pass (cutoff=0.0050): cat 5606, mi = 0.001, pvalue = 0.013, endolysosome membrane\n",
      "Didn't pass (cutoff=0.0050): cat 11622, mi = 0.001, pvalue = 0.014, cullin-RING ubiquitin ligase complex\n",
      "Passed: cat 7627, mi = 0.001, pvalue = 0.001, negative regulation of protein localization to plasma membrane\n",
      "Didn't pass (cutoff=0.0050): cat 1504, mi = 0.001, pvalue = 0.034, N-acetyltransferase activity\n",
      "Passed: cat 12135, mi = 0.001, pvalue = 0.003, mitochondrial acetyl-CoA biosynthetic process from pyruvate\n",
      "Didn't pass (cutoff=0.0050): cat 1119, mi = 0.001, pvalue = 0.026, tRNA aminoacylation for protein translation\n",
      "Didn't pass (cutoff=0.0050): cat 1249, mi = 0.001, pvalue = 0.026, small-subunit processome\n",
      "Didn't pass (cutoff=0.0050): cat 6074, mi = 0.001, pvalue = 0.022, negative regulation of growth\n",
      "Didn't pass (cutoff=0.0050): cat 5902, mi = 0.001, pvalue = 0.006, guanyl-nucleotide exchange factor complex\n",
      "Didn't pass (cutoff=0.0050): cat 1118, mi = 0.001, pvalue = 0.029, aminoacyl-tRNA ligase activity\n",
      "Didn't pass (cutoff=0.0050): cat 4855, mi = 0.001, pvalue = 0.016, peptidyl-threonine dephosphorylation\n",
      "Didn't pass (cutoff=0.0050): cat 5919, mi = 0.001, pvalue = 0.020, positive regulation of response to DNA damage stimulus\n",
      "Didn't pass (cutoff=0.0050): cat 520, mi = 0.001, pvalue = 0.026, antimicrobial humoral response\n",
      "Didn't pass (cutoff=0.0050): cat 680, mi = 0.001, pvalue = 0.019, high-density lipoprotein particle\n",
      "Didn't pass (cutoff=0.0050): cat 4804, mi = 0.001, pvalue = 0.031, cyclin-dependent protein kinase holoenzyme complex\n",
      "Didn't pass (cutoff=0.0050): cat 2316, mi = 0.001, pvalue = 0.035, flagellated sperm motility\n",
      "Didn't pass (cutoff=0.0050): cat 5396, mi = 0.001, pvalue = 0.038, mitochondrial fission\n",
      "Didn't pass (cutoff=0.0050): cat 5286, mi = 0.001, pvalue = 0.008, filamin binding\n",
      "Didn't pass (cutoff=0.0050): cat 7115, mi = 0.001, pvalue = 0.011, unsaturated fatty acid biosynthetic process\n",
      "Didn't pass (cutoff=0.0050): cat 6876, mi = 0.001, pvalue = 0.013, MCM complex\n",
      "Didn't pass (cutoff=0.0050): cat 1239, mi = 0.001, pvalue = 0.014, calcium-independent cell-cell adhesion via plasma membrane cell-adhesion molecules\n",
      "Didn't pass (cutoff=0.0050): cat 9333, mi = 0.001, pvalue = 0.021, ligase activity, forming aminoacyl-tRNA and related compounds\n",
      "Didn't pass (cutoff=0.0050): cat 7387, mi = 0.001, pvalue = 0.021, protein deneddylation\n",
      "Passed: cat 14624, mi = 0.001, pvalue = 0.001, condensed nuclear chromosome outer kinetochore\n",
      "Didn't pass (cutoff=0.0050): cat 1171, mi = 0.001, pvalue = 0.046, protein disulfide oxidoreductase activity\n",
      "Didn't pass (cutoff=0.0050): cat 2687, mi = 0.001, pvalue = 0.028, Golgi vesicle transport\n",
      "Didn't pass (cutoff=0.0050): cat 5445, mi = 0.001, pvalue = 0.041, preassembly of GPI anchor in ER membrane\n",
      "Didn't pass (cutoff=0.0050): cat 3479, mi = 0.001, pvalue = 0.043, phosphotyrosine residue binding\n",
      "Didn't pass (cutoff=0.0050): cat 4478, mi = 0.001, pvalue = 0.013, negative regulation of microtubule polymerization\n",
      "Didn't pass (cutoff=0.0050): cat 5446, mi = 0.001, pvalue = 0.015, glycosylphosphatidylinositol-N-acetylglucosaminyltransferase (GPI-GnT) complex\n",
      "Didn't pass (cutoff=0.0050): cat 7166, mi = 0.001, pvalue = 0.015, coenzyme biosynthetic process\n",
      "Didn't pass (cutoff=0.0050): cat 7795, mi = 0.001, pvalue = 0.015, microtubule depolymerization\n",
      "Didn't pass (cutoff=0.0050): cat 950, mi = 0.001, pvalue = 0.023, extracellular matrix structural constituent\n",
      "Didn't pass (cutoff=0.0050): cat 3533, mi = 0.001, pvalue = 0.032, cilium movement\n",
      "Didn't pass (cutoff=0.0050): cat 4549, mi = 0.001, pvalue = 0.049, carboxy-lyase activity\n",
      "Didn't pass (cutoff=0.0050): cat 3702, mi = 0.001, pvalue = 0.052, 1-acylglycerol-3-phosphate O-acyltransferase activity\n",
      "Passed: cat 8541, mi = 0.001, pvalue = 0.001, transforming growth factor beta-activated receptor activity\n",
      "Passed: cat 10784, mi = 0.001, pvalue = 0.002, oxygen transporter activity\n",
      "Passed: cat 7442, mi = 0.001, pvalue = 0.004, cell-cell adhesion via plasma-membrane adhesion molecules\n",
      "Didn't pass (cutoff=0.0050): cat 14649, mi = 0.001, pvalue = 0.006, cargo loading into vesicle\n",
      "Didn't pass (cutoff=0.0050): cat 1981, mi = 0.001, pvalue = 0.049, tumor necrosis factor-activated receptor activity\n",
      "Didn't pass (cutoff=0.0050): cat 9417, mi = 0.001, pvalue = 0.051, folic acid metabolic process\n",
      "Didn't pass (cutoff=0.0050): cat 7424, mi = 0.001, pvalue = 0.046, negative regulation of Rho protein signal transduction\n",
      "Didn't pass (cutoff=0.0050): cat 1853, mi = 0.001, pvalue = 0.053, amino acid transport\n",
      "Didn't pass (cutoff=0.0050): cat 1134, mi = 0.001, pvalue = 0.059, intracellular estrogen receptor signaling pathway\n",
      "Didn't pass (cutoff=0.0050): cat 10077, mi = 0.001, pvalue = 0.021, dolichyl diphosphate biosynthetic process\n",
      "Didn't pass (cutoff=0.0050): cat 2152, mi = 0.001, pvalue = 0.024, adenyl-nucleotide exchange factor activity\n",
      "Didn't pass (cutoff=0.0050): cat 1263, mi = 0.001, pvalue = 0.061, regulation of transcription involved in G1/S transition of mitotic cell cycle\n",
      "Didn't pass (cutoff=0.0050): cat 2037, mi = 0.001, pvalue = 0.062, Golgi stack\n",
      "Didn't pass (cutoff=0.0050): cat 1906, mi = 0.001, pvalue = 0.048, lipoprotein metabolic process\n",
      "Didn't pass (cutoff=0.0050): cat 8209, mi = 0.001, pvalue = 0.062, peroxisomal matrix\n",
      "Didn't pass (cutoff=0.0050): cat 5204, mi = 0.001, pvalue = 0.064, protein targeting to Golgi\n",
      "Didn't pass (cutoff=0.0050): cat 3661, mi = 0.001, pvalue = 0.056, chloride channel complex\n",
      "Didn't pass (cutoff=0.0050): cat 3963, mi = 0.001, pvalue = 0.064, positive regulation of telomerase RNA localization to Cajal body\n",
      "Didn't pass (cutoff=0.0050): cat 513, mi = 0.001, pvalue = 0.061, positive regulation of inflammatory response\n",
      "Didn't pass (cutoff=0.0050): cat 1852, mi = 0.001, pvalue = 0.068, amino acid transmembrane transport\n",
      "Didn't pass (cutoff=0.0050): cat 3411, mi = 0.001, pvalue = 0.005, semaphorin receptor complex\n",
      "Didn't pass (cutoff=0.0050): cat 7771, mi = 0.001, pvalue = 0.005, mitral valve morphogenesis\n",
      "Didn't pass (cutoff=0.0050): cat 271, mi = 0.001, pvalue = 0.043, positive regulation of chemokine production\n",
      "Total number of categories evaluated = 621.\n",
      "Number of categories that passed the tests = 44\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM2_1.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM2_1.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 35, 40, 0.00964813982846968\n",
      "best merge is 6, 43, 0.0133680362012217\n",
      "best merge is 35, 42, 0.0223829299517359\n",
      "best merge is 28, 41, 0.0252731281441059\n",
      "best merge is 6, 35, 0.0298323469330298\n",
      "best merge is 3, 8, 0.0458385355448687\n",
      "best merge is 6, 9, 0.0507942432464535\n",
      "best merge is 6, 14, 0.0702251264087581\n",
      "best merge is 1, 34, 0.0885458792773182\n",
      "best merge is 28, 38, 0.0905340664009972\n",
      "best merge is 21, 31, 0.125980783538065\n",
      "best merge is 6, 12, 0.126147549875273\n",
      "best merge is 3, 16, 0.127156543207959\n",
      "best merge is 1, 5, 0.139641125305533\n",
      "best merge is 2, 39, 0.145813210605831\n",
      "best merge is 17, 32, 0.146560452643423\n",
      "best merge is 6, 10, 0.14938842969644\n",
      "best merge is 28, 37, 0.169272013065491\n",
      "best merge is 6, 22, 0.176299325438505\n",
      "best merge is 3, 26, 0.193346007846585\n",
      "best merge is 6, 17, 0.23623203272437\n",
      "best merge is 18, 23, 0.274592396152666\n",
      "best merge is 0, 30, 0.276328375029394\n",
      "best merge is 19, 28, 0.278772325048763\n",
      "best merge is 27, 29, 0.281120755341209\n",
      "best merge is 3, 7, 0.294538341205937\n",
      "best merge is 1, 15, 0.303709432637404\n",
      "best merge is 4, 6, 0.328221735291812\n",
      "best merge is 13, 19, 0.340296664378518\n",
      "best merge is 1, 24, 0.341715909514691\n",
      "best merge is 25, 33, 0.347167978991811\n",
      "best merge is 21, 25, 0.400853959345268\n",
      "best merge is 1, 3, 0.421625633876175\n",
      "best merge is 11, 13, 0.45979539654912\n",
      "best merge is 0, 2, 0.534658488236175\n",
      "best merge is 21, 27, 0.578230866544992\n",
      "best merge is 18, 36, 0.608892880121264\n",
      "best merge is 0, 1, 0.657834070061959\n",
      "best merge is 0, 18, 0.750464987847936\n",
      "best merge is 20, 21, 0.794314783945782\n",
      "order classes\n",
      "17 - 1.38154178548618\n",
      "13 - 0.536603501587367\n",
      "7 - 1.61363324353993\n",
      "6 - 2.84514278384196\n",
      "4 6 43 35 40 42 9 14 12 10 22 17 32\n",
      "0 30 2 39 1 34 5 15 24 3 8 16 26 7 18 23 36\n",
      "11 13 19 28 41 38 37\n",
      "20\n",
      "21 31 25 33 27 29\n",
      "Done.Start drawing\n",
      "141.22\t0.00\n",
      "chemotaxis, GO:0006935\n",
      "phosphatidylserine binding, GO:0001786\n",
      "actin filament organization, GO:0007015\n",
      "myoblast fusion, GO:0007520\n",
      "transforming growth factor beta-activated receptor activity, GO:0005024\n",
      "negative regulation of protein localization to plasma membrane, GO:1903077\n",
      "sialyltransferase activity, GO:0008373\n",
      "learning, GO:0007612\n",
      "negative regulation of organ growth, GO:0046621\n",
      "glycosaminoglycan metabolic process, GO:0030203\n",
      "canonical glycolysis, GO:0061621\n",
      "retromer complex, GO:0030904\n",
      "IRE1-mediated unfolded protein response, GO:0036498\n",
      "HAUS complex, GO:0070652\n",
      "GDP binding, GO:0019003\n",
      "regulation of translation, GO:0006417\n",
      "transcription elongation from RNA polymerase II promoter, GO:0006368\n",
      "regulation of reactive oxygen species metabolic process, GO:2000377\n",
      "mitochondrial translational termination, GO:0070126\n",
      "spliceosomal complex, GO:0005681\n",
      "methyltransferase activity, GO:0008168\n",
      "NIK/NF-kappaB signaling, GO:0038061\n",
      "nuclear pore, GO:0005643\n",
      "mitochondrial calcium ion transmembrane transport, GO:0006851\n",
      "SRP-dependent cotranslational protein targeting to membrane, GO:0006614\n",
      "respiratory chain, GO:0070469\n",
      "mitochondrial acetyl-CoA biosynthetic process from pyruvate, GO:0061732\n",
      "thiol-dependent ubiquitinyl hydrolase activity, GO:0036459\n",
      "double-strand break repair, GO:0006302\n",
      "substantia nigra development, GO:0021762\n",
      "cilium assembly, GO:0060271\n",
      "nucleosome, GO:0000786\n",
      "cornification, GO:0070268\n",
      "cell-cell adhesion via plasma-membrane adhesion molecules, GO:0098742\n",
      "detection of chemical stimulus involved in sensory perception of bitter taste, GO:0001580\n",
      "condensed nuclear chromosome outer kinetochore, GO:0000942\n",
      "oxygen transporter activity, GO:0005344\n",
      "hormone activity, GO:0005179\n",
      "synaptic vesicle exocytosis, GO:0016079\n",
      "potassium ion transmembrane transport, GO:0071805\n",
      "spindle organization, GO:0007051\n",
      "cell communication, GO:0007154\n",
      "cell fate commitment, GO:0045165\n",
      "xenobiotic metabolic process, GO:0006805\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM2_1.csv_PAGE/filt_FC_LVM2_1.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM2_1.csv_PAGE/filt_FC_LVM2_1.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM2_1.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM2_1.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 35, 40, 0.00964813982846968\n",
      "best merge is 6, 43, 0.0133680362012217\n",
      "best merge is 35, 42, 0.0223829299517359\n",
      "best merge is 28, 41, 0.0252731281441059\n",
      "best merge is 6, 35, 0.0298323469330298\n",
      "best merge is 3, 8, 0.0458385355448687\n",
      "best merge is 6, 9, 0.0507942432464535\n",
      "best merge is 6, 14, 0.0702251264087581\n",
      "best merge is 1, 34, 0.0885458792773182\n",
      "best merge is 28, 38, 0.0905340664009972\n",
      "best merge is 21, 31, 0.125980783538065\n",
      "best merge is 6, 12, 0.126147549875273\n",
      "best merge is 3, 16, 0.127156543207959\n",
      "best merge is 1, 5, 0.139641125305533\n",
      "best merge is 2, 39, 0.145813210605831\n",
      "best merge is 17, 32, 0.146560452643423\n",
      "best merge is 6, 10, 0.14938842969644\n",
      "best merge is 28, 37, 0.169272013065491\n",
      "best merge is 6, 22, 0.176299325438505\n",
      "best merge is 3, 26, 0.193346007846585\n",
      "best merge is 6, 17, 0.23623203272437\n",
      "best merge is 18, 23, 0.274592396152666\n",
      "best merge is 0, 30, 0.276328375029394\n",
      "best merge is 19, 28, 0.278772325048763\n",
      "best merge is 27, 29, 0.281120755341209\n",
      "best merge is 3, 7, 0.294538341205937\n",
      "best merge is 1, 15, 0.303709432637404\n",
      "best merge is 4, 6, 0.328221735291812\n",
      "best merge is 13, 19, 0.340296664378518\n",
      "best merge is 1, 24, 0.341715909514691\n",
      "best merge is 25, 33, 0.347167978991811\n",
      "best merge is 21, 25, 0.400853959345268\n",
      "best merge is 1, 3, 0.421625633876175\n",
      "best merge is 11, 13, 0.45979539654912\n",
      "best merge is 0, 2, 0.534658488236175\n",
      "best merge is 21, 27, 0.578230866544992\n",
      "best merge is 18, 36, 0.608892880121264\n",
      "best merge is 0, 1, 0.657834070061959\n",
      "best merge is 0, 18, 0.750464987847936\n",
      "best merge is 20, 21, 0.794314783945782\n",
      "order classes\n",
      "17 - 1.38154178548618\n",
      "13 - 0.536603501587367\n",
      "7 - 1.61363324353993\n",
      "6 - 2.84514278384196\n",
      "4 6 43 35 40 42 9 14 12 10 22 17 32\n",
      "0 30 2 39 1 34 5 15 24 3 8 16 26 7 18 23 36\n",
      "11 13 19 28 41 38 37\n",
      "20\n",
      "21 31 25 33 27 29\n",
      "Done.-19.8973941368078\t42\t0\n",
      "-19.7605863192182\t42\t0.205211726384365\n",
      "-19.7070528253788\t42\t0.273615635179153\n",
      "-19.6713638294859\t42\t0.312278714063164\n",
      "-19.6386489165841\t42\t0.344993626965019\n",
      "-19.6044469621867\t42\t0.377708539866874\n",
      "-19.5642968418071\t42\t0.413397535759807\n",
      "-19.4973799745079\t42\t0.458008780625974\n",
      "1.27361563517915\t42\t0.547231270358306\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl/fc_values/filt_FC_LVM2_1.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "sem --wait done\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Error: /undefined in NaN\n",
      "Operand stack:\n",
      "   498.132\n",
      "Execution stack:\n",
      "   %interp_exit   .runexec2   --nostringval--   --nostringval--   --nostringval--   2   %stopped_push   --nostringval--   --nostringval--   --nostringval--   false   1   %stopped_push   1967   1   3   %oparray_pop   1966   1   3   %oparray_pop   --nostringval--   1950   1   3   %oparray_pop   1836   1   3   %oparray_pop   --nostringval--   %errorexec_pop   .runexec2   --nostringval--   --nostringval--   --nostringval--   2   %stopped_push   --nostringval--\n",
      "Dictionary stack:\n",
      "   --dict:1194/1684(ro)(G)--   --dict:0/20(G)--   --dict:90/200(L)--\n",
      "Current allocation mode is local\n",
      "Last OS error: No such file or directory\n",
      "Current file position is 6861\n",
      "GPL Ghostscript 9.18: Unrecoverable error, exit code 1\n",
      "Error: /undefined in NaN\n",
      "Operand stack:\n",
      "   498.132\n",
      "Execution stack:\n",
      "   %interp_exit   .runexec2   --nostringval--   --nostringval--   --nostringval--   2   %stopped_push   --nostringval--   --nostringval--   --nostringval--   false   1   %stopped_push   1967   1   3   %oparray_pop   1966   1   3   %oparray_pop   --nostringval--   1950   1   3   %oparray_pop   1836   1   3   %oparray_pop   --nostringval--   %errorexec_pop   .runexec2   --nostringval--   --nostringval--   --nostringval--   2   %stopped_push   --nostringval--\n",
      "Dictionary stack:\n",
      "   --dict:1194/1684(ro)(G)--   --dict:0/20(G)--   --dict:90/200(L)--\n",
      "Current allocation mode is local\n",
      "Last OS error: No such file or directory\n",
      "Current file position is 6921\n",
      "GPL Ghostscript 9.18: Unrecoverable error, exit code 1\n",
      "Error: /undefined in NaN\n",
      "Operand stack:\n",
      "   498.132\n",
      "Execution stack:\n",
      "   %interp_exit   .runexec2   --nostringval--   --nostringval--   --nostringval--   2   %stopped_push   --nostringval--   --nostringval--   --nostringval--   false   1   %stopped_push   1967   1   3   %oparray_pop   1966   1   3   %oparray_pop   --nostringval--   1950   1   3   %oparray_pop   1836   1   3   %oparray_pop   --nostringval--   %errorexec_pop   .runexec2   --nostringval--   --nostringval--   --nostringval--   2   %stopped_push   --nostringval--\n",
      "Dictionary stack:\n",
      "   --dict:1194/1684(ro)(G)--   --dict:0/20(G)--   --dict:90/200(L)--\n",
      "Current allocation mode is local\n",
      "Last OS error: No such file or directory\n",
      "Current file position is 6921\n",
      "GPL Ghostscript 9.18: Unrecoverable error, exit code 1\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "nohup sh run.sh iPAGE_JY.sh &"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
