{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# This is used for running data vs human_ensembl_encode_tf annotations located in: \n",
    "### /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_encode_tf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import pickle\n",
    "import seaborn as sns\n",
    "import scipy.stats\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### get index of file names to loop through "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "cd fc_values\n",
    "ls FC* > filelist.txt\n",
    "cd .."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### get ensemble genes represented in iPAGE database"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash \n",
    "\n",
    "cp /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_encode_tf/human_ensembl_encode_tf_index.txt human_ensembl_index.txt\n",
    "awk '{print $1}' human_ensembl_index.txt > ensembl_index.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### make sure that genes in file match the index "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "###################!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n",
    "###################REMEMBER TO SPECIFY INDEX USED IN THE IPAGE COMMAND!!!!!!!!\n",
    "###################!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n",
    "\n",
    "\n",
    "#####################\n",
    "#######read in data from logfc calculations\n",
    "#####################\n",
    "####define index file to loop through\n",
    "index_file = \"fc_values/filelist.txt\"\n",
    "####define matching ensg index\n",
    "GO_index_filename = \"ensembl_index.txt\"\n",
    "genes_with_GOs = []\n",
    "with open(GO_index_filename,'r') as rf:\n",
    "    for line in rf:\n",
    "        gene_name = line.split('\\n')[0]\n",
    "        genes_with_GOs.append(gene_name)\n",
    "genes_with_GOs_set = set(genes_with_GOs)\n",
    "####open file and loop through each line\n",
    "###\n",
    "iPAGE_sh = \"iPAGE_JY.sh\"\n",
    "cwd = os.getcwd()+\"/\"\n",
    "os.system(\"rm \"+iPAGE_sh)\n",
    "###\n",
    "with open(index_file,'r') as rf:\n",
    "    for line in rf:\n",
    "        #####do this to each file\n",
    "        ###read in file into pandas df\n",
    "        FC_file = \"fc_values/\"+line.split('\\n')[0]\n",
    "        FC_df = pd.read_csv(FC_file,sep=',',index_col=0)\n",
    "        FC_df.columns = ['log2FoldChange']\n",
    "        FC_df.index.names = ['Gene']\n",
    "        ###filter\n",
    "        FC_df = FC_df.loc[[x for x in FC_df.index if x in genes_with_GOs_set]]\n",
    "        ###save\n",
    "        save_file = \"fc_values/filt_\"+line.split('\\n')[0]\n",
    "        FC_df.to_csv(save_file,sep='\\t',header=True)\n",
    "        ###write final iPAGE command to an ipage .sh file\n",
    "        iPAGE_cmd = \"perl /nvme/bins/iPAGEv1.0/page.pl --expfile \"+cwd+save_file+\" --species=human_ensembl_encode_tf --exptype=continuous --max_p=0.1 --ebins=9 \\n\"\n",
    "        with open(iPAGE_sh, \"a\") as myfile:\n",
    "            myfile.write(iPAGE_cmd)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## then run the command from this folder:\n",
    "\n",
    "nohup sh run.sh iPAGE_JY.sh &"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM1_1.csv --species=human_ensembl_encode_tf --exptype=continuous --max_p=0.1 --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM1_2.csv --species=human_ensembl_encode_tf --exptype=continuous --max_p=0.1 --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM1_3.csv --species=human_ensembl_encode_tf --exptype=continuous --max_p=0.1 --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM1_4.csv --species=human_ensembl_encode_tf --exptype=continuous --max_p=0.1 --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM2_1.csv --species=human_ensembl_encode_tf --exptype=continuous --max_p=0.1 --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM2_2.csv --species=human_ensembl_encode_tf --exptype=continuous --max_p=0.1 --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM2_3.csv --species=human_ensembl_encode_tf --exptype=continuous --max_p=0.1 --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM2_4.csv --species=human_ensembl_encode_tf --exptype=continuous --max_p=0.1 --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM2_5.csv --species=human_ensembl_encode_tf --exptype=continuous --max_p=0.1 --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_P_LVM1_1.csv --species=human_ensembl_encode_tf --exptype=continuous --max_p=0.1 --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_P_LVM1_2.csv --species=human_ensembl_encode_tf --exptype=continuous --max_p=0.1 --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_P_LVM1_3.csv --species=human_ensembl_encode_tf --exptype=continuous --max_p=0.1 --ebins=9\n",
      "done\n",
      "perl /nvme/bins/iPAGEv1.0/page.pl --expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_P_LVM1_4.csv --species=human_ensembl_encode_tf --exptype=continuous --max_p=0.1 --ebins=9\n",
      "done\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM1_1.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_encode_tf.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_encode_tf/human_ensembl_encode_tf.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM1_1.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 11438... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 181\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 11438 out of 11438 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 11438\t181\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1271 hits go_terms_num = 181\n",
      "cat 12, mi = 0.00099702, BRF1_ChIP-seq_regulon\n",
      "cat 115, mi = 0.00075915, PPARGC1A_ChIP-seq_regulon\n",
      "cat 114, mi = 0.00073299, POU5F1_ChIP-seq_regulon\n",
      "cat 167, mi = 0.00059073, XRCC4_ChIP-seq_regulon\n",
      "cat 13, mi = 0.00048745, BRF2_ChIP-seq_regulon\n",
      "cat 9, mi = 0.00037502, BDP1_ChIP-seq_regulon\n",
      "cat 112, mi = 0.00027504, POLR3G_ChIP-seq_regulon\n",
      "cat 131, mi = 0.00019052, SMARCA4_ChIP-seq_regulon\n",
      "cat 0, mi = 0.00000000, ARID3A_ChIP-seq_regulon\n",
      "cat 1, mi = 0.00000000, ATF1_ChIP-seq_regulon\n",
      "cat 2, mi = 0.00000000, ATF2_ChIP-seq_regulon\n",
      "cat 3, mi = 0.00000000, ATF3_ChIP-seq_regulon\n",
      "cat 4, mi = 0.00000000, BACH1_ChIP-seq_regulon\n",
      "cat 5, mi = 0.00000000, BATF_ChIP-seq_regulon\n",
      "cat 6, mi = 0.00000000, BCL11A_ChIP-seq_regulon\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 12, mi = 0.001, pvalue = 0.074, BRF1_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.1000): cat 131, mi = 0.000, pvalue = 0.936, SMARCA4_ChIP-seq_regulon\n",
      "Didn't pass: cat 0, mi = 0.000, pvalue = 1.000, ARID3A_ChIP-seq_regulon\n",
      "Didn't pass: cat 1, mi = 0.000, pvalue = 1.000, ATF1_ChIP-seq_regulon\n",
      "Didn't pass: cat 2, mi = 0.000, pvalue = 1.000, ATF2_ChIP-seq_regulon\n",
      "Didn't pass: cat 3, mi = 0.000, pvalue = 1.000, ATF3_ChIP-seq_regulon\n",
      "Didn't pass: cat 4, mi = 0.000, pvalue = 1.000, BACH1_ChIP-seq_regulon\n",
      "Didn't pass: cat 5, mi = 0.000, pvalue = 1.000, BATF_ChIP-seq_regulon\n",
      "Didn't pass: cat 6, mi = 0.000, pvalue = 1.000, BCL11A_ChIP-seq_regulon\n",
      "Didn't pass: cat 7, mi = 0.000, pvalue = 1.000, BCL3_ChIP-seq_regulon\n",
      "Didn't pass: cat 8, mi = 0.000, pvalue = 1.000, BCLAF1_ChIP-seq_regulon\n",
      "Didn't pass: cat 10, mi = 0.000, pvalue = 1.000, BHLHE40_ChIP-seq_regulon\n",
      "Didn't pass: cat 11, mi = 0.000, pvalue = 1.000, BRCA1_ChIP-seq_regulon\n",
      "Didn't pass: cat 14, mi = 0.000, pvalue = 1.000, CBX2_ChIP-seq_regulon\n",
      "Didn't pass: cat 15, mi = 0.000, pvalue = 1.000, CBX3_ChIP-seq_regulon\n",
      "Didn't pass: cat 16, mi = 0.000, pvalue = 1.000, CBX8_ChIP-seq_regulon\n",
      "Didn't pass: cat 17, mi = 0.000, pvalue = 1.000, CCNT2_ChIP-seq_regulon\n",
      "Didn't pass: cat 18, mi = 0.000, pvalue = 1.000, CEBPB_ChIP-seq_regulon\n",
      "Didn't pass: cat 19, mi = 0.000, pvalue = 1.000, CEBPD_ChIP-seq_regulon\n",
      "Didn't pass: cat 20, mi = 0.000, pvalue = 1.000, CEBPZ_ChIP-seq_regulon\n",
      "Didn't pass: cat 21, mi = 0.000, pvalue = 1.000, CHD1_ChIP-seq_regulon\n",
      "Total number of categories evaluated = 17.\n",
      "Number of categories that passed the tests = 1\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM1_1.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM1_1.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "23.77\t0.00\n",
      "BRF1_ChIP-seq_regulon, BRF1\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM1_1.csv_PAGE/filt_FC_LVM1_1.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM1_1.csv_PAGE/filt_FC_LVM1_1.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM1_1.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM1_1.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-19.2843920908708\t42\t0\n",
      "-18.4274295330248\t42\t1.43121581825831\n",
      "-18.2065628944047\t42\t1.71392511569205\n",
      "-18.0475389145982\t42\t1.87294909549853\n",
      "-17.8885149347918\t42\t2.03197307530501\n",
      "-17.7206562894405\t42\t2.19099705511149\n",
      "-17.51745898191\t42\t2.36769036600757\n",
      "-17.199411022297\t42\t2.59739167017249\n",
      "2.50189314261674\t42\t3.00378628523349\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM1_1.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM1_3.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_encode_tf.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_encode_tf/human_ensembl_encode_tf.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM1_3.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 11438... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 181\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 11438 out of 11438 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 11438\t181\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1271 hits go_terms_num = 181\n",
      "cat 115, mi = 0.00107859, PPARGC1A_ChIP-seq_regulon\n",
      "cat 112, mi = 0.00080070, POLR3G_ChIP-seq_regulon\n",
      "cat 131, mi = 0.00078613, SMARCA4_ChIP-seq_regulon\n",
      "cat 13, mi = 0.00059631, BRF2_ChIP-seq_regulon\n",
      "cat 9, mi = 0.00050579, BDP1_ChIP-seq_regulon\n",
      "cat 114, mi = 0.00049400, POU5F1_ChIP-seq_regulon\n",
      "cat 12, mi = 0.00047787, BRF1_ChIP-seq_regulon\n",
      "cat 167, mi = 0.00041578, XRCC4_ChIP-seq_regulon\n",
      "cat 0, mi = 0.00000000, ARID3A_ChIP-seq_regulon\n",
      "cat 1, mi = 0.00000000, ATF1_ChIP-seq_regulon\n",
      "cat 2, mi = 0.00000000, ATF2_ChIP-seq_regulon\n",
      "cat 3, mi = 0.00000000, ATF3_ChIP-seq_regulon\n",
      "cat 4, mi = 0.00000000, BACH1_ChIP-seq_regulon\n",
      "cat 5, mi = 0.00000000, BATF_ChIP-seq_regulon\n",
      "cat 6, mi = 0.00000000, BCL11A_ChIP-seq_regulon\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 115, mi = 0.001, pvalue = 0.049, PPARGC1A_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.1000): cat 112, mi = 0.001, pvalue = 0.186, POLR3G_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.1000): cat 131, mi = 0.001, pvalue = 0.183, SMARCA4_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.1000): cat 9, mi = 0.001, pvalue = 0.451, BDP1_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.1000): cat 114, mi = 0.000, pvalue = 0.465, POU5F1_ChIP-seq_regulon\n",
      "Didn't pass: cat 0, mi = 0.000, pvalue = 1.000, ARID3A_ChIP-seq_regulon\n",
      "Didn't pass: cat 1, mi = 0.000, pvalue = 1.000, ATF1_ChIP-seq_regulon\n",
      "Didn't pass: cat 2, mi = 0.000, pvalue = 1.000, ATF2_ChIP-seq_regulon\n",
      "Didn't pass: cat 3, mi = 0.000, pvalue = 1.000, ATF3_ChIP-seq_regulon\n",
      "Didn't pass: cat 4, mi = 0.000, pvalue = 1.000, BACH1_ChIP-seq_regulon\n",
      "Didn't pass: cat 5, mi = 0.000, pvalue = 1.000, BATF_ChIP-seq_regulon\n",
      "Didn't pass: cat 6, mi = 0.000, pvalue = 1.000, BCL11A_ChIP-seq_regulon\n",
      "Didn't pass: cat 7, mi = 0.000, pvalue = 1.000, BCL3_ChIP-seq_regulon\n",
      "Didn't pass: cat 8, mi = 0.000, pvalue = 1.000, BCLAF1_ChIP-seq_regulon\n",
      "Didn't pass: cat 10, mi = 0.000, pvalue = 1.000, BHLHE40_ChIP-seq_regulon\n",
      "Didn't pass: cat 11, mi = 0.000, pvalue = 1.000, BRCA1_ChIP-seq_regulon\n",
      "Didn't pass: cat 14, mi = 0.000, pvalue = 1.000, CBX2_ChIP-seq_regulon\n",
      "Didn't pass: cat 15, mi = 0.000, pvalue = 1.000, CBX3_ChIP-seq_regulon\n",
      "Didn't pass: cat 16, mi = 0.000, pvalue = 1.000, CBX8_ChIP-seq_regulon\n",
      "Didn't pass: cat 17, mi = 0.000, pvalue = 1.000, CCNT2_ChIP-seq_regulon\n",
      "Didn't pass: cat 18, mi = 0.000, pvalue = 1.000, CEBPB_ChIP-seq_regulon\n",
      "Total number of categories evaluated = 14.\n",
      "Number of categories that passed the tests = 1\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM1_3.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM1_3.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "84.94\t0.00\n",
      "PPARGC1A_ChIP-seq_regulon, PPARGC1A\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM1_3.csv_PAGE/filt_FC_LVM1_3.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM1_3.csv_PAGE/filt_FC_LVM1_3.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM1_3.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM1_3.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-19.849187661879\t42\t0\n",
      "-19.6588179891688\t42\t0.301624676242053\n",
      "-19.5920649870497\t42\t0.380739345420297\n",
      "-19.5401459854015\t42\t0.435130680480339\n",
      "-19.4906993171651\t42\t0.484577348716741\n",
      "-19.4387803155168\t42\t0.534024016953143\n",
      "-19.3744996468095\t42\t0.588415352013186\n",
      "-19.2706616435131\t42\t0.662585354367789\n",
      "1.39804567930304\t42\t0.796091358606075\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM1_3.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM1_4.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_encode_tf.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_encode_tf/human_ensembl_encode_tf.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM1_4.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 11438... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 181\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 11438 out of 11438 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 11438\t181\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1271 hits go_terms_num = 181\n",
      "cat 115, mi = 0.00103157, PPARGC1A_ChIP-seq_regulon\n",
      "cat 9, mi = 0.00060267, BDP1_ChIP-seq_regulon\n",
      "cat 167, mi = 0.00059073, XRCC4_ChIP-seq_regulon\n",
      "cat 12, mi = 0.00052932, BRF1_ChIP-seq_regulon\n",
      "cat 112, mi = 0.00052542, POLR3G_ChIP-seq_regulon\n",
      "cat 13, mi = 0.00048745, BRF2_ChIP-seq_regulon\n",
      "cat 114, mi = 0.00048179, POU5F1_ChIP-seq_regulon\n",
      "cat 131, mi = 0.00043553, SMARCA4_ChIP-seq_regulon\n",
      "cat 0, mi = 0.00000000, ARID3A_ChIP-seq_regulon\n",
      "cat 1, mi = 0.00000000, ATF1_ChIP-seq_regulon\n",
      "cat 2, mi = 0.00000000, ATF2_ChIP-seq_regulon\n",
      "cat 3, mi = 0.00000000, ATF3_ChIP-seq_regulon\n",
      "cat 4, mi = 0.00000000, BACH1_ChIP-seq_regulon\n",
      "cat 5, mi = 0.00000000, BATF_ChIP-seq_regulon\n",
      "cat 6, mi = 0.00000000, BCL11A_ChIP-seq_regulon\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 115, mi = 0.001, pvalue = 0.071, PPARGC1A_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.1000): cat 9, mi = 0.001, pvalue = 0.315, BDP1_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.1000): cat 112, mi = 0.001, pvalue = 0.502, POLR3G_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.1000): cat 114, mi = 0.000, pvalue = 0.486, POU5F1_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.1000): cat 131, mi = 0.000, pvalue = 0.578, SMARCA4_ChIP-seq_regulon\n",
      "Didn't pass: cat 0, mi = 0.000, pvalue = 1.000, ARID3A_ChIP-seq_regulon\n",
      "Didn't pass: cat 1, mi = 0.000, pvalue = 1.000, ATF1_ChIP-seq_regulon\n",
      "Didn't pass: cat 2, mi = 0.000, pvalue = 1.000, ATF2_ChIP-seq_regulon\n",
      "Didn't pass: cat 3, mi = 0.000, pvalue = 1.000, ATF3_ChIP-seq_regulon\n",
      "Didn't pass: cat 4, mi = 0.000, pvalue = 1.000, BACH1_ChIP-seq_regulon\n",
      "Didn't pass: cat 5, mi = 0.000, pvalue = 1.000, BATF_ChIP-seq_regulon\n",
      "Didn't pass: cat 6, mi = 0.000, pvalue = 1.000, BCL11A_ChIP-seq_regulon\n",
      "Didn't pass: cat 7, mi = 0.000, pvalue = 1.000, BCL3_ChIP-seq_regulon\n",
      "Didn't pass: cat 8, mi = 0.000, pvalue = 1.000, BCLAF1_ChIP-seq_regulon\n",
      "Didn't pass: cat 10, mi = 0.000, pvalue = 1.000, BHLHE40_ChIP-seq_regulon\n",
      "Didn't pass: cat 11, mi = 0.000, pvalue = 1.000, BRCA1_ChIP-seq_regulon\n",
      "Didn't pass: cat 14, mi = 0.000, pvalue = 1.000, CBX2_ChIP-seq_regulon\n",
      "Didn't pass: cat 15, mi = 0.000, pvalue = 1.000, CBX3_ChIP-seq_regulon\n",
      "Didn't pass: cat 16, mi = 0.000, pvalue = 1.000, CBX8_ChIP-seq_regulon\n",
      "Didn't pass: cat 17, mi = 0.000, pvalue = 1.000, CCNT2_ChIP-seq_regulon\n",
      "Didn't pass: cat 18, mi = 0.000, pvalue = 1.000, CEBPB_ChIP-seq_regulon\n",
      "Total number of categories evaluated = 14.\n",
      "Number of categories that passed the tests = 1\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM1_4.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM1_4.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "26.31\t0.00\n",
      "PPARGC1A_ChIP-seq_regulon, PPARGC1A\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM1_4.csv_PAGE/filt_FC_LVM1_4.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM1_4.csv_PAGE/filt_FC_LVM1_4.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM1_4.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM1_4.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-19.521094640821\t42\t0\n",
      "-18.8905359179019\t42\t0.957810718358039\n",
      "-18.6431014823261\t42\t1.26111744583808\n",
      "-18.4675028506271\t42\t1.45267958950969\n",
      "-18.3078677309008\t42\t1.61231470923603\n",
      "-18.1482326111745\t42\t1.77194982896237\n",
      "-17.9726339794755\t42\t1.93158494868871\n",
      "-17.709236031927\t42\t2.12314709236032\n",
      "2.22919042189282\t42\t2.45838084378563\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM1_4.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM2_1.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_encode_tf.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_encode_tf/human_ensembl_encode_tf.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM2_1.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 11438... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 181\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 11438 out of 11438 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 11438\t181\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1271 hits go_terms_num = 181\n",
      "cat 12, mi = 0.00098849, BRF1_ChIP-seq_regulon\n",
      "cat 115, mi = 0.00096557, PPARGC1A_ChIP-seq_regulon\n",
      "cat 9, mi = 0.00079041, BDP1_ChIP-seq_regulon\n",
      "cat 112, mi = 0.00055905, POLR3G_ChIP-seq_regulon\n",
      "cat 114, mi = 0.00052770, POU5F1_ChIP-seq_regulon\n",
      "cat 13, mi = 0.00048745, BRF2_ChIP-seq_regulon\n",
      "cat 167, mi = 0.00041578, XRCC4_ChIP-seq_regulon\n",
      "cat 131, mi = 0.00032291, SMARCA4_ChIP-seq_regulon\n",
      "cat 0, mi = 0.00000000, ARID3A_ChIP-seq_regulon\n",
      "cat 1, mi = 0.00000000, ATF1_ChIP-seq_regulon\n",
      "cat 2, mi = 0.00000000, ATF2_ChIP-seq_regulon\n",
      "cat 3, mi = 0.00000000, ATF3_ChIP-seq_regulon\n",
      "cat 4, mi = 0.00000000, BACH1_ChIP-seq_regulon\n",
      "cat 5, mi = 0.00000000, BATF_ChIP-seq_regulon\n",
      "cat 6, mi = 0.00000000, BCL11A_ChIP-seq_regulon\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 12, mi = 0.001, pvalue = 0.079, BRF1_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.1000): cat 131, mi = 0.000, pvalue = 0.777, SMARCA4_ChIP-seq_regulon\n",
      "Didn't pass: cat 0, mi = 0.000, pvalue = 1.000, ARID3A_ChIP-seq_regulon\n",
      "Didn't pass: cat 1, mi = 0.000, pvalue = 1.000, ATF1_ChIP-seq_regulon\n",
      "Didn't pass: cat 2, mi = 0.000, pvalue = 1.000, ATF2_ChIP-seq_regulon\n",
      "Didn't pass: cat 3, mi = 0.000, pvalue = 1.000, ATF3_ChIP-seq_regulon\n",
      "Didn't pass: cat 4, mi = 0.000, pvalue = 1.000, BACH1_ChIP-seq_regulon\n",
      "Didn't pass: cat 5, mi = 0.000, pvalue = 1.000, BATF_ChIP-seq_regulon\n",
      "Didn't pass: cat 6, mi = 0.000, pvalue = 1.000, BCL11A_ChIP-seq_regulon\n",
      "Didn't pass: cat 7, mi = 0.000, pvalue = 1.000, BCL3_ChIP-seq_regulon\n",
      "Didn't pass: cat 8, mi = 0.000, pvalue = 1.000, BCLAF1_ChIP-seq_regulon\n",
      "Didn't pass: cat 10, mi = 0.000, pvalue = 1.000, BHLHE40_ChIP-seq_regulon\n",
      "Didn't pass: cat 11, mi = 0.000, pvalue = 1.000, BRCA1_ChIP-seq_regulon\n",
      "Didn't pass: cat 14, mi = 0.000, pvalue = 1.000, CBX2_ChIP-seq_regulon\n",
      "Didn't pass: cat 15, mi = 0.000, pvalue = 1.000, CBX3_ChIP-seq_regulon\n",
      "Didn't pass: cat 16, mi = 0.000, pvalue = 1.000, CBX8_ChIP-seq_regulon\n",
      "Didn't pass: cat 17, mi = 0.000, pvalue = 1.000, CCNT2_ChIP-seq_regulon\n",
      "Didn't pass: cat 18, mi = 0.000, pvalue = 1.000, CEBPB_ChIP-seq_regulon\n",
      "Didn't pass: cat 19, mi = 0.000, pvalue = 1.000, CEBPD_ChIP-seq_regulon\n",
      "Didn't pass: cat 20, mi = 0.000, pvalue = 1.000, CEBPZ_ChIP-seq_regulon\n",
      "Didn't pass: cat 21, mi = 0.000, pvalue = 1.000, CHD1_ChIP-seq_regulon\n",
      "Total number of categories evaluated = 17.\n",
      "Number of categories that passed the tests = 1\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM2_1.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM2_1.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "123.78\t0.00\n",
      "BRF1_ChIP-seq_regulon, BRF1\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM2_1.csv_PAGE/filt_FC_LVM2_1.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM2_1.csv_PAGE/filt_FC_LVM2_1.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM2_1.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM2_1.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-19.876151236064\t42\t0\n",
      "-19.7149781871062\t42\t0.247697527872031\n",
      "-19.6572952011634\t42\t0.322346097915657\n",
      "-19.6182743577315\t42\t0.363063499757635\n",
      "-19.5826466311197\t42\t0.400387784779447\n",
      "-19.5453223460979\t42\t0.434318952981095\n",
      "-19.5012118274358\t42\t0.475036354823073\n",
      "-19.4265632573921\t42\t0.522539990305381\n",
      "1.31216674745516\t42\t0.624333494910325\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM2_1.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM1_2.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_encode_tf.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_encode_tf/human_ensembl_encode_tf.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM1_2.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 11438... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 181\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 11438 out of 11438 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 11438\t181\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1271 hits go_terms_num = 181\n",
      "cat 115, mi = 0.00103137, PPARGC1A_ChIP-seq_regulon\n",
      "cat 114, mi = 0.00091804, POU5F1_ChIP-seq_regulon\n",
      "cat 13, mi = 0.00077146, BRF2_ChIP-seq_regulon\n",
      "cat 112, mi = 0.00064889, POLR3G_ChIP-seq_regulon\n",
      "cat 167, mi = 0.00059073, XRCC4_ChIP-seq_regulon\n",
      "cat 12, mi = 0.00057624, BRF1_ChIP-seq_regulon\n",
      "cat 9, mi = 0.00054792, BDP1_ChIP-seq_regulon\n",
      "cat 131, mi = 0.00035674, SMARCA4_ChIP-seq_regulon\n",
      "cat 0, mi = 0.00000000, ARID3A_ChIP-seq_regulon\n",
      "cat 1, mi = 0.00000000, ATF1_ChIP-seq_regulon\n",
      "cat 2, mi = 0.00000000, ATF2_ChIP-seq_regulon\n",
      "cat 3, mi = 0.00000000, ATF3_ChIP-seq_regulon\n",
      "cat 4, mi = 0.00000000, BACH1_ChIP-seq_regulon\n",
      "cat 5, mi = 0.00000000, BATF_ChIP-seq_regulon\n",
      "cat 6, mi = 0.00000000, BCL11A_ChIP-seq_regulon\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 115, mi = 0.001, pvalue = 0.068, PPARGC1A_ChIP-seq_regulon\n",
      "Passed: cat 114, mi = 0.001, pvalue = 0.075, POU5F1_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.1000): cat 131, mi = 0.000, pvalue = 0.733, SMARCA4_ChIP-seq_regulon\n",
      "Didn't pass: cat 0, mi = 0.000, pvalue = 1.000, ARID3A_ChIP-seq_regulon\n",
      "Didn't pass: cat 1, mi = 0.000, pvalue = 1.000, ATF1_ChIP-seq_regulon\n",
      "Didn't pass: cat 2, mi = 0.000, pvalue = 1.000, ATF2_ChIP-seq_regulon\n",
      "Didn't pass: cat 3, mi = 0.000, pvalue = 1.000, ATF3_ChIP-seq_regulon\n",
      "Didn't pass: cat 4, mi = 0.000, pvalue = 1.000, BACH1_ChIP-seq_regulon\n",
      "Didn't pass: cat 5, mi = 0.000, pvalue = 1.000, BATF_ChIP-seq_regulon\n",
      "Didn't pass: cat 6, mi = 0.000, pvalue = 1.000, BCL11A_ChIP-seq_regulon\n",
      "Didn't pass: cat 7, mi = 0.000, pvalue = 1.000, BCL3_ChIP-seq_regulon\n",
      "Didn't pass: cat 8, mi = 0.000, pvalue = 1.000, BCLAF1_ChIP-seq_regulon\n",
      "Didn't pass: cat 10, mi = 0.000, pvalue = 1.000, BHLHE40_ChIP-seq_regulon\n",
      "Didn't pass: cat 11, mi = 0.000, pvalue = 1.000, BRCA1_ChIP-seq_regulon\n",
      "Didn't pass: cat 14, mi = 0.000, pvalue = 1.000, CBX2_ChIP-seq_regulon\n",
      "Didn't pass: cat 15, mi = 0.000, pvalue = 1.000, CBX3_ChIP-seq_regulon\n",
      "Didn't pass: cat 16, mi = 0.000, pvalue = 1.000, CBX8_ChIP-seq_regulon\n",
      "Didn't pass: cat 17, mi = 0.000, pvalue = 1.000, CCNT2_ChIP-seq_regulon\n",
      "Didn't pass: cat 18, mi = 0.000, pvalue = 1.000, CEBPB_ChIP-seq_regulon\n",
      "Didn't pass: cat 19, mi = 0.000, pvalue = 1.000, CEBPD_ChIP-seq_regulon\n",
      "Didn't pass: cat 20, mi = 0.000, pvalue = 1.000, CEBPZ_ChIP-seq_regulon\n",
      "Didn't pass: cat 21, mi = 0.000, pvalue = 1.000, CHD1_ChIP-seq_regulon\n",
      "Total number of categories evaluated = 17.\n",
      "Number of categories that passed the tests = 2\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM1_2.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM1_2.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "62.96\t0.00\n",
      "POU5F1_ChIP-seq_regulon, POU5F1\n",
      "PPARGC1A_ChIP-seq_regulon, PPARGC1A\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM1_2.csv_PAGE/filt_FC_LVM1_2.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM1_2.csv_PAGE/filt_FC_LVM1_2.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM1_2.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM1_2.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-19.7164866581957\t42\t0\n",
      "-19.34625158831\t42\t0.56702668360864\n",
      "-19.209498094028\t42\t0.740470139771283\n",
      "-19.1161054637865\t42\t0.840533672172808\n",
      "-19.0293837357052\t42\t0.92725540025413\n",
      "-18.9393265565438\t42\t1.01397712833545\n",
      "-18.8292566709022\t42\t1.10736975857687\n",
      "-18.6524777636595\t42\t1.23411689961881\n",
      "1.73046378653113\t42\t1.46092757306226\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM1_2.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM2_3.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_encode_tf.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_encode_tf/human_ensembl_encode_tf.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM2_3.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 11438... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 181\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 11438 out of 11438 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 11438\t181\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1271 hits go_terms_num = 181\n",
      "cat 13, mi = 0.00129673, BRF2_ChIP-seq_regulon\n",
      "cat 115, mi = 0.00114419, PPARGC1A_ChIP-seq_regulon\n",
      "cat 114, mi = 0.00098227, POU5F1_ChIP-seq_regulon\n",
      "cat 9, mi = 0.00059004, BDP1_ChIP-seq_regulon\n",
      "cat 131, mi = 0.00043403, SMARCA4_ChIP-seq_regulon\n",
      "cat 167, mi = 0.00041578, XRCC4_ChIP-seq_regulon\n",
      "cat 112, mi = 0.00031820, POLR3G_ChIP-seq_regulon\n",
      "cat 12, mi = 0.00018980, BRF1_ChIP-seq_regulon\n",
      "cat 0, mi = 0.00000000, ARID3A_ChIP-seq_regulon\n",
      "cat 1, mi = 0.00000000, ATF1_ChIP-seq_regulon\n",
      "cat 2, mi = 0.00000000, ATF2_ChIP-seq_regulon\n",
      "cat 3, mi = 0.00000000, ATF3_ChIP-seq_regulon\n",
      "cat 4, mi = 0.00000000, BACH1_ChIP-seq_regulon\n",
      "cat 5, mi = 0.00000000, BATF_ChIP-seq_regulon\n",
      "cat 6, mi = 0.00000000, BCL11A_ChIP-seq_regulon\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 13, mi = 0.001, pvalue = 0.006, BRF2_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.1000): cat 131, mi = 0.000, pvalue = 0.590, SMARCA4_ChIP-seq_regulon\n",
      "Didn't pass: cat 0, mi = 0.000, pvalue = 1.000, ARID3A_ChIP-seq_regulon\n",
      "Didn't pass: cat 1, mi = 0.000, pvalue = 1.000, ATF1_ChIP-seq_regulon\n",
      "Didn't pass: cat 2, mi = 0.000, pvalue = 1.000, ATF2_ChIP-seq_regulon\n",
      "Didn't pass: cat 3, mi = 0.000, pvalue = 1.000, ATF3_ChIP-seq_regulon\n",
      "Didn't pass: cat 4, mi = 0.000, pvalue = 1.000, BACH1_ChIP-seq_regulon\n",
      "Didn't pass: cat 5, mi = 0.000, pvalue = 1.000, BATF_ChIP-seq_regulon\n",
      "Didn't pass: cat 6, mi = 0.000, pvalue = 1.000, BCL11A_ChIP-seq_regulon\n",
      "Didn't pass: cat 7, mi = 0.000, pvalue = 1.000, BCL3_ChIP-seq_regulon\n",
      "Didn't pass: cat 8, mi = 0.000, pvalue = 1.000, BCLAF1_ChIP-seq_regulon\n",
      "Didn't pass: cat 10, mi = 0.000, pvalue = 1.000, BHLHE40_ChIP-seq_regulon\n",
      "Didn't pass: cat 11, mi = 0.000, pvalue = 1.000, BRCA1_ChIP-seq_regulon\n",
      "Didn't pass: cat 14, mi = 0.000, pvalue = 1.000, CBX2_ChIP-seq_regulon\n",
      "Didn't pass: cat 15, mi = 0.000, pvalue = 1.000, CBX3_ChIP-seq_regulon\n",
      "Didn't pass: cat 16, mi = 0.000, pvalue = 1.000, CBX8_ChIP-seq_regulon\n",
      "Didn't pass: cat 17, mi = 0.000, pvalue = 1.000, CCNT2_ChIP-seq_regulon\n",
      "Didn't pass: cat 18, mi = 0.000, pvalue = 1.000, CEBPB_ChIP-seq_regulon\n",
      "Didn't pass: cat 19, mi = 0.000, pvalue = 1.000, CEBPD_ChIP-seq_regulon\n",
      "Didn't pass: cat 20, mi = 0.000, pvalue = 1.000, CEBPZ_ChIP-seq_regulon\n",
      "Didn't pass: cat 21, mi = 0.000, pvalue = 1.000, CHD1_ChIP-seq_regulon\n",
      "Total number of categories evaluated = 17.\n",
      "Number of categories that passed the tests = 1\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM2_3.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM2_3.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "79.76\t0.00\n",
      "BRF2_ChIP-seq_regulon, BRF2\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM2_3.csv_PAGE/filt_FC_LVM2_3.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM2_3.csv_PAGE/filt_FC_LVM2_3.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM2_3.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM2_3.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-19.8077983951856\t42\t0\n",
      "-19.5339769307924\t42\t0.384403209628887\n",
      "-19.402331995988\t42\t0.547642928786359\n",
      "-19.312813440321\t42\t0.647693079237713\n",
      "-19.2311935807422\t42\t0.726680040120361\n",
      "-19.1469408224674\t42\t0.810932798395186\n",
      "-19.0416248746239\t42\t0.89518555667001\n",
      "-18.8704864593781\t42\t1.02156469408225\n",
      "1.61873119358074\t42\t1.23746238716148\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM2_3.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM2_4.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_encode_tf.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_encode_tf/human_ensembl_encode_tf.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM2_4.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 11438... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 181\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 11438 out of 11438 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 11438\t181\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1271 hits go_terms_num = 181\n",
      "cat 115, mi = 0.00135894, PPARGC1A_ChIP-seq_regulon\n",
      "cat 112, mi = 0.00066645, POLR3G_ChIP-seq_regulon\n",
      "cat 12, mi = 0.00061900, BRF1_ChIP-seq_regulon\n",
      "cat 131, mi = 0.00059849, SMARCA4_ChIP-seq_regulon\n",
      "cat 114, mi = 0.00054875, POU5F1_ChIP-seq_regulon\n",
      "cat 13, mi = 0.00048745, BRF2_ChIP-seq_regulon\n",
      "cat 167, mi = 0.00041578, XRCC4_ChIP-seq_regulon\n",
      "cat 9, mi = 0.00031480, BDP1_ChIP-seq_regulon\n",
      "cat 0, mi = 0.00000000, ARID3A_ChIP-seq_regulon\n",
      "cat 1, mi = 0.00000000, ATF1_ChIP-seq_regulon\n",
      "cat 2, mi = 0.00000000, ATF2_ChIP-seq_regulon\n",
      "cat 3, mi = 0.00000000, ATF3_ChIP-seq_regulon\n",
      "cat 4, mi = 0.00000000, BACH1_ChIP-seq_regulon\n",
      "cat 5, mi = 0.00000000, BATF_ChIP-seq_regulon\n",
      "cat 6, mi = 0.00000000, BCL11A_ChIP-seq_regulon\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 115, mi = 0.001, pvalue = 0.009, PPARGC1A_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.1000): cat 112, mi = 0.001, pvalue = 0.313, POLR3G_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.1000): cat 131, mi = 0.001, pvalue = 0.355, SMARCA4_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.1000): cat 114, mi = 0.001, pvalue = 0.384, POU5F1_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.1000): cat 9, mi = 0.000, pvalue = 0.776, BDP1_ChIP-seq_regulon\n",
      "Didn't pass: cat 0, mi = 0.000, pvalue = 1.000, ARID3A_ChIP-seq_regulon\n",
      "Didn't pass: cat 1, mi = 0.000, pvalue = 1.000, ATF1_ChIP-seq_regulon\n",
      "Didn't pass: cat 2, mi = 0.000, pvalue = 1.000, ATF2_ChIP-seq_regulon\n",
      "Didn't pass: cat 3, mi = 0.000, pvalue = 1.000, ATF3_ChIP-seq_regulon\n",
      "Didn't pass: cat 4, mi = 0.000, pvalue = 1.000, BACH1_ChIP-seq_regulon\n",
      "Didn't pass: cat 5, mi = 0.000, pvalue = 1.000, BATF_ChIP-seq_regulon\n",
      "Didn't pass: cat 6, mi = 0.000, pvalue = 1.000, BCL11A_ChIP-seq_regulon\n",
      "Didn't pass: cat 7, mi = 0.000, pvalue = 1.000, BCL3_ChIP-seq_regulon\n",
      "Didn't pass: cat 8, mi = 0.000, pvalue = 1.000, BCLAF1_ChIP-seq_regulon\n",
      "Didn't pass: cat 10, mi = 0.000, pvalue = 1.000, BHLHE40_ChIP-seq_regulon\n",
      "Didn't pass: cat 11, mi = 0.000, pvalue = 1.000, BRCA1_ChIP-seq_regulon\n",
      "Didn't pass: cat 14, mi = 0.000, pvalue = 1.000, CBX2_ChIP-seq_regulon\n",
      "Didn't pass: cat 15, mi = 0.000, pvalue = 1.000, CBX3_ChIP-seq_regulon\n",
      "Didn't pass: cat 16, mi = 0.000, pvalue = 1.000, CBX8_ChIP-seq_regulon\n",
      "Didn't pass: cat 17, mi = 0.000, pvalue = 1.000, CCNT2_ChIP-seq_regulon\n",
      "Didn't pass: cat 18, mi = 0.000, pvalue = 1.000, CEBPB_ChIP-seq_regulon\n",
      "Total number of categories evaluated = 14.\n",
      "Number of categories that passed the tests = 1\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM2_4.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM2_4.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "100.30\t0.00\n",
      "PPARGC1A_ChIP-seq_regulon, PPARGC1A\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM2_4.csv_PAGE/filt_FC_LVM2_4.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM2_4.csv_PAGE/filt_FC_LVM2_4.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM2_4.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM2_4.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-19.8890329012961\t42\t0\n",
      "-19.7361914257228\t42\t0.221934197407777\n",
      "-19.6670987038883\t42\t0.30568295114656\n",
      "-19.6168494516451\t42\t0.36011964107677\n",
      "-19.5686939182453\t42\t0.406181455633101\n",
      "-19.5142572283151\t42\t0.456430707876371\n",
      "-19.4430707876371\t42\t0.515054835493519\n",
      "-19.3279162512463\t42\t0.598803589232303\n",
      "1.37268195413758\t42\t0.745363908275175\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM2_4.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM2_5.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_encode_tf.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_encode_tf/human_ensembl_encode_tf.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM2_5.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 11438... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 181\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 11438 out of 11438 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 11438\t181\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1271 hits go_terms_num = 181\n",
      "cat 9, mi = 0.00091724, BDP1_ChIP-seq_regulon\n",
      "cat 115, mi = 0.00079905, PPARGC1A_ChIP-seq_regulon\n",
      "cat 112, mi = 0.00065742, POLR3G_ChIP-seq_regulon\n",
      "cat 131, mi = 0.00053229, SMARCA4_ChIP-seq_regulon\n",
      "cat 13, mi = 0.00048725, BRF2_ChIP-seq_regulon\n",
      "cat 167, mi = 0.00041578, XRCC4_ChIP-seq_regulon\n",
      "cat 12, mi = 0.00038840, BRF1_ChIP-seq_regulon\n",
      "cat 114, mi = 0.00030981, POU5F1_ChIP-seq_regulon\n",
      "cat 0, mi = 0.00000000, ARID3A_ChIP-seq_regulon\n",
      "cat 1, mi = 0.00000000, ATF1_ChIP-seq_regulon\n",
      "cat 2, mi = 0.00000000, ATF2_ChIP-seq_regulon\n",
      "cat 3, mi = 0.00000000, ATF3_ChIP-seq_regulon\n",
      "cat 4, mi = 0.00000000, BACH1_ChIP-seq_regulon\n",
      "cat 5, mi = 0.00000000, BATF_ChIP-seq_regulon\n",
      "cat 6, mi = 0.00000000, BCL11A_ChIP-seq_regulon\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 9, mi = 0.001, pvalue = 0.083, BDP1_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.1000): cat 115, mi = 0.001, pvalue = 0.188, PPARGC1A_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.1000): cat 131, mi = 0.001, pvalue = 0.462, SMARCA4_ChIP-seq_regulon\n",
      "Didn't pass: cat 167, mi = 0.000, pvalue = 1.000, XRCC4_ChIP-seq_regulon\n",
      "Didn't pass: cat 0, mi = 0.000, pvalue = 1.000, ARID3A_ChIP-seq_regulon\n",
      "Didn't pass: cat 1, mi = 0.000, pvalue = 1.000, ATF1_ChIP-seq_regulon\n",
      "Didn't pass: cat 2, mi = 0.000, pvalue = 1.000, ATF2_ChIP-seq_regulon\n",
      "Didn't pass: cat 3, mi = 0.000, pvalue = 1.000, ATF3_ChIP-seq_regulon\n",
      "Didn't pass: cat 4, mi = 0.000, pvalue = 1.000, BACH1_ChIP-seq_regulon\n",
      "Didn't pass: cat 5, mi = 0.000, pvalue = 1.000, BATF_ChIP-seq_regulon\n",
      "Didn't pass: cat 6, mi = 0.000, pvalue = 1.000, BCL11A_ChIP-seq_regulon\n",
      "Didn't pass: cat 7, mi = 0.000, pvalue = 1.000, BCL3_ChIP-seq_regulon\n",
      "Didn't pass: cat 8, mi = 0.000, pvalue = 1.000, BCLAF1_ChIP-seq_regulon\n",
      "Didn't pass: cat 10, mi = 0.000, pvalue = 1.000, BHLHE40_ChIP-seq_regulon\n",
      "Didn't pass: cat 11, mi = 0.000, pvalue = 1.000, BRCA1_ChIP-seq_regulon\n",
      "Didn't pass: cat 14, mi = 0.000, pvalue = 1.000, CBX2_ChIP-seq_regulon\n",
      "Didn't pass: cat 15, mi = 0.000, pvalue = 1.000, CBX3_ChIP-seq_regulon\n",
      "Didn't pass: cat 16, mi = 0.000, pvalue = 1.000, CBX8_ChIP-seq_regulon\n",
      "Didn't pass: cat 17, mi = 0.000, pvalue = 1.000, CCNT2_ChIP-seq_regulon\n",
      "Didn't pass: cat 18, mi = 0.000, pvalue = 1.000, CEBPB_ChIP-seq_regulon\n",
      "Didn't pass: cat 19, mi = 0.000, pvalue = 1.000, CEBPD_ChIP-seq_regulon\n",
      "Total number of categories evaluated = 15.\n",
      "Number of categories that passed the tests = 1\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM2_5.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM2_5.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "80.82\t0.00\n",
      "BDP1_ChIP-seq_regulon, BDP1\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM2_5.csv_PAGE/filt_FC_LVM2_5.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM2_5.csv_PAGE/filt_FC_LVM2_5.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM2_5.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM2_5.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-19.8726800296956\t42\t0\n",
      "-19.6855976243504\t42\t0.25463994060876\n",
      "-19.586859688196\t42\t0.374164810690423\n",
      "-19.5141054194506\t42\t0.452115812917595\n",
      "-19.449146250928\t42\t0.519673348181143\n",
      "-19.3815887156644\t42\t0.582034149962881\n",
      "-19.3036377134373\t42\t0.654788418708241\n",
      "-19.1945063103192\t42\t0.737936154417223\n",
      "1.43652561247216\t42\t0.873051224944321\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM2_5.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_P_LVM1_2.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_encode_tf.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_encode_tf/human_ensembl_encode_tf.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_P_LVM1_2.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 11438... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 181\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 11438 out of 11438 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 11438\t181\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1271 hits go_terms_num = 181\n",
      "cat 114, mi = 0.00151278, POU5F1_ChIP-seq_regulon\n",
      "cat 9, mi = 0.00112075, BDP1_ChIP-seq_regulon\n",
      "cat 13, mi = 0.00101272, BRF2_ChIP-seq_regulon\n",
      "cat 131, mi = 0.00088626, SMARCA4_ChIP-seq_regulon\n",
      "cat 112, mi = 0.00077501, POLR3G_ChIP-seq_regulon\n",
      "cat 12, mi = 0.00071586, BRF1_ChIP-seq_regulon\n",
      "cat 115, mi = 0.00043158, PPARGC1A_ChIP-seq_regulon\n",
      "cat 167, mi = 0.00041578, XRCC4_ChIP-seq_regulon\n",
      "cat 0, mi = 0.00000000, ARID3A_ChIP-seq_regulon\n",
      "cat 1, mi = 0.00000000, ATF1_ChIP-seq_regulon\n",
      "cat 2, mi = 0.00000000, ATF2_ChIP-seq_regulon\n",
      "cat 3, mi = 0.00000000, ATF3_ChIP-seq_regulon\n",
      "cat 4, mi = 0.00000000, BACH1_ChIP-seq_regulon\n",
      "cat 5, mi = 0.00000000, BATF_ChIP-seq_regulon\n",
      "cat 6, mi = 0.00000000, BCL11A_ChIP-seq_regulon\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 114, mi = 0.002, pvalue = 0.002, POU5F1_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.1000): cat 131, mi = 0.001, pvalue = 0.124, SMARCA4_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.1000): cat 115, mi = 0.000, pvalue = 0.638, PPARGC1A_ChIP-seq_regulon\n",
      "Didn't pass: cat 0, mi = 0.000, pvalue = 1.000, ARID3A_ChIP-seq_regulon\n",
      "Didn't pass: cat 1, mi = 0.000, pvalue = 1.000, ATF1_ChIP-seq_regulon\n",
      "Didn't pass: cat 2, mi = 0.000, pvalue = 1.000, ATF2_ChIP-seq_regulon\n",
      "Didn't pass: cat 3, mi = 0.000, pvalue = 1.000, ATF3_ChIP-seq_regulon\n",
      "Didn't pass: cat 4, mi = 0.000, pvalue = 1.000, BACH1_ChIP-seq_regulon\n",
      "Didn't pass: cat 5, mi = 0.000, pvalue = 1.000, BATF_ChIP-seq_regulon\n",
      "Didn't pass: cat 6, mi = 0.000, pvalue = 1.000, BCL11A_ChIP-seq_regulon\n",
      "Didn't pass: cat 7, mi = 0.000, pvalue = 1.000, BCL3_ChIP-seq_regulon\n",
      "Didn't pass: cat 8, mi = 0.000, pvalue = 1.000, BCLAF1_ChIP-seq_regulon\n",
      "Didn't pass: cat 10, mi = 0.000, pvalue = 1.000, BHLHE40_ChIP-seq_regulon\n",
      "Didn't pass: cat 11, mi = 0.000, pvalue = 1.000, BRCA1_ChIP-seq_regulon\n",
      "Didn't pass: cat 14, mi = 0.000, pvalue = 1.000, CBX2_ChIP-seq_regulon\n",
      "Didn't pass: cat 15, mi = 0.000, pvalue = 1.000, CBX3_ChIP-seq_regulon\n",
      "Didn't pass: cat 16, mi = 0.000, pvalue = 1.000, CBX8_ChIP-seq_regulon\n",
      "Didn't pass: cat 17, mi = 0.000, pvalue = 1.000, CCNT2_ChIP-seq_regulon\n",
      "Didn't pass: cat 18, mi = 0.000, pvalue = 1.000, CEBPB_ChIP-seq_regulon\n",
      "Didn't pass: cat 19, mi = 0.000, pvalue = 1.000, CEBPD_ChIP-seq_regulon\n",
      "Didn't pass: cat 20, mi = 0.000, pvalue = 1.000, CEBPZ_ChIP-seq_regulon\n",
      "Total number of categories evaluated = 16.\n",
      "Number of categories that passed the tests = 1\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_P_LVM1_2.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_P_LVM1_2.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "5.29\t0.00\n",
      "POU5F1_ChIP-seq_regulon, POU5F1\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_P_LVM1_2.csv_PAGE/filt_FC_P_LVM1_2.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_P_LVM1_2.csv_PAGE/filt_FC_P_LVM1_2.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_P_LVM1_2.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_P_LVM1_2.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-16.8241965973535\t42\t0\n",
      "-13.1323251417769\t42\t6.35160680529301\n",
      "-12.3383742911153\t42\t7.38374291115312\n",
      "-11.8223062381853\t42\t7.93950850661626\n",
      "-11.3856332703214\t42\t8.41587901701323\n",
      "-10.9092627599244\t42\t8.81285444234405\n",
      "-10.3137996219282\t42\t9.36862003780718\n",
      "-9.44045368620038\t42\t10.0037807183365\n",
      "6.55765595463138\t42\t11.1153119092628\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_P_LVM1_2.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM2_2.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_encode_tf.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_encode_tf/human_ensembl_encode_tf.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM2_2.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 11438... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 181\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 11438 out of 11438 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 11438\t181\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1271 hits go_terms_num = 181\n",
      "cat 115, mi = 0.00121912, PPARGC1A_ChIP-seq_regulon\n",
      "cat 114, mi = 0.00076561, POU5F1_ChIP-seq_regulon\n",
      "cat 9, mi = 0.00073510, BDP1_ChIP-seq_regulon\n",
      "cat 112, mi = 0.00059112, POLR3G_ChIP-seq_regulon\n",
      "cat 167, mi = 0.00059073, XRCC4_ChIP-seq_regulon\n",
      "cat 13, mi = 0.00055345, BRF2_ChIP-seq_regulon\n",
      "cat 131, mi = 0.00046620, SMARCA4_ChIP-seq_regulon\n",
      "cat 12, mi = 0.00046312, BRF1_ChIP-seq_regulon\n",
      "cat 0, mi = 0.00000000, ARID3A_ChIP-seq_regulon\n",
      "cat 1, mi = 0.00000000, ATF1_ChIP-seq_regulon\n",
      "cat 2, mi = 0.00000000, ATF2_ChIP-seq_regulon\n",
      "cat 3, mi = 0.00000000, ATF3_ChIP-seq_regulon\n",
      "cat 4, mi = 0.00000000, BACH1_ChIP-seq_regulon\n",
      "cat 5, mi = 0.00000000, BATF_ChIP-seq_regulon\n",
      "cat 6, mi = 0.00000000, BCL11A_ChIP-seq_regulon\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 115, mi = 0.001, pvalue = 0.024, PPARGC1A_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.1000): cat 114, mi = 0.001, pvalue = 0.155, POU5F1_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.1000): cat 9, mi = 0.001, pvalue = 0.186, BDP1_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.1000): cat 112, mi = 0.001, pvalue = 0.415, POLR3G_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.1000): cat 131, mi = 0.000, pvalue = 0.549, SMARCA4_ChIP-seq_regulon\n",
      "Didn't pass: cat 0, mi = 0.000, pvalue = 1.000, ARID3A_ChIP-seq_regulon\n",
      "Didn't pass: cat 1, mi = 0.000, pvalue = 1.000, ATF1_ChIP-seq_regulon\n",
      "Didn't pass: cat 2, mi = 0.000, pvalue = 1.000, ATF2_ChIP-seq_regulon\n",
      "Didn't pass: cat 3, mi = 0.000, pvalue = 1.000, ATF3_ChIP-seq_regulon\n",
      "Didn't pass: cat 4, mi = 0.000, pvalue = 1.000, BACH1_ChIP-seq_regulon\n",
      "Didn't pass: cat 5, mi = 0.000, pvalue = 1.000, BATF_ChIP-seq_regulon\n",
      "Didn't pass: cat 6, mi = 0.000, pvalue = 1.000, BCL11A_ChIP-seq_regulon\n",
      "Didn't pass: cat 7, mi = 0.000, pvalue = 1.000, BCL3_ChIP-seq_regulon\n",
      "Didn't pass: cat 8, mi = 0.000, pvalue = 1.000, BCLAF1_ChIP-seq_regulon\n",
      "Didn't pass: cat 10, mi = 0.000, pvalue = 1.000, BHLHE40_ChIP-seq_regulon\n",
      "Didn't pass: cat 11, mi = 0.000, pvalue = 1.000, BRCA1_ChIP-seq_regulon\n",
      "Didn't pass: cat 14, mi = 0.000, pvalue = 1.000, CBX2_ChIP-seq_regulon\n",
      "Didn't pass: cat 15, mi = 0.000, pvalue = 1.000, CBX3_ChIP-seq_regulon\n",
      "Didn't pass: cat 16, mi = 0.000, pvalue = 1.000, CBX8_ChIP-seq_regulon\n",
      "Didn't pass: cat 17, mi = 0.000, pvalue = 1.000, CCNT2_ChIP-seq_regulon\n",
      "Didn't pass: cat 18, mi = 0.000, pvalue = 1.000, CEBPB_ChIP-seq_regulon\n",
      "Total number of categories evaluated = 14.\n",
      "Number of categories that passed the tests = 1\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM2_2.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM2_2.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "101.03\t0.00\n",
      "PPARGC1A_ChIP-seq_regulon, PPARGC1A\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM2_2.csv_PAGE/filt_FC_LVM2_2.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM2_2.csv_PAGE/filt_FC_LVM2_2.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM2_2.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM2_2.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-19.8503414827279\t42\t0\n",
      "-19.6507967930318\t42\t0.299317034544195\n",
      "-19.5718103533604\t42\t0.39908937939226\n",
      "-19.517766999901\t42\t0.457289913886964\n",
      "-19.467880827477\t42\t0.507176086310997\n",
      "-19.4138374740176\t42\t0.557062258735029\n",
      "-19.3473225774522\t42\t0.615262793229734\n",
      "-19.2433930515688\t42\t0.690092051865782\n",
      "1.41156092249827\t42\t0.823121844996536\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_LVM2_2.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_P_LVM1_4.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_encode_tf.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_encode_tf/human_ensembl_encode_tf.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_P_LVM1_4.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 11438... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 181\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 11438 out of 11438 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 11438\t181\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1271 hits go_terms_num = 181\n",
      "cat 115, mi = 0.00134012, PPARGC1A_ChIP-seq_regulon\n",
      "cat 114, mi = 0.00071279, POU5F1_ChIP-seq_regulon\n",
      "cat 131, mi = 0.00068757, SMARCA4_ChIP-seq_regulon\n",
      "cat 13, mi = 0.00048735, BRF2_ChIP-seq_regulon\n",
      "cat 112, mi = 0.00041597, POLR3G_ChIP-seq_regulon\n",
      "cat 167, mi = 0.00041588, XRCC4_ChIP-seq_regulon\n",
      "cat 9, mi = 0.00031470, BDP1_ChIP-seq_regulon\n",
      "cat 12, mi = 0.00018970, BRF1_ChIP-seq_regulon\n",
      "cat 0, mi = 0.00000000, ARID3A_ChIP-seq_regulon\n",
      "cat 1, mi = 0.00000000, ATF1_ChIP-seq_regulon\n",
      "cat 2, mi = 0.00000000, ATF2_ChIP-seq_regulon\n",
      "cat 3, mi = 0.00000000, ATF3_ChIP-seq_regulon\n",
      "cat 4, mi = 0.00000000, BACH1_ChIP-seq_regulon\n",
      "cat 5, mi = 0.00000000, BATF_ChIP-seq_regulon\n",
      "cat 6, mi = 0.00000000, BCL11A_ChIP-seq_regulon\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 115, mi = 0.001, pvalue = 0.010, PPARGC1A_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.1000): cat 114, mi = 0.001, pvalue = 0.189, POU5F1_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.1000): cat 131, mi = 0.001, pvalue = 0.277, SMARCA4_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.1000): cat 112, mi = 0.000, pvalue = 0.681, POLR3G_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.1000): cat 9, mi = 0.000, pvalue = 0.768, BDP1_ChIP-seq_regulon\n",
      "Didn't pass: cat 0, mi = 0.000, pvalue = 1.000, ARID3A_ChIP-seq_regulon\n",
      "Didn't pass: cat 1, mi = 0.000, pvalue = 1.000, ATF1_ChIP-seq_regulon\n",
      "Didn't pass: cat 2, mi = 0.000, pvalue = 1.000, ATF2_ChIP-seq_regulon\n",
      "Didn't pass: cat 3, mi = 0.000, pvalue = 1.000, ATF3_ChIP-seq_regulon\n",
      "Didn't pass: cat 4, mi = 0.000, pvalue = 1.000, BACH1_ChIP-seq_regulon\n",
      "Didn't pass: cat 5, mi = 0.000, pvalue = 1.000, BATF_ChIP-seq_regulon\n",
      "Didn't pass: cat 6, mi = 0.000, pvalue = 1.000, BCL11A_ChIP-seq_regulon\n",
      "Didn't pass: cat 7, mi = 0.000, pvalue = 1.000, BCL3_ChIP-seq_regulon\n",
      "Didn't pass: cat 8, mi = 0.000, pvalue = 1.000, BCLAF1_ChIP-seq_regulon\n",
      "Didn't pass: cat 10, mi = 0.000, pvalue = 1.000, BHLHE40_ChIP-seq_regulon\n",
      "Didn't pass: cat 11, mi = 0.000, pvalue = 1.000, BRCA1_ChIP-seq_regulon\n",
      "Didn't pass: cat 14, mi = 0.000, pvalue = 1.000, CBX2_ChIP-seq_regulon\n",
      "Didn't pass: cat 15, mi = 0.000, pvalue = 1.000, CBX3_ChIP-seq_regulon\n",
      "Didn't pass: cat 16, mi = 0.000, pvalue = 1.000, CBX8_ChIP-seq_regulon\n",
      "Didn't pass: cat 17, mi = 0.000, pvalue = 1.000, CCNT2_ChIP-seq_regulon\n",
      "Didn't pass: cat 18, mi = 0.000, pvalue = 1.000, CEBPB_ChIP-seq_regulon\n",
      "Total number of categories evaluated = 14.\n",
      "Number of categories that passed the tests = 1\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_P_LVM1_4.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_P_LVM1_4.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "110.21\t0.00\n",
      "PPARGC1A_ChIP-seq_regulon, PPARGC1A\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_P_LVM1_4.csv_PAGE/filt_FC_P_LVM1_4.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_P_LVM1_4.csv_PAGE/filt_FC_P_LVM1_4.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_P_LVM1_4.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_P_LVM1_4.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-19.9980945467743\t42\t0\n",
      "-19.9923781870974\t42\t0.00381090645132021\n",
      "-19.9847563741947\t42\t0.0114327193539606\n",
      "-19.9790400145177\t42\t0.019054532256601\n",
      "-19.9733236548408\t42\t0.0228654387079212\n",
      "-19.9676072951638\t42\t0.0304872516105617\n",
      "-19.8285092096906\t42\t0.0342981580618819\n",
      "-19.4779058161691\t42\t0.308683422556937\n",
      "1.3677524725524\t42\t0.7355049451048\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_P_LVM1_4.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_P_LVM1_1.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_encode_tf.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_encode_tf/human_ensembl_encode_tf.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_P_LVM1_1.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 11438... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 181\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 11438 out of 11438 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 11438\t181\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1271 hits go_terms_num = 181\n",
      "cat 115, mi = 0.00208459, PPARGC1A_ChIP-seq_regulon\n",
      "cat 131, mi = 0.00106709, SMARCA4_ChIP-seq_regulon\n",
      "cat 13, mi = 0.00066241, BRF2_ChIP-seq_regulon\n",
      "cat 167, mi = 0.00059093, XRCC4_ChIP-seq_regulon\n",
      "cat 9, mi = 0.00056257, BDP1_ChIP-seq_regulon\n",
      "cat 112, mi = 0.00055699, POLR3G_ChIP-seq_regulon\n",
      "cat 12, mi = 0.00026679, BRF1_ChIP-seq_regulon\n",
      "cat 114, mi = 0.00019484, POU5F1_ChIP-seq_regulon\n",
      "cat 0, mi = 0.00000000, ARID3A_ChIP-seq_regulon\n",
      "cat 1, mi = 0.00000000, ATF1_ChIP-seq_regulon\n",
      "cat 2, mi = 0.00000000, ATF2_ChIP-seq_regulon\n",
      "cat 3, mi = 0.00000000, ATF3_ChIP-seq_regulon\n",
      "cat 4, mi = 0.00000000, BACH1_ChIP-seq_regulon\n",
      "cat 5, mi = 0.00000000, BATF_ChIP-seq_regulon\n",
      "cat 6, mi = 0.00000000, BCL11A_ChIP-seq_regulon\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 115, mi = 0.002, pvalue = 0.000, PPARGC1A_ChIP-seq_regulon\n",
      "Passed: cat 131, mi = 0.001, pvalue = 0.051, SMARCA4_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.1000): cat 9, mi = 0.001, pvalue = 0.369, BDP1_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.1000): cat 112, mi = 0.001, pvalue = 0.473, POLR3G_ChIP-seq_regulon\n",
      "Didn't pass: cat 0, mi = 0.000, pvalue = 1.000, ARID3A_ChIP-seq_regulon\n",
      "Didn't pass: cat 1, mi = 0.000, pvalue = 1.000, ATF1_ChIP-seq_regulon\n",
      "Didn't pass: cat 2, mi = 0.000, pvalue = 1.000, ATF2_ChIP-seq_regulon\n",
      "Didn't pass: cat 3, mi = 0.000, pvalue = 1.000, ATF3_ChIP-seq_regulon\n",
      "Didn't pass: cat 4, mi = 0.000, pvalue = 1.000, BACH1_ChIP-seq_regulon\n",
      "Didn't pass: cat 5, mi = 0.000, pvalue = 1.000, BATF_ChIP-seq_regulon\n",
      "Didn't pass: cat 6, mi = 0.000, pvalue = 1.000, BCL11A_ChIP-seq_regulon\n",
      "Didn't pass: cat 7, mi = 0.000, pvalue = 1.000, BCL3_ChIP-seq_regulon\n",
      "Didn't pass: cat 8, mi = 0.000, pvalue = 1.000, BCLAF1_ChIP-seq_regulon\n",
      "Didn't pass: cat 10, mi = 0.000, pvalue = 1.000, BHLHE40_ChIP-seq_regulon\n",
      "Didn't pass: cat 11, mi = 0.000, pvalue = 1.000, BRCA1_ChIP-seq_regulon\n",
      "Didn't pass: cat 14, mi = 0.000, pvalue = 1.000, CBX2_ChIP-seq_regulon\n",
      "Didn't pass: cat 15, mi = 0.000, pvalue = 1.000, CBX3_ChIP-seq_regulon\n",
      "Didn't pass: cat 16, mi = 0.000, pvalue = 1.000, CBX8_ChIP-seq_regulon\n",
      "Didn't pass: cat 17, mi = 0.000, pvalue = 1.000, CCNT2_ChIP-seq_regulon\n",
      "Didn't pass: cat 18, mi = 0.000, pvalue = 1.000, CEBPB_ChIP-seq_regulon\n",
      "Didn't pass: cat 19, mi = 0.000, pvalue = 1.000, CEBPD_ChIP-seq_regulon\n",
      "Didn't pass: cat 20, mi = 0.000, pvalue = 1.000, CEBPZ_ChIP-seq_regulon\n",
      "Total number of categories evaluated = 16.\n",
      "Number of categories that passed the tests = 2\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_P_LVM1_1.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_P_LVM1_1.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "280.55\t0.00\n",
      "PPARGC1A_ChIP-seq_regulon, PPARGC1A\n",
      "SMARCA4_ChIP-seq_regulon, SMARCA4\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_P_LVM1_1.csv_PAGE/filt_FC_P_LVM1_1.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_P_LVM1_1.csv_PAGE/filt_FC_P_LVM1_1.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_P_LVM1_1.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_P_LVM1_1.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-19.9985029406523\t42\t0\n",
      "-19.9962573516307\t42\t0.00299411869541971\n",
      "-19.9940117626092\t42\t0.00449117804312957\n",
      "-19.9917661735876\t42\t0.00748529673854928\n",
      "-19.989520584566\t42\t0.00898235608625913\n",
      "-19.9865264658706\t42\t0.0119764747816788\n",
      "-19.9146676171805\t42\t0.0149705934770986\n",
      "-19.7537337373017\t42\t0.155694172161825\n",
      "1.16841917661736\t42\t0.336838353234718\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_P_LVM1_1.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "homologiesfile: \n",
      "genelist: \n",
      "REMOVING DUPLICATES\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/remove_homologous_sequences_withseed.pl -expfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_P_LVM1_3.csv -quantized 0 -genelist /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/seqnames/human_ensembl_encode_tf.txt -dupfile /nvme/bins/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_encode_tf/human_ensembl_encode_tf.homologies -outfile /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_P_LVM1_3.csv.nodups -ebins 9 \n",
      "\n",
      "DUPLICATION REMOVAL CANCELLED -- homologiesfile does not exist\n",
      "\n",
      "Reading expression data ... 1 by 11438... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 181\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 11438 out of 11438 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 11438\t181\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "1271 hits go_terms_num = 181\n",
      "cat 115, mi = 0.00095684, PPARGC1A_ChIP-seq_regulon\n",
      "cat 131, mi = 0.00081996, SMARCA4_ChIP-seq_regulon\n",
      "cat 167, mi = 0.00059073, XRCC4_ChIP-seq_regulon\n",
      "cat 13, mi = 0.00048755, BRF2_ChIP-seq_regulon\n",
      "cat 112, mi = 0.00040724, POLR3G_ChIP-seq_regulon\n",
      "cat 114, mi = 0.00039230, POU5F1_ChIP-seq_regulon\n",
      "cat 9, mi = 0.00037748, BDP1_ChIP-seq_regulon\n",
      "cat 12, mi = 0.00027914, BRF1_ChIP-seq_regulon\n",
      "cat 0, mi = 0.00000000, ARID3A_ChIP-seq_regulon\n",
      "cat 1, mi = 0.00000000, ATF1_ChIP-seq_regulon\n",
      "cat 2, mi = 0.00000000, ATF2_ChIP-seq_regulon\n",
      "cat 3, mi = 0.00000000, ATF3_ChIP-seq_regulon\n",
      "cat 4, mi = 0.00000000, BACH1_ChIP-seq_regulon\n",
      "cat 5, mi = 0.00000000, BATF_ChIP-seq_regulon\n",
      "cat 6, mi = 0.00000000, BCL11A_ChIP-seq_regulon\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 115, mi = 0.001, pvalue = 0.093, PPARGC1A_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.1000): cat 131, mi = 0.001, pvalue = 0.167, SMARCA4_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.1000): cat 112, mi = 0.000, pvalue = 0.698, POLR3G_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.1000): cat 114, mi = 0.000, pvalue = 0.639, POU5F1_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.1000): cat 9, mi = 0.000, pvalue = 0.660, BDP1_ChIP-seq_regulon\n",
      "Didn't pass: cat 0, mi = 0.000, pvalue = 1.000, ARID3A_ChIP-seq_regulon\n",
      "Didn't pass: cat 1, mi = 0.000, pvalue = 1.000, ATF1_ChIP-seq_regulon\n",
      "Didn't pass: cat 2, mi = 0.000, pvalue = 1.000, ATF2_ChIP-seq_regulon\n",
      "Didn't pass: cat 3, mi = 0.000, pvalue = 1.000, ATF3_ChIP-seq_regulon\n",
      "Didn't pass: cat 4, mi = 0.000, pvalue = 1.000, BACH1_ChIP-seq_regulon\n",
      "Didn't pass: cat 5, mi = 0.000, pvalue = 1.000, BATF_ChIP-seq_regulon\n",
      "Didn't pass: cat 6, mi = 0.000, pvalue = 1.000, BCL11A_ChIP-seq_regulon\n",
      "Didn't pass: cat 7, mi = 0.000, pvalue = 1.000, BCL3_ChIP-seq_regulon\n",
      "Didn't pass: cat 8, mi = 0.000, pvalue = 1.000, BCLAF1_ChIP-seq_regulon\n",
      "Didn't pass: cat 10, mi = 0.000, pvalue = 1.000, BHLHE40_ChIP-seq_regulon\n",
      "Didn't pass: cat 11, mi = 0.000, pvalue = 1.000, BRCA1_ChIP-seq_regulon\n",
      "Didn't pass: cat 14, mi = 0.000, pvalue = 1.000, CBX2_ChIP-seq_regulon\n",
      "Didn't pass: cat 15, mi = 0.000, pvalue = 1.000, CBX3_ChIP-seq_regulon\n",
      "Didn't pass: cat 16, mi = 0.000, pvalue = 1.000, CBX8_ChIP-seq_regulon\n",
      "Didn't pass: cat 17, mi = 0.000, pvalue = 1.000, CCNT2_ChIP-seq_regulon\n",
      "Didn't pass: cat 18, mi = 0.000, pvalue = 1.000, CEBPB_ChIP-seq_regulon\n",
      "Total number of categories evaluated = 14.\n",
      "Number of categories that passed the tests = 1\n",
      "Number of clusters: 9\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_P_LVM1_3.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_P_LVM1_3.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "15.99\t0.00\n",
      "PPARGC1A_ChIP-seq_regulon, PPARGC1A\n",
      "Outputing EPS file /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_P_LVM1_3.csv_PAGE/filt_FC_P_LVM1_3.csv.summary.eps\n",
      "Convert to PDF /avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_P_LVM1_3.csv_PAGE/filt_FC_P_LVM1_3.csv.summary.pdf\n",
      "Finished.\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_P_LVM1_3.csv_PAGE/pvmatrix.txt --expfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_P_LVM1_3.csv --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-19.3958724202627\t42\t0\n",
      "-18.6341463414634\t42\t1.20825515947467\n",
      "-18.3846153846154\t42\t1.5234521575985\n",
      "-18.2007504690432\t42\t1.70731707317073\n",
      "-18.0168855534709\t42\t1.89118198874296\n",
      "-17.8330206378987\t42\t2.0750469043152\n",
      "-17.5966228893058\t42\t2.25891181988743\n",
      "-17.2157598499062\t42\t2.54784240150094\n",
      "2.51031894934334\t42\t3.02063789868668\n",
      "perl /nvme/bins/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=/avicenna/Johnny/scRNAseq/metastatic_evolution/ipage_human_ensembl_encode_tf/fc_values/filt_FC_P_LVM1_3.csv_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "sem --wait done\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "nohup sh run.sh iPAGE_JY.sh &"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
